{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f27d4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    out_Seq = ''\n",
    "    \n",
    "    \n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = SeqIO.parse((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'fasta')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in fasta_file:\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = out_Seq.reverse_complement()\n",
    "            break\n",
    "                \n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    \n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4900aae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "85ed0aea",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sequence(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = SeqIO.parse((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'fasta')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in fasta_file:\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "#             out_Seq = Seq(sequence[start-1:end])\n",
    "#             if reverse_c == 1:\n",
    "#                 sequence = out_Seq.reverse_complement()\n",
    "            return(sequence)\n",
    "    \n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d454c091",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,305124,305209,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,320919,321101,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,50,53\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,321823,321936,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,36,37\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,323924,324004,0,Y,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,9,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,324401,324573,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,324709,324872,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,325346,325463,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,325705,325884,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,326150,326408,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,326744,326947,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,327161,327300,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,45,46\n",
      "\n",
      "Pieris_napi\n",
      "Aporia_crataegi,OU538732.1,328018,328181,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n",
      "Pieris_napi\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSG\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGL\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGLP\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGLPW\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGLPWP\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGLPWPL\n",
      "SLVDNIVCGDIGDVSETECAAVRNNVPLSAAAARDSPPVDFCVTPPPQSPQPNDGNGEAAMSVIMSLLEADAGLGGPVNFSGLPWPLP\n",
      "Aporia_crataegi,OU538732.1,330215,330484,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,83,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Aporia_crataegi,OU538732.1,305124,305209,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Aporia_crataegi,OU538732.1,320919,321101,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,50,53\n",
      "Aporia_crataegi,OU538732.1,321823,321936,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,36,37\n",
      "Aporia_crataegi,OU538732.1,323924,324004,0,Y,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,9,26,26\n",
      "Aporia_crataegi,OU538732.1,324401,324573,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Aporia_crataegi,OU538732.1,324709,324872,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Aporia_crataegi,OU538732.1,325346,325463,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Aporia_crataegi,OU538732.1,325705,325884,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Aporia_crataegi,OU538732.1,326150,326408,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Aporia_crataegi,OU538732.1,326744,326947,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Aporia_crataegi,OU538732.1,327161,327300,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,45,46\n",
      "Aporia_crataegi,OU538732.1,328018,328181,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Aporia_crataegi,OU538732.1,330215,330484,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,83,90\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5912030,5912115,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5901996,5902169,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,7,52,53\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5901264,5901377,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5900724,5900804,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,3,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5900067,5900239,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5899820,5899983,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5899629,5899746,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5899377,5899556,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5899055,5899307,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5898775,5898978,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5898514,5898653,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,43,46\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5897871,5898034,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n",
      "Pieris_napi\n",
      "Leptophobia_aripa,OX637275.1,5896486,5896752,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Leptophobia_aripa,OX637275.1,5912030,5912115,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Leptophobia_aripa,OX637275.1,5901996,5902169,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,7,52,53\n",
      "Leptophobia_aripa,OX637275.1,5901264,5901377,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "Leptophobia_aripa,OX637275.1,5900724,5900804,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,3,26,26\n",
      "Leptophobia_aripa,OX637275.1,5900067,5900239,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Leptophobia_aripa,OX637275.1,5899820,5899983,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Leptophobia_aripa,OX637275.1,5899629,5899746,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Leptophobia_aripa,OX637275.1,5899377,5899556,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Leptophobia_aripa,OX637275.1,5899055,5899307,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Leptophobia_aripa,OX637275.1,5898775,5898978,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Leptophobia_aripa,OX637275.1,5898514,5898653,1,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,43,46\n",
      "Leptophobia_aripa,OX637275.1,5897871,5898034,1,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Leptophobia_aripa,OX637275.1,5896486,5896752,1,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,90,90\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2795729,2795814,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2804782,2804943,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,52,53\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2805840,2805950,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,2,37,37\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2806358,2806438,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2806764,2806936,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2807009,2807172,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2807264,2807381,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2808363,2808542,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2808621,2808873,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2809064,2809267,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2809370,2809509,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2810572,2810735,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pieris_napi\n",
      "Pieris_brassicae,NC_059680.1,2811653,2811922,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_brassicae,NC_059680.1,2795729,2795814,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Pieris_brassicae,NC_059680.1,2804782,2804943,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,52,53\n",
      "Pieris_brassicae,NC_059680.1,2805840,2805950,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,2,37,37\n",
      "Pieris_brassicae,NC_059680.1,2806358,2806438,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "Pieris_brassicae,NC_059680.1,2806764,2806936,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Pieris_brassicae,NC_059680.1,2807009,2807172,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Pieris_brassicae,NC_059680.1,2807264,2807381,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Pieris_brassicae,NC_059680.1,2808363,2808542,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Pieris_brassicae,NC_059680.1,2808621,2808873,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Pieris_brassicae,NC_059680.1,2809064,2809267,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Pieris_brassicae,NC_059680.1,2809370,2809509,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "Pieris_brassicae,NC_059680.1,2810572,2810735,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Pieris_brassicae,NC_059680.1,2811653,2811922,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,2,90,90\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2501166,2501251,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2511103,2511264,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,52,53\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2512320,2512430,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,2,37,37\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2513481,2513561,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2513968,2514140,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2514951,2515114,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2515194,2515311,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2516034,2516213,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2516298,2516550,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2517146,2517349,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2517426,2517565,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2519179,2519342,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_mannii,CM054800.1,2520503,2520772,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_mannii,CM054800.1,2501166,2501251,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Pieris_mannii,CM054800.1,2511103,2511264,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,52,53\n",
      "Pieris_mannii,CM054800.1,2512320,2512430,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,2,37,37\n",
      "Pieris_mannii,CM054800.1,2513481,2513561,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "Pieris_mannii,CM054800.1,2513968,2514140,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Pieris_mannii,CM054800.1,2514951,2515114,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Pieris_mannii,CM054800.1,2515194,2515311,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Pieris_mannii,CM054800.1,2516034,2516213,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Pieris_mannii,CM054800.1,2516298,2516550,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Pieris_mannii,CM054800.1,2517146,2517349,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Pieris_mannii,CM054800.1,2517426,2517565,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "Pieris_mannii,CM054800.1,2519179,2519342,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Pieris_mannii,CM054800.1,2520503,2520772,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,482378,482463,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,495729,495890,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,53,53\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,496743,496856,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,497787,497867,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,498645,498817,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,499197,499360,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,499710,499827,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,500487,500666,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,501158,501410,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,502019,502222,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,502912,503051,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,504087,504250,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_melete,CAVNZK010000127.1,505073,505342,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_melete,CAVNZK010000127.1,482378,482463,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Pieris_melete,CAVNZK010000127.1,495729,495890,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,53,53\n",
      "Pieris_melete,CAVNZK010000127.1,496743,496856,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "Pieris_melete,CAVNZK010000127.1,497787,497867,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "Pieris_melete,CAVNZK010000127.1,498645,498817,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Pieris_melete,CAVNZK010000127.1,499197,499360,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Pieris_melete,CAVNZK010000127.1,499710,499827,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Pieris_melete,CAVNZK010000127.1,500487,500666,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Pieris_melete,CAVNZK010000127.1,501158,501410,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Pieris_melete,CAVNZK010000127.1,502019,502222,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Pieris_melete,CAVNZK010000127.1,502912,503051,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "Pieris_melete,CAVNZK010000127.1,504087,504250,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Pieris_melete,CAVNZK010000127.1,505073,505342,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2828816,2828901,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2841007,2841168,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,53,53\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2842963,2843076,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2843974,2844054,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2844824,2844996,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2845341,2845504,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2845589,2845706,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2847030,2847209,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2847627,2847879,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2848933,2849136,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2849564,2849703,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2851587,2851750,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_napi,NC_062259.1,2852756,2853025,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_napi,NC_062259.1,2828816,2828901,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Pieris_napi,NC_062259.1,2841007,2841168,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,53,53\n",
      "Pieris_napi,NC_062259.1,2842963,2843076,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "Pieris_napi,NC_062259.1,2843974,2844054,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "Pieris_napi,NC_062259.1,2844824,2844996,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Pieris_napi,NC_062259.1,2845341,2845504,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Pieris_napi,NC_062259.1,2845589,2845706,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Pieris_napi,NC_062259.1,2847030,2847209,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Pieris_napi,NC_062259.1,2847627,2847879,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Pieris_napi,NC_062259.1,2848933,2849136,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Pieris_napi,NC_062259.1,2849564,2849703,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "Pieris_napi,NC_062259.1,2851587,2851750,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,54,54\n",
      "Pieris_napi,NC_062259.1,2852756,2853025,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2063858,2063943,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2072484,2072645,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,50,53\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2073374,2073487,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2074516,2074596,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2074978,2075150,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2075622,2075785,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2075860,2075977,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2076760,2076939,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2077022,2077274,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2078017,2078220,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2078294,2078433,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2079228,2079391,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "\n",
      "Pieris_napi\n",
      "Pieris_rapae,NC_059534.1,2080540,2080809,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_rapae,NC_059534.1,2063858,2063943,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "Pieris_rapae,NC_059534.1,2072484,2072645,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_2,1,50,53\n",
      "Pieris_rapae,NC_059534.1,2073374,2073487,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_3,1,37,37\n",
      "Pieris_rapae,NC_059534.1,2074516,2074596,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_4,1,26,26\n",
      "Pieris_rapae,NC_059534.1,2074978,2075150,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_5,1,57,57\n",
      "Pieris_rapae,NC_059534.1,2075622,2075785,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_6,1,54,54\n",
      "Pieris_rapae,NC_059534.1,2075860,2075977,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_1_query_Exon_7,1,39,39\n",
      "Pieris_rapae,NC_059534.1,2076760,2076939,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_1_query_Exon_8,1,59,59\n",
      "Pieris_rapae,NC_059534.1,2077022,2077274,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_2_query_Exon_9,1,83,83\n",
      "Pieris_rapae,NC_059534.1,2078017,2078220,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_2_query_Exon_10,1,67,67\n",
      "Pieris_rapae,NC_059534.1,2078294,2078433,0,N,Pieris_napi_XM_047668504.1_Frame_1_rightoh_1_query_Exon_11,1,46,46\n",
      "Pieris_rapae,NC_059534.1,2079228,2079391,0,N,Pieris_napi_XM_047668504.1_Frame_2_rightoh_0_query_Exon_12,1,53,54\n",
      "Pieris_rapae,NC_059534.1,2080540,2080809,0,N,Pieris_napi_XM_047668504.1_Frame_0_rightoh_0_query_Exon_13,1,90,90\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#%reset -f\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "\n",
    "family_group = \"4.Pierinae\"\n",
    "blast_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/10.Cycle Exon Analysis/{family_group}\"\n",
    "query_location = f\"{blast_location}/1.Query\"\n",
    "query_species = \"Pieris_napi\"\n",
    "query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "if \"desktop.ini\" in query_transcript_list:\n",
    "    query_transcript_list.remove(\"desktop.ini\")\n",
    "\n",
    "query_transcript = query_transcript_list[0]\n",
    "#Get Species list, the list may be a list of files or a single species\n",
    "\n",
    "species_list = os.listdir(f\"{blast_location}/1.Blast_result\")\n",
    "# species_list = [\"Pararge_aegeria\"]\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove('desktop.ini')\n",
    "\n",
    "#get list of query names\n",
    "\n",
    "query_file = SeqIO.parse(f\"{blast_location}/1.Query/{query_species}/{query_transcript}/query_protein.fa\", 'fasta')\n",
    "query_name_list = []\n",
    "\n",
    "for records in query_file:\n",
    "    query_name_list.append(records.id)\n",
    "\n",
    "\n",
    "\n",
    "for species in species_list:\n",
    "    # print(species_name)\n",
    "    # species = species_name.split(\".\")[1]\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    \n",
    "#Run for each query in the query list\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "#         print(f\"Query name = {query_name}\")\n",
    "        Length_switch = \"0\"\n",
    "        with open(f\"{blast_location}/1.Blast_result/{species}/{species}_blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "#                 print(lines)\n",
    "#                 assert False\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "#                 query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "#                 query_species = str(query_species_split[0].split(\".\")[1]+\"_\"+query_species_split[1].rstrip())\n",
    "                print(query_species)\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            length = 0\n",
    "            start = 000\n",
    "            stop = 000\n",
    "            query_start_coor = 000\n",
    "            query_stop_coor = 000\n",
    "            error = \"Y\"\n",
    "            complement  = 0\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "        \n",
    "    \n",
    "    \n",
    "        \n",
    "\n",
    "#Check for Met or STOP at the start and end\n",
    "        sequence = ''\n",
    "        while True and (start != 0 or stop != 0):\n",
    "            if (query_name == query_name_list[0] or query_name == query_name_list[-1]):\n",
    "                frame = 1\n",
    "#                 translated_sequence = (sequence_extractor(species, scaff, int(complement), start, stop,frame ))\n",
    "                if sequence == '':\n",
    "                    sequence = get_sequence(species, scaff, int(complement), start, stop,frame)\n",
    "                if int(complement) == 1:\n",
    "                    translated_sequence = Seq(sequence[start-1:stop]).reverse_complement().translate()\n",
    "                else:\n",
    "                    translated_sequence = Seq(sequence[start-1:stop]).translate()\n",
    "                if translated_sequence == old_trans:\n",
    "                    break\n",
    "                    \n",
    "#Check for Met at the beginning of first exon\n",
    "\n",
    "                if query_name == query_name_list[0] :\n",
    "                    if translated_sequence[0] != \"M\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            start = int(start) - 3\n",
    "                        if complement == \"1\":\n",
    "                            stop = int(stop) + 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if \"*\" in translated_sequence:\n",
    "                        error = \"Y\"\n",
    "                        break\n",
    "                    if translated_sequence[0] == \"M\":\n",
    "                        break\n",
    "\n",
    "#Check for stop at the end of last exon\n",
    "\n",
    "                if query_name == query_name_list[-1]:\n",
    "                    if translated_sequence[-1] != \"*\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            stop = int(stop) + 3\n",
    "                        if complement == \"1\":\n",
    "                            start = int(start) - 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if translated_sequence[-1] == \"*\":\n",
    "\n",
    "                        break\n",
    "            else:\n",
    "                break\n",
    "        \n",
    "        \n",
    "#Add offset at the beginning and end of each exon\n",
    "        if (start != 0 or stop != 0):\n",
    "            \n",
    "            start_modifier = int(query_name.split(\"Frame\")[1][1])\n",
    "            stop_modifier = int(query_name.split(\"rightoh\")[1][1])\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0            \n",
    "\n",
    "        seq_length = query_length\n",
    "        \n",
    "        \n",
    "#Get exon start and end by adding or removing codons\n",
    "        \n",
    "    #if not the beginning\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "            \n",
    "           \n",
    "        output_format = str(species.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        print(output_format)\n",
    "#         assert False\n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "   \n",
    "    output_file = open(f\"{blast_location}/1.Blast_result/{species}/{species}_coordinates_old.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fa07f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465fa823",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
