{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "bac35abe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"8.Lycaenidae\"\n",
    "number_of_exons = 13\n",
    "\n",
    "location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "14fde9ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ MDDDGDDK---------DDTK----\n",
      "Aricia_artaxerx MDDDGDDK---------DDTK----\n",
      "Lycaena_phlaeas MDDDGDDK---------DDTK----\n",
      "Celastrina_argi MEDDGDDK---------DDTK----\n",
      "Cyaniris_semiar MEDDGDDK---------DDTK----\n",
      "Eumaeus_atala_E MEDDGDDK---------DDTK----\n",
      "Glaucopsyche_al MEDDGDDK---------DDTK----\n",
      "Lysandra_corido MEDDGDDK---------DDTK----\n",
      "Phengaris_arion MEDDGDDK---------DDTK----\n",
      "Plebejus_argus_ MEDDGDDK---------DDTK----\n",
      "Polyommatus_ica MEDDGDDK---------DDTK----\n",
      "Polyommatus_iph MEDDGDDK---------DDTK----\n",
      "Helleia_helle_E M-DDGHERPVAMSSRTLHTHERRYS\n",
      "Lysandra_bellar W-------------RTTETTRMTQN\n",
      "                                 .  .    \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "continue? Lysandra_bellar\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lysandra_bellargus_Exon_1_HG995319.1_11311478_11311515_left_1_right_1\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Species_name? Lysandra_bellargus\n",
      "Scaffold: HG995319.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_2.fa\n",
      "Exon_3.fa\n",
      "Exon_4.fa\n",
      "Exon_5.fa\n",
      "Exon_6.fa\n",
      "Exon_7.fa\n",
      "Exon_8.fa\n",
      "Exon_9.fa\n",
      "Exon_10.fa\n",
      "Exon_11.fa\n",
      "Exon_12.fa\n",
      "Exon_13.fa\n",
      "alignment_clustal_Exon_1.fa.txt\n",
      "alignment_clustal_Exon_2.fa.txt\n",
      "alignment_clustal_Exon_3.fa.txt\n",
      "alignment_clustal_Exon_4.fa.txt\n",
      "alignment_clustal_Exon_5.fa.txt\n",
      "alignment_clustal_Exon_6.fa.txt\n",
      "alignment_clustal_Exon_7.fa.txt\n",
      "alignment_clustal_Exon_8.fa.txt\n",
      "alignment_clustal_Exon_9.fa.txt\n",
      "alignment_clustal_Exon_10.fa.txt\n",
      "alignment_clustal_Exon_11.fa.txt\n",
      "alignment_clustal_Exon_12.fa.txt\n",
      "alignment_clustal_Exon_13.fa.txt\n",
      "error_exons.txt\n",
      "desktop.ini\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\") :\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        # error_exon = input(\"continue?\")\n",
    "        error_exon = \"y\"\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "                        # break\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
