{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7d7302c9-3ae7-41d6-bb94-577086e84ee0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Heliconius_charithonia-period', 'Napeogenes_sylphis-period', 'Mechanitis_messenoides-period', 'Mechanitis_mazaeus-period', 'Aricia_artaxerxes-period', 'Aricia_agestis-period', 'Polyommatus_iphigenia-period', 'Phengaris_arion-period', 'Celastrina_argiolus-period', 'Cyaniris_semiargus-period', 'Heliconius_nattereri-period', 'Dircenna_loreta-period', 'Eueides_isabella-period', 'Melitaea_cinxia-period', 'Lasiommata_megera-period', 'Pararge_aegeria-period', 'Danaus_chrysippus-period', 'Nymphalis_c-album-period', 'Coenonympha_glycerion-period', 'Maniola_hyperantus-period', 'Brenthis_hecate-period', 'Fabriciana_adippe-period', 'Nymphalis_urticae-period', 'Calycopis_cecrops-period', 'Hypolimnas_misippus-period', 'Hipparchia_semele-period', 'Maniola_jurtina-period', 'Melanargia_galathea-period', 'Mellicta_athalia-period', 'Oeneis_ivallda-period', 'Philaethria_dido-period', 'Napeogenes_inachia-period', 'Euphydryas_editha-period', 'Nymphalis_polychloros-period', 'Melinaea_menophilus-period', 'Polyommatus_icarus-period', 'Boloria_selene-period', 'Leptidea_juvernica-period', 'Lysandra_bellargus-period', 'Leptophobia_aripa-period', 'Leptidea_sinapis-period', 'Bicyclus_anynana-period', 'Vanessa_cardui-period', 'Brenthis_daphne-period', 'Papilio_demoleus-period', 'Zerene_cesonia-period', 'Parnassius_glacialis-period', 'Ornithoptera_alexandrae-period', 'Pieris_melete-period', 'Troides_oblongomaculatus-period', 'Glaucopsyche_alexis-period', 'Plebejus_argus-period', 'Lysandra_coridon-period', 'Nymphalis_io-period', 'Phoebis_sennae-period', 'Dryadula_phaetusa-period', 'Erebia_aethiops-period', 'Leptidea_reali-period', 'Pieris_mannii-period', 'Papilio_glaucus-period', 'Argynnis_bischoffii_washingtonia-period', 'Colias_nastes-period', 'Vanessa_atalanta-period', 'Aporia_crataegi-period', 'Battus_philenor-period', 'Papilio_dardanus_tibullus-period', 'Papilio_elwesi-period', 'Vanessa_tameamea-period', 'Boloria_euphrosyne-period', 'Brenthis_ino-period', 'Melinaea_marsaeus_rileyi-period', 'Papilio_protenor-period', 'Erebia_ligea-period', 'Pieris_napi-period', 'Dryas_iulia_moderata-period', 'Pieris_brassicae-period', 'Heliconius_sara-period', 'Anthocharis_cardamines-period', 'Eurema_hecabe-period', 'Danaus_plexippus-period', 'Ornithoptera_priamus-period', 'Troides_aeacus-period', 'Colias_eurytheme-period', 'Papilio_xuthus-period', 'Papilio_machaon-period', 'Colias_croceus-period', 'Pieris_rapae-period', 'Papilio_memnon-period', 'Teinopalpus_imperialis-period', 'Papilio_polytes-period']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading alignment from ./00.Temp_alignment/Heliconius_charithonia-period.fas ...\n",
      "Extracting sufficient statistics ...\n",
      "Compacting sufficient statistics ...\n",
      "Fitting tree model to ./00.Temp_alignment/Heliconius_charithonia-period.fas using REV ...\n",
      "numpar = 183\n",
      "Done.  log(likelihood) = -148048.277255 numeval=37224\n",
      "WARNING: BFGS algorithm reached its maximum number of iterations.\n",
      "Writing model to ./01.Temp_phastcon/modelfile.mod ...\n",
      "Done.\n",
      "Reading tree model from modelfile.mod...\n",
      "Reading alignment from ../00.Temp_alignment/Heliconius_charithonia-period.fas...\n",
      "Creating 'conserved' and 'nonconserved' states in HMM...\n",
      "Computing emission probs (state 0, cat 0, mod 0)...\n",
      "Computing emission probs (state 1, cat 1, mod 1)...\n",
      "Finding MLE for ([tree models])...\n",
      "Writing re-estimated tree models to mytrees.cons.mod and mytrees.noncons.mod...\n",
      "Done.\n",
      "Reading tree model from mytrees.cons.mod...\n",
      "Reading tree model from mytrees.noncons.mod...\n",
      "Reading alignment from ../00.Temp_alignment/Heliconius_charithonia-period.fas...\n",
      "Creating 'conserved' and 'nonconserved' states in HMM...\n",
      "Computing emission probs (state 0, cat 0, mod 0)...\n",
      "Computing emission probs (state 1, cat 1, mod 1)...\n",
      "Computing posterior probabilities...\n",
      "Done.\n",
      "Reading alignment from ./00.Temp_alignment/Napeogenes_sylphis-period.fas ...\n",
      "Extracting sufficient statistics ...\n",
      "Compacting sufficient statistics ...\n",
      "Fitting tree model to ./00.Temp_alignment/Napeogenes_sylphis-period.fas using REV ...\n",
      "numpar = 183\n",
      "Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x7f03752d8110>>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/saurav/miniconda3/envs/genomic_work/lib/python3.11/site-packages/ipykernel/ipkernel.py\", line 790, in _clean_thread_parent_frames\n",
      "    active_threads = {thread.ident for thread in threading.enumerate()}\n",
      "                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/saurav/miniconda3/envs/genomic_work/lib/python3.11/site-packages/ipykernel/ipkernel.py\", line 790, in <setcomp>\n",
      "    active_threads = {thread.ident for thread in threading.enumerate()}\n",
      "                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "KeyboardInterrupt: \n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 34\u001b[0m\n\u001b[1;32m     32\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/00.Temp_alignment/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlist_of_sequences_ids[i]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.fas\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m out_file:\n\u001b[1;32m     33\u001b[0m     out_file\u001b[38;5;241m.\u001b[39mwrite(output)\n\u001b[0;32m---> 34\u001b[0m \u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mcd \u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mlocation\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mphyloFit --tree period.masceDNA.fas.treefile --subst-mod REV --out-root \u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m./01.Temp_phastcon/modelfile\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m \u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m./00.Temp_alignment/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mlist_of_sequences_ids\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m.fas\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshell\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m     36\u001b[0m subprocess\u001b[38;5;241m.\u001b[39mrun(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcd \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/01.Temp_phastcon\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mphastCons --target-coverage 0.25 --expected-length 12 --estimate-trees mytrees ../00.Temp_alignment/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlist_of_sequences_ids[i]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.fas modelfile.mod --no-post-probs\u001b[39m\u001b[38;5;124m'\u001b[39m, shell \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m     38\u001b[0m subprocess\u001b[38;5;241m.\u001b[39mrun(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcd \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/01.Temp_phastcon\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mphastCons --target-coverage 0.25 --expected-length 12 ../00.Temp_alignment/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlist_of_sequences_ids[i]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.fas mytrees.cons.mod,mytrees.noncons.mod > scores.wig\u001b[39m\u001b[38;5;124m'\u001b[39m, shell \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "File \u001b[0;32m~/miniconda3/envs/genomic_work/lib/python3.11/subprocess.py:550\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    548\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Popen(\u001b[38;5;241m*\u001b[39mpopenargs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs) \u001b[38;5;28;01mas\u001b[39;00m process:\n\u001b[1;32m    549\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 550\u001b[0m         stdout, stderr \u001b[38;5;241m=\u001b[39m \u001b[43mprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcommunicate\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    551\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m TimeoutExpired \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[1;32m    552\u001b[0m         process\u001b[38;5;241m.\u001b[39mkill()\n",
      "File \u001b[0;32m~/miniconda3/envs/genomic_work/lib/python3.11/subprocess.py:1201\u001b[0m, in \u001b[0;36mPopen.communicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1199\u001b[0m         stderr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mread()\n\u001b[1;32m   1200\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mclose()\n\u001b[0;32m-> 1201\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1202\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1203\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/miniconda3/envs/genomic_work/lib/python3.11/subprocess.py:1264\u001b[0m, in \u001b[0;36mPopen.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1262\u001b[0m     endtime \u001b[38;5;241m=\u001b[39m _time() \u001b[38;5;241m+\u001b[39m timeout\n\u001b[1;32m   1263\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1264\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1265\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m:\n\u001b[1;32m   1266\u001b[0m     \u001b[38;5;66;03m# https://bugs.python.org/issue25942\u001b[39;00m\n\u001b[1;32m   1267\u001b[0m     \u001b[38;5;66;03m# The first keyboard interrupt waits briefly for the child to\u001b[39;00m\n\u001b[1;32m   1268\u001b[0m     \u001b[38;5;66;03m# exit under the common assumption that it also received the ^C\u001b[39;00m\n\u001b[1;32m   1269\u001b[0m     \u001b[38;5;66;03m# generated SIGINT and will exit rapidly.\u001b[39;00m\n\u001b[1;32m   1270\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/miniconda3/envs/genomic_work/lib/python3.11/subprocess.py:2053\u001b[0m, in \u001b[0;36mPopen._wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   2051\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   2052\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m  \u001b[38;5;66;03m# Another thread waited.\u001b[39;00m\n\u001b[0;32m-> 2053\u001b[0m (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_try_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2054\u001b[0m \u001b[38;5;66;03m# Check the pid and loop as waitpid has been known to\u001b[39;00m\n\u001b[1;32m   2055\u001b[0m \u001b[38;5;66;03m# return 0 even without WNOHANG in odd situations.\u001b[39;00m\n\u001b[1;32m   2056\u001b[0m \u001b[38;5;66;03m# http://bugs.python.org/issue14396.\u001b[39;00m\n\u001b[1;32m   2057\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m pid \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid:\n",
      "File \u001b[0;32m~/miniconda3/envs/genomic_work/lib/python3.11/subprocess.py:2011\u001b[0m, in \u001b[0;36mPopen._try_wait\u001b[0;34m(self, wait_flags)\u001b[0m\n\u001b[1;32m   2009\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"All callers to this function MUST hold self._waitpid_lock.\"\"\"\u001b[39;00m\n\u001b[1;32m   2010\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 2011\u001b[0m     (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwaitpid\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpid\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwait_flags\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2012\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mChildProcessError\u001b[39;00m:\n\u001b[1;32m   2013\u001b[0m     \u001b[38;5;66;03m# This happens if SIGCLD is set to be ignored or waiting\u001b[39;00m\n\u001b[1;32m   2014\u001b[0m     \u001b[38;5;66;03m# for child processes has otherwise been disabled for our\u001b[39;00m\n\u001b[1;32m   2015\u001b[0m     \u001b[38;5;66;03m# process.  This child is dead, we can't get the status.\u001b[39;00m\n\u001b[1;32m   2016\u001b[0m     pid \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/11.Alignment\"\n",
    "\n",
    "fasta_file = SeqIO.parse(f\"{location}/period.masceDNA.fas\", 'fasta')\n",
    "\n",
    "list_of_sequences_ids = []\n",
    "\n",
    "for records in fasta_file:\n",
    "    list_of_sequences_ids.append(records.id)\n",
    "print(list_of_sequences_ids)\n",
    "# list_of_folders = os.listdir(f\"{location}\")\n",
    "# print(list_of_folders)\n",
    "# if \"00.Temp_alignment\" in list_of_folders:\n",
    "#     print(\"AA\")\n",
    "#     subprocess.run(f'rm -r \"{location}/00.Temp_alignment/\"', shell = True)\n",
    "\n",
    "# os.mkdir(f\"{location}/00.Temp_alignment\")\n",
    "\n",
    "\n",
    "for i in range(len(list_of_sequences_ids)):\n",
    "    output = ''\n",
    "    fasta_file = SeqIO.parse(f\"{location}/period.masceDNA.fas\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        sequence = f\">{records.id}\\n{records.seq}\\n\"\n",
    "        if list_of_sequences_ids[i] == records.id:\n",
    "            output = sequence + output\n",
    "        else:\n",
    "            output += sequence\n",
    "    with open(f\"{location}/00.Temp_alignment/{list_of_sequences_ids[i]}.fas\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "    # subprocess.run(f'cd \"{location}\"\\nphyloFit --tree period.masceDNA.fas.treefile --subst-mod REV --out-root \"./01.Temp_phastcon/modelfile\" \"./00.Temp_alignment/{list_of_sequences_ids[i]}.fas\"', shell = True)\n",
    "    \n",
    "    # subprocess.run(f'cd \"{location}/01.Temp_phastcon\"\\nphastCons --target-coverage 0.25 --expected-length 12 --estimate-trees mytrees ../00.Temp_alignment/{list_of_sequences_ids[i]}.fas modelfile.mod --no-post-probs', shell = True)\n",
    "    \n",
    "    subprocess.run(f'cd \"{location}/01.Temp_phastcon\"\\nphastCons --target-coverage 0.25 --expected-length 12 ../00.Temp_alignment/{list_of_sequences_ids[i]}.fas mytrees.cons.mod,mytrees.noncons.mod > scores.wig', shell = True)\n",
    "\n",
    "    fasta_file = SeqIO.parse(f\"{location}/00.Temp_alignment/{list_of_sequences_ids[i]}.fas\", 'fasta')\n",
    "\n",
    "    for records in fasta_file:\n",
    "        sequence = records.seq\n",
    "        sequence_name = records.id\n",
    "        break\n",
    "    \n",
    "    with open(f\"{location}/01.Temp_phastcon/scores.wig\", 'r') as score_file:\n",
    "        scores = score_file.readlines()\n",
    "    \n",
    "    output = 'Pos\\tScore'\n",
    "    score_counter = 1\n",
    "    for j in range(len(sequence)):\n",
    "        if sequence[j] != \"-\":\n",
    "            output += f\"\\n{j+1}\\t{scores[score_counter].strip()}\"\n",
    "            score_counter += 1\n",
    "        else:\n",
    "            output += f\"\\n{j+1}\\tN/A\"\n",
    "    \n",
    "    # print(output)\n",
    "    with open(f\"{location}/01.Temp_phastcon/scores_fixed_{list_of_sequences_ids[i]}.wig\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "        # assert False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74eca1fc-bc9e-4369-8829-4f42cbdb79f2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
