{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "78df7b31",
   "metadata": {},
   "source": [
    "# Parse from JSON file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e36396fc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import json\n",
    "with open (\"F:/genome_test/ncbi_dataset/data/assembly_data_report.jsonl\", 'r') as json_file:\n",
    "    json_file_as_list = json_file.readlines()\n",
    "\n",
    "output = \"Accession\\tNumber of Scaffolds\\tContig N50\\n\"\n",
    "for accession_detail_line in json_file_as_list:\n",
    "    accession_detail = json.loads(accession_detail_line)\n",
    "    contig_n50 = (int((accession_detail[\"assemblyStats\"][\"contigN50\"]))/1000000)\n",
    "    accession_number = ((accession_detail[\"accession\"]))\n",
    "    scaffold_number = int(accession_detail[\"assemblyStats\"][\"numberOfScaffolds\"])\n",
    "#     print(accession_number,scaffold_number,contig_n50 )\n",
    "    output = output + (f\"{accession_number}\\t{scaffold_number}\\t{contig_n50}\\n\")\n",
    "    \n",
    "with open (\"F:/genome_test/temp_details.txt\",'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f56063a7",
   "metadata": {},
   "source": [
    "# Parse from Txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95dd231d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "\n",
    "with open(\"J:/genome_test/test2/gcf.txt\",'r') as list_file:\n",
    "    \n",
    "    list_of_accessions = list_file.readlines()\n",
    "\n",
    "output = \"Accession\\tScaffold_number\\tContig_N50\\n\"\n",
    "for i in range(len(list_of_accessions)):\n",
    "    accession = list_of_accessions[i].rstrip()\n",
    "#     print(accession)\n",
    "#     break\n",
    "    \n",
    "    location = f\"J:/genome_test/test2/genbank/invertebrate/{accession}\"\n",
    "    try:\n",
    "        files_in_accession_folder = os.listdir(location)\n",
    "    except FileNotFoundError:\n",
    "        output += f\"{accession}\\tN/A\\tN/A\\n\"\n",
    "        continue\n",
    "    for file in files_in_accession_folder:\n",
    "        if \"assembly_stats\" in file:\n",
    "           with open(f\"{location}/{file}\", 'r') as assembly_stat_file:\n",
    "                stat_file_list = assembly_stat_file.readlines()\n",
    "                break\n",
    "    \n",
    "    for lines in stat_file_list:\n",
    "        if \"all\tall\tall\tall\tscaffold-count\" in lines:\n",
    "            scaffold_count = lines.split(\"\t\")[-1].rstrip()\n",
    "        if \"all\tall\tall\tall\tcontig-N50\" in lines:\n",
    "            contig_n50 = int(lines.split(\"\t\")[-1].rstrip())/1000000\n",
    "#     print(scaffold_count, contig_n50)\n",
    "    output += f\"{accession}\\t{scaffold_count}\\t{contig_n50}\\n\"\n",
    "    \n",
    "with open(\"J:/genome_test/test2/accession_details.txt\",'w') as output_file:\n",
    "    output_file.write(output)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b52d3e58",
   "metadata": {},
   "source": [
    "# Code for extraction and makeblastdb\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de32354e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "location = \"J:/Genomes_2023-12-26\"\n",
    "\n",
    "list_of_species = os.listdir(location)\n",
    "output = '#!/bin/zsh\\n'\n",
    "for species in list_of_species:\n",
    "    files_list = os.listdir(f\"{location}/{species}\")\n",
    "    for files in files_list:\n",
    "        if \".gz\" in files:\n",
    "            output += f\"echo {species} \\ngzip -d /mnt/j/Genomes_2023-12-26/{species}/{files}\\n\"\n",
    "            \n",
    "#             print(species)\n",
    "\n",
    "print(output)\n",
    "            \n",
    "# with open(\"J:/My Drive/Circadian Rhythm Genes Project/1.Codes/bash scripts/unzip.sh\", \"w\") as out_file:\n",
    "#     out_file.write(output)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ec95918",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "location = \"J:/Genomes_2023-12-26\"\n",
    "\n",
    "list_of_species = os.listdir(location)\n",
    "output = '#!/bin/zsh\\n'\n",
    "for species in list_of_species:\n",
    "    files_list = os.listdir(f\"{location}/{species}\")\n",
    "    for files in files_list:\n",
    "        if \"genomic.fna\" in files[-11:]:\n",
    "            output += f\"echo {species} \\nmakeblastdb -in /mnt/j/Genomes_2023-12-26/{species}/{files} -dbtype nucl\\n\"\n",
    "    \n",
    "#             print(species)\n",
    "\n",
    "# print(output)\n",
    "            \n",
    "with open(\"K:/My Drive/Circadian Rhythm Genes Project/1.Codes/bash scripts/makeblastdb.sh\", \"w\") as out_file:\n",
    "    out_file.write(output)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f09b92b4",
   "metadata": {},
   "source": [
    "# Parse gene table from NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fd2bc12b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12492407', '12492237']\n"
     ]
    }
   ],
   "source": [
    "import re \n",
    "from Bio import Entrez, SeqIO, Align\n",
    "from Bio.Align import substitution_matrices\n",
    "from io import StringIO\n",
    "\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.mode = 'global'\n",
    "aligner.substitution_matrix = substitution_matrices.load(\"BLOSUM62\")\n",
    "aligner.open_gap_score = -10\n",
    "aligner.extend_gap_score = -0.5\n",
    "aligner.target_end_gap_score = 0.0\n",
    "aligner.query_end_gap_score = 0.0\n",
    "\n",
    "\n",
    "Entrez.email = 'saubaral@gmail.com'\n",
    "\n",
    "\n",
    "def get_protein_sequence(ncbi_protein_id):\n",
    "    handle = Entrez.efetch(db=\"protein\", id=ncbi_protein_id, rettype=\"fasta\", retmode=\"text\")\n",
    "    record = handle.read()\n",
    "    fasta_io = StringIO(record) \n",
    "    protein_sequence_element = SeqIO.parse(fasta_io, \"fasta\")\n",
    "    for record in protein_sequence_element:\n",
    "        return(record)\n",
    "\n",
    "\n",
    "species = \"1.Per period[Bombyx mori]\"\n",
    "bombyx_protein_accession = \"NP_001036975.1\"\n",
    "bombyx_protein = get_protein_sequence(bombyx_protein_accession)\n",
    "\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table.txt\",\"r\") as gene_table_file:\n",
    "    gene_table_list = gene_table_file.readlines()\n",
    "\n",
    "    \n",
    "exon_list = \"Exon1,Exon2,Exon3,Exon4,Exon5,Exon6,Exon7,Exon8,Exon9,Exon10,Exon11,Exon12,Exon13,Exon14,Exon15,Exon16,Exon17,Exon18,Exon19,Exon20,Exon21,Exon22,Exon23,Exon24,Exon25,Exon26,Exon27,Exon28,Exon29,Exon30,Exon31,Exon32,Exon33,Exon34\".split(\",\")\n",
    "i = 0\n",
    "switch = 0\n",
    "output = \"mRNA Accession\\tExon Number\\tCoding Length\\tGenome Start Point\\tGenome Stop Point\\n\" \n",
    "output2 = \"mRNA Accession\\tProtein Accession\\tExon Number\\tAlignment Score with Bombyx mori\\n\"\n",
    "exon_counter = 0\n",
    "\n",
    "while i < (len(gene_table_list)):\n",
    "    \n",
    "    if \"Exon table for  mRNA  \" in gene_table_list[i]:\n",
    "        gene_table_line = gene_table_list[i]\n",
    "        while re.search(\"[ ]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"  \",\" \")\n",
    "            \n",
    "        mrna_accession = gene_table_line.split(\" \")[4]\n",
    "        protein_accession = gene_table_line.split(\" \")[7].rstrip()\n",
    "#         print(get_protein_sequence(protein_accession))\n",
    "#         alignment_score = aligner.score(bombyx_protein.seq, get_protein_sequence(protein_accession).seq)\n",
    "#         print(alignment_score)\n",
    "        \n",
    "#         for alignment in alignments:\n",
    "#             print(\"Score = %.1f:\" % alignment.score)\n",
    "#         assert False\n",
    "        \n",
    "#         print(f\"{mrna_accession}\\t{protein_accession}\\t{alignment_score}\", end = \"\\t\")\n",
    "        \n",
    "        switch = 1\n",
    "        i += 3\n",
    "    if switch == 1:\n",
    "\n",
    "        gene_table_line = gene_table_list[i]\n",
    "        while re.search(\"[\\t]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"\\t\\t\",\"\\t\")\n",
    "#         print(len(gene_table_line.split(\"\\t\")))\n",
    "        if len(gene_table_line.split(\"\\t\")) == 7 or len(gene_table_line.split(\"\\t\")) == 6:\n",
    "            \n",
    "            coding_exon_length =  ((gene_table_line.split(\"\\t\")[5]))\n",
    "            genome_start_point,genome_end_point = ((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "            print((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "            output += f\"{mrna_accession}\\t{exon_list[exon_counter]}\\t{coding_exon_length.rstrip()}\\t{genome_start_point}\\t{genome_end_point}\\n\"\n",
    "            \n",
    "        \n",
    "            exon_counter += 1\n",
    "        \n",
    "        if switch == 1 and re.search(\"^mRNA\",gene_table_list[i]):\n",
    "            switch = 0\n",
    "            print(exon_counter)\n",
    "#             output2 += f\"{mrna_accession}\\t{protein_accession}\\t{exon_counter}\\t{alignment_score}\\n\"\n",
    "            output += \"\\n\\n\"\n",
    "            exon_counter = 0\n",
    "            \n",
    "    i += 1\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table_processed1.tsv\",'w') as out_file:\n",
    "    out_file.write(output)\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table_alignment_score1.tsv\",'w') as out_file:\n",
    "    out_file.write(output2)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "351cc700",
   "metadata": {},
   "source": [
    "# Download ncbi sequence using accession"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "dd002047",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">XM_022976811.1 PREDICTED: Spodoptera litura period circadian protein (LOC111360701), transcript variant X3, mRNA\n",
      "GCTCCGGCTAGCTGAACGTTTTGTGAGAAGTAACTCGTAACTCATTCGTTTTGTCGTTTGTTCTTTTATT\n",
      "AGGTACTGTTAAACTTATATATTCGTCGATCAATATTAGATGTCGAAAAGTTTAAAAGTGCTATAATTTG\n",
      "TTACGATAATAACGTACTAATTACTGTTTGAAACTTGATTAACTAAAGTGACCCTTGTGAAAACGGAATA\n",
      "TTCACCTACTCTAAACAAGGATAACGTGATTTCGGCATTGTTTCTTTTTAAAGACCATCAATGCAACATG\n",
      "GACAACTTGGATGATTCAGAGAACAATGCGAAGATTTCAGATTCGGCATACTCCAACAGTTGCAGTAATA\n",
      "GTCAATCTAGAAGGAGTCATAGTTCTAAGTCAACACATTCTGGAAGCAATTCCAGTGGCAGCAGTGGCTA\n",
      "TGGCGGGAAACCCTCCACATCTGGAAGCAGCAACAACTTGGGTCAACCACCTAAAAAAGATAAGGAACCC\n",
      "AAGAAAAAGAAGCCTCAAATTGAAACAATGGTCCCAGATGTAGTTGTGGAAACGGAGGCAAGGGCGCCTG\n",
      "AGCCGGCCCCTACATTTGATGCTCCGAAAGAAGATCAGCTAGATGTTATCCCGTCGCTACCACCTGCGCA\n",
      "TGCGCAATCAGAGAAGGGAGTGGAGAACATGGACATATGTACCCCTGAGTTGAATACTTTAAAAGATGAG\n",
      "GTGGTTTCCTGCAACATATCTCAGGTTAATCCTGCTTCAAGCTTTGTTACCGGTCGTCCCCTGCAAGCTT\n",
      "CTTATTGCTCGGTACAAAGTAAACGATACCAAGAGATTGGCCGCTTCGAGATGCGATGTGAGAGCTCTGA\n",
      "ACAGTTGCGTTGCAAGGATGGATTTTCATGCGTGATATCTATGCACGATGGTGTCGTAATGTACGCAACG\n",
      "TCATCATTGACGACTACTCTTGGTTTCCCCAAAGACATGTGGATTGGAAGATCATTCATCGACTTCGTTC\n",
      "ATCCAAGGGATCGCAACACCTTTGCATCACAAATCACGAATGGTTTGGCCGTCCCTAAAATTGTTAATGG\n",
      "TACGCAAGAAAAAGTTCAATCTCCAGCTAACTCGGTGTCAACAATGGTTTGTCGGATACGGCGGTACAGA\n",
      "GGCCTCACTACAGGGTTCGGTGTTAAAGAGAGAGTCGTGGGGTTCATGCCGTTCCTGCTCAAACTCACGT\n",
      "TCAAAAACATAAGCGATGAAGAGGGTAAAGTCATCTACTTGGTTATACAGGCGACACCATTTTTCTCTGC\n",
      "GTTTAAAATGCCGAACGAGGTGGTGGTAAAAGCGGTGCCTTTTGTCATAAGGCATGCGGCTAACGGTCAC\n",
      "ATTGAATATGTGGATCCTGAGTCTGTTCCGTACCTGGGGTACTTGCCTCAAGACCTCGCTGACAAGGACG\n",
      "CGCTGATGCTATACCATCCTGATGACTTGATGTATCTACGACAAGTGTATGATACCATTGTCAAAGAAGG\n",
      "AGGTCTCCCTAGATCTAAGGCTTATCGTATGATTACACAAAACGGTGACTACATTCGGCTGGAAACGGAG\n",
      "TGGTCATCGTTCATCAATCCTTGGTCTAAGAAACTGGAGTTCGTTATTGGAAAACACCATATTTTAGAGG\n",
      "GCCCGACCAACCCAGACGTCTTTCAATGTCCTGAGCCGGAAAAATCAATGAAAGTTCCCGAAGAAGAAAA\n",
      "AAACAAGGCACAAATGCTGAGAGAGACCATTATTCGAACAATGAATGAGGCTCTGACTAAACCCGCGGAA\n",
      "ATAGCGAAGCAACAGATGAGCAAGCGTTGTCAAGATCTTGCTTCTTTCATGGAGAGTCTAATGGAGGAAG\n",
      "TGCCCAAGACTGATGAGGAACTGCGCTTAGAGATTCACGATCCTGATCACAGCTATTATGAAAGGGACTC\n",
      "GGTGATGCTTGGTGGGATTTCCCCACACCATGACTATTATGACAGCAAGTCTAGTACTGAGACTCCGTTG\n",
      "AGTTACAACCAATTGAACTATAATGAAACGCTGCAGAGGTATTTCGATAGTCACCAACCTATCTCCTATG\n",
      "AGGACTACAACACGGTCACCGGCGAAAACATTCTCGGACTGAAAGATCCGAGACCAATCATAAATCATTG\n",
      "CCTGTCTCCGATGGCGCAGCATTCGGGAGACTCTGGGGAAATGACCTGTTCGTCTGATTCCAACGGCATG\n",
      "GTCATGGGCAGTAACAGTCCCGTGATGCCCCTCGGAGACTACCAACCAATACGTCTCACTGAGGCGTTGC\n",
      "TTAGCAAGCACAATGCTGAGATGGAAAAAGAGTTAATGAAGATGCATAGAGAGACTCGTTCTAGCAGTAA\n",
      "GGGGGATCGGGAGAAGAACTCAAATGAGACTAGGAAGAAGAAGAAGGAGCACTTGGCGCGTTGCAACGCG\n",
      "TCCTACCATCCTACTAGTGCTGGCGCCACCAATACTGAACATCAGCAACCTCACGGTGTAAAGCGTACAT\n",
      "CTAAAATGATGGACGCTGAGGCCGGCGCTCACAAGCACCACTGCCCGTCGCCGCGTCAGGCGCGTCGCAA\n",
      "TAAACGCACGTTGTCTACCAACGCGGCCGTCGCCCAACCATCTACCAGCGTTACCACGACCACTGTCGGT\n",
      "CAGTTGGCTACAGCATCGAATCATTGGCCAACCCCTCCTGCAAACAATATGAACACGTTCATCTTGGGCG\n",
      "TAGGAATCCCTCAGCAGATGTCTATAATGAGCCCCATGCCTCATCCTATGCAGGCCATGCACACTATGCC\n",
      "GGCGGTGCCAGGCATGTTCCCGATGTACTACGCGCCGGCGGCAGCGCCGCAACCGATGCCAACAAGCTCA\n",
      "GCGCATCATCACATTCCTTCAAGCAGCCAGCATCAGTTTCCTGCACCCATGATGATGTACAGCCAGCCAA\n",
      "TGTACGGAGCTCCTTTCATGTACTCGCCAATGGCCCAACAAATGTCTTACCCCATGCAGCAATCTCACAT\n",
      "GATGGCACAGTCCATGCAACAATACACCAATACCATGAACCCTTTAGGGTTGACCAGCAGCAACTATGAA\n",
      "GAGGCTTGTAAGCCCAGTTTAACATTGAGGCCAGGCAAGGTGCAAGGCAGTGTGTGGCGCGACAAGAAAA\n",
      "GAGCAGAATCTCAAGGCACATCTTCAAAAACAGACAACGGCAGCGCAGATAGTAACTCATCTACGAATGC\n",
      "GTTGAATCGGGTGTCGGGGACGCAAAAAAGTCACAGTGGGAACGAGAAGTCTACGATATACTCGACTGAC\n",
      "ATCAGATCTTCAAACAGTATGAGCGCTAAGAGCACAGAGGCTGTGCCGAACACTCCACCTTCAGCCAGGG\n",
      "AGAGTCGCCTCAGCAAGCTGAATCGGCTGGGCAACAGCGAGGAGACCCCCGACAAGACTGATGGAGAGTC\n",
      "TAGCTACTCGTCCTTCTACTCTTCCTTTTTTAAGACTGAGTCCGGTAGTGCCGAGGATAGTGGTGACGGC\n",
      "AAGAATGGCAAGGACAAGCAAGCTAGGAGCCCCATAAATCTGACATCTTCATACCATGTATCTCACCCAA\n",
      "CGAAGAAGGTTGCCAGAAGGAAAATGGAACCGCCATGGTTGGAACAGGTATGCGTGACATCAGAGTTAGT\n",
      "CTACAAGTACCAAATTCTTACCAAGTCTATGGAGGAGACTTTATCCTCTGACAAACAGAAGATGAAAACT\n",
      "CTGGAACAGCCTTCACTAGTGAACGAGCAGCTGAGCCAGTTGTACCTAGACCTTCAGCTGGAAGGTGCTG\n",
      "CGGCGAGGCTGACCTTGGAGGAAGGTATCACTAGTTCCAGTAGTTCCGGTGAAGAAACCACCACAGTTAC\n",
      "AAAGCCTGCTTCTAACAGAGCCTCAACTTCTGGAAAAAGCGAAGGAAAGAGACATACGACTTCACGTAGG\n",
      "AAACGGGAGTACAGCAAGTTAGTTATGATTTACGAAGAAGATGCGCCGTTGCCGCCGCCTGCTGAAGACG\n",
      "TCGTGGAAGGGGGTGAGGCATCCAGCTCGTCCACTTGTTAAATTAGGACCGTCAAAGGTCTAGTCGCAAG\n",
      "TCTTCGTCACCCTCGTCACATTATGTCGTAGCAATGCACAATAATAGTAAATTAATTTGCATTTAAGCGT\n",
      "CATGATAAAACGCACGACGGTTCAGGGAATATTAAACTTCAAATATTCGTAAATAGACATATAGTTGTGT\n",
      "GTAAGTATGAACACGTGTCGTAGCTGTGTGTTTGACGGGCCTTTCTGTACTATGATGAAGATATCGTGCT\n",
      "TGAGCTCGTGTCTTCGAACCCAGATCTCATTCAGTTCTCGAGTTAACTTGAATGCAGTAGATGTGTGGTG\n",
      "TGATTCTAACTGTAACTTAAGTCTCTGTGTTACATTTATCTTATGTATAGACTGCCATTAGCACCAAATA\n",
      "ATAGTGATCGTAGACTACTTTGCTTAATTGATAAATATGGAAATTGAATTATTTTTGTTACGATTTTGTT\n",
      "GCGTGTGTTATTATTTTGTTTGTATTTTATGAAACAGATTCTAATACTGAATTTTAAAATATTTGATATA\n",
      "AACGTGAGTTAATCTGGGAATGTTAAGAATACAAGATTTTGTTTGTCCTATTCGAATCTGTGATACTCAA\n",
      "ATTCCTATTACAGATAGGTAGTACAACTAGATAAAAAAGAGATTTATTAATATAATTATGTATCTATGTC\n",
      "ATTTAATGTAAATGACTTATTAATATTATTATATTGGGGAAAGTAGTGAACTCAGTTTGTGCGCTTATTG\n",
      "TATCTTTAGTTCTCTTAACATTCCGTTTGGTGCTCTGTTGCTCACAATGGCTTGAAAAGTATTTTAACAG\n",
      "GGTAAAGTATTATGATGAATCAGAATCAAATATTATATTTTCATAATAGGCAAGTTTCCGACTAGTCAAG\n",
      "ATAAATACTTTTTTGCAAAGTCAAAGGATATTTTTTTTTGATTTTGTAGGTACGAGATTGGGTGCTGTGA\n",
      "CATAACAGCATACAGTTTTCTTATGTTTCTAAGAAAATACGAAAATTTCGTATTAAAGAAAATGACTTTT\n",
      "TTTTTTAATTTTACTGTATTATAAAGTCAAATTACTGACTAAGGTATTCTATGAGTTTTGTAGATACCTA\n",
      "CAAATCTTCGCAGTCTGTTTGTTGTTATGTCACAATAATTCCACGTCAAGTACTGCCAAATAATGCACTC\n",
      "ATTTATTTAAAAATATTTGAAAATTAAGTAAATTAATTACATGATCGAATCAATTAATGAAAGTATGATT\n",
      "ATTTTTAATATTCTATTGTATTAATTGGTCGAAATAATGTATTGTTGACTGTGGAAACTACCCTACTTTA\n",
      "TTTATACACATAAATATATATATATGAGCGTTATGTTTATTGAATTATACTTACAGTCAAAGACTTATTG\n",
      "CATTTACTCGATCACGCAAGACATTGCTATGTTGGTTGTGCGTTGTCCTATTTATGCATTTGTAATAAAT\n",
      "AAAATAAAGAGCTGGACTGACTTGAACGTATTGTGATATGAAAGAATTGTCAGATTTATTATCATTTGTT\n",
      "GTATACTTAAGTGGTAATTTTTGCGTTATCGAGTGATTATTATGTTAAACGTGTCTTAGTTTTATTTTAT\n",
      "GAAACAAAGTTGTCAAAATGAACCCATCATCGTAACAGGTGCCTTAGAATCGATCTTTAAACTATTGTTC\n",
      "AGTGTCAATTTAATCGATCAAAATTACAATAAATTGATTTTATTTTGTTTCACGATGTCATTAGAGTAGG\n",
      "GTGTGTTGATTAAGCTGTTGGCAAAAGTCCGTGCAGGGTTTGACATTATCAACTTTTGTTTAGTATTTCC\n",
      "TACTAAAAGAATAACAATCTTTTTGAGATATCTCTTTATATTTAAATCGAGTTTGTAATTAAATAATTGA\n",
      "TATAATGCATTTATTATAAATGGTATATGTGATTGACGAATTGATGAGTTTTTAATGTGTTGAAAGTCTA\n",
      "GTTTTGTATGACTCGAAATATATTTTTTAAAAGAGAAATTTTAAAAGTTATGGAATTGTCGTGCGGTCTT\n",
      "TAAATGCCAAGGTTACCAAATTGATGTAGGTTATAGAAAATTTGAATACTTAAATGAGGCATTCATTCCT\n",
      "ACGTTGAGAATCAGAACAGACAGAATAGGTGACTGTGGTTACCTTAGTCCTAGTTAAGATATACAGTAAG\n",
      "TGTGATTAACGTTAAGTTTGATACGATAGTTATTTGTAAATATTATTTGTGCACTAACGATTACAATAAG\n",
      "TTGGTGTTTACTATTATGTTAAATTAAAGTTATCCGTTTCTAATAGAGACGTATAAGAGAATTTTATCAC\n",
      "TAAATTGCAATATAGTATAGTATTCATGCTGTTTTATAGTTGAATACATAGTTTATTACATGCATAGATG\n",
      "TTTCTTTTTATTTTTATCTTGTGTGTGGTAACTGTCCTACCATTTTTGTCAATAAAAGAGAAAGAAATGT\n",
      "TGAATTTATTACATGCCAGTAATTTTATTCTTATTTTGAATTAAAGTGGCGATTAAATAGAAGCGGTATT\n",
      "TTTTTTAATGGTAAATTCTAGTTTCCTTTTAATAAACCGTGCCGTTATTTTGTCGTCTATCACTTTAGTC\n",
      "TTCTGTCTAATTTAGATTATTAAAGCTTATAAGGGAGGGGAGGAGAATTACGTTTTTCTTAAAATCATTG\n",
      "CCATTAAATAATTAATTTTATTTGCTAATTATTACGATTCTGAAAACGAATTGGCGTTTATTTTATTGAT\n",
      "GTTGCTTATTATTTTTAAAATAATTGTATTATGAGAAAATTGAAATGTTACGAATAATTCACCAATATAT\n",
      "GACATTTCTGTGAAATAAATGATTGGAAAGTGAATA\n",
      "\n",
      "\n",
      "XM_022976811.1\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez, SeqIO\n",
    "from io import StringIO\n",
    "\n",
    "Entrez.email = 'saubaral@gmail.com'\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"XM_022976811.1\", rettype=\"fasta\", retmode=\"text\")\n",
    "# print (handle)\n",
    "record = handle.read()\n",
    "print(record)\n",
    "fasta_io = StringIO(record) \n",
    "record1 = SeqIO.parse(fasta_io, \"fasta\")\n",
    "for sequences in record1:\n",
    "    print(sequences.id)\n",
    "# record.write(record.rstrip('\\n'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b53dc81",
   "metadata": {},
   "source": [
    "# Parse BLAST result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "e5196c32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bombyx mori period (Per), mRNA\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "\n",
    "file_list = {}\n",
    "enumerator = 0\n",
    "with open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/T0260275016-Alignment-Descriptions.csv\", 'r') as blast_out_file:\n",
    "    blast_out_file_content = csv.reader(blast_out_file, delimiter=',', quotechar='\"')\n",
    "\n",
    "    \n",
    "    for data in blast_out_file_content:\n",
    "        file_list[str(enumerator)] = data\n",
    "        enumerator += 1\n",
    "#         break\n",
    "\n",
    "output = (\"\\t\".join(file_list[str(0)]))\n",
    "\n",
    "\n",
    "for i in range(1,len(file_list)-1):\n",
    "    try:\n",
    "        for k in file_list[str(i)]:\n",
    "#             print (k)\n",
    "            if \"NM_\" in k:\n",
    "                print(file_list[str(i)][0])\n",
    "                output+= \"\\n\"+((\"\\t\".join(file_list[str(i)])))\n",
    "    #         break\n",
    "    except:\n",
    "        print(f\"Error{file_list[str(i)]}\")\n",
    "\n",
    "# output = \"\\n\".join(out_list)\n",
    "\n",
    "# with open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/filtered_blast.tsv\", 'w') as blast_output_file:\n",
    "#     blast_output_file.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4714dfa8",
   "metadata": {},
   "source": [
    "# Genomes with annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "78271f0b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species\tAnnotation\n",
      "Abrostola_tripartita\tNO\n",
      "Abrostola_triplasia\tNO\n",
      "Acentria_ephemerella\tNO\n",
      "Achlya_flavicornis\tNO\n",
      "Acleris_cristana\tNO\n",
      "Acleris_emargana\tNO\n",
      "Acleris_holmiana\tNO\n",
      "Acleris_literana\tNO\n",
      "Acleris_sparsana\tNO\n",
      "Acrobasis_consociella\tNO\n",
      "Acrobasis_repandana\tNO\n",
      "Acrobasis_suavella\tNO\n",
      "Acronicta_aceris\tNO\n",
      "Acronicta_leporina\tNO\n",
      "Acronicta_psi\tNO\n",
      "Adoxophyes_honmai\tNO\n",
      "Agonopterix_arenella\tNO\n",
      "Agonopterix_heracliana\tNO\n",
      "Agonopterix_subpropinquella\tNO\n",
      "Agriopis_aurantiaria\tNO\n",
      "Agriopis_leucophaearia\tNO\n",
      "Agriopis_marginaria\tNO\n",
      "Agriphila_geniculea\tNO\n",
      "Agriphila_straminella\tNO\n",
      "Agriphila_tristella\tNO\n",
      "Agrochola_circellaris\tNO\n",
      "Agrochola_litura\tNO\n",
      "Agrochola_lota\tNO\n",
      "Agrochola_lychnidis\tNO\n",
      "Agrochola_macilenta\tNO\n",
      "Agrotis_clavis\tNO\n",
      "Agrotis_exclamationis\tNO\n",
      "Agrotis_ipsilon\tNO\n",
      "Agrotis_puta\tNO\n",
      "Alcis_repandata\tNO\n",
      "Allophyes_oxyacanthae\tNO\n",
      "Alsophila_aescularia\tNO\n",
      "Amorpha_juglandis\tNO\n",
      "Amphipoea_lucens\tNO\n",
      "Amphipoea_oculea\tNO\n",
      "Amphipyra_berbera\tNO\n",
      "Amphipyra_tragopoginis\tNO\n",
      "Amyelois_transitella\tYes\n",
      "Anania_crocealis\tNO\n",
      "Anania_fuscalis\tNO\n",
      "Anania_hortulata\tNO\n",
      "Anarsia_innoxiella\tNO\n",
      "Anorthoa_munda\tNO\n",
      "Antheraea_mylitta\tNO\n",
      "Antheraea_pernyi\tNO\n",
      "Anthocharis_cardamines\tNO\n",
      "Anticlea_derivata\tNO\n",
      "Antitype_chi\tNO\n",
      "Apamea_anceps\tNO\n",
      "Apamea_crenata\tNO\n",
      "Apamea_epomidion\tNO\n",
      "Apamea_monoglypha\tNO\n",
      "Apamea_sordens\tNO\n",
      "Apeira_syringaria\tNO\n",
      "Aplocera_efformata\tNO\n",
      "Apocheima_hispidaria\tNO\n",
      "Apoda_limacodes\tNO\n",
      "Apomyelois_bistriatella\tNO\n",
      "Aporia_crataegi\tNO\n",
      "Aporophyla_lueneburgensis\tNO\n",
      "Aporophyla_nigra\tNO\n",
      "Apotomis_betuletana\tNO\n",
      "Apotomis_capreana\tNO\n",
      "Apotomis_turbidana\tNO\n",
      "Aproaerema_taeniolella\tNO\n",
      "Archips_crataeganus\tNO\n",
      "Archips_xylosteana\tNO\n",
      "Arctia_plantaginis\tYes\n",
      "Argynnis_bischoffii_washingtonia\tNO\n",
      "Argyresthia_albistria\tNO\n",
      "Argyresthia_goedartella\tNO\n",
      "Aricia_agestis\tYes\n",
      "Aricia_artaxerxes\tNO\n",
      "Asteroscopus_sphinx\tNO\n",
      "Atethmia_centrago\tNO\n",
      "Athetis_lepigone\tNO\n",
      "Athrips_mouffetella\tNO\n",
      "Autographa_gamma\tNO\n",
      "Autographa_pulchrina\tNO\n",
      "Baronia_brevicornis\tNO\n",
      "Battus_philenor\tNO\n",
      "Bedellia_somnulentella\tNO\n",
      "Bembecia_ichneumoniformis\tNO\n",
      "Bicyclus_anynana\tYes\n",
      "Biston_betularia\tNO\n",
      "Biston_stratarius\tNO\n",
      "Blastobasis_adustella\tNO\n",
      "Blastobasis_lacticolella\tNO\n",
      "Boloria_euphrosyne\tNO\n",
      "Boloria_selene\tNO\n",
      "Bombyx_mandarina\tNO\n",
      "Bombyx_mori\tYes\n",
      "Brachylomia_viminalis\tNO\n",
      "Brenthis_daphne\tNO\n",
      "Brenthis_hecate\tNO\n",
      "Brenthis_ino\tYes\n",
      "Cabera_pusaria\tNO\n",
      "Calamotropha_paludella\tNO\n",
      "Calephelis_nemesis\tNO\n",
      "Calephelis_virginiensis\tNO\n",
      "Callimorpha_dominula\tNO\n",
      "Calycopis_cecrops\tNO\n",
      "Campaea_margaritaria\tNO\n",
      "Camptogramma_bilineatum\tNO\n",
      "Caradrina_clavipalpis\tNO\n",
      "Caradrina_kadenii\tNO\n",
      "Carcina_quercana\tNO\n",
      "Carpatolechia_fugitivella\tNO\n",
      "Carposina_sasakii\tNO\n",
      "Carterocephalus_palaemon\tNO\n",
      "Catocala_fraxini\tNO\n",
      "Catocala_nupta\tNO\n",
      "Catocala_sponsa\tNO\n",
      "Catoptria_pinella\tNO\n",
      "Celastrina_argiolus\tNO\n",
      "Ceramica_pisi\tNO\n",
      "Cerastis_leucographa\tNO\n",
      "Cerastis_rubricosa\tNO\n",
      "Charanyca_ferruginea\tNO\n",
      "Charaxes_varanes\tNO\n",
      "Chesias_legatella\tNO\n",
      "Chilo_suppressalis\tYes\n",
      "Chloroclysta_siterata\tNO\n",
      "Chloroclystis_v-ata\tNO\n",
      "Choreutis_nemorana\tNO\n",
      "Choristoneura_fumiferana\tYes\n",
      "Chrysodeixis_includens\tYes\n",
      "Chrysoteuchia_culmella\tNO\n",
      "Clepsis_dumicolana\tNO\n",
      "Clostera_curtula\tNO\n",
      "Cnaphalocrocis_medinalis\tNO\n",
      "Coenonympha_glycerion\tNO\n",
      "Coleophora_deauratella\tNO\n",
      "Coleophora_flavipennella\tNO\n",
      "Colias_behrii\tNO\n",
      "Colias_croceus\tYes\n",
      "Colias_eurytheme\tNO\n",
      "Colias_nastes\tNO\n",
      "Colostygia_pectinataria\tNO\n",
      "Conistra_vaccinii\tNO\n",
      "Conogethes_punctiferalis\tNO\n",
      "Conopomorpha_cramerella\tNO\n",
      "Cosmia_pyralina\tNO\n",
      "Cosmia_trapezina\tNO\n",
      "Cosmorhoe_ocellata\tNO\n",
      "Crambus_lathoniellus\tNO\n",
      "Craniophora_ligustri\tNO\n",
      "Crocallis_elinguaria\tNO\n",
      "Cyaniris_semiargus\tNO\n",
      "Cybosia_mesomella\tNO\n",
      "Cyclargus_thomasi\tNO\n",
      "Cyclophora_albipunctata\tNO\n",
      "Cyclophora_punctaria\tNO\n",
      "Cydalima_perspectalis\tNO\n",
      "Cydia_amplana\tNO\n",
      "Cydia_fagiglandana\tNO\n",
      "Cydia_pomonella\tNO\n",
      "Cydia_splendana\tNO\n",
      "Cydia_strobilella\tNO\n",
      "Danaus_chrysippus\tNO\n",
      "Danaus_plexippus\tYes\n",
      "Dasypolia_templi\tNO\n",
      "Deilephila_elpenor\tNO\n",
      "Deilephila_porcellus\tNO\n",
      "Dendrolimus_kikuchii\tYes\n",
      "Dendrolimus_pini\tNO\n",
      "Dendrolimus_punctatus\tNO\n",
      "Diachrysia_chrysitis\tNO\n",
      "Diaphora_mendica\tNO\n",
      "Diarsia_brunnea\tNO\n",
      "Diarsia_dahlii\tNO\n",
      "Diarsia_mendica\tNO\n",
      "Diarsia_rubi\tNO\n",
      "Diatraea_saccharalis\tYes\n",
      "Dicycla_oo\tNO\n",
      "Diloba_caeruleocephala\tNO\n",
      "Dircenna_loreta\tNO\n",
      "Ditula_angustiorana\tNO\n",
      "Drepana_arcuata\tNO\n",
      "Drepana_falcataria\tNO\n",
      "Dryadula_phaetusa\tNO\n",
      "Dryas_iulia_moderata\tNO\n",
      "Drymonia_ruficornis\tNO\n",
      "Dryobota_labecula\tNO\n",
      "Dryobotodes_eremita\tNO\n",
      "Ecliptopera_silaceata\tNO\n",
      "Ectropis_crepuscularia\tNO\n",
      "Ectropis_grisescens\tNO\n",
      "Eilema_caniola\tNO\n",
      "Eilema_depressum\tNO\n",
      "Eilema_griseolum\tNO\n",
      "Eilema_sororculum\tNO\n",
      "Electrophaes_corylata\tNO\n",
      "Elegia_similella\tNO\n",
      "Elophila_nymphaeata\tNO\n",
      "Elymnias_hypermnestra\tNO\n",
      "Emmelina_monodactyla\tNO\n",
      "Endotricha_flammealis\tNO\n",
      "Ennomos_erosarius\tNO\n",
      "Ennomos_fuscantarius\tNO\n",
      "Ennomos_quercinarius\tNO\n",
      "Ephestia_elutella\tNO\n",
      "Epiblema_foenella\tNO\n",
      "Epinotia_bilunana\tNO\n",
      "Epinotia_brunnichana\tNO\n",
      "Epinotia_demarniana\tNO\n",
      "Epinotia_nisella\tNO\n",
      "Epinotia_ramella\tNO\n",
      "Epirrhoe_alternata\tNO\n",
      "Epirrhoe_tristata\tNO\n",
      "Epirrita_christyi\tNO\n",
      "Erannis_defoliaria\tNO\n",
      "Erebia_aethiops\tNO\n",
      "Erebia_ligea\tNO\n",
      "Eremobia_ochroleuca\tNO\n",
      "Erynnis_tages\tNO\n",
      "Esperia_sulphurella\tNO\n",
      "Ethmia_dodecea\tNO\n",
      "Euclidia_mi\tNO\n",
      "Eucosma_cana\tNO\n",
      "Eudemis_profundana\tNO\n",
      "Eudonia_lacustrata\tNO\n",
      "Eudonia_mercurella\tNO\n",
      "Eudonia_truncicolella\tNO\n",
      "Eueides_isabella\tNO\n",
      "Eugnorisma_glareosa\tNO\n",
      "Eulithis_prunata\tNO\n",
      "Eulithis_testata\tNO\n",
      "Eumaeus_atala\tNO\n",
      "Eumeta_japonica\tYes\n",
      "Euphydryas_editha\tYes\n",
      "Eupithecia_abbreviata\tNO\n",
      "Eupithecia_centaureata\tNO\n",
      "Eupithecia_dodoneata\tNO\n",
      "Eupithecia_exiguata\tNO\n",
      "Eupithecia_insigniata\tNO\n",
      "Eupithecia_inturbata\tNO\n",
      "Eupithecia_subfuscata\tNO\n",
      "Eupithecia_subumbrata\tNO\n",
      "Eupithecia_tripunctaria\tNO\n",
      "Eupithecia_vulgata\tNO\n",
      "Euplagia_quadripunctaria\tNO\n",
      "Euplexia_lucipara\tNO\n",
      "Euproctis_similis\tNO\n",
      "Eupsilia_transversa\tNO\n",
      "Eurema_hecabe\tNO\n",
      "Eurois_occulta\tNO\n",
      "Euspilapteryx_auroguttella\tNO\n",
      "Euzophera_pinguis\tNO\n",
      "Fabriciana_adippe\tNO\n",
      "Falcaria_lacertinaria\tNO\n",
      "Fissipunctia_ypsillon\tNO\n",
      "Furcula_furcula\tNO\n",
      "Galleria_mellonella\tNO\n",
      "Gandaritis_pyraliata\tNO\n",
      "Glaucopsyche_alexis\tNO\n",
      "Globia_sparganii\tNO\n",
      "Gortyna_flavago\tNO\n",
      "Grapholita_molesta\tNO\n",
      "Griposia_aprilina\tNO\n",
      "Gymnoscelis_rufifasciata\tNO\n",
      "Habrosyne_pyritoides\tNO\n",
      "Hebomoia_glaucippe\tNO\n",
      "Hecatera_dysodea\tNO\n",
      "Hedya_salicella\tNO\n",
      "Heliconius_charithonia\tNO\n",
      "Heliconius_nattereri\tNO\n",
      "Heliconius_sara\tNO\n",
      "Helicoverpa_armigera\tYes\n",
      "Helicoverpa_assulta\tNO\n",
      "Helicoverpa_zea\tYes\n",
      "Heliothis_peltigera\tNO\n",
      "Heliothis_virescens\tNO\n",
      "Helleia_helle\tNO\n",
      "Hemaris_fuciformis\tNO\n",
      "Hemistola_chrysoprasaria\tNO\n",
      "Hemithea_aestivaria\tNO\n",
      "Heortia_vitessoides\tNO\n",
      "Herminia_tarsipennalis\tNO\n",
      "Hesperia_comma\tNO\n",
      "Hestina_assimilis\tNO\n",
      "Hipparchia_semele\tNO\n",
      "Hofmannophila_pseudospretella\tNO\n",
      "Hoplodrina_ambigua\tNO\n",
      "Hoplodrina_blanda\tNO\n",
      "Horisme_vitalbata\tNO\n",
      "Hydraecia_micacea\tNO\n",
      "Hydriomena_furcata\tNO\n",
      "Hylaea_fasciaria\tNO\n",
      "Hyles_euphorbiae\tNO\n",
      "Hyles_vespertilio\tNO\n",
      "Hypena_proboscidalis\tNO\n",
      "Hypercompe_scribonia\tNO\n",
      "Hyphantria_cunea\tNO\n",
      "Hypolimnas_misippus\tNO\n",
      "Hypomecis_punctinalis\tNO\n",
      "Hyposmocoma_kahamanoa\tYes\n",
      "Hyppa_rectilinea\tNO\n",
      "Hypsopygia_costalis\tNO\n",
      "Idaea_aversata\tNO\n",
      "Idaea_biselata\tNO\n",
      "Idaea_dimidiata\tNO\n",
      "Idaea_straminata\tNO\n",
      "Incurvaria_masculella\tNO\n",
      "Iphiclides_podalirius\tYes\n",
      "Lacanobia_oleracea\tNO\n",
      "Lacanobia_wlatinum\tNO\n",
      "Lampropteryx_suffumata\tNO\n",
      "Laothoe_populi\tNO\n",
      "Lapara_coniferarum\tNO\n",
      "Lasiommata_megera\tNO\n",
      "Laspeyria_flexula\tNO\n",
      "Lathronympha_strigana\tNO\n",
      "Leguminivora_glycinivorella\tYes\n",
      "Leptidea_juvernica\tNO\n",
      "Leptidea_reali\tNO\n",
      "Leptidea_sinapis\tYes\n",
      "Leptophobia_aripa\tNO\n",
      "Lerema_accius\tNO\n",
      "Leucania_comma\tNO\n",
      "Leucinodes_orbonalis\tNO\n",
      "Leucoma_salicis\tNO\n",
      "Ligdia_adustata\tNO\n",
      "Limenitis_camilla\tNO\n",
      "Lithophane_leautieri\tNO\n",
      "Lithophane_ornitopus\tNO\n",
      "Lithophane_semibrunnea\tNO\n",
      "Lithophane_socia\tNO\n",
      "Lithosia_quadra\tNO\n",
      "Lobophora_halterata\tNO\n",
      "Lomographa_bimaculata\tNO\n",
      "Luehdorfia_chinensis\tNO\n",
      "Luffia_ferchaultella\tNO\n",
      "Luperina_nickerlii\tNO\n",
      "Luperina_testacea\tNO\n",
      "Lycaena_phlaeas\tNO\n",
      "Lycia_hirtaria\tNO\n",
      "Lycophotia_porphyrea\tNO\n",
      "Lymantria_dispar\tNO\n",
      "Lymantria_monacha\tNO\n",
      "Lysandra_bellargus\tNO\n",
      "Lysandra_coridon\tNO\n",
      "Macaria_notata\tNO\n",
      "Malacosoma_neustria\tNO\n",
      "Mamestra_brassicae\tNO\n",
      "Mamestra_configurata\tYes\n",
      "Manduca_sexta\tYes\n",
      "Maniola_hyperantus\tYes\n",
      "Maniola_jurtina\tYes\n",
      "Marasmarcha_lunaedactyla\tNO\n",
      "Marasmia_exigua\tNO\n",
      "Mechanitis_mazaeus\tNO\n",
      "Mechanitis_messenoides\tNO\n",
      "Mecyna_flavalis\tNO\n",
      "Meganola_albula\tNO\n",
      "Megathymus_violae\tNO\n",
      "Melanargia_galathea\tNO\n",
      "Melanchra_persicariae\tNO\n",
      "Melinaea_marsaeus_rileyi\tNO\n",
      "Melinaea_menophilus_n._ssp._AW-2005\tNO\n",
      "Melitaea_cinxia\tYes\n",
      "Mellicta_athalia\tNO\n",
      "Mesoligia_furuncula\tNO\n",
      "Metalampra_italica\tNO\n",
      "Micropterix_aruncella\tNO\n",
      "Miltochrista_miniata\tNO\n",
      "Mimas_tiliae\tNO\n",
      "Monopis_laevigella\tNO\n",
      "Musotima_nitidalis\tNO\n",
      "Mythimna_albipuncta\tNO\n",
      "Mythimna_ferrago\tNO\n",
      "Mythimna_impura\tNO\n",
      "Mythimna_lalbum\tNO\n",
      "Mythimna_loreyi\tYes\n",
      "Mythimna_pallens\tNO\n",
      "Mythimna_separata\tYes\n",
      "Mythimna_vitellina\tNO\n",
      "Napeogenes_inachia\tNO\n",
      "Napeogenes_sylphis\tNO\n",
      "Nematopogon_swammerdamellus\tNO\n",
      "Noctua_comes\tNO\n",
      "Noctua_fimbriata\tNO\n",
      "Noctua_janthe\tNO\n",
      "Noctua_janthina\tNO\n",
      "Noctua_pronuba\tNO\n",
      "Nomophila_noctuella\tNO\n",
      "Notocelia_uddmanniana\tNO\n",
      "Notodonta_dromedarius\tNO\n",
      "Notodonta_ziczac\tNO\n",
      "Nudaria_mundana\tNO\n",
      "Nycteola_revayana\tNO\n",
      "Nymphalis_c-album\tNO\n",
      "Nymphalis_io\tYes\n",
      "Nymphalis_polychloros\tNO\n",
      "Nymphalis_urticae\tNO\n",
      "Nymphula_nitidulata\tNO\n",
      "Ochlodes_sylvanus\tNO\n",
      "Ochropacha_duplaris\tNO\n",
      "Ochropleura_leucogaster\tNO\n",
      "Ochropleura_plecta\tNO\n",
      "Oegoconia_quadripuncta\tNO\n",
      "Oeneis_ivallda\tNO\n",
      "Oligia_fasciuncula\tNO\n",
      "Oligia_latruncula\tNO\n",
      "Oligia_strigilis\tNO\n",
      "Omphaloscelis_lunosa\tNO\n",
      "Operophtera_brumata\tNO\n",
      "Opisthograptis_luteolata\tNO\n",
      "Orgyia_antiqua\tNO\n",
      "Ornithoptera_alexandrae\tNO\n",
      "Ornithoptera_priamus\tNO\n",
      "Orthosia_gothica\tNO\n",
      "Orthosia_gracilis\tNO\n",
      "Orthosia_incerta\tNO\n",
      "Ostrinia_furnacalis\tYes\n",
      "Ostrinia_nubilalis\tNO\n",
      "Pammene_aurita\tNO\n",
      "Pammene_fasciana\tNO\n",
      "Pandemis_cinnamomeana\tNO\n",
      "Pandemis_corylana\tNO\n",
      "Pandemis_heparana\tNO\n",
      "Papilio_dardanus_tibullus\tNO\n",
      "Papilio_demoleus\tNO\n",
      "Papilio_elwesi\tNO\n",
      "Papilio_gigon\tNO\n",
      "Papilio_glaucus\tNO\n",
      "Papilio_machaon\tYes\n",
      "Papilio_memnon\tNO\n",
      "Papilio_polytes\tYes\n",
      "Papilio_protenor\tNO\n",
      "Papilio_xuthus\tYes\n",
      "Parapoynx_stratiotata\tNO\n",
      "Pararge_aegeria\tYes\n",
      "Parascotia_fuliginaria\tNO\n",
      "Parnassius_glacialis\tNO\n",
      "Pasiphila_rectangulata\tNO\n",
      "Pectinophora_gossypiella\tYes\n",
      "Pelosia_muscerda\tNO\n",
      "Peribatodes_rhomboidaria\tNO\n",
      "Perizoma_affinitatum\tNO\n",
      "Perizoma_flavofasciatum\tNO\n",
      "Petrophora_chlorosata\tNO\n",
      "Phalera_bucephala\tNO\n",
      "Phengaris_arion\tNO\n",
      "Pheosia_gnoma\tNO\n",
      "Pheosia_tremula\tNO\n",
      "Philaethria_dido\tNO\n",
      "Philereme_vetulata\tNO\n",
      "Phlogophora_meticulosa\tNO\n",
      "Phoebis_sennae\tNO\n",
      "Phragmatobia_fuliginosa\tNO\n",
      "Pieris_brassicae\tYes\n",
      "Pieris_mannii\tNO\n",
      "Pieris_melete\tNO\n",
      "Pieris_napi\tYes\n",
      "Pieris_rapae\tYes\n",
      "Plagodis_dolabraria\tNO\n",
      "Plagodis_pulveraria\tNO\n",
      "Plebejus_argus\tNO\n",
      "Plemyria_rubiginata\tNO\n",
      "Plodia_interpunctella\tYes\n",
      "Plusia_festucae\tNO\n",
      "Plutella_xylostella\tYes\n",
      "Polia_nebulosa\tNO\n",
      "Polymixis_flavicincta\tNO\n",
      "Polymixis_lichenea\tNO\n",
      "Polyommatus_icarus\tNO\n",
      "Polyommatus_iphigenia\tNO\n",
      "Polyploca_ridens\tNO\n",
      "Protodeltote_pygarga\tNO\n",
      "Pseudoips_prasinana\tNO\n",
      "Ptilodon_capucinus\tNO\n",
      "Pyralis_farinalis\tNO\n",
      "Pyrausta_aurata\tNO\n",
      "Pyrausta_nigrata\tNO\n",
      "Pyrgus_malvae\tNO\n",
      "Pyrrhia_umbra\tNO\n",
      "Rhodophaea_formosa\tNO\n",
      "Samia_ricini\tNO\n",
      "Saturnia_japonica\tNO\n",
      "Saturnia_pavonia\tNO\n",
      "Schrankia_costaestrigalis\tNO\n",
      "Scotopteryx_bipunctaria\tNO\n",
      "Scrobipalpa_costella\tNO\n",
      "Selenia_dentaria\tNO\n",
      "Sericinus_montela\tNO\n",
      "Sesamia_nonagrioides\tNO\n",
      "Sesia_apiformis\tNO\n",
      "Sesia_bembeciformis\tNO\n",
      "Shargacucullia_verbasci\tNO\n",
      "Sphinx_pinastri\tNO\n",
      "Spilarctia_lutea\tNO\n",
      "Spilosoma_lubricipeda\tNO\n",
      "Spodoptera_exigua\tYes\n",
      "Spodoptera_frugiperda\tYes\n",
      "Spodoptera_littoralis\tYes\n",
      "Spodoptera_litura\tYes\n",
      "Stenoptilia_bipunctidactyla\tNO\n",
      "Subacronicta_megacephala\tNO\n",
      "Synanthedon_andrenaeformis\tNO\n",
      "Synanthedon_formicaeformis\tNO\n",
      "Synanthedon_myopaeformis\tNO\n",
      "Synanthedon_tipuliformis\tNO\n",
      "Synanthedon_vespiformis\tNO\n",
      "Tachystola_acroxantha\tNO\n",
      "Teinopalpus_imperialis\tNO\n",
      "Teleiodes_luculella\tNO\n",
      "Tethea_ocularis\tNO\n",
      "Tetheella_fluctuosa\tNO\n",
      "Thalpophila_matura\tNO\n",
      "Thera_britannica\tNO\n",
      "Thera_obeliscata\tNO\n",
      "Theretra_japonica\tNO\n",
      "Tholera_decimalis\tNO\n",
      "Thumatha_senex\tNO\n",
      "Thyatira_batis\tNO\n",
      "Thymelicus_acteon\tNO\n",
      "Thymelicus_sylvestris\tNO\n",
      "Tiliacea_aurago\tNO\n",
      "Timandra_comae\tNO\n",
      "Tinea_pellionella\tNO\n",
      "Tinea_semifulvella\tNO\n",
      "Tinea_trinotella\tNO\n",
      "Tortricodes_alternella\tNO\n",
      "Tortrix_viridana\tNO\n",
      "Trichoplusia_ni\tYes\n",
      "Trilocha_varians\tNO\n",
      "Trisateles_emortualis\tNO\n",
      "Troides_aeacus\tNO\n",
      "Troides_oblongomaculatus\tNO\n",
      "Tuta_absoluta\tNO\n",
      "Tyria_jacobaeae\tNO\n",
      "Udea_ferrugalis\tNO\n",
      "Udea_olivalis\tNO\n",
      "Urbanus_simplicius\tNO\n",
      "Vanessa_atalanta\tYes\n",
      "Vanessa_cardui\tYes\n",
      "Vanessa_tameamea\tYes\n",
      "Venusia_cambrica\tNO\n",
      "Watsonalla_binaria\tNO\n",
      "Xanthia_icteritia\tNO\n",
      "Xanthia_togata\tNO\n",
      "Xanthorhoe_designata\tNO\n",
      "Xanthorhoe_spadicearia\tNO\n",
      "Xestia_ashworthii\tNO\n",
      "Xestia_c-nigrum\tNO\n",
      "Xestia_rhomboidea\tNO\n",
      "Xestia_sexstrigata\tNO\n",
      "Xestia_xanthographa\tNO\n",
      "Xylocampa_areola\tNO\n",
      "Yponomeuta_cagnagella\tNO\n",
      "Yponomeuta_malinellus\tNO\n",
      "Yponomeuta_plumbellus\tNO\n",
      "Yponomeuta_sedellus\tNO\n",
      "Ypsolopha_scabrella\tNO\n",
      "Ypsolopha_sequella\tNO\n",
      "Zelleria_hepariella\tNO\n",
      "Zerene_cesonia\tYes\n",
      "Zeuzera_pyrina\tNO\n",
      "Zygaena_filipendulae\tNO\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "list_of_genomes = os.listdir(\"J:/Genomes_2023-12-26\")\n",
    "output = \"Species\\tAnnotation\\n\"\n",
    "for species in list_of_genomes:\n",
    "    list_of_files= os.listdir(f\"J:/Genomes_2023-12-26/{species}\")\n",
    "    annotation = \"NO\"\n",
    "    for file in list_of_files:\n",
    "        if \"genomic.gff\" in file:\n",
    "            annotation = \"Yes\"\n",
    "#             print(species)\n",
    "    \n",
    "    output += f\"{species}\\t{annotation}\\n\"\n",
    "\n",
    "print(output)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7a56c32",
   "metadata": {},
   "source": [
    "# Parse fasta query file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e9e74324",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon 1, Length = 89\n",
      "Name:NC_051358.1:c12522661-12522573\n",
      "\n",
      "\n",
      "ATG GAT AAC CTT GAA \n",
      "\n",
      " M   D   N   L   E  \n",
      "\n",
      "AAC TCT GAG AAC ACT \n",
      "\n",
      " N   S   E   N   T  \n",
      "\n",
      "GAC AAA ATA TCA GAT \n",
      "\n",
      " D   K   I   S   D  \n",
      "\n",
      "TCG GCA TAC TCA AAC \n",
      "\n",
      " S   A   Y   S   N  \n",
      "\n",
      "AGC TGC AGT TAC AGT \n",
      "\n",
      " S   C   S   Y   S  \n",
      "\n",
      "CAA TCA AGA AGA AG \n",
      "\n",
      " Q   S   R   R     \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = MDNLENSENTDKISDSAYSNSCSYSQSRR\n",
      "translated_length = 29\n",
      "\n",
      "Exon 2, Length = 84\n",
      "Name:NC_051358.1:c12521246-12521163\n",
      "\n",
      "\n",
      "T CAC AGC TCA AAA TCT \n",
      "\n",
      "   H   S   S   K   S  \n",
      "\n",
      "ACT CAT TCC GGG AGT \n",
      "\n",
      " T   H   S   G   S  \n",
      "\n",
      "TAT TCT AGC GGC AGT \n",
      "\n",
      " Y   S   S   G   S  \n",
      "\n",
      "AGC GGA TAC GGA GGT \n",
      "\n",
      " S   G   Y   G   G  \n",
      "\n",
      "CGA CAT GTT ACT GGC \n",
      "\n",
      " R   H   V   T   G  \n",
      "\n",
      "ACT AAC AG \n",
      "\n",
      " T   N     \n",
      "\n",
      "1\n",
      "\n",
      "translated = HSSKSTHSGSYSSGSSGYGGRHVTGTN\n",
      "translated_length = 27\n",
      "\n",
      "Exon 3, Length = 155\n",
      "Name:NC_051358.1:c12520999-12520845\n",
      "\n",
      "\n",
      "T AAT AAG TTG GCT CAG \n",
      "\n",
      "   N   K   L   A   Q  \n",
      "\n",
      "TCA CCA AAT AAA GAT \n",
      "\n",
      " S   P   N   K   D  \n",
      "\n",
      "AAA GAG TTA AAG AAA \n",
      "\n",
      " K   E   L   K   K  \n",
      "\n",
      "AAG AAA CAA TCA CAA \n",
      "\n",
      " K   K   Q   S   Q  \n",
      "\n",
      "ACA GAG CTA ACG ACT \n",
      "\n",
      " T   E   L   T   T  \n",
      "\n",
      "ACC AAA GAA CAA AAT \n",
      "\n",
      " T   K   E   Q   N  \n",
      "\n",
      "GAA GGA GAA AGC GGA \n",
      "\n",
      " E   G   E   S   G  \n",
      "\n",
      "CCT CAA GAA TCT GTG \n",
      "\n",
      " P   Q   E   S   V  \n",
      "\n",
      "AAC CTG GAA ACT CTA \n",
      "\n",
      " N   L   E   T   L  \n",
      "\n",
      "GAG GAG GAG AAG CAT \n",
      "\n",
      " E   E   E   K   H  \n",
      "\n",
      "GAT G \n",
      "\n",
      " D     \n",
      "\n",
      "2\n",
      "\n",
      "translated = NKLAQSPNKDKELKKKKQSQTELTTTKEQNEGESGPQESVNLETLEEEKHD\n",
      "translated_length = 51\n",
      "\n",
      "Exon 4, Length = 95\n",
      "Name:NC_051358.1:c12514423-12514329\n",
      "\n",
      "\n",
      "TT ACG AGC GCA CCC TCT \n",
      "\n",
      "    T   S   A   P   S  \n",
      "\n",
      "CCA GCG CAT GTA CAG \n",
      "\n",
      " P   A   H   V   Q  \n",
      "\n",
      "GTC GAG AAG GGA CCT \n",
      "\n",
      " V   E   K   G   P  \n",
      "\n",
      "GAG AAT ATG GAT ATA \n",
      "\n",
      " E   N   M   D   I  \n",
      "\n",
      "GCT TCA ACA GAC ACT \n",
      "\n",
      " A   S   T   D   T  \n",
      "\n",
      "GAA GGC GAC AAG GTA \n",
      "\n",
      " E   G   D   K   V  \n",
      "\n",
      "GAG \n",
      "\n",
      " E  \n",
      "\n",
      "0\n",
      "\n",
      "translated = TSAPSPAHVQVEKGPENMDIASTDTEGDKVE\n",
      "translated_length = 31\n",
      "\n",
      "Exon 5, Length = 78\n",
      "Name:NC_051358.1:c12513351-12513274\n",
      "\n",
      "\n",
      "GTT TTA TCG TGT GAC \n",
      "\n",
      " V   L   S   C   D  \n",
      "\n",
      "CTG CAA CCG CAG AAC \n",
      "\n",
      " L   Q   P   Q   N  \n",
      "\n",
      "TCT ACT GTC GAT GTA \n",
      "\n",
      " S   T   V   D   V  \n",
      "\n",
      "GTT ACG AGT CGT TCG \n",
      "\n",
      " V   T   S   R   S  \n",
      "\n",
      "AAT CTA ATC TGT TAC \n",
      "\n",
      " N   L   I   C   Y  \n",
      "\n",
      "ACT \n",
      "\n",
      " T  \n",
      "\n",
      "0\n",
      "\n",
      "translated = VLSCDLQPQNSTVDVVTSRSNLICYT\n",
      "translated_length = 26\n",
      "\n",
      "Exon 6, Length = 132\n",
      "Name:NC_051358.1:c12512245-12512114\n",
      "\n",
      "\n",
      "GAT GGA TTT TCA TGC \n",
      "\n",
      " D   G   F   S   C  \n",
      "\n",
      "GTT ATA TCG ATG CAC \n",
      "\n",
      " V   I   S   M   H  \n",
      "\n",
      "GAT GGA ATC GTA ATG \n",
      "\n",
      " D   G   I   V   M  \n",
      "\n",
      "TAC ACG ACA TCA TCG \n",
      "\n",
      " Y   T   T   S   S  \n",
      "\n",
      "CTG ACG GCT ATA CTC \n",
      "\n",
      " L   T   A   I   L  \n",
      "\n",
      "GGT TTC CCG AAG GAC \n",
      "\n",
      " G   F   P   K   D  \n",
      "\n",
      "ATG TGG ATC GGG AGG \n",
      "\n",
      " M   W   I   G   R  \n",
      "\n",
      "TCA TTT ATC GAT TTC \n",
      "\n",
      " S   F   I   D   F  \n",
      "\n",
      "GTT CAT CCA AGG \n",
      "\n",
      " V   H   P   R  \n",
      "\n",
      "0\n",
      "\n",
      "translated = DGFSCVISMHDGIVMYTTSSLTAILGFPKDMWIGRSFIDFVHPR\n",
      "translated_length = 44\n",
      "\n",
      "Exon 7, Length = 76\n",
      "Name:NC_051358.1:c12511968-12511893\n",
      "\n",
      "\n",
      "GAC CGT AGC ACG TTC \n",
      "\n",
      " D   R   S   T   F  \n",
      "\n",
      "GCG TCA CAG ATA ACC \n",
      "\n",
      " A   S   Q   I   T  \n",
      "\n",
      "AGT GGT TTG GCC GTC \n",
      "\n",
      " S   G   L   A   V  \n",
      "\n",
      "CCG AAA ACT GCA AAC \n",
      "\n",
      " P   K   T   A   N  \n",
      "\n",
      "GGT ACG CAA GAA AAA \n",
      "\n",
      " G   T   Q   E   K  \n",
      "\n",
      "G \n",
      "\n",
      "   \n",
      "\n",
      "2\n",
      "\n",
      "translated = DRSTFASQITSGLAVPKTANGTQEK\n",
      "translated_length = 25\n",
      "\n",
      "Exon 8, Length = 204\n",
      "Name:NC_051358.1:c12510243-12510040\n",
      "\n",
      "\n",
      "CT CAA TCG CCG GGT AAT \n",
      "\n",
      "    Q   S   P   G   N  \n",
      "\n",
      "TCT GGA TCG ACT ATG \n",
      "\n",
      " S   G   S   T   M  \n",
      "\n",
      "GTG TGT CGG ATA AGA \n",
      "\n",
      " V   C   R   I   R  \n",
      "\n",
      "CGT TAC AGA GGT CTG \n",
      "\n",
      " R   Y   R   G   L  \n",
      "\n",
      "AGC ACG GGG TTC GGA \n",
      "\n",
      " S   T   G   F   G  \n",
      "\n",
      "GTT AAG GAA AGA GTG \n",
      "\n",
      " V   K   E   R   V  \n",
      "\n",
      "GTC ACG TTC ATG CCG \n",
      "\n",
      " V   T   F   M   P  \n",
      "\n",
      "TTC CTT TTG AAA TTC \n",
      "\n",
      " F   L   L   K   F  \n",
      "\n",
      "ACG TTC AAG AAC ATA \n",
      "\n",
      " T   F   K   N   I  \n",
      "\n",
      "AGC GAT GAA GAG GGA \n",
      "\n",
      " S   D   E   E   G  \n",
      "\n",
      "AAT GTC ATC TAC CTC \n",
      "\n",
      " N   V   I   Y   L  \n",
      "\n",
      "GTC ATA CAG GCC ACA \n",
      "\n",
      " V   I   Q   A   T  \n",
      "\n",
      "CCA TTT TTT TCA GCG \n",
      "\n",
      " P   F   F   S   A  \n",
      "\n",
      "TTC AAA A \n",
      "\n",
      " F   K     \n",
      "\n",
      "2\n",
      "\n",
      "translated = QSPGNSGSTMVCRIRRYRGLSTGFGVKERVVTFMPFLLKFTFKNISDEEGNVIYLVIQATPFFSAFK\n",
      "translated_length = 67\n",
      "\n",
      "Exon 9, Length = 229\n",
      "Name:NC_051358.1:c12508701-12508473\n",
      "\n",
      "\n",
      "CG TCG TTT GAA ATA TTA \n",
      "\n",
      "    S   F   E   I   L  \n",
      "\n",
      "CCG AAA GTT AAT CCT \n",
      "\n",
      " P   K   V   N   P  \n",
      "\n",
      "TTC GTG ATG CGT CAC \n",
      "\n",
      " F   V   M   R   H  \n",
      "\n",
      "TCC GCG AAT GGT AAT \n",
      "\n",
      " S   A   N   G   N  \n",
      "\n",
      "CTC GAG TAC TTA GAC \n",
      "\n",
      " L   E   Y   L   D  \n",
      "\n",
      "CCA GAA TCT GTA CCT \n",
      "\n",
      " P   E   S   V   P  \n",
      "\n",
      "TAT CTC GGA TAT TTA \n",
      "\n",
      " Y   L   G   Y   L  \n",
      "\n",
      "CCT CAA GAT GTC CAA \n",
      "\n",
      " P   Q   D   V   Q  \n",
      "\n",
      "GAC AAG GAC GCA TTG \n",
      "\n",
      " D   K   D   A   L  \n",
      "\n",
      "CAA CTC TAT CAT CCG \n",
      "\n",
      " Q   L   Y   H   P  \n",
      "\n",
      "GAA GAT TTG GAA TAC \n",
      "\n",
      " E   D   L   E   Y  \n",
      "\n",
      "CTC CAG CAA GTA TAC \n",
      "\n",
      " L   Q   Q   V   Y  \n",
      "\n",
      "GAG GTA ATC GTT AAG \n",
      "\n",
      " E   V   I   V   K  \n",
      "\n",
      "GAT GGT GGC ATG CCG \n",
      "\n",
      " D   G   G   M   P  \n",
      "\n",
      "CGT TCC AAA ACT TAC \n",
      "\n",
      " R   S   K   T   Y  \n",
      "\n",
      "AG \n",
      "\n",
      "   \n",
      "\n",
      "1\n",
      "\n",
      "translated = SFEILPKVNPFVMRHSANGNLEYLDPESVPYLGYLPQDVQDKDALQLYHPEDLEYLQQVYEVIVKDGGMPRSKTY\n",
      "translated_length = 75\n",
      "\n",
      "Exon 10, Length = 112\n",
      "Name:NC_051358.1:c12507855-12507744\n",
      "\n",
      "\n",
      "A ATG ATG ACA CAA AAC \n",
      "\n",
      "   M   M   T   Q   N  \n",
      "\n",
      "GGT GAT TAC ATA AAA \n",
      "\n",
      " G   D   Y   I   K  \n",
      "\n",
      "ATC GAA ACC GAA TGG \n",
      "\n",
      " I   E   T   E   W  \n",
      "\n",
      "TCT TCG TTT ATT AAT \n",
      "\n",
      " S   S   F   I   N  \n",
      "\n",
      "CCT TGG TCG AAG AAG \n",
      "\n",
      " P   W   S   K   K  \n",
      "\n",
      "CTG GAA TTC GTC ATC \n",
      "\n",
      " L   E   F   V   I  \n",
      "\n",
      "GGG AAA CAT TAC ATC \n",
      "\n",
      " G   K   H   Y   I  \n",
      "\n",
      "ATT GAA \n",
      "\n",
      " I   E  \n",
      "\n",
      "0\n",
      "\n",
      "translated = MMTQNGDYIKIETEWSSFINPWSKKLEFVIGKHYIIE\n",
      "translated_length = 37\n",
      "\n",
      "Exon 11, Length = 120\n",
      "Name:NC_051358.1:c12506233-12506114\n",
      "\n",
      "\n",
      "GGC CCC GAG AAT CCA \n",
      "\n",
      " G   P   E   N   P  \n",
      "\n",
      "GAT GTA TTC CAG TCC \n",
      "\n",
      " D   V   F   Q   S  \n",
      "\n",
      "CAA GAT CCG GAG AAA \n",
      "\n",
      " Q   D   P   E   K  \n",
      "\n",
      "CAT ACA AAA TTA TGC \n",
      "\n",
      " H   T   K   L   C  \n",
      "\n",
      "GAC GAA CAA ATA AAA \n",
      "\n",
      " D   E   Q   I   K  \n",
      "\n",
      "AAG TCC ATG GTA TTT \n",
      "\n",
      " K   S   M   V   F  \n",
      "\n",
      "CGC GAG AAT ATC GTT \n",
      "\n",
      " R   E   N   I   V  \n",
      "\n",
      "AAA CTT ATG AAT GAG \n",
      "\n",
      " K   L   M   N   E  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = GPENPDVFQSQDPEKHTKLCDEQIKKSMVFRENIVKLMNE\n",
      "translated_length = 40\n",
      "\n",
      "Exon 12, Length = 150\n",
      "Name:NC_051358.1:c12505865-12505716\n",
      "\n",
      "\n",
      "GCT CTA ACG AAA CCA \n",
      "\n",
      " A   L   T   K   P  \n",
      "\n",
      "GCG GAA GTT GCC AAA \n",
      "\n",
      " A   E   V   A   K  \n",
      "\n",
      "CAG CAA ATG AGC AAG \n",
      "\n",
      " Q   Q   M   S   K  \n",
      "\n",
      "CGT TGC CAG GAC TTA \n",
      "\n",
      " R   C   Q   D   L  \n",
      "\n",
      "GCG TCG TTC ATG GAG \n",
      "\n",
      " A   S   F   M   E  \n",
      "\n",
      "AGC CTC ATG GAG GAA \n",
      "\n",
      " S   L   M   E   E  \n",
      "\n",
      "CCA CCG AAA AAT GAC \n",
      "\n",
      " P   P   K   N   D  \n",
      "\n",
      "GAG GAA TTA CGC CTA \n",
      "\n",
      " E   E   L   R   L  \n",
      "\n",
      "GAA ATA CAG GAT CCG \n",
      "\n",
      " E   I   Q   D   P  \n",
      "\n",
      "GAC CAC AGT TAT TAT \n",
      "\n",
      " D   H   S   Y   Y  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = ALTKPAEVAKQQMSKRCQDLASFMESLMEEPPKNDEELRLEIQDPDHSYY\n",
      "translated_length = 50\n",
      "\n",
      "Exon 13, Length = 119\n",
      "Name:NC_051358.1:c12503906-12503788\n",
      "\n",
      "\n",
      "GAG CGT GAT TCG GTA \n",
      "\n",
      " E   R   D   S   V  \n",
      "\n",
      "ATG CTC GGC GGG ATA \n",
      "\n",
      " M   L   G   G   I  \n",
      "\n",
      "TCT CCC CAT CAC GAC \n",
      "\n",
      " S   P   H   H   D  \n",
      "\n",
      "TAC AAT GAC AGC AAA \n",
      "\n",
      " Y   N   D   S   K  \n",
      "\n",
      "TCA AGC ACA GAA ACA \n",
      "\n",
      " S   S   T   E   T  \n",
      "\n",
      "CCG CTG AGT TAT AAT \n",
      "\n",
      " P   L   S   Y   N  \n",
      "\n",
      "CAA CTA AAC TAT AAC \n",
      "\n",
      " Q   L   N   Y   N  \n",
      "\n",
      "GAG ACA TTA CAA AG \n",
      "\n",
      " E   T   L   Q     \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = ERDSVMLGGISPHHDYNDSKSSTETPLSYNQLNYNETLQ\n",
      "translated_length = 39\n",
      "\n",
      "Exon 14, Length = 90\n",
      "Name:NC_051358.1:c12503481-12503392\n",
      "\n",
      "\n",
      "A TAT TTC GAC AGT CAT \n",
      "\n",
      "   Y   F   D   S   H  \n",
      "\n",
      "GAA CCC TTT TCG AGC \n",
      "\n",
      " E   P   F   S   S  \n",
      "\n",
      "GAG AAC TTC AAT CCC \n",
      "\n",
      " E   N   F   N   P  \n",
      "\n",
      "TTG ATT GTG GCA AAG \n",
      "\n",
      " L   I   V   A   K  \n",
      "\n",
      "GAG CCA AAT AAT TTA \n",
      "\n",
      " E   P   N   N   L  \n",
      "\n",
      "AAA TCG ATA CTA AG \n",
      "\n",
      " K   S   I   L     \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = YFDSHEPFSSENFNPLIVAKEPNNLKSIL\n",
      "translated_length = 29\n",
      "\n",
      "Exon 15, Length = 153\n",
      "Name:NC_051358.1:c12502301-12502149\n",
      "\n",
      "\n",
      "T ACT TGT TTA TCA CCA \n",
      "\n",
      "   T   C   L   S   P  \n",
      "\n",
      "ATG GCT CAG AAC TCT \n",
      "\n",
      " M   A   Q   N   S  \n",
      "\n",
      "GGC GAT TCG ACC GAG \n",
      "\n",
      " G   D   S   T   E  \n",
      "\n",
      "TTG AAT AGT TCG TCC \n",
      "\n",
      " L   N   S   S   S  \n",
      "\n",
      "GAT TCT AGT AGT ATG \n",
      "\n",
      " D   S   S   S   M  \n",
      "\n",
      "GCC CTG GGA GGA GTA \n",
      "\n",
      " A   L   G   G   V  \n",
      "\n",
      "AGT CCT GTG GGT ATA \n",
      "\n",
      " S   P   V   G   I  \n",
      "\n",
      "TTA GGC GAC TAC CAC \n",
      "\n",
      " L   G   D   Y   H  \n",
      "\n",
      "CAC GTT CGG CTC ACC \n",
      "\n",
      " H   V   R   L   T  \n",
      "\n",
      "GAG TTC CTA CTA ACC \n",
      "\n",
      " E   F   L   L   T  \n",
      "\n",
      "AA \n",
      "\n",
      "   \n",
      "\n",
      "1\n",
      "\n",
      "translated = TCLSPMAQNSGDSTELNSSSDSSSMALGGVSPVGILGDYHHVRLTEFLLT\n",
      "translated_length = 50\n",
      "\n",
      "Exon 16, Length = 190\n",
      "Name:NC_051358.1:c12501492-12501303\n",
      "\n",
      "\n",
      "A CAC AAT GAT GAA ATG \n",
      "\n",
      "   H   N   D   E   M  \n",
      "\n",
      "GAA AAG GAA CTG ATA \n",
      "\n",
      " E   K   E   L   I  \n",
      "\n",
      "AAT ATG CAT AGA GAG \n",
      "\n",
      " N   M   H   R   E  \n",
      "\n",
      "AGC CGG TCC AAT AGT \n",
      "\n",
      " S   R   S   N   S  \n",
      "\n",
      "AAA GGA GAA AGA GAT \n",
      "\n",
      " K   G   E   R   D  \n",
      "\n",
      "AAG ACA TCT AAC GAA \n",
      "\n",
      " K   T   S   N   E  \n",
      "\n",
      "ACG AGA CAG AAG AAA \n",
      "\n",
      " T   R   Q   K   K  \n",
      "\n",
      "AAG GAA CAT CTA GCG \n",
      "\n",
      " K   E   H   L   A  \n",
      "\n",
      "AGG TGT AAA GCT TCG \n",
      "\n",
      " R   C   K   A   S  \n",
      "\n",
      "TTT CAT CCA ACC GCG \n",
      "\n",
      " F   H   P   T   A  \n",
      "\n",
      "ACC AGC ACC ACA CCT \n",
      "\n",
      " T   S   T   T   P  \n",
      "\n",
      "GTG GAC AAA GAA GTG \n",
      "\n",
      " V   D   K   E   V  \n",
      "\n",
      "TAT AAA AAA \n",
      "\n",
      " Y   K   K  \n",
      "\n",
      "0\n",
      "\n",
      "translated = HNDEMEKELINMHRESRSNSKGERDKTSNETRQKKKEHLARCKASFHPTATSTTPVDKEVYKK\n",
      "translated_length = 63\n",
      "\n",
      "Exon 17, Length = 148\n",
      "Name:NC_051358.1:c12498481-12498334\n",
      "\n",
      "\n",
      "CCT CAC GGT GTC AAA \n",
      "\n",
      " P   H   G   V   K  \n",
      "\n",
      "CGT GCT TCG AAA CAT \n",
      "\n",
      " R   A   S   K   H  \n",
      "\n",
      "ATC GAA ACG GAA ACT \n",
      "\n",
      " I   E   T   E   T  \n",
      "\n",
      "GTA TCG CAC AAA TAC \n",
      "\n",
      " V   S   H   K   Y  \n",
      "\n",
      "CAT TGT CCA TCG CCC \n",
      "\n",
      " H   C   P   S   P  \n",
      "\n",
      "AGA GCG TCT AGA CCT \n",
      "\n",
      " R   A   S   R   P  \n",
      "\n",
      "AGG CAA ACC ACC AGT \n",
      "\n",
      " R   Q   T   T   S  \n",
      "\n",
      "GCG GCG CCT GTA CAA \n",
      "\n",
      " A   A   P   V   Q  \n",
      "\n",
      "ACT TCG TCC ACT TTG \n",
      "\n",
      " T   S   S   T   L  \n",
      "\n",
      "ACT ACT TCT GTT G \n",
      "\n",
      " T   T   S   V     \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = PHGVKRASKHIETETVSHKYHCPSPRASRPRQTTSAAPVQTSSTLTTSV\n",
      "translated_length = 49\n",
      "\n",
      "Exon 18, Length = 114\n",
      "Name:NC_051358.1:c12497509-12497396\n",
      "\n",
      "\n",
      "CC TCG ACT ACC TGG CCA \n",
      "\n",
      "    S   T   T   W   P  \n",
      "\n",
      "CCC AGC ACC AAT GCT \n",
      "\n",
      " P   S   T   N   A  \n",
      "\n",
      "GCT GGA AAT ATG AAC \n",
      "\n",
      " A   G   N   M   N  \n",
      "\n",
      "ACA TTC ATA CTA GGA \n",
      "\n",
      " T   F   I   L   G  \n",
      "\n",
      "GTC GGA ATG GCT CCA \n",
      "\n",
      " V   G   M   A   P  \n",
      "\n",
      "CAG ATG TCC CTT ATA \n",
      "\n",
      " Q   M   S   L   I  \n",
      "\n",
      "AGT CCG GTT CCG CCA \n",
      "\n",
      " S   P   V   P   P  \n",
      "\n",
      "ATG GCA G \n",
      "\n",
      " M   A     \n",
      "\n",
      "2\n",
      "\n",
      "translated = STTWPPSTNAAGNMNTFILGVGMAPQMSLISPVPPMA\n",
      "translated_length = 37\n",
      "\n",
      "Exon 19, Length = 117\n",
      "Name:NC_051358.1:c12496602-12496486\n",
      "\n",
      "\n",
      "GA ATG CTT CCG TTG TAT \n",
      "\n",
      "    M   L   P   L   Y  \n",
      "\n",
      "TAC ACT CCG ATG GCC \n",
      "\n",
      " Y   T   P   M   A  \n",
      "\n",
      "ACA ATG GCT CCC GTG \n",
      "\n",
      " T   M   A   P   V  \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCT TCC ACA TCG GAG \n",
      "\n",
      " P   S   T   S   E  \n",
      "\n",
      "GCC GCG AAC CAT CAA \n",
      "\n",
      " A   A   N   H   Q  \n",
      "\n",
      "AAT CTT CAT AAT AAT \n",
      "\n",
      " N   L   H   N   N  \n",
      "\n",
      "CCA CAG CAG TAT GCT \n",
      "\n",
      " P   Q   Q   Y   A  \n",
      "\n",
      "CCA CCT CCA A \n",
      "\n",
      " P   P   P     \n",
      "\n",
      "2\n",
      "\n",
      "translated = MLPLYYTPMATMAPVPSTSEAANHQNLHNNPQQYAPPP\n",
      "translated_length = 38\n",
      "\n",
      "Exon 20, Length = 185\n",
      "Name:NC_051358.1:c12494243-12494059\n",
      "\n",
      "\n",
      "TG CAA TGT GTA TTG TAC \n",
      "\n",
      "    Q   C   V   L   Y  \n",
      "\n",
      "GGT CAG CCT ATA TAC \n",
      "\n",
      " G   Q   P   I   Y  \n",
      "\n",
      "GGT CAG CCG ATG TAC \n",
      "\n",
      " G   Q   P   M   Y  \n",
      "\n",
      "AGT TCG CCG TTC GTG \n",
      "\n",
      " S   S   P   F   V  \n",
      "\n",
      "TAC TCA CCA ATG AAC \n",
      "\n",
      " Y   S   P   M   N  \n",
      "\n",
      "CCG CAC ACT AAC TAT \n",
      "\n",
      " P   H   T   N   Y  \n",
      "\n",
      "CCT ATG CAA CAG ACC \n",
      "\n",
      " P   M   Q   Q   T  \n",
      "\n",
      "ACG CCG CAA CCG AAT \n",
      "\n",
      " T   P   Q   P   N  \n",
      "\n",
      "GCT CAG TTC ACA CCC \n",
      "\n",
      " A   Q   F   T   P  \n",
      "\n",
      "ACA AAC ACA ATG AAT \n",
      "\n",
      " T   N   T   M   N  \n",
      "\n",
      "CCC TTG TGT TTG GCC \n",
      "\n",
      " P   L   C   L   A  \n",
      "\n",
      "AAT AGC AAC TAT GAA \n",
      "\n",
      " N   S   N   Y   E  \n",
      "\n",
      "GAG \n",
      "\n",
      " E  \n",
      "\n",
      "0\n",
      "\n",
      "translated = QCVLYGQPIYGQPMYSSPFVYSPMNPHTNYPMQQTTPQPNAQFTPTNTMNPLCLANSNYEE\n",
      "translated_length = 61\n",
      "\n",
      "Exon 21, Length = 201\n",
      "Name:NC_051358.1:c12483156-12482956\n",
      "\n",
      "\n",
      "AGA ACA GAA GCA GGG \n",
      "\n",
      " R   T   E   A   G  \n",
      "\n",
      "CAT GCC ACA ACG CCA \n",
      "\n",
      " H   A   T   T   P  \n",
      "\n",
      "TCT AAA GAC ATT AAG \n",
      "\n",
      " S   K   D   I   K  \n",
      "\n",
      "AAA ACG AGC AGG TTG \n",
      "\n",
      " K   T   S   R   L  \n",
      "\n",
      "GGT AAC ACC GAG GAG \n",
      "\n",
      " G   N   T   E   E  \n",
      "\n",
      "ATT GTA GAC AAA ACT \n",
      "\n",
      " I   V   D   K   T  \n",
      "\n",
      "GAC GGA GAA TCT AGC \n",
      "\n",
      " D   G   E   S   S  \n",
      "\n",
      "TAT TCC TCG TTT TAC \n",
      "\n",
      " Y   S   S   F   Y  \n",
      "\n",
      "TCG TCG TTT TTC AAG \n",
      "\n",
      " S   S   F   F   K  \n",
      "\n",
      "ACG GAA TCT GGC AGC \n",
      "\n",
      " T   E   S   G   S  \n",
      "\n",
      "GCG GAA GAG AGT GGC \n",
      "\n",
      " A   E   E   S   G  \n",
      "\n",
      "GAA AGC AAG AAA AGG \n",
      "\n",
      " E   S   K   K   R  \n",
      "\n",
      "GAC ACA AGA AAT CAC \n",
      "\n",
      " D   T   R   N   H  \n",
      "\n",
      "GTC AAA \n",
      "\n",
      " V   K  \n",
      "\n",
      "0\n",
      "\n",
      "translated = RTEAGHATTPSKDIKKTSRLGNTEEIVDKTDGESSYSSFYSSFFKTESGSAEESGESKKRDTRNHVK\n",
      "translated_length = 67\n",
      "\n",
      "Exon 22, Length = 135\n",
      "Name:NC_051358.1:c12482268-12482134\n",
      "\n",
      "\n",
      "TTT TGG GAT AAA AAC \n",
      "\n",
      " F   W   D   K   N  \n",
      "\n",
      "TAT CGT GGC GAC GAT \n",
      "\n",
      " Y   R   G   D   D  \n",
      "\n",
      "GGG ATG GCG ATG CCT \n",
      "\n",
      " G   M   A   M   P  \n",
      "\n",
      "GGT TCA ACG GAG GTC \n",
      "\n",
      " G   S   T   E   V  \n",
      "\n",
      "CGC CAA ACT TTT TCC \n",
      "\n",
      " R   Q   T   F   S  \n",
      "\n",
      "GAT CAC CAT ATT CCA \n",
      "\n",
      " D   H   H   I   P  \n",
      "\n",
      "GCG AAG GTA CCC AGA \n",
      "\n",
      " A   K   V   P   R  \n",
      "\n",
      "CGA AAC ATT GAA CCA \n",
      "\n",
      " R   N   I   E   P  \n",
      "\n",
      "CCA TGG ATG GAG CAG \n",
      "\n",
      " P   W   M   E   Q  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = FWDKNYRGDDGMAMPGSTEVRQTFSDHHIPAKVPRRNIEPPWMEQ\n",
      "translated_length = 45\n",
      "\n",
      "Exon 23, Length = 102\n",
      "Name:NC_051358.1:c12481645-12481544\n",
      "\n",
      "\n",
      "GTA TGC CTG ACG TCA \n",
      "\n",
      " V   C   L   T   S  \n",
      "\n",
      "GAG CTC GTC TAC AAT \n",
      "\n",
      " E   L   V   Y   N  \n",
      "\n",
      "TAC CAG ATA AGG ACA \n",
      "\n",
      " Y   Q   I   R   T  \n",
      "\n",
      "AAA ACA ATG GAG GAA \n",
      "\n",
      " K   T   M   E   E  \n",
      "\n",
      "GTA CTT TCA TCT GAC \n",
      "\n",
      " V   L   S   S   D  \n",
      "\n",
      "AAG CAG AAG ATT AGT \n",
      "\n",
      " K   Q   K   I   S  \n",
      "\n",
      "AAC CTT GAG CAG \n",
      "\n",
      " N   L   E   Q  \n",
      "\n",
      "0\n",
      "\n",
      "translated = VCLTSELVYNYQIRTKTMEEVLSSDKQKISNLEQ\n",
      "translated_length = 34\n",
      "\n",
      "Exon 24, Length = 249\n",
      "Name:NC_051358.1:c12478465-12478217\n",
      "\n",
      "\n",
      "CCG TCA CTA GTG AAC \n",
      "\n",
      " P   S   L   V   N  \n",
      "\n",
      "GAG CAA CTT GGT CAG \n",
      "\n",
      " E   Q   L   G   Q  \n",
      "\n",
      "CTG TAC CTG GAC CTT \n",
      "\n",
      " L   Y   L   D   L  \n",
      "\n",
      "CAG TTG GAA GGT GTC \n",
      "\n",
      " Q   L   E   G   V  \n",
      "\n",
      "GCC GCG CGA CTC ACC \n",
      "\n",
      " A   A   R   L   T  \n",
      "\n",
      "CTC GAG GAG GGT ATC \n",
      "\n",
      " L   E   E   G   I  \n",
      "\n",
      "ACC AGT TCC AGC AGT \n",
      "\n",
      " T   S   S   S   S  \n",
      "\n",
      "TCC GGT GAA GAG ACA \n",
      "\n",
      " S   G   E   E   T  \n",
      "\n",
      "ACA ACC TCG TCC AAG \n",
      "\n",
      " T   T   S   S   K  \n",
      "\n",
      "AGG AAA CGG GAA TAC \n",
      "\n",
      " R   K   R   E   Y  \n",
      "\n",
      "AGC AAA TTG GTT ATG \n",
      "\n",
      " S   K   L   V   M  \n",
      "\n",
      "ATA TAT GAA GAG AAC \n",
      "\n",
      " I   Y   E   E   N  \n",
      "\n",
      "GCT CCC TTC CCA CCT \n",
      "\n",
      " A   P   F   P   P  \n",
      "\n",
      "CCT TCT TCT GAA GAG \n",
      "\n",
      " P   S   S   E   E  \n",
      "\n",
      "ATA CAA GAG ATA CCG \n",
      "\n",
      " I   Q   E   I   P  \n",
      "\n",
      "ACA GCC AAT CAG TCC \n",
      "\n",
      " T   A   N   Q   S  \n",
      "\n",
      "GAG TCC TAG \n",
      "\n",
      " E   S   *  \n",
      "\n",
      "0\n",
      "\n",
      "translated = PSLVNEQLGQLYLDLQLEGVAARLTLEEGITSSSSSGEETTTSSKRKREYSKLVMIYEENAPFPPPSSEEIQEIPTANQSES*\n",
      "translated_length = 83\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "def print_codon_translated(exon_sequence,exon_counter,exon_overhang):\n",
    "    counter = 0\n",
    "    codon_counter = 0\n",
    "    protein_print_overhang = 0\n",
    "    if exon_overhang != 0:\n",
    "        print(f\"{exon_sequence[counter : counter + exon_overhang]} \", end = '')\n",
    "        protein_print_overhang = 1\n",
    "        \n",
    "    counter = counter + exon_overhang\n",
    "    start_position = counter\n",
    "    while counter < len(exon_sequence):\n",
    "        print(f\"{exon_sequence[counter : counter + 3]} \", end = '')\n",
    "        counter += 3\n",
    "        codon_counter += 1\n",
    "        if codon_counter%5 == 0:\n",
    "            print(\"\\n\")\n",
    "            if protein_print_overhang == 1:\n",
    "                print (f\"{' '*exon_overhang} \", end = '')\n",
    "                protein_print_overhang = 0\n",
    "            \n",
    "            while start_position < counter:\n",
    "                print(f\" {Seq(exon_sequence[start_position : start_position + 3]).translate()}  \", end = '')\n",
    "                start_position += 3\n",
    "            print(\"\\n\")\n",
    "    \n",
    "    print(\"\\n\")\n",
    "    if protein_print_overhang == 1:\n",
    "        print (f\"{' '*exon_overhang} \", end = '')\n",
    "        protein_print_overhang = 0\n",
    "\n",
    "    while start_position < counter:\n",
    "        print(f\" {Seq(exon_sequence[start_position : start_position + 3]).translate()}  \", end = '')\n",
    "        start_position += 3\n",
    "    print(\"\\n\")\n",
    "    return\n",
    "\n",
    "\n",
    "query_rna_file = SeqIO.parse(\"H:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_rna.txt\", 'fasta')\n",
    "exon_counter = 1\n",
    "full_sequence = ''\n",
    "exon_overhang = 0\n",
    "exon_3_overhang = 0\n",
    "exon_5_overhang = 0\n",
    "\n",
    "output_protein_query = ''\n",
    "\n",
    "for sequences in query_rna_file:\n",
    "    \n",
    "    exon_sequence = str(sequences.seq)\n",
    "    print(f\"Exon {exon_counter}, Length = {len(sequences.seq)}\\nName:{sequences.id}\\n\\n\")\n",
    "    \n",
    "    print_codon_translated(exon_sequence,exon_counter,exon_overhang)\n",
    "    \n",
    "    exon_without_overhang_length = len(exon_sequence) - exon_overhang\n",
    "    exon_5_overhang = exon_overhang\n",
    "    exon_overhang = 3 - exon_without_overhang_length % 3\n",
    "    if exon_overhang == 3:\n",
    "        exon_overhang = 0\n",
    "   \n",
    "    print(exon_overhang)\n",
    "    translated = (sequences.seq[exon_5_overhang:]).translate()\n",
    "    print (f\"\\ntranslated = {translated}\\ntranslated_length = {len(translated)}\\n\")\n",
    "    \n",
    "    output_protein_query += f\">Exon_{exon_counter}_Bombyx_mori_NM_001043510.1\\n{translated}\\n\\n\"\n",
    "    exon_counter += 1\n",
    "    \n",
    "# print(output_protein_query)\n",
    "\n",
    "with open(\"H:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_protein.txt\", 'w') as out_file:\n",
    "    out_file.write(output_protein_query)\n",
    "    \n",
    "    \n",
    "    \n",
    "#     break\n",
    "#     counter = 0\n",
    "# #     while counter < len(exon_sequence):\n",
    "        \n",
    "    \n",
    "    \n",
    "#     if len(sequences.seq)%3 != 0:\n",
    "#         print(sequences.seq, len(sequences.seq))\n",
    "#         break\n",
    "# #         my_seq = Seq(full_sequence)\n",
    "\n",
    "# #     print(my_seq.translate())\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18914ba6",
   "metadata": {},
   "source": [
    "# tblastn for all annotated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "2dcfa6b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Achroia_grisella not present in genome folder\n",
      "Achroia_grisella_2 not present in genome folder\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "species_folder = \"H:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/1.Blast_output\"\n",
    "genome_folder = \"J:/Genomes_2023-12-26/\"\n",
    "genomes_list = os.listdir(genome_folder)\n",
    "blast_command = ''\n",
    "\n",
    "list_of_species = os.listdir(species_folder)\n",
    "for species_name in list_of_species:\n",
    "    if \".ini\" not in species_name:\n",
    "        species = species_name.split(\".\")[1]\n",
    "#         if species in genomes_list:\n",
    "#             print(species)\n",
    "    if species in genomes_list:\n",
    "        list_of_files_in_genome_folder = os.listdir(genome_folder + species)\n",
    "        for file in list_of_files_in_genome_folder:\n",
    "            if \"genomic.fna\" == file[-11:]:\n",
    "                genome_file = file\n",
    "#                 print(genome_file)\n",
    "        blast_command += f'tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_protein.txt\" -db  \"/mnt/j/Genomes_2023-12-26/{species}/{genome_file}\" -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/{species_name}/{species}_blast_out.htm\" -html\\n'\n",
    "    else:\n",
    "        print(f\"{species} not present in genome folder\")\n",
    "\n",
    "    \n",
    "with open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/tblastn_out.txt\", 'w') as out_file:\n",
    "    out_file.write(blast_command)\n",
    "#     print(blast_command)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d89d980a",
   "metadata": {},
   "source": [
    "# Extract a specific exon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "ea7409c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tcGAAGTCAACCCACTCTGGAAGCAATTCGAGTGGCAGTAGCGGTTATGGCGGGAAACCATCTACATCCGGTTACAG\n",
      ">Exon2_Papilio_polytes_NW_013528315.1_485269_485346\n",
      "RSQPTLEAIRVAVAVMAGNHLHPVT\n"
     ]
    }
   ],
   "source": [
    "%reset -f\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "import os\n",
    "\n",
    "def get_genome_file(genome_folder,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(genome_folder + species)\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if \"genomic.fna\" == file[-11:]:\n",
    "            genome_file = file\n",
    "    return (genome_file)\n",
    "\n",
    "\n",
    "scaffold = \"NW_013528315.1\"\n",
    "species = \"Papilio_polytes\"\n",
    "exon_start = 485269\n",
    "exon_end = 485346\n",
    "offset_5, offset_3 = 1,2\n",
    "genome_folder = \"J:/Genomes_2023-12-26/\"\n",
    "reverse = 0\n",
    "\n",
    "\n",
    "# print(exon_start, exon_end)\n",
    "\n",
    "genome_file_name = get_genome_file(genome_folder,species)\n",
    "\n",
    "parsed_genome_file = SeqIO.parse(f\"{genome_folder+species}/{genome_file_name}\",\"fasta\" )\n",
    "\n",
    "for sequences in parsed_genome_file:\n",
    "    if sequences.id == scaffold:\n",
    "        exon_sequence = (str(sequences.seq)[exon_start:exon_end])\n",
    "\n",
    "if reverse == 1:\n",
    "    exon_sequence = Seq(exon_sequence).reverse_complement()\n",
    "exon_sequence_translated = Seq(exon_sequence)[offset_5:].translate()\n",
    "\n",
    "print(exon_sequence)\n",
    "print(f\">Exon2_{species}_{scaffold}_{exon_start}_{exon_end}\\n{exon_sequence_translated}\")\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "e8e769e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    \n",
    "    entries = os.listdir(\"F:/Genomes_2023-12-26/\"+Species)\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    #print(genome_name)\n",
    "    #Genome_name = \"GCA_014332785.1_AM_v1.0_genomic.fna\"\n",
    "    \n",
    "#    reverse_c = 0 #1 = yes, 0 = no\n",
    "    # for pasting 3713291\t3713431\n",
    "#    exon = \"idh\"\n",
    "#    start = 3713291\n",
    "         \n",
    "#    end = 3713434                  \n",
    "    out_Seq = ''\n",
    "    fasta_file = open((\"F:/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'r')\n",
    "    for record in SeqIO.parse(fasta_file,\"fasta\"):\n",
    "    #    print (record.id)\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = str(out_Seq)\n",
    "                dna_seq = Seq(out_Seq)\n",
    "                out_Seq = dna_seq.reverse_complement()\n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "#        return (out_trans.translate()) \n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    fasta_file.close()\n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#print(sequence_extractor(\"Carposina_sasakii\", \"CP053153.1\", 1, 10055832, 10056257 ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "41a91958",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.Galleria_mellonella\n",
      "28.666666666666668 29 26.1\n",
      "2.Galleria_mellonell NW_026442079.1 7100050 7099964 1 N\n",
      "ERror\n"
     ]
    }
   ],
   "source": [
    "# rom sequence_extraction_called import *\n",
    "\n",
    "\n",
    "# phylo_species = open(\"C:/Users/sauba/Desktop/Work_Stuff/MRJP/0.for_automation_temp_files/phylo_species.txt\",'r')\n",
    "# species_list = phylo_species.readlines()\n",
    "species_list = [\"2.Galleria_mellonella\"]\n",
    "#print(species_list)\n",
    "query_name_list = [\"Exon_1\",\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\",\"Exon_13\",\"Exon_14\",\"Exon_15\",\"Exon_16\",\"Exon_17\",\"Exon_18\",\"Exon_19\",\"Exon_20\",\"Exon_21\",\"Exon_22\",\"Exon_23\",\"Exon_24\"]\n",
    "#species_to_test = [\"Amyelois_transitella\",\"Bicyclus_anynana\",\"Helicoverpa_armigera\",\"Manduca_sexta\",\"Zerene_cesonia\",]\n",
    "#query_length_list = [144,40,75,38,27,37,47]\n",
    "seq_modi = [[0,2],[1,2],[1,1],[2,0],[0,0],[0,0],[0,1],[2,1],[2,2],[1,0],[0,0],[0,0],[0,2],[1,2],[1,2],[1,0],[0,1],[2,1],[2,1],[2,0],[0,0],[0,0],[0,0],[0,0]]\n",
    "#query_length_list = [90,66,49,61,64,44,35]\n",
    "#seq_modi = [[0,0,90],[0,0,66],[0,1,49],[2,2,61],[1,2,64],[1,0,44],[0,0,35],]\n",
    "#try:\n",
    "for species_name in species_list:\n",
    "    try:\n",
    "        print(species_name)\n",
    "        header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "        Output_Sequence = header\n",
    "        scaff = \"Intial_value\"\n",
    "        scaff_old = \"Intial_value\"\n",
    "        old_end = 0\n",
    "        for i in range(len(query_name_list)):\n",
    "            query_name = query_name_list[i]\n",
    "            Length_switch = \"0\"\n",
    "    #        query_length = query_length_list[i]\n",
    "            tblast_out = open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/\"+species_name+\"/Galleria_mellonella_blast_out.txt\",'r')\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "            #print(lines_in_file)\n",
    "            result_section_switch = 0\n",
    "            start_coor_switch = 0\n",
    "            query_start_coor_switch = 0\n",
    "            stop_coor_switch = 0\n",
    "            error = \"N\"\n",
    "            break_switch = 0\n",
    "    #        scaff = ''\n",
    "            start = 0\n",
    "            stop = 0\n",
    "            start_coor = 0\n",
    "            stop_coor = 0\n",
    "            \n",
    "            for lines in lines_in_file:\n",
    "                \n",
    "                if query_name in lines:\n",
    "                    result_section_switch = 1\n",
    "                    query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                    query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "#                     print(query_species)\n",
    "#                     print(query_name)\n",
    "    #                break\n",
    "                if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "                    result_section_switch == 0\n",
    "                    break\n",
    "                    \n",
    "    #                if result_section_switch == 1 and query_species not in species_to_test:\n",
    "    #                    print(f\"{species_name.rstrip()} has query {query_species}\")\n",
    "    #                    break_switch = 1\n",
    "    #                    break\n",
    "                    \n",
    "                if result_section_switch == 1:\n",
    "                    if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                        query_length = int(lines.split(\"=\")[1][:-1])\n",
    "                        Length_switch = 1\n",
    "    #                    print(query_length)\n",
    "    #                    break\n",
    "                    if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "        #                print (lines)\n",
    "                        break\n",
    "                    if \">\" in lines:\n",
    "                        \n",
    "                        scaff = lines.split(\" \")[0][1:]\n",
    "                        if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                            error = \"Y\"\n",
    "                        scaff_old = scaff\n",
    "                    if \"Sbjct\" in lines:\n",
    "                        if start_coor_switch == 0:\n",
    "                            start_coor = int(lines.split(\" \")[2])\n",
    "                            start_coor_switch = 1\n",
    "                        stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    if \"Query\" in lines and \"=\" not in lines:\n",
    "                        if query_start_coor_switch == 0:\n",
    "    #                        print(lines)\n",
    "                            query_start_coor = int(lines.split(\" \")[2])\n",
    "                            query_start_coor_switch = 1\n",
    "                        query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "        #                print (stop_coor)\n",
    "                    \n",
    "    #            print(query_name, start_coor, stop_coor, query_start_coor, query_stop_coor)\n",
    "            if break_switch == 1:\n",
    "                break\n",
    "            \n",
    "            if start_coor < stop_coor:\n",
    "                complement = \"0\"\n",
    "                length = (stop_coor-start_coor)/3\n",
    "                start = start_coor\n",
    "                stop = stop_coor\n",
    "             \n",
    "            elif start_coor > stop_coor:\n",
    "                complement = \"1\"\n",
    "                length = (-stop_coor+start_coor)/3\n",
    "                start = stop_coor\n",
    "                stop = start_coor\n",
    "             \n",
    "            else:\n",
    "                error = \"Y\"\n",
    "            print(length, query_length,query_length - 0.1*query_length )\n",
    "            if length < query_length - 0.1*query_length:\n",
    "                error = \"Y\"\n",
    "    #        translated_sequence = (sequence_extractor(species_name.split(\"\\n\")[0], scaff, int(complement), start, stop ))\n",
    "    #        print (translated_sequence[0])\n",
    "            old_trans = ''\n",
    "#             print(start, stop)\n",
    "            while True and (start != 0 or stop != 0):\n",
    "                if (query_name == \"Exon_1\" or query_name == query_name_list[-1]):\n",
    "                    frame = 1\n",
    "                    translated_sequence = (sequence_extractor(species_name.split(\".\")[1], scaff, int(complement), start, stop,frame ))\n",
    "    #                    print(species_name.split(\"\\n\")[0], scaff, int(complement), start, stop,frame, query_name )\n",
    "                    if translated_sequence == old_trans:\n",
    "                        break\n",
    "                    if query_name == \"Exon_1\" :\n",
    "                        if translated_sequence[0] != \"M\":\n",
    "                            print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                            if complement == \"0\":\n",
    "                                start = int(start) - 3\n",
    "                            if complement == \"1\":\n",
    "                                stop = int(stop) + 3\n",
    "                            old_trans = translated_sequence\n",
    "                        if \"*\" in translated_sequence:\n",
    "                            error = \"Y\"\n",
    "                            break\n",
    "                        if translated_sequence[0] == \"M\":\n",
    "    #                            print(translated_sequence)\n",
    "    #                            print(start, stop)\n",
    "                            break\n",
    "    #                    elif query_name == \"Exon1\" and query_species in species_without_exon1:\n",
    "    #                        print(\"Species_without_exon1\")\n",
    "    #                        if complement == \"0\":\n",
    "    #                            start = int(start) - 3*(int(query_start_coor)-1)\n",
    "    #                        if complement == \"1\":\n",
    "    #                            stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    #                        break\n",
    "                            \n",
    "                    if query_name == query_name_list[-1]:\n",
    "                        if translated_sequence[-1] != \"*\":\n",
    "                            print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                            if complement == \"0\":\n",
    "                                stop = int(stop) + 3\n",
    "                            if complement == \"1\":\n",
    "                                start = int(start) - 3\n",
    "                            old_trans = translated_sequence\n",
    "                        if translated_sequence[-1] == \"*\":\n",
    "    #                            print(translated_sequence)\n",
    "    #                            print(start, stop)\n",
    "                            break\n",
    "                else:\n",
    "                    break\n",
    "            \n",
    "            if (start != 0 or stop != 0):\n",
    "                start_modifier = seq_modi[i][0]\n",
    "                stop_modifier = seq_modi[i][1]\n",
    "            else:\n",
    "                start_modifier = 0\n",
    "                stop_modifier = 0            \n",
    "            seq_length = query_length\n",
    "            if query_start_coor != \"1\" and query_name != \"Exon_1\":\n",
    "                if complement == \"0\":\n",
    "                    start = int(start) - 3*(int(query_start_coor)-1)\n",
    "                if complement == \"1\":\n",
    "                    stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    #        print (query_stop_coor, seq_length)\n",
    "            if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "                if complement == \"0\":\n",
    "                    stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "                if complement == \"1\":\n",
    "                    start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    "    #        print(start, stop)\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - int(start_modifier)\n",
    "                stop = int(stop) +  int(stop_modifier)\n",
    "                if old_end != 0 and old_end > stop:\n",
    "                    \n",
    "                    error = \"Y\"\n",
    "                old_end = stop\n",
    "                \n",
    "            if complement == \"1\":\n",
    "                start = int(start) - int(stop_modifier)\n",
    "                stop = int(stop) +  int(start_modifier)\n",
    "                if old_end != 0 and old_end < stop:\n",
    "                    error = \"Y\"\n",
    "                old_end = stop\n",
    "    #        print(start, stop)\n",
    "            \n",
    "    #        print(translated_sequence)\n",
    "    \n",
    "            \n",
    "            if start == 0 or stop == 0:\n",
    "                error = \"Y\"\n",
    "            output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "            Output_Sequence = Output_Sequence + output_format\n",
    "        \n",
    "        print(Output_Sequence)\n",
    "        output_file = open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/\"+species_name+f\"/{species_name}_coordinates.csv\",'w')\n",
    "        output_file.write(Output_Sequence)\n",
    "        output_file.close()\n",
    "        #        break\n",
    "    except:\n",
    "        print(species_name[:-1], scaff, start_coor, stop_coor, complement, error)\n",
    "        print(\"ERror\")\n",
    "##        break\n",
    "# =============================================================================\n",
    "# except:\n",
    "#     \n",
    "#     print(species_name[:-1], scaff, start_coor, stop_coor, complement, error)\n",
    "# #    break\n",
    "# =============================================================================\n",
    "#print(Output_Sequence)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "2dc0a539",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.Galleria_mellonella\n",
      "28.666666666666668 29 23.2\n",
      "28.666666666666668 27 21.6\n",
      "28.666666666666668 51 40.8\n",
      "20.666666666666668 31 24.8\n",
      "18.666666666666668 26 20.8\n",
      "43.666666666666664 44 35.2\n",
      "23.666666666666668 25 20.0\n",
      "66.66666666666667 67 53.6\n",
      "72.66666666666667 75 60.0\n",
      "36.666666666666664 37 29.6\n",
      "39.666666666666664 40 32.0\n",
      "49.666666666666664 50 40.0\n",
      "38.666666666666664 39 31.2\n",
      "12.666666666666666 29 23.2\n",
      "47.666666666666664 50 40.0\n",
      "61.666666666666664 63 50.4\n",
      "46.666666666666664 49 39.2\n",
      "32.666666666666664 37 29.6\n",
      "24.666666666666668 38 30.4\n",
      "53.666666666666664 61 48.8\n",
      "37.666666666666664 67 53.6\n",
      "40.666666666666664 45 36.0\n",
      "33.666666666666664 34 27.2\n",
      "62.666666666666664 83 66.4\n",
      "PSLVNEQLGQLYVDLQLEGVASRLTLEEGITSSSSSGEETLTTTPKVRNYAFILNICNNYETF\n",
      "PSLVNEQLGQLYVDLQLEGVASRLTLEEGITSSSSSGEETLTTTPKVRNYAFILNICNNYETFL\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "2.Galleria_mellonella,NW_026442079.1,7099962,7100050,1,N,Exon_1,1,29,29\n",
      "2.Galleria_mellonella,NW_026442079.1,6,0,1,Y,Exon_2,1,29,27\n",
      "2.Galleria_mellonella,NW_026442079.1,-66,0,1,Y,Exon_3,1,29,51\n",
      "2.Galleria_mellonella,NW_026442079.1,7089119,7089213,1,Y,Exon_4,4,24,31\n",
      "2.Galleria_mellonella,NW_026442037.1,4374380,4374457,0,Y,Exon_5,8,26,26\n",
      "2.Galleria_mellonella,NW_026442079.1,7087776,7087907,1,Y,Exon_6,1,44,44\n",
      "2.Galleria_mellonella,NW_026442079.1,7086337,7086412,1,N,Exon_7,1,24,25\n",
      "2.Galleria_mellonella,NW_026442079.1,7085154,7085357,1,N,Exon_8,1,67,67\n",
      "2.Galleria_mellonella,NW_026442079.1,7084583,7084811,1,N,Exon_9,3,75,75\n",
      "2.Galleria_mellonella,NW_026442079.1,7083898,7084009,1,N,Exon_10,1,37,37\n",
      "2.Galleria_mellonella,NW_026442079.1,7083303,7083422,1,N,Exon_11,1,40,40\n",
      "2.Galleria_mellonella,NW_026442079.1,7082905,7083054,1,N,Exon_12,1,50,50\n",
      "2.Galleria_mellonella,NW_026442079.1,7082581,7082699,1,N,Exon_13,1,39,39\n",
      "2.Galleria_mellonella,NW_026442079.1,7082200,7082289,1,Y,Exon_14,1,13,29\n",
      "2.Galleria_mellonella,NW_026442079.1,7081413,7081562,1,N,Exon_15,1,49,50\n",
      "2.Galleria_mellonella,NW_026442079.1,7080734,7080938,1,N,Exon_16,1,57,63\n",
      "2.Galleria_mellonella,NW_026442079.1,7079386,7079530,1,N,Exon_17,2,49,49\n",
      "2.Galleria_mellonella,NW_026442079.1,7077431,7077538,1,N,Exon_18,1,35,37\n",
      "2.Galleria_mellonella,NW_026442151.1,951569,951685,1,Y,Exon_19,2,26,38\n",
      "2.Galleria_mellonella,NW_026442079.1,7075882,7076045,1,Y,Exon_20,1,61,61\n",
      "2.Galleria_mellonella,NW_026442079.1,7074018,7074218,1,Y,Exon_21,22,59,67\n",
      "2.Galleria_mellonella,NW_026442079.1,7073368,7073490,1,N,Exon_22,1,45,45\n",
      "2.Galleria_mellonella,NW_026442079.1,7072977,7073078,1,N,Exon_23,1,34,34\n",
      "2.Galleria_mellonella,NW_026442079.1,7072445,7072639,1,Y,Exon_24,1,63,83\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# rom sequence_extraction_called import *\n",
    "\n",
    "\n",
    "# phylo_species = open(\"C:/Users/sauba/Desktop/Work_Stuff/MRJP/0.for_automation_temp_files/phylo_species.txt\",'r')\n",
    "# species_list = phylo_species.readlines()\n",
    "species_list = [\"2.Galleria_mellonella\"]\n",
    "#print(species_list)\n",
    "query_name_list = [\"Exon_1\",\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\",\"Exon_13\",\"Exon_14\",\"Exon_15\",\"Exon_16\",\"Exon_17\",\"Exon_18\",\"Exon_19\",\"Exon_20\",\"Exon_21\",\"Exon_22\",\"Exon_23\",\"Exon_24\"]\n",
    "#species_to_test = [\"Amyelois_transitella\",\"Bicyclus_anynana\",\"Helicoverpa_armigera\",\"Manduca_sexta\",\"Zerene_cesonia\",]\n",
    "#query_length_list = [144,40,75,38,27,37,47]\n",
    "seq_modi = [[0,2],[1,2],[1,1],[2,0],[0,0],[0,0],[0,1],[2,1],[2,2],[1,0],[0,0],[0,0],[0,2],[1,2],[1,2],[1,0],[0,1],[2,1],[2,1],[2,0],[0,0],[0,0],[0,0],[0,0]]\n",
    "#query_length_list = [90,66,49,61,64,44,35]\n",
    "#seq_modi = [[0,0,90],[0,0,66],[0,1,49],[2,2,61],[1,2,64],[1,0,44],[0,0,35],]\n",
    "#try:\n",
    "for species_name in species_list:\n",
    "#     try:\n",
    "    print(species_name)\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "#        query_length = query_length_list[i]\n",
    "        tblast_out = open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/\"+species_name+\"/Galleria_mellonella_blast_out.txt\",'r')\n",
    "        lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "#        scaff = ''\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "            if query_name in lines:\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "#                     print(query_species)\n",
    "#                     print(query_name)\n",
    "#                break\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "                result_section_switch == 0\n",
    "                break\n",
    "\n",
    "#                if result_section_switch == 1 and query_species not in species_to_test:\n",
    "#                    print(f\"{species_name.rstrip()} has query {query_species}\")\n",
    "#                    break_switch = 1\n",
    "#                    break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                    query_length = int(lines.split(\"=\")[1][:-1])\n",
    "                    Length_switch = 1\n",
    "#                    print(query_length)\n",
    "#                    break\n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "                if \">\" in lines:\n",
    "\n",
    "                    scaff = lines.split(\" \")[0][1:]\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                if \"Sbjct\" in lines:\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "    #                print (stop_coor)\n",
    "\n",
    "#            print(query_name, start_coor, stop_coor, query_start_coor, query_stop_coor)\n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\"\n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\"\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        print(length, query_length,query_length - 0.2*query_length )\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "#        translated_sequence = (sequence_extractor(species_name.split(\"\\n\")[0], scaff, int(complement), start, stop ))\n",
    "#        print (translated_sequence[0])\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "        while True and (start != 0 or stop != 0):\n",
    "            if (query_name == \"Exon_1\" or query_name == query_name_list[-1]):\n",
    "                frame = 1\n",
    "                translated_sequence = (sequence_extractor(species_name.split(\".\")[1], scaff, int(complement), start, stop,frame ))\n",
    "#                    print(species_name.split(\"\\n\")[0], scaff, int(complement), start, stop,frame, query_name )\n",
    "                if translated_sequence == old_trans:\n",
    "                    break\n",
    "                if query_name == \"Exon_1\" :\n",
    "                    if translated_sequence[0] != \"M\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            start = int(start) - 3\n",
    "                        if complement == \"1\":\n",
    "                            stop = int(stop) + 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if \"*\" in translated_sequence:\n",
    "                        error = \"Y\"\n",
    "                        break\n",
    "                    if translated_sequence[0] == \"M\":\n",
    "#                            print(translated_sequence)\n",
    "#                            print(start, stop)\n",
    "                        break\n",
    "#                    elif query_name == \"Exon1\" and query_species in species_without_exon1:\n",
    "#                        print(\"Species_without_exon1\")\n",
    "#                        if complement == \"0\":\n",
    "#                            start = int(start) - 3*(int(query_start_coor)-1)\n",
    "#                        if complement == \"1\":\n",
    "#                            stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "#                        break\n",
    "\n",
    "                if query_name == query_name_list[-1]:\n",
    "                    if translated_sequence[-1] != \"*\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            stop = int(stop) + 3\n",
    "                        if complement == \"1\":\n",
    "                            start = int(start) - 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if translated_sequence[-1] == \"*\":\n",
    "#                            print(translated_sequence)\n",
    "#                            print(start, stop)\n",
    "                        break\n",
    "            else:\n",
    "                break\n",
    "\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0            \n",
    "        seq_length = query_length\n",
    "        if query_start_coor != \"1\" and query_name != \"Exon_1\":\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)\n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "#        print (query_stop_coor, seq_length)\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    "#        print(start, stop)\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "#        print(start, stop)\n",
    "\n",
    "#        print(translated_sequence)\n",
    "\n",
    "\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "        output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "    output_file = open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/\"+species_name+f\"/{species_name}_coordinates.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n",
    "        #        break\n",
    "#     except:\n",
    "#         print(species_name[:-1], scaff, start_coor, stop_coor, complement, error)\n",
    "#         print(\"ERror\")\n",
    "##        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28df9159",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71c2e163",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98e69a07",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd6e4602",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bc7b6ab",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea8d7e34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1d0c6a1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4a8e29a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d697c37b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76941efc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2ad1e85",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
