{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3247b82e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0653b64e",
   "metadata": {},
   "outputs": [],
   "source": [
    "scaffold = \"\"\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "086c0ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "#     print(list_of_files_in_genome_folder)\n",
    "    if \"Temp\" in list_of_files_in_genome_folder:\n",
    "        subprocess.run(f'rm -r \"{genome_location}/{species}/Temp\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{genome_location}/{species}/Temp\")\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9d4c6ece",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_query_folder(annotated_species_location,gene_name,species):\n",
    "    import time\n",
    "\n",
    "\n",
    "    list_of_files = os.listdir(annotated_species_location)\n",
    "    if species in list_of_files:\n",
    "        subprocess.run(f'rm -r \"{annotated_species_location}/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    time.sleep(2.5)\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts/{gene_name}\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7cb40a07",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus XM_045638192.1\n",
      "+\n",
      "{'Exon_1': ['1366117', '1366370'], 'Exon_2': ['1366617', '1366800'], 'Exon_3': ['1367167', '1367301'], 'Exon_4': ['1367600', '1367856'], 'Exon_5': ['1368893', '1369090'], 'Exon_6': ['1369807', '1369989'], 'Exon_7': ['1370342', '1370474'], 'Exon_8': ['1370849', '1371003'], 'Exon_9': ['1371576', '1371694'], 'Exon_10': ['1372548', '1372701'], 'Exon_11': ['1376780', '1376860'], 'Exon_12': ['1377199', '1377308'], 'Exon_13': ['1377521', '1377571'], 'Exon_14': ['1377880', '1377949'], 'Exon_15': ['1379870', '1380068'], 'Exon_16': ['1381168', '1381316'], 'Exon_17': ['1385397', '1385491'], 'Exon_18': ['1390782', '1390982'], 'Exon_19': ['1392580', '1392723'], 'Exon_20': ['1393077', '1393262'], 'Exon_21': ['1396658', '1396854'], 'Exon_22': ['1397436', '1398048'], 'Exon_23': ['1403828', '1403907']}\n",
      "exon counter 1\n",
      ">Colias_croceus_XM_045638192.1_Exon1\n",
      "ATGAGTTCCTTACTGTCTGCTGAACTTTCTGCAACATGCAATGCACTtggaacttttaataaaaagactGGAAAATACATGATAGATGACTTCACATTAAATACagttaaagatttaataaGATACATGCGCAGGGATGGTGAGGATCACGAAATTCGAAGGCATTTAGGACAGACTAAAATTCTTCAAAGTGACTTGTTACCAATGTTAGTTGACCATTGGGAAAACAATGAACTTTTTGATGTAACTCTTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon2\n",
      "gtTAGTCGTTAACTTAACAAACCCAGCACTATTACTATTCAGAGAAGAAGTGCCAGTAGAAAGAACAGCTCGACACAATTATTTGCAAATTCTATCCCACCTACAGTCTTATAAAGAAGAGGCGTTTAGTAAAGTTGAGAATTGGCAGGTGTTCGCTAAAAAACTGACTAAAATACTGgaaatt\n",
      ">Colias_croceus_XM_045638192.1_Exon3\n",
      "gaTTGGTCTGAACGAGATGAAGACACTGGTCTAGTAATAGAAAGGATATTAATTCTAATAAGGAATGTATTACATGTACCTGCCGATCTGGACAAAGAGCGGAGACCAGATAATGATGCTAGTATTCATGATCag\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon4\n",
      "gTACTCTGGGCATTAAACCAATCAGGAATCCTAGACATAATTCTCTACATGTCGTCAGAAAACGAGAAGCAATACTTCATGCATATATTAGAGATAATCACTCACTTGCTTCGTGAGCAGAACCCAACGAGTTTAGCTGATGCTGCGTTACAGAGAAGTGTGGATGAGAAACTGAGGGATGAGCAGGAATTACTGACTATTAGAATGTCAGAGACGAAAGAGAGGGTTAATAAAGTTAGGCAGTATTCTTCGACAAG\n",
      ">Colias_croceus_XM_045638192.1_Exon5\n",
      "gcacTCAAGATTTGGAGGGACCTACGTAATACAAAACATGAAATCAATATCAGAAAATGAGCTAATATGTTTGAAGCCTCTGAACAAAATATCTAACTTAGACTTTAACGGTAGCAAGAAATCCAAGTTAGTGAAACCTAAAAACAGGAGGCCTGAAGAGAGTGGTGCTATGGAGAGACGATCGCCTTTTGCTGTTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon6\n",
      "ATTATTTCTCAAAGAGTTTTGCATAGAATTCCTAAACAGTTCTTACAACCCGCTAATGCACTATGTCAAAGATGTGCTAGTTAGAGCTAAAGCCCAACAGAACGATGAATCTTACTTCTTGTGGGCGATGAGGTTCTTTATGGAGTTCAATAGAGGACACAAGTTTGATGTCGCTTTGGTTAG\n",
      ">Colias_croceus_XM_045638192.1_Exon7\n",
      "tgAAACAATGTCAGTGCCTATGTTTCACTATGTCCAGCAGcaaatggaaaaatattacGACATGATACGAGTTGAGAAGAAGAAATATCATGCCTGGGTTCGGAGATTGCATTTAGCATTGAAGTGTTATAAg\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon8\n",
      "GAGTTATTAAACACACTTCTAAAAATGGATAAGTGTGATAGTGCTAATGTGAAGGAATCAGCAAAAGTGATAAAAAGCAATGTTTTCTATGTGCTGGAGTATCGAGAGTTCGTGCTGGCGACCATTTTGAATtatgatgataataaaatgCCGAG\n",
      ">Colias_croceus_XM_045638192.1_Exon9\n",
      "aTCTTATCTATTGGATTTGGTAGAAACAGTGCACTTGTTCTTGAAAATGTTGGAGCATTACTGCAAAAAGACTGGTCTGGTTGTTCAGAAGAAAGTTAGAAGGAAGgctaaaagtaaaa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon10\n",
      "AACAAAAGCCCCGCCCAAAGCCCCAAGCCCCCAAACCGGATATCCCGCAATGGGAAGCGGTTCGCGAGCAAATGGCGGTAGTTCTCACTTCGCAAATAGACAACTACCCTCCACCGTTTGACGCGGCGTCTGATACACCGATTGATCAGCAGAA\n",
      ">Colias_croceus_XM_045638192.1_Exon11\n",
      "GGAGGATTGTATGAAGAAAATACAGAAACTGATGCGCGAACATAAATATGAGGACGCTGTTGGCACATTCCGGGCGGCGag\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon12\n",
      "AGAAGTATGGCCGGAAGAAGGTATTTTTGGTGTGGAAGGAATTCAAGAACACGAAGAGTTAGATATGCTAGAAACTATCTATAACACTGACTTGGGAGTTGAAATTACAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon13\n",
      "AAAATGTTGAAGATGACAATCAAGAAGATAATTCAGAATATGACGAAGAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon14\n",
      "atgaagaagaagaagatagAACGACTTCGGTGGTAGAAACTGAATTTGATTTCCAAGACTTCGTGCAAAg\n",
      ">Colias_croceus_XM_045638192.1_Exon15\n",
      "ATTCTGCCATCCCCGCGTAGTAAGCGCATGTGCATCACTATTAGAGAAATACGACAAAAACACTCCATACACAAACCATTGTATTGTGAAAATGCTGCACCGAATTGCATGGGATTGTAAACGACCGGCCATGATGTTCCAGGCGTCCATGTTTCTGACCTTCCAGAAGATTCTGGGAAATCGAATTGCGCATTTTAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon16\n",
      "gAACTCGAAAAATTCGCCATCTACGTAATAAGGAAATTCACAGAGATAGCCAGTAAGAGTCCAAAAGCGTTCATAGAACTATTGTTTTGGACCAACTCTAAAGATGCGATCGACGTTGAGTGTGGATACGACCTTTATACTGATTctag\n",
      ">Colias_croceus_XM_045638192.1_Exon17\n",
      "GGATAAACCAGGCAAGAATACTTGGTCAGAAGAACAAGAGGATGAATTACGGAATCTATATATGGAGAATCAGAGCAATCCACAGACAGATCAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon18\n",
      "ACGTAATAGACTGGATCCTAGACAATCTAATAGACCAGACTCGAACCCGACGCAACGTGATAGCCAAACTAAAAGACCTGGGGCTTATATTCAAAGCACCAACCAAGAAGAGTAATAAGGAGAGAAGAATACAGAAAGAGTTTAGTGAGGAAGAAGATACCTTGTTGGAAGAATTGTGGACGCAGTATAAGGATTCAGAag\n",
      ">Colias_croceus_XM_045638192.1_Exon19\n",
      "atccAATGGCGCTTATAATGCCCAGAATGCCGAAAAAACGACAAAAACGCAGTTTCATAGACCGTTTGTTAGAACTCGGCTTGATTCAAGACAAAAAGGAGTGCCGTAAGAAGAGACAGAGAAAGGAAAAGAGCGGCCCTCAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon20\n",
      "ATGACACAGACTCCGAATCCAAAGAATCGAATTCAGAAAGAGACATCTCAAGAAGTCCATCACCAGTTAGAATAAAACCAAAGAAGAAAGAACCCAGAAAGAAAGAAGAGAGGAAAAGAAATAACAAGAGAGTTGATATGAGTGCGAAAGAGATAGCAAGAGTTATGGTACAGGCTGTTGATAATg\n",
      ">Colias_croceus_XM_045638192.1_Exon21\n",
      "GTCAAACAGAAGCCCTAACCTGGCTAGCGGAAAACCTCGAAGAGGTAGCCCTAGACCAAGAAGCCGAGGGCTATGACCCTACTTCTGAAGGCACTCCCTTAGTACCTATATCCAAAGATTCCATAGACGCTATGGAAGATAAAATGTTCCAGAAGGTTCTTAAAGCTGTCGGAGTTATGCCACCTAGTGATGAACAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_XM_045638192.1_Exon22\n",
      "GAGACATACTGGCGCATACCAGCTGACTTACACCACGAAACAATAAGGAAGCGTCGAGAAATCATTCTTAAAGCTATCAACAAAGAACTCATAGTTGACACAAATATATCTGATAATGAaacaaataacattaatacaGATATGGCTAATGATTCTGATTCTAATGATAATCAAAATGATGAAAACTCTAATAATGACCATGAAAgaaataatgatattattatagatatggCTAATGATTCTGATTCTAATGATAATAGACAGATTGAAAATAATGATATACAAAGAGTTCAGAAAAGTTCACTTTTCGATACTTTGTTAAATAATGATGATAGTgatagtgatgatgatgaaaatacaacaaacaaaataaataatgatagtGATGATGAcggtgataaaaataaaacaaataatgatgtatttgataaaaataatacaagaaACAGTCGTAAAAGAGTTCGTAGCGTAGAAACAGAGGAACAAGTGAACAAATTAGCGAGAATAGACGAAGATGATGAAGATATATTGTCGTTTGCTAAGAAAACACTCAATCAAGAGATTTTACCAGACACAGAGGATATTTTAAGTGCGTTACCACGACGTAATGATG\n",
      ">Colias_croceus_XM_045638192.1_Exon23\n",
      "ATAGCAGTGATGAAGACGCGCCAATAAATATACCGAAACGGAAAAAACGTCTTGTAGTTAGCGACGATGAATCTGactaa\n",
      "254\t184\t135\t257\t198\t183\t133\t155\t119\t154\t81\t110\t51\t70\t199\t149\t95\t201\t144\t186\t197\t613\t80\t\n",
      "Zerene_cesonia XM_038367126.1\n",
      "+\n",
      "{'Exon_1': ['1190851', '1191104'], 'Exon_2': ['1191207', '1191390'], 'Exon_3': ['1191885', '1192019'], 'Exon_4': ['1192351', '1192607'], 'Exon_5': ['1192700', '1192897'], 'Exon_6': ['1193287', '1193469'], 'Exon_7': ['1193623', '1193755'], 'Exon_8': ['1194453', '1194607'], 'Exon_9': ['1194731', '1194849'], 'Exon_10': ['1195914', '1196067'], 'Exon_11': ['1200187', '1200267'], 'Exon_12': ['1201892', '1202001'], 'Exon_13': ['1202261', '1202311'], 'Exon_14': ['1203068', '1203137'], 'Exon_15': ['1203495', '1203693'], 'Exon_16': ['1204961', '1205109'], 'Exon_17': ['1208002', '1208096'], 'Exon_18': ['1210819', '1211019'], 'Exon_19': ['1214916', '1215059'], 'Exon_20': ['1215414', '1215590'], 'Exon_21': ['1218331', '1218527'], 'Exon_22': ['1218937', '1219516'], 'Exon_23': ['1219983', '1220059']}\n",
      "exon counter 1\n",
      ">Zerene_cesonia_XM_038367126.1_Exon1\n",
      "atgagttCCTTACTATCTGCTGAACTTTCTGCAACTTGCAATGCTCTTggaacatttaataaaaagaccGGAAAATACATGATAGACGACTTTACGTTAAATACagttaaagatttaataagaTACATGAGAAGGGATGGTGAGGATCACGAAATTCGAAGGCATTTAGGACAGACTAAAGTTCTACAAAGTGACTTGTTGCCTATGTTAGTTGACCATTGGGAAAACAATGAACTTTTTGATGTCACTCTGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon2\n",
      "GTTAGTCGTTAACTTAACAAACCCAGCTCTATTGCTGTTCAGAGAAGAAGTGCCCGTGGAAAGAACGGCTCGCCATAACTACTTGCAAATTCTATCCCACCTACAGTCATATAAAGAAGAGGCATTTAGCAAAGTTGAAAATTGGCAGGTGTTTGCTAAAAAGCTTACCAAGATACTTGAAATA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon3\n",
      "GATTGGTCCGAAAGAGATGAAGACACTGGATTAGTAatagaaagaattttgattCTAATACGAAATGTATTACATGTACCTGCCGACTTAGACAAAGAGAGAAGACCAGATAATGATGCTAGCATTCATGATCAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon4\n",
      "GTGCTCTGGGCATTAAACCAGTCGGGCATCCTAGACATAATCCTCTATATGTCATCGGAAAACGAAAAGCAGTACTTCATGCACATACTAGAAATCATCACACACTTGCTCCGAGAACAGAACCCCACAAGTTTAGCTGATGCTGCATTACAACGGAGTGTGGATGAGAAACTGAGGGATGAACAGGAATTACTGACTATACGAATGTCGGAGACAAAGGAGAGAGTTAATAAAGTTAGGCAATATTCTTCAACcag\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon5\n",
      "GCACTCAAGATTTGGAGGGACCTATGTGATACAAAACATGAAATCAATATcagaaaatgaattaatatgtttGAAGCCTCTGAACAAAATATCGAATTTAGACTTCAATGGTAGTAAGAAGTCTAAGTTAGTTAAACCTAAAAACAGAAGGCCTGAGGAGAGTGGCGCTATGGAGAGACGATCGCCATTTGCTGTAAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon6\n",
      "ACTGTTTCTGAAAGAGTTTTGCATAGAATTTCTGAATAGTGCGTACAATCCATTAATGCACTATGTTAAAGATGTATTAGTTAGAGCTAAAGCCCAACAAAACGATGAGTCATATTTCCTATGGGCAATGAGATTCTTTATGGAGTTTAATAGAGGACACAAGTTCGATGTTGCTTTGGTTGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon7\n",
      "TGAAACCATGTCAGTGCCTATGTTTCATTACGTCCAACAGCAAATGGAGAAATATTATGACATGATACGAGTTGAGAAGAAAAAGTACCATGCCTGGGTTCGGAGATTGCATTTAGCGTTGAAGTGTTATAAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon8\n",
      "GAGCTATTGAACACACTTCTAGTGATGGACAAGTCCGAAAGTGTCAATGTGAAGGAATCGGCGAAAGTGATAAAGAGCAATGTGTTTTATGTGTTGGAGTATAGAGAGTTCGTACTGGCGACCATTTTGaattatgatgataataaaatgcCGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon9\n",
      "ATCTTATTTATTGGATTTGGTAGAAACCGTgcatttattcttaaaaatgttGGAGCATTATTGCAAAAAGACCGGTCTAGTGGTTCAGAAGAAAATTCGTAGGAAGGCTAAAAGCAAAA\n",
      ">Zerene_cesonia_XM_038367126.1_Exon10\n",
      "AACCAAAGCCCCGGCCGAAGCCCCAAGCCCCCAAACCGGATATCCCCCAGTGGGAAGCAGTCCGCGAGCAAATGGCGGTAGTTCTCACTTCGCGAATCGACAACTACCCTCCGCCATTCGACGCAGCGTCTGACACACCGATTGACCAGCAGAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon11\n",
      "GGAAGATTGTATGAAGAAAATTCAGAAACTGATGCGCGAACAGAAATACGAGGACGCTGTCGGCACATTCCGGGCGGCGAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon12\n",
      "AGAGGTGTGGCCAGAGGAGGGTATATTTGGCGTGGAGGGAATCCAAGAGCACGAGGAGTTGGACATGCTGGAAACTATCTATAGCACAGACCTGGGAGTTGAGATTATAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon13\n",
      "ataatGTTGAAGACGATAATCAAGAAGATAATTCAGAATATGACGAAGAAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon14\n",
      "ATGAAGAAGAGGAAGACAGAACCACGGCTGTTGTGGAAACCGAATTCGATTTCCAGGACTTCGTGCAGag\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon15\n",
      "ATTCTGCCATCCCCGGGTGGTGAGCGCTTGCGCATCTCTGCTGGAGCGATATGAGAAGAACGCGCCGCACACGAACCACTGTATTGTGAAAATGCTGCACCGAATCGCGTGGGATTGCAAGCGGCCGGCCATGATGTTCCAAGCGTCTATGTTCCTGACGTTCCAGAAGATTCTTGGCAACCGTGTTGCGCATTTTAAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon16\n",
      "GAGTTAGAAAAGTTCGCGATCTACGTGTTGAGGAAGTTCAGCGAGGTGGCCGGCAAGAGTCCGAAGGCGTTCATAGAACTGCTCTTCTGGACCAGCTCCAAGGACGCCATCGACGTCGAGTGCGGTTACGACCTGTACAATGATCCTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon17\n",
      "AGATAAACCGGGAAAGAATACTTGGTCAGAAGAACAGGAAGAAGAGTTGCGGAAGCTGTATATGGAGAATCAAAGTAACCCACAAACTGATCAGG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon18\n",
      "ATGTCATAGACTGGATCCTAGACAACCTAATAGACCAGACTAGAACGCGGCGGAGTGTTATATCGAAACTAAAAGAGCTGGGGCTCATATTCAAAGCGCCGACCAAGAAGAGTAACAAAGAGAGGAGAGTTCAGAAGGAGTTCAGCGAGGAAGAAGACACTTTGTTGGAGGAACTGTGGACACAGTTTAAGGAAACGGATG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon19\n",
      "ACCCAATGCCGCTTATAATGGCCAGAATGCCCCGAAAGAGGCAAAAGCGCAGTTTCATAGACCGCTTATTGGAACTCGGCTTGATCCAAGATAAGAAGCAGTGTCGTAAGAAGAGACAGAGAAAGGAAAAGAGCGGCCCTCGAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon20\n",
      "atGATACAGAAGACGAATCGAAAGAATCAAATTCGGAAAGCGATACTTCCAGAAGTCCGTCGCCAGTGAAAATTAAGCCAAAGAAGAAAGTTCGTGAAGAGAAGAAAAGAAACAGAAAGAGAGTTGATATGAGTGCGAAAGAGATAGCTATATTGATGGTGAAAGCTGTTGATAATg\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon21\n",
      "GTCAAACGGGAGCCCTAACCTGGCTAGCCGAAAACCTAGAGGAGGTAGCGCTAGACCAAGAAGCGGAAGGATACGACCCCAGCTCTGAAGGAATTCCATTAGTGCCTATATCCAGGGATGCCGTTGAGGCCATGGATGATAATTCGTTCCAGAAAGTTCTAAGAGCTGTGGGAGTGATGCCGCCGAGTGATGAACAG\n",
      ">Zerene_cesonia_XM_038367126.1_Exon22\n",
      "gaATCATACTGGCGAATACCGGCCGACTTACATCACGAAACAATTAGAAAACGCCGCGAGATCATACTTAAAGCTATCAACAAAGAACTCATAGTCGATCCTAACATATCTGATAATGAAaccgataaaaataaagatggtGAAAATATAGACATGGCTAATGATTCTGATTCTAATGATAGTAGACAAGAAGCAAtcaatgatgataatgatatgaTTATGAATACGGCTAATGATGATTCTGATTCTAATGACATTAGACAAGACAGTGTAATTAATGATGATGCGCGAAGAGTAAGTCAGAAAAGTTCAGCTTCgcatattacaataaataatgatgcTAGTGATGACAGTGatgatgaaacaaataaaatagataatgatgataaatttaaaacaaatacaccCAGATATAGTCGTAAAAGAGTACGCAGTGTAGATATAGAGGAACAAGTTAGCAAATTGGCGAAGAttgatgaagatgatgaagATATATTGTCATTCGCCAAGAAAACGCTGAATCAAGAGCATCTGCCTGACACGGAGGATATTTTGAGTGCTTTGCCACAACGCGATAATG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Zerene_cesonia_XM_038367126.1_Exon23\n",
      "ATAGCAGTGATGAAGACGCGCCCATAAATATATCGAAAAGGAAAAAACGTCTGGTTGTTAGTGATGACTCTGAATAG\n",
      "254\t184\t135\t257\t198\t183\t133\t155\t119\t154\t81\t110\t51\t70\t199\t149\t95\t201\t144\t177\t197\t580\t77\t\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "# species_gene = [\"Vanessa_tameamea,XM_026631919.1\"]\n",
    "# species_gene = [\"Vanessa_tameamea,XM_026628490.1\",\"Papilio_machaon,XM_045678832.1\",\"Papilio_polytes,XM_013283593.1\",\"Papilio_xuthus,XM_013309668.1\",\"Leptidea_sinapis,XM_050825300.1\",\"Pieris_brassicae,XM_045658677.1\",\"Pieris_napi,XM_047670174.1\",\"Pieris_rapae,XM_022274940.2\"]\n",
    "species_gene = [\"Colias_croceus,XM_045638192.1\",\"Zerene_cesonia,XM_038367126.1\"]\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\"\n",
    "annotated_species_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/15.Timeout_Exon_Analysis/1.Annotated Species\"\n",
    "try:\n",
    "    os.mkdir(annotated_species_location)\n",
    "except:\n",
    "    pass\n",
    "for entries in species_gene:\n",
    "\n",
    "    species,gene_name = entries.split(\",\")\n",
    "    print(species,gene_name)\n",
    "\n",
    "    genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    query_species = species\n",
    "    create_query_folder(annotated_species_location,gene_name,species)\n",
    "    list_of_files = os.listdir(f\"{genome_location}/{species}\")\n",
    "    # print(list_of_files)\n",
    "    for file_names in list_of_files:\n",
    "        if file_names.endswith(\"_genomic.gff\"):\n",
    "            gff3_file = file_names\n",
    "\n",
    "    with open(f\"{genome_location}/{species}/{gff3_file}\") as gff_file_open:\n",
    "        gff_file_list = gff_file_open.readlines()\n",
    "    complement = ''\n",
    "    exon_count = 1\n",
    "    gene_found = 0\n",
    "    exon_dictionary = {}\n",
    "    for line in gff_file_list:\n",
    "        if gene_name in line:\n",
    "            if \"CDS\" in line.split(\"\\t\"):\n",
    "\n",
    "    #             print(line)\n",
    "                gene_found = 1\n",
    "                if complement == '':\n",
    "                    complement = line.split(\"\\t\")[6]\n",
    "                exon_start = line.split(\"\\t\")[3]\n",
    "                exon_stop = line.split(\"\\t\")[4]\n",
    "                scaffold = line.split(\"\\t\")[0]\n",
    "                exon_dictionary[f\"Exon_{exon_count}\"] = [exon_start,exon_stop]\n",
    "                exon_count += 1\n",
    "            if \"exon\" in line.split(\"\\t\") and gene_found == 1:\n",
    "                break\n",
    "    print(complement)\n",
    "    print(exon_dictionary)\n",
    "    if complement == \"+\":\n",
    "        exon_counter = 1\n",
    "    elif complement == \"-\":\n",
    "        exon_counter = len(exon_dictionary)\n",
    "    else:\n",
    "        print(\"Error!!\")\n",
    "        assert False\n",
    "\n",
    "    # list_of_query_species = os.listdir(query_location)\n",
    "    # if query_species not in list_of_query_species:\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # if \"Transcripts\" not\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\n",
    "\n",
    "    print_out = ''\n",
    "    print(\"exon counter\", exon_counter)        \n",
    "    for i in range(1,len(exon_dictionary)+1):\n",
    "\n",
    "        exon_start, exon_stop = exon_dictionary[f\"Exon_{exon_counter}\"]\n",
    "    #     print(exon_start,exon_stop)\n",
    "        subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{exon_start}-{exon_stop} > \"{genome_location}/{species}/Temp/temp_gene.fa\"', shell = True)\n",
    "        exon_sequence_file = SeqIO.parse(f\"{genome_location}/{species}/Temp/temp_gene.fa\", \"fasta\")\n",
    "        sequence = ''\n",
    "\n",
    "        for sequences in exon_sequence_file:\n",
    "    #         print(sequences.id)\n",
    "            if complement == \"-\":\n",
    "                sequence = sequences.seq.reverse_complement() \n",
    "\n",
    "    #             print(sequences.seq.reverse_complement().translate())\n",
    "            else:\n",
    "                sequence = sequences.seq\n",
    "    #             print(sequences.seq)\n",
    "    #             print(sequences.seq.translate())\n",
    "            print_out += str(len(sequence))+\"\\t\"\n",
    "            output = f\">{query_species}_{gene_name}_Exon{exon_counter}\\n{sequence}\"\n",
    "        with open(f\"{annotated_species_location}/{query_species}/Transcripts/{gene_name}/Exon{exon_counter}.fa\", 'w') as outfile:\n",
    "            outfile.write(output)\n",
    "        print(output)\n",
    "        if complement == \"-\":\n",
    "            exon_counter -= 1\n",
    "        else:\n",
    "            exon_counter += 1\n",
    "\n",
    "    #         if gene_location in line.split(\"\\t\"):\n",
    "    #           print(line)\n",
    "    #           break\n",
    "    print(print_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "111273de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0002681f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
