{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3795c17b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1826b335",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import io"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1e064f82",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "69f4b3dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# list_of_species = os.listdir(\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\")\n",
    "list_of_species = [\"Hestina assimilis\",\"Charaxes varanes\",\"Danaus chrysippus\",\"Danaus plexippus\",\"Dircenna loreta\",\"Mechanitis mazaeus\",\"Mechanitis messenoides\",\"Melinaea marsaeus rileyi\",\"Melinaea menophilus\",\"Napeogenes inachia\",\"Napeogenes sylphis\",\"Argynnis bischoffii washingtonia\",\"Boloria euphrosyne\",\"Boloria selene\",\"Brenthis daphne\",\"Brenthis hecate\",\"Brenthis ino\",\"Dryadula phaetusa\",\"Dryas iulia moderata\",\"Eueides isabella\",\"Fabriciana adippe\",\"Heliconius charithonia\",\"Heliconius nattereri\",\"Heliconius sara\",\"Philaethria dido\",\"Limenitis camilla\",\"Euphydryas editha\",\"Hypolimnas misippus\",\"Melitaea cinxia\",\"Mellicta athalia\",\"Nymphalis c-album\",\"Nymphalis io\",\"Nymphalis polychloros\",\"Nymphalis urticae\",\"Vanessa atalanta\",\"Vanessa cardui\",\"Vanessa tameamea\"]\n",
    "query_species = \"Danaus_plexippus\"\n",
    "family_group = \"6.Heliconiinae_Danainae_Nymphalinae\"\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Query\"\n",
    "query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "if \"desktop.ini\" in query_transcript_list:\n",
    "    query_transcript_list.remove(\"desktop.ini\")\n",
    "\n",
    "query_transcript = query_transcript_list[0]\n",
    "# list_of_species.remove(query_species)\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "list_of_genomes = os.listdir(\"/mnt/f/Genomes_2023-12-26\")\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Query/{query_species}/{query_transcript}/query_protein.fa\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4dee5727",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/mnt/f/Genomes_2023-12-26/Melinaea_menophilus_n._ssp._AW-2005'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 25\u001b[0m\n\u001b[1;32m     17\u001b[0m     genome_location \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/mnt/f/Genomes_2023-12-26\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     18\u001b[0m \u001b[38;5;66;03m#     for genome in list_of_genomes:\u001b[39;00m\n\u001b[1;32m     19\u001b[0m \u001b[38;5;66;03m#         if species == genome:\u001b[39;00m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;66;03m#             files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     23\u001b[0m \u001b[38;5;66;03m#                     genome_file = file\u001b[39;00m\n\u001b[1;32m     24\u001b[0m \u001b[38;5;66;03m#                     break\u001b[39;00m\n\u001b[0;32m---> 25\u001b[0m     genome_file \u001b[38;5;241m=\u001b[39m \u001b[43mget_genome_file\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     27\u001b[0m     output \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mecho \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mtblastn -seg no -query \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquery_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m -db \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/mnt/f/Genomes_2023-12-26/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mgenome_file\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m -num_alignments 3 -out \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mblast_output_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_blast_out.htm\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m -html \u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m tblastn -seg no -query \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquery_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m -db \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/mnt/f/Genomes_2023-12-26/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mgenome_file\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m -num_alignments 3 -out \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mblast_output_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_blast_out.txt\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m     29\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m io\u001b[38;5;241m.\u001b[39mopen(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfamily_group\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/tblastn_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquery_species\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.sh\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m\"\u001b[39m, newline\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m outfile:\n",
      "Cell \u001b[0;32mIn[3], line 2\u001b[0m, in \u001b[0;36mget_genome_file\u001b[0;34m(genome_location, species)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_genome_file\u001b[39m(genome_location,species):\n\u001b[0;32m----> 2\u001b[0m     list_of_files_in_genome_folder \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlistdir\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgenome_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m list_of_files_in_genome_folder:\n\u001b[1;32m      4\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m file\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_genomic.fna\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/mnt/f/Genomes_2023-12-26/Melinaea_menophilus_n._ssp._AW-2005'"
     ]
    }
   ],
   "source": [
    "genome_file = ''\n",
    "output = ''\n",
    "for species in list_of_species:\n",
    "    species = species.replace(\" \", \"_\")\n",
    "    # if \"\n",
    "    # species = species_folder.split(\".\")[1]\n",
    "    blast_output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    try:\n",
    "        os.mkdir(blast_output_location)\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "    blast_output_folders = os.listdir(blast_output_location)\n",
    "    if species not in blast_output_folders:\n",
    "        os.mkdir(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Blast_result/{species}\")\n",
    "    \n",
    "    genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "#     for genome in list_of_genomes:\n",
    "#         if species == genome:\n",
    "#             files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\n",
    "#             for file in files_in_genome_folder:\n",
    "#                 if re.search('_genomic.fna$', file):\n",
    "#                     genome_file = file\n",
    "#                     break\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    output += f'echo {species}\\ntblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.htm\" -html \\n\\n tblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.txt\"\\n\\n'\n",
    "    \n",
    "with io.open(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/tblastn_{query_species}.sh\", \"w\", newline='\\n') as outfile:\n",
    "    outfile.write(output)\n",
    "#             print(\"here\",genome)\n",
    "#     print(species_folder)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00391970",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
