{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5bbf344f-f378-45b2-81af-6a50b8999644",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Heliconius_charithonia-period', 'Napeogenes_sylphis-period', 'Mechanitis_messenoides-period', 'Mechanitis_mazaeus-period', 'Aricia_artaxerxes-period', 'Aricia_agestis-period', 'Polyommatus_iphigenia-period', 'Phengaris_arion-period', 'Celastrina_argiolus-period', 'Cyaniris_semiargus-period', 'Heliconius_nattereri-period', 'Dircenna_loreta-period', 'Eueides_isabella-period', 'Melitaea_cinxia-period', 'Lasiommata_megera-period', 'Pararge_aegeria-period', 'Danaus_chrysippus-period', 'Nymphalis_c-album-period', 'Coenonympha_glycerion-period', 'Maniola_hyperantus-period', 'Brenthis_hecate-period', 'Fabriciana_adippe-period', 'Nymphalis_urticae-period', 'Calycopis_cecrops-period', 'Hypolimnas_misippus-period', 'Hipparchia_semele-period', 'Maniola_jurtina-period', 'Melanargia_galathea-period', 'Mellicta_athalia-period', 'Oeneis_ivallda-period', 'Philaethria_dido-period', 'Napeogenes_inachia-period', 'Euphydryas_editha-period', 'Nymphalis_polychloros-period', 'Melinaea_menophilus-period', 'Polyommatus_icarus-period', 'Boloria_selene-period', 'Leptidea_juvernica-period', 'Lysandra_bellargus-period', 'Leptophobia_aripa-period', 'Leptidea_sinapis-period', 'Bicyclus_anynana-period', 'Vanessa_cardui-period', 'Brenthis_daphne-period', 'Papilio_demoleus-period', 'Zerene_cesonia-period', 'Parnassius_glacialis-period', 'Ornithoptera_alexandrae-period', 'Pieris_melete-period', 'Troides_oblongomaculatus-period', 'Glaucopsyche_alexis-period', 'Plebejus_argus-period', 'Lysandra_coridon-period', 'Nymphalis_io-period', 'Phoebis_sennae-period', 'Dryadula_phaetusa-period', 'Erebia_aethiops-period', 'Leptidea_reali-period', 'Pieris_mannii-period', 'Papilio_glaucus-period', 'Argynnis_bischoffii_washingtonia-period', 'Colias_nastes-period', 'Vanessa_atalanta-period', 'Aporia_crataegi-period', 'Battus_philenor-period', 'Papilio_dardanus_tibullus-period', 'Papilio_elwesi-period', 'Vanessa_tameamea-period', 'Boloria_euphrosyne-period', 'Brenthis_ino-period', 'Melinaea_marsaeus_rileyi-period', 'Papilio_protenor-period', 'Erebia_ligea-period', 'Pieris_napi-period', 'Dryas_iulia_moderata-period', 'Pieris_brassicae-period', 'Heliconius_sara-period', 'Anthocharis_cardamines-period', 'Eurema_hecabe-period', 'Danaus_plexippus-period', 'Ornithoptera_priamus-period', 'Troides_aeacus-period', 'Colias_eurytheme-period', 'Papilio_xuthus-period', 'Papilio_machaon-period', 'Colias_croceus-period', 'Pieris_rapae-period', 'Papilio_memnon-period', 'Teinopalpus_imperialis-period', 'Papilio_polytes-period']\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/11.Alignment\"\n",
    "\n",
    "fasta_file = SeqIO.parse(f\"{location}/period.masceDNA.fas\", 'fasta')\n",
    "\n",
    "list_of_sequences_ids = []\n",
    "\n",
    "for records in fasta_file:\n",
    "    list_of_sequences_ids.append(records.id)\n",
    "print(list_of_sequences_ids)\n",
    "# list_of_folders = os.listdir(f\"{location}\")\n",
    "# print(list_of_folders)\n",
    "# if \"00.Temp_alignment\" in list_of_folders:\n",
    "#     print(\"AA\")\n",
    "#     subprocess.run(f'rm -r \"{location}/00.Temp_alignment/\"', shell = True)\n",
    "\n",
    "# os.mkdir(f\"{location}/00.Temp_alignment\")\n",
    "\n",
    "\n",
    "for i in range(len(list_of_sequences_ids)):\n",
    "    output = ''\n",
    "    fasta_file = SeqIO.parse(f\"{location}/period.masceDNA.fas\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        sequence = f\">{records.id}\\n{records.seq}\\n\"\n",
    "        if list_of_sequences_ids[i] == records.id:\n",
    "            output = sequence + output\n",
    "        else:\n",
    "            output += sequence\n",
    "    with open(f\"{location}/00.Temp_alignment_phyloP/{list_of_sequences_ids[i]}.fas\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "    # subprocess.run(f'cd \"{location}\"\\nphyloFit --tree period.masceDNA.fas.treefile --subst-mod REV --out-root \"./02.Temp_phyloP/modelfile\" \"./00.Temp_alignment/{list_of_sequences_ids[i]}.fas\"', shell = True)\n",
    "    subprocess.run(f'cd \"{location}/02.Temp_phyloP\"\\nphyloP --mode CONACC --method LRT --wig-scores ../modelfile.mod ../00.Temp_alignment_phyloP/{list_of_sequences_ids[i]}.fas> phyloPscores.wig', shell = True)\n",
    "    \n",
    "    fasta_file = SeqIO.parse(f\"{location}/00.Temp_alignment_phyloP/{list_of_sequences_ids[i]}.fas\", 'fasta')\n",
    "\n",
    "    for records in fasta_file:\n",
    "        sequence = records.seq\n",
    "        sequence_name = records.id\n",
    "        break\n",
    "    \n",
    "    with open(f\"{location}/02.Temp_phyloP/phyloPscores.wig\", 'r') as score_file:\n",
    "        scores = score_file.readlines()\n",
    "    \n",
    "    output = 'Pos\\tScore'\n",
    "    score_counter = 1\n",
    "    for j in range(len(sequence)):\n",
    "        if sequence[j] != \"-\":\n",
    "            output += f\"\\n{j+1}\\t{scores[score_counter].strip()}\"\n",
    "            score_counter += 1\n",
    "        else:\n",
    "            output += f\"\\n{j+1}\\tN/A\"\n",
    "    \n",
    "    # print(output)\n",
    "    with open(f\"{location}/02.Temp_phyloP/scores_fixed_{list_of_sequences_ids[i]}.wig\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "        # assert False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f208b92c-0d6e-4f68-a2fe-aa5c7bcb4539",
   "metadata": {},
   "outputs": [],
   "source": [
    "scores = []\n",
    "for i in range(len(list_of_sequences_ids)):\n",
    "    with open(f\"{location}/02.Temp_phyloP/scores_fixed_{list_of_sequences_ids[i]}.wig\", 'r') as score_file:\n",
    "        scores.append(score_file.readlines())\n",
    "output = \"Pos\\tScore\"\n",
    "for i in range(1,len(scores[0])):\n",
    "    data_counter = 0\n",
    "    average = 0\n",
    "    for j in range(len(list_of_sequences_ids)):\n",
    "        if scores[j][i].strip().split(\"\\t\")[1] != \"N/A\":\n",
    "            # print ((scores[j][i]).strip())\n",
    "            average += float(scores[j][i].strip().split(\"\\t\")[1])\n",
    "            data_counter += 1\n",
    "    average = round(average / data_counter,3)\n",
    "    output += f\"\\n{i}\\t{average}\"\n",
    "    # print(i,average)\n",
    "        \n",
    "    # break\n",
    "# print(output)\n",
    "with open(f\"{location}/scores_phyloP.wig\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a6a5af4d-1044-4d40-8723-545b4f7d8f87",
   "metadata": {},
   "outputs": [],
   "source": [
    "output_split  = output.split(\"\\n\")\n",
    "output2 = \"Site\\tRate\"\n",
    "counter = 1\n",
    "site_number = 1\n",
    "while counter < len(output_split):\n",
    "    new_rate = (float(output_split[counter].split(\"\\t\")[1]) + float(output_split[counter+1].split(\"\\t\")[1]) + float(output_split[counter+2].split(\"\\t\")[1]))/3\n",
    "    output2 += f\"\\n{site_number}\\t{new_rate}\"\n",
    "    site_number += 1\n",
    "    counter += 3\n",
    "with open(f\"{location}/02.phyloP/scores_phyloP_codon.txt\", 'w') as out_file:\n",
    "  out_file.write(output2)\n",
    "# print(output2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a16865d-bbae-4349-b996-aa4b188c0f3d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
