{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ca9450f9-ef13-41b6-b314-7d7f064f8bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='cd \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/11.Alignment//05.InterProScan\"\\n~/pfam_scan/pfam_scan.py -out ./domains_unaligned.csv ./period.masceDNA_without_STOP_translated_test.fas ~/pfam/', returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/11.Alignment/\"\n",
    "\n",
    "subprocess.run(f'cd \"{location}/05.InterProScan\"\\n~/pfam_scan/pfam_scan.py -out ./domains_unaligned.csv ./period.masceDNA_without_STOP_translated_test.fas ~/pfam/', shell = True)\n",
    "\n",
    "\n",
    "# output = ''\n",
    "# with open(\"/mnt/g/My Drive/Circadian Rhythm review/01.Period/06.Sequences from paper accession/04.Cry2/domains_unaligned.csv\", 'r') as domain_position_file:\n",
    "#     domain_position_list = domain_position_file.readlines()\n",
    "# output = ''\n",
    "# for i in range(1, len(domain_position_list)):\n",
    "#     domain_position_split = (domain_position_list[i]).split(\",\")\n",
    "#     print(domain_position_list[i])\n",
    "    \n",
    "#     transcript_id = domain_position_split[0]\n",
    "    \n",
    "#     alignment_file = SeqIO.parse(\"/mnt/g/My Drive/Circadian Rhythm review/01.Period/06.Sequences from paper accession/04.Cry2/for_domain_search.fa\", 'fasta')\n",
    "#     seq_name, sequence = '',''\n",
    "#     for records in alignment_file:\n",
    "#         print\n",
    "#         if transcript_id in records.id:\n",
    "    \n",
    "#             seq_name, sequence = records.id, records.seq\n",
    "#             break\n",
    "#     residue_number = 0\n",
    "#     seq_counter = 0\n",
    "#     domain_start = int(domain_position_split[1])\n",
    "#     domain_stop = int(domain_position_split[2])\n",
    "#     # print(domain_start, domain_stop) \n",
    "#     while True:\n",
    "#         # print(seq_counter, residue_number)\n",
    "#         current_residue = sequence[residue_number]\n",
    "#         if current_residue != \"X\": \n",
    "#             seq_counter += 1\n",
    "#             if domain_start == seq_counter:\n",
    "#                 domain_position_list[i] = domain_position_list[i].replace(f\",{domain_start},\",f\",{residue_number},\")\n",
    "#                 # print(\"\\n\",domain_position_list[i])\n",
    "#                 # domain_start = residue_number\n",
    "#         # print(seq_counter)\n",
    "#         if seq_counter == domain_stop:\n",
    "#             domain_position_list[i] = domain_position_list[i].replace(f\",{domain_stop},\",f\",{residue_number},\")\n",
    "#             # domain_stop = residue_number\n",
    "#             break\n",
    "    \n",
    "#         residue_number += 1\n",
    "    \n",
    "#     # print(domain_start, domain_stop) \n",
    "#     print(domain_position_list[i])\n",
    "# with open(\"/mnt/g/My Drive/Circadian Rhythm review/01.Period/06.Sequences from paper accession/04.Cry2/domains.csv\", 'w') as out_file:\n",
    "#     out_file.write(\"\".join(domain_position_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96fade2a-8f89-4b90-a670-ff595c73c0a5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
