{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "16c8a0c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b2a20a4d",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Papilio_polytes LOC106110082\n",
      "\n",
      "17.Papilio_polytes\n",
      "Bombyx_mandarina LOC114246916\n",
      "\n",
      "18.Bombyx_mandarina\n",
      "Trichoplusia_ni LOC113493972\n",
      "\n",
      "19.Trichoplusia_ni\n",
      "Vanessa_tameamea LOC113403059\n",
      "\n",
      "20.Vanessa_tameamea\n",
      "Amyelois_transitella LOC106130500\n",
      "\n",
      "21.Amyelois_transitella\n",
      "Pieris_brassicae LOC123718685\n",
      "\n",
      "22.Pieris_brassicae\n",
      "Vanessa_atalanta LOC125076188\n",
      "\n",
      "23.Vanessa_atalanta\n",
      "Spodoptera_frugiperda LOC118261937\n",
      "\n",
      "24.Spodoptera_frugiperda\n",
      "Pieris_napi LOC125062481\n",
      "\n",
      "25.Pieris_napi\n",
      "Vanessa_cardui LOC124543195\n",
      "\n",
      "26.Vanessa_cardui\n",
      "Maniola_hyperantus LOC117995700\n",
      "\n",
      "27.Maniola_hyperantus\n",
      "Maniola_jurtina LOC123880638\n",
      "\n",
      "28.Maniola_jurtina\n",
      "Nymphalis_io LOC126780285\n",
      "\n",
      "29.Nymphalis_io\n",
      "Leptidea_sinapis LOC126979027\n",
      "\n",
      "30.Leptidea_sinapis\n",
      "Leguminivora_glycinivorella LOC125240683\n",
      "\n",
      "31.Leguminivora_glycinivorella\n",
      "Colias_croceus LOC123705108\n",
      "\n",
      "32.Colias_croceus\n",
      "Pararge_aegeria LOC120636016\n",
      "\n",
      "33.Pararge_aegeria\n",
      "Achroia_grisella LOC131841862\n",
      "\n",
      "35.Achroia_grisella_2\n",
      "Melitaea_cinxia LOC123668776\n",
      "\n",
      "36.Melitaea_cinxia\n",
      "Aricia_agestis LOC121739130\n",
      "\n",
      "37.Aricia_agestis\n",
      "Papilio_machaon LOC106719908\n",
      "\n",
      "38.Papilio_machaon\n",
      "Ostrinia_furnacalis LOC114365005\n",
      "\n",
      "43.Ostrinia_furnacalis\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import os\n",
    "\n",
    "location = \"J:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period\"\n",
    "list_of_species = os.listdir(f\"{location}/0.Species\")\n",
    "# gene = \"LOC106113722\"\n",
    "# species = \"Papilio xuthus\".replace(\" \", \"_\")\n",
    "species_folder = ''\n",
    "\n",
    "with open(f\"{location}/gene_detail.txt\", 'r') as gene_file:\n",
    "    gene_list = gene_file.readlines()\n",
    "\n",
    "for i in range(1, len(gene_list)):\n",
    "    gene_list_split = gene_list[i].split(\"\\t\")\n",
    "    species = gene_list_split[0].replace(\" \", \"_\")\n",
    "    gene = gene_list_split[2]\n",
    "    print(species, gene)\n",
    "#     \n",
    "#     print (gene_list[i])\n",
    "\n",
    "    for folder_name in list_of_species:\n",
    "        if species in folder_name:\n",
    "            species_folder = folder_name\n",
    "            print(folder_name)\n",
    "            break\n",
    "\n",
    "    url=f'https://www.ncbi.nlm.nih.gov/gene/?term={gene}&report=gene_table&format=text'\n",
    "    r = requests.get(url)\n",
    "    soup = BeautifulSoup(r.text, \"xml\")\n",
    "#     print(soup.text)\n",
    "\n",
    "    with open(f\"{location}/0.Species/{species_folder}/Gene_table.txt\", 'w') as out_file:\n",
    "        out_file.write(soup.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "76534cba",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.Bombyx_mori\n",
      "26\n",
      "26\n",
      "24\n",
      "26\n",
      "25\n",
      "25\n",
      "24\n",
      "24\n",
      "24\n",
      "25\n",
      "2.Galleria_mellonella\n",
      "28\n",
      "27\n",
      "28\n",
      "3.Spodoptera_litura\n",
      "29\n",
      "29\n",
      "28\n",
      "30\n",
      "29\n",
      "29\n",
      "27\n",
      "28\n",
      "28\n",
      "28\n",
      "29\n",
      "29\n",
      "30\n",
      "26\n",
      "43.Ostrinia_furnacalis\n",
      "27\n",
      "26\n",
      "27\n",
      "27\n",
      "25\n",
      "25\n",
      "25\n",
      "32.Colias_croceus\n",
      "27\n",
      "27\n",
      "26\n",
      "31.Leguminivora_glycinivorella\n",
      "26\n",
      "26\n",
      "26\n",
      "30.Leptidea_sinapis\n",
      "26\n",
      "25\n",
      "26\n",
      "29.Nymphalis_io\n",
      "28\n",
      "27\n",
      "28\n",
      "28.Maniola_jurtina\n",
      "27\n",
      "27\n",
      "27\n",
      "26\n",
      "24\n",
      "27.Maniola_hyperantus\n",
      "26.Vanessa_cardui\n",
      "29\n",
      "29\n",
      "25\n",
      "27\n",
      "28\n",
      "28\n",
      "25.Pieris_napi\n",
      "26\n",
      "27\n",
      "21\n",
      "27\n",
      "27\n",
      "26\n",
      "24.Spodoptera_frugiperda\n",
      "30\n",
      "30\n",
      "29\n",
      "30\n",
      "29\n",
      "27\n",
      "27\n",
      "29\n",
      "27\n",
      "30\n",
      "27\n",
      "28\n",
      "27\n",
      "27\n",
      "28\n",
      "29\n",
      "29\n",
      "29\n",
      "30\n",
      "27\n",
      "28\n",
      "28\n",
      "29\n",
      "30\n",
      "26\n",
      "27\n",
      "28\n",
      "27\n",
      "28\n",
      "29\n",
      "25\n",
      "27\n",
      "23.Vanessa_atalanta\n",
      "27\n",
      "28\n",
      "28\n",
      "27\n",
      "28\n",
      "28\n",
      "29\n",
      "29\n",
      "29\n",
      "29\n",
      "28\n",
      "29\n",
      "29\n",
      "29\n",
      "23\n",
      "22.Pieris_brassicae\n",
      "27\n",
      "26\n",
      "27\n",
      "26\n",
      "27\n",
      "27\n",
      "26\n",
      "27\n",
      "27\n",
      "27\n",
      "21.Amyelois_transitella\n",
      "24\n",
      "24\n",
      "24\n",
      "20.Vanessa_tameamea\n",
      "28\n",
      "19.Trichoplusia_ni\n",
      "29\n",
      "28\n",
      "27\n",
      "28\n",
      "28\n",
      "28\n",
      "26\n",
      "29\n",
      "28\n",
      "26\n",
      "28\n",
      "18.Bombyx_mandarina\n",
      "26\n",
      "26\n",
      "25\n",
      "23\n",
      "23\n",
      "24\n",
      "24\n",
      "38.Papilio_machaon\n",
      "37.Aricia_agestis\n",
      "36.Melitaea_cinxia\n",
      "33.Pararge_aegeria\n",
      "26\n",
      "17.Papilio_polytes\n",
      "26\n",
      "27\n",
      "27\n",
      "27\n",
      "26\n",
      "27\n",
      "25\n",
      "24\n",
      "16.Papilio_xuthus\n",
      "27\n",
      "25\n",
      "27\n",
      "27\n",
      "15.Hyposmocoma_kahamanoa\n",
      "27\n",
      "27\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "25\n",
      "14.Zerene_cesonia\n",
      "27\n",
      "26\n",
      "27\n",
      "27\n",
      "13.Danaus_plexippus\n",
      "26\n",
      "25\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "12.Manduca_sexta\n",
      "27\n",
      "27\n",
      "27\n",
      "25\n",
      "26\n",
      "27\n",
      "25\n",
      "26\n",
      "27\n",
      "10.Plutella_xylostella\n",
      "28\n",
      "28\n",
      "9.Pieris_rapae\n",
      "27\n",
      "26\n",
      "27\n",
      "8.Plodia_interpunctella\n",
      "25\n",
      "25\n",
      "25\n",
      "7.Helicoverpa_zea\n",
      "26\n",
      "28\n",
      "27\n",
      "28\n",
      "26\n",
      "25\n",
      "28\n",
      "28\n",
      "27\n",
      "26\n",
      "28\n",
      "27\n",
      "6.Pectinophora_gossypiella\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "26\n",
      "5.Bicyclus_anynana\n",
      "27\n",
      "26\n",
      "26\n",
      "27\n",
      "26\n",
      "26\n",
      "4.Helicoverpa_armigera\n",
      "27\n",
      "26\n",
      "27\n",
      "27\n",
      "27\n",
      "25\n",
      "26\n",
      "26\n",
      "27\n",
      "27\n",
      "24\n",
      "23\n"
     ]
    }
   ],
   "source": [
    "import re \n",
    "\n",
    "\n",
    "location = \"J:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period\"\n",
    "list_of_species = os.listdir(f\"{location}/0.Species\")\n",
    "list_of_species.remove(\"names\")\n",
    "list_of_species.remove(\"desktop.ini\")\n",
    "species_folder = ''\n",
    "exon_list = \"Exon1,Exon2,Exon3,Exon4,Exon5,Exon6,Exon7,Exon8,Exon9,Exon10,Exon11,Exon12,Exon13,Exon14,Exon15,Exon16,Exon17,Exon18,Exon19,Exon20,Exon21,Exon22,Exon23,Exon24,Exon25,Exon26,Exon27,Exon28,Exon29,Exon30,Exon31,Exon32,Exon33,Exon34\".split(\",\")\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    with open(f\"{location}/0.Species/{species}/Gene_table.txt\",\"r\") as gene_table_file:\n",
    "        gene_table_list = gene_table_file.readlines()\n",
    "#         print(gene_table_list)\n",
    "\n",
    "    \n",
    "\n",
    "    i = 0\n",
    "    switch = 0\n",
    "    output = \"mRNA Accession\\tExon Number\\tCoding Length\\tGenome Start Point\\tGenome Stop Point\\n\" \n",
    "    output2 = \"mRNA Accession\\tProtein Accession\\tExon Number\\tAlignment Score with Bombyx mori\\n\"\n",
    "    exon_counter = 0\n",
    "\n",
    "    while i < (len(gene_table_list)):\n",
    "\n",
    "        if \"Exon table for  mRNA  \" in gene_table_list[i]:\n",
    "            gene_table_line = gene_table_list[i]\n",
    "            while re.search(\"[ ]{2,}\", gene_table_line):\n",
    "                gene_table_line = gene_table_line.replace(\"  \",\" \")\n",
    "\n",
    "            mrna_accession = gene_table_line.split(\" \")[4]\n",
    "        \n",
    "            switch = 1\n",
    "            i += 3\n",
    "        if switch == 1:\n",
    "\n",
    "            gene_table_line = gene_table_list[i]\n",
    "            while re.search(\"[\\t]{2,}\", gene_table_line):\n",
    "                gene_table_line = gene_table_line.replace(\"\\t\\t\",\"\\t\")\n",
    "            if len(gene_table_line.split(\"\\t\")) == 7 or len(gene_table_line.split(\"\\t\")) == 6:\n",
    "\n",
    "                coding_exon_length =  ((gene_table_line.split(\"\\t\")[5]))\n",
    "                genome_start_point,genome_end_point = ((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "#                 print((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "                output += f\"{mrna_accession}\\t{exon_list[exon_counter]}\\t{coding_exon_length.rstrip()}\\t{genome_start_point}\\t{genome_end_point}\\n\"\n",
    "            \n",
    "        \n",
    "                exon_counter += 1\n",
    "\n",
    "            if switch == 1 and re.search(\"^mRNA\",gene_table_list[i]):\n",
    "                switch = 0\n",
    "                print(exon_counter)\n",
    "                output += \"\\n\\n\"\n",
    "                exon_counter = 0\n",
    "\n",
    "        i += 1\n",
    "\n",
    "    with open(f\"{location}/0.Species/{species}/Gene_table_processed.tsv\",'w') as out_file:\n",
    "        out_file.write(output)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f3fe19b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6322ec9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
