{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2948fa21-144a-459a-8838-5b897826c2e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rps5\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "wg\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "rps2\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "mdh\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "idh\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "ddc\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "gapdh\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "ef1\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "hcl\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "cat\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n",
      "cad\n",
      "3.Satyrine\n",
      "4.Pierinae\n",
      "5.Coliadinae\n",
      "6.Heliconiinae_Danainae_Nymphalinae\n",
      "7.Papilionidae\n",
      "8.Lycaenidae\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny\"\n",
    "family_group_list = [\"3.Satyrine\",\"4.Pierinae\",\"5.Coliadinae\",\"6.Heliconiinae_Danainae_Nymphalinae\",\"7.Papilionidae\",\"8.Lycaenidae\"]\n",
    "\n",
    "gene_list = os.listdir(f\"{location}/3.Satyrine/3.Extracted Sequences/\")\n",
    "if \"desktop.ini\" in gene_list:\n",
    "    gene_list.remove(\"desktop.ini\")\n",
    "\n",
    "for gene_name in gene_list:\n",
    "    print(gene_name)\n",
    "    output = ''\n",
    "    for family_group in family_group_list:\n",
    "        print(family_group)\n",
    "        list_of_species_files = os.listdir(f\"{location}/{family_group}/3.Extracted Sequences/{gene_name}\")\n",
    "        if \"desktop.ini\" in list_of_species_files:\n",
    "            list_of_species_files.remove(\"desktop.ini\")\n",
    "        for species_name in list_of_species_files:\n",
    "            # print(species_name)\n",
    "            fasta_file= SeqIO.parse(f\"{location}/{family_group}/3.Extracted Sequences/{gene_name}/{species_name}\", 'fasta')\n",
    "            for sequences in fasta_file:\n",
    "                output += f\">{sequences.id.replace(gene_name, '')[:-1]}\\n{sequences.seq}\\n\"\n",
    "    with open(f\"{location}/10.All_gene_sequences/{gene_name}.fas\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "    # print(output)\n",
    "    # break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8ae2a607-5b42-45b6-a95b-61260e265671",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'aligned_file' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[20], line 11\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m file_name \u001b[38;5;129;01min\u001b[39;00m list_of_files:\n\u001b[1;32m      9\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m file_name\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124maligned.fas.best.fas\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m     10\u001b[0m         \u001b[38;5;66;03m# print(file_name)\u001b[39;00m\n\u001b[0;32m---> 11\u001b[0m         \u001b[43maligned_file\u001b[49m\u001b[38;5;241m.\u001b[39mappend(file_name)\n\u001b[1;32m     12\u001b[0m         fasta_file \u001b[38;5;241m=\u001b[39m SeqIO\u001b[38;5;241m.\u001b[39mparse(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/10.All_gene_sequences/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile_name\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     13\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m records \u001b[38;5;129;01min\u001b[39;00m fasta_file:\n",
      "\u001b[0;31mNameError\u001b[0m: name 'aligned_file' is not defined"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny\"\n",
    "\n",
    "list_of_files = os.listdir(f\"{location}/10.All_gene_sequences\")\n",
    "list_of_species = []\n",
    "aligned_files = []\n",
    "for file_name in list_of_files:\n",
    "    if file_name.endswith(\"aligned.fas.best.fas\"):\n",
    "        # print(file_name)\n",
    "        aligned_files.append(file_name)\n",
    "        fasta_file = SeqIO.parse(f\"{location}/10.All_gene_sequences/{file_name}\", \"fasta\")\n",
    "        for records in fasta_file:\n",
    "            if records.id not in list_of_species:\n",
    "                list_of_species.append(records.id)\n",
    "output = ''\n",
    "for species in list_of_species:\n",
    "    full_sequence = ''\n",
    "    for file_name in aligned_files:\n",
    "        sequence = ''\n",
    "        sequence_length = ''\n",
    "        fasta_file = SeqIO.parse(f\"{location}/10.All_gene_sequences/{file_name}\", \"fasta\")\n",
    "        for records in fasta_file:\n",
    "            sequence_length = len(records.seq)\n",
    "            if records.id == species:\n",
    "                sequence = records.seq\n",
    "        if sequence == '':\n",
    "            print(file_name,\"\\t\\t\",species, \"Missing\")\n",
    "            sequence = \"N\"*sequence_length\n",
    "        full_sequence += sequence\n",
    "    print(species, len(full_sequence))\n",
    "    output += f\">{species}\\n{full_sequence}\\n\\n\"\n",
    "\n",
    "with open(f\"{location}/11.Combined_again/combined_file.fas\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81695ee1-bfcf-4e43-bcff-d593123ecdfa",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
