{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "36d22ad3-a85e-44b3-85e6-fc9cb5b48684",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_blast_with_new_query(annotated_genome_location,\n",
    "                             annotated_species_name,\n",
    "                             error_exon, \n",
    "                            query_species,\n",
    "                             genome_location, \n",
    "                             species):\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\"\n",
    "    genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "    out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "\n",
    "    with open(f\"{out_location}/new_query.txt\", 'r') as query_file_open:\n",
    "        query_file_lines = query_file_open.readlines()\n",
    "    if (len(query_file_lines)) == 1:\n",
    "        return (\"Query_error\")\n",
    "    print(len(query_file_lines[1])) \n",
    "    if (len(query_file_lines[1])) < 5:\n",
    "        return (\"Query_error\")\n",
    "    # print(len(query_file_lines[1]))\n",
    "    # genome_fragment_out = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/local_genomic_fragment.fa\"\n",
    "    # with io.open(genome_fragment_out,'w') as out_file:\n",
    "    #     output = f\">Genome_fragment_{error_exon}\\n{gene_sequence}\"\n",
    "    #     out_file.write(output)\n",
    "\n",
    "    local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "    # cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "    # # os.system(f'{cd_command}')\n",
    "    # subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # # print(mkdb_command)\n",
    "    # # os.system(f'{mkdb_command}')\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.htm -html'\n",
    "    # print(blast_command)\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # subprocess.run(f'{blast_command}', shell = True)\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.txt'\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6e7d0eb2-6d4d-44b8-90ab-76c864362837",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bd69b123-ff19-4205-83ee-039f63b0fc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_query_info(query_location,query_species,query_transcript,query_exon):\n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "    return(left_overhang,right_overhang,original_query_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "37864dc3-b190-4ffc-bd3c-ec61c8d7d4d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                       annotated_species_name,\n",
    "                                       error_exon,\n",
    "                                       left_overhang,\n",
    "                                       right_overhang,                                                                                               \n",
    "                                       original_query_name,\n",
    "                                      query_species_original ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "    with open(f\"{blast_location}/new_query.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out_genome_fragment.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "        genome_file = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa\", 'fasta')\n",
    "        print(\"reached here\")\n",
    "        print(f\"Before splice: {start},{stop}\")\n",
    "        acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name,query_species_original)\n",
    "        print(f\"after splice: {start},{stop}\")\n",
    "        for records in genome_file:\n",
    "            old_start = start\n",
    "            old_stop = stop\n",
    "            ag = \"N\"\n",
    "            gt = \"N\"\n",
    "            stop_counter = 0\n",
    "            while True:\n",
    "                print(f\"sequence:\\n{records.seq[start+start_modifier-1:stop]}\")\n",
    "                translated_sequence = records.seq[start+start_modifier-1:stop].translate()\n",
    "                print(f\"sequence:\\n{translated_sequence}\")\n",
    "                if \"*\" in translated_sequence:\n",
    "                    stop_counter +=1\n",
    "                    if ag == \"N\":\n",
    "                        start = old_start + 3*stop_counter\n",
    "                    if gt ==\"N\":\n",
    "                        stop = old_stop - 3*stop_counter\n",
    "                print(f\"left = {records.seq[start-3:start-1]}, right ={(records.seq[stop:stop + 2])}, stop_counter = {stop_counter}\"  ), \n",
    "                if (records.seq[start-3:start-1]).lower() == \"ag\" and ag != \"Y\":\n",
    "                    \n",
    "                    ag = \"Y\"\n",
    "                    \n",
    "                elif ag != \"Y\":\n",
    "                    start -= 3\n",
    "                    \n",
    "                if (records.seq[stop:stop + 2]).lower() == \"gt\" and gt != \"Y\":\n",
    "                    gt = \"Y\"\n",
    "                elif gt != \"Y\":\n",
    "                    stop +=3\n",
    "                if old_start - start > 1000 or stop - old_stop > 1000:\n",
    "                    break\n",
    "                if gt == \"Y\" and ag == \"Y\":\n",
    "                    gt_ag = \"Y\"\n",
    "                    break\n",
    "                if stop_counter > 10:\n",
    "                    gt = \"Y\"\n",
    "                    ag = \"Y\"\n",
    "                \n",
    "        \n",
    "        query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast/new_query_spliced.txt\"\n",
    "        with open(query_location , 'w') as query_file_new:\n",
    "            # sequence_translated = records.seq[start+start_modifier-1:stop]\n",
    "            # print(sequence_translated)\n",
    "            \n",
    "            sequence_translated = records.seq[start+start_modifier-1:stop].translate()\n",
    "            print(sequence_translated)\n",
    "            # proceed_test = input(\"Proceed with this?\")\n",
    "            # while True:\n",
    "            #     if proceed_test.lower()[0] == \"n\":\n",
    "            #         assert False\n",
    "            #     elif proceed_test.lower()[0] == \"y\":\n",
    "            #         break\n",
    "            if \"*\" in sequence_translated:\n",
    "                print(\"Errror in Spliced query\")\n",
    "                assert False\n",
    "            \n",
    "            output = f\">{original_query_name[1:]}\\n{sequence_translated}\"\n",
    "            query_file_new.write(output)\n",
    "\n",
    "        genome_file = get_genome_file(genome_location,species)\n",
    "        genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "        out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "        \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.htm\" -html'\n",
    "        # print(blast_command)\n",
    "        # subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.txt\"'\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "\n",
    "                \n",
    "        # acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name)\n",
    "\n",
    "        if acceptor == \"Y\" and donor  == \"Y\":\n",
    "            splice_prediction = \"Y\"\n",
    "        else:\n",
    "            splice_prediction = \"N\"\n",
    "            \n",
    "        return(start,stop,gt_ag, splice_prediction) \n",
    "        # output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaffold) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        # print(output_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f27f9cc5-c2a8-4cb7-83e1-7104ee982cd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_spiceator_result(start_coordinate, stop_coordinate,annotated_genome_location,annotated_species_name,query_species_original):\n",
    "    results_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/\"\n",
    "#     list_of_files_here = os.listdir(results_location)\n",
    "#     spliceator_results_file = ''\n",
    "#     for files in list_of_files_here:\n",
    "#         if files.startswith(\"Spliceator_results\"):\n",
    "#             spliceator_results_file = files\n",
    "#     if spliceator_results_file == '':\n",
    "#         print(\"Splice file missing\")\n",
    "#         assert False\n",
    "\n",
    "#     with open(f\"{results_location}/{spliceator_results_file}\", 'r') as splice_file:\n",
    "#         splice_file_content = splice_file.readlines()\n",
    "\n",
    "    splice_acceptor_presence = \"N\"\n",
    "    splice_donor_presence = \"N\"\n",
    "    splice_donor = ''\n",
    "    splice_acceptor = ''\n",
    "    start_coordinate = 0\n",
    "    stop_coordinate = 0\n",
    "#     for lines in splice_file_content:\n",
    "#         line_split = lines.split(\";\")\n",
    "#         if line_split[0]==\"Acceptor\" and splice_acceptor_presence == \"N\":\n",
    "#             acceptor_start = int(line_split[1])\n",
    "#             acceptor_end = acceptor_start + len(line_split[3])\n",
    "            \n",
    "#             if start_coordinate >=acceptor_start and start_coordinate <=acceptor_end:\n",
    "#                 splice_acceptor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if start_coordinate >= acceptor_start:\n",
    "#                 #         start_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_acceptor = lines\n",
    "#         if line_split[0]==\"Donor\" and splice_donor_presence == \"N\":\n",
    "#             donor_start = int(line_split[1])\n",
    "#             donor_end = donor_start + len(line_split[3])\n",
    "#             if stop_coordinate >=donor_start and stop_coordinate <=donor_end:\n",
    "#                 splice_donor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if stop_coordinate >=donor_start:\n",
    "#                 #         stop_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_donor = lines\n",
    "\n",
    "    return(splice_acceptor_presence, splice_donor_presence, splice_donor, splice_acceptor)\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50c956e3-7a49-4f1f-b0e6-bac420e663ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, ag_gt, splice_prediction,original_query_name, query_species ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "    with open(f\"{blast_location}/new_query_spliced.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "    \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "            # print(lines)\n",
    "            # print(query_name)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        print(f\"start_coordinate : {start_coor},stop_coordinate : {stop_coor}\")\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "            \n",
    "        output_format = str(species_name)+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \",\" + ag_gt + \",\" + splice_prediction +\"\\n\"  \n",
    "        # print(Output_Sequence)\n",
    "        return(output_format)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4a79568c-ea02-45a4-88d2-b10f23f091e7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24']\n",
      "Leptidea_juvernica\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 2340,2423\n",
      "after splice: 2340,2423\n",
      "sequence:\n",
      "aataGTTCACATTCTACACATTGTGGCAGTCACTCAAGTGGCAGCAGTGGATATGGAGGAGAACCTACCAAATCCGGTTATAG\n",
      "sequence:\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "start_coordinate : 23463731,stop_coordinate : 23463651\n",
      "2340 2423 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23463649,23463732,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_3\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 4250,4401\n",
      "after splice: 4250,4401\n",
      "sequence:\n",
      "ACTAATAATCTCAGCCAACCCCCGGAAAAGAAAAGGAAGATAAACAAGAAGAAGACAGCAGAATGtgaaaaacttaaaagtaaaaagaagGACCCAAGTCCTCCAAAATTTGAAATAGAAGTAACATTGAAAGAACAAGAAGCGAATGGCG\n",
      "sequence:\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "[[1, 1]]\n",
      "start_coordinate : 23461821,stop_coordinate : 23461672\n",
      "4250 4401 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23461671,23461822,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_1_query_Exon_3,1,50,50,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_4\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 5776,5888\n",
      "after splice: 5776,5888\n",
      "sequence:\n",
      "ATTGAAGTTTCGAAGACACAAATGTCGGAATTCGGAAGTAACGTTGAAAATGTGGACAGCTCAACGGAACCAGAACCACCGAATTGCGCAAACgctattaatgataataag\n",
      "sequence:\n",
      "IEVSKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "IEVSKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "[[2, 0]]\n",
      "start_coordinate : 23460294,stop_coordinate : 23460184\n",
      "5776 5888 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23460184,23460296,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_4,1,37,37,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_5\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "19\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 6235,6291\n",
      "after splice: 6235,6291\n",
      "sequence:\n",
      "AAAAGTTCCAGCAATATTCCAAGTTCCACGTCGATGGATTCAATTCCTTTTAATGCT\n",
      "sequence:\n",
      "KSSSNIPSSTSMDSIPFNA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSSSNIPSSTSMDSIPFNA\n",
      "[[0, 0]]\n",
      "start_coordinate : 23459837,stop_coordinate : 23459781\n",
      "6235 6291 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23459781,23459837,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_5,1,19,19,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_6\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 6737,6868\n",
      "after splice: 6737,6868\n",
      "sequence:\n",
      "gaaTGCTTTTCGTGTGTAATGTCGATGCAAGATGGTGTTGTTATGCACACTACTCCAGGAATAACATCAACTTTAGGCTACCCTAAGGATTTCTGGGTGGGAAGGCCGTTTATAGAGTTCCTTCATCAGTGg\n",
      "sequence:\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "[[0, 0]]\n",
      "start_coordinate : 23459335,stop_coordinate : 23459204\n",
      "6737 6868 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23459204,23459335,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_7\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "23\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 9630,9699\n",
      "after splice: 9630,9699\n",
      "sequence:\n",
      "gACCGGAAGACCTTCGAATCGGAATTAAATGATTTGGTTGCCGAAAATGTAAATAGTACAAAGAAAAAAg\n",
      "sequence:\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "[[0, 1]]\n",
      "start_coordinate : 23456442,stop_coordinate : 23456374\n",
      "9630 9699 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23456373,23456442,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_7,1,23,23,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_8\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "65\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 9838,10035\n",
      "after splice: 9838,10035\n",
      "sequence:\n",
      "AATGGTGACATGACTACGACAGCCGTATGCCGGTTACGATGCTATAAAGCGCTATCTGCCGGTTTCGAAATAAGGGAGACAATCGCTACTTTTACGCCATTCCTACTAAAGCTATCATTCAGAAAAATATGCGATGAAGAGGGCGAAGTGTGGTATCTTCTTGTTCAGGCCACGCCATTCATTTCTGCATTTAAAG\n",
      "sequence:\n",
      "NGDMTTTAVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NGDMTTTAVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "[[2, 1]]\n",
      "start_coordinate : 23456232,stop_coordinate : 23456038\n",
      "9838 10035 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23456037,23456234,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_8,1,65,65,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_9\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 11005,11233\n",
      "after splice: 11005,11233\n",
      "sequence:\n",
      "CCGAACGAGGTAGTAACAAAGGCGGTAGCCTTCGTATTAAGACATTCCTCAACGGGGGAAATAAAGTACATAGATACGGAGTCAGTGCCGTATCTGGGATATTCCCCGCAAGATATTGTTAACAAAGACGCTATCGCCATGTACCATCCTTATGATCTGGAATATTTACTGCATGTGTACGAAACTATCATGAAAGTAGGAAGCGCACCACGCTCGTCGCCTTACAG\n",
      "sequence:\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 23455065,stop_coordinate : 23454841\n",
      "11005 11233 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23454839,23455067,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_10\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 13317,13428\n",
      "after splice: 13317,13428\n",
      "sequence:\n",
      "atgATGACACAAAACGGTGATTATGTAAAATTGGAGACAGAATGGTCGTCTTTTTTCAATCCGTGGTCGAGAAAACTGGAATTCATTATTGGTAAACACTTGATACTCGAg\n",
      "sequence:\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "[[1, 0]]\n",
      "start_coordinate : 23452754,stop_coordinate : 23452644\n",
      "13317 13428 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23452644,23452755,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_11\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 14904,15023\n",
      "after splice: 14904,15023\n",
      "sequence:\n",
      "GGTCCAACTAACCCAGATGTGTTTCAAAGCCCGAACCCggaagtacaaaataaatttcctGAGGATGATGATAATAAGGCCAAATCGTTGAGGGAAAACATCATAAAGATATTAAATGAG\n",
      "sequence:\n",
      "GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 23451168,stop_coordinate : 23451049\n",
      "14904 15023 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23451049,23451168,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_12\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 15412,15561\n",
      "after splice: 15412,15561\n",
      "sequence:\n",
      "CCTTCAACTAAATCCGAAGATTTAGCAAAACAGGAGATTGCGATTCGATGCCAGAAAGTAGCCTCTATCATTGAAAGCTTAAACGAAAAAGCACCGAATGTAGACAAAGAGATTTGGATCGATATTCAAGAGCCTGAAAATGTtaattat\n",
      "sequence:\n",
      "PSTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "PSTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "[[0, 0]]\n",
      "start_coordinate : 23450660,stop_coordinate : 23450511\n",
      "15412 15561 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23450511,23450660,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_13\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 16507,16625\n",
      "after splice: 16507,16625\n",
      "sequence:\n",
      "GACAGGGATTCAGTTCTGCTCGGGAGCATATCTCCTCACACCGACAGTATTGTCAGCAAGGCGAGCACCGAAACTCAAGTCAACTACAATCAATTGAACACAGATCAAAATTTAGACAG\n",
      "sequence:\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDQNLD\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDQNLD\n",
      "[[0, 2]]\n",
      "start_coordinate : 23449565,stop_coordinate : 23449449\n",
      "16507 16625 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23449447,23449565,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_14\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "33\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 19732,19833\n",
      "after splice: 19732,19833\n",
      "sequence:\n",
      "TACTTCAACCGTTGGTTAGAACATTGTGTGGAAGAGGAGATAGCGGAGGCCGAACCTAATAATGGTGAAGCTAATTTGGAGAATTCTAATGAGACTAATGg\n",
      "sequence:\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "[[1, 2]]\n",
      "start_coordinate : 23446339,stop_coordinate : 23446241\n",
      "19732 19833 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23446239,23446340,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_14,1,33,33,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_15\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "45\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 23193,23330\n",
      "after splice: 23193,23330\n",
      "sequence:\n",
      "GAATTGGCGTCGGTGGAGGAGCAGTCCGAAGCTAGCAAGATGGCCGATTTCTTCAAGTCCAGTGTTACATTGATACCTAATAATCACGGCCATGCTGATTATAAACCCACAAGATTGtctgaaaatattttagaaaa\n",
      "sequence:\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "[[1, 2]]\n",
      "start_coordinate : 23442878,stop_coordinate : 23442744\n",
      "23193 23330 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23442742,23442879,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_15,1,45,45,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_16\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "55\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 23929,24094\n",
      "after splice: 23929,24094\n",
      "sequence:\n",
      "CATGATTCGATAATGGAGATGGATCTAGTGAAAAAGCATCGAGAAATCCCGAAAGAAAATAGAAAGAAATATTCAAAGGAGGTGAGGCTAAAGCAAAAGGAGCATTTGGCAAGATGTAATGCAACGTTCACACCAGCCACTGCGGGGTTACTGGCTGACTTACAg\n",
      "sequence:\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 23442142,stop_coordinate : 23441978\n",
      "23929 24094 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23441978,23442143,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_16,1,55,55,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_17\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "47\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 25771,25912\n",
      "after splice: 25771,25912\n",
      "sequence:\n",
      "tcgAATGGCCTAAAGCGTACATCTAAATTCTGTGAAGAGTCTGACGGTGTCAATTATCTCTATTCAATTGCAAAACAACCTCGGCGTGATGGGACAGGAACTAGCTCCCCTCGTTTGTACGTTAATTTACCTAACAGTGATG\n",
      "sequence:\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "[[0, 1]]\n",
      "start_coordinate : 23440301,stop_coordinate : 23440161\n",
      "25771 25912 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23440160,23440301,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_17,1,47,47,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_18\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 26250,26357\n",
      "after splice: 26250,26357\n",
      "sequence:\n",
      "ACAACTTCAGTGCCGCCAAATAAAGAGAATAGCGCCAATCAATTTTCATTTGAAGTTGGCTTTCCCCAACAGATGTCGCCTTTGGGCCCGAGTGTGGCAACGCAAG\n",
      "sequence:\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPSVATQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPSVATQ\n",
      "[[2, 1]]\n",
      "start_coordinate : 23439820,stop_coordinate : 23439716\n",
      "26250 26357 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23439715,23439822,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_18,1,35,35,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_19\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "43\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 28141,28271\n",
      "after splice: 28141,28271\n",
      "sequence:\n",
      "CCTTTTATGTCACCTGGACCGTATTTTAATGGAACGCACGCGCATTTGTATGAGGAAATGCAGCAGGAGAAAAACCATCCATCTACATCAAAGCAACCGCTTCCCCTACCTACCGACAACTACCAGGAG\n",
      "sequence:\n",
      "PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "[[2, 0]]\n",
      "start_coordinate : 23437929,stop_coordinate : 23437801\n",
      "28141 28271 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23437801,23437931,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_19,1,43,43,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_20\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "49\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 30103,30249\n",
      "after splice: 30103,30249\n",
      "sequence:\n",
      "GCATATAAACTCATTATTCCACCGAAGACGAGCAAGTCGATACCTTCCAGTGAATGGAGAAAAAAGAGAACGAAACTCCAGGCACGTGCCAATAATATCAATGAAAGCAGTGGCGACAGTATATCATCAGCGAACGTTTCTTATTTG\n",
      "sequence:\n",
      "AYKLIIPPKTSKSIPSSEWRKKRTKLQARANNINESSGDSISSANVSYL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "AYKLIIPPKTSKSIPSSEWRKKRTKLQARANNINESSGDSISSANVSYL\n",
      "[[0, 0]]\n",
      "start_coordinate : 23435969,stop_coordinate : 23435823\n",
      "30103 30249 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23435823,23435969,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_20,1,49,49,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_21\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 34607,34768\n",
      "after splice: 34607,34768\n",
      "sequence:\n",
      "AGTACAAACGCAGTTACTTTAAGCCAGATCAACGAGAAAGATCCGATTGCCAACAGCGGGACGAATGCAGATAAGACTGATGATGAATCTTGGTTTTCTTCATTTTCCTCTCACATATATAAGAGTGAAACTGGCAGCCCAGAGGAATCCGAAATGAAAAAG\n",
      "sequence:\n",
      "STNAVTLSQINEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "STNAVTLSQINEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "[[0, 0]]\n",
      "start_coordinate : 23431465,stop_coordinate : 23431304\n",
      "34607 34768 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23431304,23431465,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_21,1,54,54,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_22\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 36957,37043\n",
      "after splice: 36957,37043\n",
      "sequence:\n",
      "AAATCTACGATATCATCACTACAAACCAATCGTCAAACAATAAATAAGGTAGCAATGAAGAAAATAGATCCCCCGTGGCTGGATGAT\n",
      "sequence:\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "[[0, 0]]\n",
      "start_coordinate : 23429115,stop_coordinate : 23429029\n",
      "36957 37043 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23429029,23429115,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_22,1,29,29,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_23\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 37691,37792\n",
      "after splice: 37691,37792\n",
      "sequence:\n",
      "GTGTGCCTTACTCCTgagataatatacaaatatcaaGTCAAAACCAGAAGCTTGGCGGAAGTACTTGCGGAAGACAAGAAGAAACTCGAAGCCCTAGAGCAG\n",
      "sequence:\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 23428381,stop_coordinate : 23428280\n",
      "37691 37792 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23428280,23428381,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_23,1,34,34,Y,N\n",
      "\n",
      "Leptidea_juvernica\n",
      "Processing Exon_24\n",
      "Leptidea_juvernica_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 39156,39287\n",
      "after splice: 39156,39287\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACATCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACA\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGHPEKLTLEEGITSSSSSGDDTTSNT\n",
      "left = ag, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACATCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACAGAG\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGHPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PSLVNEQLSQLYYELKLEGHPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "[[0, 0]]\n",
      "start_coordinate : 23426916,stop_coordinate : 23426782\n",
      "39156 39290 Y N\n",
      "Leptidea_juvernica,CATIXQ010000001.1,23426782,23426916,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_24,1,45,45,Y,N\n",
      "\n",
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24']\n",
      "Leptidea_reali\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 3074,3157\n",
      "after splice: 3074,3157\n",
      "sequence:\n",
      "aataGTTCACATTCTACACATTGTGGCAGTCACTCAAGTGGCAGCAGTGGATATGGAGGAGAACCTACCAAATCCGGTTATAG\n",
      "sequence:\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "[[1, 2]]\n",
      "start_coordinate : 20869210,stop_coordinate : 20869130\n",
      "3074 3157 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20869128,20869211,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_3\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 4962,5113\n",
      "after splice: 4962,5113\n",
      "sequence:\n",
      "ACTAATAATCTCAGCCAACCCCCGGAAAAGAAAAGGAAGATAAACAAGAAGAAGACAGCAGAATGtgaaaaacttaaaagtaaaaagaagGACCCAAGTCCTccaaaatttgaaattgaagtAACATTGAAAGAACAAGAAGCGAATGGCG\n",
      "sequence:\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 1]]\n",
      "start_coordinate : 20867322,stop_coordinate : 20867173\n",
      "4962 5113 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20867172,20867323,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_1_query_Exon_3,1,50,50,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_4\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 6505,6617\n",
      "after splice: 6505,6617\n",
      "sequence:\n",
      "ATTGAAGCTTCGAAGACACAAATGTCGGAATTCGGAAGTAACGTTGAAAATGTGGACAGCTCAACGGAACCAGAACCACCGAATTGCGCAAACGCTACTAATGATAATAAG\n",
      "sequence:\n",
      "IEASKTQMSEFGSNVENVDSSTEPEPPNCANATNDNK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "IEASKTQMSEFGSNVENVDSSTEPEPPNCANATNDNK\n",
      "[[2, 0]]\n",
      "start_coordinate : 20865778,stop_coordinate : 20865668\n",
      "6505 6617 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20865668,20865780,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_4,1,37,37,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_5\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "19\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 6965,7021\n",
      "after splice: 6965,7021\n",
      "sequence:\n",
      "AAAAGTTCCAGCAATATTCCAAGTTCCACGTCGATGGATTCAATTCCTTTTAATGCT\n",
      "sequence:\n",
      "KSSSNIPSSTSMDSIPFNA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSSSNIPSSTSMDSIPFNA\n",
      "[[0, 0]]\n",
      "start_coordinate : 20865320,stop_coordinate : 20865264\n",
      "6965 7021 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20865264,20865320,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_5,1,19,19,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_6\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 7467,7598\n",
      "after splice: 7467,7598\n",
      "sequence:\n",
      "gaaTGCTTTTCGTGTGTAATGTCGATGCAAGATGGTGTTGTTATGCACACTACTCCAGGAATAACATCAACTTTAGGCTACCCTAAGGATTTCTGGGTGGGAAGGCCGTTTATAGAGTTCCTTCATCAGTGg\n",
      "sequence:\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "[[0, 0]]\n",
      "start_coordinate : 20864818,stop_coordinate : 20864687\n",
      "7467 7598 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20864687,20864818,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_7\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "23\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 11226,11295\n",
      "after splice: 11226,11295\n",
      "sequence:\n",
      "gACCGGAAGACCTTCGAATCGGAATTAAATGATTTGGTTGCCGAAAATGTAAATAGTACAAAGAAAAAAg\n",
      "sequence:\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "[[0, 1]]\n",
      "start_coordinate : 20861059,stop_coordinate : 20860991\n",
      "11226 11295 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20860990,20861059,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_7,1,23,23,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_8\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "65\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 11434,11631\n",
      "after splice: 11434,11631\n",
      "sequence:\n",
      "AATGGTGACATGACTACGACAGTCGTATGCCGGTTACGATGCTATAAAGCGCTATCTGCCGGTTTCGAAATAAGGGAGACAATCGCTACTTTTACGCCATTCCTACTAAAGCTATCATTCAGAAAAATATGCGACGAAGAGGGCGAAGTGTGGTATCTTCTTGTTCAGGCCACGCCATTCATTTCTGCATTTAAAG\n",
      "sequence:\n",
      "NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "[[2, 1]]\n",
      "start_coordinate : 20860849,stop_coordinate : 20860655\n",
      "11434 11631 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20860654,20860851,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_8,1,65,65,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_9\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 12535,12763\n",
      "after splice: 12535,12763\n",
      "sequence:\n",
      "CCGAACGAGGTAGTAACAAAGGCGGTAGCCTTCGTATTAAGACATTCCTCAACGGGGGAAATAAAGTACATAGATACGGAGTCAGTGCCGTATCTGGGATATTCCCCGCAAGATATTGTTAACAAAGACGCTATCGCCATGTACCATCCTTATGATCTGGAATATTTACTGCATGTGTACGAAACTATCATGAAAGTAGGAAGCGCACCACGCTCGTCGCCTTACAG\n",
      "sequence:\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 20859748,stop_coordinate : 20859524\n",
      "12535 12763 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20859522,20859750,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_10\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 14871,14982\n",
      "after splice: 14871,14982\n",
      "sequence:\n",
      "atgATGACACAAAACGGTGATTATGTAAAATTGGAGACAGAATGGTCGTCTTTTTTCAATCCGTGGTCGAGAAAACTGGAATTCATTATTGGTAAACACTTGATACTCGAG\n",
      "sequence:\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "left = ag, right =Gt, stop_counter = 0\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "[[1, 0]]\n",
      "start_coordinate : 20857413,stop_coordinate : 20857303\n",
      "14871 14982 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20857303,20857414,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_11\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 16447,16566\n",
      "after splice: 16447,16566\n",
      "sequence:\n",
      "GGTCCAACTAACCCAGATGTGTTTCAAAGCCCGAACCCggaagtacaaaataaatttcctGAGGATGATGATAATAAGGCCAAATCGTTGAGGGAAAACATCATAAAGATACTAAATGAG\n",
      "sequence:\n",
      "GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 20855838,stop_coordinate : 20855719\n",
      "16447 16566 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20855719,20855838,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_12\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 16972,17121\n",
      "after splice: 16972,17121\n",
      "sequence:\n",
      "cCTACAACTAAATCCGAAGATTTAGCCAAACAGGAGATAGCGATTCGATGCCAGAAAGTAGCCTCTATCATTGAAAGCTTAAACGAAAAAGCACCGAATGTAGACAAAGAGATTTGGATCGATATTCAAGAGCCTGAAAATGTtaattat\n",
      "sequence:\n",
      "PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "[[0, 0]]\n",
      "start_coordinate : 20855313,stop_coordinate : 20855164\n",
      "16972 17121 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20855164,20855313,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_13\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 18168,18286\n",
      "after splice: 18168,18286\n",
      "sequence:\n",
      "GACAGGGATTCAGTTCTGCTTGGGAGCATATCTCCTCACACCGACAGTATTGTCAGCAAGGCGAGCACCGAAACTCAAGTCAACTACAATCAATTGAACACCGATGAAAATTTAGACAG\n",
      "sequence:\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "[[0, 2]]\n",
      "start_coordinate : 20854117,stop_coordinate : 20854001\n",
      "18168 18286 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20853999,20854117,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_14\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "33\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 21272,21373\n",
      "after splice: 21272,21373\n",
      "sequence:\n",
      "TACTTCAACCGTTGGTTAGAACATTGTGTGGAAGAGGAGATAGCGGAGGCCGAACCTAATAATGGTGAAGCTAATTTGGAGAATTCTAATGAGACTAATGg\n",
      "sequence:\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "[[1, 2]]\n",
      "start_coordinate : 20851012,stop_coordinate : 20850914\n",
      "21272 21373 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20850912,20851013,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_14,1,33,33,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_15\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "45\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 24704,24841\n",
      "after splice: 24704,24841\n",
      "sequence:\n",
      "GAATTGGCGTCGGTGGAGGAGCAGTCCGAAGCTAGCAAGATGGCCGATTTCTTCAAGTCCAGTGTTACATTGATACCTAATAATCACGGCCATGCTGATTATAAACCCACAAGATTGtctgaaaatattttagaaaa\n",
      "sequence:\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "[[1, 2]]\n",
      "start_coordinate : 20847580,stop_coordinate : 20847446\n",
      "24704 24841 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20847444,20847581,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_15,1,45,45,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_16\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "55\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 25442,25607\n",
      "after splice: 25442,25607\n",
      "sequence:\n",
      "CATGATTCGATAATGGAGATGGATCTAGTGAAAAAGCATCGAGAAATCCCGAAAGAAAATAGAAAGAAATATTCAAAGGAGGTGAGGCTAAAGCAAAAGGAGCATTTGGCAAGATGTAATGCAACGTTCACACCAGCCACTGCGGGGTTACTGGCTGACTTACAg\n",
      "sequence:\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 20846842,stop_coordinate : 20846678\n",
      "25442 25607 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20846678,20846843,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_16,1,55,55,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_17\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "47\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 27411,27552\n",
      "after splice: 27411,27552\n",
      "sequence:\n",
      "tcgAATGGCCTAAAGCGTACATCTAAATTCTGTGAAGAGTCTGACGGTGTCAATTATCTCTATTCAATTGCAAAACAACCTCGGCGTGATGGGACAGGAACTAGCTCCCCTCGTTTGTACGTTAATTTACCTAACAGTGATG\n",
      "sequence:\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "[[0, 1]]\n",
      "start_coordinate : 20844874,stop_coordinate : 20844734\n",
      "27411 27552 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20844733,20844874,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_17,1,47,47,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_18\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 27890,27997\n",
      "after splice: 27890,27997\n",
      "sequence:\n",
      "ACAACTTCAGTGCCGCCAAATAAAGAGAATAGCGCCAATCAATTTTCATTTGAAGTTGGCTTTCCCCAACAGATGTCGCCTTTGGGCCCGAATGTGGCAACGCAAG\n",
      "sequence:\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "[[2, 1]]\n",
      "start_coordinate : 20844393,stop_coordinate : 20844289\n",
      "27890 27997 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20844288,20844395,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_18,1,35,35,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_19\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "43\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 32035,32165\n",
      "after splice: 32035,32165\n",
      "sequence:\n",
      "CCTTTTATGTCACCTGGACCGTATTTTAATGGAACGCACGCGCATTTGTATGAGGAAATGCAGCAGGAGAAAAACCATCCATCTACATCAAAGCAACCGCTTCCCCTACCTACCGACAACTACCAGGAG\n",
      "sequence:\n",
      "PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "[[2, 0]]\n",
      "start_coordinate : 20840248,stop_coordinate : 20840120\n",
      "32035 32165 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20840120,20840250,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_19,1,43,43,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_20\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "49\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 33900,34046\n",
      "after splice: 33900,34046\n",
      "sequence:\n",
      "GCATGTAAACTCATTATTCCACCGAAGACGAGCAAGTCGATATCTTCCAGTGAATGGAGAAAAAAGAGAACGAAACTCCAGGCACGTGCCAATAATATCAATGAAAGCAGTGGCGACAGTATATCATCAGCGAACGTTTCTAATTTG\n",
      "sequence:\n",
      "ACKLIIPPKTSKSISSSEWRKKRTKLQARANNINESSGDSISSANVSNL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ACKLIIPPKTSKSISSSEWRKKRTKLQARANNINESSGDSISSANVSNL\n",
      "[[0, 0]]\n",
      "start_coordinate : 20838385,stop_coordinate : 20838239\n",
      "33900 34046 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20838239,20838385,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_20,1,49,49,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_21\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 38429,38590\n",
      "after splice: 38429,38590\n",
      "sequence:\n",
      "AGTACAAACGCAGTTACTTTAAGCCAGAACAACGAGAAAGATCCGATTGCCAACAGCGGGACGAATGCAGATAAGACTGATGATGAATCTTGGTTTTCTTCATTTTCCTCTCACATATATAAGAGTGAAACTGGCAGCCCAGAGGAATCCGAAATGAAAAAG\n",
      "sequence:\n",
      "STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "[[0, 0]]\n",
      "start_coordinate : 20833856,stop_coordinate : 20833695\n",
      "38429 38590 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20833695,20833856,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_21,1,54,54,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_22\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 40721,40807\n",
      "after splice: 40721,40807\n",
      "sequence:\n",
      "AAATCTACGATATCATCACTACAAACCAATCGTCAAACAATAAATAAGGTAGCAATGAAGAAAATAGATCCCCCGTGGCTGGATGAT\n",
      "sequence:\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "[[0, 0]]\n",
      "start_coordinate : 20831564,stop_coordinate : 20831478\n",
      "40721 40807 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20831478,20831564,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_22,1,29,29,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_23\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 41440,41541\n",
      "after splice: 41440,41541\n",
      "sequence:\n",
      "GTGTGCCTTACTCCTgagataatatacaaatatcaaGTCAAAACCAGAAGCTTGGCGGAAGTACTTGCGGAAGACAAGAAGAAACTCGAAGCCCTAGAGCAG\n",
      "sequence:\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 20830845,stop_coordinate : 20830744\n",
      "41440 41541 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20830744,20830845,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_23,1,34,34,Y,N\n",
      "\n",
      "Leptidea_reali\n",
      "Processing Exon_24\n",
      "Leptidea_reali_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 42556,42687\n",
      "after splice: 42556,42687\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACTTCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACA\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNT\n",
      "left = ag, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACTTCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACAGAG\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "[[0, 0]]\n",
      "start_coordinate : 20829729,stop_coordinate : 20829595\n",
      "42556 42690 Y N\n",
      "Leptidea_reali,CATIXK010000002.1,20829595,20829729,1,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_24,1,45,45,Y,N\n",
      "\n",
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24']\n",
      "Leptidea_sinapis\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 2955,3038\n",
      "after splice: 2955,3038\n",
      "sequence:\n",
      "aataGTTCACATTCTACACATTGTGGCAGTCACTCAAGTGGCAGCAGTGGATATGGAGGAGAACCTACTAAATCCGGTTATAG\n",
      "sequence:\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "[[1, 2]]\n",
      "start_coordinate : 12065349,stop_coordinate : 12065429\n",
      "2955 3038 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12065348,12065431,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_3\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 4841,4992\n",
      "after splice: 4841,4992\n",
      "sequence:\n",
      "ACTAATAATCTCAGCCAACCCCCGGAAAAGAAAAGGAAGATAAACAAGAAGAAGACAGCAGAATGTGaaaaacttaaaagtaaaaagaagGACCCAAGTCCTCCAAAATTTGAAATAGAAGTAACATTGAAAGAACAAGAAGCGAATGGCG\n",
      "sequence:\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "[[1, 1]]\n",
      "start_coordinate : 12067235,stop_coordinate : 12067384\n",
      "4841 4992 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12067234,12067385,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_1_query_Exon_3,1,50,50,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_4\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 6351,6463\n",
      "after splice: 6351,6463\n",
      "sequence:\n",
      "ATTGAAGCTTCGAAGACACAAATGTCGGAATTCGGAAGTAACGTTGAAAATGTGGACAGCTCAACGGAACCAGAACCACCGAATTGCGCAAACgctattaatgataataag\n",
      "sequence:\n",
      "IEASKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "IEASKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "[[2, 0]]\n",
      "start_coordinate : 12068746,stop_coordinate : 12068856\n",
      "6351 6463 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12068744,12068856,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_4,1,37,37,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_5\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 6825,6881\n",
      "after splice: 6825,6881\n",
      "sequence:\n",
      "AAAAGTTCCAGCAATATTCCAAGTTCCCCGTCGATGGATTCAATTCCTTTTAATGCT\n",
      "sequence:\n",
      "KSSSNIPSSPSMDSIPFNA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSSSNIPSSPSMDSIPFNA\n",
      "[[0, 0]]\n",
      "start_coordinate : 12069218,stop_coordinate : 12069274\n",
      "6825 6881 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12069218,12069274,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_5,1,19,19,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_6\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 7331,7462\n",
      "after splice: 7331,7462\n",
      "sequence:\n",
      "gaaTGCTTTTCGTGTGTAATGTCGATGCAAGATGGTGTTGTTATGCACACTACTCCAGGAATAACATCAACTTTAGGCTACCCTAAGGATTTCTGGGTGGGAAGGCCGTTTATAGAGTTCCTTCATCAGTGg\n",
      "sequence:\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "[[0, 0]]\n",
      "start_coordinate : 12069724,stop_coordinate : 12069855\n",
      "7331 7462 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12069724,12069855,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_7\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "23\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 11072,11141\n",
      "after splice: 11072,11141\n",
      "sequence:\n",
      "gACCGGAAGACCTTCGAATCGGAATTAAATGATTTGGTTGCCGAAAATGTAAATAGTACAAAGAAAAAAG\n",
      "sequence:\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "DRKTFESELNDLVAENVNSTKKK\n",
      "[[0, 1]]\n",
      "start_coordinate : 12073465,stop_coordinate : 12073533\n",
      "11072 11141 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12073465,12073534,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_7,1,23,23,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_8\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "65\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 11280,11477\n",
      "after splice: 11280,11477\n",
      "sequence:\n",
      "AATGGTGACATGACTACGACAGTCGTATGCCGGTTAAGATGCTATAAAGCGCTATCCGCCGGTTTCGAAATAAGGGAGACAATCGCTACTTTTACGCCATTCCTACTAAAGCTATCATTCAGAAAAATATGCGACGAAGAGGGCGAAGTGTGGTATCTTCTTGTTCAGGCCACGCCATTCATTTCTGCATTTAAAG\n",
      "sequence:\n",
      "NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPFISAFK\n",
      "[[2, 1]]\n",
      "start_coordinate : 12073675,stop_coordinate : 12073869\n",
      "11280 11477 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12073673,12073870,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_8,1,65,65,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_9\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 12249,12477\n",
      "after splice: 12249,12477\n",
      "sequence:\n",
      "CCGAACGAGGTAGTAACAAAGGCGGTAGCCTTCGTATTAAGACATTCCTCAACGGGGGAAATAAAGTACATAGATACGGAGTCAGTGCCGTATCTGGGATATTCCCCGCAAGATATTGTTAACAAAGACGCTATCGCCATGTACCACCCCTATGATCTGGAATATTTACTGCATGTGTACGAAACTATCATGAAAGTAGGAAGCGCACCACGCTCGTCGCCTTACAG\n",
      "sequence:\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVYETIMKVGSAPRSSPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 12074644,stop_coordinate : 12074868\n",
      "12249 12477 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12074642,12074870,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_10\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 14593,14704\n",
      "after splice: 14593,14704\n",
      "sequence:\n",
      "ATGATGACACAAAACGGTGATTATGTAAAATTGGAGACAGAATGGTCGTCTTTTTTCAATCCGTGGTCGAGAAAACTGGAATTCATTATTGGTAAACACTTGATACTCGAg\n",
      "sequence:\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "[[1, 0]]\n",
      "start_coordinate : 12076987,stop_coordinate : 12077097\n",
      "14593 14704 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12076986,12077097,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_11\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 16178,16297\n",
      "after splice: 16178,16297\n",
      "sequence:\n",
      "GGTCCAACTAACCCAGATGTGTTTCAAAGCCCGAACCCGGAAGTACAAAATCAATTTCCTGAGGATGATGATAATAAGGCCAAATCGTTGAGGGAAAACATCATAAAGATACTAAATGAG\n",
      "sequence:\n",
      "GPTNPDVFQSPNPEVQNQFPEDDDNKAKSLRENIIKILNE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "GPTNPDVFQSPNPEVQNQFPEDDDNKAKSLRENIIKILNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 12078571,stop_coordinate : 12078690\n",
      "16178 16297 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12078571,12078690,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_12\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 16730,16879\n",
      "after splice: 16730,16879\n",
      "sequence:\n",
      "cCTACAACTAAATCCGAAGATTTAGCCAAACAGGAGATAGCGATTCGATGCCAGAAAGTAGCCTCTATCATTGAAAGCCTAAACGAAAAAGCACCGAATGTAGACAAAGAGATTTGGATCGATATTCAAGAGCCTGAAAATGTTAattat\n",
      "sequence:\n",
      "PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "[[0, 0]]\n",
      "start_coordinate : 12079123,stop_coordinate : 12079272\n",
      "16730 16879 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12079123,12079272,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_13\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 17925,18043\n",
      "after splice: 17925,18043\n",
      "sequence:\n",
      "GACAGGGATTCAGTTCTGCTTGGGAGCATATCTCCTCACACCGACAGTATTGTCAGCAAGGCGAGCACCGAAACTCAAGTCAACTACAATCAATTGAACACCGATGAAAATTTAGACAG\n",
      "sequence:\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "[[0, 2]]\n",
      "start_coordinate : 12080318,stop_coordinate : 12080434\n",
      "17925 18043 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12080318,12080436,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_14\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "33\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 21554,21655\n",
      "after splice: 21554,21655\n",
      "sequence:\n",
      "TACTTCAACCGTTGGTTAGAACATTGTGTGGAAGAGGAGATAGCGGAGGCCGAACCTAATAATGGTGAAGCTAATTTGGAGAATTCTAATGAGGCTAATGg\n",
      "sequence:\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNEAN\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "YFNRWLEHCVEEEIAEAEPNNGEANLENSNEAN\n",
      "[[1, 2]]\n",
      "start_coordinate : 12083948,stop_coordinate : 12084046\n",
      "21554 21655 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12083947,12084048,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_14,1,33,33,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_15\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "45\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 25011,25148\n",
      "after splice: 25011,25148\n",
      "sequence:\n",
      "GAATTGGCGTCGGTGGAGGAGCAGTCCGAAGCTAGCAAGATGGCCGATTTCTTCAAGTCCAGTGTTACATTGATACCTAATAATCACGGCCATGCTGATTATAAACCCACAAGATTGtctgaaaatattttagaaaa\n",
      "sequence:\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "[[1, 2]]\n",
      "start_coordinate : 12087405,stop_coordinate : 12087539\n",
      "25011 25148 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12087404,12087541,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_2_query_Exon_15,1,45,45,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_16\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "55\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 25748,25913\n",
      "after splice: 25748,25913\n",
      "sequence:\n",
      "CATGATTCGATAATGGAGATGGATCTAGTGAAAAAGCATCGAGAAATCCCGAAAGAAAATAGAAAGAAATATTCAAAGGAGGTGAGGTTAAAGCAAAAGGAGCATTTGGCAAGATGTAATGCAACGTTCACACCAGCCACTGCGGGGTTACTGGCTGACTTACAg\n",
      "sequence:\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 12088142,stop_coordinate : 12088306\n",
      "25748 25913 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12088141,12088306,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_1_rightoh_0_query_Exon_16,1,55,55,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_17\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "47\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 27839,27980\n",
      "after splice: 27839,27980\n",
      "sequence:\n",
      "tcgAATGGCCTAAAGCGTACATCTAAATTCTGTGAAGAGTCTGACGGTGTCAATTATCTCTATTCAATTGCAAAACAACCTCGGCGTGATGGGACAGGAACTAGCTCCCCTCGTTTGTACGTTAATTTACCTAACAGTGATG\n",
      "sequence:\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 1]]\n",
      "start_coordinate : 12090232,stop_coordinate : 12090372\n",
      "27839 27980 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12090232,12090373,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_1_query_Exon_17,1,47,47,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_18\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 28318,28425\n",
      "after splice: 28318,28425\n",
      "sequence:\n",
      "ACAACTTCAGTGCCGCCAAATAAAGAGAATAGCGCCAATCAATTTTCATTTGAAGTTGGCTTTCCCCAACAGATGTCGCCTTTGGGCCCGAATGTGGCAACGCAAG\n",
      "sequence:\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "[[2, 1]]\n",
      "start_coordinate : 12090713,stop_coordinate : 12090817\n",
      "28318 28425 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12090711,12090818,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_1_query_Exon_18,1,35,35,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_19\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "43\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 32736,32866\n",
      "after splice: 32736,32866\n",
      "sequence:\n",
      "CCTTTTTTGTCACCTGGACCGTATTTTAATGGAACGCACGCGCATTTGTATGAGGAAATGCAGCAGGAGAAAAACCATCCATCTACATCAAAGCAACCGCTTCCCCTACCTACCGACAACTACCAGGAG\n",
      "sequence:\n",
      "PFLSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PFLSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "[[2, 0]]\n",
      "start_coordinate : 12095131,stop_coordinate : 12095259\n",
      "32736 32866 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12095129,12095259,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_2_rightoh_0_query_Exon_19,1,43,43,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_20\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "49\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 34594,34740\n",
      "after splice: 34594,34740\n",
      "sequence:\n",
      "GCATGTAAACTCATTATTCCACCGAAGACGAGCAAGTCGATATCTTCCAGTGAATGGAGAAAAAAGAGAACGAAACTCCAGACACGTGCCAATAATATCAATGAAAGCAGTGGCGACAGTATATCATCAGCGAACGTTTCTAATTTG\n",
      "sequence:\n",
      "ACKLIIPPKTSKSISSSEWRKKRTKLQTRANNINESSGDSISSANVSNL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ACKLIIPPKTSKSISSSEWRKKRTKLQTRANNINESSGDSISSANVSNL\n",
      "[[0, 0]]\n",
      "start_coordinate : 12096987,stop_coordinate : 12097133\n",
      "34594 34740 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12096987,12097133,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_20,1,49,49,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_21\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 39029,39190\n",
      "after splice: 39029,39190\n",
      "sequence:\n",
      "AGTACAAACGCAGTTACTTTAAGCCAGAACAACGAGAAAGATCCGATTGCCAACAGCGGGACGAATGCAGATAAGACTGATGATGAATCTTGGTTTTCTTCATTTTCCTCTCACATATATAAGAGTGAAACTGGCAGCCCAGAGGAATCCGAAATGAAAAAG\n",
      "sequence:\n",
      "STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "[[0, 0]]\n",
      "start_coordinate : 12101422,stop_coordinate : 12101583\n",
      "39029 39190 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12101422,12101583,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_21,1,54,54,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_22\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 41375,41461\n",
      "after splice: 41375,41461\n",
      "sequence:\n",
      "AAATCTACGATATCATCACTACAAACCAATCGTCAAACAATAAATAAGGTAGCAATGAAGAAAATAGATCCCCCGTGGCTGGATGAT\n",
      "sequence:\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "[[0, 0]]\n",
      "start_coordinate : 12103768,stop_coordinate : 12103854\n",
      "41375 41461 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12103768,12103854,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_22,1,29,29,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_23\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 42720,42821\n",
      "after splice: 42720,42821\n",
      "sequence:\n",
      "GTGTGCCTTACTCCTgagataatatacaaatatcaaGTCAAAACCAGAAGCTTGGCGGAAGTACTTGCGGAAGACAAGAAGAAACTCGAAGCCCTAGAGCAG\n",
      "sequence:\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 12105113,stop_coordinate : 12105214\n",
      "42720 42821 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12105113,12105214,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_23,1,34,34,Y,N\n",
      "\n",
      "Leptidea_sinapis\n",
      "Processing Exon_24\n",
      "Leptidea_sinapis_coordinates_30.Leptidea_sinapis.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 44195,44326\n",
      "after splice: 44195,44326\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACTTCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACA\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNT\n",
      "left = ag, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "ccGAGCTTGGTCAACGAACAATTGAGCCAGTTGTATTACGAGCTCAAATTAGAGGGACTTCCGGAAAAATTGACTTTAGAGGAAGGAATAACGAGTTCTAGCAGTTCTGGTGATGATACCACTAGTAACACAGAG\n",
      "sequence:\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "[[0, 0]]\n",
      "start_coordinate : 12106588,stop_coordinate : 12106722\n",
      "44195 44329 Y N\n",
      "Leptidea_sinapis,NC_066312.1,12106588,12106722,0,N,30.Leptidea_sinapis_XM_050828181.1_Frame_0_rightoh_0_query_Exon_24,1,45,45,Y,N\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess\n",
    "annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/6.Leptidea/1.Blast_result\"\n",
    "# annotated_genome_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "# species_list = [\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Bicyclus_anynana\"]\n",
    "\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "    \n",
    "blast_error = []\n",
    "# query_species = \n",
    "for annotated_species_name in species_list:\n",
    "    # annotated_species_name = \"Lasiommata_megera\"\n",
    "    species = annotated_species_name\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/6.Leptidea/1.Query\"\n",
    "    # query_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Query\"\n",
    "    query_species = \"30.Leptidea_sinapis\" \n",
    "    query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "    if \"desktop.ini\" in query_transcript_list:\n",
    "        query_transcript_list.remove(\"desktop.ini\")\n",
    "    \n",
    "    query_transcript = query_transcript_list[0]\n",
    "    \n",
    "    genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "    # genome_location = \"/mnt/g/Genomes_2023-12-26\"\n",
    "    list_of_exons_folders = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    # print(list_of_exons_folders)\n",
    "    error_exon_list = []\n",
    "    for folders in list_of_exons_folders:\n",
    "        \n",
    "        if folders.startswith(\"Exon\"):\n",
    "            if int(folders.split(\"_\")[1]) > 1:\n",
    "                error_exon_list.append(folders)\n",
    "    print(error_exon_list)\n",
    "    # break\n",
    "    \n",
    "    \n",
    "#     error_exon_list = [\"Exon_5\"]\n",
    "    for error_exon in error_exon_list:\n",
    "        print(annotated_species_name)\n",
    "        print(f\"Processing {error_exon}\")\n",
    "        coordinate_location = f\"{annotated_genome_location}/{species}\"\n",
    "        files_in_coordinate_location = os.listdir(coordinate_location)\n",
    "        coordinate_backup = 0\n",
    "        query_coordinate = 0\n",
    "        # print(files_in_coordinate_location)\n",
    "        for file_names in files_in_coordinate_location:\n",
    "            if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                coordinate_backup = 1\n",
    "            if file_names.endswith(f\"_coordinates_{query_species}.csv\"):\n",
    "                print(file_names)\n",
    "                query_coordinate = 1\n",
    "        print(query_coordinate, \"query coordinate\")\n",
    "        if query_coordinate == 0:\n",
    "            print(file_names)\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates_old.csv\" \"{coordinate_location}/{species}_coordinates_{query_species}.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        if coordinate_backup == 0:\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates.csv\" \"{coordinate_location}/{species}_coordinates_old.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        \n",
    "        query_error = run_blast_with_new_query(annotated_genome_location,\n",
    "                                     annotated_species_name,\n",
    "                                     error_exon, \n",
    "                                     query_species,\n",
    "                                     genome_location, \n",
    "                                     species )\n",
    "    \n",
    "        if query_error == \"Query_error\":\n",
    "            print(query_error)\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        left_overhang,right_overhang,original_query_name = get_query_info(query_location,query_species,query_transcript,error_exon)\n",
    "        \n",
    "        try:\n",
    "            start_coordinate,stop_coordinate,gt_ag, splice_prediction = process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                                                                                       annotated_species_name,\n",
    "                                                                                                       error_exon,\n",
    "                                                                                                       left_overhang,\n",
    "                                                                                                       right_overhang,                                                                                               \n",
    "                                                                                                       original_query_name,\n",
    "                                                                                                      query_species)\n",
    "\n",
    "            new_coordinate_file_line = process_genome_blast_file(annotated_genome_location, \n",
    "                                                         annotated_species_name,\n",
    "                                                         error_exon,\n",
    "                                                         left_overhang,\n",
    "                                                         right_overhang,\n",
    "                                                         gt_ag, \n",
    "                                                         splice_prediction,\n",
    "                                                         original_query_name,\n",
    "                                                            query_species)\n",
    "        except:\n",
    "            print(\"Error in Blast\")\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        print(start_coordinate,stop_coordinate,gt_ag, splice_prediction)\n",
    "        print(new_coordinate_file_line)\n",
    "        output = ''\n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'r') as open_coordinate_file:\n",
    "            coordinate_file_list = open_coordinate_file.readlines()\n",
    "            # print(coordinate_file_list)\n",
    "            # assert False\n",
    "        for lines in coordinate_file_list:\n",
    "            if lines.split(\",\")[6].endswith(error_exon):\n",
    "                lines = new_coordinate_file_line\n",
    "            output += lines\n",
    "        \n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'w') as out_coordinate_file:\n",
    "            out_coordinate_file.write(output)\n",
    "\n",
    "\n",
    "print(\"\\n\".join(blast_error))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c490af29-b58c-4481-8429-8779374d7ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, \"1\", splice_prediction,original_query_name, query_species )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4fff37c-451f-48f4-8a15-eba962efc380",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
