{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14f15e7c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_15\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        overhang = int(records.id.split(\"_\")[-3])\n",
    "        \n",
    "        translated_sequence = records.seq[overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        alignment_start, alignment_stop = alignments[0].coordinates[1][1],alignments[0].coordinates[1][2]\n",
    "        print(records.seq[alignment_start*3-overhang+1:])\n",
    "#         \n",
    "#         print(sequence)\n",
    "        current_position = (alignment_start)*3\n",
    "        if (sequence[current_position-2:current_position]).lower() == \"ag\":\n",
    "            print(\"AG\\n\\n\\n\")\n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_start)*3)\n",
    "                else:\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_start)*3)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3-overhang+1:]}'\n",
    "                print(output)\n",
    "                out_file.write(output)\n",
    "#             assert False\n",
    "            break\n",
    "            \n",
    "            \n",
    "#         while current_position >0:\n",
    "# #             print((sequence[current_position:]))\n",
    "#             print((sequence[current_position-2:current_position]).lower(), end = \",\")\n",
    "# #             print((sequence[current_position+1:current_position+3]).lower()== \"ag\", end = \",\")\n",
    "#             if (sequence[current_position-2:current_position]).lower() == \"ag\":\n",
    "#                 break\n",
    "#             print(current_position,len(sequence))\n",
    "#             if current_position >= len(sequence):\n",
    "#                 break\n",
    "#             current_position -= 3\n",
    "# #             assert False\n",
    "#         print(sequence[current_position+3:])\n",
    "#         assert False\n",
    "#     break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dfc6fc1",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_7\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        l_overhang = int(records.id.split(\"_\")[-3])\n",
    "        r_overhang = int(records.id.split(\"_\")[-1])\n",
    "        translated_sequence = records.seq[l_overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        print(alignments[0].coordinates)\n",
    "        alignment_start = int(input(\"Alignment Start?\"))\n",
    "        alignment_stop = int(input(\"Alignment stop?\"))\n",
    "#         alignment_start, alignment_stop = alignments[0].coordinates[1][1],alignments[0].coordinates[1][2]\n",
    "        print(records.seq[alignment_start*3:(alignment_stop)*3+r_overhang])\n",
    "#         assert False\n",
    "#         \n",
    "#         print(sequence)\n",
    "#         current_position = (alignment_start)*3\n",
    "        found = 0\n",
    "        while alignment_stop < len(records.seq):\n",
    "        \n",
    "            if (sequence[(alignment_stop)*3+r_overhang:(alignment_stop)*3+r_overhang+2]).lower() == \"gt\":\n",
    "                print(\"GT\")\n",
    "                print(records.seq[alignment_start*3:(alignment_stop)*3+r_overhang])\n",
    "                print((records.seq[alignment_start*3:(alignment_stop)*3+r_overhang]).translate())\n",
    "                found = 1\n",
    "                break\n",
    "            alignment_stop+= 1\n",
    "        if found == 1:   \n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_stop)*3)\n",
    "                else:\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_stop)*3)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3:(alignment_stop)*3+r_overhang]}'\n",
    "                print(output+\"\\n\\n\")\n",
    "                out_file.write(output)\n",
    "# #             assert False\n",
    "#             break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71a58540",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_4\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        l_overhang = int(records.id.split(\"_\")[-3])\n",
    "        r_overhang = int(records.id.split(\"_\")[-1])\n",
    "        translated_sequence = records.seq[l_overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        print(alignments[0].coordinates)\n",
    "#         alignment_start = int(input(\"Alignment Start?\"))\n",
    "#         alignment_stop = int(input(\"Alignment stop?\"))\n",
    "        alignment_start, alignment_stop = alignments[0].coordinates[1][1],int(len(records.seq)/3)\n",
    "        print(alignment_start, alignment_stop)\n",
    "        check = input(\"This OK?\")\n",
    "        if check[0].lower() == \"n\":\n",
    "            alignment_start = int(input(\"Alignment Start?\"))\n",
    "#             alignment_stop = int(input(\"Alignment stop?\"))\n",
    "        print(records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang])\n",
    "#         assert False\n",
    "#         \n",
    "#         print(sequence)\n",
    "#         current_position = (alignment_start)*3\n",
    "        found = 0\n",
    "    \n",
    "        while alignment_start > 0:\n",
    "            print(sequence[(alignment_start)*3+l_overhang:(alignment_start)*3+l_overhang+3].lower())\n",
    "            if (sequence[(alignment_start)*3+l_overhang:(alignment_start)*3+l_overhang+3].lower()) == \"atg\":\n",
    "                print(\"ATG\")\n",
    "                print(records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang])\n",
    "                print((records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang]).translate())\n",
    "                found = 1\n",
    "                break\n",
    "            alignment_start-= 1\n",
    "        if found == 1:   \n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_start)*3+l_overhang)\n",
    "                else:\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_start)*3-l_overhang)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang]}'\n",
    "                print(output+\"\\n\\n\")\n",
    "                out_file.write(output)\n",
    "# # #             assert False\n",
    "# #             break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7383825b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
