{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "513c93f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'coordinates_file' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_2160\\3986772126.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     66\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     67\u001b[0m     \u001b[1;31m# Opening coordinate file\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 68\u001b[1;33m     \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"{location}/{folders}/{coordinates_file}\"\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mcoordinates_file_open\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     69\u001b[0m         \u001b[0mcoordinates_file_lines\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcoordinates_file_open\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadlines\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     70\u001b[0m     \u001b[0mstart_coordinate\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'coordinates_file' is not defined"
     ]
    }
   ],
   "source": [
    "import os\n",
    "location_root = \"H:/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast\"\n",
    "location = f\"{location_root}/1.Blast_output\"\n",
    "genome_location = \"F:/Genomes_2023-12-26\"\n",
    "list_of_dirs = os.listdir(location)\n",
    "list_of_dirs = [\"Bicyclus_anynana\"]\n",
    "\n",
    "def check_annotation_for_matching_exon(genome_location,genome,annotation_file,scaffold,start_coordinate,stop_coordinate):\n",
    "    with open(f\"{genome_location}/{genome}/{annotation_file}\") as annotation_file_open:\n",
    "        annotation_file_lines = annotation_file_open.readlines()\n",
    "        scaffold_found = 0\n",
    "        old_scaffold = ''\n",
    "        for line in (annotation_file_lines):\n",
    "            if scaffold in line:\n",
    "                \n",
    "                \n",
    "                if \"CDS\" in line:\n",
    "                    scaffold_found = 1\n",
    "                    line_split = line.split(\"\\t\")\n",
    "                    if start_coordinate in line_split and stop_coordinate in line_split:\n",
    "#                         line_split = line.split(\"\\t\")\n",
    "#                         test_coordinate_star\n",
    "                        print(start_coordinate, stop_coordinate)\n",
    "                        print(line)\n",
    "                        return (\"Match\")\n",
    "            \n",
    "            elif scaffold_found == 1:\n",
    "                return(\"No Match\")\n",
    "        return(\"No Match\")\n",
    "\n",
    "# print(list_of_dirs)\n",
    "output = \"Species\\tExon1\\tExon2\\tExon3\\tExon4\\tExon5\\tExon6\\tExon7\\tExon8\\tExon9\\tExon10\\tExon11\\tExon12\\tExon13\\tExon14\\tExon15\\tExon16\\tExon17\\tExon18\\tExon19\\tExon20\\tExon21\\tExon22\\tExon23\\tExon24\"\n",
    "for folders in list_of_dirs:\n",
    "    if \"supporting_files\" in folders:\n",
    "        list_of_dirs.remove(folders)\n",
    "    \n",
    "    \n",
    "    if \".ini\" in folders:\n",
    "        list_of_dirs.remove(folders)\n",
    "    \n",
    "# list_of_dirs = [\"Pararge_aegeria\"]\n",
    "for folders in list_of_dirs:\n",
    "    print(folders)\n",
    "    genome = ''\n",
    "    annotation_file = ''\n",
    "    match_list = []\n",
    "    prediction_list = []\n",
    "    \n",
    "#         print(folders)\n",
    "    list_of_files = os.listdir(f\"{location}/{folders}\")\n",
    "    for file_name in list_of_files:\n",
    "        if \"_coordinates.csv\" in file_name:\n",
    "            coordinates_file = file_name\n",
    "            if \".\" in folders:\n",
    "                genome = folders.split(\".\")[1]\n",
    "            else:\n",
    "                genome = folders\n",
    "#         print(genome)\n",
    "    genome_files = os.listdir(f\"{genome_location}/{genome}\")\n",
    "#     print(genome_files)\n",
    "    for files in genome_files:\n",
    "        if \".gff\" in files:\n",
    "#             print(files)\n",
    "            annotation_file = files\n",
    "    \n",
    "    \n",
    "    # Opening coordinate file\n",
    "    with open(f\"{location}/{folders}/{coordinates_file}\") as coordinates_file_open:\n",
    "        coordinates_file_lines = coordinates_file_open.readlines()\n",
    "    start_coordinate = 0\n",
    "    stop_coordinate = 0\n",
    "    for line_number in range(1,len(coordinates_file_lines)):\n",
    "        coordinates_file_lines_list = (coordinates_file_lines[line_number]).split(\",\")\n",
    "        exon = coordinates_file_lines_list[6]\n",
    "        scaffold = coordinates_file_lines_list[1]\n",
    "        start_coordinate = coordinates_file_lines_list[2]\n",
    "        stop_coordinate = coordinates_file_lines_list[3]\n",
    "        complement = coordinates_file_lines_list[4]\n",
    "        prediction = coordinates_file_lines_list[5]\n",
    "        print(exon)\n",
    "        match  = check_annotation_for_matching_exon(genome_location,genome,annotation_file,scaffold,start_coordinate,stop_coordinate)\n",
    "#         print(f\"{genome} {(exon,match)}\")\n",
    "        match_list.append((exon,match))\n",
    "        prediction_list.append((exon,prediction))\n",
    "    \n",
    "    #fill output file\n",
    "#     print(match_list)\n",
    "#     assert False\n",
    "    output += f\"\\n{genome} True\"\n",
    "    for exon_result in match_list:\n",
    "        exon_match = exon_result[1]\n",
    "        output += f\"\\t{exon_match}\"\n",
    "    \n",
    "    output += f\"\\n{genome} Predicted\"\n",
    "    for prediction_result in prediction_list:\n",
    "        prediction_match = prediction_result[1]\n",
    "        if prediction_match == \"N\":\n",
    "            prediction_match = \"Match\"\n",
    "        else:\n",
    "            prediction_match = \"No Match\"\n",
    "        \n",
    "        output += f\"\\t{prediction_match}\"\n",
    "    print(output)\n",
    "#     assert False\n",
    "with open(f\"{location_root}/3.Blast_processed/exon_match.csv\",'w') as out_file:\n",
    "    out_file.write(output)\n",
    "    \n",
    "#         print(match_list)\n",
    "#         assert False\n",
    "        #Opening annotation file\n",
    "        \n",
    "#         print(exon)\n",
    "#     \n",
    "        \n",
    "    \n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "331f2ad9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4784a43e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
