{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "619b7f03",
   "metadata": {},
   "outputs": [],
   "source": [
    "family_group = \"5.Coliadinae\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e7832546",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "   \n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp\")\n",
    "\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\"')\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\", \"fasta\")\n",
    "    # genome = SeqIO.parse(\"/mnt/c/Users/sauba/Downloads/Compressed/ncbi_dataset_7/ncbi_dataset/data/GCA_917051295.2/GCA_917051295.2_ilEreLige1.2_genomic.fna\", 'fasta')\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    return (gene_sequence)\n",
    "\n",
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)   \n",
    "\n",
    "def get_sequence(blast_output_location,genome_location, species, genome_file,annotated_genome_location,annotated_species_name):\n",
    "    list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        if file_names.endswith(\"_coordinates.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "\n",
    "    with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "\n",
    "    upstream_exon_line =coordinate_file_lines[1].split(\",\")\n",
    "    if upstream_exon_line[5] == \"Y\":\n",
    "        print(\"First Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    downstream_exon_line = coordinate_file_lines[-1].split(\",\")\n",
    "    if downstream_exon_line[5] == \"Y\":\n",
    "        print(\"Last Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    \n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        \n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        \n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    padding_error = ''\n",
    "    \n",
    "    if gene_start < 100000:\n",
    "        print(f\"Error{gene_start}\\n Changed Padding\")\n",
    "        print(species)\n",
    "        assert False\n",
    "    gene_start -= 100000\n",
    "    gene_end +=  100000\n",
    "    \n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    print(\"BB\")\n",
    "    return(gene_sequence,scaffold,gene_start,gene_end,padding_error )\n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2b83123a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_species_folder(output_location, species):\n",
    "    list_of_folders = os.listdir(f\"{output_location}/5.For Email_padded/\")\n",
    "    if species not in list_of_folders:\n",
    "#         subprocess.run(f'rm -r \"{output_location}/5.For Email_padded/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#         import time\n",
    "\n",
    "#         time.sleep(5)\n",
    "        os.mkdir(f\"{output_location}/5.For Email_padded/{species}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e0414369",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "ename": "UnboundLocalError",
     "evalue": "local variable 'gene_sequence' referenced before assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 23\u001b[0m\n\u001b[1;32m     20\u001b[0m     genome_file \u001b[38;5;241m=\u001b[39m get_genome_file(genome_location,species)\n\u001b[1;32m     21\u001b[0m     annotated_species_name \u001b[38;5;241m=\u001b[39m get_annotated_genome_name(annotated_genome_location, species)\n\u001b[0;32m---> 23\u001b[0m     gene_sequence,scaffold, gene_start, gene_end, padding_error \u001b[38;5;241m=\u001b[39m \u001b[43mget_sequence\u001b[49m\u001b[43m(\u001b[49m\u001b[43mblast_output_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgenome_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     24\u001b[0m \u001b[43m                                                               \u001b[49m\u001b[43mannotated_genome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mannotated_species_name\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     25\u001b[0m     error_species \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpadding_error\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     26\u001b[0m \u001b[38;5;66;03m#     print(gene_sequence,scaffold, gene_start, gene_end)\u001b[39;00m\n",
      "Cell \u001b[0;32mIn[3], line 73\u001b[0m, in \u001b[0;36mget_sequence\u001b[0;34m(blast_output_location, genome_location, species, genome_file, annotated_genome_location, annotated_species_name)\u001b[0m\n\u001b[1;32m     70\u001b[0m gene_start \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m100000\u001b[39m\n\u001b[1;32m     71\u001b[0m gene_end \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m  \u001b[38;5;241m100000\u001b[39m\n\u001b[0;32m---> 73\u001b[0m gene_sequence \u001b[38;5;241m=\u001b[39m \u001b[43mget_gene_sequence\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgenome_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mscaffold\u001b[49m\u001b[43m,\u001b[49m\u001b[43mgene_start\u001b[49m\u001b[43m,\u001b[49m\u001b[43mgene_end\u001b[49m\u001b[43m,\u001b[49m\u001b[43mcomplement\u001b[49m\u001b[43m,\u001b[49m\u001b[43mannotated_genome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mannotated_species_name\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mBB\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     75\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m(gene_sequence,scaffold,gene_start,gene_end,padding_error )\n",
      "Cell \u001b[0;32mIn[3], line 22\u001b[0m, in \u001b[0;36mget_gene_sequence\u001b[0;34m(genome_location, species, genome_file, scaffold, gene_start, gene_end, complement, annotated_genome_location, annotated_species_name)\u001b[0m\n\u001b[1;32m     20\u001b[0m         gene_sequence \u001b[38;5;241m=\u001b[39m gene_sequence\u001b[38;5;241m.\u001b[39mreverse_complement()\n\u001b[1;32m     21\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[0;32m---> 22\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m (\u001b[43mgene_sequence\u001b[49m)\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: local variable 'gene_sequence' referenced before assignment"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import io\n",
    "import subprocess\n",
    "genome_location =  \"/mnt/f/Genomes_2023-12-26\"\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "\n",
    "blast_output_location = f\"{output_location}/1.Blast_result\"\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "\n",
    "error_species = ''\n",
    "# list_of_species = [\"Anthocharis_cardamines\"]\n",
    "for species in list_of_species:\n",
    "    make_species_folder(output_location, species)\n",
    "    annotated_genome_location = f\"{output_location}/1.Blast_result\"\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "\n",
    "    gene_sequence,scaffold, gene_start, gene_end, padding_error = get_sequence(blast_output_location,genome_location, species, genome_file,\n",
    "                                                               annotated_genome_location,annotated_species_name)\n",
    "    error_species += f\"{padding_error}\\n\"\n",
    "#     print(gene_sequence,scaffold, gene_start, gene_end)\n",
    "    with open(f\"{output_location}/5.For Email_padded/{species}/Genome_fragment.fa\",'w') as out_file:\n",
    "        output = f\">{species}_{scaffold}_{gene_start}_{gene_end}\\n{gene_sequence}\"\n",
    "        out_file.write(output)\n",
    "\n",
    "with open(f\"{output_location}/5.For Email_padded/error_species.txt\",'w') as error_out_file:\n",
    "    error_out_file.write(error_species)\n",
    "    \n",
    "    \n",
    "#     local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "#     cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "#         # # os.system(f'{cd_command}')\n",
    "#     subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "#         # # print(mkdb_command)\n",
    "#         # # os.system(f'{mkdb_command}')\n",
    "#     # print(gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a4b9d370",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/5.For Email_padded/Colias_croceus/Genome_fragment.fa'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 15\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;66;03m#     print(gene_start)\u001b[39;00m\n\u001b[1;32m     14\u001b[0m     output \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m---> 15\u001b[0m     genome_fragment_file \u001b[38;5;241m=\u001b[39m \u001b[43mSeqIO\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparse\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43moutput_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/5.For Email_padded/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/Genome_fragment.fa\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfasta\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     16\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m sequences \u001b[38;5;129;01min\u001b[39;00m genome_fragment_file:\n\u001b[1;32m     17\u001b[0m         genome_name \u001b[38;5;241m=\u001b[39m sequences\u001b[38;5;241m.\u001b[39mid\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/Bio/SeqIO/__init__.py:613\u001b[0m, in \u001b[0;36mparse\u001b[0;34m(handle, format, alphabet)\u001b[0m\n\u001b[1;32m    611\u001b[0m iterator_generator \u001b[38;5;241m=\u001b[39m _FormatToIterator\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;28mformat\u001b[39m)\n\u001b[1;32m    612\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m iterator_generator:\n\u001b[0;32m--> 613\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43miterator_generator\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    614\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mformat\u001b[39m \u001b[38;5;129;01min\u001b[39;00m AlignIO\u001b[38;5;241m.\u001b[39m_FormatToIterator:\n\u001b[1;32m    615\u001b[0m     \u001b[38;5;66;03m# Use Bio.AlignIO to read in the alignments\u001b[39;00m\n\u001b[1;32m    616\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m (r \u001b[38;5;28;01mfor\u001b[39;00m alignment \u001b[38;5;129;01min\u001b[39;00m AlignIO\u001b[38;5;241m.\u001b[39mparse(handle, \u001b[38;5;28mformat\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m r \u001b[38;5;129;01min\u001b[39;00m alignment)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/Bio/SeqIO/FastaIO.py:190\u001b[0m, in \u001b[0;36mFastaIterator.__init__\u001b[0;34m(self, source, alphabet)\u001b[0m\n\u001b[1;32m    188\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m alphabet \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    189\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mThe alphabet argument is no longer supported\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 190\u001b[0m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__init__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mt\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfmt\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mFasta\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/Bio/SeqIO/Interfaces.py:58\u001b[0m, in \u001b[0;36mSequenceIterator.__init__\u001b[0;34m(self, source, alphabet, mode, fmt)\u001b[0m\n\u001b[1;32m     56\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mThe alphabet argument is no longer supported\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     57\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(source, _PathLikeTypes):\n\u001b[0;32m---> 58\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstream \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     59\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mshould_close_stream \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m     60\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/5.For Email_padded/Colias_croceus/Genome_fragment.fa'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Anthocharis_cardamines\"]\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "#     print(gene_start)\n",
    "    output = ''\n",
    "    genome_fragment_file = SeqIO.parse(f\"{output_location}/5.For Email_padded/{species}/Genome_fragment.fa\", \"fasta\")\n",
    "    for sequences in genome_fragment_file:\n",
    "        genome_name = sequences.id\n",
    "        fragment_length = len(sequences.seq)\n",
    "        print(fragment_length)\n",
    "    list_of_variant_files = os.listdir(f\"{output_location}/4.Bed File/{species}/\")\n",
    "    if 'desktop.ini' in list_of_variant_files:\n",
    "        list_of_variant_files.remove('desktop.ini')\n",
    "#     start = 0\n",
    "    \n",
    "    for variant in list_of_variant_files:    \n",
    "        with open(f\"{output_location}/4.Bed File/{species}/{variant}\", 'r') as var_file:\n",
    "            var_list = var_file.readlines()\n",
    "        \n",
    "        if int((var_list[0].split(\"\\t\")[1])) <int((var_list[1].split(\"\\t\")[1])):\n",
    "            complement = 0\n",
    "            gene_start = int((var_list[0].split(\"\\t\")[1]))\n",
    "        else:\n",
    "            complement = 1\n",
    "            gene_start = int((var_list[-1].split(\"\\t\")[1])) \n",
    "        output = ''\n",
    "#         print(var_list)\n",
    "        for gene_line in  var_list:\n",
    "            \n",
    "            gene_line_split = gene_line.split(\"\\t\")\n",
    "#             print(gene_line, var_list)\n",
    "            gene_line_split[0] = genome_name\n",
    "            if complement == 0:\n",
    "                gene_line_split[1] = str(int(gene_line_split[1]) - gene_start + 100000)\n",
    "                gene_line_split[2] = str(int(gene_line_split[2]) - gene_start + 100000)\n",
    "            if complement == 1:\n",
    "                old_gene_line_split_1 = gene_line_split[1]\n",
    "                gene_line_split[1] = str(fragment_length - (int(gene_line_split[2]) - gene_start + 100000))\n",
    "                gene_line_split[2] = str(fragment_length - (int(old_gene_line_split_1) - gene_start + 100000))\n",
    "            output += \"\\t\".join(gene_line_split)\n",
    "        \n",
    "#         print(output)\n",
    "        with open(f\"{output_location}/5.For Email_padded/{species}/{variant}\", 'w') as out_file:\n",
    "            out_file.write(output)\n",
    "#         print(complement, gene_start)\n",
    "#         assert False\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f157c3a4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "098ed3a0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
