{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "81fcfca7-185d-4284-bb3f-d800723fc4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b257780-eea0-47d9-b846-654cff8fa8e1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9993f28b-5774-4b4f-a38f-63c393b15b08",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length):\n",
    "    for offset in range(3):\n",
    "        translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "        for i in range(len(translated_sequence)):\n",
    "            if len(translated_sequence[i])> 0.8*query_length:\n",
    "                sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "                # print(i+1, offset)\n",
    "                \n",
    "                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                    output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "236f726f-7cfc-4839-871c-505018689085",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_mafft(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    location = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\n",
    "    list_of_files_to_run_mafft_on = os.listdir(location)\n",
    "    for file in list_of_files_to_run_mafft_on:\n",
    "        if file.endswith(\".fa\"):\n",
    "            command = f'\"mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "            # print(command)\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            # os.system(f'{command}')\n",
    "            command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/{file}\" > \"{location}/alignment_clustal_{file}.txt\"'\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            \n",
    "            # os.system(f'{command}')\n",
    "    return(location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "df06ca7c-0540-4e5e-a02b-2da57d4eecfb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def process_mafft_output(mafft_run_folder, error_exon):\n",
    "    list_of_files_in_mafft_run_folder = os.listdir(mafft_run_folder)\n",
    "    score_output = []\n",
    "    score = 99\n",
    "    min_score_sequence = ''\n",
    "    alignment_file = ''\n",
    "    for file in list_of_files_in_mafft_run_folder:\n",
    "        if file.endswith(\".fa.hat2\"):\n",
    "            with io.open(f\"{mafft_run_folder}/{file}\", 'r') as dist_matrix_file:\n",
    "                dist_matrix_list = dist_matrix_file.readlines()\n",
    "            \n",
    "            sequence_name = dist_matrix_list[-2].rstrip().split(\"=\")[1]\n",
    "            distance_score = float(dist_matrix_list[-1].rstrip())\n",
    "            if len(score_output) < 5:\n",
    "                score_output.append( [sequence_name,distance_score])\n",
    "            else:\n",
    "                for i in range(len(score_output)):\n",
    "                    score_at_this_index = score_output[i][1]\n",
    "                    if distance_score < score_at_this_index:\n",
    "                        score_output[i] = [sequence_name,distance_score]\n",
    "                        break\n",
    "            if distance_score < score:\n",
    "                score = distance_score\n",
    "                min_score_sequence = sequence_name\n",
    "                alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "    # print(score_output)\n",
    "    print(f\"min = {min_score_sequence}, {score}\" )\n",
    "    # score_out_merged = '\\n'.join(score_output)\n",
    "    print(f\"5 top scores:\\n{score_output}\")\n",
    "    print(f\"{mafft_run_folder}/{alignment_file}\")\n",
    "    \n",
    "    alignment_file = SeqIO.parse(f\"{mafft_run_folder}/{alignment_file}\", 'fasta')\n",
    "    # print (records.id)\n",
    "    fasta_start_position = 0\n",
    "    fasta_end_position = 0\n",
    "    start_switch = 0\n",
    "    end_switch = 0\n",
    "    alignment_name = ''\n",
    "    for records in alignment_file:\n",
    "        \n",
    "        \n",
    "        \n",
    "        if error_exon in records.id:\n",
    "            gap_counter = 0\n",
    "            base_counter = 0\n",
    "            for current_position in range(len(records.seq)):\n",
    "                sequence_length = len(records.seq) - records.seq.count('-')\n",
    "                # print(f\"fasta_end_position {fasta_end_position} fasta_start_position {fasta_start_position}\")\n",
    "                # print(f\"current_position = {current_position}, {len(records.seq)}\")\n",
    "                # print(records.seq[current_position])\n",
    "                # print(\"fasta_start_position\",fasta_start_position)\n",
    "                # print(\"start_switch\",start_switch)\n",
    "                \n",
    "                # print(gap_counter, base_counter)\n",
    "                # print(5,0.2*sequence_length)\n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and gap_counter > 3 and (len(records.seq[:current_position]) - records.seq[:current_position].count('-')) < (0.1*sequence_length):\n",
    "                #  :\n",
    "                    # print(\"\\n\\nhere\\n\\n\")\n",
    "                    start_switch = 0\n",
    "                    gap_counter = 0\n",
    "\n",
    "                if \"-\" not in records.seq[current_position] and start_switch == 0 :\n",
    "                    fasta_start_position = current_position\n",
    "                    start_switch = 1\n",
    "                \n",
    "                    \n",
    "                \n",
    "                    \n",
    "                if end_switch == 1 and \"-\" not in records.seq[current_position] and (len(records.seq[current_position:]) - records.seq[current_position:].count('-')) > 0.1*sequence_length :\n",
    "                    end_switch = 0\n",
    "                    gap_counter = 0\n",
    "                \n",
    "                    \n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0:\n",
    "                    # print(f\"base_counter {base_counter}\")\n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    fasta_end_position = current_position\n",
    "                    end_switch = 1\n",
    "                if \"-\" in records.seq[current_position]:\n",
    "                    gap_counter += 1\n",
    "                else:\n",
    "                    gap_counter = 0\n",
    "                    base_counter += 1\n",
    "        else:\n",
    "            alignment_name = records.id\n",
    "            if fasta_end_position == 0:\n",
    "                \n",
    "                fasta_end_position = current_position\n",
    "            end_switch = 1\n",
    "            start_switch = 1\n",
    "\n",
    "        if start_switch == 1 and end_switch == 1:\n",
    "            print(fasta_start_position, fasta_end_position)\n",
    "            gene_sequence = records.seq[fasta_start_position:fasta_end_position]\n",
    "            print(f\"{records.id}\\n{gene_sequence}\")\n",
    "    return(gene_sequence, alignment_name)\n",
    "\n",
    "# process_mafft_output(mafft_run_folder, error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "87615294-93f0-483b-b9f7-fad66730cd1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n",
    "def check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    \n",
    "    list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/\")\n",
    "    \n",
    "    if error_exon not in list_of_exon_directories:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "    elif error_exon in list_of_exon_directories:\n",
    "        list_of_folders_1 = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        if \"for_alignment\" not in list_of_folders_1:\n",
    "            os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        for file in list_of_files:\n",
    "            os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50742e4c-f9ab-433d-96a1-bde1f9d1c255",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "    print(\"Getting Gene\")\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp\")\n",
    "\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    # subprocess.run(\"pwd\")\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3cb097ff-9369-40fd-9ae5-7587c792ba31",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mafft_process(previous_exon_coordinates,\n",
    "                  next_exon_coordinates,\n",
    "                  current_exon_coordinates,\n",
    "                  query_species,\n",
    "                  query_transcript,\n",
    "                  query_location,\n",
    "                  annotated_genome_location,\n",
    "                 genome_location):\n",
    "    query_exon = current_exon_coordinates[6].split(\"query\")[-1][1:]\n",
    "    error_exon = query_exon\n",
    "    # print(\"AA\")\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    upstream_exon_line, downstream_exon_line = previous_exon_coordinates.split(\",\"),next_exon_coordinates.split(\",\")\n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    \n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "        original_query = query_file_list[1]\n",
    "        \n",
    "    query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "    \n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "    # print(annotated_species_name)\n",
    "\n",
    "    # print(annotated_genome_location)\n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    # assert False\n",
    "    check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon)\n",
    "    \n",
    "    \n",
    "    make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length)\n",
    "    \n",
    "    return(annotated_species_name, error_exon,species, genome_file,gene_sequence,left_overhang,right_overhang,scaffold, original_query_name,original_query )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e3790f71-89c6-4326-a178-d4147df81ed0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12340826,12340909,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_2,1,27,27\n",
      "\n",
      "1\n",
      "4\n",
      "Colias_croceus,NC_059568.1,12338695,12338780,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Colias_croceus,NC_059568.1,12346414,12346502,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_4,1,29,29\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set43_frame1, 0.143\n",
      "5 top scores:\n",
      "[['set43_frame1', 0.143], ['set13_frame2', 1.363], ['set18_frame2', 1.587], ['set145_frame1', 1.656], ['set107_frame1', 1.679]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_2/for_alignment/alignment_Exon_2_translated_genomic_sequence_43_frame1.fa.txt\n",
      "5 32\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_2\n",
      "RSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "5 32\n",
      "set43_frame1\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Original query = RSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "New query = RSSHSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12345251,12345405,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_1_query_Exon_3,1,51,51\n",
      "\n",
      "1\n",
      "5\n",
      "Colias_croceus,NC_059568.1,12338695,12338780,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Colias_croceus,NC_059568.1,12346988,12347044,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_5,1,19,19\n",
      "\n",
      "Getting Gene\n",
      "min = set145_frame1, 1.554\n",
      "5 top scores:\n",
      "[['set145_frame1', 1.554], ['set57_frame2', 1.799], ['set185_frame2', 1.814], ['set45_frame0', 1.816], ['set93_frame1', 1.83]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_3/for_alignment/alignment_Exon_3_translated_genomic_sequence_145_frame1.fa.txt\n",
      "7 65\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_1_query_Exon_3\n",
      "INNESQPPEKRSKDKDAKKKIPLQGLQTDLKVTEVINENECVPEIAPIFESSKEDISD\n",
      "7 65\n",
      "set145_frame1\n",
      "SNNVSQPPVELRCKVKKKKKQPEETL-------EVQEEEPSPPEPEPEVKPVEEECVD\n",
      "Original query = INNESQPPEKRSKDKDAKKKIPLQGLQTDLKVTEVINENECVPEIAPIFESSKEDISD\n",
      "New query = SNNVSQPPVELRCKVKKKKKQPEETL-------EVQEEEPSPPEPEPEVKPVEEECVD\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12346414,12346502,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_4,1,29,29\n",
      "\n",
      "2\n",
      "6\n",
      "Colias_croceus,NC_059568.1,12340826,12340909,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_2,1,27,27\n",
      "\n",
      "Colias_croceus,NC_059568.1,12348126,12348257,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_6,1,44,44\n",
      "\n",
      "Getting Gene\n",
      "min = set127_frame2, 1.493\n",
      "5 top scores:\n",
      "[['set127_frame2', 1.493], ['set147_frame2', 1.65], ['set130_frame2', 1.699], ['set136_frame2', 1.721], ['set23_frame0', 1.745]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_4/for_alignment/alignment_Exon_4_translated_genomic_sequence_127_frame2.fa.txt\n",
      "7 38\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_0_query_Exon_4\n",
      "AQTQSPGLVQPEKGPETMEVS--DRTSGKDE\n",
      "7 38\n",
      "set127_frame2\n",
      "AAASSPAQPAEASCVESMDISTPERSSVKDK\n",
      "Original query = AQTQSPGLVQPEKGPETMEVSDRTSGKDE\n",
      "New query = AAASSPAQPAEASCVESMDISTPERSSVKDK\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12346988,12347044,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_5,1,19,19\n",
      "\n",
      "3\n",
      "7\n",
      "Colias_croceus,NC_059568.1,12345251,12345405,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_1_query_Exon_3,1,51,51\n",
      "\n",
      "Colias_croceus,NC_059568.1,12349535,12349610,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Getting Gene\n",
      "min = set62_frame0, 1.628\n",
      "5 top scores:\n",
      "[['set62_frame0', 1.628], ['set43_frame2', 1.644], ['set84_frame0', 1.675], ['set88_frame2', 1.679], ['set6_frame2', 1.695]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_5/for_alignment/alignment_Exon_5_translated_genomic_sequence_62_frame0.fa.txt\n",
      "15 40\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_5\n",
      "PVVCDTPVGTPVMFVTSDPNSTQTY\n",
      "15 40\n",
      "set62_frame0\n",
      "SCVISMQDGV-VMYTTSSLTNTLGF\n",
      "Original query = PVVCDTPVGTPVMFVTSDPNSTQTY\n",
      "New query = SCVISMQDGV-VMYTTSSLTNTLGF\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12348126,12348257,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_6,1,44,44\n",
      "\n",
      "4\n",
      "8\n",
      "Colias_croceus,NC_059568.1,12346414,12346502,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_4,1,29,29\n",
      "\n",
      "Colias_croceus,NC_059568.1,12351686,12351877,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_8,1,63,63\n",
      "\n",
      "Getting Gene\n",
      "min = set38_frame1, 0.171\n",
      "5 top scores:\n",
      "[['set38_frame1', 0.171], ['set51_frame1', 1.738], ['set67_frame1', 1.775], ['set74_frame0', 1.768], ['set6_frame2', 1.789]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_6/for_alignment/alignment_Exon_6_translated_genomic_sequence_38_frame1.fa.txt\n",
      "12 56\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_6\n",
      "DGFSCVISLQDGIVMYTTSSITTALGFPKDMWIGRSFIDFVHQR\n",
      "12 56\n",
      "set38_frame1\n",
      "DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "Original query = DGFSCVISLQDGIVMYTTSSITTALGFPKDMWIGRSFIDFVHQR\n",
      "New query = DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12349535,12349610,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "5\n",
      "9\n",
      "Colias_croceus,NC_059568.1,12346988,12347044,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_5,1,19,19\n",
      "\n",
      "Colias_croceus,NC_059568.1,12352091,12352319,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_2_query_Exon_9,1,75,75\n",
      "\n",
      "Getting Gene\n",
      "min = set57_frame1, 0.23\n",
      "5 top scores:\n",
      "[['set57_frame1', 0.23], ['set29_frame2', 1.499], ['set86_frame2', 1.545], ['set79_frame2', 1.654], ['set72_frame2', 1.686]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_7/for_alignment/alignment_Exon_7_translated_genomic_sequence_57_frame1.fa.txt\n",
      "31 56\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_1_query_Exon_7\n",
      "DRNTFASQITNGLAAPKSVNGTQTK\n",
      "31 56\n",
      "set57_frame1\n",
      "DRNTFASLITNGLAAPKTVNGTQKK\n",
      "Original query = DRNTFASQITNGLAAPKSVNGTQTK\n",
      "New query = DRNTFASLITNGLAAPKTVNGTQKK\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12351686,12351877,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_8,1,63,63\n",
      "\n",
      "6\n",
      "10\n",
      "Colias_croceus,NC_059568.1,12348126,12348257,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_6,1,44,44\n",
      "\n",
      "Colias_croceus,NC_059568.1,12353151,12353262,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "Getting Gene\n",
      "min = set79_frame2, 0.557\n",
      "5 top scores:\n",
      "[['set79_frame2', 0.557], ['set64_frame2', 1.826], ['set84_frame2', 1.83], ['set82_frame0', 1.717], ['set109_frame1', 1.803]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_8/for_alignment/alignment_Exon_8_translated_genomic_sequence_79_frame2.fa.txt\n",
      "0 70\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_8\n",
      "PDL---GDHVSTMVCRIRRYKGLNTGFGIKDRVVTYMPFLLKFFFKNMNEDEGQVMYLVIQASPFFSAFK\n",
      "0 70\n",
      "set79_frame2\n",
      "LVIYSKGNTVSTMVCRIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "Original query = PDLGDHVSTMVCRIRRYKGLNTGFGIKDRVVTYMPFLLKFFFKNMNEDEGQVMYLVIQASPFFSAFK\n",
      "New query = LVIYSKGNTVSTMVCRIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12352091,12352319,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_2_query_Exon_9,1,75,75\n",
      "\n",
      "7\n",
      "11\n",
      "Colias_croceus,NC_059568.1,12349535,12349610,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Colias_croceus,NC_059568.1,12353380,12353499,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "Getting Gene\n",
      "min = set51_frame2, 0.524\n",
      "5 top scores:\n",
      "[['set24_frame0', 1.889], ['set46_frame2', 1.833], ['set51_frame2', 0.524], ['set71_frame2', 1.885]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_9/for_alignment/alignment_Exon_9_translated_genomic_sequence_51_frame2.fa.txt\n",
      "20 95\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_2_query_Exon_9\n",
      "PNEIVTNPKPFVMRHSPSGRLDYIDPESVPYLGYLPQDIIEKDVLLLYHPSDLFYLRQVYETIVKEGGEQRSKPY\n",
      "20 95\n",
      "set51_frame2\n",
      "PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "Original query = PNEIVTNPKPFVMRHSPSGRLDYIDPESVPYLGYLPQDIIEKDVLLLYHPSDLFYLRQVYETIVKEGGEQRSKPY\n",
      "New query = PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12353151,12353262,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "8\n",
      "12\n",
      "Colias_croceus,NC_059568.1,12351686,12351877,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_8,1,63,63\n",
      "\n",
      "Colias_croceus,NC_059568.1,12353838,12353987,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_12,1,50,50\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting Gene\n",
      "min = set26_frame0, 0.136\n",
      "5 top scores:\n",
      "[['set2_frame2', 1.783], ['set6_frame0', 1.73], ['set17_frame1', 1.809], ['set26_frame0', 0.136], ['set26_frame2', 1.816]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_10/for_alignment/alignment_Exon_10_translated_genomic_sequence_26_frame0.fa.txt\n",
      "56 93\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_0_query_Exon_10\n",
      "MMAQNGDYLKLETEWTSFINPWSRKLEFVIGKHFIIE\n",
      "56 93\n",
      "set26_frame0\n",
      "MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "Original query = MMAQNGDYLKLETEWTSFINPWSRKLEFVIGKHFIIE\n",
      "New query = MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12353380,12353499,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "9\n",
      "13\n",
      "Colias_croceus,NC_059568.1,12352091,12352319,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_2_query_Exon_9,1,75,75\n",
      "\n",
      "Colias_croceus,NC_059568.1,12354335,12354453,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Getting Gene\n",
      "min = set24_frame2, 0.821\n",
      "5 top scores:\n",
      "[['set24_frame2', 0.821], ['set32_frame1', 1.8], ['set9_frame2', 1.81], ['set26_frame0', 1.823], ['set32_frame0', 1.831]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_11/for_alignment/alignment_Exon_11_translated_genomic_sequence_24_frame2.fa.txt\n",
      "0 40\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_11\n",
      "GPTNPDVFQTPEPDKFFKKYTEEETKAQAIRENMIRIMNE\n",
      "0 40\n",
      "set24_frame2\n",
      "GPNNPDVFQSADIEKSFKFSEDEKNKAQTLRESIVRIMNE\n",
      "Original query = GPTNPDVFQTPEPDKFFKKYTEEETKAQAIRENMIRIMNE\n",
      "New query = GPNNPDVFQSADIEKSFKFSEDEKNKAQTLRESIVRIMNE\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12353838,12353987,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_12,1,50,50\n",
      "\n",
      "10\n",
      "14\n",
      "Colias_croceus,NC_059568.1,12353151,12353262,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "Colias_croceus,NC_059568.1,12354586,12354684,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_14,1,32,32\n",
      "\n",
      "Getting Gene\n",
      "min = set14_frame0, 0.559\n",
      "5 top scores:\n",
      "[['set14_frame0', 0.559], ['set27_frame0', 1.85], ['set4_frame1', 1.78], ['set31_frame2', 1.808]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_12/for_alignment/alignment_Exon_12_translated_genomic_sequence_14_frame0.fa.txt\n",
      "14 64\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_12\n",
      "VLTKPAEAAKQQMSKRCQDLASFMESLIEEAPKPEEELRLVIQDADHSCY\n",
      "14 64\n",
      "set14_frame0\n",
      "LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "Original query = VLTKPAEAAKQQMSKRCQDLASFMESLIEEAPKPEEELRLVIQDADHSCY\n",
      "New query = LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12354335,12354453,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "11\n",
      "15\n",
      "Colias_croceus,NC_059568.1,12353380,12353499,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "Colias_croceus,NC_059568.1,12355091,12355231,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_15,1,46,46\n",
      "\n",
      "Getting Gene\n",
      "min = set25_frame2, 0.331\n",
      "5 top scores:\n",
      "[['set25_frame2', 0.331], ['set30_frame2', 1.741], ['set19_frame0', 1.722], ['set18_frame1', 1.821], ['set29_frame0', 1.754]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_13/for_alignment/alignment_Exon_13_translated_genomic_sequence_25_frame2.fa.txt\n",
      "6 45\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_2_query_Exon_13\n",
      "ERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYNENLQ\n",
      "6 45\n",
      "set25_frame2\n",
      "ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "Original query = ERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYNENLQ\n",
      "New query = ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12354586,12354684,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_14,1,32,32\n",
      "\n",
      "12\n",
      "16\n",
      "Colias_croceus,NC_059568.1,12353838,12353987,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_12,1,50,50\n",
      "\n",
      "Colias_croceus,NC_059568.1,12356800,12356977,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_16,1,59,59\n",
      "\n",
      "Getting Gene\n",
      "min = set17_frame0, 1.634\n",
      "5 top scores:\n",
      "[['set29_frame1', 1.682], ['set53_frame1', 1.711], ['set58_frame2', 1.724], ['set17_frame0', 1.634], ['set79_frame2', 1.739]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_14/for_alignment/alignment_Exon_14_translated_genomic_sequence_17_frame0.fa.txt\n",
      "8 42\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_14\n",
      "YFDSHEPYNVEKTDALAICDHANTSEMQEPVAGT\n",
      "8 42\n",
      "set17_frame0\n",
      "YFDSHLLFSEENNSDTPSDRPSGNKNIKTNSSGY\n",
      "Original query = YFDSHEPYNVEKTDALAICDHANTSEMQEPVAGT\n",
      "New query = YFDSHLLFSEENNSDTPSDRPSGNKNIKTNSSGY\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12355091,12355231,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_15,1,46,46\n",
      "\n",
      "13\n",
      "17\n",
      "Colias_croceus,NC_059568.1,12354335,12354453,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Colias_croceus,NC_059568.1,12357405,12357552,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_17,1,49,49\n",
      "\n",
      "Getting Gene\n",
      "min = set16_frame0, 1.453\n",
      "5 top scores:\n",
      "[['set16_frame0', 1.453], ['set56_frame2', 1.764], ['set23_frame2', 1.825], ['set26_frame2', 1.841], ['set62_frame0', 1.829]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_15/for_alignment/alignment_Exon_15_translated_genomic_sequence_16_frame0.fa.txt\n",
      "5 52\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_15\n",
      "RCILSNPGELDERASTSDSNAAAGASTSRAGTSIEYPALRLTESLLN\n",
      "5 52\n",
      "set16_frame0\n",
      "SPMAQNSGDFGDMTSSCDSSALALSNSTPMGDY--YQPVRLTESALS\n",
      "Original query = RCILSNPGELDERASTSDSNAAAGASTSRAGTSIEYPALRLTESLLN\n",
      "New query = SPMAQNSGDFGDMTSSCDSSALALSNSTPMGDY--YQPVRLTESALS\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12356800,12356977,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_16,1,59,59\n",
      "\n",
      "14\n",
      "18\n",
      "Colias_croceus,NC_059568.1,12354586,12354684,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_14,1,32,32\n",
      "\n",
      "Colias_croceus,NC_059568.1,12358110,12358223,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_18,1,37,37\n",
      "\n",
      "Getting Gene\n",
      "min = set51_frame2, 0.642\n",
      "5 top scores:\n",
      "[['set51_frame2', 0.642], ['set76_frame2', 1.825], ['set21_frame2', 1.856], ['set63_frame0', 1.699], ['set35_frame1', 1.882]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_16/for_alignment/alignment_Exon_16_translated_genomic_sequence_51_frame2.fa.txt\n",
      "15 70\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_0_query_Exon_16\n",
      "HNIDMEKELLKKHRETRSSSKGEREKASNESRQKKKEHLARCNALYQPTTAGLSA\n",
      "15 70\n",
      "set51_frame2\n",
      "HNAVMEKELMKLHREIKTVTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLST\n",
      "Original query = HNIDMEKELLKKHRETRSSSKGEREKASNESRQKKKEHLARCNALYQPTTAGLSADK\n",
      "New query = HNAVMEKELMKLHREIKTVTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLST\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12357405,12357552,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_17,1,49,49\n",
      "\n",
      "15\n",
      "19\n",
      "Colias_croceus,NC_059568.1,12355091,12355231,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_15,1,46,46\n",
      "\n",
      "Colias_croceus,NC_059568.1,12358864,12358971,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_19,1,35,35\n",
      "\n",
      "Getting Gene\n",
      "min = set50_frame2, 1.382\n",
      "5 top scores:\n",
      "[['set50_frame2', 1.382], ['set42_frame2', 1.748], ['set54_frame0', 1.758], ['set65_frame0', 1.812], ['set9_frame1', 1.822]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_17/for_alignment/alignment_Exon_17_translated_genomic_sequence_50_frame2.fa.txt\n",
      "19 65\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_1_query_Exon_17\n",
      "AHGLKRASKQTEDGTT-HKHRCSSPRNRRRFSQKVTNSPPLQATNN\n",
      "19 65\n",
      "set50_frame2\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRR--KQSTSAAPAHPSNT\n",
      "Original query = AHGLKRASKQTEDGTTHKHRCSSPRNRRRFSQKVTNSPPLQATNN\n",
      "New query = PHGLKRSSRHSDDSETGYKYRYSTAKLPRR--KQSTSAAPAHPSNT\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12358110,12358223,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_18,1,37,37\n",
      "\n",
      "16\n",
      "20\n",
      "Colias_croceus,NC_059568.1,12356800,12356977,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_0_query_Exon_16,1,59,59\n",
      "\n",
      "Colias_croceus,NC_059568.1,12359458,12359621,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_20,1,54,54\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting Gene\n",
      "min = set27_frame1, 1.009\n",
      "5 top scores:\n",
      "[['set27_frame1', 1.009], ['set9_frame2', 1.754], ['set40_frame2', 1.798], ['set10_frame2', 1.824], ['set50_frame2', 1.834]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_18/for_alignment/alignment_Exon_18_translated_genomic_sequence_27_frame1.fa.txt\n",
      "5 40\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_18\n",
      "PSPWPPNP--VNNMNAFILGLGIPPQMSVMSPMTV\n",
      "5 40\n",
      "set27_frame1\n",
      "SSPMPTNSMNMNNMNTFFLGVSIPQQMSMMSPVAA\n",
      "Original query = PSPWPPNPVNNMNAFILGLGIPPQMSVMSPMTVP\n",
      "New query = SSPMPTNSMNMNNMNTFFLGVSIPQQMSMMSPVAA\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12358864,12358971,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_19,1,35,35\n",
      "\n",
      "17\n",
      "21\n",
      "Colias_croceus,NC_059568.1,12357405,12357552,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_1_query_Exon_17,1,49,49\n",
      "\n",
      "Colias_croceus,NC_059568.1,12362060,12362203,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_21,1,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set32_frame0, 1.61\n",
      "5 top scores:\n",
      "[['set14_frame2', 1.696], ['set54_frame1', 1.744], ['set32_frame0', 1.61], ['set57_frame1', 1.748], ['set49_frame0', 1.699]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_19/for_alignment/alignment_Exon_19_translated_genomic_sequence_32_frame0.fa.txt\n",
      "43 81\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_19\n",
      "ALPM-YYTPTSTQQMPSRFNPNTNLPH----AHYHRYQ\n",
      "43 81\n",
      "set32_frame0\n",
      "MFPICYAMPASSQQM---MQNDTEMPNQGPSTSNHPFT\n",
      "Original query = ALPMYYTPTSTQQMPSRFNPNTNLPHAHYHRYQ\n",
      "New query = MFPICYAMPASSQQM---MQNDTEMPNQGPSTSNHPFT\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12359458,12359621,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_20,1,54,54\n",
      "\n",
      "18\n",
      "22\n",
      "Colias_croceus,NC_059568.1,12358110,12358223,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_18,1,37,37\n",
      "\n",
      "Colias_croceus,NC_059568.1,12365953,12366114,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_22,1,54,54\n",
      "\n",
      "Getting Gene\n",
      "min = set31_frame1, 1.092\n",
      "5 top scores:\n",
      "[['set31_frame1', 1.092], ['set116_frame2', 1.762], ['set160_frame2', 1.769], ['set114_frame2', 1.852], ['set84_frame0', 1.803]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_20/for_alignment/alignment_Exon_20_translated_genomic_sequence_31_frame1.fa.txt\n",
      "0 57\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_0_query_Exon_20\n",
      "QC---MMLGQTVYGSPFMYSAIDPQ--MSYALQQSFVPQ-QTPNTQPLELSTSNYEE\n",
      "0 57\n",
      "set31_frame1\n",
      "LSVPYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "Original query = QCMMLGQTVYGSPFMYSAIDPQMSYALQQSFVPQQTPNTQPLELSTSNYEE\n",
      "New query = LSVPYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12362060,12362203,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_21,1,48,48\n",
      "\n",
      "19\n",
      "23\n",
      "Colias_croceus,NC_059568.1,12358864,12358971,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_1_query_Exon_19,1,35,35\n",
      "\n",
      "Colias_croceus,NC_059568.1,12370479,12370565,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_23,1,29,29\n",
      "\n",
      "Getting Gene\n",
      "min = set65_frame2, 1.56\n",
      "5 top scores:\n",
      "[['set65_frame2', 1.56], ['set50_frame0', 1.745], ['set257_frame2', 1.747], ['set151_frame1', 1.781], ['set38_frame2', 1.792]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_21/for_alignment/alignment_Exon_21_translated_genomic_sequence_65_frame2.fa.txt\n",
      "10 58\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_21\n",
      "ACKLTMPQKLGKSYFGNMRREKMAEVRANKGGHGRNIIGSSSVGASNR\n",
      "10 58\n",
      "set65_frame2\n",
      "ACKLTFPTK-SKLVPSSTWREKRAEQLAAAAKKDKSSGTSNSSAAVSN\n",
      "Original query = ACKLTMPQKLGKSYFGNMRREKMAEVRANKGGHGRNIIGSSSVGASNR\n",
      "New query = ACKLTFPTK-SKLVPSSTWREKRAEQLAAAAKKDKSSGTSNSSAAVSN\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12365953,12366114,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_22,1,54,54\n",
      "\n",
      "20\n",
      "24\n",
      "Colias_croceus,NC_059568.1,12359458,12359621,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_20,1,54,54\n",
      "\n",
      "Colias_croceus,NC_059568.1,12370857,12370958,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_24,1,34,34\n",
      "\n",
      "Getting Gene\n",
      "min = set131_frame2, 1.104\n",
      "5 top scores:\n",
      "[['set131_frame2', 1.104], ['set153_frame2', 1.799], ['set59_frame1', 1.823], ['set56_frame0', 1.794], ['set74_frame0', 1.816]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_22/for_alignment/alignment_Exon_22_translated_genomic_sequence_131_frame2.fa.txt\n",
      "6 16\n",
      "set131_frame2\n",
      "SANAATENRS\n",
      "Original query = STDTMTDERIKRRLDNSDSNNDKTDGESSYSSFYSSFFKTDSGSGSDTRHQNNKEDNKTP\n",
      "New query = SANAATENRS\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12370479,12370565,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_23,1,29,29\n",
      "\n",
      "21\n",
      "25\n",
      "Colias_croceus,NC_059568.1,12362060,12362203,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_21,1,48,48\n",
      "\n",
      "Colias_croceus,NC_059568.1,12371299,12371442,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_25,1,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set184_frame2, 1.29\n",
      "5 top scores:\n",
      "[['set184_frame2', 1.29], ['set30_frame1', 1.679], ['set14_frame2', 1.708], ['set105_frame2', 1.725], ['set182_frame1', 1.757]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_23/for_alignment/alignment_Exon_23_translated_genomic_sequence_184_frame2.fa.txt\n",
      "0 80\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_23\n",
      "YRH--KTNGKFWHNSDVSMTNV----------------------------------------ANARKIPRRKMDPPWMEQ\n",
      "0 80\n",
      "set184_frame2\n",
      "SKMYGGTVGTSTLTTNISIVMIDSYLKILNSSLKYLYFGIEILIIVVFLRQRSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "Original query = YRHKTNGKFWHNSDVSMTNVANARKIPRRKMDPPWMEQ\n",
      "New query = SKMYGGTVGTSTLTTNISIVMIDSYLKILNSSLKYLYFGIEILIIVVFLRQRSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12370857,12370958,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_24,1,34,34\n",
      "\n",
      "22\n",
      "25\n",
      "Colias_croceus,NC_059568.1,12365953,12366114,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_22,1,54,54\n",
      "\n",
      "Colias_croceus,NC_059568.1,12371299,12371442,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_25,1,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set107_frame0, 0.505\n",
      "5 top scores:\n",
      "[['set107_frame0', 0.505], ['set88_frame2', 1.648], ['set82_frame2', 1.687], ['set70_frame2', 1.717], ['set63_frame2', 1.741]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_24/for_alignment/alignment_Exon_24_translated_genomic_sequence_107_frame0.fa.txt\n",
      "3 37\n",
      "28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_24\n",
      "VCVTSELIYKYQIVTKNIDEVLSSDKEKLKQLEQ\n",
      "3 37\n",
      "set107_frame0\n",
      "VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "Original query = VCVTSELIYKYQIVTKNIDEVLSSDKEKLKQLEQ\n",
      "New query = VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12371299,12371442,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_25,1,48,48\n",
      "\n",
      "23\n",
      "25\n",
      "Colias_croceus,NC_059568.1,12370479,12370565,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_23,1,29,29\n",
      "\n",
      "Colias_croceus,NC_059568.1,12371299,12371442,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_0_query_Exon_25,1,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set18_frame2, 1.795\n",
      "5 top scores:\n",
      "[['set9_frame0', 1.805], ['set7_frame1', 1.811], ['set18_frame2', 1.795]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result/Colias_croceus/Period_gene_genomic_sequence_individual_exon_28.Maniola_jurtina/Exon_25/for_alignment/alignment_Exon_25_translated_genomic_sequence_18_frame2.fa.txt\n",
      "0 17\n",
      "set18_frame2\n",
      "V-----KLNSFHINLYL\n",
      "Original query = PSLVNEQLSQLYLDLQLEGVAARLTLEEGITSSSSSGEESTAKSSK\n",
      "New query = V-----KLNSFHINLYL\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold = NC_059568.1, start = 12338695, end = 12338780\n",
      "Colias_croceus,NC_059568.1,12340826,12340909,0,N,32.Colias_croceus_XM_045653731.1_Frame_1_rightoh_2_query_Exon_2,1,27,27\n",
      "\n",
      "1\n",
      "4\n",
      "Colias_croceus,NC_059568.1,12338695,12338780,0,N,32.Colias_croceus_XM_045653731.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Colias_croceus,NC_059568.1,12346414,12346502,0,N,32.Colias_croceus_XM_045653731.1_Frame_2_rightoh_0_query_Exon_4,1,29,29\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 112\u001b[0m\n\u001b[1;32m    102\u001b[0m \u001b[38;5;28mprint\u001b[39m(next_exon_coordinates)\n\u001b[1;32m    103\u001b[0m annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query \u001b[38;5;241m=\u001b[39m mafft_process(previous_exon_coordinates,\n\u001b[1;32m    104\u001b[0m                                                                                                   next_exon_coordinates,\n\u001b[1;32m    105\u001b[0m                                                                                                   current_exon_coordinates,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    109\u001b[0m                                                                                                   annotated_genome_location,\n\u001b[1;32m    110\u001b[0m                                                                                                      genome_location)\n\u001b[0;32m--> 112\u001b[0m mafft_run_folder \u001b[38;5;241m=\u001b[39m \u001b[43mrun_mafft\u001b[49m\u001b[43m(\u001b[49m\u001b[43mannotated_genome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mannotated_species_name\u001b[49m\u001b[43m,\u001b[49m\u001b[43merror_exon\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    114\u001b[0m possible_gene_sequence, alignment_name \u001b[38;5;241m=\u001b[39m process_mafft_output(mafft_run_folder, error_exon)\n\u001b[1;32m    116\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mOriginal query = \u001b[39m\u001b[38;5;132;01m{\u001b[39;00moriginal_query\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[0;32mIn[3], line 11\u001b[0m, in \u001b[0;36mrun_mafft\u001b[0;34m(annotated_genome_location, annotated_species_name, error_exon)\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[38;5;66;03m# os.system(f'{command}')\u001b[39;00m\n\u001b[1;32m     10\u001b[0m         command \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmafft\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m --localpair --clustalout --maxiterate 16 --reorder \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m > \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlocation\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/alignment_clustal_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.txt\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m---> 11\u001b[0m         \u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mcommand\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshell\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstderr\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDEVNULL\u001b[49m\u001b[43m)\u001b[49m \n\u001b[1;32m     13\u001b[0m         \u001b[38;5;66;03m# os.system(f'{command}')\u001b[39;00m\n\u001b[1;32m     14\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m(location)\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:491\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    489\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Popen(\u001b[38;5;241m*\u001b[39mpopenargs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs) \u001b[38;5;28;01mas\u001b[39;00m process:\n\u001b[1;32m    490\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 491\u001b[0m         stdout, stderr \u001b[38;5;241m=\u001b[39m \u001b[43mprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcommunicate\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    492\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m TimeoutExpired \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[1;32m    493\u001b[0m         process\u001b[38;5;241m.\u001b[39mkill()\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1016\u001b[0m, in \u001b[0;36mPopen.communicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1014\u001b[0m         stderr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mread()\n\u001b[1;32m   1015\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mclose()\n\u001b[0;32m-> 1016\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1017\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1018\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1079\u001b[0m, in \u001b[0;36mPopen.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1077\u001b[0m     endtime \u001b[38;5;241m=\u001b[39m _time() \u001b[38;5;241m+\u001b[39m timeout\n\u001b[1;32m   1078\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1079\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1080\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m:\n\u001b[1;32m   1081\u001b[0m     \u001b[38;5;66;03m# https://bugs.python.org/issue25942\u001b[39;00m\n\u001b[1;32m   1082\u001b[0m     \u001b[38;5;66;03m# The first keyboard interrupt waits briefly for the child to\u001b[39;00m\n\u001b[1;32m   1083\u001b[0m     \u001b[38;5;66;03m# exit under the common assumption that it also received the ^C\u001b[39;00m\n\u001b[1;32m   1084\u001b[0m     \u001b[38;5;66;03m# generated SIGINT and will exit rapidly.\u001b[39;00m\n\u001b[1;32m   1085\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1804\u001b[0m, in \u001b[0;36mPopen._wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1802\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1803\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m  \u001b[38;5;66;03m# Another thread waited.\u001b[39;00m\n\u001b[0;32m-> 1804\u001b[0m (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_try_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1805\u001b[0m \u001b[38;5;66;03m# Check the pid and loop as waitpid has been known to\u001b[39;00m\n\u001b[1;32m   1806\u001b[0m \u001b[38;5;66;03m# return 0 even without WNOHANG in odd situations.\u001b[39;00m\n\u001b[1;32m   1807\u001b[0m \u001b[38;5;66;03m# http://bugs.python.org/issue14396.\u001b[39;00m\n\u001b[1;32m   1808\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m pid \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1762\u001b[0m, in \u001b[0;36mPopen._try_wait\u001b[0;34m(self, wait_flags)\u001b[0m\n\u001b[1;32m   1760\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"All callers to this function MUST hold self._waitpid_lock.\"\"\"\u001b[39;00m\n\u001b[1;32m   1761\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1762\u001b[0m     (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwaitpid\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpid\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwait_flags\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1763\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mChildProcessError\u001b[39;00m:\n\u001b[1;32m   1764\u001b[0m     \u001b[38;5;66;03m# This happens if SIGCLD is set to be ignored or waiting\u001b[39;00m\n\u001b[1;32m   1765\u001b[0m     \u001b[38;5;66;03m# for child processes has otherwise been disabled for our\u001b[39;00m\n\u001b[1;32m   1766\u001b[0m     \u001b[38;5;66;03m# process.  This child is dead, we can't get the status.\u001b[39;00m\n\u001b[1;32m   1767\u001b[0m     pid \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "\n",
    "blast_output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result\"\n",
    "# blast_output_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "species_list = os.listdir(blast_output_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "for species in species_list:\n",
    "    annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result\"\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Query\"\n",
    "    list_of_query_species = os.listdir(query_location)\n",
    "    list_of_query_species.remove(\"desktop.ini\")\n",
    "    for query_species in list_of_query_species:\n",
    "        list_of_query_transcripts = os.listdir(f\"{query_location}/{query_species}\")\n",
    "        list_of_query_transcripts.remove(\"desktop.ini\")\n",
    "        for query_transcript in list_of_query_transcripts:\n",
    "\n",
    "\n",
    "            genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "\n",
    "\n",
    "\n",
    "            output_coordinate_file = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "            exons_to_check_list = [\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\",\"Exon_13\",\"Exon_14\",\"Exon_15\",\"Exon_16\",\"Exon_17\",\"Exon_18\",\"Exon_19\",\"Exon_20\",\"Exon_21\",\"Exon_22\",\"Exon_23\",\"Exon_24\",\"Exon_25\"]\n",
    "#             exons_to_check_list = [\"Exon_21\"]\n",
    "\n",
    "            list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "            coordinate_file_name = ''\n",
    "        #     print(list_of_files_in_species_folder)\n",
    "            for file_names in list_of_files_in_species_folder:\n",
    "\n",
    "                if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                    coordinate_file_name = file_names\n",
    "            if coordinate_file_name =='':\n",
    "                print(f\"Coordinate file error\")\n",
    "                assert False\n",
    "\n",
    "            with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "                coordinate_file_lines = temp_file_open.readlines()\n",
    "            # print(coordinate_file_lines[0])\n",
    "            for exons_to_check in exons_to_check_list:\n",
    "\n",
    "                for i in range(1,len(coordinate_file_lines)):\n",
    "                    coordinate_file_lines_split = coordinate_file_lines[i].split(\",\")\n",
    "                    current_exon_coordinates = coordinate_file_lines[i].split(\",\")\n",
    "                    if i == 1:\n",
    "                        scaffold_first_exon = coordinate_file_lines[i].split(\",\")[1]\n",
    "                        first_exon_start, first_exon_end = coordinate_file_lines[i].split(\",\")[2],coordinate_file_lines[i].split(\",\")[3]\n",
    "                    # if coordinate_file_lines_split[5] == \"Y\":\n",
    "                    if  coordinate_file_lines_split[6].endswith(exons_to_check):\n",
    "                        if i == 1:\n",
    "                            print(f'First Exon has errors\\n{coordinate_file_lines[i]}')\n",
    "                            assert False\n",
    "                        else:\n",
    "                            print(f\"Scaffold = {scaffold_first_exon}, start = {first_exon_start}, end = {first_exon_end}\")\n",
    "                            print(coordinate_file_lines[i])\n",
    "                            # process_current_exon = input(\"Error Found! Process?\")\n",
    "                            process_current_exon = 'y'\n",
    "                            if process_current_exon.lower()[0] == \"y\":\n",
    "                                if i>2:\n",
    "                                    previous_exon_number = i-2\n",
    "                                else:\n",
    "                                    previous_exon_number = i-1\n",
    "                                if i+2 <=int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = i+2\n",
    "                                else:\n",
    "                                    next_exon_number = i+1\n",
    "                                if i >= int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = int(exons_to_check_list[-1].split(\"_\")[1])\n",
    "#                                     continue\n",
    "\n",
    "                                '''\n",
    "                                the above code ensures that missing exons do not cause issues\n",
    "                                '''\n",
    "\n",
    "                                previous_exon_coordinates = ''\n",
    "                                next_exon_coordinates = ''\n",
    "                                while True:\n",
    "                                    print(previous_exon_number)\n",
    "                                    if coordinate_file_lines[previous_exon_number].split(\",\")[5] == \"N\":\n",
    "                                        previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                        break\n",
    "                                    # else:\n",
    "                                    #     proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[previous_exon_number]}\\nProceed?\")\n",
    "                                    #     if proceed_prompt.lower()[0] == \"y\":\n",
    "                                    #         previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                    #         break\n",
    "                                    previous_exon_number -= 1\n",
    "                                while True:\n",
    "                                    print(next_exon_number)\n",
    "                                    try:\n",
    "                                        if coordinate_file_lines[next_exon_number].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                                            break\n",
    "                                    except:\n",
    "                                        if coordinate_file_lines[next_exon_number-1].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number-1]\n",
    "                                            break\n",
    "                                   \n",
    "                                    next_exon_number += 1\n",
    "\n",
    "\n",
    "                                print(previous_exon_coordinates)\n",
    "                                print(next_exon_coordinates)\n",
    "                                annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query = mafft_process(previous_exon_coordinates,\n",
    "                                                                                                                                  next_exon_coordinates,\n",
    "                                                                                                                                  current_exon_coordinates,\n",
    "                                                                                                                                  query_species,\n",
    "                                                                                                                                  query_transcript,\n",
    "                                                                                                                                  query_location,\n",
    "                                                                                                                                  annotated_genome_location,\n",
    "                                                                                                                                     genome_location)\n",
    "\n",
    "                                mafft_run_folder = run_mafft(annotated_genome_location,annotated_species_name,error_exon)\n",
    "\n",
    "                                possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder, error_exon)\n",
    "\n",
    "                                print(f\"Original query = {original_query}\")\n",
    "                                print(f\"New query = {possible_gene_sequence}\")\n",
    "                                \n",
    "                                list_of_folders_inside_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "                                if \"for_blast\" not in list_of_folders_inside_exon_folder:\n",
    "                                    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                else:\n",
    "                                    list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                    for file in list_of_files:\n",
    "                                        os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/{file}\")\n",
    "\n",
    "\n",
    "                                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\",'w') as query_file:\n",
    "                                    query = f\">Query_{error_exon}_{alignment_name}\\n{possible_gene_sequence}\"\n",
    "                                    query_file.write(query)\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccdf583e-4fa4-4fac-a4d4-98a069f1d391",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307c7dce-21b5-432f-926a-024803e6dad6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
