{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "006dd81d-4a68-4fa1-9154-6e64cc0dd7fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Bicyclus_anynana', 'Coenonympha_glycerion', 'Erebia_aethiops', 'Erebia_ligea', 'Hipparchia_semele', 'Lasiommata_megera', 'Maniola_hyperantus', 'Maniola_jurtina', 'Melanargia_galathea', 'Oeneis_ivallda', 'Pararge_aegeria', 'Anthocharis_cardamines', 'Aporia_crataegi', 'Leptophobia_aripa', 'Pieris_brassicae', 'Pieris_mannii', 'Pieris_melete', 'Pieris_napi', 'Pieris_rapae', 'Colias_croceus', 'Colias_eurytheme', 'Colias_nastes', 'Phoebis_sennae', 'Zerene_cesonia', 'Leptidea_juvernica', 'Leptidea_reali', 'Leptidea_sinapis', 'Hestina_assimilis', 'Danaus_chrysippus', 'Dircenna_loreta', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Melinaea_marsaeus_rileyi', 'Melinaea_menophilus', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Argynnis_bischoffii_washingtonia', 'Boloria_euphrosyne', 'Boloria_selene', 'Brenthis_daphne', 'Brenthis_hecate', 'Brenthis_ino', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Eueides_isabella', 'Fabriciana_adippe', 'Heliconius_charithonia', 'Heliconius_nattereri', 'Philaethria_dido', 'Limenitis_camilla', 'Euphydryas_editha', 'Hypolimnas_misippus', 'Melitaea_cinxia', 'Mellicta_athalia', 'Nymphalis_c', 'Nymphalis_io', 'Nymphalis_polychloros', 'Nymphalis_urticae', 'Vanessa_atalanta', 'Vanessa_cardui', 'Vanessa_tameamea', 'Iphiclides_podalirius', 'Luehdorfia_chinensis', 'Ornithoptera_alexandrae', 'Ornithoptera_priamus', 'Papilio_dardanus_tibullus', 'Papilio_demoleus', 'Papilio_elwesi', 'Papilio_gigon', 'Papilio_machaon', 'Papilio_memnon', 'Papilio_protenor', 'Papilio_xuthus', 'Parnassius_glacialis', 'Sericinus_montela', 'Teinopalpus_imperialis', 'Troides_aeacus', 'Troides_oblongomaculatus', 'Aricia_agestis', 'Aricia_artaxerxes', 'Calycopis_cecrops', 'Celastrina_argiolus', 'Cyaniris_semiargus', 'Cyclargus_thomasi', 'Eumaeus_atala', 'Glaucopsyche_alexis', 'Helleia_helle', 'Lycaena_phlaeas', 'Lysandra_bellargus', 'Lysandra_coridon', 'Phengaris_arion', 'Plebejus_argus', 'Polyommatus_icarus', 'Polyommatus_iphigenia', 'Calephelis_nemesis', 'Calephelis_virginiensis']\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import Phylo\n",
    "import os\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/\"\n",
    "list_of_files = os.listdir(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/\")\n",
    "for file_name in list_of_files:\n",
    "    if file_name.endswith(\"_aligned.fas\"):\n",
    "        output = ''\n",
    "        sequence_file = SeqIO.parse(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/{file_name}\", 'fasta')\n",
    "        species_list_1 = []\n",
    "        for records in sequence_file:\n",
    "            species_list_1.append((\"_\".join(records.id.split(\"_\")[:-1])).split(\"-\")[0])\n",
    "            output += f\">{('_'.join(records.id.split('_')[:-1])).split('-')[0]}\\n{records.seq}\\n\\n\"\n",
    "        with open(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/2.Gene_sequences_final/ng_{file_name}\", 'w') as out_file:\n",
    "            out_file.write(output)\n",
    "\n",
    "print(species_list_1)    \n",
    "# with open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "27a79091-f6ac-4b55-9895-d7506b905520",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_plexippus\n"
     ]
    }
   ],
   "source": [
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny\"\n",
    "sequence_file = SeqIO.parse(f\"{location}/10.All_gene_sequences/cad.fas\", 'fasta')\n",
    "species_list_1 = []\n",
    "for records in sequence_file:\n",
    "    species = records.id.split('-')[0]\n",
    "    species_list_1.append(species)\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/\"\n",
    "period_sequence_file = SeqIO.parse(f\"{location}/6.Period Exon Analysis/11.Alignment/00.Alignment/period_aligned_prank.fas.best.fas\", 'fasta')\n",
    "\n",
    "for records in period_sequence_file:\n",
    "    species = records.id.split('-')[0]\n",
    "    if species not in species_list_1:\n",
    "        print(species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a45bedff-a438-4f48-9621-968c74cff5f0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
