{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "68b4eef9-b0f6-4717-83e0-f0f257b87e6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "74\n",
      "wg.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "cad.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "cat.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "ddc.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "ef1.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "gapdh.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "hcl.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "idh.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "mdh.fas.aligned.fas.best.fas Species_left =  ['Battus_philenor'] 73\n",
      "rps2.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "rps5.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "period.fas.aligned.fas.best.fas Species_left =  ['Lycaena_phlaeas'] 73\n",
      "timeless.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "clock.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "cycle.fas.aligned.fas.best.fas Species_left =  ['Teinopalpus_imperialis'] 73\n",
      "cwo.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "cry1a.fas.aligned.fas.best.fas Species_left =  ['Leptidea_reali'] 73\n",
      "cry1b.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "cry2.fas.aligned.fas.best.fas Species_left =  [] 74\n",
      "timeout.fas.aligned.fas.best.fas Species_left =  [] 74\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import Phylo\n",
    "import os\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization\"\n",
    "\n",
    "tree = Phylo.read(f\"{location}/2.For_branch_length_optimization_new/pruned_tree.nwk\", 'newick')\n",
    "species_list = [term.name for term in tree.get_terminals()]\n",
    "print(len(species_list))\n",
    "# for species in species_list:\n",
    "#     print(species)\n",
    "files_list = os.listdir(f\"{location}/2.Gene_sequence_pruned\")\n",
    "if \"desktop.ini\" in files_list:\n",
    "    files_list.remove(\"desktop.ini\")\n",
    "    \n",
    "for files in files_list:\n",
    "    fasta_file = SeqIO.parse(f\"{location}/2.Gene_sequence_pruned/{files}\", 'fasta')\n",
    "    tree = Phylo.read(f\"{location}/2.For_branch_length_optimization_new/pruned_tree.nwk\", 'newick')\n",
    "    species_list = [term.name for term in tree.get_terminals()]\n",
    "    species_2 = []\n",
    "    for records in fasta_file:\n",
    "        species_2.append(records.id)\n",
    "        for species in species_list:\n",
    "            if species == records.id:\n",
    "                # print(species)\n",
    "                species_list.remove(species)\n",
    "    print(files,\"Species_left = \",species_list, len(species_2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3acfe7af-c7b9-4449-aac8-837f36097889",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import Phylo\n",
    "import os\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization\"\n",
    "tree = Phylo.read(f\"{location}/2.For_branch_length_optimization_new/pruned_tree.nwk\", 'newick')\n",
    "\n",
    "tree.prune('Leptidea_reali')\n",
    "Phylo.write(tree, f\"{location}/2.For_branch_length_optimization_new/cty1a_pruned_tree.nwk\", \"newick\")\n",
    "\n",
    "# species_list = [term.name for term in tree.get_terminals()]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "26764208-7105-499c-9857-36fc9fc11152",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization\"\n",
    "file_list = os.listdir(f\"{location}/4.HYPHY_trees\")\n",
    "if \"desktop.ini\" in file_list:\n",
    "    file_list.remove(\"desktop.ini\")\n",
    "output = ''\n",
    "for file in file_list:\n",
    "    gene_name = file.split('.')[0]\n",
    "    with open(f\"{location}/4.HYPHY_trees/{file}\", 'r') as tree_file:\n",
    "        output += f\"{gene_name}\\t{tree_file.readlines()[0]}\\n\"\n",
    "with open(f\"{location}/combined_new2.nwk\",'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "7641e015-07a4-44e2-b255-da0836adcca1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pervasive [419, 505, 1242, 2033] [1183, 1194]\n",
      "Episodic [43, 421, 491, 502, 710, 726, 946, 1086, 1242, 1368, 1425, 1557, 1568, 1696, 1761, 1892, 1941, 1950, 1975, 2029, 2033, 2053] [356, 362, 544, 560, 752, 821, 1063, 1113, 1120, 1137, 1176, 1183, 1187, 1188, 1191, 1194, 1207]\n",
      "1207\n",
      "[1833, 2033]\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "from Bio import SeqIO\n",
    "gene = \"timeless\"\n",
    "gene_folder = \"7.Timeless Exon Analysis\"\n",
    "\n",
    "location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/{gene_folder}/\"\n",
    "family_list = [\"Lycaenidae\",\"Nymphalidae\",\"Papilionidae\",\"Pieridae\"]\n",
    "family = \"Nymphalidae\"\n",
    "# for family in family_list:\n",
    "\n",
    "trimmed_fasta_list = []\n",
    "untrimmed_fasta_list = []\n",
    "\n",
    "\n",
    "untimmed_fasta_file = SeqIO.parse(f\"{location}/12.Family_specific_tests_{gene}/{family}/{family}_{gene}_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "untrimmed_sequence_list = []\n",
    "for records in untimmed_fasta_file:\n",
    "    untrimmed_sequence_list.append(records.seq.translate())\n",
    "for i in range(len(untrimmed_sequence_list[0])):\n",
    "    dic_key = ''\n",
    "    for aligned_sequence in untrimmed_sequence_list:\n",
    "        dic_key += aligned_sequence[i]\n",
    "    untrimmed_fasta_list.append(dic_key)\n",
    "\n",
    "\n",
    "timmed_fasta_file = SeqIO.parse(f\"{location}/12.Family_specific_tests_{gene}/{family}/{family}_{gene}_nostop.fas.aligned.fas.trimal.fas\", 'fasta')\n",
    "trimmed_sequence_list = []\n",
    "for records in timmed_fasta_file:\n",
    "    trimmed_sequence_list.append(records.seq.translate())\n",
    "for i in range(len(trimmed_sequence_list[0])):\n",
    "    dic_key = ''\n",
    "    for aligned_sequence in trimmed_sequence_list:\n",
    "        dic_key += aligned_sequence[i]\n",
    "    trimmed_fasta_list.append(dic_key)\n",
    "# print(trimmed_fasta_dictionary)\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/03.FEL/{family}_{gene}_nostop.fas.aligned.fas.trimal.fas.FEL.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "pervasive_sites_trimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][4]) < 0.05:\n",
    "        if float(list_of_values[i][0]) < float(list_of_values[i][1]):\n",
    "            pervasive_sites_trimmed.append(i+1)\n",
    "\n",
    "\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/03.FEL/{family}_{gene}_nostop.fas.aligned.fas.best.fas.FEL.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "pervasive_sites_untrimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][4]) < 0.05:\n",
    "        if float(list_of_values[i][0]) < float(list_of_values[i][1]):\n",
    "            pervasive_sites_untrimmed.append(i+1)\n",
    "\n",
    "print(\"Pervasive\",pervasive_sites_untrimmed, pervasive_sites_trimmed)\n",
    "\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/04.MEME/{family}_{gene}_nostop.fas.aligned.fas.trimal.fas.MEME.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "episodic_sites_trimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][6]) < 0.05:\n",
    "        episodic_sites_trimmed.append(i+1)\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/04.MEME/{family}_{gene}_nostop.fas.aligned.fas.best.fas.MEME.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "episodic_sites_untrimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][6]) < 0.05:\n",
    "        \n",
    "        episodic_sites_untrimmed.append(i+1)\n",
    "\n",
    "print(\"Episodic\",episodic_sites_untrimmed, episodic_sites_trimmed)\n",
    "\n",
    "            \n",
    "    # break\n",
    "pervasive_old = pervasive_sites_trimmed\n",
    "pervasive = []\n",
    "episodic_old = episodic_sites_trimmed\n",
    "episodic = []\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/1.trimal_outputs/columns_kept.txt\", 'r') as column_file:\n",
    "    columns = column_file.readlines()[0].split(\"\\t\")[1].split(\",\")\n",
    "# print(columns)\n",
    "# for i in range(len(columns)):\n",
    "#     print(i,columns[i])\n",
    "for i in pervasive_old:\n",
    "    # print(columns[i])\n",
    "    pervasive.append(int(untrimmed_fasta_list.index(trimmed_fasta_list[int(i-1)]))+1)\n",
    "    # if int(columns[i]) in pervasive_sites_untrimmed:\n",
    "        \n",
    "    #     print(\"predicted from old = \",columns[i])\n",
    "    #     print(\"predicted from new = \",)\n",
    "    # else:\n",
    "    #     print(f\"Pervasive Site_trimmed = {i}, Site_untrimmed = {columns[i]}\")\n",
    "    #     print(\"predicted from new = \",untrimmed_fasta_list.index(trimmed_fasta_list[int(i)]))\n",
    "    #     pervasive.append(int(input(\"Proper Site\")))\n",
    "print(len(trimmed_fasta_list))\n",
    "for i in episodic_old:\n",
    "    episodic.append(untrimmed_fasta_list.index(trimmed_fasta_list[int(i-1)])+1)\n",
    "    # if int(columns[i]) in episodic_sites_untrimmed:\n",
    "    #     episodic.append(int(columns[i]))\n",
    "    #     print(\"predicted from new = \",untrimmed_fasta_list.index(trimmed_fasta_list[int(i)]))\n",
    "    # else:\n",
    "    #     print(f\"Episodic Site_trimmed = {i}, Site_untrimmed = {columns[i]}\")\n",
    "    #     episodic.append(int(input(\"Proper Site\")))\n",
    "    #     print(\"predicted from new = \",untrimmed_fasta_list.index(trimmed_fasta_list[int(i)]))\n",
    "print(pervasive)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f75b9d52-9051-4c81-8585-adf26dfde039",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/03.FEL/{family}_{gene}_nostop.fas.aligned.fas.trimal.fas.FEL.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "pervasive_sites_trimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][4]) < 0.05:\n",
    "        if float(list_of_values[i][0]) < float(list_of_values[i][1]):\n",
    "            pervasive_sites_trimmed.append(i+1)\n",
    "\n",
    "\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/03.FEL/{family}_{gene}_nostop.fas.aligned.fas.best.fas.FEL.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "pervasive_sites_untrimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][4]) < 0.05:\n",
    "        if float(list_of_values[i][0]) < float(list_of_values[i][1]):\n",
    "            pervasive_sites_untrimmed.append(i+1)\n",
    "\n",
    "print(\"Pervasive\",pervasive_sites_untrimmed, pervasive_sites_trimmed)\n",
    "\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/04.MEME/{family}_{gene}_nostop.fas.aligned.fas.trimal.fas.MEME.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "episodic_sites_trimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][6]) < 0.05:\n",
    "        episodic_sites_trimmed.append(i+1)\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/04.MEME/{family}_{gene}_nostop.fas.aligned.fas.best.fas.MEME.json\", 'r') as f:\n",
    "    json_file = json.load(f)\n",
    "\n",
    "list_of_values = (json_file[\"MLE\"][\"content\"][\"0\"])\n",
    "episodic_sites_untrimmed = []\n",
    "for i in range(len(list_of_values)):\n",
    "    if float(list_of_values[i][6]) < 0.05:\n",
    "        \n",
    "        episodic_sites_untrimmed.append(i+1)\n",
    "\n",
    "print(\"Episodic\",episodic_sites_untrimmed, episodic_sites_trimmed)\n",
    "\n",
    "            \n",
    "    # break\n",
    "pervasive_old = pervasive_sites_trimmed\n",
    "pervasive = []\n",
    "episodic_old = episodic_sites_trimmed\n",
    "episodic = []\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests_{gene}/{family}/1.trimal_outputs/columns_kept.txt\", 'r') as column_file:\n",
    "    columns = column_file.readlines()[0].split(\"\\t\")[1].split(\",\")\n",
    "# print(columns)\n",
    "# for i in range(len(columns)):\n",
    "#     print(i,columns[i])\n",
    "for i in pervasive_old:\n",
    "    # print(columns[i])\n",
    "    \n",
    "    if int(columns[i]) in pervasive_sites_untrimmed:\n",
    "        pervasive.append(int(columns[i]))\n",
    "    else:\n",
    "        print(f\"Pervasive Site_trimmed = {i}, Site_untrimmed = {columns[i]}\")\n",
    "        pervasive.append(int(input(\"Proper Site\")))\n",
    "for i in episodic_old:\n",
    "    if int(columns[i]) in episodic_sites_untrimmed:\n",
    "        episodic.append(int(columns[i]))\n",
    "    else:\n",
    "        print(f\"Episodic Site_trimmed = {i}, Site_untrimmed = {columns[i]}\")\n",
    "        episodic.append(int(input(\"Proper Site\")))\n",
    "print(pervasive)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
