{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f27d4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    out_Seq = ''\n",
    "    \n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = open((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'r')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in SeqIO.parse(fasta_file,\"fasta\"):\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = out_Seq.reverse_complement()\n",
    "                \n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    fasta_file.close()\n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4900aae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d454c091",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069086.1,11483690,11484343,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069085.1,15161126,15161650,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069088.1,14786334,14786852,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069091.1,4302552,4303541,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069087.1,7018552,7019775,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069085.1,7848095,7848799,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069101.1,8211585,8212580,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069083.1,16542533,16543918,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069095.1,13512390,13513166,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069110.1,13793257,13794717,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Bicyclus_anynana,NC_069087.1,9201226,9204366,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Bicyclus_anynana,NC_069086.1,11483690,11484343,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Bicyclus_anynana,NC_069085.1,15161126,15161650,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Bicyclus_anynana,NC_069088.1,14786334,14786852,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Bicyclus_anynana,NC_069091.1,4302552,4303541,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Bicyclus_anynana,NC_069087.1,7018552,7019775,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "Bicyclus_anynana,NC_069085.1,7848095,7848799,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Bicyclus_anynana,NC_069101.1,8211585,8212580,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Bicyclus_anynana,NC_069083.1,16542533,16543918,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Bicyclus_anynana,NC_069095.1,13512390,13513166,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Bicyclus_anynana,NC_069110.1,13793257,13794717,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Bicyclus_anynana,NC_069087.1,9201226,9204366,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979630.1,7820902,7821558,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979629.1,14988771,14989295,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979632.1,14129484,14130002,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979635.1,12735815,12736804,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979628.1,12388155,12389378,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979629.1,7749400,7750104,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979645.1,10857308,10858303,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979627.1,16008458,16009843,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979642.1,698669,699445,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979633.1,12279944,12281404,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion,OY979628.1,10475365,10478505,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Coenonympha_glycerion,OY979630.1,7820902,7821558,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Coenonympha_glycerion,OY979629.1,14988771,14989295,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Coenonympha_glycerion,OY979632.1,14129484,14130002,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Coenonympha_glycerion,OY979635.1,12735815,12736804,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Coenonympha_glycerion,OY979628.1,12388155,12389378,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Coenonympha_glycerion,OY979629.1,7749400,7750104,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Coenonympha_glycerion,OY979645.1,10857308,10858303,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Coenonympha_glycerion,OY979627.1,16008458,16009843,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Coenonympha_glycerion,OY979642.1,698669,699445,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Coenonympha_glycerion,OY979633.1,12279944,12281404,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Coenonympha_glycerion,OY979628.1,10475365,10478505,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000007.1,1016982,1017638,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000927.1,4751645,4752169,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000847.1,1601378,1601896,0,Y,Bicyclus_anynana_rps2,51,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000867.1,968690,969679,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000925.1,2247539,2248762,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000070.1,1882844,1883548,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000282.1,4061040,4062035,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000030.1,2553063,2554448,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000493.1,1793205,1793981,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000008.1,768485,769945,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Elymnias_hypermnestra,JACTOM010000152.1,1887649,1890789,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Elymnias_hypermnestra,JACTOM010000007.1,1016982,1017638,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Elymnias_hypermnestra,JACTOM010000927.1,4751645,4752169,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "Elymnias_hypermnestra,JACTOM010000847.1,1601378,1601896,0,Y,Bicyclus_anynana_rps2,51,173,173\n",
      "Elymnias_hypermnestra,JACTOM010000867.1,968690,969679,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Elymnias_hypermnestra,JACTOM010000925.1,2247539,2248762,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Elymnias_hypermnestra,JACTOM010000070.1,1882844,1883548,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Elymnias_hypermnestra,JACTOM010000282.1,4061040,4062035,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Elymnias_hypermnestra,JACTOM010000030.1,2553063,2554448,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Elymnias_hypermnestra,JACTOM010000493.1,1793205,1793981,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Elymnias_hypermnestra,JACTOM010000008.1,768485,769945,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Elymnias_hypermnestra,JACTOM010000152.1,1887649,1890789,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281084.1,18743085,18743741,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281090.1,15362590,15363114,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281092.1,14629826,14630344,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281093.1,13826017,13827006,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281091.1,12191359,12192582,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281090.1,8241243,8241947,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281096.1,4663891,4664886,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281089.1,16682893,16684278,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281087.1,8016928,8017704,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281079.1,13375704,13377167,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_aethiops,OV281091.1,10159919,10163059,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Erebia_aethiops,OV281084.1,18743085,18743741,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Erebia_aethiops,OV281090.1,15362590,15363114,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Erebia_aethiops,OV281092.1,14629826,14630344,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Erebia_aethiops,OV281093.1,13826017,13827006,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Erebia_aethiops,OV281091.1,12191359,12192582,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Erebia_aethiops,OV281090.1,8241243,8241947,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Erebia_aethiops,OV281096.1,4663891,4664886,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Erebia_aethiops,OV281089.1,16682893,16684278,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Erebia_aethiops,OV281087.1,8016928,8017704,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Erebia_aethiops,OV281079.1,13375704,13377167,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Erebia_aethiops,OV281091.1,10159919,10163059,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785220.1,8978403,8979059,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785219.1,17596455,17596979,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785224.1,16037586,16038104,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785227.1,15392748,15393737,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785223.1,13779872,13781095,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785219.1,9144482,9145186,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785238.1,4900149,4901144,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785226.1,2300356,2301741,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785232.1,848750,849526,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785222.1,7157559,7159019,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Erebia_ligea,OU785223.1,11304140,11307280,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Erebia_ligea,OU785220.1,8978403,8979059,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Erebia_ligea,OU785219.1,17596455,17596979,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Erebia_ligea,OU785224.1,16037586,16038104,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Erebia_ligea,OU785227.1,15392748,15393737,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Erebia_ligea,OU785223.1,13779872,13781095,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Erebia_ligea,OU785219.1,9144482,9145186,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Erebia_ligea,OU785238.1,4900149,4901144,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Erebia_ligea,OU785226.1,2300356,2301741,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Erebia_ligea,OU785232.1,848750,849526,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Erebia_ligea,OU785222.1,7157559,7159019,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "Erebia_ligea,OU785223.1,11304140,11307280,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121711.1,6975906,6976559,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121709.1,13598235,13598759,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121713.1,3786478,3786996,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121717.1,11087930,11088919,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121712.1,5946971,5948194,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121709.1,6965158,6965862,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121726.1,3753749,3754744,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121710.1,3011375,3012760,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121720.1,533162,533938,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121737.1,10203542,10205002,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Hipparchia_semele,OW121712.1,7770471,7773611,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Hipparchia_semele,OW121711.1,6975906,6976559,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Hipparchia_semele,OW121709.1,13598235,13598759,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Hipparchia_semele,OW121713.1,3786478,3786996,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Hipparchia_semele,OW121717.1,11087930,11088919,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Hipparchia_semele,OW121712.1,5946971,5948194,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "Hipparchia_semele,OW121709.1,6965158,6965862,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Hipparchia_semele,OW121726.1,3753749,3754744,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Hipparchia_semele,OW121710.1,3011375,3012760,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Hipparchia_semele,OW121720.1,533162,533938,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Hipparchia_semele,OW121737.1,10203542,10205002,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Hipparchia_semele,OW121712.1,7770471,7773611,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743311.1,11543352,11544005,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743309.1,4648199,4648723,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743310.1,4630672,4631190,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743315.1,13702087,13703076,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743312.1,12210320,12211543,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743309.1,12053849,12054553,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743325.1,11792062,11793057,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743308.1,16986516,16987901,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743320.1,648135,648911,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743307.1,10296459,10297919,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Lasiommata_megera,OV743312.1,10059843,10062983,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Lasiommata_megera,OV743311.1,11543352,11544005,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Lasiommata_megera,OV743309.1,4648199,4648723,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "Lasiommata_megera,OV743310.1,4630672,4631190,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Lasiommata_megera,OV743315.1,13702087,13703076,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Lasiommata_megera,OV743312.1,12210320,12211543,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Lasiommata_megera,OV743309.1,12053849,12054553,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Lasiommata_megera,OV743325.1,11792062,11793057,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Lasiommata_megera,OV743308.1,16986516,16987901,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Lasiommata_megera,OV743320.1,648135,648911,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Lasiommata_megera,OV743307.1,10296459,10297919,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "Lasiommata_megera,OV743312.1,10059843,10062983,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048539.1,10202149,10202805,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048536.1,4352786,4353310,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048541.1,3966082,3966600,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048543.1,11944985,11945974,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048538.1,11287973,11289196,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048536.1,11309462,11310166,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048554.1,3891938,3892933,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048537.1,3312241,3313626,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048546.1,665453,666229,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048564.1,10991294,10992757,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_hyperantus,NC_048538.1,9357346,9360486,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Maniola_hyperantus,NC_048539.1,10202149,10202805,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Maniola_hyperantus,NC_048536.1,4352786,4353310,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "Maniola_hyperantus,NC_048541.1,3966082,3966600,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Maniola_hyperantus,NC_048543.1,11944985,11945974,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Maniola_hyperantus,NC_048538.1,11287973,11289196,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Maniola_hyperantus,NC_048536.1,11309462,11310166,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Maniola_hyperantus,NC_048554.1,3891938,3892933,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Maniola_hyperantus,NC_048537.1,3312241,3313626,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Maniola_hyperantus,NC_048546.1,665453,666229,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Maniola_hyperantus,NC_048564.1,10991294,10992757,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Maniola_hyperantus,NC_048538.1,9357346,9360486,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060032.1,9743325,9743981,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060029.1,13386780,13387304,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060034.1,12449502,12450020,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060039.1,3931442,3932431,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060031.1,6199013,6200236,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060029.1,6955070,6955774,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060045.1,9898902,9899897,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060030.1,2967844,2969229,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060040.1,14514977,14515753,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060058.1,15091157,15092620,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Maniola_jurtina,NC_060031.1,8035326,8038466,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Maniola_jurtina,NC_060032.1,9743325,9743981,0,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Maniola_jurtina,NC_060029.1,13386780,13387304,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Maniola_jurtina,NC_060034.1,12449502,12450020,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Maniola_jurtina,NC_060039.1,3931442,3932431,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Maniola_jurtina,NC_060031.1,6199013,6200236,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "Maniola_jurtina,NC_060029.1,6955070,6955774,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Maniola_jurtina,NC_060045.1,9898902,9899897,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Maniola_jurtina,NC_060030.1,2967844,2969229,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Maniola_jurtina,NC_060040.1,14514977,14515753,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Maniola_jurtina,NC_060058.1,15091157,15092620,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "Maniola_jurtina,NC_060031.1,8035326,8038466,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049863.1,10562285,10562947,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049862.1,5936256,5936780,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049866.1,19012510,19013028,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049867.1,5791100,5792089,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049864.1,9295900,9297123,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049862.1,15576853,15577557,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049876.1,14341017,14342012,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049859.1,4737416,4738801,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049870.1,21268797,21269573,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049860.1,17416576,17418036,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Melanargia_galathea,OV049864.1,12320829,12323969,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Melanargia_galathea,OV049863.1,10562285,10562947,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Melanargia_galathea,OV049862.1,5936256,5936780,0,N,Bicyclus_anynana_wg,1,175,175\n",
      "Melanargia_galathea,OV049866.1,19012510,19013028,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Melanargia_galathea,OV049867.1,5791100,5792089,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Melanargia_galathea,OV049864.1,9295900,9297123,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "Melanargia_galathea,OV049862.1,15576853,15577557,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Melanargia_galathea,OV049876.1,14341017,14342012,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Melanargia_galathea,OV049859.1,4737416,4738801,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Melanargia_galathea,OV049870.1,21268797,21269573,1,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Melanargia_galathea,OV049860.1,17416576,17418036,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Melanargia_galathea,OV049864.1,12320829,12323969,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000005.1,18297517,18298173,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000002.1,7243721,7244245,0,N,Bicyclus_anynana_wg,2,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000030.1,1499809,1500327,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000010.1,6701190,6702179,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000006.1,10148375,10149598,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000002.1,18679339,18680043,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000019.1,17249100,17250095,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000003.1,5871996,5873381,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000015.1,1187252,1188028,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000001.1,24020291,24021751,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Oeneis_ivallda,JARPMR010000006.1,13750256,13753396,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Oeneis_ivallda,JARPMR010000005.1,18297517,18298173,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Oeneis_ivallda,JARPMR010000002.1,7243721,7244245,0,N,Bicyclus_anynana_wg,2,175,175\n",
      "Oeneis_ivallda,JARPMR010000030.1,1499809,1500327,0,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Oeneis_ivallda,JARPMR010000010.1,6701190,6702179,1,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Oeneis_ivallda,JARPMR010000006.1,10148375,10149598,0,N,Bicyclus_anynana_idh,1,408,408\n",
      "Oeneis_ivallda,JARPMR010000002.1,18679339,18680043,1,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Oeneis_ivallda,JARPMR010000019.1,17249100,17250095,1,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Oeneis_ivallda,JARPMR010000003.1,5871996,5873381,0,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Oeneis_ivallda,JARPMR010000015.1,1187252,1188028,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Oeneis_ivallda,JARPMR010000001.1,24020291,24021751,1,N,Bicyclus_anynana_cat,1,487,487\n",
      "Oeneis_ivallda,JARPMR010000006.1,13750256,13753396,0,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053183.1,8818510,8819163,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053182.1,16434376,16434900,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053184.1,15813827,15814345,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053190.1,14291871,14292860,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053185.1,13048460,13049683,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053182.1,8903608,8904312,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053196.1,5263527,5264522,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053181.1,17873394,17874779,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053194.1,712344,713120,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053208.1,10063224,10064684,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria,NC_053185.1,10705223,10708363,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pararge_aegeria,NC_053183.1,8818510,8819163,1,N,Bicyclus_anynana_rps5,1,218,218\n",
      "Pararge_aegeria,NC_053182.1,16434376,16434900,1,N,Bicyclus_anynana_wg,1,175,175\n",
      "Pararge_aegeria,NC_053184.1,15813827,15814345,1,N,Bicyclus_anynana_rps2,1,173,173\n",
      "Pararge_aegeria,NC_053190.1,14291871,14292860,0,N,Bicyclus_anynana_mdh,1,330,330\n",
      "Pararge_aegeria,NC_053185.1,13048460,13049683,1,N,Bicyclus_anynana_idh,1,408,408\n",
      "Pararge_aegeria,NC_053182.1,8903608,8904312,0,N,Bicyclus_anynana_ddc,1,235,235\n",
      "Pararge_aegeria,NC_053196.1,5263527,5264522,0,N,Bicyclus_anynana_gapdh,1,332,332\n",
      "Pararge_aegeria,NC_053181.1,17873394,17874779,1,N,Bicyclus_anynana_ef1,1,462,462\n",
      "Pararge_aegeria,NC_053194.1,712344,713120,0,N,Bicyclus_anynana_hcl,1,259,259\n",
      "Pararge_aegeria,NC_053208.1,10063224,10064684,0,N,Bicyclus_anynana_cat,1,487,487\n",
      "Pararge_aegeria,NC_053185.1,10705223,10708363,1,N,Bicyclus_anynana_cad,1,1047,1047\n",
      "\n"
     ]
    },
    {
     "ename": "NotADirectoryError",
     "evalue": "[Errno 20] Not a directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/3.Satyrine/1.Blast_result/desktop.ini/desktop.ini_blast_out.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNotADirectoryError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 29\u001b[0m\n\u001b[1;32m     27\u001b[0m query_name \u001b[38;5;241m=\u001b[39m query_name_list[i]\n\u001b[1;32m     28\u001b[0m Length_switch \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m0\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m---> 29\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mblast_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/1.Blast_result/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m_blast_out.txt\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m tblast_out:\n\u001b[1;32m     30\u001b[0m     lines_in_file \u001b[38;5;241m=\u001b[39m tblast_out\u001b[38;5;241m.\u001b[39mreadlines()\n\u001b[1;32m     31\u001b[0m \u001b[38;5;66;03m#print(lines_in_file)\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py:284\u001b[0m, in \u001b[0;36m_modified_open\u001b[0;34m(file, *args, **kwargs)\u001b[0m\n\u001b[1;32m    277\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m {\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m2\u001b[39m}:\n\u001b[1;32m    278\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m    279\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIPython won\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt let you open fd=\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m by default \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    280\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mas it is likely to crash IPython. If you know what you are doing, \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    281\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124myou can use builtins\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m open.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    282\u001b[0m     )\n\u001b[0;32m--> 284\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mio_open\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfile\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mNotADirectoryError\u001b[0m: [Errno 20] Not a directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/3.Satyrine/1.Blast_result/desktop.ini/desktop.ini_blast_out.txt'"
     ]
    }
   ],
   "source": [
    "#%reset -f\n",
    "blast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/3.Satyrine\"\n",
    "query_species = \"Bicyclus_anynana\"\n",
    "\n",
    "\n",
    "species_list = os.listdir(f\"{blast_location}/1.Blast_result\")\n",
    "\n",
    "query_file = SeqIO.parse(f\"{blast_location}/1.Query/query.txt\", 'fasta')\n",
    "query_name_list = []\n",
    "\n",
    "for records in query_file:\n",
    "    query_name_list.append(records.id)\n",
    "\n",
    "\n",
    "\n",
    "for species in species_list:\n",
    "    # print(species_name)\n",
    "    # species = species_name.split(\".\")[1]\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    \n",
    "#Run for each query in the query list\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        with open(f\"{blast_location}/1.Blast_result/{species}/{species}_blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "#                 print(query_species_split)\n",
    "                query_species = str(query_species_split[0]+\"_\"+query_species_split[1].rstrip())\n",
    "                print(query_species)\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "#                     if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "#                         error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "\n",
    "\n",
    "# #Check for Met or STOP at the start and end\n",
    "#         while True and (start != 0 or stop != 0):\n",
    "#             if (query_name == query_name_list[0] or query_name == query_name_list[-1]):\n",
    "#                 frame = 1\n",
    "#                 translated_sequence = (sequence_extractor(species, scaff, int(complement), start, stop,frame ))\n",
    "\n",
    "#                 if translated_sequence == old_trans:\n",
    "#                     break\n",
    "                    \n",
    "# #Check for Met at the beginning of first exon\n",
    "\n",
    "#                 if query_name == query_name_list[0] :\n",
    "#                     if translated_sequence[0] != \"M\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             start = int(start) - 3\n",
    "#                         if complement == \"1\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if \"*\" in translated_sequence:\n",
    "#                         error = \"Y\"\n",
    "#                         break\n",
    "#                     if translated_sequence[0] == \"M\":\n",
    "#                         break\n",
    "\n",
    "# #Check for stop at the end of last exon\n",
    "\n",
    "#                 if query_name == query_name_list[-1]:\n",
    "#                     if translated_sequence[-1] != \"*\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         if complement == \"1\":\n",
    "#                             start = int(start) - 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if translated_sequence[-1] == \"*\":\n",
    "\n",
    "#                         break\n",
    "#             else:\n",
    "#                 break\n",
    "        \n",
    "        \n",
    "#Add offset at the beginning and end of each exon\n",
    "#         if (start != 0 or stop != 0):\n",
    "            \n",
    "#             start_modifier = int(query_name.split(\"Frame\")[1][1])\n",
    "#             stop_modifier = int(query_name.split(\"rightoh\")[1][1])\n",
    "#         else:\n",
    "        start_modifier = 0\n",
    "        stop_modifier = 0            \n",
    "\n",
    "        seq_length = query_length\n",
    "        \n",
    "        \n",
    "#Get exon start and end by adding or removing codons\n",
    "        \n",
    "    #if not the beginning\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "#             if old_end != 0 and old_end > stop:\n",
    "\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "#             if old_end != 0 and old_end < stop:\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "            \n",
    "           \n",
    "        output_format = str(species.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        print(output_format)\n",
    "#         assert False\n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "    output_file = open(f\"{blast_location}/1.Blast_result/{species}/{species}_coordinates.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fa07f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465fa823",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
