{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "231732a3-2c48-45d2-8251-0572299fdc0b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      " wg \n",
      "\n",
      "\n",
      "103 107\n",
      "103 105\n",
      "\n",
      "\n",
      " cad \n",
      "\n",
      "\n",
      "105 105\n",
      "105 104\n",
      "\n",
      "\n",
      " cat \n",
      "\n",
      "\n",
      "104 104\n",
      "104 102\n",
      "\n",
      "\n",
      " ddc \n",
      "\n",
      "\n",
      "106 102\n",
      "106 102\n",
      "\n",
      "\n",
      " ef1 \n",
      "\n",
      "\n",
      "103 102\n",
      "103 101\n",
      "\n",
      "\n",
      " gapdh \n",
      "\n",
      "\n",
      "106 101\n",
      "106 101\n",
      "\n",
      "\n",
      " hcl \n",
      "\n",
      "\n",
      "106 101\n",
      "106 101\n",
      "\n",
      "\n",
      " idh \n",
      "\n",
      "\n",
      "105 101\n",
      "105 100\n",
      "\n",
      "\n",
      " mdh \n",
      "\n",
      "\n",
      "103 100\n",
      "103 98\n",
      "\n",
      "\n",
      " rps2 \n",
      "\n",
      "\n",
      "104 98\n",
      "104 97\n",
      "\n",
      "\n",
      " rps5 \n",
      "\n",
      "\n",
      "106 97\n",
      "106 97\n",
      "\n",
      "\n",
      " period \n",
      "\n",
      "\n",
      "90 97\n",
      "90 90\n",
      "\n",
      "\n",
      " timeless \n",
      "\n",
      "\n",
      "96 90\n",
      "96 85\n",
      "\n",
      "\n",
      " clock \n",
      "\n",
      "\n",
      "95 85\n",
      "95 81\n",
      "\n",
      "\n",
      " cycle \n",
      "\n",
      "\n",
      "91 81\n",
      "91 79\n",
      "\n",
      "\n",
      " cwo \n",
      "\n",
      "\n",
      "97 79\n",
      "97 78\n",
      "\n",
      "\n",
      " cry1a \n",
      "\n",
      "\n",
      "92 78\n",
      "92 77\n",
      "\n",
      "\n",
      " cry1b \n",
      "\n",
      "\n",
      "100 77\n",
      "100 77\n",
      "\n",
      "\n",
      " cry2 \n",
      "\n",
      "\n",
      "98 77\n",
      "98 75\n",
      "\n",
      "\n",
      " timeout \n",
      "\n",
      "\n",
      "93 75\n",
      "93 74\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import Phylo\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project\"\n",
    "\n",
    "list_of_gene_files = os.listdir(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/\")\n",
    "if \"desktop.ini\" in list_of_gene_files:\n",
    "    list_of_gene_files.remove(\"desktop.ini\")\n",
    "full_species_list = []\n",
    "for gene_file_name in list_of_gene_files:\n",
    "    gene_name = gene_file_name.split(\".\")[0]\n",
    "    # print(\"\\n\\n\",gene_name,\"\\n\\n\")\n",
    "\n",
    "    \n",
    "    fasta_file = SeqIO.parse(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/{gene_file_name}\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        # print(records.id)\n",
    "        if records.id not in full_species_list:\n",
    "            full_species_list.append(records.id)\n",
    "for gene_file_name in list_of_gene_files:\n",
    "    gene_specific_species_list = []\n",
    "    gene_name = gene_file_name.split(\".\")[0]\n",
    "    print(\"\\n\\n\",gene_name,\"\\n\\n\")\n",
    "    # if \"period\" in gene_file_name:\n",
    "    fasta_file = SeqIO.parse(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/{gene_file_name}\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        gene_specific_species_list.append(records.id)\n",
    "    print(len(gene_specific_species_list),len(full_species_list))\n",
    "    for species_name in full_species_list:\n",
    "        if species_name not in gene_specific_species_list:\n",
    "            full_species_list.remove(species_name)\n",
    "    print(len(gene_specific_species_list),len(full_species_list)) \n",
    "\n",
    "for gene_file_name in list_of_gene_files:\n",
    "    output = ''\n",
    "    fasta_file = SeqIO.parse(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/1.Gene_sequences/{gene_file_name}\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        if records.id in full_species_list:\n",
    "            output += f\">{records.id}\\n{records.seq}\\n\"\n",
    "    with open(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/2.Gene_sequence_pruned/{gene_file_name}\",'w') as out_file:\n",
    "        out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "838d3f24-b293-49cf-ad83-8b10ae1e3bed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Calephelis_nemesis', 'Aricia_agestis', 'Aricia_artaxerxes', 'Polyommatus_icarus', 'Polyommatus_iphigenia', 'Cyaniris_semiargus', 'Lysandra_bellargus', 'Lysandra_coridon', 'Plebejus_argus', 'Cyclargus_thomasi', 'Glaucopsyche_alexis', 'Helleia_helle', 'Lycaena_phlaeas', 'Calycopis_cecrops', 'Eumaeus_atala', 'Eueides_isabella', 'Heliconius_nattereri', 'Heliconius_charithonia', 'Heliconius_sara', 'Philaethria_dido', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Argynnis_bischoffii_washingtonia', 'Fabriciana_adippe', 'Brenthis_daphne', 'Brenthis_ino', 'Brenthis_hecate', 'Boloria_euphrosyne', 'Boloria_selene', 'Hypolimnas_misippus', 'Nymphalis_c', 'Nymphalis_polychloros', 'Vanessa_atalanta', 'Vanessa_tameamea', 'Nymphalis_io', 'Nymphalis_urticae', 'Euphydryas_editha', 'Melitaea_cinxia', 'Mellicta_athalia', 'Bicyclus_anynana', 'Lasiommata_megera', 'Erebia_aethiops', 'Erebia_ligea', 'Maniola_hyperantus', 'Maniola_jurtina', 'Melanargia_galathea', 'Hipparchia_semele', 'Oeneis_ivallda', 'Coenonympha_glycerion', 'Elymnias_hypermnestra', 'Danaus_chrysippus', 'Melinaea_marsaeus_rileyi', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Dircenna_loreta', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Leptidea_juvernica', 'Leptidea_reali', 'Anthocharis_cardamines', 'Aporia_crataegi', 'Leptophobia_aripa', 'Pieris_brassicae', 'Pieris_melete', 'Pieris_mannii', 'Colias_behrii', 'Colias_croceus', 'Colias_eurytheme', 'Zerene_cesonia', 'Phoebis_sennae', 'Eurema_hecabe', 'Baronia_brevicornis', 'Parnassius_glacialis', 'Sericinus_montela', 'Luehdorfia_chinensis', 'Ornithoptera_alexandrae', 'Ornithoptera_priamus', 'Troides_aeacus', 'Troides_oblongomaculatus', 'Papilio_dardanus_tibullus', 'Papilio_polytes', 'Papilio_protenor', 'Papilio_memnon', 'Papilio_demoleus', 'Papilio_elwesi', 'Papilio_glaucus', 'Papilio_xuthus', 'Papilio_machaon', 'Colias_nastes', 'Coenonympha_glycerion2']\n"
     ]
    }
   ],
   "source": [
    "print(full_species_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "6e94fb9e-9cee-48c3-9ad5-e728ab76d117",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wg\n",
      "cad\n",
      "cat\n",
      "ddc\n",
      "ef1\n",
      "gapdh\n",
      "hcl\n",
      "idh\n",
      "mdh\n",
      "rps2\n",
      "rps5\n",
      "period\n",
      "timeless\n",
      "clock\n",
      "cycle\n",
      "cwo\n",
      "cry1a\n",
      "cry1b\n",
      "cry2\n",
      "timeout\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import Phylo\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project\"\n",
    "\n",
    "list_of_gene_files = os.listdir(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/2.Gene_sequence_pruned/\")\n",
    "if \"desktop.ini\" in list_of_gene_files:\n",
    "    list_of_gene_files.remove(\"desktop.ini\")\n",
    "\n",
    "for gene_file_name in list_of_gene_files:\n",
    "    gene_name = gene_file_name.split(\".\")[0]\n",
    "    print(gene_name)\n",
    "\n",
    "    species_list = []\n",
    "    fasta_file = SeqIO.parse(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/2.Gene_sequence_pruned/{gene_file_name}\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "    \n",
    "        species_list.append(records.id)\n",
    "        \n",
    "    \n",
    "    tree = Phylo.read(f\"{location}/20.RERconverge/3.HYPHY branch length optimization/combined_file.fas.treefile_no_branch_length_fig.nwk\", 'newick')\n",
    "    tree_species_list = [term.name for term in tree.get_terminals()]\n",
    "    for species in tree_species_list:\n",
    "        if species not in species_list:\n",
    "            # print(species)\n",
    "            tree.prune(species)\n",
    "    Phylo.write(tree, f\"{location}/20.RERconverge/3.HYPHY branch length optimization/2.For_branch_length_optimization_new/{gene_name}_pruned_tree.nwk\", \"newick\")\n",
    "    # subprocess.run(f\"cd '{location}/20.RERconverge/3.HYPHY branch length optimization'\\niqtree2 -s ./2.Gene_sequence_pruned/{gene_name}.fas.aligned.fas.best.fas -g ./2.For_branch_length_optimization_new/{gene_name}_pruned_tree.nwk -m MFP -pre ./2.For_branch_length_optimization_new/{gene_name}_recalibrated_tree\", shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "efe337bf-1695-472c-a76d-1fb5238c4a89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.3.5 for Linux x86 64-bit built Jul 22 2024\n",
      "Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor, Heiko Schmidt,\n",
      "Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong\n",
      "\n",
      "Host:    DESKTOP-CM7UJAK (AVX2, FMA3, 7 GB RAM)\n",
      "Command: iqtree2 -s ./1.Gene_sequences/wg.fas.aligned.fas.best.fas -g ./2.For_branch_length_optimization/wg_pruned_tree.nwk -m MFP -pre ./2.For_branch_length_optimization/wg_recalibrated_tree\n",
      "Seed:    481265 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Wed Oct  2 14:39:25 2024\n",
      "Kernel:  AVX+FMA - 1 threads (4 CPU cores detected)\n",
      "\n",
      "HINT: Use -nt option to specify number of threads because your CPU has 4 cores!\n",
      "HINT: -nt AUTO will automatically determine the best number of threads to use.\n",
      "\n",
      "Reading alignment file ./1.Gene_sequences/wg.fas.aligned.fas.best.fas ... Fasta format detected\n",
      "Reading fasta file: done in 0.0084418 secs\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 103 sequences with 534 columns, 284 distinct patterns\n",
      "267 parsimony-informative, 11 singleton sites, 256 constant sites\n",
      "                                  Gap/Ambiguity  Composition  p-value\n",
      "Analyzing sequences: done in 0.0001043 secs\n",
      "   1  Calephelis_nemesis                  1.69%    failed      0.60%\n",
      "   2  Calephelis_virginiensis             1.69%    failed      0.13%\n",
      "   3  Aricia_agestis                      1.69%    failed      0.00%\n",
      "   4  Aricia_artaxerxes                   1.69%    failed      0.00%\n",
      "   5  Polyommatus_icarus                  1.69%    failed      0.00%\n",
      "   6  Polyommatus_iphigenia               1.69%    failed      0.00%\n",
      "   7  Cyaniris_semiargus                  1.69%    failed      0.00%\n",
      "   8  Lysandra_bellargus                  1.69%    failed      0.03%\n",
      "   9  Lysandra_coridon                    1.69%    failed      0.00%\n",
      "  10  Plebejus_argus                      1.69%    failed      0.00%\n",
      "  11  Cyclargus_thomasi                   1.69%    failed      0.16%\n",
      "  12  Celastrina_argiolus                 1.69%    failed      0.13%\n",
      "  13  Glaucopsyche_alexis                 1.69%    passed     17.87%\n",
      "  14  Phengaris_arion                     1.69%    passed     45.99%\n",
      "  15  Helleia_helle                       1.69%    passed     65.73%\n",
      "  16  Lycaena_phlaeas                     1.69%    passed     80.15%\n",
      "  17  Calycopis_cecrops                   1.69%    passed     32.90%\n",
      "  18  Eumaeus_atala                       1.69%    passed     62.01%\n",
      "  19  Eueides_isabella                    1.69%    passed     51.02%\n",
      "  20  Heliconius_nattereri                1.69%    passed     56.11%\n",
      "  21  Heliconius_charithonia              1.69%    passed     81.01%\n",
      "  22  Heliconius_sara                     1.69%    passed     64.85%\n",
      "  23  Philaethria_dido                    1.69%    passed     56.11%\n",
      "  24  Dryadula_phaetusa                   1.69%    passed     54.33%\n",
      "  25  Dryas_iulia_moderata                1.69%    passed     65.67%\n",
      "  26  Argynnis_bischoffii_washingtonia    1.69%    passed     14.29%\n",
      "  27  Fabriciana_adippe                   1.69%    passed     31.86%\n",
      "  28  Brenthis_daphne                     1.69%    passed     50.34%\n",
      "  29  Brenthis_ino                        1.69%    passed     60.44%\n",
      "  30  Brenthis_hecate                     1.69%    passed     55.78%\n",
      "  31  Boloria_euphrosyne                  1.69%    passed     34.89%\n",
      "  32  Boloria_selene                      1.69%    passed     16.92%\n",
      "  33  Limenitis_camilla                   1.69%    passed     90.32%\n",
      "  34  Hestina_assimilis                   1.69%    passed     89.68%\n",
      "  35  Hypolimnas_misippus                 1.69%    passed     96.12%\n",
      "  36  Nymphalis_c                         1.69%    passed     83.57%\n",
      "  37  Nymphalis_polychloros               1.69%    passed     45.78%\n",
      "  38  Vanessa_atalanta                    1.69%    passed     65.48%\n",
      "  39  Vanessa_tameamea                    1.69%    passed     70.96%\n",
      "  40  Vanessa_cardui                      1.69%    passed     98.32%\n",
      "  41  Nymphalis_io                        2.25%    passed     15.62%\n",
      "  42  Nymphalis_urticae                   2.25%    passed     45.07%\n",
      "  43  Euphydryas_editha                   1.12%    passed      8.00%\n",
      "  44  Melitaea_cinxia                     1.12%    passed     17.02%\n",
      "  45  Mellicta_athalia                    1.12%    passed     14.09%\n",
      "  46  Bicyclus_anynana                    1.69%    passed     10.74%\n",
      "  47  Pararge_aegeria                     1.69%    passed     94.30%\n",
      "  48  Lasiommata_megera                   1.69%    passed     23.01%\n",
      "  49  Erebia_aethiops                     1.69%    passed     72.31%\n",
      "  50  Erebia_ligea                        1.69%    passed     43.15%\n",
      "  51  Maniola_hyperantus                  1.69%    passed     53.10%\n",
      "  52  Maniola_jurtina                     1.69%    failed      1.00%\n",
      "  53  Melanargia_galathea                 1.69%    failed      0.45%\n",
      "  54  Hipparchia_semele                   1.69%    passed     24.87%\n",
      "  55  Oeneis_ivallda                      1.69%    passed     37.74%\n",
      "  56  Coenonympha_glycerion               1.69%    failed      1.94%\n",
      "  57  Elymnias_hypermnestra               1.69%    passed     32.29%\n",
      "  58  Danaus_chrysippus                   1.69%    passed     14.85%\n",
      "  59  Melinaea_marsaeus_rileyi            1.69%    passed     56.29%\n",
      "  60  Melinaea_menophilus                 1.69%    passed     46.56%\n",
      "  61  Mechanitis_mazaeus                  1.69%    passed     38.19%\n",
      "  62  Mechanitis_messenoides              1.69%    passed     38.67%\n",
      "  63  Dircenna_loreta                     1.69%    passed     58.72%\n",
      "  64  Napeogenes_inachia                  1.69%    passed     56.21%\n",
      "  65  Napeogenes_sylphis                  1.69%    passed     47.43%\n",
      "  66  Leptidea_juvernica                  1.69%    failed      1.61%\n",
      "  67  Leptidea_reali                      1.69%    failed      1.53%\n",
      "  68  Leptidea_sinapis                    1.69%    failed      1.48%\n",
      "  69  Anthocharis_cardamines              1.69%    passed     97.60%\n",
      "  70  Aporia_crataegi                     1.69%    passed     29.94%\n",
      "  71  Leptophobia_aripa                   1.69%    passed     83.79%\n",
      "  72  Pieris_brassicae                    1.69%    passed     83.23%\n",
      "  73  Pieris_melete                       1.69%    passed     98.09%\n",
      "  74  Pieris_napi                         1.69%    passed     97.49%\n",
      "  75  Pieris_mannii                       1.69%    passed     87.91%\n",
      "  76  Pieris_rapae                        1.69%    passed     96.93%\n",
      "  77  Colias_behrii                       1.69%    passed     73.05%\n",
      "  78  Colias_croceus                      1.69%    passed     72.21%\n",
      "  79  Colias_eurytheme                    1.69%    passed     63.29%\n",
      "  80  Zerene_cesonia                      1.69%    passed     94.62%\n",
      "  81  Phoebis_sennae                      1.69%    passed     68.18%\n",
      "  82  Eurema_hecabe                       1.69%    passed     99.52%\n",
      "  83  Baronia_brevicornis                 1.69%    passed     65.67%\n",
      "  84  Battus_philenor                     1.69%    passed     63.01%\n",
      "  85  Parnassius_glacialis                1.69%    passed     37.48%\n",
      "  86  Sericinus_montela                   1.69%    failed      1.35%\n",
      "  87  Luehdorfia_chinensis                1.69%    passed     57.73%\n",
      "  88  Teinopalpus_imperialis              1.69%    passed     75.91%\n",
      "  89  Ornithoptera_alexandrae             1.69%    passed     70.96%\n",
      "  90  Ornithoptera_priamus                1.69%    passed     80.12%\n",
      "  91  Troides_aeacus                      1.69%    passed     93.72%\n",
      "  92  Troides_oblongomaculatus            1.69%    passed     93.72%\n",
      "  93  Papilio_dardanus_tibullus           1.69%    passed     90.56%\n",
      "  94  Papilio_polytes                     1.69%    passed     83.71%\n",
      "  95  Papilio_protenor                    1.69%    passed     90.42%\n",
      "  96  Papilio_memnon                      1.69%    passed     68.35%\n",
      "  97  Papilio_demoleus                    1.69%    passed     40.05%\n",
      "  98  Papilio_elwesi                      1.69%    passed     93.49%\n",
      "  99  Papilio_glaucus                     1.69%    passed     89.25%\n",
      " 100  Papilio_xuthus                      1.69%    passed     55.95%\n",
      " 101  Papilio_machaon                     1.69%    passed     34.83%\n",
      " 102  Papilio_gigon                       1.69%    passed     98.13%\n",
      " 103  Iphiclides_podalirius               1.69%    failed      0.03%\n",
      "****  TOTAL                               1.68%  20 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "Reading constraint tree ./2.For_branch_length_optimization/wg_pruned_tree.nwk...\n",
      "WARNING: Rooted constraint tree will be treated as unrooted tree\n",
      "\n",
      "WARNING: Rooted constraint tree will be treated as unrooted tree\n",
      "Creating fast initial parsimony tree by random order stepwise addition...\n",
      "WARNING: Constraint tree has all taxa and is bifurcating, which strictly enforces final tree!\n",
      "0.045 seconds, parsimony score: 0 (based on 278 sites)\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -12175.137\n",
      "2. Current log-likelihood: -12173.611\n",
      "Optimal log-likelihood: -12172.710\n",
      "Rate parameters:  A-C: 1.23956  A-G: 4.32561  A-T: 1.41838  C-G: 0.59750  C-T: 7.34141  G-T: 1.00000\n",
      "Base frequencies:  A: 0.246  C: 0.257  G: 0.292  T: 0.205\n",
      "Proportion of invariable sites: 0.432\n",
      "Gamma shape alpha: 0.834\n",
      "Parameters optimization took 2 rounds (0.358 sec)\n",
      "Time for fast ML tree search: 1.019 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 9 MB RAM!\n",
      "ModelFinder will test up to 484 DNA models (sample size: 534 epsilon: 0.100) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "  1  GTR+F         14372.698    211 29167.396    29445.234    30070.559\n",
      "  2  GTR+F+I       12799.049    212 26022.099    26303.445    26929.543\n",
      "  3  GTR+F+G4      12251.412    212 24926.825    25208.171    25834.269\n",
      "  4  GTR+F+I+G4    12172.491    213 24770.981    25055.869    25682.705\n",
      "  5  GTR+F+R2      12405.375    213 25236.750    25521.638    26148.474\n",
      "  6  GTR+F+R3      12220.852    215 24871.705    25163.780    25791.990\n",
      "  7  GTR+F+R4      12171.752    217 24777.505    25076.910    25706.351\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "KeyboardInterrupt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import Phylo\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization/\"\n",
    "\n",
    "list_of_gene_files = os.listdir(f\"{location}/1.Gene_sequences\")\n",
    "for gene_file_name in list_of_gene_files:\n",
    "    gene_name = gene_file_name.split(\".\")[0]\n",
    "    \n",
    "    subprocess.run(f\"cd '{location}'\\niqtree2 -s ./1.Gene_sequences/{gene_file_name} -g ./2.For_branch_length_optimization/{gene_name}_pruned_tree.nwk -m MFP -pre ./2.For_branch_length_optimization/{gene_name}_recalibrated_tree\", shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "a3067c7b-e72a-418f-967a-5f4036c861e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import Phylo\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization\"\n",
    "output = \"\"\n",
    "list_of_files = os.listdir(f\"{location}/2.For_branch_length_optimization_new\")\n",
    "\n",
    "for file_name in list_of_files:\n",
    "    if file_name.endswith(\"recalibrated_tree.treefile\"):\n",
    "        # tree = Phylo.read(f\"{location}/2.For_branch_length_optimization_new/{file_name}\", 'newick')\n",
    "        # tree_species_list2 = [term.name for term in tree.get_terminals()]\n",
    "        # # print(file_name, len(tree_species_list2))\n",
    "        # for species in tree_species_list2:\n",
    "        #     if \"c-album\" in species:\n",
    "        #         print(file_name, species)\n",
    "        with open(f\"{location}/2.For_branch_length_optimization_new/{file_name}\", 'r') as tree_file:\n",
    "            output += f\">{file_name.split('_')[0]}\\t{tree_file.readlines()[0]}\"\n",
    "with open(f\"{location}/combined_tree.nwk\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "# print(output)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "44a767e1-ddc0-44bc-bd24-030c25c2b338",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calephelis_nemesis\n",
      "Calephelis_virginiensis\n",
      "Aricia_agestis\n",
      "Aricia_artaxerxes\n",
      "Polyommatus_icarus\n",
      "Polyommatus_iphigenia\n",
      "Lysandra_bellargus\n",
      "Lysandra_coridon\n",
      "Cyaniris_semiargus\n",
      "Plebejus_argus\n",
      "Cyclargus_thomasi\n",
      "Celastrina_argiolus\n",
      "Glaucopsyche_alexis\n",
      "Phengaris_arion\n",
      "Helleia_helle\n",
      "Lycaena_phlaeas\n",
      "Calycopis_cecrops\n",
      "Eumaeus_atala\n",
      "Eueides_isabella\n",
      "Heliconius_nattereri\n",
      "Heliconius_charithonia\n",
      "Heliconius_sara\n",
      "Philaethria_dido\n",
      "Dryadula_phaetusa\n",
      "Dryas_iulia_moderata\n",
      "Argynnis_bischoffii_washingtonia\n",
      "Fabriciana_adippe\n",
      "Boloria_euphrosyne\n",
      "Boloria_selene\n",
      "Brenthis_daphne\n",
      "Brenthis_ino\n",
      "Brenthis_hecate\n",
      "Limenitis_camilla\n",
      "Hestina_assimilis\n",
      "Hypolimnas_misippus\n",
      "Euphydryas_editha\n",
      "Melitaea_cinxia\n",
      "Mellicta_athalia\n",
      "Nymphalis_c\n",
      "Nymphalis_polychloros\n",
      "Nymphalis_io\n",
      "Nymphalis_urticae\n",
      "Vanessa_atalanta\n",
      "Vanessa_tameamea\n",
      "Vanessa_cardui\n",
      "Bicyclus_anynana\n",
      "Pararge_aegeria\n",
      "Lasiommata_megera\n",
      "Erebia_aethiops\n",
      "Erebia_ligea\n",
      "Maniola_hyperantus\n",
      "Maniola_jurtina\n",
      "Melanargia_galathea\n",
      "Hipparchia_semele\n",
      "Oeneis_ivallda\n",
      "Coenonympha_glycerion\n",
      "Elymnias_hypermnestra\n",
      "Danaus_chrysippus\n",
      "Melinaea_marsaeus_rileyi\n",
      "Melinaea_menophilus\n",
      "Mechanitis_mazaeus\n",
      "Mechanitis_messenoides\n",
      "Dircenna_loreta\n",
      "Napeogenes_inachia\n",
      "Napeogenes_sylphis\n",
      "Leptidea_juvernica\n",
      "Leptidea_reali\n",
      "Leptidea_sinapis\n",
      "Anthocharis_cardamines\n",
      "Aporia_crataegi\n",
      "Leptophobia_aripa\n",
      "Pieris_brassicae\n",
      "Pieris_melete\n",
      "Pieris_napi\n",
      "Pieris_mannii\n",
      "Pieris_rapae\n",
      "Colias_behrii\n",
      "Colias_eurytheme\n",
      "Colias_croceus\n",
      "Zerene_cesonia\n",
      "Phoebis_sennae\n",
      "Eurema_hecabe\n",
      "Baronia_brevicornis\n",
      "Battus_philenor\n",
      "Ornithoptera_alexandrae\n",
      "Ornithoptera_priamus\n",
      "Troides_aeacus\n",
      "Troides_oblongomaculatus\n",
      "Papilio_dardanus_tibullus\n",
      "Papilio_demoleus\n",
      "Papilio_polytes\n",
      "Papilio_protenor\n",
      "Papilio_memnon\n",
      "Papilio_xuthus\n",
      "Papilio_machaon\n",
      "Papilio_gigon\n",
      "Papilio_elwesi\n",
      "Papilio_glaucus\n",
      "Parnassius_glacialis\n",
      "Sericinus_montela\n",
      "Luehdorfia_chinensis\n",
      "Teinopalpus_imperialis\n",
      "Iphiclides_podalirius\n",
      "Charaxes_varanes\n",
      "Colias_nastes\n",
      "Danaus_plexippus\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import Phylo\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/3.HYPHY branch length optimization\"\n",
    "output = \"\"\n",
    "list_of_files = os.listdir(f\"{location}/2.For_branch_length_optimization\")\n",
    "species_list = []\n",
    "for file_name in list_of_files:\n",
    "    if file_name.endswith(\"recalibrated_tree.treefile\"):\n",
    "        tree = Phylo.read(f\"{location}/2.For_branch_length_optimization/{file_name}\", 'newick')\n",
    "        tree_species_list2 = [term.name for term in tree.get_terminals()]\n",
    "        for species in tree_species_list2:\n",
    "            if species not in species_list:\n",
    "                if species == \"Coenonympha_glycerion2\":\n",
    "                    print(file_name)\n",
    "                species_list.append(species)\n",
    "\n",
    "for species in species_list:\n",
    "    print(species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "4798611f-da5c-4e41-a701-fc550a6b978c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import Phylo\n",
    "\n",
    "tree = Phylo.read(f\"{location}/2.For_branch_length_optimization/cycle_recalibrated_tree.treefile\", 'newick')\n",
    "tree.prune(\"Coenonympha_glycerion2\")\n",
    "Phylo.write(tree, f\"{location}/2.For_branch_length_optimization/cycle_recalibrated_tree.treefile_2\", \"newick\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aad8f961-2757-40c8-9059-bb42931df6cd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
