{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9976d94f-7c44-49b6-9db0-8995edc0bb76",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lycaenidae_period.fas\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "family, gene = \"Lycaenidae\", \"period\"\n",
    "location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/12.Family_specific_tests/{family}\"\n",
    "file_name = f'{family}_{gene}.fas'\n",
    "\n",
    "subprocess.run(f'cd {location}\\nprank -d={file_name} -o={file_name}.aligned.fas -codon', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "\n",
    "\n",
    "# location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/11.Alignment\"\n",
    "\n",
    "output1 = \"\"\n",
    "output2 = ''\n",
    "\n",
    "untranslated = SeqIO.parse(f\"{location}/{file_name}.aligned.fas.best.fas\", 'fasta')\n",
    "\n",
    "for records in untranslated:\n",
    "    output1 += f\">{records.id}\\n{records.seq.replace('-','')}\\n\\n\"\n",
    "    output2 += f\">{records.id}\\n{records.seq.translate()}\\n\\n\"\n",
    "print(output1)\n",
    "with open(f\"{location}/{file_name}.aligned.fas.ungapped.fas\", 'w') as out_file:\n",
    "    out_file.write(output1)\n",
    "\n",
    "with open(f\"{location}/{file_name}.aligned.fas.translated.fas\", 'w') as out_file:\n",
    "    out_file.write(output2)\n",
    "\n",
    "subprocess.run(f'cd {location}\\ntrimal -colnumbering -htmlout trimal_output.html -in {file_name}.aligned.fas.translated.fas -out {file_name}.aligned.fas.trimal.fas -automated1 > columns_kept.txt', shell = True, stderr = subprocess.DEVNULL)\n",
    "subprocess.run(f'cd {location}\\ntrimal -backtrans {file_name}.aligned.fas.ungapped.fas -htmlout trimal_output.html -in {file_name}.aligned.fas.translated.fas -out {file_name}.aligned.fas.trimal.fas -automated1', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "subprocess.run(f'cd {location}\\niqtree -s {file_name}.aligned.fas.trimal.fas -m MFP -t RANDOM -bb 1000 -alrt 1000 -bnni -nt AUTO', shell = True, stderr = subprocess.DEVNULL)\n",
    "subprocess.run(f'cd {location}\\niqtree -s {file_name}.aligned.fas.best.fas -m MFP -t RANDOM -bb 1000 -alrt 1000 -bnni -nt AUTO', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "subprocess.run(f'cd {location}\\nhyphy busted --alignment {file_name}.aligned.fas.trimal.fas --tree {file_name}.aligned.fas.trimal.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "subprocess.run(f'cd {location}\\nhyphy busted --alignment {file_name}.aligned.fas.best.fas --tree{file_name}.aligned.fas.best.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37fb6277-6209-4c59-a897-3abc20b1a7cb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
