{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ee2d2aae-e268-4258-bc01-a00d928ca22a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Erebia_aethiops\n",
      "Exon_1 Match\n",
      "[['Maniola_jurtina', ['7624959', '7625041']], ['Pararge_aegeria', ['7624959', '7625041']], ['Bicyclus_anynana', ['7624959', '7625041']]]\n",
      "Exon_2 Match\n",
      "[['Maniola_jurtina', ['7627065', '7627148']], ['Pararge_aegeria', ['7627065', '7627148']], ['Bicyclus_anynana', ['7627065', '7627148']]]\n",
      "Exon_3 Match\n",
      "[['Maniola_jurtina', ['7627773', '7627945']], ['Pararge_aegeria', ['7627773', '7627945']], ['Bicyclus_anynana', ['7627773', '7627945']]]\n",
      "Exon_4 mismatch\n",
      "[['Maniola_jurtina', ['7628823', '7628917']], ['Pararge_aegeria', ['7628823', '7628917']], ['Bicyclus_anynana', ['7628823', '7628887']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_4 75_frame1 0.646 \n",
      "CQQQTQSPASAQTEKGPELMEVSDNSSGKEQ\n",
      "Pararge_aegeria Exon_4 75_frame1 1.219 \n",
      "CQQQTQSPASAQTEKGPELMEVSDNSSGKEQ\n",
      "Bicyclus_anynana Exon_4 75_frame1 1.115 \n",
      "CQQQTQSPASAQTEKGPELME\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? 1\n",
      "Check Alignment?? n\n",
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_5 mismatch\n",
      "[['Maniola_jurtina', ['7630073', '7630147']], ['Pararge_aegeria', ['102', '27']], ['Bicyclus_anynana', ['7630073', '7630147']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? n\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_6 Match\n",
      "[['Maniola_jurtina', ['7630393', '7630524']], ['Pararge_aegeria', ['7630393', '7630524']], ['Bicyclus_anynana', ['7630393', '7630524']]]\n",
      "Exon_7 Match\n",
      "[['Maniola_jurtina', ['7631378', '7631453']], ['Pararge_aegeria', ['7631378', '7631453']], ['Bicyclus_anynana', ['7631378', '7631453']]]\n",
      "Exon_8 Match\n",
      "[['Maniola_jurtina', ['7631881', '7632084']], ['Pararge_aegeria', ['7631881', '7632084']], ['Bicyclus_anynana', ['7631881', '7632084']]]\n",
      "Exon_9 Match\n",
      "[['Maniola_jurtina', ['7632564', '7632792']], ['Pararge_aegeria', ['7632564', '7632792']], ['Bicyclus_anynana', ['7632564', '7632792']]]\n",
      "Exon_10 Match\n",
      "[['Maniola_jurtina', ['7633837', '7633948']], ['Pararge_aegeria', ['7633837', '7633948']], ['Bicyclus_anynana', ['7633837', '7633948']]]\n",
      "Exon_11 Match\n",
      "[['Maniola_jurtina', ['7634044', '7634163']], ['Pararge_aegeria', ['7634044', '7634163']], ['Bicyclus_anynana', ['7634044', '7634163']]]\n",
      "Exon_12 Match\n",
      "[['Maniola_jurtina', ['7634969', '7635118']], ['Pararge_aegeria', ['7634969', '7635118']], ['Bicyclus_anynana', ['7634969', '7635118']]]\n",
      "Exon_13 mismatch\n",
      "[['Maniola_jurtina', ['7635881', '7635999']], ['Pararge_aegeria', ['7635884', '7635999']], ['Bicyclus_anynana', ['7635881', '7635999']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_13 44_frame2 0.081 \n",
      "ERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQ\n",
      "Pararge_aegeria Exon_13 44_frame2 0.53 \n",
      "RDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQ\n",
      "Bicyclus_anynana Exon_13 44_frame2 0.332 \n",
      "ERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQ\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt -----------ERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYNENLQ-\n",
      "set44_frame2    HFNSVIFSITQERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQR\n",
      "                           **********************************::*** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege -------------DSVMLGGVSPHHDHNDSMTNKDTPLTYKQLNYNDNLQ-\n",
      "set44_frame2    HFNSVIFSITQERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQR\n",
      "                             *******:*****:*** *..:*.*:*:****::*** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn -----------ERDSVMLGGISPHHDNNDSNTSTETPLTYNQLNYNENLQ-\n",
      "set44_frame2    HFNSVIFSITQERDSVMLGGISPHHDYNDSHTSSETTLSYNQLNYKQNLQR\n",
      "                           *************** ***:**:**.*:******::*** \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_14 Match\n",
      "[['Maniola_jurtina', ['7636136', '7636240']], ['Pararge_aegeria', ['7636136', '7636240']], ['Bicyclus_anynana', ['7636136', '7636240']]]\n",
      "Exon_15 mismatch\n",
      "[['Maniola_jurtina', ['7636666', '7636818']], ['Pararge_aegeria', ['7636666', '7636803']], ['Bicyclus_anynana', ['7636666', '7636818']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_15 15_frame2 0.84 \n",
      "KCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLTESLLK\n",
      "Pararge_aegeria Exon_15 15_frame2 1.331 \n",
      "KCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLT\n",
      "Bicyclus_anynana Exon_15 15_frame2 1.005 \n",
      "KCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLTESLLK\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt --------RCI---LSNPGELDERASTSDSNAAAGASTSRAGTSIEYPALRLTESLLN--\n",
      "set15_frame2    YLSLCTFSKCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLTESLLKKS\n",
      "                        :*:    ..*.*:*  .*****.**..***.***** :**::*******:  \n",
      "\n",
      "28.Maniola_jurt -\n",
      "set15_frame2    V\n",
      "                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege K------------SASTVGEIEDMATSSDSSVVAPTTIS---SSCGYPTFRLTESLLN--\n",
      "set15_frame2    YLSLCTFSKCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLTESLLKKS\n",
      "                             ::.: .*::  .::****....:: .   :*  ***.*******:  \n",
      "\n",
      "33.Pararge_aege -\n",
      "set15_frame2    V\n",
      "                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn ------FSKNFLSMVNSSGEFDEMASTSDSSTAAVASSNNACSSNGFQALRLTESLLN--\n",
      "set15_frame2    YLSLCTFSKCVSLTSNSPDEFDVSGSTSDSSAAVAASTNRAGTSNKYPTVRLTESLLKKS\n",
      "                      *** .    **..***  .******:*..**:*.* :** : ::*******:  \n",
      "\n",
      "5.Bicyclus_anyn -\n",
      "set15_frame2    V\n",
      "                 \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_16 Match\n",
      "[['Maniola_jurtina', ['7637174', '7637342']], ['Pararge_aegeria', ['7637174', '7637342']], ['Bicyclus_anynana', ['7637174', '7637342']]]\n",
      "Exon_17 Match\n",
      "[['Maniola_jurtina', ['7638011', '7638152']], ['Pararge_aegeria', ['7638011', '7638152']], ['Bicyclus_anynana', ['7638011', '7638152']]]\n",
      "Exon_18 Match\n",
      "[['Maniola_jurtina', ['7638571', '7638675']], ['Pararge_aegeria', ['7638571', '7638675']], ['Bicyclus_anynana', ['7638571', '7638675']]]\n",
      "Exon_19 mismatch\n",
      "[['Maniola_jurtina', ['7639269', '7639370']], ['Pararge_aegeria', ['7639269', '7639334']], ['Bicyclus_anynana', ['7639269', '7639370']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_19 32_frame0 0.577 \n",
      "VLPVYYTPMTNQQMPPRFNTNSAMPHAHYYRYP\n",
      "Pararge_aegeria Exon_19 32_frame0 1.295 \n",
      "VLPVYYTPMTNQQMPPRFNTN\n",
      "Bicyclus_anynana Exon_19 32_frame0 1.286 \n",
      "VLPVYYTPMTNQQMPPRFNTNSAMPHAHYYRYP\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt -----ALPMYYTPTSTQQMPSRFNPNTNLPHAHYHRYQ----\n",
      "set32_frame0    YCFSGVLPVYYTPMTNQQMPPRFNTNSAMPHAHYYRYPSESL\n",
      "                     .**:**** :.****.***.*: :*****:**     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege -----LLPVDYAPPMPYQVPPR----SDLAAPHYDRHS----\n",
      "set32_frame0    YCFSGVLPVYYTPMTNQQMPPRFNTNSAMPHAHYYRYPSESL\n",
      "                     :*** *:*    *:***    * :. .** *:.    \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn -----PVPMYYIPAAPYQMAPKSEAGPSASQAQYQRHS----\n",
      "set32_frame0    YCFSGVLPVYYTPMTNQQMPPRFNTNSAMPHAHYYRYPSESL\n",
      "                      :*:** * :  **.*: ::..: .:*:* *:.    \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_20 mismatch\n",
      "[['Maniola_jurtina', ['7639930', '7640106']], ['Pararge_aegeria', ['7639930', '7640046']], ['Bicyclus_anynana', ['13907473', '13907595']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_20 26_frame0 0.608 \n",
      "VNVQTLNPHIQYMQQGQMNVPDVVFQMNSVFSTNGEPHSVNPNIVPHNQYQQQHQNNQ\n",
      "Pararge_aegeria Exon_20 26_frame0 1.66 \n",
      "VNVQTLNPHIQYMQQGQMNVPDVVFQMNSVFSTNGEPH\n",
      "Bicyclus_anynana Exon_20 44_frame1 1.678 \n",
      "Issue with query\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt ---------VSTVNPHMQYMQQGQINESDVVFQMNSVFATSGEANRVHPNNIPPTHYQQI\n",
      "set26_frame0    HYSTISGVNVQTLNPHIQYMQQGQMNVPDVVFQMNSVFSTNGEPHSVNPNIVPHNQYQQQ\n",
      "                         *.*:***:*******:* .**********:*.**.: *:** :* .:*** \n",
      "\n",
      "28.Maniola_jurt HQSNQ---\n",
      "set26_frame0    HQNNQCKF\n",
      "                **.**   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege ---------MAPVNPQMQYHHMPQ--------QMNPHCSTGAV-----PNRYMHNNMQPA\n",
      "set26_frame0    HYSTISGVNVQTLNPHIQYMQQGQMNVPDVVFQMNSVFSTNGEPHSVNPNIVPHNQYQQQ\n",
      "                         : .:**::** :  *        ***.  **..      **   **: *  \n",
      "\n",
      "33.Pararge_aege YPVHVPNPQ---\n",
      "set26_frame0    H----QNNQCKF\n",
      "                :     * *   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn -MNG--------------------------------FSI---------------------\n",
      "set44_frame1    GFNAGVSIQRTAPCPSLKIVFMVDCIIREVYLIKSFFSVQCMMLGQTVYGPPFMYAPIDP\n",
      "                 :*.                                **:                     \n",
      "\n",
      "5.Bicyclus_anyn -------PYMGHQQNMN--PAGRSSRTETANANMNMQPNVYSP\n",
      "set44_frame1    QMSYALQPSFIPQQTLNTPPLGFSSSTYEEVSNNLLQ----YL\n",
      "                       * :  **.:*  * * ** *    :*  :*      \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_21 Match\n",
      "[['Maniola_jurtina', ['7640747', '7640901']], ['Pararge_aegeria', ['7640747', '7640901']], ['Bicyclus_anynana', ['7640747', '7640901']]]\n",
      "Exon_22 mismatch\n",
      "[['Maniola_jurtina', ['7642098', '7642244']], ['Bicyclus_anynana', ['7639194', '7639238']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_22 73_frame0 1.776 \n",
      "LTLAYELLQNKHKSSETNVSFLRSPDAVATNANARKIPRRKMDPPWMEQ\n",
      "Pararge_aegeria Exon_22 62_frame1 1.075 \n",
      "STDAMTGESLKRRLDTSDSNNDKTDGESSYSSFYSSFFKTDSGSGSDTRNQNNKGHVKAS\n",
      "Bicyclus_anynana Exon_22 12_frame0 1.708 \n",
      "GSRERNFLTIHKYYY\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt --------------------------ACKLTMPQKLGKSYFGNMRREKMAEVRANKGGHG\n",
      "set73_frame0    VRRLFTYSYFNKNNKTNFIPTKRYFFVKKYLIPMQLTLAYELLQNKHKSSETNVS-----\n",
      "                                          . *  :* :*  :*    .:.* :*....     \n",
      "\n",
      "28.Maniola_jurt RNIIGSSSVGASNR--------------------------\n",
      "set73_frame0    --FLRSPDAVATNANARKIPRRKMDPPWMEQVSFVYLRYI\n",
      "                  :: *... *:*                           \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege ----NKADSASAPK-KARLEHSDT-NEKTDGESSYSSFYSSFFKTDSGSNEESVSKSRGK\n",
      "set62_frame1    VVFQQSTDAMTGESLKRRLDTSDSNNDKTDGESSYSSFYSSFFKTDSGSGSDTRNQN-NK\n",
      "                    :.:*: :. . * **: **: *:**********************..:: .:. .*\n",
      "\n",
      "33.Pararge_aege RGSKAS------\n",
      "set62_frame1    GHVKASVSILTK\n",
      "                   ***      \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn ACKLTIPMKYV--QSCNGHGRREKNV--------ECRANNGVHGRDISSSFAGASNR\n",
      "set12_frame0    NFH---PLRGLKDQAAGGQGSRERNFLTIHKYYYVCNKNSIV----FQVSYRCIILR\n",
      "                  :   *:: :  *:..*:* **:*.         *. *. *    :. *:     *\n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 n\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_23 Match\n",
      "[['Maniola_jurtina', ['7641724', '7641903']], ['Pararge_aegeria', ['7641724', '7641903']], ['Bicyclus_anynana', ['7641724', '7641903']]]\n",
      "Exon_24 mismatch\n",
      "[['Maniola_jurtina', ['7642137', '7642244']], ['Pararge_aegeria', ['7642131', '7642244']], ['Bicyclus_anynana', ['7642137', '7642244']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_24 73_frame0 0.884 \n",
      "Issue with query\n",
      "Pararge_aegeria Exon_24 74_frame1 0.834 \n",
      "VCVSSELIYKYQILSKNIDEVLSSDKEKLKQLEQ\n",
      "Bicyclus_anynana Exon_24 73_frame0 1.152 \n",
      "NKHKSSETN\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt Y--------------------------------------------RHK---TNGKFWHNS\n",
      "set73_frame0    VRRLFTYSYFNKNNKTNFIPTKRYFFVKKYLIPMQLTLAYELLQNKHKSSETNVSFLRSP\n",
      "                                                             :**   ** .* :..\n",
      "\n",
      "28.Maniola_jurt DVSMTNVANARKIPRRKMDPPWMEQ---------\n",
      "set73_frame0    DAVATN-ANARKIPRRKMDPPWMEQVSFVYLRYI\n",
      "                *.  ** ******************         \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege VNATSDLFYKYQIPTRDCNEVLSNDKEKLKLLDQ--------------\n",
      "set74_frame1    VCVSSELIYKYQILSKNIDEVLSSDKEKLKQLEQVSLFIFLLPQMRHF\n",
      "                * .:*:*:***** ::: :****.****** *:*              \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn --------------------------------------------NKYKILSS-----KTP\n",
      "set73_frame0    VRRLFTYSYFNKNNKTNFIPTKRYFFVKKYLIPMQLTLAYELLQNKHKSSETNVSFLRSP\n",
      "                                                            **:*  .:     ::*\n",
      "\n",
      "5.Bicyclus_anyn TIPSTYVPNDRKVARRKMEPPWMEQ---------\n",
      "set73_frame0    DAVATNA-NARKIPRRKMDPPWMEQVSFVYLRYI\n",
      "                   :* . * **:.****:******         \n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 n\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_25 Match\n",
      "[['Maniola_jurtina', ['7642564', '7642665']], ['Pararge_aegeria', ['7642564', '7642665']], ['Bicyclus_anynana', ['7642564', '7642665']]]\n",
      "Exon_26 mismatch\n",
      "[['Maniola_jurtina', ['7643259', '7643396']], ['Pararge_aegeria', ['7642098', '7642151']], ['Bicyclus_anynana', ['7643259', '7643381']]]\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Proceed? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina Exon_26 100_frame0 0.144 \n",
      "PALVNEQLSQLYLDLQLGGVAARLTLEEGITSSSSSGEESTVRSSK\n",
      "Pararge_aegeria Exon_26 73_frame0 1.728 \n",
      "Issue with query\n",
      "Bicyclus_anynana Exon_26 100_frame0 0.749 \n",
      "PALVNEQLSQLYLDLQLGGVAARLTLEEGITSSSSSGEEST\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Check Alignment?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "28.Maniola_jurt PSLVNEQLSQLYLDLQLEGVAARLTLEEGITSSSSSGEESTAKSSK-----\n",
      "set100_frame0   PALVNEQLSQLYLDLQLGGVAARLTLEEGITSSSSSGEESTVRSSKVWRVR\n",
      "                *:*************** ***********************.:***     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "33.Pararge_aege --------------------------------PRRRNLQYDRLALIHEADA---------\n",
      "set73_frame0    VRRLFTYSYFNKNNKTNFIPTKRYFFVKKYLIPMQLTLAYELLQNKHKSSETNVSFLRSP\n",
      "                                                * : .* *: *   *::.          \n",
      "\n",
      "33.Pararge_aege ----------PMPPPEMDVPMLSPGS----SSS\n",
      "set73_frame0    DAVATNANARKIPRRKMDPPWMEQVSFVYLRYI\n",
      "                           :*  :** * :.  *       \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.520)\n",
      "\n",
      "\n",
      "5.Bicyclus_anyn SPLVCEQLSQLYLDLQLQGVAARLTLEDGITSSSSSGEDTP-NHPKPSTSK\n",
      "set100_frame0   PALVNEQLSQLYLDLQLGGVAARLTLEEGITSSSSSGEESTVRSSKVWRVR\n",
      "                ..** ************ *********:**********::. . .*    :\n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 3 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_27 Match\n",
      "[['Maniola_jurtina', ['7643993', '7644121']], ['Pararge_aegeria', ['7643993', '7644121']], ['Bicyclus_anynana', ['7643993', '7644121']]]\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Erebia_aethiops,OV281079.1,7624959,7625041,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "Erebia_aethiops,OV281079.1,7627065,7627148,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "Erebia_aethiops,OV281079.1,7627773,7627945,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_1_query_Exon_3,1,57,57,Y,N\n",
      "Erebia_aethiops,OV281079.1,7628823,7628917,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_0_query_Exon_4,1,31,31,Y,N\n",
      "Erebia_aethiops,OV281079.1,000,000,0,Y,Error_Exon_5,00,00,00\n",
      "Erebia_aethiops,OV281079.1,7630393,7630524,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "Erebia_aethiops,OV281079.1,7631378,7631453,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_1_query_Exon_7,1,25,25,Y,N\n",
      "Erebia_aethiops,OV281079.1,7631881,7632084,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_8,1,67,67,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7632564,7632792,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7633837,7633948,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "Erebia_aethiops,OV281079.1,7634044,7634163,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7634969,7635118,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "Erebia_aethiops,OV281079.1,7635881,7635999,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7636136,7636240,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_14,1,34,34,Y,N\n",
      "Erebia_aethiops,OV281079.1,7636666,7636818,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_2_query_Exon_15,1,50,50,Y,N\n",
      "Erebia_aethiops,OV281079.1,7637174,7637342,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_1_rightoh_0_query_Exon_16,1,56,56,Y,N\n",
      "Erebia_aethiops,OV281079.1,7638011,7638152,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_1_query_Exon_17,1,47,47,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7638571,7638675,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_18,1,34,34,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7639269,7639370,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_19,1,33,33,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7639930,7640106,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_1_query_Exon_20,1,58,58,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7640747,7640901,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_2_rightoh_0_query_Exon_21,1,51,51,Y,Y\n",
      "Erebia_aethiops,OV281079.1,000,000,0,Y,Error_Exon_22,00,00,00\n",
      "Erebia_aethiops,OV281079.1,7641724,7641903,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_23,1,60,60,Y,N\n",
      "Erebia_aethiops,OV281079.1,000,000,0,Y,Error_Exon_24,00,00,00\n",
      "Erebia_aethiops,OV281079.1,7642564,7642665,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_25,1,34,34,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7643259,7643396,0,N,28.Maniola_jurtina_XM_045928854.1_Frame_0_rightoh_0_query_Exon_26,1,46,46,Y,Y\n",
      "Erebia_aethiops,OV281079.1,7643993,7644121,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_0_query_Exon_27,1,31,32\n",
      "\n",
      "Erebia_ligea\n",
      "Exon_1 Match\n",
      "[['Maniola_jurtina', ['13282649', '13282731']], ['Pararge_aegeria', ['13282649', '13282731']], ['Bicyclus_anynana', ['13282649', '13282731']]]\n",
      "Exon_2 Match\n",
      "[['Maniola_jurtina', ['13280169', '13280252']], ['Pararge_aegeria', ['13280169', '13280252']], ['Bicyclus_anynana', ['13280169', '13280252']]]\n",
      "Exon_3 Match\n",
      "[['Maniola_jurtina', ['13279313', '13279485']], ['Pararge_aegeria', ['13279313', '13279485']], ['Bicyclus_anynana', ['13279313', '13279485']]]\n",
      "Exon_4 mismatch\n",
      "[['Maniola_jurtina', ['13278401', '13278495']], ['Pararge_aegeria', ['13278401', '13278495']], ['Bicyclus_anynana', ['13278431', '13278495']]]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "Interrupted by user",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 77\u001b[0m\n\u001b[1;32m     60\u001b[0m \u001b[38;5;66;03m# for query_species_name in query_species_list:\u001b[39;00m\n\u001b[1;32m     61\u001b[0m \u001b[38;5;66;03m#     list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\u001b[39;00m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;66;03m#     for folders in list_of_folders:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     73\u001b[0m \u001b[38;5;66;03m#             assert False\u001b[39;00m\n\u001b[1;32m     74\u001b[0m              \u001b[38;5;66;03m# \u001b[39;00m\n\u001b[1;32m     76\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m---> 77\u001b[0m     choice \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43minput\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mProceed?\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     78\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m choice[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124my\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mor\u001b[39;00m choice[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mn\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m     79\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/ipykernel/kernelbase.py:1270\u001b[0m, in \u001b[0;36mKernel.raw_input\u001b[0;34m(self, prompt)\u001b[0m\n\u001b[1;32m   1268\u001b[0m     msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraw_input was called, but this frontend does not support input requests.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1269\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m StdinNotImplementedError(msg)\n\u001b[0;32m-> 1270\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_input_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1271\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mprompt\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1272\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_parent_ident\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mshell\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1273\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_parent\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mshell\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1274\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpassword\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m   1275\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/ipykernel/kernelbase.py:1313\u001b[0m, in \u001b[0;36mKernel._input_request\u001b[0;34m(self, prompt, ident, parent, password)\u001b[0m\n\u001b[1;32m   1310\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m:\n\u001b[1;32m   1311\u001b[0m     \u001b[38;5;66;03m# re-raise KeyboardInterrupt, to truncate traceback\u001b[39;00m\n\u001b[1;32m   1312\u001b[0m     msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInterrupted by user\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1313\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m(msg) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1314\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[1;32m   1315\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlog\u001b[38;5;241m.\u001b[39mwarning(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid Message:\u001b[39m\u001b[38;5;124m\"\u001b[39m, exc_info\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: Interrupted by user"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "species_list = [\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "species_list = [\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "\n",
    "for species in species_list:\n",
    "    print(species)\n",
    "    blast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{blast_location}/{species}\")\n",
    "    query_species_list = []\n",
    "    coordinate_detail_dictionary = {}\n",
    "    coordinate_info_dictionary = {}\n",
    "    scaffold = ''\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\".csv\") and len(file.split(\"_\")) >= 5:\n",
    "            query_species = f\"{file.split('_')[3].split('.')[1]}_{file.split('_')[4].split('.')[0]}\"\n",
    "            query_species_list.append(query_species)\n",
    "            coordinate_detail_dictionary[query_species] = {}\n",
    "            with open(f\"{blast_location}/{species}/{file}\", 'r') as coordinate_file_open:\n",
    "                coordinate_file_lines_list  =  coordinate_file_open.readlines()\n",
    "            if scaffold == '':\n",
    "                scaffold = coordinate_file_lines_list[1].split(\",\")[1]\n",
    "    \n",
    "            for i in range(1,len(coordinate_file_lines_list)):\n",
    "                \n",
    "                coordinate_file_lines_split = coordinate_file_lines_list[i].split(\",\")\n",
    "                start,stop = coordinate_file_lines_split[2],coordinate_file_lines_split[3]\n",
    "                if query_species == \"Pararge_aegeria\" and i > 21 and i < 26:\n",
    "                    exon = f\"Exon_{i+1}\"\n",
    "                else:\n",
    "                    exon = f\"Exon_{i}\"\n",
    "                if exon not in coordinate_info_dictionary:\n",
    "                    coordinate_info_dictionary[exon] = {}\n",
    "                coordinate_info_dictionary[exon][query_species] = [start,stop]\n",
    "                coordinate_detail_dictionary[query_species][exon] = coordinate_file_lines_list[i]\n",
    "            \n",
    "            # break\n",
    "            # print(query_species)\n",
    "            # print(file)\n",
    "    # print(coordinate_info_dictionary)\n",
    "    # assert False\n",
    "    output = coordinate_file_lines_list[0]\n",
    "    for exon,species_specific_coordinates in coordinate_info_dictionary.items():\n",
    "        # print(exon,species_specific_coordinates)\n",
    "        coordinates_list = []\n",
    "        previous_coordinates = \"\"\n",
    "        match = 1\n",
    "        for query_species, coordinates in coordinate_info_dictionary[exon].items():\n",
    "            coordinates_list.append([query_species,coordinates])\n",
    "            if previous_coordinates != '':\n",
    "                if coordinates != previous_coordinates:\n",
    "                    match = 0\n",
    "                    \n",
    "            previous_coordinates = coordinates\n",
    "            \n",
    "        if match == 0:\n",
    "            print(exon,\"mismatch\")\n",
    "            print(coordinates_list)\n",
    "            # for query_species_name in query_species_list:\n",
    "            #     list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "            #     for folders in list_of_folders:\n",
    "            #         if folders.endswith(query_species_name):\n",
    "            #             # print()\n",
    "                        \n",
    "            #             with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "            #                 # alignment_detail = query_file_open.readlines()[0]\n",
    "            #                 alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                            \n",
    "            #                 with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "            #                     distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "            #             print(query_species_name,exon, alignment_detail, distance_score)\n",
    "            #             assert False\n",
    "                         # \n",
    "                \n",
    "            while True:\n",
    "                choice = input(\"Proceed?\")\n",
    "                if choice[0].lower() == \"y\" or choice[0].lower() == \"n\":\n",
    "                    break\n",
    "            \n",
    "            while True:\n",
    "                if choice[0].lower() == \"y\":\n",
    "                    alignment_detail_dictionary = {}\n",
    "                    for query_species_name in query_species_list:\n",
    "                        list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                        for folders in list_of_folders:\n",
    "                            if folders.endswith(query_species_name):\n",
    "                                # print()\n",
    "                                \n",
    "                                with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "                                    # alignment_detail = query_file_open.readlines()[0]\n",
    "                                    alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                                    alignment_detail_dictionary[query_species_name] = alignment_detail\n",
    "                                with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "                                    distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "                                try:\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query_spliced.txt\", 'r') as new_query_file:\n",
    "                                        new_query = (new_query_file.readlines()[-1].rstrip())\n",
    "                                except:\n",
    "                                    new_query = \"Issue with query\"\n",
    "                                print(query_species_name,exon, alignment_detail, distance_score,f\"\\n{new_query}\")\n",
    "                    while True:            \n",
    "                        check_alignment = input(\"Check Alignment??\")\n",
    "                        if check_alignment[0].lower() == \"y\" or check_alignment[0].lower() == \"n\":\n",
    "                            break\n",
    "                    if check_alignment[0].lower() == \"y\":\n",
    "                        for query_species_name in query_species_list:\n",
    "                            list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                            for folders in list_of_folders:\n",
    "                                if folders.endswith(query_species_name):\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/alignment_clustal_{exon}_translated_genomic_sequence_{alignment_detail_dictionary[query_species_name]}.fa.txt\", 'r') as clustal_alignment_file:\n",
    "                                        clustal_alignment = clustal_alignment_file.readlines()\n",
    "                                    print(\"\".join(clustal_alignment))\n",
    "                                # assert False\n",
    "                    while True:\n",
    "                        try:\n",
    "                            species_number = (input(f\"Choose Species 1 - {len(query_species_list)}\"))\n",
    "                            \n",
    "                            \n",
    "                            if species_number[0].lower() == 'n':\n",
    "                                output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                                break \n",
    "                            output += coordinate_detail_dictionary[query_species_list[int(species_number)-1]][exon]\n",
    "                            break\n",
    "                        except:\n",
    "                            print(\"Retry\")\n",
    "                    break\n",
    "                elif choice[0].lower() == \"n\":\n",
    "                    output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                    break\n",
    "                \n",
    "                \n",
    "        else:\n",
    "            print(exon, \"Match\")\n",
    "            print(coordinates_list)\n",
    "            output += coordinate_detail_dictionary[query_species_list[0]][exon]\n",
    "        # assert False\n",
    "    # print(coordinate_detail_dictionary)\n",
    "    print(output)\n",
    "    with open(f\"{blast_location}/{species}/final_coordinates.csv\", \"w\") as out_file:\n",
    "        out_file.write(output)\n",
    "    \n",
    "    # 'Coenonympha_glycerion,OY979626.1,7015875,7015963,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7892a6a6-9400-414d-bbaa-cf893036e0d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
