{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"3.Satyrine\"\n",
    "number_of_exons = 27\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan MDNMDDSDPNAKISDSAYSTSCSNSQSRR\n",
      "Coenonympha_gly MDNTDDSENNAKVSDSAYSNSCSNSQSRR\n",
      "Hipparchia_seme MDNLDDSENNAKISDSAYSNSCSNSQSRR\n",
      "Maniola_hyperan MDNLDDSENNAKVSDSAYSNSCSNSQSRR\n",
      "Maniola_jurtina MDNLDESENNAKVSDSAYSNSCSNSQSRR\n",
      "Oeneis_ivallda_ MENLDDSENNAKISDSAYSNSCSNSQSRR\n",
      "Lasiommata_mege MDNMDDSENNVKISDSAYSNSCSNSQSRR\n",
      "Melanargia_gala MDTLDDSENTAKISDSAYSNSCSNSQSRR\n",
      "Elymnias_hyperm MANLDDSENNVKISDSAYSTSCSNSQSRR\n",
      "Erebia_aethiops MDSLDVSD--AKVSDSAYSNSCSNSQSRR\n",
      "Erebia_ligea_Ex MDNLDVSD--AKVSDSAYSNSCSNSQSRR\n",
      "Pararge_aegeria MNKMDCSENNAKVSDAGYSNSCTNSQSRR\n",
      "                * . * *:  .*:**:.**.**:******\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan HSSKSTH----SSGSSGYGGKQTTSGY\n",
      "Coenonympha_gly HSSKSTHSGSNSSGSSGYGGKPSTSGF\n",
      "Maniola_hyperan HSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "Erebia_aethiops HSSKSTRSGSNSSGSSGYGGKPSTSGY\n",
      "Erebia_ligea_Ex HSSKSTRSGSNSSGSSGYGGKPSTSGY\n",
      "Hipparchia_seme HSSKSTRSGSNSSGSSGYGGKPSTSGY\n",
      "Melanargia_gala HSSKSTRSGSNSSGSSGYGGKPSTSGY\n",
      "Oeneis_ivallda_ HSSKSTRSGSNSSGSSGYGGKPSTSGY\n",
      "Maniola_jurtina RSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "Pararge_aegeria HSSKSTHSESNSSGSSGYGGKPSSSGY\n",
      "Lasiommata_mege HSSKSTHSGSNSSGSSGYGGKPS-SGY\n",
      "Elymnias_hyperm HSSKSTHSGSHSSGSSGYGGKTSTSDY\n",
      "                :*****:    ********** : *.:\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan NNNQTQSPEKRPKEKELKKKKTQLTTQNDVQVVEAKIEIVCDVTMKPVF--EPLPEEISD\n",
      "Coenonympha_gly INTQSQPSEKRSKEKDAKKKIPMQSVQTELQTTDVQNETELVPVTVPSM--IETSAEIVE\n",
      "Erebia_aethiops -TNQSQPADKRCKDKETKKKKPLQALQTDITVSEVIKENECISEPAPIFTTESPKDDIGD\n",
      "Erebia_ligea_Ex -TNQSQPADKRCKDKETKKKKPLQALQTDLKVSEVIKENECISETAPIITAESLKDDISH\n",
      "Hipparchia_seme SNNQSQPTEKRSKDKDAKKKKPLQNIQTDLQGAEVKNVIECIPEIVPVF--ESSKEEIND\n",
      "Oeneis_ivallda_ SNNQSQPTEKRSKDKDTKKKKPLQNIQTDLQVSEVKNVIECIPEIVPVI--ESSKKEITD\n",
      "Maniola_hyperan -NNQGQPAEKRSKDKDAKKKKPLQSLQTDLQVTEVINETECVRELAPIL--ESSKEEISD\n",
      "Maniola_jurtina INNESQPPEKRSKDKDAKKKIPLQGLQTDLKVTEVINENECVPEIAPIF--ESSKEDISD\n",
      "Melanargia_gala SNIQIEPTEKRTKDKEIKKKKPLQSLQTDLQVTEAKNEIECIPENVPIF--ESSKEDIND\n",
      "Lasiommata_mege SNN--QAAEKRAKDKESKKKKTLQGIQKDLQVTVSKNEIESIPEHAPIF--EFSKEESN-\n",
      "Pararge_aegeria SNNQSQIAEKRNKEKELKKKRPMQFIQSDLKANMAKKAIQKIPLLTPVF--EFSKAEI--\n",
      "Elymnias_hyperm SNNQSQPQVAHVRDQEFKKNNPV--LTETWKLGEVTNETECVPEPAPSL--VASQSNESN\n",
      "                 .   :    : :::: **: .                        * :       :   \n",
      "\n",
      "Bicyclus_anynan --\n",
      "Coenonympha_gly I-\n",
      "Erebia_aethiops --\n",
      "Erebia_ligea_Ex --\n",
      "Hipparchia_seme --\n",
      "Oeneis_ivallda_ --\n",
      "Maniola_hyperan --\n",
      "Maniola_jurtina --\n",
      "Melanargia_gala --\n",
      "Lasiommata_mege --\n",
      "Pararge_aegeria --\n",
      "Elymnias_hyperm KK\n",
      "                  \n",
      "\n",
      "continue?Elymnias_hyperm\n",
      "Elymnias_hypermnestra_Exon_3_JACTOM010000032.1_219447_219622_left_1_right_1\n",
      "Species_name?Elymnias_hypermnestra\n",
      "Scaffold:JACTOM010000032.1\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan --VPSPASGVVKMEKGPEL-DVSDKKS--GKDE\n",
      "Coenonympha_gly --ARSPTPGLAQAEVAPEAIEVCEKSS--GKDE\n",
      "Erebia_aethiops CQQQTQSPASAQTEKGPELMEVSDNSS--GKEQ\n",
      "Erebia_ligea_Ex CQQQTQSPVSAQTEKGPELMEVSDDSS--GKEQ\n",
      "Maniola_hyperan --TQTQLPGSVQPEDGPEPMEVSDRTS--GKDE\n",
      "Maniola_jurtina --AQTQSPGLVQPEKGPETMEVSDRTS--GKDE\n",
      "Hipparchia_seme --PQIPTPGLAQIEKVPDLMEVSDRNS--GKDE\n",
      "Oeneis_ivallda_ --PQLPTPGLAQIDNVPVLMELSDRSS--GKDE\n",
      "Melanargia_gala --APIPTPVLAQTEKVPDLMEETDKSN--GKDE\n",
      "Lasiommata_mege --VPMPSPALAQMEKGPELMEISERAN--GKEE\n",
      "Pararge_aegeria --AKMPNHGLALKENKPEKMDISDRSS--GKDE\n",
      "Elymnias_hyperm --AETPTTSKAMIENKPDLRGIADSVEDGSQDM\n",
      "                          .  :  *      :  .  .:: \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
