{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85ae0a5c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_15\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        overhang = int(records.id.split(\"_\")[-3])\n",
    "        \n",
    "        translated_sequence = records.seq[overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        alignment_start, alignment_stop = alignments[0].coordinates[1][1],alignments[0].coordinates[1][2]\n",
    "        print(records.seq[alignment_start*3-overhang+1:])\n",
    "#         \n",
    "#         print(sequence)\n",
    "        current_position = (alignment_start)*3\n",
    "        if (sequence[current_position-2:current_position]).lower() == \"ag\":\n",
    "            print(\"AG\\n\\n\\n\")\n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_start)*3)\n",
    "                else:\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_start)*3)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3-overhang+1:]}'\n",
    "                print(output)\n",
    "                out_file.write(output)\n",
    "#             assert False\n",
    "            break\n",
    "            \n",
    "            \n",
    "#         while current_position >0:\n",
    "# #             print((sequence[current_position:]))\n",
    "#             print((sequence[current_position-2:current_position]).lower(), end = \",\")\n",
    "# #             print((sequence[current_position+1:current_position+3]).lower()== \"ag\", end = \",\")\n",
    "#             if (sequence[current_position-2:current_position]).lower() == \"ag\":\n",
    "#                 break\n",
    "#             print(current_position,len(sequence))\n",
    "#             if current_position >= len(sequence):\n",
    "#                 break\n",
    "#             current_position -= 3\n",
    "# #             assert False\n",
    "#         print(sequence[current_position+3:])\n",
    "#         assert False\n",
    "#     break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ffad589",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_7\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        l_overhang = int(records.id.split(\"_\")[-3])\n",
    "        r_overhang = int(records.id.split(\"_\")[-1])\n",
    "        translated_sequence = records.seq[l_overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        print(alignments[0].coordinates)\n",
    "        alignment_start = int(input(\"Alignment Start?\"))\n",
    "        alignment_stop = int(input(\"Alignment stop?\"))\n",
    "#         alignment_start, alignment_stop = alignments[0].coordinates[1][1],alignments[0].coordinates[1][2]\n",
    "        print(records.seq[alignment_start*3:(alignment_stop)*3+r_overhang])\n",
    "#         assert False\n",
    "#         \n",
    "#         print(sequence)\n",
    "#         current_position = (alignment_start)*3\n",
    "        found = 0\n",
    "        while alignment_stop < len(records.seq):\n",
    "        \n",
    "            if (sequence[(alignment_stop)*3+r_overhang:(alignment_stop)*3+r_overhang+2]).lower() == \"gt\":\n",
    "                print(\"GT\")\n",
    "                print(records.seq[alignment_start*3:(alignment_stop)*3+r_overhang])\n",
    "                print((records.seq[alignment_start*3:(alignment_stop)*3+r_overhang]).translate())\n",
    "                found = 1\n",
    "                break\n",
    "            alignment_stop+= 1\n",
    "        if found == 1:   \n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_stop)*3)\n",
    "                else:\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_stop)*3)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3:(alignment_stop)*3+r_overhang]}'\n",
    "                print(output+\"\\n\\n\")\n",
    "                out_file.write(output)\n",
    "# #             assert False\n",
    "#             break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "1881ee7b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "target            0 ----------ME------VSDRTSGKDE 12\n",
      "                  0 ----------||------|||..||||| 28\n",
      "query             0 VPSPASGVVKMEKGPELDVSDKKSGKDE 28\n",
      "\n",
      "[[ 0  0  2  2 12]\n",
      " [ 0 10 12 18 28]]\n",
      "10 28\n",
      "This OK?y\n",
      "ATGGAGAAGGGACCTGAGTTGGATGTATCTGATAAAAAAAGTGGAAAAGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAGAAGGGACCTGAGTTGGATGTATCTGATAAAAAAAGTGGAAAAGACG\n",
      "MEKGPELDVSDKKSGKD\n",
      "complement =  0\n",
      ">Bicyclus_anynana_Exon_4_NC_069110.1_7835512_7835513_left_2_right_0\n",
      "ATGGAGAAGGGACCTGAGTTGGATGTATCTGATAAAAAAAGTGGAAAAGACG\n",
      "\n",
      "\n",
      "Coenonympha_glycerion\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------.||....||||| 29\n",
      "query             0 ARSPTPGLAQAEVAPEAIEVCEKSSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATCGAAGTGTGTGAAAAGAGCAGCGGAAAAGACG\n",
      "atc\n",
      "gca\n",
      "gag\n",
      "ccc\n",
      "gcg\n",
      "gtg\n",
      "gag\n",
      "gcg\n",
      "caa\n",
      "gcg\n",
      "ttg\n",
      "gga\n",
      "ccc\n",
      "aca\n",
      "cca\n",
      "tct\n",
      "cga\n",
      "Elymnias_hypermnestra\n",
      "target            0 ME-------------------VSDRTSGKDE 12\n",
      "                  0 .|-------------------.|......|. 31\n",
      "query             0 AETPTTSKAMIENKPDLRGIADSVEDGSQDM 31\n",
      "\n",
      "[[ 0  2  2 12]\n",
      " [ 0  2 21 31]]\n",
      "2 31\n",
      "This OK?n\n",
      "Alignment Start?21\n",
      "GACTCCGTTGAAGATGGCTCGCAAGATA\n",
      "gac\n",
      "gct\n",
      "atc\n",
      "ggc\n",
      "agg\n",
      "ctc\n",
      "gac\n",
      "cct\n",
      "aaa\n",
      "aat\n",
      "gaa\n",
      "ata\n",
      "atg\n",
      "ATG\n",
      "ATGATAGAAAATAAACCTGACCTCAGGGGCATCGCTGACTCCGTTGAAGATGGCTCGCAAGATA\n",
      "MIENKPDLRGIADSVEDGSQD\n",
      "complement =  1\n",
      ">Elymnias_hypermnestra_Exon_4_JACTOM010000032.1_216289_216290_left_2_right_0\n",
      "ATGATAGAAAATAAACCTGACCTCAGGGGCATCGCTGACTCCGTTGAAGATGGCTCGCAAGATA\n",
      "\n",
      "\n",
      "Erebia_aethiops\n",
      "target            0 -------------------MEVSDRTSGKDE 12\n",
      "                  0 -------------------|||||..|||.. 31\n",
      "query             0 CQQQTQSPASAQTEKGPELMEVSDNSSGKEQ 31\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 19 31]]\n",
      "19 31\n",
      "This OK?y\n",
      "ATGGAAGTGTCTGATAATAGCAGCGGAAAAGAAC\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGTGTCTGATAATAGCAGCGGAAAAGAAC\n",
      "MEVSDNSSGKE\n",
      "complement =  0\n",
      ">Erebia_aethiops_Exon_4_OV281079.1_7628823_7628824_left_2_right_0\n",
      "ATGGAAGTGTCTGATAATAGCAGCGGAAAAGAAC\n",
      "\n",
      "\n",
      "Erebia_ligea\n",
      "target            0 -------------------MEVSDRTSGKDE 12\n",
      "                  0 -------------------|||||..|||.. 31\n",
      "query             0 CQQQTQSPVSAQTEKGPELMEVSDDSSGKEQ 31\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 19 31]]\n",
      "19 31\n",
      "This OK?y\n",
      "ATGGAAGTATCTGATGATAGCAGTGGAAAAGAAC\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGTATCTGATGATAGCAGTGGAAAAGAAC\n",
      "MEVSDDSSGKE\n",
      "complement =  1\n",
      ">Erebia_ligea_Exon_4_OU785222.1_13278494_13278495_left_2_right_0\n",
      "ATGGAAGTATCTGATGATAGCAGTGGAAAAGAAC\n",
      "\n",
      "\n",
      "Hipparchia_semele\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------||||||.||||| 29\n",
      "query             0 PQIPTPGLAQIEKVPDLMEVSDRNSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAAGTATCCGACAGAAACAGTGGGAAAGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGTATCCGACAGAAACAGTGGGAAAGACG\n",
      "MEVSDRNSGKD\n",
      "complement =  1\n",
      ">Hipparchia_semele_Exon_4_OW121737.1_511099_511100_left_2_right_0\n",
      "ATGGAAGTATCCGACAGAAACAGTGGGAAAGACG\n",
      "\n",
      "\n",
      "Lasiommata_megera\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------||.|.|..||.| 29\n",
      "query             0 VPMPSPALAQMEKGPELMEISERANGKEE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAAATATCTGAAAGGGCCAATGGAAAAGAAG\n",
      "atg\n",
      "ATG\n",
      "ATGGAAATATCTGAAAGGGCCAATGGAAAAGAAG\n",
      "MEISERANGKE\n",
      "complement =  1\n",
      ">Lasiommata_megera_Exon_4_OV743307.1_17364065_17364066_left_2_right_0\n",
      "ATGGAAATATCTGAAAGGGCCAATGGAAAAGAAG\n",
      "\n",
      "\n",
      "Maniola_hyperantus\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------|||||||||||| 29\n",
      "query             0 TQTQLPGSVQPEDGPEPMEVSDRTSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAAGTATCTGACAGGACCAGTGGAAAAGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGTATCTGACAGGACCAGTGGAAAAGACG\n",
      "MEVSDRTSGKD\n",
      "complement =  0\n",
      ">Maniola_hyperantus_Exon_4_NC_048564.1_6318417_6318418_left_2_right_0\n",
      "ATGGAAGTATCTGACAGGACCAGTGGAAAAGACG\n",
      "\n",
      "\n",
      "Maniola_jurtina\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------|||||||||||| 29\n",
      "query             0 AQTQSPGLVQPEKGPETMEVSDRTSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAAGTATCTGATAGGACGAGTGGAAAAGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGTATCTGATAGGACGAGTGGAAAAGACG\n",
      "MEVSDRTSGKD\n",
      "complement =  0\n",
      ">Maniola_jurtina_Exon_4_NC_060058.1_5750025_5750026_left_2_right_0\n",
      "ATGGAAGTATCTGATAGGACGAGTGGAAAAGACG\n",
      "\n",
      "\n",
      "Melanargia_galathea\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------||..|...|||| 29\n",
      "query             0 APIPTPVLAQTEKVPDLMEETDKSNGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAAGAGACTGATAAGAGCAATGGAAAGGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAAGAGACTGATAAGAGCAATGGAAAGGACG\n",
      "MEETDKSNGKD\n",
      "complement =  0\n",
      ">Melanargia_galathea_Exon_4_OV049860.1_10029176_10029177_left_2_right_0\n",
      "ATGGAAGAGACTGATAAGAGCAATGGAAAGGACG\n",
      "\n",
      "\n",
      "Oeneis_ivallda\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------||.|||.||||| 29\n",
      "query             0 PQLPTPGLAQIDNVPVLMELSDRSSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGAATTGTCTGATAGGAGCAGTGGAAAAGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGAATTGTCTGATAGGAGCAGTGGAAAAGACG\n",
      "MELSDRSSGKD\n",
      "complement =  0\n",
      ">Oeneis_ivallda_Exon_4_JARPMR010000001.1_13909515_13909516_left_2_right_0\n",
      "ATGGAATTGTCTGATAGGAGCAGTGGAAAAGACG\n",
      "\n",
      "\n",
      "Pararge_aegeria\n",
      "target            0 -----------------MEVSDRTSGKDE 12\n",
      "                  0 -----------------|..|||.||||| 29\n",
      "query             0 AKMPNHGLALKENKPEKMDISDRSSGKDE 29\n",
      "\n",
      "[[ 0  0 12]\n",
      " [ 0 17 29]]\n",
      "17 29\n",
      "This OK?y\n",
      "ATGGACATATCTGACAGAAGCAGCGGAAAGGACG\n",
      "atg\n",
      "ATG\n",
      "ATGGACATATCTGACAGAAGCAGCGGAAAGGACG\n",
      "MDISDRSSGKD\n",
      "complement =  1\n",
      ">Pararge_aegeria_Exon_4_NC_053208.1_16925975_16925976_left_2_right_0\n",
      "ATGGACATATCTGACAGAAGCAGCGGAAAGGACG\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Align\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.open_gap_score = -1\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "alternate_exon = \"Exon_4\"\n",
    "\n",
    "alternate_exon_fasta = SeqIO.parse(f\"{output_location}/2.Alternate_exons/{alternate_exon}.txt\", \"fasta\")\n",
    "for sequences_record in alternate_exon_fasta:\n",
    "    alternate_sequence = sequences_record.seq\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "list_of_species.remove('desktop.ini')\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    fasta_file = SeqIO.parse(f\"{output_location}/2.Final_output/{species}/{alternate_exon}.fa\",'fasta')\n",
    "    for records in fasta_file:\n",
    "        l_overhang = int(records.id.split(\"_\")[-3])\n",
    "        r_overhang = int(records.id.split(\"_\")[-1])\n",
    "        translated_sequence = records.seq[l_overhang:].translate()\n",
    "        sequence = records.seq\n",
    "#         print(alternate_sequence,translated_sequence)\n",
    "        alignments = aligner.align(alternate_sequence, translated_sequence)\n",
    "        print(alignments[0])\n",
    "        print(alignments[0].coordinates)\n",
    "#         alignment_start = int(input(\"Alignment Start?\"))\n",
    "#         alignment_stop = int(input(\"Alignment stop?\"))\n",
    "        alignment_start, alignment_stop = alignments[0].coordinates[1][1],int(len(records.seq)/3)\n",
    "        print(alignment_start, alignment_stop)\n",
    "        check = input(\"This OK?\")\n",
    "        if check[0].lower() == \"n\":\n",
    "            alignment_start = int(input(\"Alignment Start?\"))\n",
    "#             alignment_stop = int(input(\"Alignment stop?\"))\n",
    "        print(records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang])\n",
    "#         assert False\n",
    "#         \n",
    "#         print(sequence)\n",
    "#         current_position = (alignment_start)*3\n",
    "        found = 0\n",
    "    \n",
    "        while alignment_start > 0:\n",
    "            print(sequence[(alignment_start)*3+l_overhang:(alignment_start)*3+l_overhang+3].lower())\n",
    "            if (sequence[(alignment_start)*3+l_overhang:(alignment_start)*3+l_overhang+3].lower()) == \"atg\":\n",
    "                print(\"ATG\")\n",
    "                print(records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang])\n",
    "                print((records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang]).translate())\n",
    "                found = 1\n",
    "                break\n",
    "            alignment_start-= 1\n",
    "        if found == 1:   \n",
    "            with open(f\"{output_location}/2.Final_output/{species}/{alternate_exon}_alternate.fa\", 'w') as out_file:\n",
    "                name_split = records.id.split(\"_\")\n",
    "                with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\",'r') as coordinate_file:\n",
    "                    complement = coordinate_file.readlines()[1].split(\",\")[4]\n",
    "                print(\"complement = \",complement)\n",
    "                if complement == \"0\":\n",
    "                    name_split[-6] = str(int(name_split[-6])+(alignment_start)*3)\n",
    "                else:\n",
    "                    name_split[-5] = str(int(name_split[-5])-(alignment_start)*3)\n",
    "                output = f'>{\"_\".join(name_split)}\\n{records.seq[alignment_start*3+l_overhang:(alignment_stop)*3+r_overhang]}'\n",
    "                print(output+\"\\n\\n\")\n",
    "                out_file.write(output)\n",
    "# # #             assert False\n",
    "# #             break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4662577e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
