{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "4b77441b-77ad-4c18-a3f9-bab5117e5325",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coenonympha_glycerion\n",
      "Exon_1\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_2\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_3\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_4\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_5\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_6\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_7\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tMatch\tNot Match\n",
      "Pararge_aegeria\tMatch\tNot Match\n",
      "Exon_8\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_9\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_10\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_11\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_12\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_13\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_14\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_15\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_16\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_17\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tMatch\tNot Match\n",
      "Pararge_aegeria\tMatch\tNot Match\n",
      "Exon_18\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_19\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_20\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tNot Match\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_21\tStart\tStop\n",
      "Old\tMatch\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_22\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tNot Match\tNot Match\n",
      "Pararge_aegeria\tNot Match\tNot Match\n",
      "Exon_23\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_24\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_25\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_26\tStart\tStop\n",
      "Old\tNot Match\tNot Match\n",
      "Maniola_jurtina\tMatch\tNot Match\n",
      "Pararge_aegeria\tMatch\tMatch\n",
      "Exon_27\tStart\tStop\n",
      "Old\tMatch\tMatch\n",
      "Maniola_jurtina\tMatch\tMatch\n",
      "Pararge_aegeria\tMatch\tMatch\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "species_list = [\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "species = \"Coenonympha_glycerion\"\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "list_of_files = os.listdir(f\"{location}/{species}\")\n",
    "coordinate_files_list = []\n",
    "output = f'{species}'\n",
    "for file in list_of_files:\n",
    "    if file.endswith(\"_coordinates.csv\"):\n",
    "        with open(f\"{location}/{species}/{file}\", 'r') as coordinate_file:\n",
    "            coordinate_file_list = coordinate_file.readlines()\n",
    "    elif file.endswith(\".csv\"):\n",
    "        # if \"_old\" in file:\n",
    "        #     # output += \"\\told\\tQuery_name\"\n",
    "        # else:\n",
    "        #     species_name = file.split(\".\")[1]\n",
    "        #     output += f\"\\t{species_name}\\tQuery_name\"\n",
    "        coordinate_files_list.append(file)\n",
    "\n",
    "# print(output)\n",
    "# print(coordinate_files_list)\n",
    "\n",
    "for i in range(1,len(coordinate_file_list)):\n",
    "    coordinate_split = (coordinate_file_list[i].split(\",\"))\n",
    "    # print(coordinate_split)\n",
    "    exon = coordinate_split[6].split(\"Exon\")[1][1:]\n",
    "    start_coordinate = coordinate_split[2]\n",
    "    stop_coordinate = coordinate_split[3]\n",
    "    output += f\"\\nExon_{exon}\\tStart\\tStop\"\n",
    "    for alternative_coordinate_file in coordinate_files_list:\n",
    "        with open(f\"{location}/{species}/{alternative_coordinate_file}\", 'r') as alternative_coordinate_file_open:\n",
    "            alternative_coordinate_file_open_list = alternative_coordinate_file_open.readlines()\n",
    "        if \"_old\" in alternative_coordinate_file:\n",
    "            species_name = \"Old\"\n",
    "        else:\n",
    "            species_name = alternative_coordinate_file.split(\".\")[1]\n",
    "        output += f\"\\n{species_name}\"\n",
    "        match = 0\n",
    "        for j in range(1, len(alternative_coordinate_file_open_list)):\n",
    "            alternative_coordinate_file_split = alternative_coordinate_file_open_list[j].split(\",\")\n",
    "            alternative_start_coordinate = alternative_coordinate_file_split[2]\n",
    "            alternative_stop_coordinate = alternative_coordinate_file_split[3]\n",
    "            if alternative_start_coordinate == start_coordinate:\n",
    "                output += f\"\\tMatch\"\n",
    "                match = 1\n",
    "                if alternative_stop_coordinate == stop_coordinate:\n",
    "                    output += f\"\\tMatch\"\n",
    "                else:\n",
    "                    output += f\"\\tNot Match\"\n",
    "                break\n",
    "\n",
    "        if match == 0:\n",
    "            output += f\"\\tNot Match\\tNot Match\"\n",
    "        # output += \"\\n\"\n",
    "            # if alternative_start_coordinate == start_coordinate:\n",
    "print(output)\n",
    "with open(f\"{location}/{species}/match_mismatch.tsv\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee2d2aae-e268-4258-bc01-a00d928ca22a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### import os\n",
    "\n",
    "species_list = [\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "\n",
    "for species in species_list:\n",
    "    print(species)\n",
    "    blast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{blast_location}/{species}\")\n",
    "    query_species_list = []\n",
    "    coordinate_detail_dictionary = {}\n",
    "    coordinate_info_dictionary = {}\n",
    "    scaffold = ''\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\".csv\") and len(file.split(\"_\")) >= 5:\n",
    "            query_species = f\"{file.split('_')[3].split('.')[1]}_{file.split('_')[4].split('.')[0]}\"\n",
    "            query_species_list.append(query_species)\n",
    "            coordinate_detail_dictionary[query_species] = {}\n",
    "            with open(f\"{blast_location}/{species}/{file}\", 'r') as coordinate_file_open:\n",
    "                coordinate_file_lines_list  =  coordinate_file_open.readlines()\n",
    "            if scaffold == '':\n",
    "                scaffold = coordinate_file_lines_list[1].split(\",\")[1]\n",
    "    \n",
    "            for i in range(1,len(coordinate_file_lines_list)):\n",
    "                \n",
    "                coordinate_file_lines_split = coordinate_file_lines_list[i].split(\",\")\n",
    "                start,stop = coordinate_file_lines_split[2],coordinate_file_lines_split[3]\n",
    "                if query_species == \"Pararge_aegeria\" and i > 21 and i < 26:\n",
    "                    exon = f\"Exon_{i+1}\"\n",
    "                else:\n",
    "                    exon = f\"Exon_{i}\"\n",
    "                if exon not in coordinate_info_dictionary:\n",
    "                    coordinate_info_dictionary[exon] = {}\n",
    "                coordinate_info_dictionary[exon][query_species] = [start,stop]\n",
    "                coordinate_detail_dictionary[query_species][exon] = coordinate_file_lines_list[i]\n",
    "            \n",
    "            # break\n",
    "            # print(query_species)\n",
    "            # print(file)\n",
    "    # print(coordinate_info_dictionary)\n",
    "    # assert False\n",
    "    output = coordinate_file_lines_list[0]\n",
    "    for exon,species_specific_coordinates in coordinate_info_dictionary.items():\n",
    "        # print(exon,species_specific_coordinates)\n",
    "        coordinates_list = []\n",
    "        previous_coordinates = \"\"\n",
    "        match = 1\n",
    "        for query_species, coordinates in coordinate_info_dictionary[exon].items():\n",
    "            coordinates_list.append([query_species,coordinates])\n",
    "            if previous_coordinates != '':\n",
    "                if coordinates != previous_coordinates:\n",
    "                    match = 0\n",
    "                    \n",
    "            previous_coordinates = coordinates\n",
    "            \n",
    "        if match == 0:\n",
    "            print(exon,\"mismatch\")\n",
    "            print(coordinates_list)\n",
    "            # for query_species_name in query_species_list:\n",
    "            #     list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "            #     for folders in list_of_folders:\n",
    "            #         if folders.endswith(query_species_name):\n",
    "            #             # print()\n",
    "                        \n",
    "            #             with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "            #                 # alignment_detail = query_file_open.readlines()[0]\n",
    "            #                 alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                            \n",
    "            #                 with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "            #                     distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "            #             print(query_species_name,exon, alignment_detail, distance_score)\n",
    "            #             assert False\n",
    "                         # \n",
    "                \n",
    "            while True:\n",
    "                choice = input(\"Proceed?\")\n",
    "                if choice[0].lower() == \"y\" or choice[0].lower() == \"n\":\n",
    "                    break\n",
    "            \n",
    "            while True:\n",
    "                if choice[0].lower() == \"y\":\n",
    "                    for query_species_name in query_species_list:\n",
    "                        list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                        for folders in list_of_folders:\n",
    "                            if folders.endswith(query_species_name):\n",
    "                                # print()\n",
    "                                \n",
    "                                with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "                                    # alignment_detail = query_file_open.readlines()[0]\n",
    "                                    alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                                    \n",
    "                                with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "                                    distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "                                try:\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query_spliced.txt\", 'r') as new_query_file:\n",
    "                                        new_query = (new_query_file.readlines()[-1].rstrip())\n",
    "                                except:\n",
    "                                    new_query = \"Issue with query\"\n",
    "                                print(query_species_name,exon, alignment_detail, distance_score,f\"\\n{new_query}\")\n",
    "                                \n",
    "                    check_alignment = input(\"Check Alignment??\")\n",
    "                    if check_alignment[0].lower() == \"y\":\n",
    "                        for query_species_name in query_species_list:\n",
    "                            list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                            for folders in list_of_folders:\n",
    "                                if folders.endswith(query_species_name):\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/alignment_clustal_{exon}_translated_genomic_sequence_{alignment_detail}.fa.txt\", 'r') as clustal_alignment_file:\n",
    "                                        clustal_alignment = clustal_alignment_file.readlines()\n",
    "                                    print(\"\".join(clustal_alignment))\n",
    "                                # assert False\n",
    "                    while True:\n",
    "                        try:\n",
    "                            species_number = (input(f\"Choose Species 1 - {len(query_species_list)}\"))\n",
    "                            \n",
    "                            \n",
    "                            if species_number[0].lower() == 'n':\n",
    "                                output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                                break \n",
    "                            output += coordinate_detail_dictionary[query_species_list[int(species_number)-1]][exon]\n",
    "                            break\n",
    "                        except:\n",
    "                            print(\"Retry\")\n",
    "                    break\n",
    "                elif choice[0].lower() == \"n\":\n",
    "                    output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                    break\n",
    "                \n",
    "                \n",
    "        else:\n",
    "            print(exon, \"Match\")\n",
    "            print(coordinates_list)\n",
    "            output += coordinate_detail_dictionary[query_species_list[0]][exon]\n",
    "        # assert False\n",
    "    # print(coordinate_detail_dictionary)\n",
    "    print(output)\n",
    "    with open(f\"{blast_location}/{species}/final_coordinates.csv\", \"w\") as out_file:\n",
    "        out_file.write(output)\n",
    "    \n",
    "    # 'Coenonympha_glycerion,OY979626.1,7015875,7015963,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7892a6a6-9400-414d-bbaa-cf893036e0d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
