{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3dfd4b8a-39f1-4f27-b6c3-57a9b7c41468",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Bicyclus_anynana', 'Brenthis_ino', 'Danaus_plexippus', 'Maniola_hyperantus', 'Maniola_jurtina', 'Melitaea_cinxia', 'Nymphalis_io', 'Pararge_aegeria', 'Vanessa_atalanta', 'Vanessa_cardui', 'Vanessa_tameamea', 'Argynnis_bischoffii_washingtonia', 'Boloria_euphrosyne', 'Boloria_selene', 'Brenthis_daphne', 'Brenthis_hecate', 'Charaxes_varanes', 'Coenonympha_glycerion', 'Danaus_chrysippus', 'Dircenna_loreta', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Elymnias_hypermnestra', 'Erebia_aethiops', 'Erebia_ligea', 'Eueides_isabella', 'Fabriciana_adippe', 'Heliconius_charithonia', 'Heliconius_nattereri', 'Heliconius_sara', 'Hestina_assimilis', 'Hipparchia_semele', 'Hypolimnas_misippus', 'Lasiommata_megera', 'Limenitis_camilla', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Melanargia_galathea', 'Melinaea_marsaeus_rileyi', 'Melinaea_menophilus', 'Mellicta_athalia', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Nymphalis_c', 'Nymphalis_polychloros', 'Nymphalis_urticae', 'Oeneis_ivallda', 'Philaethria_dido', 'Euphydryas_editha']\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/\"\n",
    "with open(f\"{location}/Documentation/List of Species/species_family.csv\", 'r') as species_family_file:\n",
    "    species_family_list = species_family_file.readlines()\n",
    "species_list = []\n",
    "for species_family in species_family_list:\n",
    "    family,species = species_family.strip().split(\",\")\n",
    "    if family == \"Nymphalidae\":\n",
    "        species_list.append(species.split(\"-\")[0])\n",
    "        \n",
    "print(species_list)\n",
    "fasta_file = SeqIO.parse(f\"{location}/8.For Phylogeny/2.Tree/combined_aligned.fas\", 'fasta')\n",
    "output = ''\n",
    "for records in fasta_file:\n",
    "    if records.id.split(\"-\")[0] in species_list:\n",
    "        output += f\">{records.id.split('-')[0]}\\n{records.seq}\\n\"\n",
    "\n",
    "with open(f\"{location}/19.Evolutionary_rate_co-variation(ERC)/2.For_base_tree_topology/Nymphalidae_fasta.fas\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f253acf8-60ba-433f-a7bc-dd52e755d12b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Erebia_aethiops', 'Erebia_ligea', 'Maniola_hyperantus', 'Maniola_jurtina', 'Melanargia_galathea', 'Hipparchia_semele', 'Oeneis_ivallda', 'Bicyclus_anynana', 'Lasiommata_megera', 'Pararge_aegeria', 'Coenonympha_glycerion', 'Danaus_chrysippus', 'Melinaea_marsaeus_rileyi', 'Melinaea_menophilus', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Dircenna_loreta', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Limenitis_camilla', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Philaethria_dido', 'Eueides_isabella', 'Heliconius_charithonia', 'Heliconius_nattereri', 'Brenthis_daphne', 'Brenthis_ino', 'Brenthis_hecate', 'Argynnis_bischoffii_washingtonia', 'Fabriciana_adippe', 'Boloria_euphrosyne', 'Boloria_selene', 'Hestina_assimilis', 'Melitaea_cinxia', 'Mellicta_athalia', 'Euphydryas_editha', 'Hypolimnas_misippus', 'Vanessa_atalanta', 'Vanessa_tameamea', 'Vanessa_cardui', 'Nymphalis_c', 'Nymphalis_polychloros', 'Nymphalis_io', 'Nymphalis_urticae']\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import Phylo\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/\"\n",
    "gene_folder_list = [\"6.Period Exon Analysis\",\"7.Timeless Exon Analysis\",\"9.CLOCK Exon Analysis\",\"10.Cycle_Exon_Analysis\",\"11.CWO_Exon_Analysis\",\"12.Cryptochrome_1a_Exon_Analysis\",\"13.Cryptochrome_1b_Exon_Analysis\",\"14.Cryptochrome_2_Exon_Analysis\",\"15.Timeout_Exon_Analysis\"]\n",
    "gene_list = [\"period\",\"timeless\",\"CLOCK\",\"Cycle\",\"CWO\",\"cry1a\",\"cry1b\",\"cry2\",\"timeout\"]\n",
    "\n",
    "print(tree_species_list)\n",
    "for gene_folder, gene in zip(gene_folder_list,gene_list):\n",
    "    tree = Phylo.read(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/2.For_base_tree_topology/Nymphalidae_fasta_just_topology.nwk\", 'newick')\n",
    "\n",
    "    tree_species_list = [term.name for term in tree.get_terminals()]\n",
    "    gene_species_list = []\n",
    "    location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/{gene_folder}/12.Family_specific_tests_{gene}/Nymphalidae\"\n",
    "    fasta_file = SeqIO.parse(f\"{location}/Nymphalidae_{gene}_nostop.fas.aligned.fas.best.fas\",  \"fasta\")\n",
    "\n",
    "    for records in fasta_file:\n",
    "        gene_species_list.append(records.id.split(\"-\")[0])\n",
    "    for species in tree_species_list:\n",
    "        if species not in gene_species_list:\n",
    "            tree.prune(species)\n",
    "    tree_species_list = [term.name for term in tree.get_terminals()]\n",
    "    fasta_file = SeqIO.parse(f\"{location}/Nymphalidae_{gene}_nostop.fas.aligned.fas.best.fas\",  \"fasta\")\n",
    "    fasta_output = ''\n",
    "    for records in fasta_file:\n",
    "        if records.id.split(\"-\")[0] in tree_species_list:\n",
    "            fasta_output += f\">{records.id.split('-')[0]}\\n{records.seq}\\n\"\n",
    "            # print(records.id.split(\"-\")[0])\n",
    "    with open(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/3.CodeML_branch_length_optimization/alignments/{gene}_align.fas\",'w') as out_file:\n",
    "        out_file.write(fasta_output)\n",
    "    Phylo.write(tree, f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/3.CodeML_branch_length_optimization/trees/{gene}_tree.nwk\", \"newick\")    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2307dc90-c5d6-4812-8d57-b186c0194208",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Bicyclus_anynana', 'Lasiommata_megera', 'Pararge_aegeria', 'Coenonympha_glycerion', 'Erebia_aethiops', 'Erebia_ligea', 'Maniola_hyperantus', 'Maniola_jurtina', 'Hipparchia_semele', 'Oeneis_ivallda', 'Melanargia_galathea', 'Euphydryas_editha', 'Melitaea_cinxia', 'Mellicta_athalia', 'Hypolimnas_misippus', 'Nymphalis_c', 'Nymphalis_polychloros', 'Nymphalis_io', 'Nymphalis_urticae', 'Vanessa_atalanta', 'Vanessa_tameamea', 'Vanessa_cardui', 'Eueides_isabella', 'Heliconius_charithonia', 'Heliconius_nattereri', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Philaethria_dido', 'Argynnis_bischoffii_washingtonia', 'Fabriciana_adippe', 'Brenthis_daphne', 'Brenthis_ino', 'Brenthis_hecate', 'Boloria_euphrosyne', 'Boloria_selene', 'Danaus_chrysippus', 'Dircenna_loreta', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Melinaea_marsaeus_rileyi', 'Melinaea_menophilus', 'Hestina_assimilis', 'Limenitis_camilla']\n",
      "['Bicyclus_anynana', 'Lasiommata_megera', 'Pararge_aegeria', 'Coenonympha_glycerion', 'Erebia_aethiops', 'Erebia_ligea', 'Maniola_hyperantus', 'Maniola_jurtina', 'Hipparchia_semele', 'Oeneis_ivallda', 'Melanargia_galathea', 'Euphydryas_editha', 'Melitaea_cinxia', 'Mellicta_athalia', 'Hypolimnas_misippus', 'Nymphalis_c', 'Nymphalis_polychloros', 'Nymphalis_io', 'Nymphalis_urticae', 'Vanessa_atalanta', 'Vanessa_tameamea', 'Vanessa_cardui', 'Eueides_isabella', 'Heliconius_charithonia', 'Heliconius_nattereri', 'Dryadula_phaetusa', 'Dryas_iulia_moderata', 'Philaethria_dido', 'Argynnis_bischoffii_washingtonia', 'Fabriciana_adippe', 'Brenthis_daphne', 'Brenthis_ino', 'Brenthis_hecate', 'Boloria_euphrosyne', 'Boloria_selene', 'Dircenna_loreta', 'Napeogenes_inachia', 'Napeogenes_sylphis', 'Mechanitis_mazaeus', 'Mechanitis_messenoides', 'Melinaea_marsaeus_rileyi', 'Melinaea_menophilus']\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/3.hyphy_branch_length_optimization/alignments\"\n",
    "list_of_files = os.listdir(location)\n",
    "if \"desktop.ini\" in list_of_files:\n",
    "    list_of_files.remove(\"desktop.ini\")\n",
    "species_list = []\n",
    "for files in list_of_files:\n",
    "    fasta_file = SeqIO.parse(f\"{location}/{files}\", 'fasta')\n",
    "\n",
    "    for record in fasta_file:\n",
    "        if record.id not in species_list:\n",
    "            species_list.append(record.id)\n",
    "\n",
    "print(species_list)\n",
    "for files in list_of_files:\n",
    "    fasta_file = SeqIO.parse(f\"{location}/{files}\", 'fasta')\n",
    "    species_in_fasta_file = []\n",
    "    for record in fasta_file:\n",
    "        species_in_fasta_file.append(record.id)\n",
    "    for species in species_list:\n",
    "        if species not in species_in_fasta_file:\n",
    "            species_list.remove(species)\n",
    "\n",
    "print(species_list)\n",
    "\n",
    "\n",
    "for files in list_of_files:\n",
    "    output = ''\n",
    "    fasta_file = SeqIO.parse(f\"{location}/{files}\", 'fasta')\n",
    "    for record in fasta_file:\n",
    "        if record.id in species_list:\n",
    "            output += f\">{record.id}\\n{record.seq}\\n\"\n",
    "\n",
    "    with open(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/4.HYPHY_branch_length_optimization/{files}\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "\n",
    "combined_fasta_file = SeqIO.parse(\"/mnt/g/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/2.Tree/combined_aligned.fas\",'fasta')\n",
    "output = ''\n",
    "for records in combined_fasta_file:\n",
    "    if records.id in species_list:\n",
    "        output += f\">{records.id}\\n{records.seq}\\n\"\n",
    "with open(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/19.Evolutionary_rate_co-variation(ERC)/4.HYPHY_branch_length_optimization/for_tree.fas\", 'w') as out_file:\n",
    "    out_file.write(output)        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8929cb5-0a62-472b-a2bf-0462effa07c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta_file = "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
