{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2b2fddd4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "GCF_947172395.1_ilBicAnyn1.1_genomic.fna\n",
      "8759179 8759216\n",
      "0 2\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MEDDGDDKDDFK\n",
      ">Bicyclus_anynana_Exon_1_NC_069110.1_8759179_8759216_left_0_right_2\n",
      "ATGGAGGACGATGGGGATGACAAAGATGATTTCAAAAG\n",
      "8758140 8758288\n",
      "1 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_2_NC_069110.1_8758140_8758288_left_1_right_1\n",
      "gAGGACACGTAATCTTAGTGAAAAGAAAAGACGTGATCAATTTAATACGCTTGTTCATGAACTGGGAGCAATGATCTCATCAAATAACAGAAAAATGGATAAATCAACAATTCTCCGAACAACCATATACTTCTTAAAGAATCATAATG\n",
      "8757713 8757894\n",
      "2 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_3_NC_069110.1_8757713_8757894_left_2_right_0\n",
      "AAATAACAGCACGAGCAAGAGCCCACGATGTGCAAGACGATTGGAAACCTGCTTTCTTGACTAACGAAGAGTTCACATATTTACTACTTGAGGCTTTGGAGGGTTTTGTCATAGTATTTTCAGCAAATGGACGCATCATTTATGTCTCTGAAAGTGTTGCATCTCTATTAGGCCATAACCcg\n",
      "8757213 8757330\n",
      "0 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_4_NC_069110.1_8757213_8757330_left_0_right_1\n",
      "aatgatattattaataaaaatatatatgatttaGTATATAAAGATGATCGCTGTAATCTTTATGGATTACTACAAAATCCTGATACATCAGTGCATGCAGCGAAAAACGACAATGCAG\n",
      "8753106 8753201\n",
      "2 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_5_NC_069110.1_8753106_8753201_left_2_right_1\n",
      "acAATGAGATAAATTTCCAATGTCTCATGCGTAGAGGTACGCTGGATTTCCGTGATGACGTCAGCTACGAACTTATTCAATTTAATGGATACTTCC\n",
      "8751939 8751999\n",
      "2 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_6_NC_069110.1_8751939_8751999_left_2_right_2\n",
      "GTACTGACATAGAGCAGGATGAAAACTGCGATTTAACTTTTGAAGATCCACATGATTCCAg\n",
      "8748753 8748991\n",
      "1 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_7_NC_069110.1_8748753_8748991_left_1_right_1\n",
      "GGTTGTTTACTTCTGTACGGCCAAGCTGGGCACTCCTCAACTAATCCGGGATGTATCGCTTGTGGATCCAGATCGTAACGAATTTGTTTCACGACATAGTTTAGAATGGAAATTTCTCTTTCTGGACCACCGCGCCCCACCAATCATTGGTTACCTGCCATTTGAAGTGCTCGGCACATCGGGCTATGATTATTACCATTTTGACGATCTTGAGAAAGTAATTACGTGCCATGAAGCAT\n",
      "8747915 8748062\n",
      "2 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_8_NC_069110.1_8747915_8748062_left_2_right_2\n",
      "TGATGCAAAAGGGGGAGTTAACATCCTGCTACTACAGATTTCTAACAAAGGGTCAGCAATGGATTTGGCTTCAGTCACGCTTCTATATAACTTATCACCAGTGGCATTCCAAGCCCGAGTTTGTTGTATGCACTCATCGTGTGGTTAG\n",
      "8747496 8747673\n",
      "1 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_9_NC_069110.1_8747496_8747673_left_1_right_0\n",
      "CTATGTCGATGTACAGACACCCACAAAGCAAGAAAATATAGAGCAAGACACTAGCACTGATTGTGAACCAAGCCACAGTGGTTTAAAGGAGCCTTTATCAGAGGAGGCTACAGCAGTATCACCTTCTTACATGTCCGAATCCAGTGATGCTTACGGGAATTGCATCTCACAGATACAG\n",
      "8746856 8747050\n",
      "0 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_10_NC_069110.1_8746856_8747050_left_0_right_0\n",
      "CAGCAATCACAATCAGTGAAGTCAATGCCAACTTCAGCGGACAGCGTGTGTGCAACGGTGGCCTCCGCTGGCACGCCCGTCACTCCCGCGGCCAGCTGGCCCTCTCGCTCGCCGTACCAGCCCCACGGCGTCGGCTCCGACACCGGCTCCGTGTCCGGCGACTCGCGGTCGTCGCAGCGCAACAGTTCTCAAGAT\n",
      "8746134 8746295\n",
      "0 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_11_NC_069110.1_8746134_8746295_left_0_right_0\n",
      "ACACAAAATGTGCATGAGGCAACACTCGTTCCGCAGCACAATATTGGAGCCCAGTATTTGGAGCCAGTTCCGTATGTAGGCCCTGTCAGTATGCCTGGTGTACTTCCTATGTCTCTGCCCCCGCTACCGGTCATCGTATCACCGAATCAGGCGCAACTACAG\n",
      "8742083 8742205\n",
      "0 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_12_NC_069110.1_8742083_8742205_left_0_right_0\n",
      "GAGCAGTTGCAGCGCACGCACCGCGAGCTGCAGCACATGATATTCCGCCAACAGGAGGAGCTGCGGTTGGTCAAGGAGCAGCTGATGCTGGCACGGCTCGGCATTTTTCAGCCCATGATTAGT\n",
      "8738842 8740872\n",
      "0 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_13_NC_069110.1_8738842_8740872_left_0_right_0\n",
      "GTTCAAGATCCTTATGGCCATAATGAAGAAATATTGCATGTTCCGCGCATTCCACCACAAATAGTATACGACGGGACACATCCAAGGGCCCTTGGCGGCTACACGCCGCATCAGCTACACCAGCCGCAGCCGCTGCAGTTACCTCAGGCACAGGCTCTACAGGCGCAAGCGCTTCAGGCACAAGCGCACCAGGCGCAGGCGCAACAGCACCAGTCAGCAGTGCACGAGTCGCAGATGCTTCGCTCGCCAGTGCATCAGCCAAACACTTCACAGTCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGGAACAGTCACAGATGCAGCTATCGCCAGTGCAACAGTCACAGATGAAGTTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAAATGAAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGGAACAGTCACAGATGCAGCTATCGCCAGTGGAACAGTCACAGATGCAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGGAACAGTCACAGATGCAGCTATCGCCAGTGCAACAGTCACAGATGCAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAGATGCAGCTATCGCCAGTGCAACAGTCACAGATGAAGCTATCGCCAGTGCAACAGTCACAGATGCAGCTATCGCCAGTGCAACAATCACAGACGCAGCTATCGCCAGTGCAACAGTCACAGACACCTCAGGTGCAAACTTTTCAGGTGCAACTGCACCCAGAGCAGTCTCATCCTTTGCCATTACACCAGTCGCAGGCTCCTCAGGCTCAGACGCTACAGGTGCAGGCGCCCCAGTTGAAGGCTCATCACTCATCAGTACCAAAGGCACAGGTGCCTCAGGCTCAAGCTCTTCAGATTCAAGGGCACCATGTGCCGACTCATCGTACTCCAGTGCAACAGGCTCAGGCGATTCAGGTGCAAGCGCACCAAGTTCAGGCGCTTCAGGCACAAGCACATCAGGTGCAGTCGCTTCAGGCACAAGCAAACCAGGTGCAAGCGCTTCAGGCACAAGCCCATCACTCGCAAGGGCGCCAGGTGCAAACGCTTCAGGCACAAGCACATCACTCGTCAGGGCGCCAGGTGCAAACGCTTCAGGCACAAGCCCATCACTCGTCAGGGCGCCAAGCGCAAGCTCCTCAGGTACAGACGCTGCAGGCGCCGTCACACCAGGCGCACCTGCTGCAGGCACAAGGGCACCAGGCGCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCACACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACTCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCTTCAGTCCCAGATGCAGCAGTCACAGACGCGTCAGAGTCAGGCGCACCAGTCCCAGATGCACCAAGTGCATGCGCATCCGCTCCAGCCGCAGTCGGACCACCCCGCGCGTGGAGCCCAGAACCAGTACAATCCAAATATGTCTCGTGCTCCACAGTAA\n",
      "Coenonympha_glycerion\n",
      "GCA_963855885.1_ilCoeGlyc1.1_genomic.fna\n",
      "8028554 8028591\n",
      "0 2\n",
      "Getting Gene\n",
      "MEEDGEDKDDSK\n",
      ">Coenonympha_glycerion_Exon_1_OY979626.1_8028554_8028591_left_0_right_2\n",
      "ATGGAAGAAGACGGCGAAGATAAAGATGACTCCAAAAG\n",
      "8026636 8026784\n",
      "1 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_2_OY979626.1_8026636_8026784_left_1_right_1\n",
      "GAAGACACGTAACCTAAGTGAGAAGAAGCGCAGAGACCAATTTAATATGCTCATTAATGAACTGGGCTCAATGGTTACAACCAATAATAGGAAGATGGACAAATCTACTGTCCTGAGGTCTACTATATCTTTTTTGAAGAACCACAATg\n",
      "8025728 8025909\n",
      "2 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_3_OY979626.1_8025728_8025909_left_2_right_0\n",
      "AAACAACAGTAAGATCAAGAGCCCATGACGTGCAAGAAGATTGGAAGCCTGGTTTCTTGTCCAATGAAGAATTTACATATCTGTTGCTTGAAGCTCTCGAGTGTTTCATTATGGTGTTTTCAGCAAGTGGCTACATCTATTATGTGTCTGAAAGTGTCACATCATTATTGGGCCACAATCCG\n",
      "8024708 8024825\n",
      "0 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_4_OY979626.1_8024708_8024825_left_0_right_1\n",
      "aatgaaattttgaataaaagtatttttgatCTGGCATATGAGGAGGATCGCGCCAACATTTTAGAGATACTTCAAAAACCAGCTGCTTCAGTACATACCACATCGGGAATCAATTCAg\n",
      "8021461 8021556\n",
      "2 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_5_OY979626.1_8021461_8021556_left_2_right_1\n",
      "ATAACGAGATAATATTCCAGTGTCACCTACGAAGAGGCACGCTTGATCTTCAAGATGTAATCACTTATGAACTCGTACAATTCAATGGCTGTTTCC\n",
      "8020296 8020356\n",
      "2 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_6_OY979626.1_8020296_8020356_left_2_right_2\n",
      "GTACTAACACCGAACTGAGTGAAAGCTGTGATCTATCTTTTCAACAAGATCAAGAATCAAG\n",
      "8019257 8019495\n",
      "1 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_7_OY979626.1_8019257_8019495_left_1_right_1\n",
      "GCTACTTTTCGTCTGTACCGGCAAGCTGTGCACTCCGCAGCTCATCAGAGACGTGTCGCTCGTGGACCCCAGTCGCAACGAGTTCACGTCGCGGCACAGTTTGGAGTGGAAGTTCCTTTTCCTGGACCACCGCGCGCCGCCCATCATCGGGTACCTACCCTTCGAAGTGTTAGGCACGTCGGGCTATGATTACTATCACTTTGATGACCTGGAGAAGGTCGTCATGTGCCACGAAGCAT\n",
      "8018430 8018577\n",
      "2 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_8_OY979626.1_8018430_8018577_left_2_right_2\n",
      "TGATGCAGAAGGGCGAAGTGACGTCATGCTACTACAGGTTTCTAACTAAGGGGCAGCAGTGGATATGGCTTCAAACGCGTTTCTATATAACTTACCACCAGTGGAATTCCAAACCGGAGTTTGTTGTATGCACTCATCGAGTTGTAAG\n",
      "8017170 8017350\n",
      "1 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_9_OY979626.1_8017170_8017350_left_1_right_0\n",
      "TTATACTGACGTGGATGCTAGAACAAAGCAAGAGGCCGTAGAGCAAGACGCAAAGAGCGACGCTGAGCCGGGCGTGAGTGGACTCAACGAGCCGTCCCCCGATCGCCTCGCCGCCGTGTCGCCATCTTACATGTCAGATAACAGTGAAGCTTTCGGCAACTCCTATCATTCTGTTTCACAG\n",
      "8016337 8016525\n",
      "0 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_10_OY979626.1_8016337_8016525_left_0_right_0\n",
      "CCGCCATCGATAAAGTCGGCCGCGACATCAGCGGGCAGCAcaagcgcggcggcggcgacgggcGGCACGGCGACCACGGCGGCCGCGTCCTGGACGACGCGCCCGTCGCACATCCGCTACGCCGCCGGCTCCGACACCGCCTCCGTCTCGGGCGAGTCGCGCTCCTCGCAGTGGAACTCCTCGCAACAA\n",
      "8014779 8014952\n",
      "0 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_11_OY979626.1_8014779_8014952_left_0_right_0\n",
      "CGGTGTGTTTGGTCGCAGGCGCTGCGGCTGCCGGAGCTGGTGCCGCAGCACGGCATCGGCGCGCAGTACCTGGAGCCCGTGCCGTACGTGAGCGCGGTGGGCGTGCCCGGCGTGCTGCCGCTGACGCTGCCCGCGCTGCCCGTCATCGTCTCGCCGGACCAGGCGCAGCTGCAG\n",
      "8014015 8014137\n",
      "0 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_12_OY979626.1_8014015_8014137_left_0_right_0\n",
      "GAGCAGCTGCAGCGCACGCACCGCGAGCTGCAGGAGCTGATCGCGCGGCAGCAGGAGGAGCTGCGCCAGGTGAAGGAGCAGCTGCTGCTGGCGCGCCTGGGCATCCTCCAGCCCATCATTAAC\n",
      "8008322 8008531\n",
      "0 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_13_OY979626.1_8008322_8008531_left_0_right_0\n",
      "gTGCAAGATCCGTACGTGCAAACAGCGGAGTTGCAGCAGAGTCCCCGTCTCCCAGGACAGTTAGTGTACGAGGGTGGTCAACCGCGGCCCATGGCGGGGTATCCGCTCTCGCCGCACCAACCGCACCAACCGCACCAACCGCACCAACCACCGTCTGACCACCCGGCGCACTACTCCCAGAACCAGCTGCACCCGAACATGCCTCAGTAG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Elymnias_hypermnestra\n",
      "GCA_015832385.1_UofC_Ehyp_1.0_genomic.fna\n",
      "960613 960650\n",
      "0 2\n",
      "Getting Gene\n",
      "MEDDGDDKDDSK\n",
      ">Elymnias_hypermnestra_Exon_1_JACTOM010000009.1_960613_960650_left_0_right_2\n",
      "ATGGAGGACGATGGAGACGACAAAGATGACTCTAAAAG\n",
      "959018 959166\n",
      "1 1\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_2_JACTOM010000009.1_959018_959166_left_1_right_1\n",
      "GAGGACTCGTAACCTAAGTGAGAAAAGGCGAAGGGATCAATTTAATATGCTTGTCAATGAGCTTGGTACAATGGTTACAGTAGGTAACAGAAAAATGGACAAATCTACAGTCCTCAAATCTACCATAGCATTTCTAAGGCATCACAATG\n",
      "958752 958933\n",
      "2 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_3_JACTOM010000009.1_958752_958933_left_2_right_0\n",
      "aAATAACTATGCGATCAAGAACATATGATGTACAAGAAAACTGGAAACCAGCTTTCCTTTCAAATGAGGAGTTTACATATTTAGTACTTGAAGCACTTGAAGGATTTGTAATGGTGTTTTCTGTAACTGGTAGCATATATTATGTATCTGAAAGTATTACATCACTATTAGGCCACAATTCG\n",
      "958181 958298\n",
      "0 1\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_4_JACTOM010000009.1_958181_958298_left_0_right_1\n",
      "AATGATATAGTAAACAAAAGTATATTTGACTTGATATTTGAAGAGGAGAGGCCTAGTTTGTTACAACTCCTCCAAAATTCTAATTCTTTCATTAATTCCTCAGAATGGTTTCTTGCag\n",
      "955439 955534\n",
      "2 1\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_5_JACTOM010000009.1_955439_955534_left_2_right_1\n",
      "AGAACGAGATTAGGTTCCAGATTCACGTGCAAAGGGGTACACTTGAAGTCCGAGATGAGGTCACATACGAACTGATTCAGTTTAGTGGGCACTTTC\n",
      "951921 951981\n",
      "2 2\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_6_JACTOM010000009.1_951921_951981_left_2_right_2\n",
      "GTGCTTATACAGAGCCAAAGGAAAACTACGATGCGTTATTTGAACAAGAACCAGAATGcag\n",
      "949934 950172\n",
      "1 1\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_7_JACTOM010000009.1_949934_950172_left_1_right_1\n",
      "ACTACTTTTCGTGTGCACCGGAAAGTTATGCACACCTCAATTGATAAGAGACGTGTCGCTTTTGGATCCTAACCGTAACGAGTTCACATCTAGACATAGCTTAGAgtggaaatttttatttctggACCATCGTGCTCCCCCGATAATCGGTTATTTACCCTTTGAAGTGCTGGGAACATCGGGATACGATTATTATCATTTCGATGACCTAGAGAAAGTTGTCACTTGCCACGAAGCAT\n",
      "948837 948984\n",
      "2 2\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_8_JACTOM010000009.1_948837_948984_left_2_right_2\n",
      "TGATGCAAAAGGGCGAATTAACATCGTGCTACTACAGGTTTCTTACTAAGGGACAGCAATGGATTTGGCTACAGACTCGATTCTATATAACTTACCACCAATGGAATTCTAAGCCAGAATTTGTAGTATGCACACATCGTGTCGTTAG\n",
      "947012 947198\n",
      "1 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_9_JACTOM010000009.1_947012_947198_left_1_right_0\n",
      "TTATGCTGAAATAGATAGGAATGCGAAGCAAGAGGCCGCAGAGGCGGAAACTCAGAGTGAACCAGAGCCTAAACAGAGCACACTGGAAGAGCCTCCGTCCGAGGAGGCAGTGGTGGCAGCGTCACCATCGTATATGTCTGAAGCCGACGCTGACGCCTTCACCAACTCCTACCATTCCATATCACAG\n",
      "945889 946077\n",
      "0 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_10_JACTOM010000009.1_945889_946077_left_0_right_0\n",
      "CCTGCATCTGTAAATTCACCACCGACATCAGTTGGTAGTAATAGTGCCACAGTGGCCACAGTTGAGACGGCTGCTACAACGGTCAGTTCTTGGCCGACACAGGCATCAGATTTACGATACTCCGCCGGATCCGATACAGCCTCCGTATCAAGAGAATCTCGATCACCTCAACAGAATAACTCACAAGAG\n",
      "939785 939940\n",
      "0 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_11_JACTOM010000009.1_939785_939940_left_0_right_0\n",
      "ACAGAGAGATCGCCAGAGCCGGCACTTGTACCCCAACACGTCATTGGGGCACAGTATTTAGAGCCAGTACCTTACGTCGGGCCAGTTAGTGTCCCGAGTGTGTTGCCGCTACCTCTGCCGTCGTTACCGGTCATCGTTGATCAGGCGCAGTTACAG\n",
      "937244 937366\n",
      "0 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_12_JACTOM010000009.1_937244_937366_left_0_right_0\n",
      "GAGCAGCTGCAGCGAACCCATCATGAATTACAGCAGATGATAGTGCGACAGCAGGAAGAGCTACGCCAGATGAAGGAACAACTGCTGTTCGCTCGCCTTGGAATTTATCCGCCCGTTATTAAC\n",
      "935910 936413\n",
      "0 0\n",
      "Getting Gene\n",
      ">Elymnias_hypermnestra_Exon_13_JACTOM010000009.1_935910_936413_left_0_right_0\n",
      "GTGCAAGATCCATATGGACGCATTCAACAAATACATCAAAACAATCCACTTATTCAAGCACAAACTATTTATGATGGCACACAGATGATACCTTTGGCTGGTTATCCACAGCAGCAATTGCCACCGATGCAAGATAACCTACACTATGCCTCCCGTCAAAACCCAAGCCGTGCCTCGCGTCAAAACCTAAATCACGCCATGTATCAAAACCCGAATCATGTCTCGCAACAAAACCCAAATCAAACCTCTCATCAAAACCCAAATCATGCCTCGCATCCAAACCTAAATCACGCCTCGCGTCAAAACCCAAATCAAGTCTCGCGTCAAAACCCAAATCACGCTTCGCGATCTGACCCAAATCACGCTACTCGACAAAACCCAAATCACCCCTCGCGTCAAAACCCAAATGACGCCTCGCGTCAGAATCCAAATCATGCATCGCGTCAAAATCCACATCGGAgccaaaatcaaaatcagtacCACCAAAACATGCCCCCTCAATAA\n",
      "Erebia_aethiops\n",
      "GCA_923060345.2_ilEreAeth2.2_genomic.fna\n",
      "8494672 8494709\n",
      "0 2\n",
      "Getting Gene\n",
      "MDDDGDDKDDSK\n",
      ">Erebia_aethiops_Exon_1_OV281079.1_8494672_8494709_left_0_right_2\n",
      "ATGGACGACGACGGAGACGACAAAGATGACTCCAAAag\n",
      "8492961 8493109\n",
      "1 1\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_2_OV281079.1_8492961_8493109_left_1_right_1\n",
      "GAAGACACGTAACCTTAGTGAGAAGAAGCGTCGAGATCAATTCAATATGCTTGTTAATGAGCTGGGAACAATGATCTCTACAAATAATCGGAAAATGGATAAATCAACTATACTCAAATCTACTATCTCCTTTTTAAAGGAGCACAATG\n",
      "8492139 8492320\n",
      "2 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_3_OV281079.1_8492139_8492320_left_2_right_0\n",
      "aaataaCAGTACGGTCAAGAGTCCATGATGTACAAGAAGACTGGAAGCCATCTTTCTTGTCAAATGAAGAATTTACATATTTAGTACTTGAGGCGCTGGAGGGATTTGTTATGGTGTTTTCGGCTACTGGTCGCATCTATTATGTGTCTGAAAGTGTAACATCTCTGTTGGGTCACAATCCG\n",
      "8491523 8491640\n",
      "0 1\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_4_OV281079.1_8491523_8491640_left_0_right_1\n",
      "aatGGCATAGtgaataaaagtatatttgattTGGTATATGAGAAGGATCACCATAATCTCTATGAATTCTTTCAAAATCCTGGTGCATCTGTGCATTCAGCACAAGGGATTAATACAG\n",
      "8489614 8489709\n",
      "2 1\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_5_OV281079.1_8489614_8489709_left_2_right_1\n",
      "AGAACGCCATAAGCTTTCAGTGTCACCTAAAAAGAGGAGCGCTTGATTTACGAGACGAGGTTACATACGAACTTGTTCAGTTTAACGGCCATTTCC\n",
      "8489309 8489366\n",
      "2 2\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_6_OV281079.1_8489309_8489366_left_2_right_2\n",
      "GTACGAATACAGAACAAAACGAAGCCGATTTATCCTTTCAACAAGACCAAGAATCCAg\n",
      "8488382 8488620\n",
      "1 1\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_7_OV281079.1_8488382_8488620_left_1_right_1\n",
      "ACTACTTTTCGTCTGTACTGGCAAGCTGTGTACGCCACAACTAATCCGGGATGTGTCACTTGTGGACCCCAACCGCAACGAGTTCACATCACGCCACAGCCTCGAATGGAAGTTCCTTTTCCTGGACCATCGTGCTTCTCCCATAATCGGTTACCTACCCTTTGAAGTGCTGGGCACATCCGGATATGATTACTATCATTTTGACGATCTAGAGAAAGTTGTTACTTGTCATGAAGCAT\n",
      "8486522 8486669\n",
      "2 2\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_8_OV281079.1_8486522_8486669_left_2_right_2\n",
      "TGATGCAAAAGGGAGAGTTGACTTCCTGCTACTACAGGTTTCTAACAAAGGGACAGCAATGGATTTGGCTCCAAACACGCTTCTATATAACATACCACCAGTGGAATTCTAAACCAGAGTTCGTTGTGTGTACACATCGTGTTGTTAG\n",
      "8485777 8485957\n",
      "1 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_9_OV281079.1_8485777_8485957_left_1_right_0\n",
      "TTATGCTGACATAGATAAGAGCACAAAGCAAGAAAATATAGAACAAGACGCTATAAGTGAAGCAGAGCCAAACCTAGGCGGCGTAAAGGAACCGTCACCAGAGGCTATACCTGCTGTTTCACCTTCTTACATGTCTGAAGCCAGTGAAGTCTTTGGGAACTGTTATCATTCACTATCACAG\n",
      "8485158 8485346\n",
      "0 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_10_OV281079.1_8485158_8485346_left_0_right_0\n",
      "CCACCATCGGTGAAGTCCATAGCAACATCGACTGGCAGCACAAGTGCGACGGTCGCCACTGTTGGCACGGCGACTACAGCGGCCACGTCTTGGCCGACGCGCGCCTCGCACATTCGGTACGCGACCGGCTCAGACACCGCGTCAGTCTCGGGCGACTCGCGATCGTCGCAGCGGAACTGCACCCAAGAA\n",
      "8482913 8483068\n",
      "0 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_11_OV281079.1_8482913_8483068_left_0_right_0\n",
      "AGCGTTCAAGAGCCTTTGCTGGTGCCGCAGCACGGTATCGGAGCTCAGTACTTGGAGCCGGTTCCGTACGTAGGTACAGTCGGCATGCCTGGAGTATTGCCTCTGCCCTTGCAGTCACTTCCGGTCATCGTGTCGCCCGACCAAGCGCAGTTACAG\n",
      "8480591 8480713\n",
      "0 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_12_OV281079.1_8480591_8480713_left_0_right_0\n",
      "GAGCAGCTGCAGCGCACCCATCGTGAGCTGCAGCAGATGATAGTGCGGCAACAGGAGGAACTGCGCCAAGTCAAGGAGCAACTGTTGCTTGCACGGCTCGGCATCTTGCAGCCCATCATTAAT\n",
      "8477430 8477612\n",
      "0 0\n",
      "Getting Gene\n",
      ">Erebia_aethiops_Exon_13_OV281079.1_8477430_8477612_left_0_right_0\n",
      "GTTCAAGATCCATACGGACATAGTGAGGAGCATCACCAAAACCCACCACGCGTTCCAGCACAAATTCTTTACGATGGTACGCAGCCGAGGCCTGTCGCCGGGTAtcccccgccgccgccgccgccgggtCTGCCCGCGCACCCCAGTTCCAACCAGTACAACGTGCAACGTCCTCCGCAGTAA\n",
      "Erebia_ligea\n",
      "GCA_917051295.2_ilEreLige1.2_genomic.fna\n",
      "12334606 12334643\n",
      "0 2\n",
      "Getting Gene\n",
      "MEDDGDDKDDSK\n",
      ">Erebia_ligea_Exon_1_OU785222.1_12334606_12334643_left_0_right_2\n",
      "ATGGAAGATGACGGAGACGACAAAGATGACTCCaaaag\n",
      "12335447 12335595\n",
      "1 1\n",
      "Getting Gene\n",
      ">Erebia_ligea_Exon_2_OU785222.1_12335447_12335595_left_1_right_1\n",
      "GAAGACACGTAACCTTAGTGAGAAGAAGCGTCGAGACCAATTCAATATGCTTGTTAATGAGCTGGGAACAATGGTCTCTACAAATAATCGGAAAATGGATAAATCAACTATACTCAAATCTACCATCTCCTTTTTAAAGGAGCACAATG\n",
      "12336352 12336533\n",
      "2 0\n",
      "Getting Gene\n",
      ">Erebia_ligea_Exon_3_OU785222.1_12336352_12336533_left_2_right_0\n",
      "AAATAACAGTACGATCAAGAGCCCATGATGTACAAGAAGACTGGAAGCCATCTTTCTTGTCAAATGAAGAATTTACATATTTAGTACTTGAGGCGCTGGAGGGATTTATTATGGTGTTTTCAGCAACTGGTCGCATCTATTATGTGTCTGAAAGTGTAACATCTCTGTTGGGCCATAATCCG\n",
      "12337023 12337140\n",
      "0 1\n",
      "Getting Gene\n",
      ">Erebia_ligea_Exon_4_OU785222.1_12337023_12337140_left_0_right_1\n",
      "aatggCTTAGTgaataaaagtatatttgatTTGGTATATGAGGAGGATCACCATAATCTCTTTGAATTCTTTGAAAATCCTAGTGCATCTGTGCATTCTGCACAAGGGATTGATACAG\n",
      "12338611 12338706\n",
      "2 1\n",
      "Getting Gene\n",
      ">Erebia_ligea_Exon_5_OU785222.1_12338611_12338706_left_2_right_1\n",
      "aGAACGCCATAAGCTTCCAGTGTCACCTACAAAGAGGAACGCTTGATTTACGAGACGAGGTTACATACGAACTTGTTCAGTTTAACGGCCATTTCC\n",
      "12338963 12339023\n",
      "2 2\n",
      "Getting Gene\n",
      ">Erebia_ligea_Exon_5_OU785222.1_12338611_12338706_left_2_right_1\n",
      "aGAACGCCATAAGCTTCCAGTGTCACCTACAAAGAGGAACGCTTGATTTACGAGACGAGGTTACATACGAACTTGTTCAGTTTAACGGCCATTTCC\n",
      "12338963 12339023\n",
      "2 2\n",
      "Getting Gene\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"3.Satyrine\"\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}\"\n",
    "make_output_folder(output_location)\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "if 'desktop.ini' in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Heliconius_charithonia\"]\n",
    "# list_of_species = [\"Maniola_hyperantus\"]\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "        \n",
    "            \n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "            print(left_oh, right_oh)\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            query_name = coordinate_line_split[6]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            continue\n",
    "            \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "       \n",
    "        current_exon = query_name.split('_')[-1]\n",
    "        if current_exon == \"1\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "            print(translated)\n",
    "            if \"*\" in translated:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "        \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "        elif current_exon == \"16\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence[:-int(left_oh)].reverse_complement.translate()\n",
    "                print(gene_sequence)\n",
    "            else:\n",
    "                translated = gene_sequence[int(left_oh):].translate()\n",
    "                print(gene_sequence)\n",
    "\n",
    "            if \"*\" in translated[:-1]:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                print(translated)\n",
    "            else:\n",
    "                \n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "#         assert False\n",
    "        else:\n",
    "            exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "            print(exon_output)\n",
    "\n",
    "            write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
