{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d86e485-0c0f-42cb-9693-c1008778bc85",
   "metadata": {},
   "outputs": [],
   "source": [
    "location = \"/mnt/griffin/saubar/busco_runs_CDS_files\"\n",
    "\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "list_of_species = os.listdir(location)\n",
    "list_of_species.remove(\"count_genes.py\")\n",
    "# print(list_of_species)\n",
    "busco_count_dictionary = {}\n",
    "busco_per_gene_count = {}\n",
    "busco_count_dictionary_checked = {}\n",
    "for i,genome_folder in enumerate(list_of_species):\n",
    "    print(i, len(list_of_species))\n",
    "    if genome_folder.endswith(\"_genomic.fna\"):\n",
    "        list_of_busco_genes = os.listdir(f'{location}/{genome_folder}/CDS')\n",
    "        genome_count = 0\n",
    "        for gene_name in list_of_busco_genes:\n",
    "            fasta_sequence = SeqIO.parse(f\"{location}/{genome_folder}/CDS/{gene_name}\", 'fasta')\n",
    "            for records in fasta_sequence:\n",
    "                if \"*\" not in records.seq.translate()[:-1]:\n",
    "                    genome_count += 1\n",
    "            if gene_name not in busco_per_gene_count:\n",
    "                busco_per_gene_count[gene_name] = 0\n",
    "            busco_per_gene_count[gene_name] += 1\n",
    "        busco_count_dictionary[genome_folder] = len(list_of_busco_genes)\n",
    "        busco_count_dictionary_checked[genome_folder] = genome_count\n",
    "output1 = \"Genome,BUSCO_gene\"\n",
    "\n",
    "for key, value in busco_count_dictionary.items():\n",
    "    output1 += f\"\\n{key},{value}\"\n",
    "with open(\"/mnt/griffin/saubar/busco_runs_CDS_files/per_genome_count_without_checking.txt\", 'w') as out_file:\n",
    "    out_file.write(output1)\n",
    "\n",
    "output1 = \"Genome,BUSCO_gene\"\n",
    "\n",
    "for key, value in busco_count_dictionary_checked.items():\n",
    "    output1 += f\"\\n{key},{value}\"\n",
    "with open(\"/mnt/griffin/saubar/busco_runs_CDS_files/per_genome_count_with_checking.txt\", 'w') as out_file:\n",
    "    out_file.write(output1)\n",
    "\n",
    "\n",
    "output1 = \"Gene,Total_count\"\n",
    "\n",
    "for key, value in busco_per_gene_count.items():\n",
    "    output1 += f\"\\n{key},{value}\"\n",
    "with open(\"/mnt/griffin/saubar/busco_runs_CDS_files/gene_count_without_checking.txt\", 'w') as out_file:\n",
    "    out_file.write(output1)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0d9df16-0b34-4089-91e7-fc130397d60e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
