{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0c9e33c0-46ae-4020-aeb4-286f443583a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calycopis_cecrops-period,169-172,211-214,322-325,511-514,562-565,772-775,1600-1603,1717-1789,1996-2002,2101-2104,2194-2200,2209-2215,2290-2296,2323-2338,2371-2380,2404-2407,2701-2707,3166-3172,3178-3187\n",
      "Polyommatus_icarus-period,70-73,163-166,169-172,226-229,322-325,562-565,1351-1354,1588-1591,1600-1603,1693-1711,1813-1855,2194-2200,2278-2281,2290-2293,2323-2338,2365-2368,2371-2380,2404-2407,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Polyommatus_iphigenia-period,70-73,163-166,169-172,211-214,226-229,322-325,562-565,1351-1354,1588-1591,1600-1603,1693-1711,1813-1855,2194-2200,2278-2281,2290-2293,2323-2338,2365-2368,2371-2380,2404-2407,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Aricia_agestis-period,70-73,163-166,169-172,211-214,322-325,562-565,1351-1354,1513-1522,1588-1591,1600-1603,1693-1711,1813-1855,2194-2200,2278-2281,2290-2293,2323-2338,2365-2368,2404-2407,2416-2431,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Aricia_artaxerxes-period,70-73,163-166,169-172,211-214,322-325,562-565,1351-1354,1513-1522,1588-1591,1600-1603,1693-1711,1813-1855,2083-2092,2194-2200,2278-2281,2290-2293,2323-2338,2365-2368,2404-2407,2416-2431,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Cyaniris_semiargus-period,70-73,163-166,169-172,211-214,562-565,1351-1354,1588-1591,1600-1603,1693-1711,1813-1855,2194-2200,2278-2281,2323-2338,2371-2380,2404-2407,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Lysandra_bellargus-period,70-73,163-166,169-172,211-214,445-448,562-565,1351-1354,1588-1591,1693-1711,1813-1855,2194-2200,2278-2281,2290-2293,2323-2338,2371-2380,2404-2407,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Lysandra_coridon-period,70-73,163-166,169-172,211-214,562-565,1351-1354,1588-1591,1693-1711,1813-1855,2194-2200,2278-2281,2290-2293,2323-2338,2371-2380,2404-2407,2437-2443,2512-2515,2566-2569,2581-2590,3178-3187,3193-3204\n",
      "Plebejus_argus-period,70-73,163-166,169-172,211-214,322-325,1351-1354,1588-1591,1600-1603,1693-1711,1813-1855,1891-1894,2194-2200,2290-2293,2371-2380,2404-2407,2512-2515,2530-2533,2566-2569,3166-3172,3178-3187,3193-3204\n",
      "Celastrina_argiolus-period,70-73,163-166,169-172,211-214,322-325,562-565,1351-1354,1588-1591,1600-1603,1684-1690,1693-1714,1777-1783,1813-1855,2290-2293,2320-2341,2371-2380,2422-2434,2512-2515,3166-3172,3193-3204\n",
      "Glaucopsyche_alexis-period,70-73,163-166,211-214,322-325,562-565,664-694,1351-1354,1540-1543,1588-1591,1600-1603,1693-1711,1813-1855,1900-1903,2194-2200,2290-2293,2323-2338,2371-2380,2404-2407,2494-2497,2512-2515,3166-3172,3193-3204\n",
      "Phengaris_arion-period,70-73,163-166,211-214,322-325,562-565,1351-1354,1588-1591,1600-1603,1693-1711,1813-1855,1900-1903,2194-2200,2290-2293,2323-2338,2371-2380,2404-2407,2512-2515,2689-2692,3166-3172,3193-3204\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "family = \"Lycaenidae\"\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/12.Family_specific_tests/\"\n",
    "\n",
    "alignment_file = SeqIO.parse(f\"{location}/{family}/{family}_period_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "output = ''\n",
    "for records in alignment_file:\n",
    "    output += records.id+ \",\"\n",
    "    gap_start = 0\n",
    "    for i in range(len(records.seq)):\n",
    "        if records.seq[i] == \"-\" and gap_start == 0:\n",
    "            output += f\"{i+1}-\"\n",
    "            gap_start = 1\n",
    "        if gap_start == 1 and records.seq[i] != \"-\":\n",
    "            output += f\"{i+1},\"\n",
    "            gap_start = 0\n",
    "    if gap_start == 1:   \n",
    "        output += f\"{i+1}\\n\"\n",
    "    else:\n",
    "        output =output[:-1]+\"\\n\"\n",
    "    \n",
    "print(output)\n",
    "with open(f\"{location}/{family}/05.Summary Figure/gap_position.txt\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cdceb785-7436-4e40-886c-90229ec04365",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calycopis_cecrops-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1817,1818-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Polyommatus_icarus-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Polyommatus_iphigenia-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Aricia_agestis-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Aricia_artaxerxes-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Cyaniris_semiargus-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Lysandra_bellargus-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Lysandra_coridon-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Plebejus_argus-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Celastrina_argiolus-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Glaucopsyche_alexis-period,1-92,93-173,174-352,353-450,451-558,559-663,664-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "Phengaris_arion-period,1-92,93-173,174-352,353-450,451-558,559-693,694-754,755-952,953-1181,1182-1293,1294-1416,1417-1548,1549-1667,1668-1769,1770-1943,1944-2118,2119-2254,2255-2395,2396-2544,2545-2724,2725-2835,2836-2937,2938-3075,3076-3204\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def determine_species_group(species_name, list_of_species_groups, location):\n",
    "    import os\n",
    "\n",
    "    for group_name in list_of_species_groups:\n",
    "        list_of_species = os.listdir(f\"{location}/{group_name}/1.Blast_result\")\n",
    "        if species_name in list_of_species:\n",
    "            return(group_name)\n",
    "\n",
    "from Bio import SeqIO\n",
    "family = \"Lycaenidae\"\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/\"\n",
    "list_of_species_groups = [\"3.Satyrine\",\"4.Pierinae\",\"5.Coliadinae\",\"6.Leptidea\",\"8.Heliconiinae_Danainae\",\"9.Papilionidae\",\"10.Lycaenidae\"]\n",
    "\n",
    "\n",
    "fasta_file = SeqIO.parse(f\"{location}/12.Family_specific_tests/{family}/{family}_period_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "output = \"\"\n",
    "for records in fasta_file:\n",
    "    if \"c-album\" in records.id:\n",
    "        species_name = f'{records.id.split(\"-\")[0]}-{records.id.split(\"-\")[1]}'\n",
    "    else:\n",
    "        species_name = (records.id.split(\"-\")[0])\n",
    "    \n",
    "    group_name = determine_species_group(species_name, list_of_species_groups, location)\n",
    "    \n",
    "    with open(f\"{location}/{group_name}/1.Blast_result/{species_name}/final_coordinates.csv\", 'r') as coordinate_file:\n",
    "        coordinate_list = coordinate_file.readlines()\n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "    \n",
    "    exon_dictionary = {}\n",
    "\n",
    "    for i in range(1,len(coordinate_list)):\n",
    "        coordinate_split = coordinate_list[i].split(\",\")\n",
    "        if coordinate_split[5] == \"N\":\n",
    "            exon_start,exon_stop = int(coordinate_split[2]),int(coordinate_split[3])\n",
    "        else:\n",
    "            exon_start,exon_stop = 0,0\n",
    "        exon_dictionary[i] = exon_stop-exon_start\n",
    "\n",
    "            \n",
    "    # print(\"\".join(coordinate_list))\n",
    "    # print(exon_dictionary)\n",
    "    pos_counter = 0\n",
    "    seq_counter = 0\n",
    "    output += f\"{records.id}\"  \n",
    "    # print(len(records.seq))\n",
    "    for key,value in exon_dictionary.items():\n",
    "        output +=f\",{pos_counter+1}-\" \n",
    "        exon_length = (seq_counter + int(value)+1)\n",
    "        # print(output, f\"Exon_{key}\", exon_length)\n",
    "        while pos_counter < len(records.seq):\n",
    "            \n",
    "            pos_counter += 1\n",
    "            # print(records.seq[pos_counter-1],pos_counter,seq_counter)\n",
    "            if records.seq[pos_counter-1] != \"-\":\n",
    "                seq_counter += 1\n",
    "                # print(records.seq[pos_counter-1],seq_counter)\n",
    "            if seq_counter == exon_length:\n",
    "                output += f\"{pos_counter}\"\n",
    "                break\n",
    "        \n",
    "            \n",
    "\n",
    "    if output[-1] == \"-\":\n",
    "        output += f\"{pos_counter}\"\n",
    "    output += \"\\n\"\n",
    "print(output)\n",
    "with open(f\"{location}/12.Family_specific_tests/{family}/05.Summary Figure/exon_position.txt\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3a838dd4-d900-4bdc-8f61-ccc206be7838",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aricia_artaxerxes-period 335 419 341 425\n",
      "Aricia_artaxerxes-period 784 952 835 1008\n",
      "Glaucopsyche_alexis-period 314 415 329 430\n",
      "Glaucopsyche_alexis-period 793 954 847 1008\n",
      "Polyommatus_iphigenia-period 795 959 840 1008\n",
      "Polyommatus_iphigenia-period 181 263 188 270\n",
      "Polyommatus_iphigenia-period 322 423 329 430\n",
      "Celastrina_argiolus-period 797 958 847 1008\n",
      "Celastrina_argiolus-period 335 419 341 425\n",
      "Aricia_agestis-period 787 955 835 1008\n",
      "Aricia_agestis-period 323 424 329 430\n",
      "Lysandra_bellargus-period 184 235 190 241\n",
      "Lysandra_bellargus-period 323 424 329 430\n",
      "Lysandra_bellargus-period 800 962 842 1008\n",
      "Lysandra_coridon-period 184 236 189 241\n",
      "Lysandra_coridon-period 799 961 842 1008\n",
      "Lysandra_coridon-period 324 425 329 430\n",
      "Cyaniris_semiargus-period 324 425 329 430\n",
      "Cyaniris_semiargus-period 183 236 188 241\n",
      "Cyaniris_semiargus-period 794 963 834 1008\n",
      "Calycopis_cecrops-period 797 957 846 1008\n",
      "Calycopis_cecrops-period 321 424 327 430\n",
      "Phengaris_arion-period 323 425 328 430\n",
      "Phengaris_arion-period 183 248 188 253\n",
      "Phengaris_arion-period 815 965 857 1008\n",
      "Plebejus_argus-period 324 425 329 430\n",
      "Plebejus_argus-period 818 969 857 1008\n",
      "Plebejus_argus-period 183 236 188 241\n",
      "Polyommatus_icarus-period 798 960 842 1008\n",
      "Polyommatus_icarus-period 182 235 188 241\n",
      "Polyommatus_icarus-period 323 424 329 430\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis\"\n",
    "with open(f\"{location}/11.Alignment/000.old_alignment_work/05.InterProScan/pfam_results.txt\", 'r') as pfam_results:\n",
    "    pfam_list = pfam_results.readlines()\n",
    "output = ''\n",
    "for line in pfam_list:\n",
    "    line_split = line.split(\"\\t\")\n",
    "    sequence_name = line_split[0]\n",
    "    region_start,region_stop = int(line_split[6]),int(line_split[7])\n",
    "    \n",
    "\n",
    "    fasta_file = SeqIO.parse(f\"{location}/12.Family_specific_tests/{family}/{family}_period_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "    for records in fasta_file:\n",
    "        if sequence_name == records.id:\n",
    "            pos_counter = 0\n",
    "            seq_counter = 0\n",
    "            new_start, new_stop = 0,0\n",
    "            sequence = records.seq.translate()\n",
    "            for residue in sequence:\n",
    "                pos_counter += 1\n",
    "                # print(residue)\n",
    "                if residue != \"-\":\n",
    "                    seq_counter += 1\n",
    "                if seq_counter == region_start:\n",
    "                    new_start = pos_counter\n",
    "                if seq_counter == region_stop:\n",
    "                    new_stop = pos_counter\n",
    "                    break\n",
    "                    \n",
    "            print(records.id,region_start,region_stop, new_start, new_stop)\n",
    "            line_split[6] = str(new_start)\n",
    "            line_split[7] = str(new_stop)\n",
    "            output += \"\\t\".join(line_split)            \n",
    "            # assert False\n",
    "\n",
    "with open(f\"{location}/12.Family_specific_tests/{family}/05.Summary Figure/pfam_results_position_fixed.txt\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "ad74242d-7ded-4639-b735-e0459ce6721c",
   "metadata": {},
   "outputs": [],
   "source": [
    "pervasive_old = [450]\n",
    "pervasive = []\n",
    "episodic_old = [31, 84, 100, 143, 450, 501, 555, 557, 659, 750, 799, 803, 972, 983]\n",
    "episodic = []\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis\"\n",
    "with open(f\"{location}/12.Family_specific_tests/{family}/1.trimal_outputs/columns_kept.txt\", 'r') as column_file:\n",
    "    columns = column_file.readlines()[0].split(\"\\t\")[1].split(\",\")\n",
    "for i in pervasive_old:\n",
    "    # print(columns[i])\n",
    "    pervasive.append(int(columns[i]))\n",
    "for i in episodic_old:\n",
    "    episodic.append(int(columns[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "3b31e9d9-57a3-41cc-b973-3f055b89d101",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3204.0\n",
      "{'Calycopis_cecrops-period': Seq('ATGGAGAGAGTCGAAGATGCGGACAAAAATAGTACGCTTCGTGACTCCGGCTAC...CAA'), 'Polyommatus_icarus-period': Seq('ATGGAGATCGTTGAAGATACGGAGAAACGTAGCGGAGTTGGCGACTCCGCGTAT...---'), 'Polyommatus_iphigenia-period': Seq('ATGGAGATTGTTGAAGATACGGAGAAACGTAGCGGAGTTGGCGACTCCGCGTAT...---'), 'Aricia_agestis-period': Seq('ATGGAGATTGTTGAAGATACAGAGAAACGTAGCGGAGTTGCTGACTCCGCTTAT...---'), 'Aricia_artaxerxes-period': Seq('ATGGAGATTGTTGAAGATACGGAGAAACGTAGCGGAGTTGCTGACTCCGCTTAT...---'), 'Cyaniris_semiargus-period': Seq('ATGGAGATTGTTGAAAATGCGGAGAAACGTAGCGGAGTTGGTGACTCCGCGTAT...---'), 'Lysandra_bellargus-period': Seq('ATGGAGATAGTTGAAGAGACGGAGAAACGTAGCGGAGTTGGTGACTCCGCATAT...---'), 'Lysandra_coridon-period': Seq('ATGGAGATAGTTGAAGAGACGGAGAAACGTAGCGGAGTTGGTGACTCCGCGTAT...---'), 'Plebejus_argus-period': Seq('ATGGAGGTTGTTGAAGAGACGGAGAAACGTAGCGGAGTTGGTGACTCCGCGTAT...---'), 'Celastrina_argiolus-period': Seq('ATGGAGATTGTTGAAGAAACGGAAAAACGTAGCGGAGTAGGTGACTCCGCTTAT...---'), 'Glaucopsyche_alexis-period': Seq('ATGGAGATCGTCGAAGAAACAGAAAAACGTAGCGGAGTAGGTGACTCCGCATAT...---'), 'Phengaris_arion-period': Seq('ATGGAGATTGTTGAAGAAACAGAAAAACGTAGCGGAGTAGATGACTCCGCATAT...---')}\n",
      "Calycopis_cecrops-period\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "Seq('-')",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[38], line 63\u001b[0m\n\u001b[1;32m     61\u001b[0m     episodic_position \u001b[38;5;241m=\u001b[39m episodic_pos\u001b[38;5;241m*\u001b[39m\u001b[38;5;241m3\u001b[39m\n\u001b[1;32m     62\u001b[0m     base \u001b[38;5;241m=\u001b[39m species_sequence_dic[species_name][episodic_position:episodic_position\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m3\u001b[39m]\u001b[38;5;241m.\u001b[39mtranslate()\n\u001b[0;32m---> 63\u001b[0m     d\u001b[38;5;241m.\u001b[39mappend(draw\u001b[38;5;241m.\u001b[39mText(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mbase\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;241m10\u001b[39m, \u001b[38;5;241m10\u001b[39m\u001b[38;5;241m+\u001b[39m((episodic_position)\u001b[38;5;241m*\u001b[39m\u001b[38;5;241m.5\u001b[39m), \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m5\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m20\u001b[39m\u001b[38;5;241m*\u001b[39m(species_number\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m1\u001b[39m)\u001b[38;5;241m+\u001b[39m \u001b[38;5;241m50\u001b[39m\u001b[38;5;241m*\u001b[39m(species_number) \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m20\u001b[39m,fill\u001b[38;5;241m=\u001b[39m\u001b[43mcolor_dictionary\u001b[49m\u001b[43m[\u001b[49m\u001b[43mbase\u001b[49m\u001b[43m]\u001b[49m))                    \n\u001b[1;32m     66\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m domain_line \u001b[38;5;129;01min\u001b[39;00m domain_list:\n\u001b[1;32m     67\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m dm[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;129;01min\u001b[39;00m domain_line:\n",
      "\u001b[0;31mKeyError\u001b[0m: Seq('-')"
     ]
    }
   ],
   "source": [
    "import drawsvg as draw\n",
    "from Bio import SeqIO\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/12.Family_specific_tests/\"\n",
    "length_factor = 1\n",
    "fasta_file = SeqIO.parse(f\"{location}/{family}/{family}_period_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "species_sequence_dic = {}\n",
    "for record in fasta_file:\n",
    "    total_length = len(record.seq)/length_factor\n",
    "    species_sequence_dic[record.id] = record.seq\n",
    "    # break\n",
    "\n",
    "color_dictionary = {\"G\":\"#EBEBEB\",\"H\":\"#8282D2\",\"P\":\"#DC9682\",\"A\":\"#C8C8C8\",\"W\":\"#B45AB4\",\"S\":\"#FA9600\",\"T\":\"#FA9600\",\"V\":\"#0F820F\",\"I\":\"#0F820F\",\"L\":\"#0F820F\",\"F\":\"#3232AA\",\"Y\":\"#3232AA\",\"C\":\"#E6E600\",\"M\":\"#E6E600\",\"D\":\"#E60A0A\",\"E\":\"#E60A0A\",\"N\":\"#00DCDC\",\"Q\":\"#00DCDC\",\"K\":\"#145AFF\",\"R\":\"#145AFF\",\"-\":\"#BEA06E\"}\n",
    "print(total_length)\n",
    "# total_length = 1947\n",
    "print(species_sequence_dic)\n",
    "from Bio import Phylo\n",
    "tree = Phylo.read(f\"{location}/{family}/Lycaenidae_period_nostop.fas.aligned.fas.best.fas.treefile_figtree\", 'newick')\n",
    "species_list = [term.name for term in tree.get_terminals()]\n",
    "# print(species_list)\n",
    "\n",
    "# dm = \"Drosophila_melanogaster-AAA28752-period,1-180,211-212,242-250,283-322,522-524,526-529,533-534,553-559,560-568,754-759,795-797,802-803,832-833,842-850,895-899,907-909,915-916,921-939,959-962,1083-1084,1086-1087,1105-1112,1114-1129,1141-1142,1147-1148,1153-1156,1176-1234,1242-1250,1255-1258,1264-1265,1307-1309,1317-1319,1343-1349,1356-1358,1360-1361,1386-1395,1428-1439,1462-1475,1530-1584,1604-1643\".split(\",\")\n",
    "with open(f\"{location}/{family}/05.Summary Figure/gap_position.txt\", 'r') as gap_file:\n",
    "    gap_list = gap_file.readlines()\n",
    "with open(f\"{location}/{family}/05.Summary Figure/pfam_results_position_fixed.txt\", 'r') as domain_file:\n",
    "    domain_list = domain_file.readlines()\n",
    "with open(f\"{location}/{family}/05.Summary Figure/exon_position.txt\", 'r') as exons_file:\n",
    "    exons_list = exons_file.readlines()\n",
    "\n",
    "# pervasive = []\n",
    "\n",
    "# episodic = [340*3]    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "d = draw.Drawing(total_length, 80*len(gap_list))    \n",
    "\n",
    "for species_number in range(len(species_list)):\n",
    "    species_name = species_list[species_number]\n",
    "    # print(species_name)\n",
    "    \n",
    "    for gap_line in range(len(gap_list)):\n",
    "        \n",
    "        dm = gap_list[gap_line].strip().split(\",\")\n",
    "        \n",
    "        if dm[0] == species_name:\n",
    "            print(dm[0])\n",
    "            d.append(draw.Lines(10, 20*(species_number+1) + 50*(species_number) + 20,\n",
    "                                 10+(total_length/10)*5, 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                10+(total_length/10)*5,20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                10,20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                close=True,\n",
    "                        fill='#eeee00',\n",
    "                        stroke='black'))\n",
    "            d.append(draw.Text(dm[0], 12, 10, 20*(species_number+1)-18+ 50*(species_number) + 20))\n",
    "            length = round((total_length)/100)\n",
    "            # print(length)\n",
    "            \n",
    "            \n",
    "            for episodic_pos in episodic:\n",
    "                episodic_position = episodic_pos*3\n",
    "                base = species_sequence_dic[species_name][episodic_position:episodic_position+3].translate()\n",
    "                d.append(draw.Text(f\"{base}\", 10, 10+((episodic_position)*.5), -5 + 20*(species_number+1)+ 50*(species_number) + 20,fill=color_dictionary[base]))                    \n",
    "\n",
    "            \n",
    "            for domain_line in domain_list:\n",
    "                if dm[0] in domain_line:\n",
    "                    domain_line_split = domain_line.strip().split(\"\\t\")\n",
    "                \n",
    "                    start, stop = int(domain_line_split[6])*3/length_factor, int(domain_line_split[7])*3/length_factor\n",
    "                    \n",
    "                    d.append(draw.Lines(10+((start-1)*.5), 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                               10 + ((stop-1)*.5), 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                               10 + ((stop-1)*.5), 20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                               10+((start-1)*.5), 20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                               close = True,\n",
    "                               fill = \"#AA336A\"))\n",
    "                    d.append(draw.Text(f\"{domain_line_split[5]}\", 10, 10+((start-1)*.5), -5 + 20*(species_number+1)+ 50*(species_number) + 20))\n",
    "               \n",
    "            \n",
    "            for i in range(1,len(dm)):\n",
    "                # print(dm[i])\n",
    "                start,stop = int(dm[i].split(\"-\")[0])/length_factor,int(dm[i].split(\"-\")[1])/length_factor\n",
    "                d.append(draw.Lines(10+((start-1)*.5), 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                   10 + ((stop-1)*.5), 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                   10 + ((stop-1)*.5), 20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                   10+((start-1)*.5), 20 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                   close = True,\n",
    "                                   fill = \"#808080\"))\n",
    "           \n",
    "            for exon_line in exons_list:\n",
    "                if dm[0] in exon_line:\n",
    "                    exon_line_split = exon_line.strip().split(\",\")\n",
    "                    # print(exon_line_split)\n",
    "                    exon_counter = 1\n",
    "                    for i in range (1, len(exon_line_split)):\n",
    "                        start,stop = int(exon_line_split[i].split(\"-\")[0])/length_factor+1,(int(exon_line_split[i].split(\"-\")[1])-1)/length_factor\n",
    "                        d.append(draw.Lines(10+((start-1)*.5), 23+ 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                            10+((start-1)*.5), 26 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                            10 + ((stop-1)*.5), 26 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                            10 + ((stop-1)*.5), 23 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                                   \n",
    "                                   \n",
    "                                   close = True,\n",
    "                                   fill = \"#00008B\"))\n",
    "                        # print(10+((start + ((stop-start)/2) -1)*.5), 35 + 20*(species_number+1)+ 30*(species_number))\n",
    "                        d.append(draw.Text(str(exon_counter), 10, 10+((start + ((stop-start)/2) -5)*.5), 35 + 20*(species_number+1)+ 50*(species_number) + 20))\n",
    "                        exon_counter += 1\n",
    "                        # assert False\n",
    "                        \n",
    "for episodic_pos in episodic:\n",
    "    episodic_position = episodic_pos*3\n",
    "    d.append(draw.Lines(10+((episodic_position)*.5),30 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                   10+((episodic_position-15)*.5),45 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                   10+((episodic_position+15)*.5),45 + 20*(species_number+1)+ 50*(species_number) + 20,\n",
    "                   \n",
    "                   close = True,\n",
    "                   fill = \"#AA336A\"))\n",
    "            \n",
    "            \n",
    "                # print(dm[i])\n",
    "                # break\n",
    "\n",
    "# d.save_svg(f\"{location}/domains.svg\" )\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "296b3618-c52f-421f-aab7-f76d8eefa28f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
