{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "12f98d0a-daf2-4ad6-aef0-7eb1394a8a41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b50aa19f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aricia_agestis\n",
      "Exon_1 Match\n",
      "[['Aricia_agestis', ['32775356', '32775504']], ['Danaus_plexippus', ['32775356', '32775504']], ['Maniola_jurtina', ['32775356', '32775504']]]\n",
      "Exon_2 Match\n",
      "[['Aricia_agestis', ['32771356', '32771504']], ['Danaus_plexippus', ['32771356', '32771504']], ['Maniola_jurtina', ['32771356', '32771504']]]\n",
      "Exon_3 Match\n",
      "[['Aricia_agestis', ['32771022', '32771203']], ['Danaus_plexippus', ['32771022', '32771203']], ['Maniola_jurtina', ['32771022', '32771203']]]\n",
      "Exon_4 Match\n",
      "[['Aricia_agestis', ['32769986', '32770103']], ['Danaus_plexippus', ['32769986', '32770103']], ['Maniola_jurtina', ['32769986', '32770103']]]\n",
      "Exon_5 Match\n",
      "[['Aricia_agestis', ['32765550', '32765645']], ['Danaus_plexippus', ['32765550', '32765645']], ['Maniola_jurtina', ['32765550', '32765645']]]\n",
      "Exon_6 mismatch\n",
      "[['Aricia_agestis', ['32763192', '32763330']], ['Danaus_plexippus', ['32763192', '32763330']], ['Maniola_jurtina', ['32770045', '32770192']]]\n",
      "Proceed?y\n",
      "Aricia_agestis Exon_6 160_frame1 0.0 \n",
      "MLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELN\n",
      "Danaus_plexippus Exon_6 160_frame1 1.268 \n",
      "MLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELN\n",
      "Maniola_jurtina Exon_6 1_frame0 1.381 \n",
      "TRVVSAQKIRRVERPLSFFIYKTFHTLFQNDIIQTNLFDLTYQDDRSN\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ ------------------------MLQYVEQTDNEEHSGYQREQES--------------\n",
      "set160_frame1   YTCTEIIFHNILSFYFILCIIFLGMLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIK\n",
      "                                        **********************              \n",
      "\n",
      "Aricia_agestis_ -------------------\n",
      "set160_frame1   EEKILKELNSCYFYTRTRI\n",
      "                                   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Danaus_plexippu --------------------------KNLESNENGHHS-YQDEHES--------------\n",
      "set160_frame1   YTCTEIIFHNILSFYFILCIIFLGMLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIK\n",
      "                                          : :*..:* .** ** *:**              \n",
      "\n",
      "Danaus_plexippu -------------------\n",
      "set160_frame1   EEKILKELNSCYFYTRTRI\n",
      "                                   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina TNTEQNEACDLSFQQDQES----------------------------------------\n",
      "set1_frame0     NDIIQTNLFDLTYQDDRSNLYNILQNNTTAADSTVTQSAGESSYYVIYVFCLYCRLQST\n",
      "                .:  *.:  **::*:*:..                                        \n",
      "\n",
      "Choose Species 1 - 31\n",
      "Exon_7 Match\n",
      "[['Aricia_agestis', ['32748502', '32748740']], ['Danaus_plexippus', ['32748502', '32748740']], ['Maniola_jurtina', ['32748502', '32748740']]]\n",
      "Exon_8 Match\n",
      "[['Aricia_agestis', ['32746812', '32746959']], ['Danaus_plexippus', ['32746812', '32746959']], ['Maniola_jurtina', ['32746812', '32746959']]]\n",
      "Exon_9 Match\n",
      "[['Aricia_agestis', ['32732447', '32732630']], ['Danaus_plexippus', ['32732447', '32732630']], ['Maniola_jurtina', ['32732447', '32732630']]]\n",
      "Exon_10 mismatch\n",
      "[['Aricia_agestis', ['32729653', '32729829']], ['Danaus_plexippus', ['32729653', '32729829']], ['Maniola_jurtina', ['32729653', '32729859']]]\n",
      "Proceed?y\n",
      "Aricia_agestis Exon_10 375_frame0 0.0 \n",
      "PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Danaus_plexippus Exon_10 375_frame0 1.403 \n",
      "PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Maniola_jurtina Exon_10 375_frame0 1.408 \n",
      "SRLTNTSPFQPSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ ----------PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRT\n",
      "set375_frame0   SRLTNTSPFQPSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRT\n",
      "                          **************************************************\n",
      "\n",
      "Aricia_agestis_ NTRDECRSG-----\n",
      "set375_frame0   NTRDECRSGVTYNV\n",
      "                *********     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Danaus_plexippu L----------ASVKSAATSGSTSATVATLGTAITTASATWPPRSSYLLYTTGSDTTSVS\n",
      "set375_frame0   SRLTNTSPFQPSSVKS--TSGSTT------GTPTASGSAAWSRVTSQVCYSE-SDRASMS\n",
      "                           :****  *****:      **. ::.**:*.  :* : *:  ** :*:*\n",
      "\n",
      "Danaus_plexippu GGSRSSQRNSSQE----------\n",
      "set375_frame0   SESRSSHRTNTRDECRSGVTYNV\n",
      "                . ****:*..:::          \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina -------PPSVKSVATSAGSASATVATVGTATTAAASWPTRASHIRYATGSDTGSVSGES\n",
      "set375_frame0   SRLTNTSPFQPSSVKSTSGSTTGTPTASGS-----AAWSRVTSQVCYSE-SDRASMSSES\n",
      "                       * . .** :::**::.* :: *:     *:*.  :*:: *:  ** .*:*.**\n",
      "\n",
      "Maniola_jurtina RSSQRNSSQE----------\n",
      "set375_frame0   RSSHRTNTRDECRSGVTYNV\n",
      "                ***:*..:::          \n",
      "\n",
      "Choose Species 1 - 31\n",
      "Exon_11 Match\n",
      "[['Aricia_agestis', ['32728335', '32728496']], ['Danaus_plexippus', ['32728335', '32728496']], ['Maniola_jurtina', ['32728335', '32728496']]]\n",
      "Exon_12 Match\n",
      "[['Aricia_agestis', ['32724296', '32724418']], ['Danaus_plexippus', ['32724296', '32724418']], ['Maniola_jurtina', ['32724296', '32724418']]]\n",
      "Exon_13 Match\n",
      "[['Aricia_agestis', ['32721852', '32722010']], ['Danaus_plexippus', ['32721852', '32722010']], ['Maniola_jurtina', ['32721852', '32722010']]]\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Aricia_agestis,NC_056428.1,32775356,32775504,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "Aricia_agestis,NC_056428.1,32771356,32771504,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_1_query_Exon_2,1,49,49,Y,N\n",
      "Aricia_agestis,NC_056428.1,32771022,32771203,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_0_query_Exon_3,1,60,60,Y,N\n",
      "Aricia_agestis,NC_056428.1,32769986,32770103,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_1_query_Exon_4,1,39,39,Y,N\n",
      "Aricia_agestis,NC_056428.1,32765550,32765645,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_1_query_Exon_5,1,31,31,Y,N\n",
      "Aricia_agestis,NC_056428.1,32763192,32763330,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_2_query_Exon_6,1,45,45,Y,N\n",
      "Aricia_agestis,NC_056428.1,32748502,32748740,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_1_query_Exon_7,1,79,79,Y,N\n",
      "Aricia_agestis,NC_056428.1,32746812,32746959,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_2_query_Exon_8,1,48,48,Y,N\n",
      "Aricia_agestis,NC_056428.1,32732447,32732630,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_0_query_Exon_9,1,61,61,Y,N\n",
      "Aricia_agestis,NC_056428.1,32729653,32729829,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_10,1,59,59,Y,N\n",
      "Aricia_agestis,NC_056428.1,32728335,32728496,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_11,1,54,54,Y,N\n",
      "Aricia_agestis,NC_056428.1,32724296,32724418,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_12,1,41,41,Y,N\n",
      "Aricia_agestis,NC_056428.1,32721852,32722010,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_13,1,53,53\n",
      "\n",
      "Aricia_artaxerxes\n",
      "Exon_1 Match\n",
      "[['Aricia_agestis', ['35290983', '35291131']], ['Danaus_plexippus', ['35290983', '35291131']], ['Maniola_jurtina', ['35290983', '35291131']]]\n",
      "Exon_2 Match\n",
      "[['Aricia_agestis', ['35286983', '35287131']], ['Danaus_plexippus', ['35286983', '35287131']], ['Maniola_jurtina', ['35286983', '35287131']]]\n",
      "Exon_3 Match\n",
      "[['Aricia_agestis', ['35286645', '35286826']], ['Danaus_plexippus', ['35286645', '35286826']], ['Maniola_jurtina', ['35286645', '35286826']]]\n",
      "Exon_4 Match\n",
      "[['Aricia_agestis', ['35280363', '35280480']], ['Danaus_plexippus', ['35280363', '35280480']], ['Maniola_jurtina', ['35280363', '35280480']]]\n",
      "Exon_5 Match\n",
      "[['Aricia_agestis', ['35275927', '35276022']], ['Danaus_plexippus', ['35275927', '35276022']], ['Maniola_jurtina', ['35275927', '35276022']]]\n",
      "Exon_6 mismatch\n",
      "[['Aricia_agestis', ['35273605', '35273743']], ['Danaus_plexippus', ['35273605', '35273743']], ['Maniola_jurtina', ['35280422', '35280569']]]\n",
      "Proceed?y\n",
      "Aricia_agestis Exon_6 152_frame1 0.0 \n",
      "MLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELN\n",
      "Danaus_plexippus Exon_6 152_frame1 1.268 \n",
      "MLQYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELN\n",
      "Maniola_jurtina Exon_6 1_frame0 1.381 \n",
      "TRVVSAHKIRRVERPLSFFIYKTFHTLFQNDIIQTNLFDLTYQDDRSN\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ ----------------------------------------------------------ML\n",
      "set152_frame1   DTAWRQTDRRTNIEVSVMGSRFYSLGTEPQNIQSNIGTEIIFHNILSFYFILCIIFLGML\n",
      "                                                                          **\n",
      "\n",
      "Aricia_agestis_ QYVEQTDNEEHSGYQREQES---------------------------------\n",
      "set152_frame1   QYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELNSCYFYTRTRI\n",
      "                ********************                                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Danaus_plexippu ------------------------------------------------------------\n",
      "set152_frame1   DTAWRQTDRRTNIEVSVMGSRFYSLGTEPQNIQSNIGTEIIFHNILSFYFILCIIFLGML\n",
      "                                                                            \n",
      "\n",
      "Danaus_plexippu KNLESNENGHHS-YQDEHES---------------------------------\n",
      "set152_frame1   QYVEQTDNEEHSGYQREQESRQVFYIYYQCSVIKEEKILKELNSCYFYTRTRI\n",
      "                : :*..:* .** ** *:**                                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina TNTEQNEACDLSFQQDQES----------------------------------------\n",
      "set1_frame0     NDIIQTNLFDLTYQDDRSNLYNILQNNTTAADSTVTQSAGESSYYVIYVFCLYCRLQST\n",
      "                .:  *.:  **::*:*:..                                        \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Choose Species 1 - 31\n",
      "Exon_7 Match\n",
      "[['Aricia_agestis', ['35257816', '35258054']], ['Danaus_plexippus', ['35257816', '35258054']], ['Maniola_jurtina', ['35257816', '35258054']]]\n",
      "Exon_8 Match\n",
      "[['Aricia_agestis', ['35256139', '35256286']], ['Danaus_plexippus', ['35256139', '35256286']], ['Maniola_jurtina', ['35256139', '35256286']]]\n",
      "Exon_9 Match\n",
      "[['Aricia_agestis', ['35227952', '35228135']], ['Danaus_plexippus', ['35227952', '35228135']], ['Maniola_jurtina', ['35227952', '35228135']]]\n",
      "Exon_10 mismatch\n",
      "[['Aricia_agestis', ['35224167', '35224343']], ['Danaus_plexippus', ['35224167', '35224343']], ['Maniola_jurtina', ['35224167', '35224373']]]\n",
      "Proceed?y\n",
      "Aricia_agestis Exon_10 693_frame2 0.0 \n",
      "PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Danaus_plexippus Exon_10 693_frame2 1.403 \n",
      "PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Maniola_jurtina Exon_10 693_frame2 1.408 \n",
      "SRLTNTSPFQPSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRTNTRDECRSG\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ ----------PSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRT\n",
      "set693_frame2   SRLTNTSPFQPSSVKSTSGSTTGTPTASGSAAWSRVTSQVCYSESDRASMSSESRSSHRT\n",
      "                          **************************************************\n",
      "\n",
      "Aricia_agestis_ NTRDECRSG-----\n",
      "set693_frame2   NTRDECRSGVTYNV\n",
      "                *********     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Danaus_plexippu L----------ASVKSAATSGSTSATVATLGTAITTASATWPPRSSYLLYTTGSDTTSVS\n",
      "set693_frame2   SRLTNTSPFQPSSVKS--TSGSTT------GTPTASGSAAWSRVTSQVCYSE-SDRASMS\n",
      "                           :****  *****:      **. ::.**:*.  :* : *:  ** :*:*\n",
      "\n",
      "Danaus_plexippu GGSRSSQRNSSQE----------\n",
      "set693_frame2   SESRSSHRTNTRDECRSGVTYNV\n",
      "                . ****:*..:::          \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina -------PPSVKSVATSAGSASATVATVGTATTAAASWPTRASHIRYATGSDTGSVSGES\n",
      "set693_frame2   SRLTNTSPFQPSSVKSTSGSTTGTPTASGS-----AAWSRVTSQVCYSE-SDRASMSSES\n",
      "                       * . .** :::**::.* :: *:     *:*.  :*:: *:  ** .*:*.**\n",
      "\n",
      "Maniola_jurtina RSSQRNSSQE----------\n",
      "set693_frame2   RSSHRTNTRDECRSGVTYNV\n",
      "                ***:*..:::          \n",
      "\n",
      "Choose Species 1 - 31\n",
      "Exon_11 Match\n",
      "[['Aricia_agestis', ['35222852', '35223013']], ['Danaus_plexippus', ['35222852', '35223013']], ['Maniola_jurtina', ['35222852', '35223013']]]\n",
      "Exon_12 Match\n",
      "[['Aricia_agestis', ['35218342', '35218464']], ['Danaus_plexippus', ['35218342', '35218464']], ['Maniola_jurtina', ['35218342', '35218464']]]\n",
      "Exon_13 Match\n",
      "[['Aricia_agestis', ['35215759', '35215917']], ['Danaus_plexippus', ['35215759', '35215917']], ['Maniola_jurtina', ['35215759', '35215917']]]\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Aricia_artaxerxes,OW569310.1,35290983,35291131,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "Aricia_artaxerxes,OW569310.1,35286983,35287131,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_1_query_Exon_2,1,49,49,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35286645,35286826,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_0_query_Exon_3,1,60,60,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35280363,35280480,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_1_query_Exon_4,1,39,39,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35275927,35276022,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_1_query_Exon_5,1,31,31,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35273605,35273743,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_2_query_Exon_6,1,45,45,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35257816,35258054,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_1_query_Exon_7,1,79,79,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35256139,35256286,1,N,Aricia_agestis_XM_042131827.1_Frame_2_rightoh_2_query_Exon_8,1,48,48,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35227952,35228135,1,N,Aricia_agestis_XM_042131827.1_Frame_1_rightoh_0_query_Exon_9,1,61,61,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35224167,35224343,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_10,1,59,59,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35222852,35223013,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_11,1,54,54,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35218342,35218464,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_12,1,41,41,Y,N\n",
      "Aricia_artaxerxes,OW569310.1,35215759,35215917,1,N,Aricia_agestis_XM_042131827.1_Frame_0_rightoh_0_query_Exon_13,1,53,53\n",
      "\n",
      "Calycopis_cecrops\n",
      "Exon_1 Match\n",
      "[['Aricia_agestis', ['161686', '161834']], ['Danaus_plexippus', ['161686', '161834']], ['Maniola_jurtina', ['161686', '161834']]]\n",
      "Exon_2 Match\n",
      "[['Aricia_agestis', ['165686', '165834']], ['Danaus_plexippus', ['165686', '165834']], ['Maniola_jurtina', ['165686', '165834']]]\n",
      "Exon_3 Match\n",
      "[['Aricia_agestis', ['166279', '166460']], ['Danaus_plexippus', ['166279', '166460']], ['Maniola_jurtina', ['166279', '166460']]]\n",
      "Exon_4 Match\n",
      "[['Aricia_agestis', ['167112', '167232']], ['Danaus_plexippus', ['167112', '167232']], ['Maniola_jurtina', ['167112', '167232']]]\n",
      "Exon_5 Match\n",
      "[['Aricia_agestis', ['168478', '168573']], ['Danaus_plexippus', ['168478', '168573']], ['Maniola_jurtina', ['168478', '168573']]]\n",
      "Exon_6 mismatch\n",
      "[['Aricia_agestis', ['175141', '175303']], ['Danaus_plexippus', ['175240', '175303']], ['Maniola_jurtina', ['167068', '167170']]]\n",
      "Proceed?y\n",
      "Aricia_agestis Exon_6 135_frame0 1.062 \n",
      "SYMRILNPNAIILINLCRCYLTPMYTVCIFFLGLNMDRNDTDESSGYQREHEP\n",
      "Danaus_plexippus Exon_6 135_frame0 1.308 \n",
      "LNMDRNDTDESSGYQREHEP\n",
      "Maniola_jurtina Exon_6 1_frame0 1.43 \n",
      "PIIFLEASCVFLFQGEVINTSLFDLIFQDDRQK\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Aricia_agestis_ ---MLQY-----------------------------------------VEQTDNEEHSGY\n",
      "set135_frame0   KKIQLSYIKKFDPSYMRILNPNAIILINLCRCYLTPMYTVCIFFLGLNMDRNDTDESSGY\n",
      "                    *.*                                         :::.*.:* ***\n",
      "\n",
      "Aricia_agestis_ QREQE-S\n",
      "set135_frame0   QREHEPR\n",
      "                ***:*  \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Danaus_plexippu K----------------------------------------------NLESNENGHHS-Y\n",
      "set135_frame0   KKIQLSYIKKFDPSYMRILNPNAIILINLCRCYLTPMYTVCIFFLGLNMDRNDTDESSGY\n",
      "                *                                              *:: *:... * *\n",
      "\n",
      "Danaus_plexippu QDEHE-S\n",
      "set135_frame0   QREHEPR\n",
      "                * ***  \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina TNTEQNEACDLSFQQDQES-------------------------------------\n",
      "set1_frame0     GEVINTSLFDLIFQDDRQKLYNILQNSTPVADTNRLLASPGWKLCIITGLLFFISN\n",
      "                 :. :..  ** **:*::.                                     \n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "family_group = \"8.Lycaenidae\"\n",
    "annotated_genome_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "# species_list = [\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Limenitis_camilla\"]\n",
    "# species_list = [\"Argynnis_bischoffii_washingtonia\",\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Bicyclus_anynana\",\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Vanessa_tameamea\"]\n",
    "for species in species_list:\n",
    "    print(species)\n",
    "    blast_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{blast_location}/{species}\")\n",
    "    query_species_list = []\n",
    "    coordinate_detail_dictionary = {}\n",
    "    coordinate_info_dictionary = {}\n",
    "    scaffold = ''\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\".csv\") and len(file.split(\"_\")) >= 5 and not(file.endswith(\"_old.csv\")) and not(file.endswith(\"_old_backup.csv\")):\n",
    "#             print(file)\n",
    "            if \".\" in file[:-4]:\n",
    "                query_species = f\"{file.split('_')[-2].split('.')[1]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            else:\n",
    "                query_species = f\"{file.split('_')[-2]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            query_species_list.append(query_species)\n",
    "            coordinate_detail_dictionary[query_species] = {}\n",
    "            with open(f\"{blast_location}/{species}/{file}\", 'r') as coordinate_file_open:\n",
    "                coordinate_file_lines_list  =  coordinate_file_open.readlines()\n",
    "            if scaffold == '':\n",
    "                scaffold = coordinate_file_lines_list[1].split(\",\")[1]\n",
    "    \n",
    "            for i in range(1,len(coordinate_file_lines_list)):\n",
    "                \n",
    "                coordinate_file_lines_split = coordinate_file_lines_list[i].split(\",\")\n",
    "                start,stop = coordinate_file_lines_split[2],coordinate_file_lines_split[3]\n",
    "                exon = f\"Exon_{i}\"\n",
    "                if exon not in coordinate_info_dictionary:\n",
    "                    coordinate_info_dictionary[exon] = {}\n",
    "                coordinate_info_dictionary[exon][query_species] = [start,stop]\n",
    "                coordinate_detail_dictionary[query_species][exon] = coordinate_file_lines_list[i]\n",
    "            \n",
    "            # break\n",
    "            # print(query_species)\n",
    "            # print(file)\n",
    "#     print(coordinate_info_dictionary)\n",
    "#     print(coordinate_detail_dictionary)\n",
    "    # assert False\n",
    "    output = coordinate_file_lines_list[0]\n",
    "    for exon,species_specific_coordinates in coordinate_info_dictionary.items():\n",
    "        # print(exon,species_specific_coordinates)\n",
    "        coordinates_list = []\n",
    "        previous_coordinates = \"\"\n",
    "        match = 1\n",
    "        for query_species, coordinates in coordinate_info_dictionary[exon].items():\n",
    "            coordinates_list.append([query_species,coordinates])\n",
    "            if previous_coordinates != '':\n",
    "                if coordinates != previous_coordinates:\n",
    "                    match = 0\n",
    "                    \n",
    "            previous_coordinates = coordinates\n",
    "       \n",
    "        \n",
    "        \n",
    "        if match == 0:\n",
    "            print(exon,\"mismatch\")\n",
    "            print(coordinates_list)\n",
    "                \n",
    "            while True:\n",
    "                choice = input(\"Proceed?\")\n",
    "                if choice[0].lower() == \"y\" or choice[0].lower() == \"n\":\n",
    "                    break\n",
    "            \n",
    "            while True:\n",
    "                if choice[0].lower() == \"y\":\n",
    "                    alignment_detail_dictionary = {}\n",
    "                    for query_species_name in query_species_list:\n",
    "                        list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                        for folders in list_of_folders:\n",
    "                            if folders.endswith(query_species_name):\n",
    "                                # print()\n",
    "                                \n",
    "                                with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "                                    # alignment_detail = query_file_open.readlines()[0]\n",
    "                                    alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                                    alignment_detail_dictionary[query_species_name] = alignment_detail\n",
    "                                with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "                                    distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "                                try:\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query_spliced.txt\", 'r') as new_query_file:\n",
    "                                        new_query = (new_query_file.readlines()[-1].rstrip())\n",
    "                                except:\n",
    "                                    new_query = \"Issue with query\"\n",
    "                                print(query_species_name,exon, alignment_detail, distance_score,f\"\\n{new_query}\")\n",
    "                    while True:            \n",
    "                        check_alignment = input(\"Check Alignment??\")\n",
    "                        if check_alignment[0].lower() == \"y\" or check_alignment[0].lower() == \"n\":\n",
    "                            break\n",
    "                    if check_alignment[0].lower() == \"y\":\n",
    "                        for query_species_name in query_species_list:\n",
    "                            list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                            for folders in list_of_folders:\n",
    "                                if folders.endswith(query_species_name):\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/alignment_clustal_{exon}_translated_genomic_sequence_{alignment_detail_dictionary[query_species_name]}.fa.txt\", 'r') as clustal_alignment_file:\n",
    "                                        clustal_alignment = clustal_alignment_file.readlines()\n",
    "                                    print(\"\".join(clustal_alignment))\n",
    "                                    \n",
    "                                # assert False\n",
    "                    while True:\n",
    "                        try:\n",
    "                            species_number = (input(f\"Choose Species 1 - {len(query_species_list)}\"))\n",
    "                            \n",
    "                            \n",
    "                            if species_number[0].lower() == 'n':\n",
    "                                output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                                break \n",
    "                            output += coordinate_detail_dictionary[query_species_list[int(species_number)-1]][exon]\n",
    "                            break\n",
    "                        except:\n",
    "                            print(\"Retry\")\n",
    "                    break\n",
    "                elif choice[0].lower() == \"n\":\n",
    "                    output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                    break\n",
    "                \n",
    "                \n",
    "        else:\n",
    "            print(exon, \"Match\")\n",
    "            print(coordinates_list)\n",
    "            output += coordinate_detail_dictionary[query_species_list[0]][exon]\n",
    "        \n",
    "       \n",
    "\n",
    "        # assert False\n",
    "    # print(coordinate_detail_dictionary)\n",
    "    print(output)\n",
    "    with open(f\"{blast_location}/{species}/final_coordinates.csv\", \"w\") as out_file:\n",
    "        out_file.write(output)\n",
    "    \n",
    "    # 'Coenonympha_glycerion,OY979626.1,7015875,7015963,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7892a6a6-9400-414d-bbaa-cf893036e0d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
