{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2b2fddd4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "OSError",
     "evalue": "[Errno 19] No such device: '/mnt/f/Genomes_2023-12-26/Leptidea_juvernica'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 13\u001b[0m\n\u001b[1;32m     11\u001b[0m error_exons \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m     12\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m species \u001b[38;5;129;01min\u001b[39;00m list_of_species:\n\u001b[0;32m---> 13\u001b[0m     genome_file \u001b[38;5;241m=\u001b[39m \u001b[43mget_genome_file\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     14\u001b[0m     \u001b[38;5;28mprint\u001b[39m(genome_file)\n\u001b[1;32m     15\u001b[0m     previous_coordinate_start \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "Cell \u001b[0;32mIn[1], line 2\u001b[0m, in \u001b[0;36mget_genome_file\u001b[0;34m(genome_location, species)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_genome_file\u001b[39m(genome_location,species):\n\u001b[0;32m----> 2\u001b[0m     list_of_files_in_genome_folder \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlistdir\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgenome_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m list_of_files_in_genome_folder:\n\u001b[1;32m      4\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m file\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_genomic.fna\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 19] No such device: '/mnt/f/Genomes_2023-12-26/Leptidea_juvernica'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "\n",
    "output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/6.Leptidea\"\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[5], query_name.split('_')[7]\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            error_exons += coordinate_file_list[i]\n",
    "        \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "        \n",
    "        exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "        print(exon_output)\n",
    "        \n",
    "        write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
