{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ac657986-6585-4fd4-9b21-3656830020c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "595bc109-0bed-4b8a-a53c-923c4732b4fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Exon_2\n",
      "stop_codon_here\n",
      "\n",
      "\n",
      "Exon 1 : 87 : 29.0 : 0 + 87 * 3 + 0\n",
      "Exon 2 : 1419 : 473.0 : 0 + 1419 * 3 + 0\n",
      "\n",
      "\n",
      "\n",
      "Exon_2\n",
      "stop_codon_here\n",
      "\n",
      "\n",
      "Exon 1 : 150 : 50.0 : 0 + 150 * 3 + 0\n",
      "Exon 2 : 1428 : 476.0 : 0 + 1428 * 3 + 0\n",
      "\n",
      "\n",
      "\n",
      "Exon_2\n",
      "stop_codon_here\n",
      "\n",
      "\n",
      "Exon 1 : 87 : 29.0 : 0 + 87 * 3 + 0\n",
      "Exon 2 : 1416 : 472.0 : 0 + 1416 * 3 + 0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "family_group = \"3.Satyrine\"\n",
    "\n",
    "trascript_source_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/14.Cryptochrome_2_Exon_Analysis/1.Annotated Species\"\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/14.Cryptochrome_2_Exon_Analysis/{family_group}/1.Query\"\n",
    "source_species_list = [\"Maniola_jurtina\",\"Bicyclus_anynana\",\"Pararge_aegeria\"]\n",
    "# source_species_list = [\"Maniola_jurtina\",\"Bicyclus_anynana\",\"Pieris_napi\"]\n",
    "# source_species_list = [\"Colias_croceus\",\"Leptidea_sinapis\",\"Zerene_cesonia\"]\n",
    "# source_species_list = [\"Danaus_plexippus\",\"Nymphalis_io\",\"Vanessa_tameamea\"]\n",
    "# source_species_list = [\"Danaus_plexippus\",\"Papilio_machaon\",\"Papilio_xuthus\",\"Zerene_cesonia\"]\n",
    "# source_species_list = [\"Danaus_plexippus\",\"Aricia_agestis\",\"Maniola_jurtina\"]\n",
    "\n",
    "for source_species in source_species_list:\n",
    "    list_of_transcripts = os.listdir(f\"{trascript_source_location}/{source_species}/Transcripts\")\n",
    "    if \"desktop.ini\" in list_of_transcripts:\n",
    "        list_of_transcripts.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "    source_transcript = list_of_transcripts[0]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    try:\n",
    "        os.mkdir(query_location)\n",
    "    except:\n",
    "        pass\n",
    "    folders_in_query = os.listdir(query_location)\n",
    "    if source_species not in folders_in_query:\n",
    "        os.mkdir(f\"{query_location}/{source_species}\")\n",
    "    transcript_folders = os.listdir(f\"{query_location}/{source_species}\")\n",
    "\n",
    "    if source_transcript not in transcript_folders:\n",
    "        os.mkdir(f\"{query_location}/{source_species}/{source_transcript}\")\n",
    "\n",
    "    list_of_exons = os.listdir(f\"{trascript_source_location}/{source_species}/Transcripts/{source_transcript}\")\n",
    "    # print(list_of_exons)\n",
    "    if \"desktop.ini\" in list_of_exons:\n",
    "        list_of_exons.remove(\"desktop.ini\")\n",
    "\n",
    "    output = ''\n",
    "    exon_dictionary = {}\n",
    "\n",
    "    for exon in list_of_exons:\n",
    "        exon_number = int(exon.split(\".\")[0].split(\"Exon\")[1])\n",
    "\n",
    "        sequence = SeqIO.parse(f\"{trascript_source_location}/{source_species}/Transcripts/{source_transcript}/{exon}\", \"fasta\")\n",
    "\n",
    "        for record in sequence:\n",
    "            # print(record.id)\n",
    "            # print(record.seq)\n",
    "            # translated_exon = record.seq[left_overhang:].translate()\n",
    "            exon_dictionary[f\"Exon_{exon_number}\"] = record.seq\n",
    "    exon_overhang = {}\n",
    "    exon_structure = \"\"\n",
    "    for exon in range(1,len(exon_dictionary)+1):\n",
    "        if exon == 1:\n",
    "            left_overhang = 0\n",
    "            right_overhang = len(exon_dictionary[f\"Exon_{exon}\"]) % 3\n",
    "            exon_overhang[f\"Exon_{exon}\"] = [left_overhang,right_overhang]\n",
    "        else:\n",
    "            left_overhang = (3-(exon_overhang[f\"Exon_{exon-1}\"][1]))%3\n",
    "            right_overhang = (len(exon_dictionary[f\"Exon_{exon}\"])-left_overhang) % 3\n",
    "            exon_overhang[f\"Exon_{exon}\"] = [left_overhang,right_overhang]\n",
    "\n",
    "        exon_length = len(exon_dictionary[f'Exon_{exon}'])\n",
    "        exon_structure += f\"Exon {exon} : {exon_length} : {exon_length/3} : {left_overhang} + {exon_length-left_overhang-right_overhang} * 3 + {right_overhang}\\n\"\n",
    "        translated_sequence = exon_dictionary[f'Exon_{exon}'][left_overhang:].translate()\n",
    "        if \"*\" in translated_sequence:\n",
    "            print(f\"\\n\\nExon_{exon}\\nstop_codon_here\\n\\n\")\n",
    "        # print(f\"Exon_{exon}\\n{translated_sequence}\")\n",
    "        with open(f\"{query_location}/{source_species}/{source_transcript}/query_Exon_{exon}.fa\", 'w') as query_file:\n",
    "            output1 = f\">{source_species}_{source_transcript}_Frame_{left_overhang}_rightoh_{right_overhang}_query_Exon_{exon}\\n{translated_sequence}\"\n",
    "            query_file.write(output1)\n",
    "        output += output1+\"\\n\\n\"\n",
    "\n",
    "    with open(f\"{query_location}/{source_species}/{source_transcript}/query_protein.fa\", 'w') as query_file:\n",
    "        query_file.write(output)\n",
    "\n",
    "        # print(f\"Exon_{exon}\",exon_dictionary[f\"Exon_{exon}\"])\n",
    "\n",
    "    # for key, value in exon_overhang.items():\n",
    "    #     print(key,value)\n",
    "    print(exon_structure)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "ef49d793-16f9-4099-b574-3a4253716ab8",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Exon_12\n",
      "stop_codon_here\n",
      "\n",
      "\n",
      "Exon 1 : 130 : 43.333333333333336 : 0 + 129 * 3 + 1\n",
      "Exon 2 : 147 : 49.0 : 2 + 144 * 3 + 1\n",
      "Exon 3 : 188 : 62.666666666666664 : 2 + 186 * 3 + 0\n",
      "Exon 4 : 117 : 39.0 : 0 + 117 * 3 + 0\n",
      "Exon 5 : 134 : 44.666666666666664 : 0 + 132 * 3 + 2\n",
      "Exon 6 : 96 : 32.0 : 1 + 93 * 3 + 2\n",
      "Exon 7 : 74 : 24.666666666666668 : 1 + 72 * 3 + 1\n",
      "Exon 8 : 127 : 42.333333333333336 : 2 + 123 * 3 + 2\n",
      "Exon 9 : 183 : 61.0 : 1 + 180 * 3 + 2\n",
      "Exon 10 : 168 : 56.0 : 1 + 165 * 3 + 2\n",
      "Exon 11 : 124 : 41.333333333333336 : 1 + 123 * 3 + 0\n",
      "Exon 12 : 120 : 40.0 : 0 + 120 * 3 + 0\n",
      "\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8fe39ee-a28f-46c7-886b-2fef23066cae",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
