{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "12f98d0a-daf2-4ad6-aef0-7eb1394a8a41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b50aa19f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coenonympha_glycerion\n",
      "Exon_1 Match\n",
      "[['Bicyclus_anynana', ['10535487', '10535583']], ['Maniola_jurtina', ['10535487', '10535583']], ['Pararge_aegeria', ['10535487', '10535583']]]\n",
      "Exon_2 Match\n",
      "[['Bicyclus_anynana', ['10533898', '10534448']], ['Maniola_jurtina', ['10533898', '10534448']], ['Pararge_aegeria', ['10533898', '10534448']]]\n",
      "Exon_3 Match\n",
      "[['Bicyclus_anynana', ['10533204', '10533357']], ['Maniola_jurtina', ['10533204', '10533357']], ['Pararge_aegeria', ['10533204', '10533357']]]\n",
      "Exon_4 Match\n",
      "[['Bicyclus_anynana', ['10532545', '10532789']], ['Maniola_jurtina', ['10532545', '10532789']], ['Pararge_aegeria', ['10532545', '10532789']]]\n",
      "Exon_5 mismatch\n",
      "[['Bicyclus_anynana', ['10532030', '10532313']], ['Maniola_jurtina', ['10532030', '10532403']], ['Pararge_aegeria', ['10532030', '10532313']]]\n",
      "Proceed?y\n",
      "Bicyclus_anynana Exon_5 15_frame0 0.673 \n",
      "ENKNSKEVPKNTTEATQKNVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRPSVTLQERRRKKIVKRGKAN\n",
      "Maniola_jurtina Exon_5 15_frame0 0.511 \n",
      "AVVFFLFLNYLLVQTSNNLMYNLTLCVYLKENKNSKEVPKNTTEATQKNVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRPSVTLQERRRKKIVKRGKAN\n",
      "Pararge_aegeria Exon_5 15_frame0 0.743 \n",
      "ENKNSKEVPKNTTEATQKNVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRPSVTLQERRRKKIVKRGKAN\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan -------------------------------------------QKSRSKEATNIADGTQN\n",
      "set15_frame0    ALIFWKKSFYHKAVVFFLFLNYLLVQTSNNLMYNLTLCVYLKENKNSKEVPKNTTEATQK\n",
      "                                                           :*. .: ..* ::.**:\n",
      "\n",
      "Bicyclus_anynan TEKKIIISETSDCGYGTQIENQESISTSSNEDDIPEKKPVHQKPHTTKPRINNKVRAISV\n",
      "set15_frame0    NVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRP-SV\n",
      "                . ** *:********** ******:*******::*.*****:***..* ::*** *. **\n",
      "\n",
      "Bicyclus_anynan TLQERKRKKIVKRGKSN-\n",
      "set15_frame0    TLQERRRKKIVKRGKANM\n",
      "                *****:*********:* \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina ---------------------------ENN---------YMKDGTNTKNVTKN-------\n",
      "set15_frame0    ALIFWKKSFYHKAVVFFLFLNYLLVQTSNNLMYNLTLCVYLKENKNSKEVPKNTTEATQK\n",
      "                                           .**         *:*:..*:*:*.**       \n",
      "\n",
      "Maniola_jurtina NEKKTVISENSDCGYGTQIENQESISTSSNEDELPAKKPVHQKPHTSKQRINNKVRPSVT\n",
      "set15_frame0    NVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRPSVT\n",
      "                * **.::**.******* ******:**********:*****:***..**::*** *****\n",
      "\n",
      "Maniola_jurtina LQERKRKKIVKRGKAN-\n",
      "set15_frame0    LQERRRKKIVKRGKANM\n",
      "                ****:*********** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Pararge_aegeria G------------------------------------------NNNMKEV-TNTTDVTKK\n",
      "set15_frame0    ALIFWKKSFYHKAVVFFLFLNYLLVQTSNNLMYNLTLCVYLKENKNSKEVPKNTTEATQK\n",
      "                .                                          *:* *** .***:.*:*\n",
      "\n",
      "Pararge_aegeria NEKKVRISENSDCVYGTQVENQASLSTSSNEDELPAKK--HQKPHTPKQRVITKVWPSAT\n",
      "set15_frame0    NVKKNIVSETSDCGYGTLIENQESMSTSSNEDELPSKKPVHKKPHKPKQKVNNKQRPSVT\n",
      "                * **  :**.*** *** :*** *:**********:**  *:***.***:* .*  **.*\n",
      "\n",
      "Pararge_aegeria PQDRKRKKLVKRGKAN-\n",
      "set15_frame0    LQERRRKKIVKRGKANM\n",
      "                 *:*:***:******* \n",
      "\n",
      "Choose Species 1 - 31\n",
      "Exon_6 Match\n",
      "[['Bicyclus_anynana', ['10531108', '10531468']], ['Maniola_jurtina', ['10531108', '10531468']], ['Pararge_aegeria', ['10531108', '10531468']]]\n",
      "Exon_7 Match\n",
      "[['Bicyclus_anynana', ['10530415', '10530606']], ['Maniola_jurtina', ['10530415', '10530606']], ['Pararge_aegeria', ['10530415', '10530606']]]\n",
      "Exon_8 Match\n",
      "[['Bicyclus_anynana', ['10530174', '10530274']], ['Maniola_jurtina', ['10530174', '10530274']], ['Pararge_aegeria', ['10530174', '10530274']]]\n",
      "Exon_9 Match\n",
      "[['Bicyclus_anynana', ['10528966', '10529164']], ['Maniola_jurtina', ['10528966', '10529164']], ['Pararge_aegeria', ['10528966', '10529164']]]\n",
      "Exon_10 Match\n",
      "[['Bicyclus_anynana', ['10528362', '10528573']], ['Maniola_jurtina', ['10528362', '10528573']], ['Pararge_aegeria', ['10528362', '10528573']]]\n",
      "Exon_11 Match\n",
      "[['Bicyclus_anynana', ['10527531', '10527955']], ['Maniola_jurtina', ['10527531', '10527955']], ['Pararge_aegeria', ['10527531', '10527955']]]\n",
      "Exon_12 Match\n",
      "[['Bicyclus_anynana', ['10527023', '10527223']], ['Maniola_jurtina', ['10527023', '10527223']], ['Pararge_aegeria', ['10527023', '10527223']]]\n",
      "Exon_13 Match\n",
      "[['Bicyclus_anynana', ['10524349', '10524526']], ['Maniola_jurtina', ['10524349', '10524526']], ['Pararge_aegeria', ['10524349', '10524526']]]\n",
      "Exon_14 Match\n",
      "[['Bicyclus_anynana', ['3942251', '3942316']], ['Maniola_jurtina', ['3942251', '3942316']], ['Pararge_aegeria', ['3942251', '3942316']]]\n",
      "Exon_15 mismatch\n",
      "[['Bicyclus_anynana', ['10523387', '10523566']], ['Maniola_jurtina', ['10523387', '10523566']], ['Pararge_aegeria', ['10523408', '10523566']]]\n",
      "Proceed?y\n",
      "Bicyclus_anynana Exon_15 13_frame1 1.468 \n",
      "SIIVEDADTMMKLVEAKEARPTPAAPLICVAQPEHLTSPDKQEPANFTGEHPEFDCCDN\n",
      "Maniola_jurtina Exon_15 13_frame1 1.32 \n",
      "SIIVEDADTMMKLVEAKEARPTPAAPLICVAQPEHLTSPDKQEPANFTGEHPEFDCCDN\n",
      "Pararge_aegeria Exon_15 13_frame1 1.706 \n",
      "SIIVEDADTMMKLVEAKEARPTPAAPLICVAQPEHLTSPDKQEPANFTGEHP\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan ------------------------------------SIIAEDEDEVMQVADEELHVSLEA\n",
      "set13_frame1    SCHRSDLDTPLPCSRPLNISRRRLVKKDVNNNVVGRSIIVEDADTMMKLVEAK-----EA\n",
      "                                                    ***.** * :*::.: :     **\n",
      "\n",
      "Bicyclus_anynan KPR---PLVTPAGPVTLMMSTIMSVKQETTNFTIGQAESDYCNN----------------\n",
      "set13_frame1    RPTPAAPLICVAQPEHLTSPD----KQEPANFTGEHPEFDCCDNRYLAIQSGRDPSRHKY\n",
      "                :*    **:  * *  *  .     ***.:***  :.* * *:*                \n",
      "\n",
      "Bicyclus_anynan ---------------------------\n",
      "set13_frame1    SARMYLFAPGPNRNQRITTSTSRRIPK\n",
      "                                           \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina ------------------------------------SIIAEDEDEVMKLEEESVRVSLEA\n",
      "set13_frame1    SCHRSDLDTPLPCSRPLNISRRRLVKKDVNNNVVGRSIIVEDADTMMKLVEAKEARPTPA\n",
      "                                                    ***.** * :*** * .   .  *\n",
      "\n",
      "Maniola_jurtina VPPVRIAT----VLSGKQEPAGFSGQ-AESDYCNN-------------------------\n",
      "set13_frame1    APLICVAQPEHLTSPDKQEPANFTGEHPEFDCCDNRYLAIQSGRDPSRHKYSARMYLFAP\n",
      "                .* : :*     . ..*****.*:*: .* * *:*                         \n",
      "\n",
      "Maniola_jurtina ------------------\n",
      "set13_frame1    GPNRNQRITTSTSRRIPK\n",
      "                                  \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Pararge_aegeria ------------------------------------SIIGENADTVIQQMDIGRVRVSRG\n",
      "set13_frame1    SCHRSDLDTPLPCSRPLNISRRRLVKKDVNNNVVGRSIIVEDADTMMKLVEAKEAR----\n",
      "                                                    *** *:***::: ::  ..*    \n",
      "\n",
      "Pararge_aegeria VICDVTTCNVPIVLPAEPMTLMMSSMSVTQDITMNFP-----------------------\n",
      "set13_frame1    -----PTPAAPLICVAQPEHLTSPDKQEPANFTGEHPEFDCCDNRYLAIQSGRDPSRHKY\n",
      "                     .*  .*::  *:*  *  .. . . ::* :.*                       \n",
      "\n",
      "Pararge_aegeria ----------------VNSPHSINCNA\n",
      "set13_frame1    SARMYLFAPGPNRNQRITTSTSRRIPK\n",
      "                                :.:. * .   \n",
      "\n",
      "Choose Species 1 - 31\n",
      "Exon_16 Match\n",
      "[['Bicyclus_anynana', ['10522891', '10523086']], ['Maniola_jurtina', ['10522891', '10523086']], ['Pararge_aegeria', ['10522891', '10523086']]]\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Coenonympha_glycerion,OY979629.1,10535487,10535583,1,N,Maniola_jurtina_XM_045907399.1_Frame_0_rightoh_1_query_Exon_1,1,33,33\n",
      "Coenonympha_glycerion,OY979629.1,10533898,10534448,1,N,Bicyclus_anynana_XM_052891332.1_Frame_2_rightoh_0_query_Exon_2,1,183,183,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10533204,10533357,1,N,Bicyclus_anynana_XM_052891332.1_Frame_0_rightoh_1_query_Exon_3,1,51,51,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10532545,10532789,1,N,Bicyclus_anynana_XM_052891332.1_Frame_2_rightoh_0_query_Exon_4,1,81,81,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10532030,10532313,1,N,Bicyclus_anynana_XM_052891332.1_Frame_0_rightoh_2_query_Exon_5,1,94,94,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10531108,10531468,1,N,Bicyclus_anynana_XM_052891332.1_Frame_1_rightoh_0_query_Exon_6,1,120,120,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10530415,10530606,1,N,Bicyclus_anynana_XM_052891332.1_Frame_0_rightoh_0_query_Exon_7,1,64,64,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10530174,10530274,1,N,Bicyclus_anynana_XM_052891332.1_Frame_0_rightoh_2_query_Exon_8,1,33,33,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10528966,10529164,1,N,Bicyclus_anynana_XM_052891332.1_Frame_1_rightoh_0_query_Exon_9,1,66,66,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10528362,10528573,1,N,Bicyclus_anynana_XM_052891332.1_Frame_0_rightoh_2_query_Exon_10,1,70,70,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10527531,10527955,1,N,Bicyclus_anynana_XM_052891332.1_Frame_1_rightoh_1_query_Exon_11,1,141,141,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10527023,10527223,1,N,Bicyclus_anynana_XM_052891332.1_Frame_2_rightoh_1_query_Exon_12,1,66,66,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10524349,10524526,1,N,Bicyclus_anynana_XM_052891332.1_Frame_2_rightoh_2_query_Exon_13,1,58,58,Y,N\n",
      "Coenonympha_glycerion,OY979647.1,3942251,3942316,1,N,Bicyclus_anynana_XM_052891332.1_Frame_1_rightoh_2_query_Exon_14,1,21,21,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10523387,10523566,1,N,Bicyclus_anynana_XM_052891332.1_Frame_1_rightoh_2_query_Exon_15,1,59,59,Y,N\n",
      "Coenonympha_glycerion,OY979629.1,10522891,10523086,1,N,Maniola_jurtina_XM_045907399.1_Frame_1_rightoh_0_query_Exon_16,1,71,78\n",
      "\n",
      "Elymnias_hypermnestra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1 Match\n",
      "[['Bicyclus_anynana', ['5063241', '5063337']], ['Maniola_jurtina', ['5063241', '5063337']], ['Pararge_aegeria', ['5063241', '5063337']]]\n",
      "Exon_2 Match\n",
      "[['Bicyclus_anynana', ['5062617', '5063167']], ['Maniola_jurtina', ['5062617', '5063167']], ['Pararge_aegeria', ['5062617', '5063167']]]\n",
      "Exon_3 Match\n",
      "[['Bicyclus_anynana', ['5061538', '5061691']], ['Maniola_jurtina', ['5061538', '5061691']], ['Pararge_aegeria', ['5061538', '5061691']]]\n",
      "Exon_4 Match\n",
      "[['Bicyclus_anynana', ['5060037', '5060269']], ['Maniola_jurtina', ['5060037', '5060269']], ['Pararge_aegeria', ['5060037', '5060269']]]\n",
      "Exon_5 Match\n",
      "[['Bicyclus_anynana', ['5059457', '5059737']], ['Maniola_jurtina', ['5059457', '5059737']], ['Pararge_aegeria', ['5059457', '5059737']]]\n",
      "Exon_6 Match\n",
      "[['Bicyclus_anynana', ['5058769', '5059129']], ['Maniola_jurtina', ['5058769', '5059129']], ['Pararge_aegeria', ['5058769', '5059129']]]\n",
      "Exon_7 Match\n",
      "[['Bicyclus_anynana', ['5057428', '5057619']], ['Maniola_jurtina', ['5057428', '5057619']], ['Pararge_aegeria', ['5057428', '5057619']]]\n",
      "Exon_8 Match\n",
      "[['Bicyclus_anynana', ['5056799', '5056893']], ['Maniola_jurtina', ['5056799', '5056893']], ['Pararge_aegeria', ['5056799', '5056893']]]\n",
      "Exon_9 Match\n",
      "[['Bicyclus_anynana', ['5056354', '5056552']], ['Maniola_jurtina', ['5056354', '5056552']], ['Pararge_aegeria', ['5056354', '5056552']]]\n",
      "Exon_10 Match\n",
      "[['Bicyclus_anynana', ['5055623', '5055834']], ['Maniola_jurtina', ['5055623', '5055834']], ['Pararge_aegeria', ['5055623', '5055834']]]\n",
      "Exon_11 Match\n",
      "[['Bicyclus_anynana', ['5054659', '5055125']], ['Maniola_jurtina', ['5054659', '5055125']], ['Pararge_aegeria', ['5054659', '5055125']]]\n",
      "Exon_12 Match\n",
      "[['Bicyclus_anynana', ['5052710', '5052910']], ['Maniola_jurtina', ['5052710', '5052910']], ['Pararge_aegeria', ['5052710', '5052910']]]\n",
      "Exon_13 Match\n",
      "[['Bicyclus_anynana', ['5052147', '5052330']], ['Maniola_jurtina', ['5052147', '5052330']], ['Pararge_aegeria', ['5052147', '5052330']]]\n",
      "Exon_14 Match\n",
      "[['Bicyclus_anynana', ['5050606', '5050707']], ['Maniola_jurtina', ['5050606', '5050707']], ['Pararge_aegeria', ['5050606', '5050707']]]\n",
      "Exon_15 mismatch\n",
      "[['Bicyclus_anynana', ['5049505', '5049567']], ['Maniola_jurtina', ['5049364', '5049567']], ['Pararge_aegeria', ['464241', '464321']]]\n",
      "Proceed?y\n",
      "Bicyclus_anynana Exon_15 57_frame1 1.368 \n",
      "SLHDDNMDSTSPNKGHNEDE\n",
      "Maniola_jurtina Exon_15 57_frame1 1.43 \n",
      "SLHDDNMDSTSPNKGHNEDEVCISLEATSHPLDETVGSGATLLSPSSMDQEPATFTIGPPESDFCNN\n",
      "Pararge_aegeria Exon_15 45_frame0 1.762 \n",
      "SCAAWLQRLPATLFISSAHLVGGLST\n",
      "Check Alignment??y\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Bicyclus_anynan SIIAED----EDEVMQVA-------DEELHVSLEAKPRPLVTPAGPVTLMMSTIMSVKQE\n",
      "set57_frame1    FIIIHINRSLHDDNMDSTSPNKGHNEDEVCISLEATSHPLDETVGSGATLLSP-SSMDQE\n",
      "                 ** .     .*: *: :       ::*: :****..:**  ..*. : ::*.  *:.**\n",
      "\n",
      "Bicyclus_anynan TTNFTIGQAESDYCNN\n",
      "set57_frame1    PATFTIGPPESDFCNN\n",
      "                .:.**** .***:***\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Maniola_jurtina SIIAED----EDEVMKL-------EEESVRVSLEAVPPVRIATVLSG---------KQEP\n",
      "set57_frame1    FIIIHINRSLHDDNMDSTSPNKGHNEDEVCISLEATSHPLDETVGSGATLLSPSSMDQEP\n",
      "                 ** .     .*: *.        :*:.* :****..     ** **         .***\n",
      "\n",
      "Maniola_jurtina AGFS-GQAESDYCNN\n",
      "set57_frame1    ATFTIGPPESDFCNN\n",
      "                * *: * .***:***\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Pararge_aegeria SIIGENADTVIQQMDIGRVRVSRGVICDVTTCNVPIVLPAEPMTLMMSSMSVTQDI-TMN\n",
      "set45_frame0    --------------------VSRNDVHKARSCAA--WLQRLPATLFISSAHLVGGLSTLR\n",
      "                                    ***. : .. :* .   *   * **::**  :. .: *:.\n",
      "\n",
      "Pararge_aegeria FPVNSPHS----------INCNA-----------------------\n",
      "set45_frame0    RPVRGRHSSTFGLQHPSVLCCQVKSSGLAAKILFRRFLAPPAPPLP\n",
      "                 **.. **          : *:.                       \n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "family_group = \"3.Satyrine\"\n",
    "annotated_genome_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Blast_result\"\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "# species_list = [\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Limenitis_camilla\"]\n",
    "# species_list = [\"Argynnis_bischoffii_washingtonia\",\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Bicyclus_anynana\",\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Vanessa_tameamea\"]\n",
    "for species in species_list:\n",
    "    print(species)\n",
    "    blast_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{blast_location}/{species}\")\n",
    "    query_species_list = []\n",
    "    coordinate_detail_dictionary = {}\n",
    "    coordinate_info_dictionary = {}\n",
    "    scaffold = ''\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\".csv\") and len(file.split(\"_\")) >= 5 and not(file.endswith(\"_old.csv\")):\n",
    "#             print(file)\n",
    "            if \".\" in file[:-4]:\n",
    "                query_species = f\"{file.split('_')[-2].split('.')[1]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            else:\n",
    "                query_species = f\"{file.split('_')[-2]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            query_species_list.append(query_species)\n",
    "            coordinate_detail_dictionary[query_species] = {}\n",
    "            with open(f\"{blast_location}/{species}/{file}\", 'r') as coordinate_file_open:\n",
    "                coordinate_file_lines_list  =  coordinate_file_open.readlines()\n",
    "            if scaffold == '':\n",
    "                scaffold = coordinate_file_lines_list[1].split(\",\")[1]\n",
    "    \n",
    "            for i in range(1,len(coordinate_file_lines_list)):\n",
    "                \n",
    "                coordinate_file_lines_split = coordinate_file_lines_list[i].split(\",\")\n",
    "                start,stop = coordinate_file_lines_split[2],coordinate_file_lines_split[3]\n",
    "                exon = f\"Exon_{i}\"\n",
    "                if exon not in coordinate_info_dictionary:\n",
    "                    coordinate_info_dictionary[exon] = {}\n",
    "                coordinate_info_dictionary[exon][query_species] = [start,stop]\n",
    "                coordinate_detail_dictionary[query_species][exon] = coordinate_file_lines_list[i]\n",
    "            \n",
    "            # break\n",
    "            # print(query_species)\n",
    "            # print(file)\n",
    "    # print(coordinate_info_dictionary)\n",
    "    # assert False\n",
    "    output = coordinate_file_lines_list[0]\n",
    "    for exon,species_specific_coordinates in coordinate_info_dictionary.items():\n",
    "        # print(exon,species_specific_coordinates)\n",
    "        coordinates_list = []\n",
    "        previous_coordinates = \"\"\n",
    "        match = 1\n",
    "        for query_species, coordinates in coordinate_info_dictionary[exon].items():\n",
    "            coordinates_list.append([query_species,coordinates])\n",
    "            if previous_coordinates != '':\n",
    "                if coordinates != previous_coordinates:\n",
    "                    match = 0\n",
    "                    \n",
    "            previous_coordinates = coordinates\n",
    "       \n",
    "        \n",
    "        \n",
    "        if match == 0:\n",
    "            print(exon,\"mismatch\")\n",
    "            print(coordinates_list)\n",
    "                \n",
    "            while True:\n",
    "                choice = input(\"Proceed?\")\n",
    "                if choice[0].lower() == \"y\" or choice[0].lower() == \"n\":\n",
    "                    break\n",
    "            \n",
    "            while True:\n",
    "                if choice[0].lower() == \"y\":\n",
    "                    alignment_detail_dictionary = {}\n",
    "                    for query_species_name in query_species_list:\n",
    "                        list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                        for folders in list_of_folders:\n",
    "                            if folders.endswith(query_species_name):\n",
    "                                # print()\n",
    "                                \n",
    "                                with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "                                    # alignment_detail = query_file_open.readlines()[0]\n",
    "                                    alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                                    alignment_detail_dictionary[query_species_name] = alignment_detail\n",
    "                                with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "                                    distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "                                try:\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query_spliced.txt\", 'r') as new_query_file:\n",
    "                                        new_query = (new_query_file.readlines()[-1].rstrip())\n",
    "                                except:\n",
    "                                    new_query = \"Issue with query\"\n",
    "                                print(query_species_name,exon, alignment_detail, distance_score,f\"\\n{new_query}\")\n",
    "                    while True:            \n",
    "                        check_alignment = input(\"Check Alignment??\")\n",
    "                        if check_alignment[0].lower() == \"y\" or check_alignment[0].lower() == \"n\":\n",
    "                            break\n",
    "                    if check_alignment[0].lower() == \"y\":\n",
    "                        for query_species_name in query_species_list:\n",
    "                            list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                            for folders in list_of_folders:\n",
    "                                if folders.endswith(query_species_name):\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/alignment_clustal_{exon}_translated_genomic_sequence_{alignment_detail_dictionary[query_species_name]}.fa.txt\", 'r') as clustal_alignment_file:\n",
    "                                        clustal_alignment = clustal_alignment_file.readlines()\n",
    "                                    print(\"\".join(clustal_alignment))\n",
    "                                    \n",
    "                                # assert False\n",
    "                    while True:\n",
    "                        try:\n",
    "                            species_number = (input(f\"Choose Species 1 - {len(query_species_list)}\"))\n",
    "                            \n",
    "                            \n",
    "                            if species_number[0].lower() == 'n':\n",
    "                                output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                                break \n",
    "                            output += coordinate_detail_dictionary[query_species_list[int(species_number)-1]][exon]\n",
    "                            break\n",
    "                        except:\n",
    "                            print(\"Retry\")\n",
    "                    break\n",
    "                elif choice[0].lower() == \"n\":\n",
    "                    output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                    break\n",
    "                \n",
    "                \n",
    "        else:\n",
    "            print(exon, \"Match\")\n",
    "            print(coordinates_list)\n",
    "            output += coordinate_detail_dictionary[query_species_list[0]][exon]\n",
    "        \n",
    "       \n",
    "\n",
    "        # assert False\n",
    "    # print(coordinate_detail_dictionary)\n",
    "    print(output)\n",
    "    with open(f\"{blast_location}/{species}/final_coordinates.csv\", \"w\") as out_file:\n",
    "        out_file.write(output)\n",
    "    \n",
    "    # 'Coenonympha_glycerion,OY979626.1,7015875,7015963,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7892a6a6-9400-414d-bbaa-cf893036e0d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
