{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b2fddd4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coenonympha_glycerion\n",
      "GCA_963855885.1_ilCoeGlyc1.1_genomic.fna\n",
      "10535487 10535583\n",
      "Getting Gene\n",
      "MEWVLRSPQIHNTFSDLGFALPDGYHINDSCN\n",
      ">Coenonympha_glycerion_Exon_1_OY979629.1_10535487_10535583_left_0_right_1\n",
      "ATGGAGTGGGTACTCAGAAGTCCCCAAATTCACAACACATTCAGCGACCTCGGATTTGCACTTCCAGATGGATATCACATAAACGATAGTTGCAACT\n",
      "10533898 10534448\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_2_OY979629.1_10533898_10534448_left_2_right_0\n",
      "CCGCTTTGGAGACCATTCTTCAAAACCTTGTGACGGAAGACAAATACTCGCGCTCATATCGCAGAAGTATTAGCTTCGgacaaaatatcaaaaaagACCTTATACCTGTGTTAATCCACACCAAAGATGATAAAACCGTTGAGTTGTTAATCAAAATATTGGTTAATTTAACGATTCCGGTGGAGTGTTTACTTTCGTTGGAAACAGTTTCACAAACCGAAATCGGGAGACACGCGATTTTAGAAATAAACAATTTGCTCTCTTGCACCAAAATTGCGTTTACCGATCACCATGTGGCAAAAGTCGTTGTGGACTTGTTGAAGAAGAACGCCGAGAATGATGAAAAGGGAAAATTGACGACAGAAAATTGTACGAATATAAGCAACACCCTTTTGTTTCTGCGTAATTTGCTCCACATACCGGAGGACATTGATTGTCACAGCCCGACGTACAGTGGCCCACCTCATACGATTCAGAACCAGATATTATGGAATTTATTCAGCCACAGTATCGACAAACTACTCATAAAATTGATGACAATTCCGGATGCG\n",
      "10533204 10533357\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_3_OY979629.1_10533204_10533357_left_0_right_1\n",
      "ATAAACTGGGGCGTGACAATGGTGCAGTTCATTGCCCTGCTGTACAAAGACCAGCATGTGTCCACTTTGCACAAACTGCTCAATTTGTGGCTCGAAACGTCACTCTCCGAAAGTTCCGAAGATAACGAAAGTAATACATCACCACCAAATAGGG\n",
      "10532545 10532789\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_4_OY979629.1_10532545_10532789_left_2_right_0\n",
      "GGAATAGTGATTGTTCGCCTATGCTAACGTCGGAACCGACGTCCGATTCGTCGGACACGGGGGCCAGTGGCAAGTGCAACGGAGAGGCCAACTTTGTAGACAACAATAGCTGGGACGACTCCTCGAGGAACGGCAACAGTGTTAAACAAACCTTCCAGGCGCAAAACGCAGATAATGCTCCGAGCGGGCCGTGTGAGGGCGGACAAACTAAATTCGCAGAACCACGCAAAGAATCTAACGAAACA\n",
      "10532030 10532313\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_5_OY979629.1_10532030_10532313_left_0_right_2\n",
      "gaaaataaaaattctaaggAAGTTCCAAAAAACACAACTGAAGCTACGCAGAAAAATGtcaagaaaaatattgtttctgaAACTTCCGACTGTGGTTACGGCACTCTTATTGAAAACCAAGAATCAATGTCGACGTCTAGCAACGAAGACGAACTGCCGTCTAAGAAGCCTGTTCACAAGAAACCGCATAAACCCAAACAGAAGGTAAACAACAAGCAGCGGCCGAGTGTCACCCTTCAGGAACGTAGAAGAAAGAAAATTGTGAAGAGGGGCAAAGCTAACAT\n",
      "10531108 10531468\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_6_OY979629.1_10531108_10531468_left_1_right_0\n",
      "AATAAACGTGCAGGGTCTATCGCACAGCACGCCGACCGACGACGATATATCCAACGTACTGAAAGAGTTCACGGTTGATTTTCTTCTGAAGGGCTACAACTCTTTGGTTCGAACTCTGCACTCCCAGATACTGACGAACGCGCAGTTCGAAATCGACACTTCCCACTTCTTCTGGCTCGTGACGTACTTCCTGAAGTTCGCGACTCAGATCGAGCTGGACCTGGAGCACGTGGGCAGCGTGCTGTCGTTCGAGATCGTGTCGTACTTGACTGCGGAAGGGGTGCACCTGTGCGAGCAGTTCGAGCTCGCCATCAAGCTAGACGGAAACGATCTCAAGCCCAGCACGAGGAGACTGCATTTG\n",
      "10530415 10530606\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_7_OY979629.1_10530415_10530606_left_0_right_0\n",
      "GTCGTAACTGCAATCCGGGAATTCATTCAAGCCATCGAAGTGTACAAAAAGTCCGCGCATATTTGCAACAACGAGAAAGAAATACTCGACAAGTTGCAAATCAAGATGTTCGAAACAGACGAGCTGCGCTTGCTGCTCGTGCTGTTGCTGAGATCCTACAACCCCAAGTACCAGTCCAAGCAGTATCTGCAG\n",
      "10530174 10530274\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_8_OY979629.1_10530174_10530274_left_0_right_2\n",
      "GACCTGGTCGTGACGAATCATATTCTTCTGTTGTTCATGGAAAGTGTTATGAAGTTGCCAAATTACAAGGGTCCTGTTACCATGGTCGATCATTTAAAACA\n",
      "10528966 10529164\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_9_OY979629.1_10528966_10529164_left_1_right_0\n",
      "GTTCTGCGCGCCCGACATAATGGAGCAGTTCGGGCTGCTGCTGGTGGAGTACGAGTCCAACGGCGAGTTCGTGAACGACTGCATCTTCACGCTGATGCACCACGTGGGCGGCGAGCTGGACAGCTTGATCTCACTGTTCCAGCCCAAGATACTCAAGACTTTCACTTTGATTTGGAAGTCGGAGTTTGAAGTTTGCGAC\n",
      "10528362 10528573\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_10_OY979629.1_10528362_10528573_left_0_right_2\n",
      "GATTGGTCAGATCTAATTGAGTATGTGATAAACACATTCGTGAAGAAGCCGTATAAACTGCAAAGCTCCGCACAAATCCACTTGAACGTTGCAATCATCGGGAACGACTCAAGATTCAAAGAAATGCTAGGCTTAGCCGAGCACACGTACGAAGAGGACGAAGAGTTGACTCTGCAAGATGATCTGACCACCGAAGTCAATCATTCTAAAGG\n",
      "10527531 10527955\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_11_OY979629.1_10527531_10527955_left_1_right_1\n",
      "TTGGAGTAAAGATGAACTATCAACGCTGAGCTGGAATTACTTGCAATGCAATGAGTCGCCAGATGTGGTCGGGGAAATCGTGAAGATGTTCCGAGAGGACGGGGTCGTGAAGAGTCGCGAGTCGGTCGTAAAAGAGCTCTACAATCAATTCATCATCAACAAAGAGCAGCTCGACAAACTCTCCAAAAAAGAAACCGAGGAAACGAGGGAGAATGATGTCGGAAAGATTTGCGAGCAACTCATACAGGACGGGAAATCTGTGTTCTTGGACTGGGTCCAAACGGTGATCCTGGAAACTTGTACTGCCAAAATATACTTGGAGAAAAAGGCTTCGAATACACGTAACGCACAGTCATGCGAGGAGTTATCAAAGAAAGTACCTGAAGATTTCCCAGTCATATCCTCCGTGTCCTATCATTCATTAC\n",
      "10527023 10527223\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_12_OY979629.1_10527023_10527223_left_2_right_1\n",
      "ttcgtAACCAATTTGTGGCACTGGTTCCATGGAATTGCCAGCAGGCGTCCATTTGCAAGGATTTGAAGTTTCTGCAACTTTTACACCGGCTTGGATTCCACATGCCAGTTGATACTGGCAAAGTTTTCATTAGAATACCGCACTACTGGACACCGGATATCCTGTATGAAATAGCCGCCAAAATATCACCTATAGATGTTT\n",
      "10524349 10524526\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_13_OY979629.1_10524349_10524526_left_2_right_2\n",
      "ctaaattaaaattttcaatgaaCGATATCTCCCAAAGTAGTATTCTGAGATCAGATCAAACGAGCAATTTGCTTCTGCCGGAGCTCTCGAAGGTGCTGTCGCCGATGGCTGACAGCTTCTATCAGATCCACAACAAGAAACAAAGCCACCTTGCATCGATTGTcaactttacgcccat\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"int\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 62\u001b[0m\n\u001b[1;32m     60\u001b[0m     query_name \u001b[38;5;241m=\u001b[39m coordinate_line_split[\u001b[38;5;241m6\u001b[39m]\n\u001b[1;32m     61\u001b[0m     exon \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExon_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mquery_name\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m---> 62\u001b[0m     error_exons \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mscaffold\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,000,000,0,Y,Error_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mexon\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m1\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,00,00,00\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     64\u001b[0m gene_sequence \u001b[38;5;241m=\u001b[39m get_gene_sequence(genome_location, \n\u001b[1;32m     65\u001b[0m                                   species, \n\u001b[1;32m     66\u001b[0m                                   genome_file, \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m                                   complement,\n\u001b[1;32m     71\u001b[0m                                   output_location)\n\u001b[1;32m     74\u001b[0m current_exon \u001b[38;5;241m=\u001b[39m query_name\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\n",
      "\u001b[0;31mTypeError\u001b[0m: can only concatenate str (not \"int\") to str"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"3.Satyrine\"\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "make_output_folder(output_location)\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "if 'desktop.ini' in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Heliconius_charithonia\"]\n",
    "# list_of_species = [\"Vanessa_tameamea\"]\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "        \n",
    "            \n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            query_name = coordinate_line_split[6]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "        \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "       \n",
    "        current_exon = query_name.split('_')[-1]\n",
    "        if current_exon == \"1\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "            print(translated)\n",
    "            if \"*\" in translated:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "        \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "        elif current_exon == \"16\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "\n",
    "            if \"*\" in translated[:-1]:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                print(translated)\n",
    "            else:\n",
    "                \n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "#         assert False\n",
    "        else:\n",
    "            exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "            print(exon_output)\n",
    "\n",
    "            write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
