{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"3.Satyrine\"\n",
    "number_of_exons = 16\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm MEWVLSGPQTHNSFSNLGYALPD-GYHINENCN\n",
      "Lasiommata_mege MEWVLRSPQIHNSFSNLGYSLPD-GYHINENCN\n",
      "Pararge_aegeria MAWVLRSPQIHNNFSDLGYALPD-GYHINDNCN\n",
      "Erebia_aethiops MEWVLRSPQIHNTFSNLGFPLPDGGYHINENCN\n",
      "Erebia_ligea_Ex MEWVLRSPQIHNTFSNLGFPLPDGGYHINENCN\n",
      "Maniola_hyperan MEWVLRSPQIHNTFSNLGFPLPDGGYHINENCN\n",
      "Maniola_jurtina MEWVLRSPQIHNTFSNLGFPLPDGGYHINENCN\n",
      "Hipparchia_seme MEWVLQSSQIHNTFSNLGYPLPDGGYHINENCN\n",
      "Melanargia_gala MEWVLRSPQIHNTFSNLGYALPDGGYHINDNCN\n",
      "Oeneis_ivallda_ MEWVQQSSQIHSTFSNLGYSLPNGGYHINENCN\n",
      "                * **  ..* *..**:**:.**: *****:***\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm TLEEILENLVTEDRYLRTYRRSISFGQNIKKDLIPLLLYTKDDKTVELLIKILVNLTIPV\n",
      "Erebia_aethiops ALESIFQSLTTEDKHLRTYRRSISFGQNIKKDLKPLLINTKDDKTTELLIKIFVNLTIPV\n",
      "Erebia_ligea_Ex ALESIFQNLTTEDKFLRTYRRSISFGQNIKKDLKPLLINTKDDKTTELLIKIFVNLTIPV\n",
      "Maniola_jurtina ALESIFQNLTTEDKHLRTYRRSISFGQNIKKDLIPLLINTKDDKTVELLIKIFVNLTIPV\n",
      "Maniola_hyperan ALESIFHNLITEDKHLRTYRRSISFGQNINKDLVPLLINTKDGKTIELLIKIFVNLTIPV\n",
      "Hipparchia_seme ALESILENLATEDKYLRTYRRSISFGQNIKKDLIPLLINTKDDKAIELLTKIFVNLTIPV\n",
      "Oeneis_ivallda_ ALESILENLATEDKYLRTYRRSISFGQNIKKDLIPLLINTKDDKAIELLTKIFVNLTIPV\n",
      "Melanargia_gala ALESILQNLATEDKHLRPYRRSISFGQNIKRDLTPLLINTKDDKAIELLIKILVNLTIPV\n",
      "Lasiommata_mege VLETILQNLVTEDKYLRTYRRSISFGQNIKKDLIPLLIYTRDDNTIELLIKILVNLTIPV\n",
      "Pararge_aegeria ALESILQNLVTEDKYLRTYRRSISFGQNIKKDLIPILINTKDDNTVELLIKILVNLTIPV\n",
      "                .** *:..* ***:.**.***********::** *:*: *:*.:: *** **:*******\n",
      "\n",
      "Elymnias_hyperm ECLLSLETVSKNDTGRHTIFEINSLLSSTKVALTDHRVTKVIMDFLKKNFDNDQIFKLSV\n",
      "Erebia_aethiops ECLLSLESISRTDIGRHTIFEINNLLSATKIALTDHRVTKIIIEFLKKNCENEQKGRLST\n",
      "Erebia_ligea_Ex ECLLSLESISRTDIGRHTIFEINNLLSATKIALTDHRVTKIIIEFLKKNSENEQKGKLST\n",
      "Maniola_jurtina ECLLSLDSVSRSDIGRHTIFEINNLLSATKIALTDHRVTKVIIEFLKKNSENEQKGKLST\n",
      "Maniola_hyperan ECLLSLDSVSRTDIGRHTIFEINNLLSSTKIALTDHRVTKVIIEFLKKNSENEQKGKLSP\n",
      "Hipparchia_seme ECLLSLESVSKTDIGRHTIFEINSLLSATKIALTDHRVTKVNIEFLKKNSENEQMGKLST\n",
      "Oeneis_ivallda_ ECLLSLELVSKTDIGRHTIFEINSLLLATKIAFTDYRVTKVNIEFLKKNSANEQMGKLST\n",
      "Melanargia_gala ECLLSLELVSKTDIGRHTIFEINNLLSATKIALTDHRVTKVIIEFLKKNSENEPKAKLSM\n",
      "Lasiommata_mege ECLLCLELVSRTDTGRHTIFEINSLLSSTKIAVTDQRVTKVIIDFLKKNADNEQKAKLSI\n",
      "Pararge_aegeria ECLLSLELISRTDSGRHAIFEINNMLSSTKTALTDHRVTKVIINFLKKNAKNEPKAKLSI\n",
      "                ****.*: :*:.* ***:*****.:* :** *.** ****: ::*****  *:   :** \n",
      "\n",
      "Elymnias_hyperm EQCTNISNTLLFLRNILHIPEDVNPQNPMFNGPPHTVQNQLIWNLFSQSIDKVLIKLMTI\n",
      "Erebia_aethiops EQCTNISNTLLILRNILHIPEDVNCHSPTYSGPPHTLQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Erebia_ligea_Ex EQCTNITNTLLILRNILHIPEDVNCHSPTYSGPPHTIQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Maniola_jurtina EQCTNISNTLLILRNILHIPEDVNCHSPTYSGPPHTIQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Maniola_hyperan EQCTNISNSLLILRNILHIPEDVNCHSPTYSGPPHTIQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Hipparchia_seme EQCTNISNTLLILRNILHIPEDVNCHSPTYSGPPHTIQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Oeneis_ivallda_ QQCTNISNTLLILRNILHIPEDVNCHSPTYSGPPHNVQNQLIWNLFSQSIDKVLIKLMII\n",
      "Melanargia_gala VQCTNINNTLLILRNILHIPEDVNYNSPTYSGPPHTIQNQLLWNLFSQSIDKVLIKLMTI\n",
      "Lasiommata_mege EQCTNISNTLLFLRNILHIPEDINCHSATYSGPPHTIQNQLLWNLFSHSIDKVLIKLMTI\n",
      "Pararge_aegeria EQCTNISNTLLFLRNIMHIPEDITCHTSPHGGPPHTVHNQLLWNLFSHSIDKVLIKLMTI\n",
      "                 *****.*:**:****:*****:. :.. ..****.::***:*****:********** *\n",
      "\n",
      "Elymnias_hyperm PAA\n",
      "Erebia_aethiops PDA\n",
      "Erebia_ligea_Ex PDA\n",
      "Maniola_jurtina PDA\n",
      "Maniola_hyperan PDA\n",
      "Hipparchia_seme PDA\n",
      "Oeneis_ivallda_ PDA\n",
      "Melanargia_gala PDA\n",
      "Lasiommata_mege PDA\n",
      "Pararge_aegeria PDA\n",
      "                * *\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm INWGVTMVQLIALLYKDQHVTTLHKLLNLWLEATLSESSEDNESNTSPPNR\n",
      "Erebia_aethiops INWGVSMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Erebia_ligea_Ex INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Maniola_hyperan INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Maniola_jurtina INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Hipparchia_seme INWGVTMVQIIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Oeneis_ivallda_ INWGVTMVQIIALVYKDQHVTTLHKLLNIWLEASLSESSEDNESNTSPPKR\n",
      "Lasiommata_mege INWGVTMVQLIALFYKDQHVTALHKLLNLWLEASISESSEDNESNTSPPNR\n",
      "Melanargia_gala INWGVTMVQLIALLYKDQHVTILHKLLNIWLEASMSESSEDNESNTSPPKG\n",
      "Pararge_aegeria INWGVTMVQLIALFYKDQHVTTLHKLLNLWLESSICESSEDNESNTSPSYR\n",
      "                *****:***:***.******* ******:***:::.************.  \n",
      "\n",
      "continue?y\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm SDHSSPMLTSDPTSDSSDTAASGKCNAVNNST-NNAWDDTSRNSESDKQNSQLENTDNIA\n",
      "Erebia_aethiops SDDSSPMLTSDPTSDSSDTGASGKCNGEPNSV-NNGWDDSSRNSESEKQNFQSRNTNNVT\n",
      "Erebia_ligea_Ex SDDSSPMLTSDPTSDSSDTGTSGKCNGEPNSVNNNGWDDSSRNSEGEKQNFQSQNTKNVT\n",
      "Maniola_hyperan SDDSSPMLTSDQTSDSSDTGASGKCNGEPNSV-NNGWDDSSRNSGSEKQNFQSQNTNNMP\n",
      "Maniola_jurtina SDDSSPMLTSDPTSDSSDTGASGKCNGEPNSV-NNGWDDSSRNSGSEKQNFQPQNTNNVP\n",
      "Hipparchia_seme SDDSSPMLTSDPTSDSSDTGISGKCNGEPNSG-NNGWDDSSGNSETDKQ-----NTNNAP\n",
      "Oeneis_ivallda_ SDDSSPMLTSDPTSDSSDTGISGKCNGERNSG-NNGWDDSSGNSETDKQ-----NTNNAP\n",
      "Melanargia_gala SDDSSPMLTSDPTSDSSDTGISGKSNGAPNSA-NSSWDDSSRNSGSDKQNFH--------\n",
      "Lasiommata_mege SDDSSPMLTSDPTSDSSDTGASGKCNGVSNSV-SKAWDDSARNNGNGKHNFQSQNADNDP\n",
      "Pararge_aegeria SDDSSLMDTSDPTSDSSDTGANRKSNRTSNLV-NNGWNLTPMNGENVKQNFQPQNTTTDP\n",
      "                **.** * *** *******. . *.*   *   ...*: :. *.   *:           \n",
      "\n",
      "Elymnias_hyperm K-ESSE---ANNIKKPEASTKQ\n",
      "Erebia_aethiops NRDSSEGDRSNFVDNHEKSCKK\n",
      "Erebia_ligea_Ex NSESSEGDRSNFVDNREKSCKK\n",
      "Maniola_hyperan N-DSCESDRSNSVDNRNSSCNK\n",
      "Maniola_jurtina N-DSCESDRSNFVDNRDNSCKK\n",
      "Hipparchia_seme Y-DSCEGDRSKFVNNREQPCKK\n",
      "Oeneis_ivallda_ Y-NSCEGDRSKFVHNREQPCKK\n",
      "Melanargia_gala ---ACKADRTNFVNNREHSCKM\n",
      "Lasiommata_mege K-TSSQGDQSNFVDNCEESREK\n",
      "Pararge_aegeria K-NSIQGGQSNLADDRRNSCEK\n",
      "                   : :   ::  ..   . : \n",
      "\n",
      "continue?y\n",
      "Exon_5.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm -EDECKDNGTNTSNNNVKKGKSVTISENSDCGYGTQIENQESISTSSNEDEVPTKKPVHQ\n",
      "Erebia_aethiops -DSNYMKEGTDTNNVTKNNEKKTVISENSDCGYGTQVENQESISTSSNEDELPAKKPVHQ\n",
      "Erebia_ligea_Ex -DTNNMKEGTDTNNVTKNNEKKTVISENSDCGYGTQVENQESISTSSNEDELPAKKPVHQ\n",
      "Maniola_hyperan -ENNYMKEGTNTNNVMKNNEKKTVISENSDCGYGTQIENQESISTSSNEDELPAKKPVHQ\n",
      "Maniola_jurtina -ENNYMKDGTNTKNVTKNNEKKTVISENSDCGYGTQIENQESISTSSNEDELPAKKPVHQ\n",
      "Hipparchia_seme -ENDYTKEDKNTTNVSKNTDMKTAISINSDCGYGTQMENQESISTSSNEDELPAKKPVHQ\n",
      "Oeneis_ivallda_ -ENDYTKEDKNTTNVSKNTDMKAAISINSDCGYGTQMENQESISTSSNEDELPTKKPVHK\n",
      "Melanargia_gala -EKNYMKEETNITNVSKNIDMKTVISMNSDCGYGTQMENQESISTSSNEDELPAKKPVHQ\n",
      "Lasiommata_mege GNNNCMKDSSNTTNATKNNEKKTVISENSDCGYGTQIENRGSISTSSNEDELPAKK--HQ\n",
      "Pararge_aegeria -GNNNMKEVTNTTDVTKKNEKKVRISENSDCVYGTQVENQASLSTSSNEDELPAKK--HQ\n",
      "                   :  .: .: .:   :   .. ** **** ****:**: *:********:*:**  *:\n",
      "\n",
      "Elymnias_hyperm KPHTSKQRVNNKARPNVTLQERKRKKFGKKSKSN\n",
      "Erebia_aethiops KPHTPKQRINIKVRPSVTLQERKRKKIVKRGKAN\n",
      "Erebia_ligea_Ex KPHTSKQRINNKVRPSVTLQERKRKKIVKRCKAN\n",
      "Maniola_hyperan KPHTTKQRINNKSRPSVTLQERKRKKIVKRGKAN\n",
      "Maniola_jurtina KPHTSKQRINNKVRPSVTLQERKRKKIVKRGKAN\n",
      "Hipparchia_seme KPHTAKQRINNKVRPSVTLQERKRKKIVKRGKAN\n",
      "Oeneis_ivallda_ KPHNAKQRTNNKVRPSVTLQERKRKKIVKRGKAN\n",
      "Melanargia_gala KPHTCKQRVNNKVRPSVTLQERKRKKIVKRGKAN\n",
      "Lasiommata_mege KPHASKQRINNKVRPSVTPQERKRKKLVKRGKAS\n",
      "Pararge_aegeria KPHTPKQRVITKVWPSATPQDRKRKKLVKRGKAN\n",
      "                ***  ***   *  *..* *:*****: *: *:.\n",
      "\n",
      "continue?y\n",
      "Exon_6.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm INAQGLAQSMPTDDDISNVLKEFTIDFLLKGYNALIRTLHSQILTNVELEIDTSHFFWLL\n",
      "Erebia_aethiops INVQGLSHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Hipparchia_seme INVQGLSHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Maniola_jurtina INVQGLSHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Melanargia_gala INVQGLSHNTPTDDDISNVLKEFTVDFLLRGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Maniola_hyperan INVQGLSHTTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHLQILTNVELEIDTSHFFWLV\n",
      "Erebia_ligea_Ex INVQGLSHNTPTDDDISIVLKEFTVDFLLKGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Oeneis_ivallda_ INVQGLTHNTSTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNVLLEIDTSHFFWLV\n",
      "Lasiommata_mege TNVQGLSHKMPTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNVQLEIDTSHFFWLV\n",
      "Pararge_aegeria TNVQGLSHNMPTDDDISNVLKEFTVDFLLKGYNSLVRTLHSQILTNAHLEIDTSHFFWLV\n",
      "                 *.***::. .****** ******:****:***:*:**** *****. ***********:\n",
      "\n",
      "Elymnias_hyperm TYFLKFATQIELDLEHVSCVLSYEIVSYLTAEGVNLCEQFELAIKLDRSDLKPCVRRLHL\n",
      "Erebia_aethiops TYFLKFATQIELDLEHVCCVLSFEVVSYLTAEGVNLCEQFELAIKLDGNYLKPIIRRLHL\n",
      "Hipparchia_seme TYFLKFATQIELDLEHVCCVLSFEIVSYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Maniola_jurtina TYFLKFATQIELDLEHVCCVLSFEIMSYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Melanargia_gala TYFLKFATQIELELEHVCCVLSFEIVSYLTAEGVNLCEQFQLAIKLDGNDLKPSIRRLHL\n",
      "Maniola_hyperan TYFLKFATQIELDLEHVCCVLSFEIVSYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Erebia_ligea_Ex TYFLKFATQIELDLEHVCCVLSFEVVSYLTAEGVNLCEQFELAIKLDGNDLKPITRRLHL\n",
      "Oeneis_ivallda_ TYFLKFATQIELDLEHVSCVLSFEVVSYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Lasiommata_mege TYFLKFATQIELDLEHVCCILSYEIVSYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Pararge_aegeria TYFLKCATQIELDLEHVCCILSFEIVSYLTAEAVNLCEQFELAIKLDGNNLKPSKRRLHL\n",
      "                ***** ******:****.*:**:*::******.*******:****** . ***  *****\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "continue?y\n",
      "Exon_7.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Elymnias_hyperm VVTAIREFVQAIEVYKKSANTSNNHNEQLDKLQLQMCETDDLRSLFVLLLKYYNPRYHSK\n",
      "Erebia_aethiops VVTAIREFVTANDVYKKSAHICNNDKDELDKLQVKMCETDELRSLLVLLLRYYNPKYHSK\n",
      "Erebia_ligea_Ex VVTAIREFVTANDVYKKSAHMCNNDKEELDKLQVKMCETDELRSLLVLLLRYYNPKYHSK\n",
      "Hipparchia_seme VVTAIREFVQAIEVYKKSAHICNNDKEELDKLQIKMCETDELRSLLVLLLRYYNPKYHSK\n",
      "Melanargia_gala VVTAIREFVQAIEVYKKSAHICNNDKEELDKLQIKMCETDELRSLLVLLLRSYNPKYHSK\n",
      "Maniola_hyperan VVTAIREFVQAIEVYKKSAHICNNDKHELDKLQVKMCETDELRSLLVLLLRYYNPKYHSK\n",
      "Maniola_jurtina VVTAIREFVQAIEVYKKSAHICNNDKDELDKLQVKMCETDELRSLLVLLLRYYNPKYQSK\n",
      "Lasiommata_mege VVTAIREFVQAIEVYKKSAHICNNDRELLDKLQLKMCETDELRSLLVLLLRYYNPKHHSK\n",
      "Oeneis_ivallda_ VVTAIREFFQAIEVYKKSAHICNNDKEELDKLQIKMCETDELRSLLVLLLRRYNQKYHSK\n",
      "Pararge_aegeria VVTAIREFIQAIDTYKKSAQICNNDKELLDNLQLKICETDELRSLLVLLLRYYNPKYQTK\n",
      "                ********. * :.*****: .**... **:**:::****:****:****: ** ::::*\n",
      "\n",
      "Elymnias_hyperm QYLQ\n",
      "Erebia_aethiops QYLQ\n",
      "Erebia_ligea_Ex QYLQ\n",
      "Hipparchia_seme QYLQ\n",
      "Melanargia_gala QYLQ\n",
      "Maniola_hyperan QYLQ\n",
      "Maniola_jurtina QYLQ\n",
      "Lasiommata_mege QYLQ\n",
      "Oeneis_ivallda_ QYLQ\n",
      "Pararge_aegeria QYLQ\n",
      "                ****\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
