{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "81fcfca7-185d-4284-bb3f-d800723fc4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b257780-eea0-47d9-b846-654cff8fa8e1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9993f28b-5774-4b4f-a38f-63c393b15b08",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length):\n",
    "    for offset in range(3):\n",
    "        translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "        for i in range(len(translated_sequence)):\n",
    "            if len(translated_sequence[i])> 0.8*query_length:\n",
    "                sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "                # print(i+1, offset)\n",
    "                \n",
    "                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                    output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "236f726f-7cfc-4839-871c-505018689085",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_mafft(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    location = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\n",
    "    list_of_files_to_run_mafft_on = os.listdir(location)\n",
    "    for file in list_of_files_to_run_mafft_on:\n",
    "        if file.endswith(\".fa\"):\n",
    "            command = f'\"mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "            # print(command)\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            # os.system(f'{command}')\n",
    "            command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/{file}\" > \"{location}/alignment_clustal_{file}.txt\"'\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            \n",
    "            # os.system(f'{command}')\n",
    "    return(location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "df06ca7c-0540-4e5e-a02b-2da57d4eecfb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def process_mafft_output(mafft_run_folder, error_exon):\n",
    "    list_of_files_in_mafft_run_folder = os.listdir(mafft_run_folder)\n",
    "    score_output = []\n",
    "    score = 99\n",
    "    min_score_sequence = ''\n",
    "    alignment_file = ''\n",
    "    for file in list_of_files_in_mafft_run_folder:\n",
    "        if file.endswith(\".fa.hat2\"):\n",
    "            with io.open(f\"{mafft_run_folder}/{file}\", 'r') as dist_matrix_file:\n",
    "                dist_matrix_list = dist_matrix_file.readlines()\n",
    "            \n",
    "            sequence_name = dist_matrix_list[-2].rstrip().split(\"=\")[1]\n",
    "            distance_score = float(dist_matrix_list[-1].rstrip())\n",
    "            if len(score_output) < 5:\n",
    "                score_output.append( [sequence_name,distance_score])\n",
    "            else:\n",
    "                for i in range(len(score_output)):\n",
    "                    score_at_this_index = score_output[i][1]\n",
    "                    if distance_score < score_at_this_index:\n",
    "                        score_output[i] = [sequence_name,distance_score]\n",
    "                        break\n",
    "            if distance_score < score:\n",
    "                score = distance_score\n",
    "                min_score_sequence = sequence_name\n",
    "                alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "    # print(score_output)\n",
    "    print(f\"min = {min_score_sequence}, {score}\" )\n",
    "    # score_out_merged = '\\n'.join(score_output)\n",
    "    print(f\"5 top scores:\\n{score_output}\")\n",
    "    print(f\"{mafft_run_folder}/{alignment_file}\")\n",
    "    \n",
    "    alignment_file = SeqIO.parse(f\"{mafft_run_folder}/{alignment_file}\", 'fasta')\n",
    "    # print (records.id)\n",
    "    fasta_start_position = 0\n",
    "    fasta_end_position = 0\n",
    "    start_switch = 0\n",
    "    end_switch = 0\n",
    "    alignment_name = ''\n",
    "    for records in alignment_file:\n",
    "        \n",
    "        \n",
    "        \n",
    "        if error_exon in records.id:\n",
    "            gap_counter = 0\n",
    "            base_counter = 0\n",
    "            for current_position in range(len(records.seq)):\n",
    "                sequence_length = len(records.seq) - records.seq.count('-')\n",
    "                # print(f\"fasta_end_position {fasta_end_position} fasta_start_position {fasta_start_position}\")\n",
    "                # print(f\"current_position = {current_position}, {len(records.seq)}\")\n",
    "                # print(records.seq[current_position])\n",
    "                # print(\"fasta_start_position\",fasta_start_position)\n",
    "                # print(\"start_switch\",start_switch)\n",
    "                \n",
    "                # print(gap_counter, base_counter)\n",
    "                # print(5,0.2*sequence_length)\n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and gap_counter > 3 and (len(records.seq[:current_position]) - records.seq[:current_position].count('-')) < (0.1*sequence_length):\n",
    "                #  :\n",
    "                    # print(\"\\n\\nhere\\n\\n\")\n",
    "                    start_switch = 0\n",
    "                    gap_counter = 0\n",
    "\n",
    "                if \"-\" not in records.seq[current_position] and start_switch == 0 :\n",
    "                    fasta_start_position = current_position\n",
    "                    start_switch = 1\n",
    "                \n",
    "                    \n",
    "                \n",
    "                    \n",
    "                if end_switch == 1 and \"-\" not in records.seq[current_position] :\n",
    "                    end_switch = 0\n",
    "                    gap_counter = 0\n",
    "                \n",
    "                    \n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0 and (len(records.seq[current_position:]) - records.seq[current_position:].count('-')) < 0.1*sequence_length:\n",
    "                    # print(f\"base_counter {base_counter}\")\n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    fasta_end_position = current_position\n",
    "                    end_switch = 1\n",
    "                if \"-\" in records.seq[current_position]:\n",
    "                    gap_counter += 1\n",
    "                else:\n",
    "                    gap_counter = 0\n",
    "                    base_counter += 1\n",
    "        else:\n",
    "            alignment_name = records.id\n",
    "            if fasta_end_position == 0:\n",
    "                \n",
    "                fasta_end_position = current_position\n",
    "            end_switch = 1\n",
    "            start_switch = 1\n",
    "\n",
    "        if start_switch == 1 and end_switch == 1:\n",
    "            print(fasta_start_position, fasta_end_position)\n",
    "            gene_sequence = records.seq[fasta_start_position:fasta_end_position]\n",
    "            print(f\"{records.id}\\n{gene_sequence}\")\n",
    "    return(gene_sequence, alignment_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "87615294-93f0-483b-b9f7-fad66730cd1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n",
    "def check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    \n",
    "    list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/\")\n",
    "    \n",
    "    if error_exon not in list_of_exon_directories:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "    elif error_exon in list_of_exon_directories:\n",
    "        list_of_folders_1 = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        if \"for_alignment\" not in list_of_folders_1:\n",
    "            os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        for file in list_of_files:\n",
    "            os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50742e4c-f9ab-433d-96a1-bde1f9d1c255",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "    print(\"Getting Gene\")\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp\")\n",
    "\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    # subprocess.run(\"pwd\")\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3cb097ff-9369-40fd-9ae5-7587c792ba31",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mafft_process(previous_exon_coordinates,\n",
    "                  next_exon_coordinates,\n",
    "                  current_exon_coordinates,\n",
    "                  query_species,\n",
    "                  query_transcript,\n",
    "                  query_location,\n",
    "                  annotated_genome_location,\n",
    "                 genome_location):\n",
    "    query_exon = current_exon_coordinates[6].split(\"query\")[-1][1:]\n",
    "    error_exon = query_exon\n",
    "    # print(\"AA\")\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    upstream_exon_line, downstream_exon_line = previous_exon_coordinates.split(\",\"),next_exon_coordinates.split(\",\")\n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        gene_end = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    \n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "        original_query = query_file_list[1]\n",
    "        \n",
    "    query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "    \n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "    # print(annotated_species_name)\n",
    "\n",
    "    # print(annotated_genome_location)\n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    # assert False\n",
    "    check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon)\n",
    "    \n",
    "    \n",
    "    make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length)\n",
    "    \n",
    "    return(annotated_species_name, error_exon,species, genome_file,gene_sequence,left_overhang,right_overhang,scaffold, original_query_name,original_query )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e3790f71-89c6-4326-a178-d4147df81ed0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold = FR989951.1, start = 4965981, end = 4966057\n",
      "Anthocharis_cardamines,FR989951.1,0,0,0,Y,22.Pieris_brassicae_XM_045675385.1_Frame_1_rightoh_2_query_Exon_2,0,0,27\n",
      "\n",
      "1\n",
      "4\n",
      "Anthocharis_cardamines,FR989951.1,4965981,4966057,1,N,22.Pieris_brassicae_XM_045675385.1_Frame_0_rightoh_2_query_Exon_1,1,28,28\n",
      "\n",
      "Anthocharis_cardamines,FR989951.1,4961428,4961519,1,N,22.Pieris_brassicae_XM_045675385.1_Frame_2_rightoh_0_query_Exon_4,2,30,30\n",
      "\n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 19] No such device: '/mnt/f/Genomes_2023-12-26/Anthocharis_cardamines'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[12], line 101\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[38;5;28mprint\u001b[39m(previous_exon_coordinates)\n\u001b[1;32m    100\u001b[0m \u001b[38;5;28mprint\u001b[39m(next_exon_coordinates)\n\u001b[0;32m--> 101\u001b[0m annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query \u001b[38;5;241m=\u001b[39m \u001b[43mmafft_process\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprevious_exon_coordinates\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    102\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mnext_exon_coordinates\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    103\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mcurrent_exon_coordinates\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    104\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mquery_species\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    105\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mquery_transcript\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    106\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mquery_location\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    107\u001b[0m \u001b[43m                                                                                                  \u001b[49m\u001b[43mannotated_genome_location\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    108\u001b[0m \u001b[43m                                                                                                     \u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    110\u001b[0m mafft_run_folder \u001b[38;5;241m=\u001b[39m run_mafft(annotated_genome_location,annotated_species_name,error_exon)\n\u001b[1;32m    112\u001b[0m \u001b[38;5;66;03m# mafft_run_folder = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\u001b[39;00m\n",
      "Cell \u001b[0;32mIn[7], line 12\u001b[0m, in \u001b[0;36mmafft_process\u001b[0;34m(previous_exon_coordinates, next_exon_coordinates, current_exon_coordinates, query_species, query_transcript, query_location, annotated_genome_location, genome_location)\u001b[0m\n\u001b[1;32m     10\u001b[0m error_exon \u001b[38;5;241m=\u001b[39m query_exon\n\u001b[1;32m     11\u001b[0m \u001b[38;5;66;03m# print(\"AA\")\u001b[39;00m\n\u001b[0;32m---> 12\u001b[0m genome_file \u001b[38;5;241m=\u001b[39m \u001b[43mget_genome_file\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     14\u001b[0m upstream_exon_line, downstream_exon_line \u001b[38;5;241m=\u001b[39m previous_exon_coordinates\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m),next_exon_coordinates\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     15\u001b[0m complement,scaffold \u001b[38;5;241m=\u001b[39m upstream_exon_line[\u001b[38;5;241m4\u001b[39m], upstream_exon_line[\u001b[38;5;241m1\u001b[39m]\n",
      "Cell \u001b[0;32mIn[5], line 2\u001b[0m, in \u001b[0;36mget_genome_file\u001b[0;34m(genome_location, species)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_genome_file\u001b[39m(genome_location,species):\n\u001b[0;32m----> 2\u001b[0m     list_of_files_in_genome_folder \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlistdir\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgenome_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m list_of_files_in_genome_folder:\n\u001b[1;32m      4\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m file\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_genomic.fna\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 19] No such device: '/mnt/f/Genomes_2023-12-26/Anthocharis_cardamines'"
     ]
    }
   ],
   "source": [
    "\n",
    "blast_output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae/1.Blast_result\"\n",
    "# blast_output_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "species_list = os.listdir(blast_output_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "# species_list.remove(\"Anthocharis_cardamines\")\n",
    "# species_list = [\"Bicyclus_anynana\",\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Lasiommata_megera\"]\n",
    "for species in species_list:\n",
    "    annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae/1.Blast_result\"\n",
    "    # annotated_genome_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae/1.Query\"\n",
    "    # query_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Query\"\n",
    "    query_species = \"5.Bicyclus_anynana\"\n",
    "    query_transcript = \"XM_024088150.2\"\n",
    "    \n",
    "    genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "    # genome_location = \"/mnt/g/Genomes_2023-12-26\"\n",
    "    \n",
    "    \n",
    "    \n",
    "    output_coordinate_file = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "    exons_to_check_list = [\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\",\"Exon_13\",\"Exon_14\",\"Exon_15\",\"Exon_16\",\"Exon_17\",\"Exon_18\",\"Exon_19\",\"Exon_20\",\"Exon_21\",\"Exon_22\",\"Exon_23\",\"Exon_24\",\"Exon_25\"]\n",
    "#     exons_to_check_list = [\"Exon_23\",\"Exon_24\",\"Exon_25\"]\n",
    "    list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "#     print(list_of_files_in_species_folder)\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        \n",
    "        if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "    if coordinate_file_name =='':\n",
    "        print(f\"Coordinate file error\")\n",
    "        assert False\n",
    "    \n",
    "    with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "    # print(coordinate_file_lines[0])\n",
    "    for exons_to_check in exons_to_check_list:\n",
    "        \n",
    "        for i in range(1,len(coordinate_file_lines)):\n",
    "            coordinate_file_lines_split = coordinate_file_lines[i].split(\",\")\n",
    "            current_exon_coordinates = coordinate_file_lines[i].split(\",\")\n",
    "            if i == 1:\n",
    "                scaffold_first_exon = coordinate_file_lines[i].split(\",\")[1]\n",
    "                first_exon_start, first_exon_end = coordinate_file_lines[i].split(\",\")[2],coordinate_file_lines[i].split(\",\")[3]\n",
    "            # if coordinate_file_lines_split[5] == \"Y\":\n",
    "            if  coordinate_file_lines_split[6].endswith(exons_to_check):\n",
    "                if i == 1:\n",
    "                    print(f'First Exon has errors\\n{coordinate_file_lines[i]}')\n",
    "                    assert False\n",
    "                else:\n",
    "                    print(f\"Scaffold = {scaffold_first_exon}, start = {first_exon_start}, end = {first_exon_end}\")\n",
    "                    print(coordinate_file_lines[i])\n",
    "                    # process_current_exon = input(\"Error Found! Process?\")\n",
    "                    process_current_exon = 'y'\n",
    "                    if process_current_exon.lower()[0] == \"y\":\n",
    "                        if i>2:\n",
    "                            previous_exon_number = i-2\n",
    "                        else:\n",
    "                            previous_exon_number = i-1\n",
    "                        if i+2 <=int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                            next_exon_number = i+2\n",
    "                        else:\n",
    "                            next_exon_number = i+1\n",
    "\n",
    "                        '''\n",
    "                        the above code ensures that missing exons do not cause issues\n",
    "                        '''\n",
    "                        \n",
    "                        previous_exon_coordinates = ''\n",
    "                        next_exon_coordinates = ''\n",
    "                        while True:\n",
    "                            print(previous_exon_number)\n",
    "                            if coordinate_file_lines[previous_exon_number].split(\",\")[5] == \"N\":\n",
    "                                previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                break\n",
    "                            # else:\n",
    "                            #     proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[previous_exon_number]}\\nProceed?\")\n",
    "                            #     if proceed_prompt.lower()[0] == \"y\":\n",
    "                            #         previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                            #         break\n",
    "                            previous_exon_number -= 1\n",
    "                        while True:\n",
    "                            print(next_exon_number)\n",
    "                            if coordinate_file_lines[next_exon_number].split(\",\")[5] == \"N\":\n",
    "                                next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                                break\n",
    "                            # else:\n",
    "                            #     proceed_prompt = input(f\"Next exon coordinate\\n{coordinate_file_lines[next_exon_number]}\\nProceed?\")\n",
    "                            #     if proceed_prompt.lower()[0] == \"y\":\n",
    "                            #         next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                            #         break\n",
    "                            \n",
    "                            next_exon_number += 1\n",
    "                \n",
    "                        \n",
    "                        print(previous_exon_coordinates)\n",
    "                        print(next_exon_coordinates)\n",
    "                        annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query = mafft_process(previous_exon_coordinates,\n",
    "                                                                                                                          next_exon_coordinates,\n",
    "                                                                                                                          current_exon_coordinates,\n",
    "                                                                                                                          query_species,\n",
    "                                                                                                                          query_transcript,\n",
    "                                                                                                                          query_location,\n",
    "                                                                                                                          annotated_genome_location,\n",
    "                                                                                                                             genome_location)\n",
    "                \n",
    "                        mafft_run_folder = run_mafft(annotated_genome_location,annotated_species_name,error_exon)\n",
    "                \n",
    "                        # mafft_run_folder = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\n",
    "                        possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder, error_exon)\n",
    "    \n",
    "                        print(f\"Original query = {original_query}\")\n",
    "                        print(f\"New query = {possible_gene_sequence}\")\n",
    "                        # while True:\n",
    "                            \n",
    "                        #     # check_gene_sequence = input(\"is the query sequence ok?\")\n",
    "                        #     if check_gene_sequence.lower()[0] == \"n\":\n",
    "                        #         possible_gene_sequence = input(\"Enter the desired query sequence :\")\n",
    "                        #         break\n",
    "                        #     if check_gene_sequence.lower()[0] == \"y\":\n",
    "                        #         break\n",
    "                \n",
    "                        list_of_folders_inside_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "                        if \"for_blast\" not in list_of_folders_inside_exon_folder:\n",
    "                            os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                        else:\n",
    "                            list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                            for file in list_of_files:\n",
    "                                os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/{file}\")\n",
    "                \n",
    "                        \n",
    "                        with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\",'w') as query_file:\n",
    "                            query = f\">Query_{error_exon}_{alignment_name}\\n{possible_gene_sequence}\"\n",
    "                            query_file.write(query)\n",
    "            #         else:\n",
    "            #             output_coordinate_file += coordinate_file_lines[i].rstrip()+\",NA,NA\\n\"\n",
    "            # else:\n",
    "            #         output_coordinate_file +=coordinate_file_lines[i].rstrip()+\",NA,NA\\n\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c1ce12a8-3e1a-4234-91c4-88677d81065f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set49_frame1, 0.457\n",
      "5 top scores:\n",
      "[['set49_frame1', 0.457], ['set31_frame2', 1.735], ['set30_frame2', 1.811], ['set16_frame2', 1.834], ['set58_frame2', 1.835]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result/Maniola_hyperantus/Period_gene_genomic_sequence_individual_exon_5.Bicyclus_anynana/Exon_12/for_alignment/alignment_Exon_12_translated_genomic_sequence_49_frame1.fa.txt\n",
      "6 56\n",
      "5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_0_query_Exon_12\n",
      "VLTKPAEMAKQHMTKRCQELASFMESLLDETPKTEEELRLEIQDTDHSYF\n",
      "6 56\n",
      "set49_frame1\n",
      "VLTKPAEAAKQQMSKRCQDLASFMESLMDETPKAEEELRLVIQDADSNCY\n"
     ]
    }
   ],
   "source": [
    "possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccdf583e-4fa4-4fac-a4d4-98a069f1d391",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f0590dc7-9a90-45d6-993b-39b995eaeb4d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set16_frame1, 1.495\n",
      "5 top scores:\n",
      "[['set16_frame1', 1.495], ['set31_frame1', 1.706], ['set49_frame2', 1.765], ['set5_frame2', 1.788], ['set100_frame2', 1.806]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result/Maniola_jurtina/Period_gene_genomic_sequence_individual_exon_33.Pararge_aegeria/Exon_20/for_alignment/alignment_Exon_20_translated_genomic_sequence_16_frame1.fa.txt\n",
      "7 63\n",
      "set16_frame1\n",
      "VSTVNPHMQYMQQGQINESDVVFQMNSVFATSGEANRVHPNNIPPTHYQQIHQSNQ\n",
      "VSTVNPHMQYMQQGQINESDVVFQMNSVFATSGEANRVHPNNIPPTHYQQIHQSNQ set16_frame1\n"
     ]
    }
   ],
   "source": [
    "error_exon = \"Exon_20\"\n",
    "mafft_run_folder = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result/Maniola_jurtina/Period_gene_genomic_sequence_individual_exon_33.Pararge_aegeria/{error_exon}/for_alignment\"\n",
    "\n",
    "possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder, error_exon)\n",
    "print(    possible_gene_sequence, alignment_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307c7dce-21b5-432f-926a-024803e6dad6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
