{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16c8a0c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b2a20a4d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Vanessa_tameamea LOC113403059\n",
      "20.Vanessa_tameamea\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/html/parser.py:171: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  k = self.parse_starttag(i)\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "# import lxml\n",
    "import os\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period\"\n",
    "list_of_species = os.listdir(f\"{location}/0.Species\")\n",
    "gene = \"LOC113403059\"\n",
    "species = \"Vanessa tameamea\".replace(\" \", \"_\")\n",
    "species_folder = ''\n",
    "\n",
    "with open(f\"{location}/gene_detail.txt\", 'r') as gene_file:\n",
    "    gene_list = gene_file.readlines()\n",
    "\n",
    "for i in range(1, len(gene_list)):\n",
    "    gene_list_split = gene_list[i].split(\"\\t\")\n",
    "    species = gene_list_split[0].replace(\" \", \"_\")\n",
    "    gene = gene_list_split[2]\n",
    "    gene = \"LOC113403059\"\n",
    "    species = \"Vanessa tameamea\".replace(\" \", \"_\")\n",
    "    print(species, gene)\n",
    "#     \n",
    "#     print (gene_list[i])\n",
    "\n",
    "    for folder_name in list_of_species:\n",
    "        if species in folder_name:\n",
    "            species_folder = folder_name\n",
    "            print(folder_name)\n",
    "            break\n",
    "\n",
    "    url=f'https://www.ncbi.nlm.nih.gov/gene/?term={gene}&report=gene_table&format=text'\n",
    "    r = requests.get(url)\n",
    "    soup = BeautifulSoup(r.text, \"html.parser\")\n",
    "#     print(soup.text)\n",
    "\n",
    "    with open(f\"{location}/0.Species/{species_folder}/Gene_table.txt\", 'w') as out_file:\n",
    "        out_file.write(soup.text)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "68b6632f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting html5lib\n",
      "  Downloading html5lib-1.1-py2.py3-none-any.whl.metadata (16 kB)\n",
      "Requirement already satisfied: six>=1.9 in /home/saurav/.local/lib/python3.11/site-packages (from html5lib) (1.16.0)\n",
      "Collecting webencodings (from html5lib)\n",
      "  Downloading webencodings-0.5.1-py2.py3-none-any.whl.metadata (2.1 kB)\n",
      "Downloading html5lib-1.1-py2.py3-none-any.whl (112 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m112.2/112.2 kB\u001b[0m \u001b[31m788.4 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hUsing cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)\n",
      "\u001b[33mDEPRECATION: distro-info 0.23ubuntu1 has a non-standard version number. pip 24.1 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of distro-info or contact the author to suggest that they release a version with a conforming version number. Discussion can be found at https://github.com/pypa/pip/issues/12063\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[33mDEPRECATION: python-debian 0.1.36ubuntu1 has a non-standard version number. pip 24.1 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of python-debian or contact the author to suggest that they release a version with a conforming version number. Discussion can be found at https://github.com/pypa/pip/issues/12063\u001b[0m\u001b[33m\n",
      "\u001b[0mInstalling collected packages: webencodings, html5lib\n",
      "Successfully installed html5lib-1.1 webencodings-0.5.1\n"
     ]
    }
   ],
   "source": [
    "!pip3 install html5lib\n",
    "# pip install html5lib\n",
    "# pip install lxml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "76534cba",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20.Vanessa_tameamea\n"
     ]
    }
   ],
   "source": [
    "import re \n",
    "\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period\"\n",
    "list_of_species = os.listdir(f\"{location}/0.Species\")\n",
    "list_of_species.remove(\"names\")\n",
    "list_of_species.remove(\"desktop.ini\")\n",
    "species_folder = ''\n",
    "list_of_species = [\"20.Vanessa_tameamea\"]\n",
    "exon_list = \"Exon1,Exon2,Exon3,Exon4,Exon5,Exon6,Exon7,Exon8,Exon9,Exon10,Exon11,Exon12,Exon13,Exon14,Exon15,Exon16,Exon17,Exon18,Exon19,Exon20,Exon21,Exon22,Exon23,Exon24,Exon25,Exon26,Exon27,Exon28,Exon29,Exon30,Exon31,Exon32,Exon33,Exon34\".split(\",\")\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    with open(f\"{location}/0.Species/{species}/Gene_table.txt\",\"r\") as gene_table_file:\n",
    "        gene_table_list = gene_table_file.readlines()\n",
    "#         print(gene_table_list)\n",
    "\n",
    "    \n",
    "\n",
    "    i = 0\n",
    "    switch = 0\n",
    "    output = \"mRNA Accession\\tExon Number\\tCoding Length\\tGenome Start Point\\tGenome Stop Point\\n\" \n",
    "    output2 = \"mRNA Accession\\tProtein Accession\\tExon Number\\tAlignment Score with Bombyx mori\\n\"\n",
    "    exon_counter = 0\n",
    "\n",
    "    while i < (len(gene_table_list)):\n",
    "\n",
    "        if \"Exon table for  mRNA  \" in gene_table_list[i]:\n",
    "            gene_table_line = gene_table_list[i]\n",
    "            while re.search(\"[ ]{2,}\", gene_table_line):\n",
    "                gene_table_line = gene_table_line.replace(\"  \",\" \")\n",
    "\n",
    "            mrna_accession = gene_table_line.split(\" \")[4]\n",
    "        \n",
    "            switch = 1\n",
    "            i += 3\n",
    "        if switch == 1:\n",
    "\n",
    "            gene_table_line = gene_table_list[i]\n",
    "            while re.search(\"[\\t]{2,}\", gene_table_line):\n",
    "                gene_table_line = gene_table_line.replace(\"\\t\\t\",\"\\t\")\n",
    "            if len(gene_table_line.split(\"\\t\")) == 7 or len(gene_table_line.split(\"\\t\")) == 6:\n",
    "\n",
    "                coding_exon_length =  ((gene_table_line.split(\"\\t\")[5]))\n",
    "                genome_start_point,genome_end_point = ((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "#                 print((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "                output += f\"{mrna_accession}\\t{exon_list[exon_counter]}\\t{coding_exon_length.rstrip()}\\t{genome_start_point}\\t{genome_end_point}\\n\"\n",
    "            \n",
    "        \n",
    "                exon_counter += 1\n",
    "\n",
    "            if switch == 1 and re.search(\"^mRNA\",gene_table_list[i]):\n",
    "                switch = 0\n",
    "                print(exon_counter)\n",
    "                output += \"\\n\\n\"\n",
    "                exon_counter = 0\n",
    "\n",
    "        i += 1\n",
    "\n",
    "    with open(f\"{location}/0.Species/{species}/Gene_table_processed.tsv\",'w') as out_file:\n",
    "        out_file.write(output)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f3fe19b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6322ec9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
