{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3247b82e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0653b64e",
   "metadata": {},
   "outputs": [],
   "source": [
    "scaffold = \"\"\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "086c0ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "#     print(list_of_files_in_genome_folder)\n",
    "    if \"Temp\" in list_of_files_in_genome_folder:\n",
    "        subprocess.run(f'rm -r \"{genome_location}/{species}/Temp\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{genome_location}/{species}/Temp\")\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9d4c6ece",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_query_folder(annotated_species_location,gene_name,species):\n",
    "    import time\n",
    "\n",
    "\n",
    "    list_of_files = os.listdir(annotated_species_location)\n",
    "    if species in list_of_files:\n",
    "        subprocess.run(f'rm -r \"{annotated_species_location}/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    time.sleep(2.5)\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts/{gene_name}\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7cb40a07",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maniola_jurtina XM_045915831.1\n",
      "+\n",
      "{'Exon_1': ['12309849', '12309978'], 'Exon_2': ['12312009', '12312155'], 'Exon_3': ['12313000', '12313187'], 'Exon_4': ['12313812', '12313928'], 'Exon_5': ['12314740', '12314873'], 'Exon_6': ['12315653', '12315748'], 'Exon_7': ['12316409', '12316482'], 'Exon_8': ['12316981', '12317107'], 'Exon_9': ['12317575', '12317757'], 'Exon_10': ['12318372', '12318539'], 'Exon_11': ['12319098', '12319221'], 'Exon_12': ['12319897', '12320016']}\n",
      "exon counter 1\n",
      ">Maniola_jurtina_XM_045915831.1_Exon1\n",
      "ATGCTCGGTGGcagcgtgctgtggttccgtcACGGGTTGCGACTGCACGACAACCCGTCTCTCAACACCGCTCTAGAGGACAAGAGCGTCCCCTTCTTCCCCGTCTTTATATTCGATGGAGAAACTGCTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon2\n",
      "GTACCAAAGTGGTAGGTTACAACCGCATGCGCTACCTACTGGAAGCTCTGCAAGACCTGGACAACCAGTTCAAGAAGTACGGCGGCAGACTGCACATGATCAAGGGCAAGCCTGAGGCTGTGTTCAGACGCCTGTGGGAGGAGTTTG\n",
      ">Maniola_jurtina_XM_045915831.1_Exon3\n",
      "GGATCCGTAAGATTTGCTTCGAGCAGGATTGCGAACCGATATGGCGCGCTAGAGACGAGTGCGTCAAGAGCCTCTGTCGTGAGATAGGTGTGGGGTGTCGTGAGAATGTTTCCCACACGCTTTGGGACCCTGACGTCATCATACGCGCCAATGGGGGCATCCCACCCCTTACCTATCAGATGTTTTTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon4\n",
      "CATACCGTAGCCATCATCGGCAATCCGCCGCGACCTGTAGATGACGTGGATTTAAGAGGAGTCAACTTTGTGGCCCTTCCCGACAGCTTCTACCAGGAGTTTACCATATTTGACAAG\n",
      ">Maniola_jurtina_XM_045915831.1_Exon5\n",
      "GCTCCAAAACCAGAAGACTTGGGAATATTTTTGGAACATGAGGATATCCGCATGATACGCTGGGTGGGCGGAGAAACTGCTGCCTTGGTCCAGATGCAGAACCGCCTCGCTGTGGAGTATGAGACGTTTTGCAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon6\n",
      "gggcTCGTATCTGCCAACGCATGGGAACCCGGATCTCTTGAGCCGGCCGATATCTTTGAGCCCTGCGCTTCGTTTCGGCTGCTTGTCTGTGCGCcg\n",
      ">Maniola_jurtina_XM_045915831.1_Exon7\n",
      "GTTCTACTGGTGCCTTCAAGATCTGTTCCAACAGGTCCATCAGGGGCGTCTGGGCTCCAGTCAATTTATTACTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon8\n",
      "GCCAGCTGATCTGGCGCGAGTACTTCTACACGATGAGCGTGAACAACCCGCAGTACGCGCAGATGGCGGGCAACCCCATCTGCCTCGACATTCCGTGGAAGGAGCCAGAGGGCGATGAGTTGGAAAG\n",
      ">Maniola_jurtina_XM_045915831.1_Exon9\n",
      "ATGGAAGGAAGGTCGTACGGGGTTCCCGTTCGTAGACGCGGCGATGCGTCAACTGCGCACAGAGGGCTGGCTGCACCACGCCGTGCGTAACACCGTTGCGTCGTTCCTCACGCGCGGTACGCTGTGGTTGTCCTGGGAACACGGGTTGCAACACTTCCTCAAATACTTACTTGATGCTGATTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon10\n",
      "GTCAGTCTGCGCCGGCAACTGGATGTGGGTGTCCTCCAGCGCGTTCGAGGCGCTGCTGGATTCGGGCGAGTGCGCGTGTCCGGTGCGCCTCGGCCGCCGCCTGGAGCCCTCGGGCCACTACGTGCGCCGCTACGTGCCCGAGCTGGCCAAGATGCCTGTCGAATACAT\n",
      ">Maniola_jurtina_XM_045915831.1_Exon11\n",
      "CTACGAGCCATGGCGCGCGCCGCTGGACGTGCAGCAGAAGGCGGAGTGCATCATCGGCCGCGACTACCCCGCGCCGGTGGTCGACCACCGCGCCGCCGCGCATCGCAACCGCAGCGCCATGCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045915831.1_Exon12\n",
      "GAGTTACGTCGCGTGCTAGAGAAGGCGCCGCCCCACTGCTGTCCGTCGTCCGAGGACGAAGTCCGTCAGTTCATGTGGCTCGGTGACGAGAACCAGATAGTGGTCAACACTGACTCCTAA\n",
      "130\t147\t188\t117\t134\t96\t74\t127\t183\t168\t124\t120\t\n",
      "Melitaea_cinxia XM_045592943.1\n",
      "+\n",
      "{'Exon_1': ['5295075', '5295204'], 'Exon_2': ['5295432', '5295578'], 'Exon_3': ['5295879', '5296066'], 'Exon_4': ['5296573', '5296689'], 'Exon_5': ['5296802', '5296935'], 'Exon_6': ['5297931', '5298026'], 'Exon_7': ['5298100', '5298173'], 'Exon_8': ['5299397', '5299523'], 'Exon_9': ['5300408', '5300590'], 'Exon_10': ['5301068', '5301235'], 'Exon_11': ['5301489', '5301612'], 'Exon_12': ['5303411', '5303527']}\n",
      "exon counter 1\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon1\n",
      "ATGATGGGTGGCAGCGTGCTATGGTTTCGGCATGGACTGCGTCTGCACGACAATCCGTCGTTGCAGGCAGCGTTACAAGATAGGAGTATACCCTTCTTTCCTGTTTTCATCTTTGACGGAGAGACGGCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon2\n",
      "GCACTAAAGAAGTGGGGTATAATCGGATGCGGTATTTGCTCGAGGCCTTGAACGATTTGGACATCCAGTTTAACAAGTATGGCGGAAAACTTCACATGATAAAAGGGAAGCCCGATGTTGTATTCCGACGTCTTTGGGAGGAACTCG\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon3\n",
      "GTATTCGCAAGATTTGCTTTGAACAGGACTGTGAGCCCATATGGCGCGCTCGTGATGAGCGCGTGCGAAACCTCTGCCGAGAAATAGGAGTGTCGTGCTGTGAAAACGTCTCTCATACCCTATGGGATCCAGATAAAGTCATAAACGCTAACGGTGGCATTCCACCACTGACCTATCAGATGTTTCTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon4\n",
      "CATACTGTTGGAATCATTGGCAACCCACCTCGCCCTGTGGATGACGTAGACTTGAAAGGAGTCAACTTTGGTACTCTTCCTAAAAGCTTCTATGAAGAATTTACTGTTTTCGATAAG\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon5\n",
      "TCCCCAAAACCGGAAGATTTGGGAGTTTTTTTGGAAAACGAGGACATTCGCATGATACGTTGGGTTGGTGGTGAGACAGCCGCGTTGAATCAGATGCAGAACCGTCTGGCCGTCGAATACGAGACTTTTCTCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon6\n",
      "agGCTCGTATTTACCGACGCATGGTAACCCTGACTTGCTCGGATCGCCTATTTCCTTGAGTCCAGCACTTCGTTTTGGCTGCCTCTCAGTTCGCCG\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon7\n",
      "ATTCTATTGGTCTGTCCAAGATCTGTTCCAACAAGTGCATCAAGGGCGTCTCTCTGCGACTCAGTTTATCACTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon8\n",
      "GTCAACTCATATGGCGCGAATATTTCTACACCATGAGCGTGAACAATCCAAACTACGCGCAGATGGCCGGCAATCCTATATGCTTGGACATACCCTGGAAGGAGCCAGAAAACGATGAACTTCAAag\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon9\n",
      "ATGGAAGGAGGGCCGCACGGGGTTCCCGTTCGTGGACGCGGCCATGCGGCAACTGCGCACGGAGGGCTGGCTGCACCACGTCGTGCGCAACACTGTCGCCTCCTTCCTCACGCGCGGCACGCTGTGGCTGTCCTGGGAGCACGGCCTGGCGCACTTCCTCAAGTACCTGCTGGACGCCGACTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon10\n",
      "GTCGGTGTGCGCCGGCAACTGGATGTGGGTATCGTCGAGCGCGTTCGAGGCGCTGCTGGACTCGGGCGAGTGCGCGTGCCCGGTGCGGCTGGGGCGCCGCCTGGAGCCCACCGGCCAGTACGTGCGCCGCTACGTGCCCGAGCTGGCGCGCATGCCCGTCGAATACAT\n",
      ">Melitaea_cinxia_XM_045592943.1_Exon11\n",
      "CTACGACCCGTGGAACGCGCCGCTGCAGGTGCAGCAGCGCGCGGGCTGCGTGGTGGGGCGGCACTACCCCGCGCGCGTCGTCGACCAGCGCGCCGCCGCGCAGCGCAACCGCGCCGCGATGCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045592943.1_Exon12\n",
      "GAACTACGTCGAATGCTCGAAAAGGTACCACCTCATTGCTGTCCTTCTTCGGAGGATGAGGTTCGCCAGTTCATGTGGCTCGGGGAAGAGACTCAGGCTGAGGTTTCTACATCTTAA\n",
      "130\t147\t188\t117\t134\t96\t74\t127\t183\t168\t124\t117\t\n",
      "Nymphalis_io XM_050489987.1\n",
      "-\n",
      "{'Exon_1': ['13032477', '13032606'], 'Exon_2': ['13032134', '13032280'], 'Exon_3': ['13031253', '13031440'], 'Exon_4': ['13030651', '13030767'], 'Exon_5': ['13029836', '13029969'], 'Exon_6': ['13029175', '13029270'], 'Exon_7': ['13028243', '13028316'], 'Exon_8': ['13027749', '13027875'], 'Exon_9': ['13027288', '13027470'], 'Exon_10': ['13026505', '13026672'], 'Exon_11': ['13026162', '13026285'], 'Exon_12': ['13025641', '13025757']}\n",
      "exon counter 12\n",
      ">Nymphalis_io_XM_050489987.1_Exon12\n",
      "gatttgcgtaaaatgcTTGAGAAGGCACCGCCTCACTGCTGTCCGTCTTCGGAGGATGAAGTCCGTCAGTTCATGTGGCTCGGCGATGAGACCCAGAATGATATACCCAGCGCCTAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon11\n",
      "CTACGAGCCGTGGAACGCGCCCCTCGAGGTGCAGGAGCGCGCGGGCTGCGTGGTGGGGCGGCACTACCCCGCGCCCGTCGTCGACCACCGCGCCGCCGCGCAGCGCAACCGCGCCAACATGCAG\n",
      ">Nymphalis_io_XM_050489987.1_Exon10\n",
      "GTCGGTGTGCGCCGGCAACTGGATGTGGGTGTCGTCCAGCGCCTTCGAGGCGCTGCTGGACTCCGGCGAGTGCGCGTGCCCCGTGCGCCTCGGCCGCCGCCTCGAGCCGTCCGGCCACTACGTGCGGCGCTACGTGCCCGAGCTGGCGCACATGCCCGTCGACTACAT\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon9\n",
      "GTGGAAGGAGGGTCGCACGGGGTTCCCGTTCGTGGACGCGGCCATGCGCCAGCTGCGCTCGGAGGGCTGGCTGCACCACGCCGTGCGCAACACCGTGGCGTCGTTCCTCACGCGCGGCACGCTGTGGCTCTCGTGGGAGCACGGCCTGGCGCACTTCCTCAAGTACCTGCTGGACGCCGACTG\n",
      ">Nymphalis_io_XM_050489987.1_Exon8\n",
      "GTCAATTGATATGGCGTGAGTATTTTTACACGATGAGCGTGAACAATCCGAACTACGCGCAGATGGCCAACAACCCGATCTGCTTGGATATACCGTGGAAGGAGCCGGAGAGTGATGAGCTGCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon7\n",
      "TTTCTATTGGTGCGTCCAAGATCTGTTCCAACAAGTGCATCAGGGACGACTCGCGTCGACTCAGTTTATAAcgg\n",
      ">Nymphalis_io_XM_050489987.1_Exon6\n",
      "GGGTTCCTATTTACCGACGCATGGTAACCCTGACCTGCTCGGACCACCGATATCCCTGAGTCCGGCCTTGCGGTTCGGCTGTCTGTCTGTGCGCCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon5\n",
      "GTTCCCAAACCGGAAGACTTGGGGGTGTTTTTGGAGAATGAAGACATACGTATGATACGCTGGGTGGGGGGTGAGACAGCTGCATTGAAACAGATGCAGAACCGCCTGGCCGTCGAGTACGAGACCTTTTGCAG\n",
      ">Nymphalis_io_XM_050489987.1_Exon4\n",
      "caCACCGTTGCCATTATCGGAGACCCGCCTCGCCCCGTGGGCGACGTGGACTTGAAGGGAGTCAACTTTGCGAGTCTTCCAAGGAGTTTCTACGACGAGTTCACTATTTTTGAAGag\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon3\n",
      "GTATACGCAAGATTTGTTTCGAGCAAGATTGTGAGCCAATATGGCGCGCACGTGACGATTGCGTGAGAAACCTCTGCCGTGAGATTGGAATTTCGTGTCACGAAAACGTTTCCCATACACTATGGGATCCAGACAGCGTCATCCAGGCTAATGGTGGAATCCCACCATTGACTTATCAAATGTTTCTG\n",
      ">Nymphalis_io_XM_050489987.1_Exon2\n",
      "GTACCAAGATCGTGGGGTACAATCGCATGCGTTACTTGTTGGAGGCGCTGAACGATTTGGATCTCCAGTTTAAGAAGTATGGCGGGAAATTGCACTTGATAAAAGGCAAGCCTGAACATGTGTTCAGGCGCTTCTGGGAAGTATTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050489987.1_Exon1\n",
      "ATGGTCGGTGGTAGCGTGCTCTGGTTCCGGCATGGGCTACGTCTGCACGATAATCCGTCGTTGCAAAACGCTCTTGAAGATATTAACACGCCATTATTCCCTATCTTCATTTTTGACGGAGAGACAGCGG\n",
      "117\t124\t168\t183\t127\t74\t96\t134\t117\t188\t147\t130\t\n",
      "Vanessa_atalanta XM_047673373.1\n",
      "+\n",
      "{'Exon_1': ['869540', '869669'], 'Exon_2': ['869855', '870001'], 'Exon_3': ['870420', '870607'], 'Exon_4': ['870743', '870859'], 'Exon_5': ['871517', '871650'], 'Exon_6': ['872046', '872141'], 'Exon_7': ['872492', '872565'], 'Exon_8': ['872637', '872763'], 'Exon_9': ['872987', '873169'], 'Exon_10': ['873924', '874091'], 'Exon_11': ['874167', '874290'], 'Exon_12': ['875060', '875176']}\n",
      "exon counter 1\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon1\n",
      "ATGGTCGGTGGCAGCGTGCTCTGGTTCCGACATGGACTACGTCTGCACGACAATCCGTCGTTGCAAAATGCCCTTGAAGACAGGCACACACCATTCTACCCTATCTTCATCTTTGATGGAGAGACAGCGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047673373.1_Exon2\n",
      "GTACGAAGATCGTAGGGTACAATCGCATGCGCTATTTGTTGGAGGCGCTGAACGATTTGGATGTCCAGTTTAAAAAGTATGGAGGAAGACTGCACATGATAAAAGGCAATCCTGATCTCGTGTTCAGGCGTCTCTGGGAAGAATCCG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon3\n",
      "GAATACGGAAGATTTGTTTCGAGCAAGATTGTGAGCCAATATGGCGCGCACGTGACGACTGCGTCCGAAACCTCTGCCGTGAGATCGGAGTTTCGTGTCGGGAAAACGTTTCCCATACACTATGGGATCCAGACAGCGTCATCCAAGCTAATGGTGGAATCCCTCCATTGACCTATCAAATGTTTCTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047673373.1_Exon4\n",
      "caCACCATTGCTATTATTGGAGACCCGCCTCGCCCTGTGGACGATGTAGACTTGAAGGGAGTCAACTTTGCGTGTCTTCCAAAAAGCTTCTACGATGAATTTACTGTGTTTGATGTG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon5\n",
      "GTCCCCAAACCTGAAGACTTGGGAGTGTTTTTGGAAAACGAAGACATTCGTATGATACGCTGGATGGGGGGCGAGACTGCTGCTTTGAACCAGATGCAGGACCGTCTGGCCGTGGAGTACGAGACGTTTTGCAG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon6\n",
      "GGGCTCCTATTTACCGACGCATGGTAACCCAGACCTGCTCGGACCACCGATATCGCTGAGTCCGGCGCTGCGGTTCGGCTGTCTGTCTGTGCGTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047673373.1_Exon7\n",
      "TTTCTATTGGTGCGTCCAAGATCTGTTCCAACAAGTGCACCAGGGACGCCTCGCTTCGACTCAGTTTATAACTG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon8\n",
      "GTCAACTGATATGGCGTGAGTATTTCTACACGATGAGCGTGAACAATCCGAACTACGCGCAGATGGCCGGCAACCCGATCTGCTTGGATATACCGTGGAAGGACCCGGAGAGCGACGAGCTCCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047673373.1_Exon9\n",
      "GTGGAAGGAGGGCCGCACGGGCTTCCCGTTCGTGGACGCGGCCATGCGGCAGCTGCGCACGGAGGGCTGGCTGCACCACGCCGTGCGCAACACCGTGGCGTCGTTCCTCACGCGCGGCACGCTGTGGCTCTCGTGGGAGCACGGCCTCGCGCACTTCCTCAAGTACCTGCTGGACGCCGACTG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon10\n",
      "GTCGGTGTGCGCCGGCAACTGGATGTGGGTGTCGTCCAGCGCGTTCGAGGCGCTGCTGGACTCGGGCGAGTGCGCGTGCCCCGTGCGCCTCGGCCGCCGCCTCGAGCCGTCCGGCCACTACGTGCGCCGCTACGTGCCCGAGCTGGCGCGCATGTCCGTCGACTACAT\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047673373.1_Exon11\n",
      "CTACGAGCCGTGGAACGCGCCGCTCGAGGTGCAGGAGCGCGCGGGCTGCGTGGTGGGCCGAGACTACCCCGCGCCCGTCGTCGACCACCGCGCCGCCGCGCAGCGCAACCGCGCCGCCATGCAG\n",
      ">Vanessa_atalanta_XM_047673373.1_Exon12\n",
      "gatTTGCGCAAAATGCTCGAGAAAGCACCGCCTCATTGCTGCCCGTCCTCGGAGGATGAAGTCCGCCAGTTCATGTGGCTCGGCGACGAGGGCCAGAACGAGATAACCAGCACCTAA\n",
      "130\t147\t188\t117\t134\t96\t74\t127\t183\t168\t124\t117\t\n",
      "Vanessa_cardui XM_047103353.1\n",
      "+\n",
      "{'Exon_1': ['920733', '920862'], 'Exon_2': ['921371', '921517'], 'Exon_3': ['921901', '922088'], 'Exon_4': ['923185', '923301'], 'Exon_5': ['923814', '923947'], 'Exon_6': ['924654', '924749'], 'Exon_7': ['925080', '925153'], 'Exon_8': ['925225', '925351'], 'Exon_9': ['926114', '926296'], 'Exon_10': ['927392', '927559'], 'Exon_11': ['927980', '928103'], 'Exon_12': ['929041', '929157']}\n",
      "exon counter 1\n",
      ">Vanessa_cardui_XM_047103353.1_Exon1\n",
      "ATGGCCGGTGGCAGCGTGCTCTGGTTCCGACATGGACTACGTCTGCATGACAATCCGTCGTTGCAAAACGCCCTTGAGGACAGGCACACACCATTCTACCCAATGTTCATCTTTGACGGAGAGACAGCGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon2\n",
      "GTACGAAGATCGTGGGGTACAATCGCATGCGTTATTTGTTGGAGGCGCTGAACGATTTAGACGTCCAGTTTAAAAAGTATGGCGGAAAATTGCACATGATAAAGGGCAAGCCTGATCTTGTGTTCAGGCGTCTCTGGGAAGAATCCG\n",
      ">Vanessa_cardui_XM_047103353.1_Exon3\n",
      "GTATACGGAAAATTTGTTTCGAGCAAGATTGCGAGCCAATATGGCGCGCTCGTGACGACTGCGTGCGAAACCTCTGCCGTGAGATCGGAGTTTCGTGTCGGGAAAACGTTTCCCACACACTATGGGATCCAGACAGCGTCATCCAAGCTAATGGTGGAATCCCTCCATTGACCTATCAGATGTTTCTG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon4\n",
      "caCACAATCGCTATTATTGGAGACCCACCTCGCCCTGTGGACGACGTTGATTTGAAGGGAGTCAACTTTGCCTGTCTTCCAAAAAGCTTCTACGATGAGTTTACTGTTTTTGAAAAG\n",
      ">Vanessa_cardui_XM_047103353.1_Exon5\n",
      "gtTCCCAAACCCGAAGACTTGGGAGTGTTTTTAGAGAACGAAGACATTCGTATGATACGCTGGATGGGGGGCGAGACTGCTGCATTGAACCAGATGCAGAACCGCCTGGCCGTGGAGTACGAGACCTTTTGCAG\n",
      ">Vanessa_cardui_XM_047103353.1_Exon6\n",
      "GGGCTCCTATTTACCGACGCACGGTAACCCCGACTTGCTCGGACCACCGATATCCCTGAGTCCGGCGTTGCGGTTCGGCTGTCTGTCTGTGCGTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon7\n",
      "TTTCTATTGGTGCGTCCAAGACCTGTTCCAACAAGTGCATCAGGGACGCCTCGCTTCTACTCAGTTTATAACGG\n",
      ">Vanessa_cardui_XM_047103353.1_Exon8\n",
      "GTCAACTGATTTGGCGCGAGTACTTCTACACGATGAGCGTGAACAATCCAAACTACGCGCAAATGTCCGGCAATCCGATCTGTTTGGATATACCCTGGAAGGAGCCGGAGAGCGATGAGCTTCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon9\n",
      "GTGGAAGGAGGGGCGCACGGGCTTCCCGTTCGTGGACGCGGCCATGCGGCAGCTGCGCGCGGAGGGCTGGCTGCACCACGCCGTGCGCAACACGGTGGCGTCGTTCCTCACGCGCGGCACGCTGTGGCTGTCGTGGGAGCACGGCCTCGCGCACTTCCTCAAGTACCTGCTCGACGCCGACTG\n",
      ">Vanessa_cardui_XM_047103353.1_Exon10\n",
      "GTCGGTGTGCGCCGGCAACTGGATGTGGGTGTCGTCGAGCGCGTTCGAGGCGCTGCTGGACTCGGGCGAGTGCGCGTGCCCCGTGCGCCTCGGCCGCCGCCTCGAGCCCTCCGGCCACTACGTGCGCCGCTACGTGCCCGAGCTGGCGCGCATGCCCGTCGACTACAT\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon11\n",
      "CTACGAGCCGTGGAACGCGCCGCTGGAGGTGCAGGAGCGCGCGGGCTGCGTGGTGGGGCGCCACTACCCCGCGCCCGTCGTCGACCACCGCGCCGCGCAGCAGCGCAACCGCGCCGCCATGCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047103353.1_Exon12\n",
      "gATTTGCGCAGAATGCTCGAGAAAGCACCGCCTCACTGCTGCCCGTCCTCGGAGGATGAGGTCCGCCAGTTCATGTGGCTCGGCGATGAGACCCAGAATGAGATAACCAGCACCTAA\n",
      "130\t147\t188\t117\t134\t96\t74\t127\t183\t168\t124\t117\t\n",
      "Vanessa_tameamea XM_064215795.1\n",
      "\n",
      "{}\n",
      "Error!!\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 59\u001b[0m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     58\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mError!!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m     61\u001b[0m \u001b[38;5;66;03m# list_of_query_species = os.listdir(query_location)\u001b[39;00m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;66;03m# if query_species not in list_of_query_species:\u001b[39;00m\n\u001b[1;32m     63\u001b[0m \u001b[38;5;66;03m#     os.mkdir(f\"{annotated_species_location}/{query_species}\")\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \u001b[38;5;66;03m# list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\u001b[39;00m\n\u001b[1;32m     65\u001b[0m \u001b[38;5;66;03m# if \"Transcripts\" not\u001b[39;00m\n\u001b[1;32m     66\u001b[0m \u001b[38;5;66;03m#     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\u001b[39;00m\n\u001b[1;32m     68\u001b[0m print_out \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "# species_gene = [\"Vanessa_tameamea,XM_026631919.1\"]\n",
    "species_gene = [\"Vanessa_tameamea,XM_026643186.1\",\"Papilio_machaon,XM_014499499.2\",\"Papilio_polytes,XM_013287971.1\",\"Papilio_xuthus,XM_013317177.1\",\"Colias_croceus,XM_045641425.1\",\"Leptidea_sinapis,XM_050828385.1\",\"Pieris_brassicae,XM_045658499.1\",\"Pieris_napi,XM_047649107.1\",\"Pieris_rapae,XM_045629235.1\",\"Zerene_cesonia,XM_038356442.1\"]\n",
    "\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\"\n",
    "annotated_species_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/12.Cryptochrome_1a_Exon_Analysis/1.Annotated Species\"\n",
    "try:\n",
    "    os.mkdir(annotated_species_location)\n",
    "except:\n",
    "    pass\n",
    "for entries in species_gene:\n",
    "\n",
    "    species,gene_name = entries.split(\",\")\n",
    "    print(species,gene_name)\n",
    "\n",
    "    genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    query_species = species\n",
    "    create_query_folder(annotated_species_location,gene_name,species)\n",
    "    list_of_files = os.listdir(f\"{genome_location}/{species}\")\n",
    "    # print(list_of_files)\n",
    "    for file_names in list_of_files:\n",
    "        if file_names.endswith(\"_genomic.gff\"):\n",
    "            gff3_file = file_names\n",
    "\n",
    "    with open(f\"{genome_location}/{species}/{gff3_file}\") as gff_file_open:\n",
    "        gff_file_list = gff_file_open.readlines()\n",
    "    complement = ''\n",
    "    exon_count = 1\n",
    "    gene_found = 0\n",
    "    exon_dictionary = {}\n",
    "    for line in gff_file_list:\n",
    "        if gene_name in line:\n",
    "            if \"CDS\" in line.split(\"\\t\"):\n",
    "\n",
    "    #             print(line)\n",
    "                gene_found = 1\n",
    "                if complement == '':\n",
    "                    complement = line.split(\"\\t\")[6]\n",
    "                exon_start = line.split(\"\\t\")[3]\n",
    "                exon_stop = line.split(\"\\t\")[4]\n",
    "                scaffold = line.split(\"\\t\")[0]\n",
    "                exon_dictionary[f\"Exon_{exon_count}\"] = [exon_start,exon_stop]\n",
    "                exon_count += 1\n",
    "            if \"exon\" in line.split(\"\\t\") and gene_found == 1:\n",
    "                break\n",
    "    print(complement)\n",
    "    print(exon_dictionary)\n",
    "    if complement == \"+\":\n",
    "        exon_counter = 1\n",
    "    elif complement == \"-\":\n",
    "        exon_counter = len(exon_dictionary)\n",
    "    else:\n",
    "        print(\"Error!!\")\n",
    "        assert False\n",
    "\n",
    "    # list_of_query_species = os.listdir(query_location)\n",
    "    # if query_species not in list_of_query_species:\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # if \"Transcripts\" not\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\n",
    "\n",
    "    print_out = ''\n",
    "    print(\"exon counter\", exon_counter)        \n",
    "    for i in range(1,len(exon_dictionary)+1):\n",
    "\n",
    "        exon_start, exon_stop = exon_dictionary[f\"Exon_{exon_counter}\"]\n",
    "    #     print(exon_start,exon_stop)\n",
    "        subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{exon_start}-{exon_stop} > \"{genome_location}/{species}/Temp/temp_gene.fa\"', shell = True)\n",
    "        exon_sequence_file = SeqIO.parse(f\"{genome_location}/{species}/Temp/temp_gene.fa\", \"fasta\")\n",
    "        sequence = ''\n",
    "\n",
    "        for sequences in exon_sequence_file:\n",
    "    #         print(sequences.id)\n",
    "            if complement == \"-\":\n",
    "                sequence = sequences.seq.reverse_complement() \n",
    "\n",
    "    #             print(sequences.seq.reverse_complement().translate())\n",
    "            else:\n",
    "                sequence = sequences.seq\n",
    "    #             print(sequences.seq)\n",
    "    #             print(sequences.seq.translate())\n",
    "            print_out += str(len(sequence))+\"\\t\"\n",
    "            output = f\">{query_species}_{gene_name}_Exon{exon_counter}\\n{sequence}\"\n",
    "        with open(f\"{annotated_species_location}/{query_species}/Transcripts/{gene_name}/Exon{exon_counter}.fa\", 'w') as outfile:\n",
    "            outfile.write(output)\n",
    "        print(output)\n",
    "        if complement == \"-\":\n",
    "            exon_counter -= 1\n",
    "        else:\n",
    "            exon_counter += 1\n",
    "\n",
    "    #         if gene_location in line.split(\"\\t\"):\n",
    "    #           print(line)\n",
    "    #           break\n",
    "    print(print_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "111273de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0002681f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
