{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2b2fddd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "GCF_947172395.1_ilBicAnyn1.1_genomic.fna\n",
      "1569832 1569923\n",
      "0 2\n",
      "Getting Gene\n",
      "MAREFTRTHEYFMALPDLQPPTGHYTFSFD\n",
      ">Bicyclus_anynana_Exon_1_NC_069110.1_1569832_1569923_left_0_right_2\n",
      "ATGGCGCGTGAATTTACACGAACCCACGAGTACTTCATGGCTTTGCCGGACTTGCAGCCTCCGACAGGGCATTACACCTTCTCCTTCGACAT\n",
      "1584792 1584971\n",
      "1 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_2_NC_069110.1_1584792_1584971_left_1_right_2\n",
      "GCCGCACACGAGCACGCTGCCGTCTCTGCAGCTGACAGTGGACACCGACATCGCGGACGCAGACATCGAGATCTGCGATGACGACGACGGCGCGGGCAGCCACGGGCGACCACAGCAGCTGCTGCAGATCACCTCGCAGCCACACGACCAGCGCAGCAAACGCAGGAACTCACAGACCCT\n",
      "1591763 1591855\n",
      "1 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_3_NC_069110.1_1591763_1591855_left_1_right_2\n",
      "cTCGGCTGCGTTTGATTTGTCGAGCTGCGAAGTTCTCTCTACTTCGTCGGTATCTACGGCCACTTCAAACaggaaaagaaaaacatcacCCTA\n",
      "1596215 1596283\n",
      "1 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_4_NC_069110.1_1596215_1596283_left_1_right_2\n",
      "caCTTGCAGTTCTTTCGAAGATGACTGCCATGATGCGAAATCTACCCGCAGCAGCATACCAGACAAAAA\n",
      "1597013 1597159\n",
      "1 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_5_NC_069110.1_1597013_1597159_left_1_right_2\n",
      "ACAAAACCACAGTGAGATTGAGAAGCGACGACGCGATAAGATGAACAACTACATATTGGAGCTGAGTTCCATGATTCCCATCTGTAACGCCATGTCGAGGAAGCTGGACAAGCTCACGGTGCTGCGTATGGCTGTGCAGCATATGAG\n",
      "1599697 1599886\n",
      "1 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_6_NC_069110.1_1599697_1599886_left_1_right_0\n",
      "GTCTGTCCGGGGCGCAATTTCTGCGGGCCCGCTGACTCTGCGGCCCAGACCCGCGTTCCTCTCGGAGCAGGATCTGAACAATATGGTGCTCAAGGCGGCGGAGGATTCCTTCTTGATGGTGGTGGGCTGCGACCGCGGACGGCTGCTCTACGTGTCATGTTCGGTCACGAAGATGTTGCAATACGATCAG\n",
      "1600596 1600713\n",
      "0 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_7_NC_069110.1_1600596_1600713_left_0_right_1\n",
      "GCGGAGCTTCTGGGCCAGAGTCTATTCGACATCTTGCATCCGAAGGACGTGGCTAAAGTGAAGGAGCAGCTGTCAGCGTCCGACCTCAATCCGCGAGAGCGATTTATTGATACCAAAA\n",
      "1601627 1601866\n",
      "2 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_8_NC_069110.1_1601627_1601866_left_2_right_1\n",
      "CCATGTTGCCTGTTAAAACGGACTACGTGTCGAGCGCGTCGCGACTGTGCCCGGGCGCGCGGCGCTCCTTCTTCTGCCGCATTAAATGCCGCGCCGCCACCGTCATCACTGCGCCCAGCTGCACTGTCACCTCGGGCTCGCTGCAGCCGCCGCACGTCAAAGAGGAGCAAGGCAGCGACTGCAACCCCACCAAGAGGAAGAAGCAAACCGAGAGGAAGTACTGCGTGGTGCAGTTCACAG\n",
      "1602339 1602555\n",
      "2 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_9_NC_069110.1_1602339_1602555_left_2_right_2\n",
      "GTTATCTGAAATCCTGGGCACCGACAGATGGCCAGGATAACGGGAATGCGGACGGGATGGCCGACGACGGTGACGCCAGCAACCTGTCCTGCCTGGTGGCGGTGGGCAGGTCTCTGACTGAGCTGACGCGCTTCATGGCTCCCGCGCCGCATCGCCGACTGTTGCAGTTCACGTCCAGGCACGCTACGGACAGCAAGTTCGTGTTTGTGGACCAGAG\n",
      "1603477 1603680\n",
      "1 2\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_10_NC_069110.1_1603477_1603680_left_1_right_2\n",
      "GGTGACGCTAGTCCTGGGCTTCCTCCCGCAAGAGCTTCTGGGCACGAGTCTGTACGAGCATGTGAACGCTTCAGAGCTGAGCGCGGTCGCGAAGACGCACAAGGCAGCGCTACTTCAAAGAGACGCTTTGAAGACGCCGCCATACAATTTCAAGAAGAAAGATGGCTCCTACATTAGGATACAGACGCATTTCAAGCCGTTCAA\n",
      "1604077 1604186\n",
      "1 1\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_11_NC_069110.1_1604077_1604186_left_1_right_1\n",
      "CAATCCGTGGACGAGGGACGTGGATTGCTTGGTCGCCAACAACACCGTGGTAGCGGATGACGCGGCCGATGTCACGCCTGAGGACAGTTCCTTTGACGTATTTCATGGAA\n",
      "1607452 1607606\n",
      "2 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_12_NC_069110.1_1607452_1607606_left_2_right_0\n",
      "AAGCTGACGCGGAGATGCAACGGTTGATCGACTCGCAGGTCGAGTCTCACAAGATCGGCTCGACGATCGCGGACGAAGTGTTGCGGCGGTCACCCACGGAGTTCTCGGCCGAACTGCCCGCCGATTTGCTTCAAGACGCAGTGTTCATTCAACAG\n",
      "1608333 1608611\n",
      "0 0\n",
      "Getting Gene\n",
      ">Bicyclus_anynana_Exon_13_NC_069110.1_1608333_1608611_left_0_right_0\n",
      "CCAACGCTGGACACCTTGCTCGGTGCAGAGGTGGACTCCTTCGAAGGACAAGTGCGTAACAACGTCCCTCTGAGTGCAGCGGGCGCCGCAGACACCCCTCCGCCCGAGGTCCACGAGCTGCCCGCCAACAGTCCGCAGCCCGCCTCGCCGCCTCTCCCCACGCTGAGCTTCGACGGCAACGGAGAGGCAACCATGGCGGTAATCATGAGCTTGCTGGAAGCCGATGCGGGTCTAGGAGGCCCCGTCAATTTCAATGGCTTACCTTGGCCGTTACCTTAA\n",
      "Coenonympha_glycerion\n",
      "GCA_963855885.1_ilCoeGlyc1.1_genomic.fna\n",
      "312508 312599\n",
      "0 2\n",
      "Getting Gene\n",
      "MARDYARAHELYLHLHELQPAHFHPYQYDL\n",
      ">Coenonympha_glycerion_Exon_1_OY979626.1_312508_312599_left_0_right_2\n",
      "ATGGCGCGAGACTACGCGCGCGCTCACGAGCTGTATCTACACCTGCATGAGCTGCAGCCGGCGCACTTCCATCCCTACCAGTATGACCTGCA\n",
      "326383 326565\n",
      "1 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_2_OY979626.1_326383_326565_left_1_right_2\n",
      "AGCGCATTCCACCGGGGCGCCGGCTTACGAGCTGACAgcgggcgcgggcgccgcgGACGGCGCCGCTGGTGCGGAAGGCGCCGGCAgcggcgcgctcgcggcgcACGCGCAGCTGCCGCAGCacgcgccgcagccgcagcgCGCGCCGCACGACCCGCGCAAGTGGCAGACCACCCACGACAT\n",
      "329746 329835\n",
      "1 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_3_OY979626.1_329746_329835_left_1_right_2\n",
      "CTCGGAGGCGTACGAGCCGGCCAGCTGCGACGCGCTCTCCGTCTCCTCGGCGTCCACCGCGGCCACGGGTAGGAAGAGGAAGCCTTCTCC\n",
      "337600 337671\n",
      "1 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_4_OY979626.1_337600_337671_left_1_right_2\n",
      "CAGCGCGAGCAGCGCTTACGACGACGACGGCACCGACGACGCGCGCTCCAGCCGCAGCGCGACCGACAAGAA\n",
      "338405 338551\n",
      "1 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_5_OY979626.1_338405_338551_left_1_right_2\n",
      "GCAAAACCACAGCGAGATCGAGAAACGACGTCGCGATAAAATGAACACCTACATAACCGAGCTCAGTGCCATGGTGCCCATGTGCGGCGCCATGGCGCGCAAGCTGGACAAGCTCACCGTGCTGCGTATGGCGGTGCAGCACATGAG\n",
      "338909 339098\n",
      "1 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_6_OY979626.1_338909_339098_left_1_right_0\n",
      "GTCGATCCGCGGCGCGCTGTCGGCGGGCCCGCTGACGGCGCGGCCGCGGCCGTCGATCGTGTCGGAGCGCGAGCTGAACGCGCTGGTGCTGCGCGCCGCCGAGGACTGCTTCCTGATGGTGGTGGGCTGCGACCGCGGCCGCCTCATGTTCGTGTCCGCCTCCGTCACCAGGCTGCTCAACTACGACCAG\n",
      "339335 339452\n",
      "0 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_7_OY979626.1_339335_339452_left_0_right_1\n",
      "TCGGAGCTGCTGGGCCAGAGCCTGTTCGACATCCTGCACCCGAAAGACGTGGCCAAGGTGAAGGAGCAGCTGTCGTCTTCCGATCTCAGCCCGCGGGAGCGCCTCATCGACGCCAAGA\n",
      "339610 339807\n",
      "2 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_8_OY979626.1_339610_339807_left_2_right_1\n",
      "CGATGCTGCCGCTGAAGGCGGAACTGGTGGTGGACGCGTCGCGGCTGTGCCCCGGCGCGCGGCGCTCGTTCCTCTGCCGCATCAAGTGCCGCGCGCCCGCCGAGCCCGCCGCCGACGCGCCGCCGCTCAAGGAGGAGGCGGACGCCAAGCCGCGCAAGAAGCACGCGGAGAGGAAGTTCTGCGTGGTGCAGTGCACGG\n",
      "339877 340108\n",
      "2 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_9_OY979626.1_339877_340108_left_2_right_2\n",
      "GCTACCTCAAGTCGTGGGCGCCGGCGGAGCTGAGCGCGggcgcggacggcgcggcgGAGGACGAGGCGTGCAACCTGTCGTGCCTGGTGGCGGTGGGGCGCGCGCTGCCgggcgccgcgcacgcgccgcgcgcgcACCACACGCGCCCGCTGCAGTTCACCGCGCGCCACGCCACCGACGGCAAGTTCCTCTTCGTGGACCAAAGGCGAGTGCACCCCCCACCACGCGACCC\n",
      "340673 340876\n",
      "1 2\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_10_OY979626.1_340673_340876_left_1_right_2\n",
      "GGTGACGCTGGCTCTGGGCTACCTCCCCCAGGAGCTGCTGGGCACGAGCCTGTACGAGCACGTGCACGCGCGCGAGCTGGGCGCGGTGGCGCGCGCGCACAaggcggcgctgctgcgcaGGGAGCCGCTGCGCACGCCGCCCTACTCCTTCAAGAAGAAGGACGGCACCTGCGCGCGCGTCCAGACGCACTTCAAGCCCTTCAA\n",
      "341486 341595\n",
      "1 1\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_11_OY979626.1_341486_341595_left_1_right_1\n",
      "GAACCCGTGGACGAAGGACGTGGAGAGCCTGGTCGCCAACAACACCGTGATGGCGGACGAGTGGGCGCCGCCCGACGCCGACGACTGCGCGAACTACGACGCCAACACGA\n",
      "342323 342510\n",
      "2 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_12_OY979626.1_342323_342510_left_2_right_0\n",
      "ACTTCGACAGCGGTCTTTTGGAGACCGACGTGGAGATGCAGCGCCTGATCGACTCGCAGGTGGAGTCGCACAAGATCGGCGCCACCATAGCCGAGGAGGCGCTGCGCCGCGCGTCCGCCGACCTGTCGCCCGACCTGCCGCCCGACCTGCCGCCCGACCTGCTACCCGACAACATGTTCAGCCAGCAG\n",
      "342798 343073\n",
      "0 0\n",
      "Getting Gene\n",
      ">Coenonympha_glycerion_Exon_13_OY979626.1_342798_343073_left_0_right_0\n",
      "TCGCTGATGGACAACCTGCTGGGCGCGGAGGGCTACGAGCCGCAGGTGCGCAACAACGTGCCGCtgagcgcggcgcgcgcgcgctcgcccgcgtCGCCGCTGCGCGAGCTGCGCCCCGCCGCCGCCAGCACGCcgcgcgcgtcgccgccgctgccgccgctcgGCCCCGACGGCAACGGCGAGGCCGCCATGGCCGTCATCATGAGCCTGCTCGAGGCCGACGCGGGGCTCGGCGGGCCCGTCAGCTTCTCGGGCCTGCCCTGGCCGCTGCCGTGA\n",
      "Lasiommata_megera\n",
      "GCA_928268935.1_ilLasMege1.1_genomic.fna\n",
      "27393173 27393264\n",
      "0 2\n",
      "Getting Gene\n",
      "MAREFPRINEFFFQLQELQPQPSHFTFPYD\n",
      ">Lasiommata_megera_Exon_1_OV743307.1_27393173_27393264_left_0_right_2\n",
      "ATGGCGCGCGAATTCCCGCGTATTAACGAGTTCTTTTTCCAACTGCAAGAGCTGCAGCCGCAGCCGTCGCATTTCACATTCCCGTACGATTT\n",
      "27416289 27416471\n",
      "1 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_2_OV743307.1_27416289_27416471_left_1_right_2\n",
      "GCCGCACACGAGCGCGCAGCCGGCCTACGAGCTGACGGTGGGCGTCGGCGTGGAGAGCGCCAGCGCCGACGACTGCTCGCGCGCGCTCGCGCTGCCGCACCTGCAGCTCGCGCACCCGCACGCGCGCTCGCAGCTGCAGCACGAAGCCAGCCTCGCCTGCAAGGAGACCGTCACCCAGAACAG\n",
      "27420347 27420439\n",
      "1 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_3_OV743307.1_27420347_27420439_left_1_right_2\n",
      "TTCGGACGGCTTCGACTTGGCGAGTTGCGATGTGCTCTCCGTGTCGTCCACTTCCACCGCCACCACCCCTAGGAAACGGAAACCGTCGCCCTA\n",
      "27428736 27428810\n",
      "1 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_4_OV743307.1_27428736_27428810_left_1_right_2\n",
      "CAGCACGTGCAGCACCTACGACGACGACGGCAACGAGGACACACGATCTAACCGCAGCAGTATACCGGATAAGCG\n",
      "27429756 27429902\n",
      "1 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_5_OV743307.1_27429756_27429902_left_1_right_2\n",
      "GCAAAACCACAGCGAGATAGAGAAGCGCCGCCGCGACAAGATGAACAACTACATCACGGAGCTGAGCGCCATGATCCCCATGTGCAGCGCCATGCCGCGCAAGCTCGACAAGCTCACCGTGCTGCGCATGGCGGTGCAGCACATCAA\n",
      "27430263 27430452\n",
      "1 0\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_6_OV743307.1_27430263_27430452_left_1_right_0\n",
      "GTCCATCCGCGGCGCCATGTCCGCGGGCCCGCTGACGCCGCGGCCGCGGCCCGCCTTCCTTCCGGAGCGGGACCTGAACGGGATGGTGCTGGAGGCGGCCAAGGACTGCTTCCTGATGGTGGTGGGCTGCGACCGCGGCCGCCTGCTCTACGTGTCCGCCTCGGTGACCAGCATGCTGCAGTACCAGAAG\n",
      "27430778 27430895\n",
      "0 1\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_7_OV743307.1_27430778_27430895_left_0_right_1\n",
      "GCGGAGCTGCTGGGACAGAGCCTGTTCGACATCCTGCACCCGAAGGACGTGGCCAAGGTGAAGGAGCAGCTGTCGTCGGCCGACCTCTGCCCGCGCGAGCGCCTCATCGACGCCAAGA\n",
      "27431863 27432048\n",
      "2 1\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_8_OV743307.1_27431863_27432048_left_2_right_1\n",
      "CCATGCTGCCGGTGAAGGCGGAGCTGGCGGCGGCGACGCCGCGCCTGGGCCCGGGCGCGCGACGCTCCTTCTTCTGCCGCATCAAGTGCCACGCCTCCGGCGGCGTCAAGGAGGAGGCCGAGCACATGCACGGCGCCCCGCGCGCCTCGCTGGCCGAGAAGAAGTACTGCGTGGTGCAGTGCACAG\n",
      "27432126 27432342\n",
      "2 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_9_OV743307.1_27432126_27432342_left_2_right_2\n",
      "GCTACCTGAAGTCGTGGCCGCCCGCCGACGTGTGCGAGCCCGGCGGCGAGGGCGCGGCGGCCGACGAGGGCGACGCCTGCAACCTGTCGTGCCTGGTGGCGGTGGGGCGCTGCCTGGACGACCTGACGCCGCTCGTGACGCCCGCGCCCTACAAGCGCCTGCTGCAGTTCACCTCGCGGCACGCCACCGACAGCAAGTTCGTGTTCGTGGACCAACG\n",
      "27432429 27432632\n",
      "1 2\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_10_OV743307.1_27432429_27432632_left_1_right_2\n",
      "AGTGACGCTGGCGCTGGGCTTCCTGCCGCAGGAGCTGCTGGGCACGAGCCTGTACGAGCACGTGGGCGCGCCGGAGCTGGGCGCGGTGGCGCGCACGCACAAGGCGGCGCTGCTGCAGCGCGGCGCGCTCAAGACGCCGCTCTACAATTTCAAGAAGAAGGACGGCTCGTACGTGCGCATGCAGACGCACTTCAAGCCTTTCAA\n",
      "27433149 27433258\n",
      "1 1\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_11_OV743307.1_27433149_27433258_left_1_right_1\n",
      "CAACCCGTGGACGAAGGACGTGGAGTGCCTGATCGCCAACAACACGGTGGTGACGGACGAGTGGGCGCAGGCCGTCGCCGGCGTCTGCGCCTACGACCTGTACAACCGAA\n",
      "27435527 27435717\n",
      "2 0\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_12_OV743307.1_27435527_27435717_left_2_right_0\n",
      "GCGAGCCGTTCGTCGGTCACGTTGATAGTGGGCTTTCAGAAGCCGACGCGGAGATGCAACGACTGATCGACTCGCAGGTGGAGTCGCACAAGATCGGCTCCACGATCGCCGAGGAGGTGTTGAGGTCGTCCACGGAGTACTCGGCTGAACCCGCGGCAGATTTGCTGCAAGACGCTGTGTTTGTCCAGCAG\n",
      "27438007 27438198\n",
      "0 0\n",
      "Getting Gene\n",
      ">Lasiommata_megera_Exon_13_OV743307.1_27438007_27438198_left_0_right_0\n",
      "GGTCGCCGCGGCTCGCCCGCGCCCGACGCGCCCTGCGCCAGCAGCCCCCCGCGCGCCTCGCCGCCCCTGGCGCCGCTAGAGCCGGACGGCAACGGCGAGGCGGCCATGGCCGTCATCATGAGCCTGCTGGAGGCCGACGCGGGGCTCGGCGGGCCCGTCAGCTTCTCCAGCCTGCCCTGGCCGCTGCCGTAG\n",
      "Pararge_aegeria\n",
      "GCF_905163445.1_ilParAegt1.1_genomic.fna\n",
      "26655097 26655188\n",
      "0 2\n",
      "Getting Gene\n",
      "MSREFPRVNEFYYHLQDLQPPTPHYTYPFE\n",
      ">Pararge_aegeria_Exon_1_NC_053208.1_26655097_26655188_left_0_right_2\n",
      "ATGTCGCGCGAATTTCCGCGTGTTAACGAGTTCTATTACCATCTGCAAGACCTGCAACCGCCGACGCCGCACTACACCTACCCGTTCGAGCT\n",
      "26693758 26693997\n",
      "1 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_2_NC_053208.1_26693758_26693997_left_1_right_2\n",
      "GCCGCACACGAGCACGCAGCCGGCCTACGAGCTGACAGTGGGTGCCGGCGTCCAGGACGCTAGCGCCGCCGGCTTTGCGGAGGTTGCGGGTGCCGGCGCGCTGCTTTCCCTGCACGCGCCGCTGCCGCACCTGCAGCACCCGCTGCCACTGCCGCTGCCGGCACCACTGGCAATGCCGCTGCCGGTGTCCCAGCGCGATGTCGCCGCCGCCGTCCTCAAGAGGAGCTCCTCGCAGAACAT\n",
      "26696605 26696694\n",
      "1 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_3_NC_053208.1_26696605_26696694_left_1_right_2\n",
      "TTCGGACGGGTTCGAGTTGACGAATTGCGAAGTGATCACCATCTCGTCCACTTCCGACGCCGCCGCCACGAGGAAGCGGAAACCATCGCC\n",
      "26702466 26702540\n",
      "1 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_4_NC_053208.1_26702466_26702540_left_1_right_2\n",
      "CAGTACGTGCAGCACCTACGACGATGACGGCAACGAGGATTCGCGATCTAACCGCAGCACCTTACCTGATAAGCG\n",
      "26704210 26704356\n",
      "1 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_5_NC_053208.1_26704210_26704356_left_1_right_2\n",
      "GCAAAACCACAGCGAGATCGAGAAGCGGCGCCGCGACAAGATGAACAACTACATCACTGAGCTTAGCTCCATGATCCCGATGTGCACCGCCATCATGCGCAAGCTCGACAAGCTCACCGTGCTGCGCATGGCGGTGCAGCACATGAA\n",
      "26705394 26705583\n",
      "1 0\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_6_NC_053208.1_26705394_26705583_left_1_right_0\n",
      "AAGCATCCGCGGCGCCATGTCGGCGGGCCCGCTGTCGGTGCGTCCGAAGCCCACCTTCCTGTCGGATCGCGCGCTCAACGGCATGGTGCTGGAGGCCGCCAAGGACTGCTTCGTCATGGTGGTGAGCTGCTCGCGCGGCCGCCTGCTCTACGTGTCCGCCTCCGTCACCAACATGCTGCAGTACGACAAG\n",
      "26706437 26706554\n",
      "0 1\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_7_NC_053208.1_26706437_26706554_left_0_right_1\n",
      "GCGGAGTTGCTGGGGCAAAGCCTGTTCGACATCCTGCACCCGAAGGACGTGGCCAAAGTGAAGGAGCAGTTGTCGTCGTCAGACCTCAGCCCGCGGGAGCGCCTCATCGACGCCAAGA\n",
      "26707150 26707353\n",
      "2 1\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_8_NC_053208.1_26707150_26707353_left_2_right_1\n",
      "CCATGCTGCCGGTGAAGTCGATGGTGTCGGGCGCGTCGCGGCTGTGCGCGGGCGCGCGGCGCTCGTTCTTCTGCCGCATCAAGTGCCGGCCCGCTACCGTCATCACGGCGACCAGCGGCGGCGCCGCGCGCGTCAAGGAGGAGCACAGCCAGCCCAAGGCGCGCAAGCTGGCCGATAAGAAGTACTGCGTCGTGCAGTGCACCG\n",
      "26708337 26708544\n",
      "2 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_9_NC_053208.1_26708337_26708544_left_2_right_2\n",
      "GCTACCTGAAGTCGTGGGCGCCCGCCAACATCAGCGAGACGGGCGATGGCGCGGCCGACGAGAGCGACTGCAACTTGTCGTGCTTGGTGGCCGTGGGGCGCTGCCTGGCCGACCTGACGCCGCTGGTGGCGCCCGCGCCGCACAAGCGCAGCCTCGCGTTCGCCTCGCGTCACACCACCGACAGCAAGTTCGTCTTCGTTGACCAGCG\n",
      "26709023 26709226\n",
      "1 2\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_10_NC_053208.1_26709023_26709226_left_1_right_2\n",
      "GGTGACGCTGGTGCTGGGCTACCTGCCGCAAGAGCTGCTGGGCACCAGTCTGTACGAGCACGTGAGCGCCCCGGAGTTGGGCGCGGTGGCCAAGACGCACAAGACTGCGCTCCTGCAGCGCAACCCGCTCAAGACGCCGCCCTACAGCTTCAAGAAGAAGGACGGCACCTACGTGCGCATACAGACGCACTTCAAACCCTTCAA\n",
      "26709795 26709904\n",
      "1 1\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_11_NC_053208.1_26709795_26709904_left_1_right_1\n",
      "CAACCCGTGGACCAAGGACGTGGAGTGCTTGATCGCCAACAACACTGTGGTGACGGAGGAGTGGGCGCAGCCGGCCGCACGTGGAGAGTGCTACGACATGTACAACAGAA\n",
      "26712373 26712563\n",
      "2 0\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_12_NC_053208.1_26712373_26712563_left_2_right_0\n",
      "GCGAGCCGTACAGCGATCAGGTTGATAGTGGGGTTTCAGAGGCCGACCTGGAGATGCAGCGCCTAATTGACTCGCAGGTGGAATCGCACAAGATCGGCTCCACGATCGTCCAGGAGGTATTGAGGGCGTCCACGGAGTTACCGTCCAATCTTTCATCAGACGTGCTTCAGGACTCTGCGTACATCCAAGAG\n",
      "26713608 26713847\n",
      "0 0\n",
      "Getting Gene\n",
      ">Pararge_aegeria_Exon_13_NC_053208.1_26713608_26713847_left_0_right_0\n",
      "CCGGAGTTGCTGGAGGGCTCGCAGCCGATGTACGAGCAGGTGCGCCACCTGGTGGCGCCGAGCGCGGACGGGGCCCCGCCGTCTCCGCGGGCGGCCACCCCTCCGCCGCGCATTGCGCCGCCGCACGAGCTGGACGGCAACGGCGAAGCCACCATGGCGGTCATCATGAGCATGCTGGAGGCCGAGGCCGGGCTGGGCGGGCCCATCAACTACTCCAGCCTGCCCTGGCCCCTGCCCTAA\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"3.Satyrine_Exon5_split\"\n",
    "\n",
    "output_location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/10.Cycle_Exon_Analysis/{family_group}\"\n",
    "make_output_folder(output_location)\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "if 'desktop.ini' in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Heliconius_charithonia\"]\n",
    "# list_of_species = [\"Maniola_hyperantus\"]\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "        \n",
    "            \n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "            print(left_oh, right_oh)\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            query_name = coordinate_line_split[6]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            continue\n",
    "            \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "       \n",
    "        current_exon = query_name.split('_')[-1]\n",
    "        if current_exon == \"1\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "            print(translated)\n",
    "            if \"*\" in translated:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "        \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "        elif current_exon == \"16\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence[:-int(left_oh)].reverse_complement.translate()\n",
    "                print(gene_sequence)\n",
    "            else:\n",
    "                translated = gene_sequence[int(left_oh):].translate()\n",
    "                print(gene_sequence)\n",
    "\n",
    "            if \"*\" in translated[:-1]:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                print(translated)\n",
    "            else:\n",
    "                \n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "#         assert False\n",
    "        else:\n",
    "            exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "            print(exon_output)\n",
    "\n",
    "            write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
