{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f27d4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    out_Seq = ''\n",
    "    \n",
    "    \n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = SeqIO.parse((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'fasta')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in fasta_file:\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = out_Seq.reverse_complement()\n",
    "            break\n",
    "                \n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    \n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4900aae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "85ed0aea",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_sequence(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = SeqIO.parse((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'fasta')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in fasta_file:\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "#             out_Seq = Seq(sequence[start-1:end])\n",
    "#             if reverse_c == 1:\n",
    "#                 sequence = out_Seq.reverse_complement()\n",
    "            return(sequence)\n",
    "    \n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d454c091",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_plexippus\n",
      "LHNLIPPSPFYYRY\n",
      "GLHNLIPPSPFYYRY\n",
      "DGLHNLIPPSPFYYRY\n",
      "ADGLHNLIPPSPFYYRY\n",
      "CADGLHNLIPPSPFYYRY\n",
      "Hestina_assimilis,CM041748.1,11457807,11457871,1,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,15,28,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18263471,18263629,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,2,52,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,0,0,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,0,0,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,0,0,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,0,0,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18268004,18268340,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18268710,18268827,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18268954,18269190,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18269273,18269486,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18269923,18270126,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18270622,18270734,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,36,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18271722,18271897,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,3,56,58\n",
      "\n",
      "Danaus_plexippus\n",
      "Hestina_assimilis,CM041759.1,18272710,18272982,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,2,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Hestina_assimilis,CM041748.1,11457807,11457871,1,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,15,28,30\n",
      "Hestina_assimilis,CM041759.1,18263471,18263629,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,2,52,55\n",
      "Hestina_assimilis,CM041759.1,0,0,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,0,0,35\n",
      "Hestina_assimilis,CM041759.1,0,0,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,0,0,23\n",
      "Hestina_assimilis,CM041759.1,18268004,18268340,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Hestina_assimilis,CM041759.1,18268710,18268827,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Hestina_assimilis,CM041759.1,18268954,18269190,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Hestina_assimilis,CM041759.1,18269273,18269486,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Hestina_assimilis,CM041759.1,18269923,18270126,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Hestina_assimilis,CM041759.1,18270622,18270734,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,36,37\n",
      "Hestina_assimilis,CM041759.1,18271722,18271897,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,3,56,58\n",
      "Hestina_assimilis,CM041759.1,18272710,18272982,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,2,90,90\n",
      "\n",
      "Danaus_plexippus\n",
      "EYPRVHDYYLQLQDYQPSPSHYSYQYDF\n",
      "REYPRVHDYYLQLQDYQPSPSHYSYQYDF\n",
      "AREYPRVHDYYLQLQDYQPSPSHYSYQYDF\n",
      "Charaxes_varanes,JANCMC010022294.1,116126,116220,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,4,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010033613.1,9807589,9807771,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,2,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010037582.1,50345,50452,1,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,6,24,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010043018.1,3111,3182,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010045438.1,1276,1612,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010033613.1,20964739,20964856,0,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010033613.1,20965574,20965810,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010033613.1,20966026,20966248,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010033613.1,20966534,20966737,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010044260.1,13931,14043,1,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,36,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Charaxes_varanes,JANCMC010044260.1,13057,13235,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,6,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPE\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEI\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEIS\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEIST\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTH\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHS\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSS\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSST\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTS\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSP\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPP\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPS\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQ\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQA\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQAL\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALH\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHL\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLN\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLND\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLNDY\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLNDYR\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLNDYRP\n",
      "TLVENILGGDSMNYSDQIRNNVTLSAAGAEGSPLPEISTHSSTSPPSQALHLNDYRPP\n",
      "Charaxes_varanes,JANCMC010044260.1,12194,12373,1,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,2,35,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Charaxes_varanes,JANCMC010022294.1,116126,116220,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,4,30,30\n",
      "Charaxes_varanes,JANCMC010033613.1,9807589,9807771,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,2,55,55\n",
      "Charaxes_varanes,JANCMC010037582.1,50345,50452,1,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,6,24,35\n",
      "Charaxes_varanes,JANCMC010043018.1,3111,3182,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Charaxes_varanes,JANCMC010045438.1,1276,1612,0,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Charaxes_varanes,JANCMC010033613.1,20964739,20964856,0,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Charaxes_varanes,JANCMC010033613.1,20965574,20965810,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Charaxes_varanes,JANCMC010033613.1,20966026,20966248,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Charaxes_varanes,JANCMC010033613.1,20966534,20966737,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Charaxes_varanes,JANCMC010044260.1,13931,14043,1,Y,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,36,37\n",
      "Charaxes_varanes,JANCMC010044260.1,13057,13235,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,6,58,58\n",
      "Charaxes_varanes,JANCMC010044260.1,12194,12373,1,Y,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,2,35,90\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,610233,610324,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,599149,599316,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,598451,598558,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,596996,597067,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,596583,596919,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,596280,596397,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,595942,596190,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,595236,595449,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,594685,594888,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,593752,593864,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,593449,593624,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_chrysippus,CAKASE010000043.1,592850,593119,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Danaus_chrysippus,CAKASE010000043.1,610233,610324,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "Danaus_chrysippus,CAKASE010000043.1,599149,599316,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "Danaus_chrysippus,CAKASE010000043.1,598451,598558,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "Danaus_chrysippus,CAKASE010000043.1,596996,597067,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Danaus_chrysippus,CAKASE010000043.1,596583,596919,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Danaus_chrysippus,CAKASE010000043.1,596280,596397,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Danaus_chrysippus,CAKASE010000043.1,595942,596190,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Danaus_chrysippus,CAKASE010000043.1,595236,595449,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Danaus_chrysippus,CAKASE010000043.1,594685,594888,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Danaus_chrysippus,CAKASE010000043.1,593752,593864,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "Danaus_chrysippus,CAKASE010000043.1,593449,593624,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "Danaus_chrysippus,CAKASE010000043.1,592850,593119,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,555458,555549,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,545185,545352,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,544441,544548,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,543163,543234,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,542739,543075,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,542419,542536,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,542061,542309,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,541653,541866,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,541166,541369,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,540654,540766,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,540354,540529,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "Danaus_plexippus,NC_083559.1,540002,540271,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Danaus_plexippus,NC_083559.1,555458,555549,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "Danaus_plexippus,NC_083559.1,545185,545352,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "Danaus_plexippus,NC_083559.1,544441,544548,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "Danaus_plexippus,NC_083559.1,543163,543234,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Danaus_plexippus,NC_083559.1,542739,543075,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Danaus_plexippus,NC_083559.1,542419,542536,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Danaus_plexippus,NC_083559.1,542061,542309,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Danaus_plexippus,NC_083559.1,541653,541866,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Danaus_plexippus,NC_083559.1,541166,541369,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Danaus_plexippus,NC_083559.1,540654,540766,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "Danaus_plexippus,NC_083559.1,540354,540529,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "Danaus_plexippus,NC_083559.1,540002,540271,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,732062,732153,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,717437,717583,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,715678,715785,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,713925,713996,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,712928,713264,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,712441,712558,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,712070,712294,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,711739,711952,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,711244,711447,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,710868,710980,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,710290,710465,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta,OY743092.1,709052,709321,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Dircenna_loreta,OY743092.1,732062,732153,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "Dircenna_loreta,OY743092.1,717437,717583,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "Dircenna_loreta,OY743092.1,715678,715785,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "Dircenna_loreta,OY743092.1,713925,713996,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Dircenna_loreta,OY743092.1,712928,713264,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Dircenna_loreta,OY743092.1,712441,712558,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Dircenna_loreta,OY743092.1,712070,712294,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Dircenna_loreta,OY743092.1,711739,711952,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Dircenna_loreta,OY743092.1,711244,711447,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Dircenna_loreta,OY743092.1,710868,710980,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "Dircenna_loreta,OY743092.1,710290,710465,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "Dircenna_loreta,OY743092.1,709052,709321,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20270161,20270252,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20285048,20285194,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20287621,20287728,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20289132,20289203,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20289534,20289870,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20290259,20290376,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20291770,20291994,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20292085,20292298,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20293187,20293390,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20293638,20293750,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_mazaeus,OY365748.1,20294010,20294185,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "SSLVDNILGVELNYSNQVRNNVPLSAAGGSSPSPEAEIPLSPLPASPPLPPLGLDGNGEAAMAVIMSLLEADAGLGGPVNISGLPWPLP\n",
      "Mechanitis_mazaeus,OY365748.1,20294289,20294558,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,89,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Mechanitis_mazaeus,OY365748.1,20270161,20270252,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "Mechanitis_mazaeus,OY365748.1,20285048,20285194,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "Mechanitis_mazaeus,OY365748.1,20287621,20287728,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "Mechanitis_mazaeus,OY365748.1,20289132,20289203,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Mechanitis_mazaeus,OY365748.1,20289534,20289870,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Mechanitis_mazaeus,OY365748.1,20290259,20290376,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Mechanitis_mazaeus,OY365748.1,20291770,20291994,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Mechanitis_mazaeus,OY365748.1,20292085,20292298,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Mechanitis_mazaeus,OY365748.1,20293187,20293390,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Mechanitis_mazaeus,OY365748.1,20293638,20293750,0,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "Mechanitis_mazaeus,OY365748.1,20294010,20294185,0,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "Mechanitis_mazaeus,OY365748.1,20294289,20294558,0,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,89,90\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,600610,600701,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,585520,585666,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,583011,583118,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,581504,581575,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,580837,581173,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,580331,580448,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,578740,578964,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,578429,578642,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,577343,577546,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,576991,577103,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,576556,576731,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "\n",
      "Danaus_plexippus\n",
      "Mechanitis_messenoides,OY365760.1,576183,576452,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Mechanitis_messenoides,OY365760.1,600610,600701,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_2_query_Exon_1,1,30,30\n",
      "Mechanitis_messenoides,OY365760.1,585520,585666,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_2,1,55,55\n",
      "Mechanitis_messenoides,OY365760.1,583011,583118,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_3,1,35,35\n",
      "Mechanitis_messenoides,OY365760.1,581504,581575,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_4,1,23,23\n",
      "Mechanitis_messenoides,OY365760.1,580837,581173,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_0_query_Exon_5,1,112,112\n",
      "Mechanitis_messenoides,OY365760.1,580331,580448,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_1_query_Exon_6,1,39,39\n",
      "Mechanitis_messenoides,OY365760.1,578740,578964,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_1_query_Exon_7,1,82,82\n",
      "Mechanitis_messenoides,OY365760.1,578429,578642,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_2_query_Exon_8,1,70,70\n",
      "Mechanitis_messenoides,OY365760.1,577343,577546,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_2_query_Exon_9,1,67,67\n",
      "Mechanitis_messenoides,OY365760.1,576991,577103,1,N,Danaus_plexippus_XM_061526053.1_Frame_1_rightoh_1_query_Exon_10,1,37,37\n",
      "Mechanitis_messenoides,OY365760.1,576556,576731,1,N,Danaus_plexippus_XM_061526053.1_Frame_2_rightoh_0_query_Exon_11,1,58,58\n",
      "Mechanitis_messenoides,OY365760.1,576183,576452,1,N,Danaus_plexippus_XM_061526053.1_Frame_0_rightoh_0_query_Exon_12,1,90,90\n",
      "\n",
      "Danaus_plexippus\n"
     ]
    }
   ],
   "source": [
    "#%reset -f\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "\n",
    "family_group = \"6.Heliconiinae_Danainae_Nymphalinae\"\n",
    "blast_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/10.Cycle Exon Analysis/{family_group}\"\n",
    "query_location = f\"{blast_location}/1.Query\"\n",
    "query_species = \"Danaus_plexippus\"\n",
    "query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "if \"desktop.ini\" in query_transcript_list:\n",
    "    query_transcript_list.remove(\"desktop.ini\")\n",
    "\n",
    "query_transcript = query_transcript_list[0]\n",
    "#Get Species list, the list may be a list of files or a single species\n",
    "\n",
    "species_list = os.listdir(f\"{blast_location}/1.Blast_result\")\n",
    "# species_list = [\"Pararge_aegeria\"]\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove('desktop.ini')\n",
    "\n",
    "#get list of query names\n",
    "\n",
    "query_file = SeqIO.parse(f\"{blast_location}/1.Query/{query_species}/{query_transcript}/query_protein.fa\", 'fasta')\n",
    "query_name_list = []\n",
    "\n",
    "for records in query_file:\n",
    "    query_name_list.append(records.id)\n",
    "\n",
    "\n",
    "\n",
    "for species in species_list:\n",
    "    # print(species_name)\n",
    "    # species = species_name.split(\".\")[1]\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    \n",
    "#Run for each query in the query list\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "#         print(f\"Query name = {query_name}\")\n",
    "        Length_switch = \"0\"\n",
    "        with open(f\"{blast_location}/1.Blast_result/{species}/{species}_blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "#                 print(lines)\n",
    "#                 assert False\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "#                 query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "#                 query_species = str(query_species_split[0].split(\".\")[1]+\"_\"+query_species_split[1].rstrip())\n",
    "                print(query_species)\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            length = 0\n",
    "            start = 000\n",
    "            stop = 000\n",
    "            query_start_coor = 000\n",
    "            query_stop_coor = 000\n",
    "            error = \"Y\"\n",
    "            complement  = 0\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "        \n",
    "    \n",
    "    \n",
    "        \n",
    "\n",
    "#Check for Met or STOP at the start and end\n",
    "        sequence = ''\n",
    "        while True and (start != 0 or stop != 0):\n",
    "            if (query_name == query_name_list[0] or query_name == query_name_list[-1]):\n",
    "                frame = 1\n",
    "#                 translated_sequence = (sequence_extractor(species, scaff, int(complement), start, stop,frame ))\n",
    "                if sequence == '':\n",
    "                    sequence = get_sequence(species, scaff, int(complement), start, stop,frame)\n",
    "                if int(complement) == 1:\n",
    "                    translated_sequence = Seq(sequence[start-1:stop]).reverse_complement().translate()\n",
    "                else:\n",
    "                    translated_sequence = Seq(sequence[start-1:stop]).translate()\n",
    "                if translated_sequence == old_trans:\n",
    "                    break\n",
    "                    \n",
    "#Check for Met at the beginning of first exon\n",
    "\n",
    "                if query_name == query_name_list[0] :\n",
    "                    if translated_sequence[0] != \"M\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            start = int(start) - 3\n",
    "                        if complement == \"1\":\n",
    "                            stop = int(stop) + 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if \"*\" in translated_sequence:\n",
    "                        error = \"Y\"\n",
    "                        break\n",
    "                    if translated_sequence[0] == \"M\":\n",
    "                        break\n",
    "\n",
    "#Check for stop at the end of last exon\n",
    "\n",
    "                if query_name == query_name_list[-1]:\n",
    "                    if translated_sequence[-1] != \"*\":\n",
    "                        print(translated_sequence)\n",
    "#                             print(start, stop)\n",
    "                        if complement == \"0\":\n",
    "                            stop = int(stop) + 3\n",
    "                        if complement == \"1\":\n",
    "                            start = int(start) - 3\n",
    "                        old_trans = translated_sequence\n",
    "                    if translated_sequence[-1] == \"*\":\n",
    "\n",
    "                        break\n",
    "            else:\n",
    "                break\n",
    "        \n",
    "        \n",
    "#Add offset at the beginning and end of each exon\n",
    "        if (start != 0 or stop != 0):\n",
    "            \n",
    "            start_modifier = int(query_name.split(\"Frame\")[1][1])\n",
    "            stop_modifier = int(query_name.split(\"rightoh\")[1][1])\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0            \n",
    "\n",
    "        seq_length = query_length\n",
    "        \n",
    "        \n",
    "#Get exon start and end by adding or removing codons\n",
    "        \n",
    "    #if not the beginning\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "            \n",
    "           \n",
    "        output_format = str(species.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        print(output_format)\n",
    "#         assert False\n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "   \n",
    "    output_file = open(f\"{blast_location}/1.Blast_result/{species}/{species}_coordinates_old.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fa07f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465fa823",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
