{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fbb53c69",
   "metadata": {},
   "outputs": [],
   "source": [
    "family_group = \"4.Pierinae\"\n",
    "def make_species_folder(output_location, species):\n",
    "    \n",
    "    list_of_folders = os.listdir(f\"{output_location}/4.Bed File/\")\n",
    "    if species not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/4.Bed File/{species}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "73f14383",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Anthocharis_cardamines Anthocharis_cardamines,FR989951.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Aporia_crataegi Aporia_crataegi,OU538732.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Aporia_crataegi Aporia_crataegi,OU538732.1,000,000,0,Y,Error_Exon_22,00,00,00\n",
      "\n",
      "Leptophobia_aripa Leptophobia_aripa,OX637275.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Pieris_brassicae Pieris_brassicae,NC_059680.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Pieris_mannii Pieris_mannii,CM054800.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Pieris_melete Pieris_melete,CAVNZK010000320.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n",
      "Pieris_rapae Pieris_rapae,NC_059534.1,000,000,0,Y,Error_Exon_20,00,00,00\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#for variant 1\n",
    "import os\n",
    "import subprocess\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders_1 = os.listdir(f\"{output_location}\")\n",
    "if \"4.Bed File\" in list_of_folders_1:\n",
    "    subprocess.run(f'rm -r \"{output_location}/4.Bed File\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{output_location}/4.Bed File\")\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "\n",
    "for species in list_of_species:\n",
    "#     print(species)\n",
    "    \n",
    "    output = ''\n",
    "    variant_presence = 1\n",
    "    \n",
    "    list_of_files_in_species_folder = os.listdir(f\"{output_location}/1.Blast_result/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        if file_names.endswith(\"_coordinates.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "    all_good = 1\n",
    "    for i in range(1,len(coordinate_file_lines)):\n",
    "#         print(i)\n",
    "        split_coordinate_line = (coordinate_file_lines[i]).split(\",\")\n",
    "        scaffold = split_coordinate_line[1]\n",
    "        complement = split_coordinate_line[4]\n",
    "        if complement == \"0\":\n",
    "            chr_start =int(split_coordinate_line[2])\n",
    "            chr_end = int(split_coordinate_line[3])+1\n",
    "        elif complement == \"1\":\n",
    "            chr_start =int(split_coordinate_line[2])\n",
    "            chr_end = int(split_coordinate_line[3])+1\n",
    "        exon_number = split_coordinate_line[6].split(\"_\")[-1] \n",
    "        if chr_start > 5 and chr_end > 5:\n",
    "            output +=f\"{scaffold}\\t{chr_start}\\t{chr_end}\\tExon_{exon_number}\\n\" \n",
    "        else:\n",
    "            print(species, coordinate_file_lines[i])\n",
    "            all_good = 0\n",
    "            \n",
    "    if all_good == 1:\n",
    "        make_species_folder(output_location, species)\n",
    "        with open(f\"{output_location}/4.Bed File/{species}/Transcript.bed\", 'w') as out_file:\n",
    "            out_file.write(output)\n",
    "#     print(output)\n",
    "    \n",
    "#     assert False\n",
    "    \n",
    "#     for exon_number in range(1,28):\n",
    "        \n",
    "        \n",
    "        \n",
    "#         with open(f\"{output_location}/2.Final_output/{species}/{exon}\", 'r') as exon_file:\n",
    "#             exon_name_split = exon_file.readlines()[0].split(\"_\")\n",
    "#             if len(exon_name_split[4]) < 4:\n",
    "#                 scaffold = exon_name_split[4]+\"_\"+ exon_name_split[5]\n",
    "\n",
    "#             else:\n",
    "#                 scaffold = exon_name_split[4]\n",
    "#             chr_start = exon_name_split[-6]\n",
    "#             chr_end = exon_name_split[-5]\n",
    "# #                 print(scaffold, chr_start,chr_end)\n",
    "#             output +=f\"{scaffold}\\t{chr_start}\\t{chr_end}\\tExon_{exon_number}\\n\" \n",
    "    \n",
    "#     if variant_presence == 1:\n",
    "#         with open(f\"{output_location}/4.Bed File/{species}/Transcript_vairant_{Transcript_variant}.bed\", 'w') as out_file:\n",
    "#             out_file.write(output)\n",
    "#     print(output)\n",
    "# #     assert False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "572f54f4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
