{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "03175890",
   "metadata": {},
   "outputs": [],
   "source": [
    "family_group = \"4.Pierinae\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cf58414a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "   \n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp\")\n",
    "\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\"')\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\", \"fasta\")\n",
    "    # genome = SeqIO.parse(\"/mnt/c/Users/sauba/Downloads/Compressed/ncbi_dataset_7/ncbi_dataset/data/GCA_917051295.2/GCA_917051295.2_ilEreLige1.2_genomic.fna\", 'fasta')\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    return (gene_sequence)\n",
    "\n",
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)   \n",
    "\n",
    "def get_sequence(blast_output_location,genome_location, species, genome_file,annotated_genome_location,annotated_species_name):\n",
    "    list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        if file_names.endswith(\"_coordinates.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "\n",
    "    with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "\n",
    "    upstream_exon_line =coordinate_file_lines[1].split(\",\")\n",
    "    if upstream_exon_line[5] == \"Y\":\n",
    "        print(\"First Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    downstream_exon_line = coordinate_file_lines[-1].split(\",\")\n",
    "    if downstream_exon_line[5] == \"Y\":\n",
    "        print(\"Last Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    \n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "\n",
    "    if gene_start < 100000:\n",
    "        print(\"Gene Start Issue\")\n",
    "        assert False\n",
    "    gene_start = gene_start - 100000\n",
    "    gene_end = gene_end + 100000\n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    print(species)\n",
    "    return(gene_sequence,scaffold,gene_start,gene_end )\n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "42d7701e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_species_folder(output_location, species):\n",
    "    list_of_folders = os.listdir(f\"{output_location}/5.For Email/\")\n",
    "    if species not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/5.For Email/{species}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c9b9527b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import io\n",
    "import subprocess\n",
    "genome_location =  \"/mnt/f/Genomes_2023-12-26\"\n",
    "# genome_location =  \"/mnt/g/Genomes_2023-12-26\"\n",
    "blast_output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}/1.Blast_result\"\n",
    "# blast_output_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "\n",
    "\n",
    "for species in list_of_species:\n",
    "\n",
    "    annotated_genome_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    # annotated_genome_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "\n",
    "    gene_sequence,scaffold, gene_start, gene_end = get_sequence(blast_output_location,genome_location, species, genome_file,\n",
    "                                                               annotated_genome_location,annotated_species_name)\n",
    "\n",
    "#     print(gene_sequence,scaffold, gene_start, gene_end)\n",
    "    with open(f\"{output_location}/5.For Email_padded/{species}/Genome_fragment.fa\",'w') as out_file:\n",
    "        output = f\">{species}_{scaffold}_{gene_start}_{gene_end}\\n{gene_sequence}\"\n",
    "        out_file.write(output)\n",
    "\n",
    "#     local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "#     cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "#         # # os.system(f'{cd_command}')\n",
    "#     subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "#         # # print(mkdb_command)\n",
    "#         # # os.system(f'{mkdb_command}')\n",
    "#     # print(gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0fc007ce",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Anthocharis_cardamines\n",
      "['FR989951.1\\t4965980\\t4966057\\tExon_1\\n', 'FR989951.1\\t4965133\\t4965217\\tExon_2\\n', 'FR989951.1\\t4962682\\t4962831\\tExon_3\\n', 'FR989951.1\\t4961427\\t4961519\\tExon_4\\n', 'FR989951.1\\t4960954\\t4961014\\tExon_5\\n', 'FR989951.1\\t4960109\\t4960241\\tExon_6\\n', 'FR989951.1\\t4959302\\t4959378\\tExon_7\\n', 'FR989951.1\\t4956461\\t4956671\\tExon_8\\n', 'FR989951.1\\t4955726\\t4955955\\tExon_9\\n', 'FR989951.1\\t4955120\\t4955232\\tExon_10\\n', 'FR989951.1\\t4954904\\t4955018\\tExon_11\\n', 'FR989951.1\\t4953826\\t4953976\\tExon_12\\n', 'FR989951.1\\t4953202\\t4953336\\tExon_13\\n', 'FR989951.1\\t4952026\\t4952122\\tExon_14\\n', 'FR989951.1\\t4951182\\t4951326\\tExon_15\\n', 'FR989951.1\\t4950163\\t4950341\\tExon_16\\n', 'FR989951.1\\t4948598\\t4948752\\tExon_17\\n', 'FR989951.1\\t4947175\\t4947283\\tExon_18\\n', 'FR989951.1\\t4946221\\t4946323\\tExon_19\\n', 'FR989951.1\\t4945380\\t4945637\\tExon_21\\n', 'FR989951.1\\t4941541\\t4941679\\tExon_22\\n', 'FR989951.1\\t4939332\\t4939497\\tExon_23\\n', 'FR989951.1\\t4938054\\t4938141\\tExon_24\\n', 'FR989951.1\\t4937374\\t4937476\\tExon_25\\n', 'FR989951.1\\t4935738\\t4935840\\tExon_26\\n']\n",
      "FR989951.1\t30242\t30319\tExon_1\n",
      "FR989951.1\t29395\t29479\tExon_2\n",
      "FR989951.1\t26944\t27093\tExon_3\n",
      "FR989951.1\t25689\t25781\tExon_4\n",
      "FR989951.1\t25216\t25276\tExon_5\n",
      "FR989951.1\t24371\t24503\tExon_6\n",
      "FR989951.1\t23564\t23640\tExon_7\n",
      "FR989951.1\t20723\t20933\tExon_8\n",
      "FR989951.1\t19988\t20217\tExon_9\n",
      "FR989951.1\t19382\t19494\tExon_10\n",
      "FR989951.1\t19166\t19280\tExon_11\n",
      "FR989951.1\t18088\t18238\tExon_12\n",
      "FR989951.1\t17464\t17598\tExon_13\n",
      "FR989951.1\t16288\t16384\tExon_14\n",
      "FR989951.1\t15444\t15588\tExon_15\n",
      "FR989951.1\t14425\t14603\tExon_16\n",
      "FR989951.1\t12860\t13014\tExon_17\n",
      "FR989951.1\t11437\t11545\tExon_18\n",
      "FR989951.1\t10483\t10585\tExon_19\n",
      "FR989951.1\t9642\t9899\tExon_21\n",
      "FR989951.1\t5803\t5941\tExon_22\n",
      "FR989951.1\t3594\t3759\tExon_23\n",
      "FR989951.1\t2316\t2403\tExon_24\n",
      "FR989951.1\t1636\t1738\tExon_25\n",
      "FR989951.1\t0\t102\tExon_26\n",
      "\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[13], line 42\u001b[0m\n\u001b[1;32m     38\u001b[0m         \u001b[38;5;28mprint\u001b[39m(output)\n\u001b[1;32m     39\u001b[0m \u001b[38;5;66;03m#         with open(f\"{output_location}/5.For Email_padded/{species}/{variant}\", 'w') as out_file:\u001b[39;00m\n\u001b[1;32m     40\u001b[0m \u001b[38;5;66;03m#             out_file.write(output)\u001b[39;00m\n\u001b[1;32m     41\u001b[0m \u001b[38;5;66;03m#         print(complement, gene_start)\u001b[39;00m\n\u001b[0;32m---> 42\u001b[0m         \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/5.For Email_padded\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "# #     print(gene_start)\n",
    "#     make_species_folder(output_location, species)\n",
    "    output = ''\n",
    "    list_of_transcript_files = os.listdir(f\"{output_location}/4.Bed File/{species}/\")\n",
    "    if 'desktop.ini' in list_of_transcript_files:\n",
    "        list_of_transcript_files.remove('desktop.ini')\n",
    "#     start = 0\n",
    "    \n",
    "    for transcript in list_of_transcript_files:    \n",
    "        with open(f\"{output_location}/4.Bed File/{species}/{transcript}\", 'r') as var_file:\n",
    "            var_list = var_file.readlines()\n",
    "        \n",
    "        if int((var_list[0].split(\"\\t\")[1])) <int((var_list[1].split(\"\\t\")[1])):\n",
    "            complement = 0\n",
    "            gene_start = int((var_list[0].split(\"\\t\")[1]))\n",
    "        else:\n",
    "            complement = 1\n",
    "            gene_start = int((var_list[-1].split(\"\\t\")[1]))\n",
    "        output = ''\n",
    "        print(var_list)\n",
    "        for gene_line in  var_list:\n",
    "            gene_line_split = gene_line.split(\"\\t\")\n",
    "#             print(gene_line, var_list)\n",
    "            gene_line_split[1] = str(int(gene_line_split[1]) - gene_start)\n",
    "            gene_line_split[2] = str(int(gene_line_split[2]) - gene_start)\n",
    "            output += \"\\t\".join(gene_line_split)\n",
    "        \n",
    "        print(output)\n",
    "#         with open(f\"{output_location}/5.For Email_padded/{species}/{variant}\", 'w') as out_file:\n",
    "#             out_file.write(output)\n",
    "#         print(complement, gene_start)\n",
    "        assert False\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "007ee2e9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
