{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0288a896",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Get_genome_name(species, genome_location):\n",
    "    import os\n",
    "    list_of_folders = os.listdir(genome_location)\n",
    "    while True:\n",
    "        if species in list_of_folders:\n",
    "            return(species)\n",
    "        else:\n",
    "            input(\"species_missing?\")\n",
    "            break\n",
    "def Genome_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            return(file)\n",
    "\n",
    "def Get_gff_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.gff\"):\n",
    "            return(file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e4c2a786",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Anthocharis_cardamines', 'Aporia_crataegi', 'Leptophobia_aripa', 'Pieris_brassicae', 'Pieris_mannii', 'Pieris_melete', 'Pieris_napi', 'Pieris_rapae']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae\"\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "\n",
    "list_of_species = os.listdir(f\"{location}/1.Blast_result/\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "print(list_of_species)\n",
    "\n",
    "for species in list_of_species:\n",
    "\n",
    "    with open(f\"{location}/1.Blast_result/{species}/{species}_coordinates_old.csv\",'r') as coordinate_file:\n",
    "        coordinate_lines = coordinate_file.readlines()\n",
    "\n",
    "\n",
    "    complement = coordinate_lines[1].split(\",\")[4]\n",
    "    if complement == \"0\":\n",
    "        start = coordinate_lines[1].split(\",\")[2]\n",
    "        stop = coordinate_lines[-1].split(\",\")[3]\n",
    "    if complement == \"1\":\n",
    "        start = coordinate_lines[-1].split(\",\")[2]\n",
    "        stop = coordinate_lines[1].split(\",\")[3]\n",
    "    scaffold = coordinate_lines[1].split(\",\")[1]\n",
    "\n",
    "    genome_folder_name = Get_genome_name(species, genome_location)\n",
    "\n",
    "    genome_species_location = f\"{genome_location}/{species}\"\n",
    "    genome_file_name = Genome_file_name(genome_species_location)\n",
    "    \n",
    "    genome_start = int(start) - 100000\n",
    "    genome_stop = int(stop) + 100000\n",
    "    \n",
    "    list_of_folders = os.listdir(f\"{location}/2.TOGA/2.Target_genomes/\")\n",
    "    if species not in list_of_folders:\n",
    "        os.mkdir(f\"{location}/2.TOGA/2.Target_genomes/{species}\")\n",
    "    \n",
    "    \n",
    "    subprocess.run(f'samtools faidx \"{genome_species_location}/{genome_file_name}\" {scaffold}:{genome_start}-{genome_stop} > \"{location}/2.TOGA/2.Target_genomes/{species}/target_genome.fa\"', shell = True)\n",
    "\n",
    "    \n",
    "#     print(species,start,stop,complement)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d81fb22",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
