{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cd3f3d9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Get_genome_name(species, genome_location):\n",
    "    import os\n",
    "    list_of_folders = os.listdir(genome_location)\n",
    "    while True:\n",
    "        if species in list_of_folders:\n",
    "            return(species)\n",
    "        else:\n",
    "            input(\"species_missing?\")\n",
    "            break\n",
    "def Genome_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            return(file)\n",
    "\n",
    "def Get_gff_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.gff\"):\n",
    "            return(file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b29b597f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_905333055.1_ilManJurt1.1_genomic.gff\n",
      "5746955 5778306\n",
      "5746955 5778306\n",
      "NC_060058.1\t100000\t100088\tExon_1\n",
      "NC_060058.1\t101874\t101957\tExon_2\n",
      "NC_060058.1\t102193\t102368\tExon_3\n",
      "NC_060058.1\t103070\t103158\tExon_4\n",
      "NC_060058.1\t103746\t103820\tExon_5\n",
      "NC_060058.1\t104023\t104154\tExon_6\n",
      "NC_060058.1\t104662\t104737\tExon_7\n",
      "NC_060058.1\t106111\t106314\tExon_8\n",
      "NC_060058.1\t107233\t107461\tExon_9\n",
      "NC_060058.1\t108934\t109045\tExon_10\n",
      "NC_060058.1\t109585\t109704\tExon_11\n",
      "NC_060058.1\t110000\t110149\tExon_12\n",
      "NC_060058.1\t111045\t111163\tExon_13\n",
      "NC_060058.1\t111368\t111472\tExon_14\n",
      "NC_060058.1\t112452\t112595\tExon_15\n",
      "NC_060058.1\t112681\t112852\tExon_16\n",
      "NC_060058.1\t113018\t113153\tExon_17\n",
      "NC_060058.1\t113508\t113612\tExon_18\n",
      "NC_060058.1\t113755\t113856\tExon_19\n",
      "NC_060058.1\t114259\t114429\tExon_20\n",
      "NC_060058.1\t115141\t115295\tExon_21\n",
      "NC_060058.1\t119170\t119313\tExon_22\n",
      "NC_060058.1\t128492\t128671\tExon_23\n",
      "NC_060058.1\t129092\t129205\tExon_24\n",
      "NC_060058.1\t130005\t130106\tExon_25\n",
      "NC_060058.1\t130657\t130794\tExon_26\n",
      "NC_060058.1\t131250\t131351\tExon_27\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5746955 5778306\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine\"\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "species = \"Maniola_jurtina\"\n",
    "transcript = \"XM_045928854.1\"\n",
    "\n",
    "with open(f\"{location}/1.Blast_result/{species}/{species}_coordinates_old.csv\",'r') as coordinate_file:\n",
    "    coordinate_lines = coordinate_file.readlines()\n",
    "\n",
    "\n",
    "complement = coordinate_lines[1].split(\",\")[4]\n",
    "if complement == \"0\":\n",
    "    start = coordinate_lines[1].split(\",\")[2]\n",
    "    stop = coordinate_lines[-1].split(\",\")[3]\n",
    "if complement == \"1\":\n",
    "    start = coordinate_lines[-1].split(\",\")[2]\n",
    "    stop = coordinate_lines[1].split(\",\")[3]\n",
    "scaffold = coordinate_lines[1].split(\",\")[1]\n",
    "\n",
    "genome_folder_name = Get_genome_name(species, genome_location)\n",
    "\n",
    "genome_species_location = f\"{genome_location}/{species}\"\n",
    "genome_file_name = Genome_file_name(genome_species_location)\n",
    "\n",
    "\n",
    "gff_file_name = Get_gff_file_name(genome_species_location)\n",
    "print(gff_file_name)\n",
    "\n",
    "with open(f\"{genome_species_location}/{gff_file_name}\", 'r') as gff_file:\n",
    "    gff_file_lines = gff_file.readlines()\n",
    "\n",
    "coordinate_switch = 0\n",
    "gene_found = 0\n",
    "exon_dictionary = {}\n",
    "exon_number = 0\n",
    "output = ''\n",
    "chromosome_start = 0\n",
    "# print(start)\n",
    "for line in gff_file_lines:\n",
    "    line_split = line.split(\"\\t\")\n",
    "    if scaffold in line_split:\n",
    "        if \"gene\" in line_split[2]:\n",
    "            if (int(line_split[4]) > int(start)) and (int(line_split[3]) < int(start)):\n",
    "                coordinate_switch = 1\n",
    "#                 print(line)\n",
    "#                 break\n",
    "    if coordinate_switch == 1:\n",
    "        if transcript in line:\n",
    "            if \"CDS\" in line_split[2]:\n",
    "                gene_found = 1\n",
    "                exon_number +=1\n",
    "\n",
    "                exon_dictionary[f\"Exon_{exon_number}\"] = line.split(\"\\t\")[0]+\"\\t\"+line.split(\"\\t\")[3]+\"\\t\"+line.split(\"\\t\")[4]\n",
    "    #             output += line.split(\"\\t\")[0]+\"\\t\"+line.split(\"\\t\")[3]+\"\\t\"+line.split(\"\\t\")[4]+\"\\n\"\n",
    "            if gene_found == 1 and \"CDS\" not in line_split[2]:\n",
    "                break\n",
    "\n",
    "if complement == \"0\":\n",
    "    chromosome_start = exon_dictionary[\"Exon_1\"].split(\"\\t\")[1]\n",
    "    chromosome_stop = exon_dictionary[f\"Exon_{exon_number}\"].split(\"\\t\")[2]\n",
    "if complement == \"1\":\n",
    "    chromosome_start = exon_dictionary[f\"Exon_{exon_number}\"].split(\"\\t\")[1]\n",
    "    chromosome_stop = exon_dictionary[\"Exon_1\"].split(\"\\t\")[2]\n",
    "print(start,stop)\n",
    "print(chromosome_start,chromosome_stop)\n",
    "\n",
    "for i in range(1,len(exon_dictionary)+1):\n",
    "    output += exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[0]+\"\\t\"+str(int(exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[1])-(int(chromosome_start))+100000)+\"\\t\"+str(int(exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[2])-(int(chromosome_start))+100000)+\"\\t\"+f\"Exon_{i}\"+\"\\n\"\n",
    "\n",
    "print(output)\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae\"\n",
    "with open(f\"{location}/2.TOGA/1.Query_genomes/{species}/gff_fragment.txt\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "    \n",
    "genome_start = int(chromosome_start) - 100000\n",
    "genome_stop = int(chromosome_stop) + 100000\n",
    "\n",
    "\n",
    "\n",
    "subprocess.run(f'samtools faidx \"{genome_species_location}/{genome_file_name}\" {scaffold}:{genome_start}-{genome_stop} > \"{location}/2.TOGA/1.Query_genomes/{species}/query_genome.fa\"', shell = True)\n",
    "\n",
    "print(start,stop)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f7f3d50b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "129418 100000\n",
      "NC_059680\t100000\t129418\tPeriod\t0\t+\t100000\t129418\t0\t27\t21,111,83,154,91,44,131,75,218,228,111,113,149,133,116,140,177,135,92,107,232,137,152,104,101,143,98,\t0,1880,2529,4373,6296,6825,7074,8929,9414,9865,10548,11329,14463,15291,16013,16678,17405,18918,19365,19573,19786,23008,23663,24567,24947,25596,29320,\n"
     ]
    }
   ],
   "source": [
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae\"\n",
    "species = \"Pieris_brassicae\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/2.TOGA/1.Query_genomes/{species}/gff_fragment.txt\", 'r') as input_file:\n",
    "    input_bed_file = input_file.readlines()\n",
    "\n",
    "chromosome_start = input_bed_file[0].split(\"\\t\")[1]\n",
    "chromosome = input_bed_file[0].split(\"\\t\")[0]\n",
    "# print(input_bed_file[-1])\n",
    "chromosome_end = input_bed_file[-1].split(\"\\t\")[2].rstrip()\n",
    "print(chromosome_end,chromosome_start)\n",
    "\n",
    "output = f\"{chromosome.split('.')[0]}\\t{chromosome_start}\\t{chromosome_end}\\tPeriod\\t0\\t+\\t{chromosome_start}\\t{chromosome_end}\\t0\\t{len(input_bed_file)}\\t\"\n",
    "\n",
    "# print(output)\n",
    "size_dictionary = {}\n",
    "sizes = ''\n",
    "pos = ''\n",
    "pos_dictionary = {}\n",
    "for i in range(0,len(input_bed_file)):\n",
    "    line_split = input_bed_file[i].split(\"\\t\")\n",
    "    sizes += f\"{int(line_split[2])-int(line_split[1])},\"\n",
    "    size_dictionary[f\"Exon_{i+1}\"] = int(line_split[2])-int(line_split[1])\n",
    "    pos_dictionary = int(line_split[1]) - 100000\n",
    "    pos += f\"{int(line_split[1]) - 100000},\"\n",
    "#     print(input_bed_file[i])\n",
    "\n",
    "# print(sizes,pos)\n",
    "    \n",
    "output += f\"{sizes}\\t{pos}\"\n",
    "print(output)\n",
    "with open(f\"{location}/2.TOGA/1.Query_genomes/{species}/gff_fragment.bed\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45b52ac4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
