{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fbb53c69",
   "metadata": {},
   "outputs": [],
   "source": [
    "family_group = \"6.Leptidea\"\n",
    "\n",
    "\n",
    "def make_species_folder(output_location, species):\n",
    "    list_of_folders = os.listdir(f\"{output_location}/4.Bed File/\")\n",
    "    if species not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/4.Bed File/{species}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "73f14383",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Leptidea_juvernica\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "CATIXQ010000001.1\t23465986\t23466072\tExon_1\n",
      "CATIXQ010000001.1\t23463649\t23463733\tExon_2\n",
      "CATIXQ010000001.1\t23461671\t23461823\tExon_3\n",
      "CATIXQ010000001.1\t23460184\t23460297\tExon_4\n",
      "CATIXQ010000001.1\t23459781\t23459838\tExon_5\n",
      "CATIXQ010000001.1\t23459204\t23459336\tExon_6\n",
      "CATIXQ010000001.1\t23456373\t23456443\tExon_7\n",
      "CATIXQ010000001.1\t23456037\t23456235\tExon_8\n",
      "CATIXQ010000001.1\t23454839\t23455068\tExon_9\n",
      "CATIXQ010000001.1\t23452644\t23452756\tExon_10\n",
      "CATIXQ010000001.1\t23451049\t23451169\tExon_11\n",
      "CATIXQ010000001.1\t23450511\t23450661\tExon_12\n",
      "CATIXQ010000001.1\t23449447\t23449566\tExon_13\n",
      "CATIXQ010000001.1\t23446239\t23446341\tExon_14\n",
      "CATIXQ010000001.1\t23442742\t23442880\tExon_15\n",
      "CATIXQ010000001.1\t23441978\t23442144\tExon_16\n",
      "CATIXQ010000001.1\t23440160\t23440302\tExon_17\n",
      "CATIXQ010000001.1\t23439715\t23439823\tExon_18\n",
      "CATIXQ010000001.1\t23437801\t23437932\tExon_19\n",
      "CATIXQ010000001.1\t23435823\t23435970\tExon_20\n",
      "CATIXQ010000001.1\t23431304\t23431466\tExon_21\n",
      "CATIXQ010000001.1\t23429029\t23429116\tExon_22\n",
      "CATIXQ010000001.1\t23428280\t23428382\tExon_23\n",
      "CATIXQ010000001.1\t23426782\t23426917\tExon_24\n",
      "CATIXQ010000001.1\t23424656\t23424761\tExon_25\n",
      "\n",
      "Leptidea_reali\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "CATIXK010000002.1\t20872199\t20872285\tExon_1\n",
      "CATIXK010000002.1\t20869128\t20869212\tExon_2\n",
      "CATIXK010000002.1\t20867172\t20867324\tExon_3\n",
      "CATIXK010000002.1\t20865668\t20865781\tExon_4\n",
      "CATIXK010000002.1\t20865264\t20865321\tExon_5\n",
      "CATIXK010000002.1\t20864687\t20864819\tExon_6\n",
      "CATIXK010000002.1\t20860990\t20861060\tExon_7\n",
      "CATIXK010000002.1\t20860654\t20860852\tExon_8\n",
      "CATIXK010000002.1\t20859522\t20859751\tExon_9\n",
      "CATIXK010000002.1\t20857303\t20857415\tExon_10\n",
      "CATIXK010000002.1\t20855719\t20855839\tExon_11\n",
      "CATIXK010000002.1\t20855164\t20855314\tExon_12\n",
      "CATIXK010000002.1\t20853999\t20854118\tExon_13\n",
      "CATIXK010000002.1\t20850912\t20851014\tExon_14\n",
      "CATIXK010000002.1\t20847444\t20847582\tExon_15\n",
      "CATIXK010000002.1\t20846678\t20846844\tExon_16\n",
      "CATIXK010000002.1\t20844733\t20844875\tExon_17\n",
      "CATIXK010000002.1\t20844288\t20844396\tExon_18\n",
      "CATIXK010000002.1\t20840120\t20840251\tExon_19\n",
      "CATIXK010000002.1\t20838239\t20838386\tExon_20\n",
      "CATIXK010000002.1\t20833695\t20833857\tExon_21\n",
      "CATIXK010000002.1\t20831478\t20831565\tExon_22\n",
      "CATIXK010000002.1\t20830744\t20830846\tExon_23\n",
      "CATIXK010000002.1\t20829595\t20829730\tExon_24\n",
      "CATIXK010000002.1\t20826883\t20826988\tExon_25\n",
      "\n",
      "Leptidea_sinapis\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "NC_066312.1\t12062394\t12062480\tExon_1\n",
      "NC_066312.1\t12065348\t12065432\tExon_2\n",
      "NC_066312.1\t12067234\t12067386\tExon_3\n",
      "NC_066312.1\t12068744\t12068857\tExon_4\n",
      "NC_066312.1\t12069218\t12069275\tExon_5\n",
      "NC_066312.1\t12069724\t12069856\tExon_6\n",
      "NC_066312.1\t12073465\t12073535\tExon_7\n",
      "NC_066312.1\t12073673\t12073871\tExon_8\n",
      "NC_066312.1\t12074642\t12074871\tExon_9\n",
      "NC_066312.1\t12076986\t12077098\tExon_10\n",
      "NC_066312.1\t12078571\t12078691\tExon_11\n",
      "NC_066312.1\t12079123\t12079273\tExon_12\n",
      "NC_066312.1\t12080318\t12080437\tExon_13\n",
      "NC_066312.1\t12083947\t12084049\tExon_14\n",
      "NC_066312.1\t12087404\t12087542\tExon_15\n",
      "NC_066312.1\t12088141\t12088307\tExon_16\n",
      "NC_066312.1\t12090232\t12090374\tExon_17\n",
      "NC_066312.1\t12090711\t12090819\tExon_18\n",
      "NC_066312.1\t12095129\t12095260\tExon_19\n",
      "NC_066312.1\t12096987\t12097134\tExon_20\n",
      "NC_066312.1\t12101422\t12101584\tExon_21\n",
      "NC_066312.1\t12103768\t12103855\tExon_22\n",
      "NC_066312.1\t12105113\t12105215\tExon_23\n",
      "NC_066312.1\t12106588\t12106723\tExon_24\n",
      "NC_066312.1\t12109248\t12109353\tExon_25\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#for variant 1\n",
    "import os\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "list_of_folders = os.listdir(f\"{output_location}\")\n",
    "if \"4.Bed File\" not in list_of_folders:\n",
    "    os.mkdir(f\"{output_location}/4.Bed File\")\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    make_species_folder(output_location, species)\n",
    "    output = ''\n",
    "    variant_presence = 1\n",
    "    \n",
    "    list_of_files_in_species_folder = os.listdir(f\"{output_location}/1.Blast_result/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        if file_names.endswith(\"_coordinates.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "\n",
    "    for i in range(1,len(coordinate_file_lines)):\n",
    "        print(i)\n",
    "        split_coordinate_line = (coordinate_file_lines[i]).split(\",\")\n",
    "        scaffold = split_coordinate_line[1]\n",
    "        complement = split_coordinate_line[4]\n",
    "        if complement == \"0\":\n",
    "            chr_start =int(split_coordinate_line[2])\n",
    "            chr_end = int(split_coordinate_line[3])+1\n",
    "        elif complement == \"1\":\n",
    "            chr_start =int(split_coordinate_line[2])\n",
    "            chr_end = int(split_coordinate_line[3])+1\n",
    "        exon_number = split_coordinate_line[6].split(\"_\")[-1] \n",
    "        if chr_start > 5 and chr_end > 5:\n",
    "            output +=f\"{scaffold}\\t{chr_start}\\t{chr_end}\\tExon_{exon_number}\\n\" \n",
    "    with open(f\"{output_location}/4.Bed File/{species}/Transcript.bed\", 'w') as out_file:\n",
    "        out_file.write(output)\n",
    "    print(output)\n",
    "    \n",
    "#     assert False\n",
    "    \n",
    "#     for exon_number in range(1,28):\n",
    "        \n",
    "        \n",
    "        \n",
    "#         with open(f\"{output_location}/2.Final_output/{species}/{exon}\", 'r') as exon_file:\n",
    "#             exon_name_split = exon_file.readlines()[0].split(\"_\")\n",
    "#             if len(exon_name_split[4]) < 4:\n",
    "#                 scaffold = exon_name_split[4]+\"_\"+ exon_name_split[5]\n",
    "\n",
    "#             else:\n",
    "#                 scaffold = exon_name_split[4]\n",
    "#             chr_start = exon_name_split[-6]\n",
    "#             chr_end = exon_name_split[-5]\n",
    "# #                 print(scaffold, chr_start,chr_end)\n",
    "#             output +=f\"{scaffold}\\t{chr_start}\\t{chr_end}\\tExon_{exon_number}\\n\" \n",
    "    \n",
    "#     if variant_presence == 1:\n",
    "#         with open(f\"{output_location}/4.Bed File/{species}/Transcript_vairant_{Transcript_variant}.bed\", 'w') as out_file:\n",
    "#             out_file.write(output)\n",
    "#     print(output)\n",
    "# #     assert False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "572f54f4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
