{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"6.Leptidea\"\n",
    "number_of_exons = 25\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern MDTLDDSEKNAKISDSAYSNSCTDSQAR\n",
      "Leptidea_reali_ MDTLDDSEKNAKISDSAYSNSCTDSQAR\n",
      "Leptidea_sinapi MDTLDDSEKNAKISDSAYSNSCTDSQAR\n",
      "                ****************************\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "Leptidea_reali_ NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "Leptidea_sinapi NSSHSTHCGSHSSGSSGYGGEPTKSGY\n",
      "                ***************************\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "Leptidea_reali_ TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "Leptidea_sinapi TNNLSQPPEKKRKINKKKTAECEKLKSKKKDPSPPKFEIEVTLKEQEANG\n",
      "                **************************************************\n",
      "\n",
      "continue?y\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern IEVSKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "Leptidea_sinapi IEASKTQMSEFGSNVENVDSSTEPEPPNCANAINDNK\n",
      "Leptidea_reali_ IEASKTQMSEFGSNVENVDSSTEPEPPNCANATNDNK\n",
      "                **.***************************** ****\n",
      "\n",
      "continue?y\n",
      "Exon_5.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern KSSSNIPSSTSMDSIPFNA\n",
      "Leptidea_reali_ KSSSNIPSSTSMDSIPFNA\n",
      "Leptidea_sinapi KSSSNIPSSPSMDSIPFNA\n",
      "                *********.*********\n",
      "\n",
      "continue?y\n",
      "Exon_6.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "Leptidea_reali_ ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "Leptidea_sinapi ECFSCVMSMQDGVVMHTTPGITSTLGYPKDFWVGRPFIEFLHQW\n",
      "                ********************************************\n",
      "\n",
      "continue?y\n",
      "Exon_7.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern DRKTFESELNDLVAENVNSTKKK\n",
      "Leptidea_reali_ DRKTFESELNDLVAENVNSTKKK\n",
      "Leptidea_sinapi DRKTFESELNDLVAENVNSTKKK\n",
      "                ***********************\n",
      "\n",
      "continue?y\n",
      "Exon_8.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern NGDMTTTAVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPF\n",
      "Leptidea_reali_ NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPF\n",
      "Leptidea_sinapi NGDMTTTVVCRLRCYKALSAGFEIRETIATFTPFLLKLSFRKICDEEGEVWYLLVQATPF\n",
      "                *******.****************************************************\n",
      "\n",
      "Leptidea_juvern ISAFK\n",
      "Leptidea_reali_ ISAFK\n",
      "Leptidea_sinapi ISAFK\n",
      "                *****\n",
      "\n",
      "continue?y\n",
      "Exon_9.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVY\n",
      "Leptidea_reali_ PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVY\n",
      "Leptidea_sinapi PNEVVTKAVAFVLRHSSTGEIKYIDTESVPYLGYSPQDIVNKDAIAMYHPYDLEYLLHVY\n",
      "                ************************************************************\n",
      "\n",
      "Leptidea_juvern ETIMKVGSAPRSSPY\n",
      "Leptidea_reali_ ETIMKVGSAPRSSPY\n",
      "Leptidea_sinapi ETIMKVGSAPRSSPY\n",
      "                ***************\n",
      "\n",
      "continue?y\n",
      "Exon_10.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "Leptidea_reali_ MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "Leptidea_sinapi MMTQNGDYVKLETEWSSFFNPWSRKLEFIIGKHLILE\n",
      "                *************************************\n",
      "\n",
      "continue?y\n",
      "Exon_11.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "Leptidea_reali_ GPTNPDVFQSPNPEVQNKFPEDDDNKAKSLRENIIKILNE\n",
      "Leptidea_sinapi GPTNPDVFQSPNPEVQNQFPEDDDNKAKSLRENIIKILNE\n",
      "                *****************:**********************\n",
      "\n",
      "continue?y\n",
      "Exon_12.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern PSTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "Leptidea_reali_ PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "Leptidea_sinapi PTTKSEDLAKQEIAIRCQKVASIIESLNEKAPNVDKEIWIDIQEPENVNY\n",
      "                *:************************************************\n",
      "\n",
      "continue?y\n",
      "Exon_13.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDQNLD\n",
      "Leptidea_reali_ DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "Leptidea_sinapi DRDSVLLGSISPHTDSIVSKASTETQVNYNQLNTDENLD\n",
      "                ***********************************:***\n",
      "\n",
      "continue?y\n",
      "Exon_14.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "Leptidea_reali_ YFNRWLEHCVEEEIAEAEPNNGEANLENSNETN\n",
      "Leptidea_sinapi YFNRWLEHCVEEEIAEAEPNNGEANLENSNEAN\n",
      "                *******************************:*\n",
      "\n",
      "continue?y\n",
      "Exon_15.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "Leptidea_reali_ ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "Leptidea_sinapi ELASVEEQSEASKMADFFKSSVTLIPNNHGHADYKPTRLSENILE\n",
      "                *********************************************\n",
      "\n",
      "continue?y\n",
      "Exon_16.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "Leptidea_reali_ HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "Leptidea_sinapi HDSIMEMDLVKKHREIPKENRKKYSKEVRLKQKEHLARCNATFTPATAGLLADLQ\n",
      "                *******************************************************\n",
      "\n",
      "continue?y\n",
      "Exon_17.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "Leptidea_reali_ SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "Leptidea_sinapi SNGLKRTSKFCEESDGVNYLYSIAKQPRRDGTGTSSPRLYVNLPNSD\n",
      "                ***********************************************\n",
      "\n",
      "continue?y\n",
      "Exon_18.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern TTSVPPNKENSANQFSFEVGFPQQMSPLGPSVATQ\n",
      "Leptidea_reali_ TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "Leptidea_sinapi TTSVPPNKENSANQFSFEVGFPQQMSPLGPNVATQ\n",
      "                ******************************.****\n",
      "\n",
      "continue?y\n",
      "Exon_19.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "Leptidea_reali_ PFMSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "Leptidea_sinapi PFLSPGPYFNGTHAHLYEEMQQEKNHPSTSKQPLPLPTDNYQE\n",
      "                **:****************************************\n",
      "\n",
      "continue?y\n",
      "Exon_20.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern AYKLIIPPKTSKSIPSSEWRKKRTKLQARANNINESSGDSISSANVSYL\n",
      "Leptidea_reali_ ACKLIIPPKTSKSISSSEWRKKRTKLQARANNINESSGDSISSANVSNL\n",
      "Leptidea_sinapi ACKLIIPPKTSKSISSSEWRKKRTKLQTRANNINESSGDSISSANVSNL\n",
      "                * ************.************:******************* *\n",
      "\n",
      "continue?y\n",
      "Exon_21.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern STNAVTLSQINEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "Leptidea_reali_ STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "Leptidea_sinapi STNAVTLSQNNEKDPIANSGTNADKTDDESWFSSFSSHIYKSETGSPEESEMKK\n",
      "                ********* ********************************************\n",
      "\n",
      "continue?y\n",
      "Exon_22.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "Leptidea_reali_ KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "Leptidea_sinapi KSTISSLQTNRQTINKVAMKKIDPPWLDD\n",
      "                *****************************\n",
      "\n",
      "continue?y\n",
      "Exon_23.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "Leptidea_reali_ VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "Leptidea_sinapi VCLTPEIIYKYQVKTRSLAEVLAEDKKKLEALEQ\n",
      "                **********************************\n",
      "\n",
      "continue?y\n",
      "Exon_24.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern PSLVNEQLSQLYYELKLEGHPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "Leptidea_reali_ PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "Leptidea_sinapi PSLVNEQLSQLYYELKLEGLPEKLTLEEGITSSSSSGDDTTSNTE\n",
      "                ******************* *************************\n",
      "\n",
      "continue?y\n",
      "Exon_25.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Leptidea_juvern DRRRKREHGKLVIIFEEDAPLPSPTEHVRNGDDN\n",
      "Leptidea_reali_ DRRRKREHGKLVIIYEEDAPLPSPTEHVRNEDDN\n",
      "Leptidea_sinapi DRRRKREHGKLVIIYEEDAPLPSPTEHVRNEDDN\n",
      "                **************:*************** ***\n",
      "\n",
      "continue?y\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
