{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "25dab8af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "02a9a36f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2fe40ae3",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/1.Blast_output\"\n",
    "annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\"\n",
    "reblast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast\"\n",
    "query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query\"\n",
    "\n",
    "species = \"Pararge_aegeria\"\n",
    "coordinate_file = f\"{species}_coordinates.csv\"\n",
    "good_exon_upstream = 6\n",
    "good_exon_downstream = 8\n",
    "error_exon = \"Exon_7\"\n",
    "\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "for file in list_of_files_in_genome_folder:\n",
    "    if file.endswith(\"_genomic.fna\"):\n",
    "        genome_file = file\n",
    "# print(genome_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8488c390",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR!!!\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "\n",
      "Pararge_aegeria,NC_053208.1,16924607,16924738,1,Y,5.Bicyclus_anynana_XM_024088150.2_query_Exon_6,1,44,44\n",
      "\n",
      "Pararge_aegeria,NC_053208.1,16923103,16923306,1,N,5.Bicyclus_anynana_XM_024088150.2_query_Exon_8,2,67,67\n",
      "\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Continue?? y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16923306 16924607\n",
      "['Pararge_aegeria', 'NC_053208.1', '16924607', '16924738', '1', 'Y', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_6', '1', '44', '44\\n']\n",
      "['Pararge_aegeria', 'NC_053208.1', '16923103', '16923306', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_8', '2', '67', '67\\n']\n"
     ]
    }
   ],
   "source": [
    "with open(f\"{blast_output_location}/{species}/{coordinate_file}\", 'r') as coordinate_open_file:\n",
    "    coordinate_file_lines = coordinate_open_file.readlines()\n",
    "    \n",
    "upstream_exon_line, downstream_exon_line = coordinate_file_lines[good_exon_upstream].split(\",\"), coordinate_file_lines[good_exon_downstream].split(\",\")\n",
    "if (\"Y\" in upstream_exon_line[5]) or (\"Y\" in downstream_exon_line[5]):\n",
    "    print(\"ERROR!!!\")\n",
    "    print(f\"{coordinate_file_lines[0]}\\n{coordinate_file_lines[good_exon_upstream]}\\n{coordinate_file_lines[good_exon_downstream]}\")\n",
    "    continue_param = input(\"Continue??\")\n",
    "    if continue_param.lower()[0] == \"n\":\n",
    "        assert False\n",
    "          \n",
    "\n",
    "complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "if complement == \"0\":\n",
    "    gene_start = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    gene_end = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "\n",
    "if complement == \"1\":\n",
    "    gene_start = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    gene_end = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "\n",
    "print(gene_start, gene_end)\n",
    "\n",
    "\n",
    "print(upstream_exon_line)\n",
    "print(downstream_exon_line)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b431689d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_053208.1\n"
     ]
    }
   ],
   "source": [
    "genome = SeqIO.parse(f\"{genome_location}/{species}/{genome_file}\", \"fasta\")\n",
    "for entries in genome:\n",
    "    # print(scaffold)\n",
    "    if scaffold == entries.id:\n",
    "        print(scaffold)\n",
    "        gene_sequence = entries.seq[gene_start-1:gene_end]\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "22e13e69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29\n",
      "GGTAAggacattataatttttaatgcatTCCCTCGAAACTTGAAATCTTCTAATCCACTAgagtttctataaaaaaaaatatggcataTTATGCCGTCATGCCAATCCATTATGGCggatatccatactaatattatgctTGTGAAATTGTttctgtttgtttctttgtttgttacctatgaaCAGCTAAACCGTTgaatcgattttcatgaaatttgacACACACGTAGTTTGCATCTTTAGTACAGTGAtttcattataacatttatattggcTGTAATATAAAAAGCGTTAACCATCTGATTTGTGAGATGGTGGAGCTAAGGCTAAAAATTCAGTATGTAACtattttaagttaaatgtaTAGCATCATTTGCATCTGAAGCCGTTTCATCATATAGATTGATTagcgttttttattatatttggtcGGAGTAATACTAATAATCTATTAGCGTGAGCAACATTAATTAATCTGTTTGacagattttattaaaatttactttcatGATATTGAACTATAGAAAGAGTAAAGGTGAAATAGAGAAGTTAGTAATACAAAGTGAATTCACAAAAccgttttattttacaaaaaattgaatgaaattataatttattttcatatttacagGATCGCAAAACCTTTGCATCCCAAATCACTAGCGGGCTTTTAGCTCCGAAATCAACTAGCGGTGAACAACCTCAAGGTAATTTCGATTGATTTAATGATCCATGCTTCCAAtagatttaatttgaaaatttgtcgTTTGTATTAGCTTCAATACATCACATTCTCAAACATGACTTCATACAACCAAACTTATCCgctacaattttataattgagATAGTGAAAAAGTTAATAGCTTCAAGATACAAATTTTACAGGGTTCTTTCATACAAACGTGTAATTAATTCTAAATGTTAtaactaagtataaaaataatgtttcctTTAAAGTGAACTGTCTTCTTGCTGATTCTAGTACTAGTATCaataatttcaattaataagTACCTCAATAAGATAAACCGCCATATTTTATTGCCGGCTGTTTCATgtaaaattatcttatttaaaaacatttaaaagacAAATTTACAAAATGTATTAAGCTTCTGTGAGAATTATGTGCACTATGTTTCATACATAGCAAAAGTTGTATTTCATTATTTGagatgaaattgaaaatttaatgttaataatttttgttttataatgcaTTTAAAGTAATCATCAACACATAACCTAACTTTTGGGATAGTTAAAATTTGTGTCGCAACTTCAATAAGAGCTTGTCCAC\n"
     ]
    }
   ],
   "source": [
    "query_species = \"5.Bicyclus_anynana\"\n",
    "query_transcript = \"XM_024088150.2\"\n",
    "query_exon = error_exon\n",
    "frame_offset = 1\n",
    "\n",
    "with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "    \n",
    "    query_fasta_sequence = \"\".join(query_file.readlines())\n",
    "\n",
    "list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "annotated_species_name = ''\n",
    "for annotated_species in list_of_annotated_genomes:\n",
    "    if annotated_species.endswith(species):\n",
    "        annotated_species_name = (annotated_species)\n",
    "        break\n",
    "if annotated_species_name == '':\n",
    "    print(f\"Error with annotated species name\")\n",
    "    assert False\n",
    "    \n",
    "list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "print(query_length)\n",
    "print((gene_sequence))\n",
    "sequence_set = ''\n",
    "\n",
    "list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\")\n",
    "if error_exon not in list_of_exon_directories:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\")\n",
    "\n",
    "# for offset in range(3):\n",
    "# #     print()\n",
    "#     translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "#     for i in range(len(translated_sequence)):\n",
    "#         if len(translated_sequence[i])> 0.8*query_length:\n",
    "#             sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "#             with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "\n",
    "#                 output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "#                 out_file.write(output)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "d442f4a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sequence_set = ''\n",
    "# for i in range(len(translated_sequence)):\n",
    "#     if len(translated_sequence[i])> 0.8*query_length:\n",
    "#         sequence_set += f\">set{i+1}\\n{translated_sequence[i]}\\n\\n\"\n",
    "# with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/{error_exon}_translated_genomic_sequence.fa\",'w') as out_file:\n",
    "\n",
    "#     output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "#     out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c73cd22d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "conda 22.9.0\n"
     ]
    }
   ],
   "source": [
    "!conda --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "41c937bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n"
     ]
    }
   ],
   "source": [
    "!wsl.exe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0637c278",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'pwd' is not recognized as an internal or external command,\n",
      "operable program or batch file.\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58086ac0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
