{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3247b82e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0653b64e",
   "metadata": {},
   "outputs": [],
   "source": [
    "scaffold = \"\"\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "086c0ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "#     print(list_of_files_in_genome_folder)\n",
    "    if \"Temp\" in list_of_files_in_genome_folder:\n",
    "        subprocess.run(f'rm -r \"{genome_location}/{species}/Temp\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{genome_location}/{species}/Temp\")\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9d4c6ece",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_query_folder(annotated_species_location,gene_name,species):\n",
    "    import time\n",
    "\n",
    "\n",
    "    list_of_files = os.listdir(annotated_species_location)\n",
    "    if species in list_of_files:\n",
    "        subprocess.run(f'rm -r \"{annotated_species_location}/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    time.sleep(2.5)\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts/{gene_name}\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7cb40a07",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_plexippus XM_032655043.2\n",
      "+\n",
      "{'Exon_1': ['1316575', '1316627'], 'Exon_2': ['1327960', '1328086'], 'Exon_3': ['1328742', '1328925'], 'Exon_4': ['1329441', '1329587'], 'Exon_5': ['1329761', '1329931'], 'Exon_6': ['1330135', '1330316'], 'Exon_7': ['1330629', '1330763'], 'Exon_8': ['1330922', '1331012'], 'Exon_9': ['1331508', '1331689']}\n",
      "exon counter 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Danaus_plexippus_XM_032655043.2_Exon1\n",
      "atggaaACACGTCATTATTGGGAAGAGAATGGACATGCCGTCAAATATGACAA\n",
      ">Danaus_plexippus_XM_032655043.2_Exon2\n",
      "CTACTCCAATGAGGAGTTCGCGCGCGAGCCGCTGAGCTTCGCGCCGCCGTCCGAGGACGAGGCGGAGTACCCTCCAGGGTACAAGAAGGGGAAGGTCTCGAGGGCAAGTCACACTGAATATGCCcgt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Danaus_plexippus_XM_032655043.2_Exon3\n",
      "CAAGATCCAATGTCGCACCGCATCATCGAGAAGCGGCGGCGGGACCGGATGAACAACTGCCTGGCGGACCTGTCCCGCCTCATACCACCCGAGTACCTCAAGAAGGGCCGCGGCCGCGTGGAGAAGACGGAGATCATAGAGATGGCCATACGACACCTCAAGTATCTACAGGACAGAGTCCACG\n",
      ">Danaus_plexippus_XM_032655043.2_Exon4\n",
      "TTCTGGAGCGGGGGTCGGAGTTCCTCGCGGGGTACCAGAGGGCGGGGGCGGAGGCGGTGCGGTTCGTGGAGCTCCAGGGCTCCCGTGACGGCCTGGCGGATCAGCTCGCTGCACACCTACACTCACACGCTGACATGATGGCCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Danaus_plexippus_XM_032655043.2_Exon5\n",
      "AAGCTGTACACGAAAAGCGTGTGTATCCGAACTCTTCGTCGGAGACGACCAGCTCGTCGAGCAGTTCCCAGGGCTTCGCCGTGAAGGTGATCCAGCGGCCGGAGCCCCCGCCCGCGTTCCCGGAGCCCTACGAGCCTGACAGACAGGAACATTTCGCGGACTGCGAGAGAT\n",
      ">Danaus_plexippus_XM_032655043.2_Exon6\n",
      "TGTCGGTGCACCAACCAGCTGTAATGGAGCCGTTGGAAGGCGAGCCTCTCCCGCTGGACGGACGAGTGAAGAAGGAAGTGACGCTGAGGAAGATTAGGAAGCCAGAACACGAGGACTACTTGCACTCGTACAAGTTCAAGAACTCCATAGAGAGGAGGTTCTCCAGGTCGCAGGACTCCGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Danaus_plexippus_XM_032655043.2_Exon7\n",
      "GGTGACGCGTGGAGCGCCGGGCCGACAGCGAAGGCCTACAATCATAAACGTCGGAGGCCTACCAAGCCCGCGCCGCCCTCCACGTCCACTTCCGCCTCGGGCTCCACCGAGGAAGCGCGCGACACCAGCCCACAA\n",
      ">Danaus_plexippus_XM_032655043.2_Exon8\n",
      "GACACGTGCAGCGACTCCCCCCACCACCACTCGTTCGACAAGCCGCCGCCGCCCGCGCAGTACGTGCCCGTGTTCGCCTTGAACGCACTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Danaus_plexippus_XM_032655043.2_Exon9\n",
      "GCAAGTACTACGTGCCGCTGAGCGTGGAGTACGGCTGCGTGTCTCGTCAGCTGGGCGCGGGCGTGACGTCACTGGAGGCGGCCGAGGCGCGCGCGCTTCACCCCGTCACCATACATGTGAACTTCCAACCCTGCATCGACTACCTCAAGCGGGAGCCCGACCCTCACTGGCGCCCGCTCTAA\n",
      "53\t127\t184\t147\t171\t182\t135\t91\t182\t\n",
      "Aricia_agestis XM_042120213.1\n",
      "-\n",
      "{'Exon_1': ['14047331', '14047383'], 'Exon_2': ['14022090', '14022204'], 'Exon_3': ['14021443', '14021626'], 'Exon_4': ['14020245', '14020406'], 'Exon_5': ['14019541', '14019732'], 'Exon_6': ['14018907', '14019076'], 'Exon_7': ['14017977', '14018105'], 'Exon_8': ['14016571', '14016664'], 'Exon_9': ['14015998', '14016158']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Aricia_agestis_XM_042120213.1_Exon9\n",
      "GTAAATACTACGTGCCGCTATGTGTGGAGCGCGCGTGTTTATCGCGGAACGGCGCGTGGGAGGGCACGGGCGCGGCGCCCCTTCACCCCGTCACCATCCACGTCAACTTCACACCGTGCGGCGCTTGCGTCAAGCGCGAGCCGGATTGGCGCCCCGGATAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Aricia_agestis_XM_042120213.1_Exon8\n",
      "GACACGTCGAGCGAGTCGCCGCACCACCACCAGTATGAGAAGCCGCTGCCTACGGCGCAGCAGTTGGTACCAGTTTTCGCGTTGAATGCGATAG\n",
      ">Aricia_agestis_XM_042120213.1_Exon7\n",
      "GCAGCTGAGAATGGTACGCGTGGTCCTCGCGCGTTCACGAAGCGGCGGCGTGCGAATAAACCCGTACCGGCGTCGACGTCGGCATCCACTTCCGGTTCCACGGAGGACGCTAGAGACACTTCGCCACAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Aricia_agestis_XM_042120213.1_Exon6\n",
      "TGCCAGTCCCGCTGTTGCCAGTGGACTTGCCGGCGGACGAGCCCCTACCTCTCGACAGGCATAAGCGCGACTTACGCAAAGTGCGCAAGCCCGACCACGATGACTACTTACACTCCTACAAGTTTAAGAATTCCATAGAGCGACGGTTTTCAAGGTCGCAAGATTCGGAG\n",
      ">Aricia_agestis_XM_042120213.1_Exon5\n",
      "CAAACGGAAGCGTTCCCCGGGAGAAGATCAGGTCCCACATCGGCGGTTCCTCGTCGGAGACGGCGAGCTCGTCCAGCAGCTCGCACGGCTACGTGATGAAGGGCGTGATCCAGCGGCCGGAGCCCCCGCCCTTCTCGGAGCCCTACGAGCCTGAACACCCCGCGGACATGTACCAGATGGAGTGTGATAGAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Aricia_agestis_XM_042120213.1_Exon4\n",
      "CTGCTGAGGGCCAGGCGCCGGAGCCCTATAGGGTGGGGTATCAGGAGGCGGTCGGCGAGGCGGTGAGATTCCTGGTGGAAGTTCAGGGGTTCGGCCCCGCTGACGGCTTGTGCGTGCAGATGGCGGCGCATCTACAACGACACTGTGAGATTGTTACTAAAG\n",
      ">Aricia_agestis_XM_042120213.1_Exon3\n",
      "CAAGACCCGATGTCGCATCGGATTATCGAGAAGCGGCGGCGAGATCGGATGAACAACTGTCTGGCGGATCTGTCCCGTCTCATACCGCCAGAGTACTTGAAGAAAGGCAGGGGAAGGGTTGAGAAGACTGAAATTATAGAAATGGCTATTCGCCATATAAAGTTCTTACAGGATAGAGTTAATG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Aricia_agestis_XM_042120213.1_Exon2\n",
      "ctACAACTCAAACGATGGTTTCGGGCGACCAGGGGAGGTGCACTACGCGCCGCCATCAGAGGACGAGGCGGAGTACCCGCAGGGCGCCTACCTCAAGAAGGGCAAGGTCTCGAGG\n",
      ">Aricia_agestis_XM_042120213.1_Exon1\n",
      "ATGGAAACGCATCATTACTGGGAGGAGAACGGGCACACCGTTAAATACGACAA\n",
      "161\t94\t129\t170\t192\t162\t184\t115\t53\t\n",
      "Bicyclus_anynana XM_024084267.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-\n",
      "{'Exon_1': ['16174532', '16174587'], 'Exon_2': ['16140561', '16140675'], 'Exon_3': ['16139077', '16139260'], 'Exon_4': ['16134878', '16135039'], 'Exon_5': ['16134194', '16134388'], 'Exon_6': ['16133453', '16133622'], 'Exon_7': ['16132312', '16132458'], 'Exon_8': ['16131805', '16131895'], 'Exon_9': ['16131508', '16131686']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bicyclus_anynana_XM_024084267.2_Exon9\n",
      "GAAAATACTATGTGCCATTAAGCGTAGACTACACCTGCATAGCACGGCAACTCGGCACCATGGACTCGGACACGCGGCCCATCACGGCGTTACATCCCGTCACCATACACGTCAACTTCCCCTGCCTCAACTACCACGTCAAGCGCGAGCCGGGCGAGCGGCAGTGGCGGCCTATGTAG\n",
      ">Bicyclus_anynana_XM_024084267.2_Exon8\n",
      "GACACGTCCAGCGAGTCTCCGCACAACCACCATTATGACAAGCCTCAGCCACCCACTCAGTACGTGCCAGTCTTCGCTCTCAATTCATTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bicyclus_anynana_XM_024084267.2_Exon7\n",
      "ATGACTGACGCAGCGGCGCACAACGGCACGAACGGCGCACCTCACGGCAAGATGCTGAAGCGGCGCCGCGCGCTCAAGCCTGCACCGCCCTCTGCATCCACGTCCGCCTCCACCTCCACCGAGGACGCGCCGGACACCTCGCCACAG\n",
      ">Bicyclus_anynana_XM_024084267.2_Exon6\n",
      "TACAAAACCAAGTGCCGGCCGAGGGCGCCACCGAGCCGCTGCAACTAGACGGCCGCGCGCGCGACGCGCTGCGGAAGGTCCGCAAGCTCGACCACAACGACGACTACCTGCACTCCTACAAGTTCAAGAACTCTATCGAGAGGCGGTTCAGCAGATCGCAGGACACAGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bicyclus_anynana_XM_024084267.2_Exon5\n",
      "GTGATGTGATGCTGCGAGAGAAGACGCGAACACATCCCGGGACGTCATCTTCAGAGACGACGAGTTCTTCGAGCAGTTCCCATGGATACGCCGTAAAGGTGATCCAGAGGCCGGAGTTGCCGCCCGCCTTCGTACCTGAACCTTATGAGCAGGATCCACCCTCTGGACAGCATTACCCCATGGAATGTGAAAGAG\n",
      ">Bicyclus_anynana_XM_024084267.2_Exon4\n",
      "CGGCGGAGCGGCAGGCCTCCGACGAGTTCCGCGGAGGCTACCAGGAGGCCGTGGCTGAGGCCGTCCGCTTCCTCGTCGATGTGCAGGGCTTCTCCCCCGCCGACGGCCTCTGCGTGCAGCTCGCCACGCATCTGCAGCGGCACTGCGATGGCGTCACCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bicyclus_anynana_XM_024084267.2_Exon3\n",
      "CAAGACCCTATGTCCCACCGGATCATAGAGAAGCGTCGTCGAGACCGCATGAACAACTGCCTGGCCGACCTCTCGCGCCTCATACCACCGGAGTACCTCAAGAAAGGCCGAGGCCGGGTGGAGAAGACAGAGATCATTGAAATGGCAATCAGACATCTCAAGTTCTTGCAGGACCGTGTTAATG\n",
      ">Bicyclus_anynana_XM_024084267.2_Exon2\n",
      "TTACAACTCCAACGATGGCTTCGGTCGCGGCGAGGCGCACGGCTACGCGCCGCCGTCGGAGGACGAGGCGGAGTACCCGCCCGGCGCGTACCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bicyclus_anynana_XM_024084267.2_Exon1\n",
      "ATGATGGAAACTCGTAACTATTGGGAGGAAAATGTCCATCCCGTTAAATATGACAA\n",
      "179\t91\t147\t170\t195\t162\t184\t115\t56\t\n",
      "Pararge_aegeria XM_039892779.1\n",
      "-\n",
      "{'Exon_1': ['17062881', '17062933'], 'Exon_2': ['17033383', '17033503'], 'Exon_3': ['17031469', '17031652'], 'Exon_4': ['17027555', '17027716'], 'Exon_5': ['17027044', '17027238'], 'Exon_6': ['17026414', '17026583'], 'Exon_7': ['17025773', '17025928'], 'Exon_8': ['17025014', '17025104'], 'Exon_9': ['17024312', '17024490']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Pararge_aegeria_XM_039892779.1_Exon9\n",
      "GAAAATACTACGTGCCACTAAGCGTAGAGTACAGTTGCATAGCACGGCAGCTGGGCGCTATGGACGCAGACTCACGCCCTGTCACAGCGTTGCATCCTGTCACCATACACGTCAACTTTCCCTGCCTCAACTACCATGTGAAACGGGAGCCCGGTGAACGGCAATGGCGACCTATGTAA\n",
      ">Pararge_aegeria_XM_039892779.1_Exon8\n",
      "GATACATCAAGTGAATCGCCTCATCACCACCAGTACGACAAGCCTCAGCTACCGACGCAATATGTGCCAGTCTTCGCTCTCAATTCATTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Pararge_aegeria_XM_039892779.1_Exon7\n",
      "GTGAAGGATTCTACACCGCCCAACGGAGTTAATGGAGCGAGTCACGGCAAGCTCTTGAAGAGGCGTCGAGCCATCAAGCCAGCTCCTCCCGCATCCACATCTGCATCCACTTCTGCATCCACCTCCACTGAAGATGCCAGGGATACCTCGCCACAG\n",
      ">Pararge_aegeria_XM_039892779.1_Exon6\n",
      "tacaaAATCCACCAGAAGCGGAGAGTGCAGAGCCCTTGCAGCTGGACGGCCGCTTCaagcgcgaggcgctgcgcaaAGTACGCAAACTGGACCACTCTGAACACTACCTCCATTCCTACAAGTTCAAGAACTCTATAGAGAGGCGATTTAGCAGATCGCTGGATACCGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Pararge_aegeria_XM_039892779.1_Exon5\n",
      "GTGACGCGATGCTACGTGAGAAGACACGGTCAAATCCCGGGACGTCATCTTCAGAGACAACAAGTTCGTCGAGCAGTTCTCAAGGATTCGCAGTCAAGGTCATCCAGAGGCCTGAAGTACCGCCGGCGTTTCAACCGGAGCCTTATGAGCAGGAACTCTCAACTGGGCCCTCATATCCTACGGAATGTGAAAGGg\n",
      ">Pararge_aegeria_XM_039892779.1_Exon4\n",
      "CGGCCGAGCGACAGGCTGGCGAGGAATTCCGCTCCGGATACCAAGAAGCCGTAGCAGAGGCTGTTCGCTTCCTCGTCGAGGTTCAGGGCTTCTCACCGAGCGACGGACTATGCGTGCAGCTCGCTTCGCATCTGCAGCGACACTGCGAAGTCGTTACCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Pararge_aegeria_XM_039892779.1_Exon3\n",
      "CAAGATCCTATGTCTCATCGGATCATAGAGAAGCGACGTCGCGATCGCATGAACAACTGCCTGGCGGACCTGTCCCGCCTCATCCCGCCGGATTACCTCAAGAAGGGCCGCGGCCGCGTTGAGAAGACTGAGATCATAGAAATGGCGATCAGACACCTCAAGTTCTTACAGGACCGCGTCAATG\n",
      ">Pararge_aegeria_XM_039892779.1_Exon2\n",
      "CTACAACTCTAACGACGAGTTTGGGAGGGGAGGCGAGCAGTTGAACTTCGCCGCGCCGCCGTCCGATGACGAGGCGGACTACCCACCCGGCGCCTATCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Pararge_aegeria_XM_039892779.1_Exon1\n",
      "ATGGAAACACGCAATTATTGGGAGGATAATTCACATCCCGTGAAATATAACAA\n",
      "179\t91\t156\t170\t195\t162\t184\t121\t53\t\n",
      "Maniola_hyperantus XM_034968762.1\n",
      "-\n",
      "{'Exon_1': ['14470553', '14470605'], 'Exon_2': ['14444731', '14444848'], 'Exon_3': ['14443730', '14443913'], 'Exon_4': ['14441662', '14441823'], 'Exon_5': ['14441211', '14441411'], 'Exon_6': ['14440731', '14440906'], 'Exon_7': ['14439981', '14440121'], 'Exon_8': ['14439753', '14439843'], 'Exon_9': ['14439350', '14439528']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_hyperantus_XM_034968762.1_Exon9\n",
      "GGAAATACTACGTGCCACTAAGCGTAGAGTACGCTTGCATAGCGCGTCAGCTTGGTGCCATGGACTCGGAGACGCGCCCCATCACTGCGTTACATCCAGTCACTATACACGTCAACTTTCCCTGCCTCAACTACCACGTGAAGCGCGAGCCAGTTGAACGGCAGTGGCGACCTATGTAA\n",
      ">Maniola_hyperantus_XM_034968762.1_Exon8\n",
      "GACACGTCAAGCGAATCCCCTCATAACCACCATTACGACAAGCCTCCACCACCGACGCAATATGTGCCAGTCTTCGCTCTCAATTCATTAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_hyperantus_XM_034968762.1_Exon7\n",
      "GCGGCGGACTGCCCTCCACACAGTGGGGTTAACGGTGCGTCTCATGGCAAGCCATATGCACACAAGCGACGGCGTGCCGCCAAGCCTGCGTCATCTGTTCCTACCTCCGCATCCACGTCTACTGAGGATACCTCGCCACAG\n",
      ">Maniola_hyperantus_XM_034968762.1_Exon6\n",
      "TACAAAATCTAGCGGCACCAGAAGCCGAAAGCCAGGAGCCCTTACCTTTGGATGGTCGGATCAAACGCGAGGCGTTGCGGAAAGTGCGCAAAATTGAACACTCGGACGATTATCTCCATTCCTACAAGTTTAAGAACTCTATAGAGAGGCGATTTAGCAGATCCCAGGATACAGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_hyperantus_XM_034968762.1_Exon5\n",
      "GAGATGCCATGACACGTAGTGAGAAGTCGAGGTCGTCAAACCCCGGGTCTTCTTCAGAGACGACCAGTTCGTCAAGCAGTTCTCACGGATTTGCCGTAAAGGTTATCCAGAGGCCCGAGCTGCCACCTACCTTCCCAGTTGAGCCCTATGAGCAGGAGCGGCCGGCGGCTGGGCCCCAATATCCTATAGAGTGTGAAAGAG\n",
      ">Maniola_hyperantus_XM_034968762.1_Exon4\n",
      "CCGCCGAGCGACAAGCGGGAGAGGAATTCCGCGCTGGCTATCAAGAGGCCGTGGCGGAGGCAGTGCGCTTTCTCGTCGAAGTGCAAGGCTTCGGACCAGGGGACGGGCTATGCATGCAGCTCAGTTCACATATGCAGCGACATTGCGATGCTGTTACCAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_hyperantus_XM_034968762.1_Exon3\n",
      "CAAGACCCAATGTCTCACCGCATCATAGAGAAGCGTCGCCGCGACCGCATGAACAACTGCCTGGCGGACCTGTCGCGCCTCATCCCCGCGGAGTACCTGAAGAAGGGTCGCGGACGAGTCGAGAAGACCGAGATCATCGAGATGGCGATACGGCACCTCAAGTTCTTGCAGGACCGCGTCAATG\n",
      ">Maniola_hyperantus_XM_034968762.1_Exon2\n",
      "CTACAATTCGAACGACGGCTTCGGGCGCACAGGGGAGCCCCTCAGCTTCGCGCCGCCGTCAGAGGACGAGGCGGAGTACCCGCCCGGCGTCTACCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_hyperantus_XM_034968762.1_Exon1\n",
      "ATGGAAACACGTAACTATTGGGAGGAGAACGGACATCCCGTTAAATATGATAA\n",
      "179\t91\t141\t176\t201\t162\t184\t118\t53\t\n",
      "Maniola_jurtina XM_045908966.1\n",
      "+\n",
      "{'Exon_1': ['13717348', '13717400'], 'Exon_2': ['13747401', '13747518'], 'Exon_3': ['13748074', '13748257'], 'Exon_4': ['13750092', '13750253'], 'Exon_5': ['13750494', '13750694'], 'Exon_6': ['13751040', '13751215'], 'Exon_7': ['13752111', '13752251'], 'Exon_8': ['13752719', '13752809'], 'Exon_9': ['13753156', '13753334']}\n",
      "exon counter 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045908966.1_Exon1\n",
      "atggaAACACGTAATTATTGGGAGGAGAACGGACATCCCGTTAAATATGACAA\n",
      ">Maniola_jurtina_XM_045908966.1_Exon2\n",
      "CTACAACTCAAACGACGGCTTCGGGCGCACAGGCGAGCCGCTCAGCTTCGCGCCGCCGTCGGAGGACGAGGCGGAGTACCCGGCCGGCGCCTACCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045908966.1_Exon3\n",
      "CAAGACCCAATGTCTCACCGCATCATAGAGAAGCGCCGCCGCGACCGCATGAACAACTGCCTGGCGGACCTGTCGCGCCTCATCCCCGCGGAGTATCTGAAGAAGGGCCGCGGAAGAGTCGAGAAGACCGAGATCATCGAGATGGCGATACGGCACCTCAAGTTCTTGCAGGACCGCGTTAATG\n",
      ">Maniola_jurtina_XM_045908966.1_Exon4\n",
      "CCGCCGAGCGACAAGCGGGCGAGGAATTCCGCGCTGGCTATCAAGAGGCCGTGGCGGAGGCAGTACGCTTTCTCGTCGAGGTCCAAGGCTTCGGACCAGGGGACGGGCTATGCGTTCAGCTCGCTTCACATATGCAGCGACATTGCGATGCTGTTACCAAAG\n",
      ">Maniola_jurtina_XM_045908966.1_Exon5\n",
      "GAGATGCTATGACACGTAGTGACAAGTCGAGGTCGTCGAATCCCGGGTCATCTTCAGAGACAACCAGTTCGTCAAGCAGTTCACACGGATTTGCTGTAAAGGTTATCCAGAGGCCTGAGCTGCCACCTGCCTTCCCAGTTGAGTCCTATGAACAGGAGCGGCCGGCGGCTGGGCCTCAATATCCTATGGAGTGTGAAAGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045908966.1_Exon6\n",
      "TACAAAATCTAGCGGCACCAGAAGCCGAAAGCCAGGAACCCTTACCCCTGGATGGTCGGATCAAACGCGAGGCGTTGCGCAAAGTGCGCAAGATTGAACACTCGGACGATTATCTCCATTCCTATAAGTTCAAGAACTCTATAGAGAGGCGATTTAGCAGATCGCAAGATACTGAG\n",
      ">Maniola_jurtina_XM_045908966.1_Exon7\n",
      "GCGGCGGACTGTCCTCCACACAACGGGGTTAACGGCGCGACTCACGGCAAGCCTTATGCGCATAAGCGACGGCGTGCCGTCAAGCCTGCGCCCTCTGCTCCCACCTCCGCATCTACGTCCACTGAGGATACCTCGCCACAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Maniola_jurtina_XM_045908966.1_Exon8\n",
      "GACACGTCAAGCGAATCCCCTCATAACCACCACTACGACAAGCCTCCACCGCCGACGCAATATGTGCCAGTCTTCGCTCTCAATTCATTAG\n",
      ">Maniola_jurtina_XM_045908966.1_Exon9\n",
      "GGAAATACTACGTGCCACTAAGCGTAGAGTACGCTTGCATAGCGCGCCAGCTCGGTGCCATGGACTCGGAGACGCGTCCCATCACGGCGTTACATCCTGTCACAATACATGTCAACTTCCCCTGTCTCAACTACCACGTGAAGCGCGAGCCAGTTGAACGGCAGTGGCGACCATTGTAA\n",
      "53\t118\t184\t162\t201\t176\t141\t91\t179\t\n",
      "Melitaea_cinxia XM_045590661.1\n",
      "-\n",
      "{'Exon_1': ['15572584', '15572758'], 'Exon_2': ['15570896', '15571125'], 'Exon_3': ['15570711', '15570775'], 'Exon_4': ['15570488', '15570620'], 'Exon_5': ['15570330', '15570383'], 'Exon_6': ['15570105', '15570225'], 'Exon_7': ['15569381', '15569557'], 'Exon_8': ['15568592', '15568758'], 'Exon_9': ['15566952', '15567098'], 'Exon_10': ['15566029', '15566119'], 'Exon_11': ['15564817', '15564995']}\n",
      "exon counter 11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon11\n",
      "GCAAGTACTACGTGCCGCTGAGCGTGGAGTACGCGTGCGTGGCGCGGCAGCTGGGCGCGCTGGAGGCGGGCGCGCGGCCGCTGACGGCGCTGCACCCCGTCACCATACACGTCAACTTCCAGCCCTGCCTCGACTACCACGTCAAGCGCGAGCAGCCCGACCACTGGCGCCCCGGCTGA\n",
      ">Melitaea_cinxia_XM_045590661.1_Exon10\n",
      "GACACATCGAGCGACTCCCCACACCACCACCAGTACGATAAGCCGCCGCCTCCGACGCAGTACGTTCCTGTTTTCGCTTTGAATTCTCTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon9\n",
      "GCGGCCGagggcgcggcgcgcgcggcgcacGGCAAGCTGTACTCGCACAAGCGCCGGCGGGCCATCaagccgcccgcgcccgcctcCACCTCCGCCTCTACCTCCGCCTCCGTCTCCACCGAAGACGCGCACGACACCTCGCCACAG\n",
      ">Melitaea_cinxia_XM_045590661.1_Exon8\n",
      "TGTCCGCGCTGCCGGCGGCGGAGGCGGACGCGCCGCTGGACGAGCGGCTCAAGCGGGAGCCCGCGCTGCGCAAAGTACGCAAGCCGGACCACGGCGACGACTACCTCCACTCGTATAAGTTCAAGAACTCCATTGAAAGACGATTCTCTAGGTCGCAGGATTCAGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon7\n",
      "TGCCCATAGACAAGTCCCGCGACAAGGCGCGCTCGCACGCCTCGTCGTCGGAGACGGCCAGCTCGTCCAGCAGTTCGCACGGGTTCGCCGTGAAGGTGATCCAGCGCCCCGAGCCGCCACCCTTCCCGGAGCCCTACGAGGAACGCCCCGAGCCCTTCGCCATGGAATGCGACAGAG\n",
      ">Melitaea_cinxia_XM_045590661.1_Exon6\n",
      "CTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTTGTACTATTATTGTCCGTGTCTGCGCTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon5\n",
      "CTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTT\n",
      ">Melitaea_cinxia_XM_045590661.1_Exon4\n",
      "CTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTTGTACTATTACTGTCCGTGTCTGCAGCTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTT\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon3\n",
      "CTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTTGTACTGTTACT\n",
      ">Melitaea_cinxia_XM_045590661.1_Exon2\n",
      "CGGAGCGCTCGGGGGAGGAGTACCGCGCCGGGTTCCAGGAGGCGGTGGGGGAGGCCGTGCGCTTCCTCGTGGAGGTGCAGGGCTTCCGGCCCGGGGACGGGCTGTGTCTGCAGCTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCACTAAAGGTACTTCTGTTGTACTGTTACTGTACGTGTCTGCAGCTCGCCTCGCACCTGCAGCGCCACTGCGAGATCGTCAC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Melitaea_cinxia_XM_045590661.1_Exon1\n",
      "ATGTCTCATCGTATCATAGAGAAGCGTCGTCGCGACCGAATGAACAACTGCCTGGCTGATCTATCCCGCCTCATACCCCCGGAGTACCTCAAGAAGGGCAGGGGACGAGTCGAGAAGACCGAAATCATAGAAATGGCGATACGTCACCTCAAGTTCCTGCAGGATCGCGTCAATG\n",
      "179\t91\t147\t167\t177\t121\t54\t133\t65\t230\t175\t\n",
      "Nymphalis_io XM_050486679.1\n",
      "-\n",
      "{'Exon_1': ['3995018', '3995070'], 'Exon_2': ['3974248', '3974362'], 'Exon_3': ['3972862', '3973045'], 'Exon_4': ['3972210', '3972371'], 'Exon_5': ['3971860', '3972042'], 'Exon_6': ['3971064', '3971239'], 'Exon_7': ['3970462', '3970611'], 'Exon_8': ['3969044', '3969134'], 'Exon_9': ['3968258', '3968436']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050486679.1_Exon9\n",
      "GCAAGTACTACGTGCCGCTGAGCGTGGAGTACGCGTGCGTGTCGCGCCAGCTGGGCGCGCTGGAGGGCGACGTGCGGCCGCTGGCGGCGCTGCACCCCGTCACCATACACGTCAACTTCCAGCCGTGCCTCAACTACCACGTCAAGCGCGAGCCCGCCGACCCCTGGCGGCCCGCCTGA\n",
      ">Nymphalis_io_XM_050486679.1_Exon8\n",
      "GACACGTCGAGTGACTCCCCCCACCACCACCAGTACGACAAGCCGCCGCCTCCCACGCAATACGTGCCAGTCTTTGCATTAAATTCGCTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050486679.1_Exon7\n",
      "ACGGCGGAGGGCGCGGGGCGCGGCGCGGCGCACGGCAAGCTGTACTCGCACAAGCGGCGCCGGCAGGCCAAGCCCGCCGCGCCCGAGTCCACCtccgcctccacctccgcctCCGCCTCCACTGAGGACGCGCAGGACACTTCGCCGCAG\n",
      ">Nymphalis_io_XM_050486679.1_Exon6\n",
      "TATCGGTGTTGCCGGCGGCTGAGACGGACGCGCCGGATGCGCCGCTCGACGGACGTCTCAAGAAAGAACCTGCGCTGCGCAAAGTGCGGAAACCCGAACATACCGACGATTACCTCCACTCCTATAAATTTAAGAATTCCATAGAAAGGCGATTTTCGAGATCGCAAGATACCGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050486679.1_Exon5\n",
      "GCGAGCACACACGTGACAAGCCACGCTCGCACCCGGGGTCGTCGTCGGAGACTGCCAGCTCGTCCAGCAGCTCCCACGGGTTTGGAGTCAAGGTGATTCAGCGGCCAGAGCTTCCGCCTTTTCCAGCGGAGACCTACGACCATGAGCGACAAGAACCACCGTATGCTATGGAATGTGATAGAG\n",
      ">Nymphalis_io_XM_050486679.1_Exon4\n",
      "CGGCGGAGCGACAGGCCGGTGACGACTACCGCGCCGGCTACCAGGAGGCGGTGGCGGAGGCCGTGAGGTTCCTCGTCGAGGTGCAGGGATTCGGCCCCGGAGACGGCCTCTGCCTGCAGCTCGCGTCACATCTACAAAGGCATTGCGAGTTTGTTACTAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050486679.1_Exon3\n",
      "CAAGATCCGATGTCTCATCGCATCATCGAGAAACGTCGTCGTGACCGAATGAACAACTGTCTGGCTGACCTTTCACGCCTCATACCTCCGGAATACCTCAAAAAGGGCAGAGGCCGCGTCGAGAAGACCGAGATCATAGAAATGGCTATACGTCACCTCAAATTCCTACAGGACCGCGTCAATG\n",
      ">Nymphalis_io_XM_050486679.1_Exon2\n",
      "TTACAATTCAAACGACGGTTTTGGTCGTGCTGGCGAGCTCAACTTCGCGGCTCCTTCGGAGGACGAGGCTGAGTATCCTCCGGCAGTATTTCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Nymphalis_io_XM_050486679.1_Exon1\n",
      "ATGGAATCACATCATTATTGGGAAGAGAATGGACATGCCGTTAAGTACGACAA\n",
      "179\t91\t150\t176\t183\t162\t184\t115\t53\t\n",
      "Vanessa_atalanta XM_047670912.1\n",
      "-\n",
      "{'Exon_1': ['3834991', '3835043'], 'Exon_2': ['3812726', '3812840'], 'Exon_3': ['3810642', '3810825'], 'Exon_4': ['3809895', '3810056'], 'Exon_5': ['3809499', '3809732'], 'Exon_6': ['3808902', '3809077'], 'Exon_7': ['3808471', '3808620'], 'Exon_8': ['3807712', '3807802'], 'Exon_9': ['3807322', '3807500']}\n",
      "exon counter 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047670912.1_Exon9\n",
      "GCAAGTACTACGTGCCGCTGAGCGTGGAGTACGCGTGCGTGTCGCGGCAGCTGGGCGCGCTGGAGGCCGACGTGCGGCCGCTGGCGGCGCTGCACCCCGTCACCATACACGTCAACTTCCAGCCCTGCCTCAACTACCACGTCAAGCGCGAGCCCGCCGACCCCTGGCGGCCCGCCTGA\n",
      ">Vanessa_atalanta_XM_047670912.1_Exon8\n",
      "GACACGTCGAGCGATTCCCCCCACCACCACCAGTACGACAAGCCGCCGCCCCCCACGCAATACGTGCCAGTATTTGCATTAAATTCACTCG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047670912.1_Exon7\n",
      "TCGGCGGAGGGCGCGGCGCACGGCAAGCTGTACTCGCACAAGCGGCGCCGACAGGCCAAGCCCTGCGCGCCGCCCTCCACCTCCGCCTCGGCCTCCGCCTCTACCTCCGCCTCCGCCTCCACGGAGGACGCACAGGACACCTCGCCGCAG\n",
      ">Vanessa_atalanta_XM_047670912.1_Exon6\n",
      "TGTCGCAGCTGCCGGCGGCCGAGGCGGACGCGCCGTGCGCGCCGCTCGACGGGCGGCTCAAGAGGGAGCCTGCGCTGCGTAAAGTGCGGAAACCTGAACACACCGACGATTACCTTCACTCgtataaatttaagaattcCATAGAAAGGCGATTTTCGAGGTCGCAAGACACCGAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047670912.1_Exon5\n",
      "GCGAGCATACGCGCGACAAACCACGCTCGCATCCCGGCTCGTCCTCGGAGACGGCCAGCTCGTCGAGCAGCTCCCACGGGTTCGGAGTCAAGGTGATCCAGCGGCCAGAACTTCCGCCATTTCCGGCGGAGACCTACGACCATGAGCGACAAGAACCACCGTATGCTATGGAATGTGATAGAGGTTATAATGCATGTATTGGCCATACATACATTTCGCGACATCCTCGTTCAG\n",
      ">Vanessa_atalanta_XM_047670912.1_Exon4\n",
      "CGGCCGAGCGGCAGGCGGGAGATGATTACCGCTCCGGATACCAGGAGGCGGTGGCAGAAGCCGTGCGGTTCCTCGTCGAGGTACAGGGCTTTGGTCCCGGAGACGGCCTCTGTCTACAGCTTGCGTCGCATCTACAGAGGCATTGCGAGATTGTTACGAAAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047670912.1_Exon3\n",
      "CAAGACCCGATGTCTCATCGCATCATCGAGAAACGGCGTCGTGACCGAATGAACAACTGCCTGGCTGACCTTTCGCGCCTCATACCCCCGGAATACCTAAAAAAGGGCAGAGGCCGCGTCGAGAAGACCGAGATCATAGAGATGGCGATACGGCACCTCAAATTCCTGCAGGACCGCGTCAATG\n",
      ">Vanessa_atalanta_XM_047670912.1_Exon2\n",
      "TTACAATTCAAACGACGGCTTTGGTCGTGCAGGCGAGCTCAACTTCGCTGCACCCTCGGAGGACGAGGCAGATTATCCTCCCGCTGCATATCTCAAGAAGGGGAAGGTCTCGAGG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_atalanta_XM_047670912.1_Exon1\n",
      "atggaatcGCATCATTATTGGGAAGAGAATGGACATGCCGTTAAGTACGACAA\n",
      "179\t91\t150\t176\t234\t162\t184\t115\t53\t\n",
      "Vanessa_cardui XM_047104248.1\n",
      "+\n",
      "{'Exon_1': ['11214284', '11214336'], 'Exon_2': ['11239689', '11239803'], 'Exon_3': ['11240543', '11240738'], 'Exon_4': ['11241254', '11241415'], 'Exon_5': ['11241868', '11242101'], 'Exon_6': ['11242982', '11243136'], 'Exon_7': ['11248840', '11248989'], 'Exon_8': ['11249347', '11249437'], 'Exon_9': ['11250791', '11250969']}\n",
      "exon counter 1\n",
      ">Vanessa_cardui_XM_047104248.1_Exon1\n",
      "atggaatcaCATCATTATTGGGAAGAGAATGGACATGCCGTTAAGTACGACAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047104248.1_Exon2\n",
      "ttaCAATTCAAACGACGGCTTTGGTCGTGCCGGCGAGCTCAACTTCGCTGCACCCTCGGAGGACGAGGCAGACTATCCTCCCGCTGCGTATCTCAAGAAGGGGAAGGTCTCGAGG\n",
      ">Vanessa_cardui_XM_047104248.1_Exon3\n",
      "gtacatttGCAGCAAGACCCGATGTCTCATCGCATCATCGAGAAACGTCGTCGTGACCGAATGAACAACTGCCTGGCTGACCTTTCCCGCCTCATACCTCCGGAGTACCTTAAAAAGGGCAGGGGCCGCGTCGAGAAGACCGAGATCATAGAGATGGCGATACGGCACCTCAAATTCCTGCAGGACCGCGTCAATG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047104248.1_Exon4\n",
      "CGGCCGAGCGCCAGGCGGGCGACGACTACCGCTCCGGCTACCAGGAGGCCGTGGCGGAGGCCGTGAGATTCCTCGTCGAGGTTCAGGGTTTTGGTCCCGGAGACGGCCTCTGTCTACAGCTTGCGTCGCATCTACAGAGGCATTGCGAAATTGTTACAAAAG\n",
      ">Vanessa_cardui_XM_047104248.1_Exon5\n",
      "GCGAGCACACGCGCGAGAAGCCCCGCTCGCACCCCGGCTCCTCCTCGGAGACGGCGAGCTCGTCCAGCAGTTCCCACGGGTTCGGAGTCAAGGTGATCCAGCGGCCGGAGCTGCCGCCCTTCCCGACGGAGACCTACGACCCTGAGCGGCAAGAACCACCGTATGCTATGGAATGTGATAGAGGTTATAATGCATGTATTGGCCATACATACATTTCGCGACACCCCTTTTCAG\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047104248.1_Exon6\n",
      "TGTCGCAGCTGCCGCCGGCCGAGGCGGACGCGCCGCTCAAGCGCGAGCCGGCGCTGCGCAAGGTGCGCAAGCCCGACCACACCGACGACTACCTCCACTCCTACAAATTTAAGAATTCCATCGAAAGGCGATTTTCGAGGTCCCAGGACACCGAG\n",
      ">Vanessa_cardui_XM_047104248.1_Exon7\n",
      "TCGGCAGAGGGCGCGGCGCACGGCAAGCTGTACTCGCACAAGCGGCGCCGACAGGCCAAGCCTCCCGCGCCGCCCTCCACCTCTGCTTCAGCCTCCGCCTCTACCTCTGCCTCCGCCTCCACGGAGGACGCACAGGACACCTCGCCACAA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Vanessa_cardui_XM_047104248.1_Exon8\n",
      "GACACGTCGAGCGATTCCCCCCACCACCACCAGTACGACAAGCCGCCGCCTCCCACGCAATACGTGCCAGTCTTTGCGTTAAATTCACTCG\n",
      ">Vanessa_cardui_XM_047104248.1_Exon9\n",
      "GCAAGTACTACGTGCCGCTGAGCGTGGAGTACGCGTGCGTGTCGCGGCAGCTGGGCGCGTTGGAGGCGGGCGTGCGGCCCCTGGCGGCGCTGCACCCCGTCACCATACACGTCAACTTCCAGCCCTGCCTCAACTACCACGTCAAGCGCGAACCCAACGACCCCTGGCGCCCTGCCTGA\n",
      "53\t115\t196\t162\t234\t155\t150\t91\t179\t\n",
      "Vanessa_tameamea XM_064215129.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "{}\n",
      "Error!!\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 59\u001b[0m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     58\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mError!!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m     61\u001b[0m \u001b[38;5;66;03m# list_of_query_species = os.listdir(query_location)\u001b[39;00m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;66;03m# if query_species not in list_of_query_species:\u001b[39;00m\n\u001b[1;32m     63\u001b[0m \u001b[38;5;66;03m#     os.mkdir(f\"{annotated_species_location}/{query_species}\")\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \u001b[38;5;66;03m# list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\u001b[39;00m\n\u001b[1;32m     65\u001b[0m \u001b[38;5;66;03m# if \"Transcripts\" not\u001b[39;00m\n\u001b[1;32m     66\u001b[0m \u001b[38;5;66;03m#     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\u001b[39;00m\n\u001b[1;32m     68\u001b[0m print_out \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "# species_gene = [\"Vanessa_tameamea,XM_026631919.1\"]\n",
    "species_gene = [\"Vanessa_tameamea,XM_026630430.1\",\"Papilio_machaon,XM_014501859.2\",\"Papilio_polytes,XM_013278899.1\",\"Papilio_xuthus,XM_013322560.1\",\"Colias_croceus,XM_045637564.1\",\"Leptidea_sinapis,XM_050824666.1\",\"Pieris_brassicae,XM_045664337.1\",\"Pieris_napi,XM_047659772.1\",\"Pieris_rapae,XM_022273623.2\",\"Zerene_cesonia,XM_038354443.1\"]\n",
    "\n",
    "# gene_name = \"XM_013278379.1\"\n",
    "# species = \"Papilio_polytes\"\n",
    "annotated_species_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/11.CWO_Exon_Analysis/1.Annotated Species\"\n",
    "try:\n",
    "    os.mkdir(annotated_species_location)\n",
    "except:\n",
    "    pass\n",
    "for entries in species_gene:\n",
    "\n",
    "    species,gene_name = entries.split(\",\")\n",
    "    print(species,gene_name)\n",
    "\n",
    "    genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    query_species = species\n",
    "    create_query_folder(annotated_species_location,gene_name,species)\n",
    "    list_of_files = os.listdir(f\"{genome_location}/{species}\")\n",
    "    # print(list_of_files)\n",
    "    for file_names in list_of_files:\n",
    "        if file_names.endswith(\"_genomic.gff\"):\n",
    "            gff3_file = file_names\n",
    "\n",
    "    with open(f\"{genome_location}/{species}/{gff3_file}\") as gff_file_open:\n",
    "        gff_file_list = gff_file_open.readlines()\n",
    "    complement = ''\n",
    "    exon_count = 1\n",
    "    gene_found = 0\n",
    "    exon_dictionary = {}\n",
    "    for line in gff_file_list:\n",
    "        if gene_name in line:\n",
    "            if \"CDS\" in line.split(\"\\t\"):\n",
    "\n",
    "    #             print(line)\n",
    "                gene_found = 1\n",
    "                if complement == '':\n",
    "                    complement = line.split(\"\\t\")[6]\n",
    "                exon_start = line.split(\"\\t\")[3]\n",
    "                exon_stop = line.split(\"\\t\")[4]\n",
    "                scaffold = line.split(\"\\t\")[0]\n",
    "                exon_dictionary[f\"Exon_{exon_count}\"] = [exon_start,exon_stop]\n",
    "                exon_count += 1\n",
    "            if \"exon\" in line.split(\"\\t\") and gene_found == 1:\n",
    "                break\n",
    "    print(complement)\n",
    "    print(exon_dictionary)\n",
    "    if complement == \"+\":\n",
    "        exon_counter = 1\n",
    "    elif complement == \"-\":\n",
    "        exon_counter = len(exon_dictionary)\n",
    "    else:\n",
    "        print(\"Error!!\")\n",
    "        assert False\n",
    "\n",
    "    # list_of_query_species = os.listdir(query_location)\n",
    "    # if query_species not in list_of_query_species:\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\n",
    "    # if \"Transcripts\" not\n",
    "    #     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\n",
    "\n",
    "    print_out = ''\n",
    "    print(\"exon counter\", exon_counter)        \n",
    "    for i in range(1,len(exon_dictionary)+1):\n",
    "\n",
    "        exon_start, exon_stop = exon_dictionary[f\"Exon_{exon_counter}\"]\n",
    "    #     print(exon_start,exon_stop)\n",
    "        subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{exon_start}-{exon_stop} > \"{genome_location}/{species}/Temp/temp_gene.fa\"', shell = True)\n",
    "        exon_sequence_file = SeqIO.parse(f\"{genome_location}/{species}/Temp/temp_gene.fa\", \"fasta\")\n",
    "        sequence = ''\n",
    "\n",
    "        for sequences in exon_sequence_file:\n",
    "    #         print(sequences.id)\n",
    "            if complement == \"-\":\n",
    "                sequence = sequences.seq.reverse_complement() \n",
    "\n",
    "    #             print(sequences.seq.reverse_complement().translate())\n",
    "            else:\n",
    "                sequence = sequences.seq\n",
    "    #             print(sequences.seq)\n",
    "    #             print(sequences.seq.translate())\n",
    "            print_out += str(len(sequence))+\"\\t\"\n",
    "            output = f\">{query_species}_{gene_name}_Exon{exon_counter}\\n{sequence}\"\n",
    "        with open(f\"{annotated_species_location}/{query_species}/Transcripts/{gene_name}/Exon{exon_counter}.fa\", 'w') as outfile:\n",
    "            outfile.write(output)\n",
    "        print(output)\n",
    "        if complement == \"-\":\n",
    "            exon_counter -= 1\n",
    "        else:\n",
    "            exon_counter += 1\n",
    "\n",
    "    #         if gene_location in line.split(\"\\t\"):\n",
    "    #           print(line)\n",
    "    #           break\n",
    "    print(print_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "111273de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0002681f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
