{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2b2fddd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion\n",
      "Erebia_aethiops\n",
      "Erebia_ligea\n",
      "Hipparchia_semele\n",
      "Lasiommata_megera\n",
      "Maniola_hyperantus\n",
      "Maniola_jurtina\n",
      "Melanargia_galathea\n",
      "Oeneis_ivallda\n",
      "\n",
      "Anthocharis_cardamines\n",
      "Aporia_crataegi\n",
      "Leptidea_juvernica\n",
      "Leptidea_reali\n",
      "Leptidea_sinapis\n",
      "Leptophobia_aripa\n",
      "Pieris_brassicae\n",
      "Pieris_mannii\n",
      "Pieris_melete\n",
      "Pieris_napi\n",
      "Pieris_rapae\n",
      "\n",
      "Colias_behrii\n",
      "Colias_croceus\n",
      "Zerene_cesonia\n",
      "\n",
      "Argynnis_bischoffii_washingtonia\n",
      "Boloria_euphrosyne\n",
      "Boloria_selene\n",
      "Brenthis_daphne\n",
      "Brenthis_hecate\n",
      "Brenthis_ino\n",
      "Danaus_chrysippus\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta\n",
      "Dryadula_phaetusa\n",
      "Dryas_iulia_moderata\n",
      "Eueides_isabella\n",
      "Euphydryas_editha\n",
      "Fabriciana_adippe\n",
      "Heliconius_charithonia\n",
      "GCA_030704555.1_Hchar_UCI_1.0_genomic.fna\n",
      "4041357 4041610\n",
      "0 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "1 Q\n",
      "gc\n",
      "MTSLLSAELSATCNALGTFNKKTKKYMIDEFTLNTVKDLIRYLRRDGEDHEIRRHFGQTKVLQTDLIPMLIDHWEKEELFDVTL\n",
      ">Heliconius_charithonia_Exon_1_CM060814.1_4041357_4041610_left_0_right_2\n",
      "ATGACTTCGTTATTATCAGCCGAACTTTCTGCCACTTGCAATGCTCTTGGAACATTTAATAAGAAAACCAAAAAGTACATGATAGATGAATTTACACTTAATACTGTTAAAGATCTTATAAGATATCTTAGACGTGATGGCGAGGACCACGAAATTCGCAGACACTTTGGACAAACTAAGGTTTTACAAACTGACTTAATACCGATGCTAATTGATCACTGGGAAAAGGAAGAATTATTTGATGTCACACTCAG\n",
      "4041731 4041914\n",
      "1 0\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "2 I\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_2_CM060814.1_4041731_4041914_left_1_right_0\n",
      "ACTTCTTGTCAATTTAACAAATCCAGCTCTTTTATTATACCGTGAAGAAGTACCAGTAGAACGAACTGCTCgtcataattatttacaaatactctcacatttacaatattataaagaggAAGCATTTACTAAAGTTGAAACATGGCAGGTCTTTGCTAAAAAGTTAGCTAAGATATTAGAAATA\n",
      "4041993 4042127\n",
      "0 0\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "3 Q\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_3_CM060814.1_4041993_4042127_left_0_right_0\n",
      "GATTGGTCTGAAAGAGATGAAGATACGGGGTTGATAATtgaaagaattttaatattgataagaaATGTATTACATGTACCTGCTGACTTAGATAAAGAACGAAGACCAGAAAATGATGCTAGTGTCCATGATcag\n",
      "4042952 4043208\n",
      "0 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "4 *\n",
      ">Heliconius_charithonia_Exon_4_CM060814.1_4042952_4043208_left_0_right_2\n",
      "GTATTATGGGCGTTAAACCAAGCTGGTATCttggatataatattatatatgtcatctgaaaatgaaaaacagtattttatgcatataatagaaattatatcaCATTTACTGCGGGAACAAAACCCGACTAGCTTAGCAGATGCAGCTTTACAAAGAAGTGTGGATGAAAAATTGAAAGACGACCAAGAATTACTAATGTTGAGAATGTCAGAGAGTAAAGAGAGAATGAATAAAGTTAAACAGTATTCGTCTActag\n",
      "4043292 4043489\n",
      "1 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "5 *\n",
      ">Heliconius_charithonia_Exon_5_CM060814.1_4043292_4043489_left_1_right_2\n",
      "ACACTCCCGCTTTGGTGGTACATATGTCATACAAAACATGAAGTCTATATCAGACAATGAGCTAATATGTCTAAAAccattaaacaaaatatctaatttagattttaatggTAGTAAAAAGTCTAAGCTAGTTAGACCTAAGAACAGAAGACCTGTTGAAAGTGGCATATTGGAAAGGAGATCAGCTTTTGCTATTAG\n",
      "4043866 4044048\n",
      "1 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "6 *\n",
      ">Heliconius_charithonia_Exon_6_CM060814.1_4043866_4044048_left_1_right_2\n",
      "attattCTTAAAAGAGTTTTGCATAGAATTTTTAAACAGTTCATATAATCCTTTAATGCATTACGTGAAAGATGTATTGGTAAGAGCGAAAGCGCAACAGAATGATGAGTCATATTACTTGTGGGCTATGAAGTTTTTTATGGAATTCAATAGAGGACATCATTTCCAAGTCGGATTAGTTAG\n",
      "4044361 4044648\n",
      "1 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "7 E\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_7_CM060814.1_4044361_4044648_left_1_right_2\n",
      "CGAAACAATGTCAGTTCCGATGTTCCATTACGTACAACAGCAAATGGAGAAATATTACGACATGATTAAAGTCGAGAAGAAGAAATTCTCCACTTGGGTGAGAAGACTACACCTCGCTCTTAGAGCTTACAAGGAACTTCTGAATACATTACTCGCTATGGACAAAAGTAACGACACAATGGTTAAAGAATCTGCTAAAGTGTTGAAAAGCAATATTTTCTATGTGTTGGAGTACAGGGAGTTCATTTTGAGTATATTTCTGAACTATGATGAGAGTAAGATGCCGAG\n",
      "4044806 4044930\n",
      "1 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "8 E\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_8_CM060814.1_4044806_4044930_left_1_right_1\n",
      "ATCATATTTAGTAGATTTGATAGAAACAGTTCATTTGTTCCTTAAAATGTTAGAGCACTATTGTAAGAGGACGGGGCTGGTGGTGCAGAAGAAGGCTCGGAAGAAGACCAAGACCAAGTCTAGAA\n",
      "4045333 4045474\n",
      "2 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "9 K\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_9_CM060814.1_4045333_4045474_left_2_right_2\n",
      "AGCAGAAGCCGGCTCCGACACTGCTGATGGGATGGGGATGGGAGCGAGTGAGGCCTCAAATGGCCGCAGTACTGAGTTCTGCTACAGAAGAGTGTCCGCCGCCTTTCGACGCGGCCTCTGACGTGCCCATTGACCAACAAAA\n",
      "4047084 4047164\n",
      "1 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "10 E\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_10_CM060814.1_4047084_4047164_left_1_right_2\n",
      "GGGAGATTGTatgaagaaaatacaaaaactaaTGCGGGACCACAAATACGAAGATGCCGTCGGAACATTCCGAGCCGCGAG\n",
      "4047343 4047449\n",
      "1 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "11 L\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_11_CM060814.1_4047343_4047449_left_1_right_1\n",
      "ggAAGTGTGGCCCGAAGAAGGGATATTTGGTGCGAGCGATATCAAAGAAGATGAAGAGTTAGATATGCTGGAGACTATTTATAACACGGACTTGGGAATGGACGctg\n",
      "4048724 4048771\n",
      "2 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "12 M\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_12_CM060814.1_4048724_4048771_left_2_right_1\n",
      "gtaCAGTTAAAGAAGAACCACAAGAAAATTCAGAATACGATGAAGAtg\n",
      "4048973 4049045\n",
      "2 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "13 K\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_13_CM060814.1_4048973_4049045_left_2_right_2\n",
      "acgaAGAAGACGAAGAAGAAAAGTCGACAGCTGTGGTTGAGACGGACTTTGATTTTCAAGATTTCGTTCTAAG\n",
      "4049350 4049548\n",
      "1 0\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "14 E\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_14_CM060814.1_4049350_4049548_left_1_right_0\n",
      "GTTCTGCCACCAGCGAGTAGTGAGCGCGTGTGTGCGTCTGTTAGAGCActacctccagaactcgccgCACACGAACCACTGCGCCGTCAAGATGCTGCACAGAGTCGCGCGGGACTGCAACCGACCGGCCGCCATGTTTCAGGCCTCGTTGTTTGTGGTCTTCCAGAGGGTGTTGGGCAACCCTGGGCCGTTATTCGAG\n",
      "4053275 4053423\n",
      "0 2\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "15 Q\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_15_CM060814.1_4053275_4053423_left_0_right_2\n",
      "GAATTGGAAAAGTTCGCCATATTTGTTCTAAGGAAATTCGCTGAAATGGCATCGAGGAGTAGCAAAGTTTTTATCGAGCTGTTATTTTGGAAGAACTCTAAAGAttgtgtggaaatcgaacacggCTACGACCTCTACAACGACCCCAG\n",
      "4056527 4056621\n",
      "1 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "16 K\n",
      "gt\n",
      "GGATAAGCCGGGGAACTCGCAGTGGACGGAGGAACAGGAGGAGGAGCTTCGTAAACTTTACCTGGAGAATCAGACCAACCCTGAAAGTGATCAAG\n",
      ">Heliconius_charithonia_Exon_16_CM060814.1_4056527_4056621_left_1_right_1\n",
      "GGATAAGCCGGGGAACTCGCAGTGGACGGAGGAACAGGAGGAGGAGCTTCGTAAACTTTACCTGGAGAATCAGACCAACCCTGAAAGTGATCAAG\n",
      "4057939 4058145\n",
      "2 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "17 K\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_17_CM060814.1_4057939_4058145_left_2_right_1\n",
      "ATGTAATCGACTGGATACTAGAAAATTTAATAGTACAAACTCGCACTCGTAGAAGTGTTATCAAAAAACTCAAAGAACTGGGACTTATCTTCAAAGCGCCGACTAAGAAAAGCAATAAGGAGAGGAGAGAAAGAGTGCCTTTGGAGTTTACTGAAGAGGAAGATCATATTTTGACGGAACTTTGGAAGGAACATGGCGAGactaaag\n",
      "4059947 4060084\n",
      "2 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "18 T\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_18_CM060814.1_4059947_4060084_left_2_right_1\n",
      "ACGCACTACACACAATAGCGTCCAGAATGCCTAACAAACGATCCAAACGAAGCCTCAAAGAGAGACTCACAGAACTCGGCTTGAGAACAAATAAGCGCAAGAAGAAAGAAGCGTTTAAAAGCGATAATAAATCGAACG\n",
      "4060470 4060646\n",
      "2 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "19 M\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_19_CM060814.1_4060470_4060646_left_2_right_1\n",
      "aaacttCATCGGATTCAGACAGTTCTAGTTCAGACAACGAGGCCGGTAGGGCTCCCAAACTTCATATAAAACCTAAGAAGAAACCGACGACACAAAGTAGAAATGAGAAACAAACAAAGAGTATGGAACTAAGTGCTAATGAAATTGCGAAACTGTTGGTTCAAGCTGTTGATGATG\n",
      "4061823 4061998\n",
      "2 0\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "20 V\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_20_CM060814.1_4061823_4061998_left_2_right_0\n",
      "ATTTAACAGAAGCATTAAAGTGGTTAGCGGAAAGTCTAGAAGAAGTAGCCCTCGACCAAGAGGCTGAAGGCTTCGATCCTCACTCCGAAGGAACACCTCTAGTACCGATATCCCGTGAAGCTATCGAAGCCATCGGGAACAAGttgtttatcaaaatattaaaaagtatcgGAGTC\n",
      "4062333 4062912\n",
      "0 1\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "21 A\n",
      "gt\n",
      ">Heliconius_charithonia_Exon_21_CM060814.1_4062333_4062912_left_0_right_1\n",
      "gaAACTTACTGGAGAATACCAAGTAACTTACACTTTGAAACTATACGCAAAAGAAGAGAAATCATTATGAAAGCTATAAACAAAGAGCTTATAATAGATAATGATAACAACACGCAGACAAATAATGATGCAAATGGAActgaaaataatgataatgcTGAATCTAGTGATGACgatgatttatttgaaaatttaagaaaaatacaaaatattcctaATAAAAGTGGTTCAAGTAAATCAATTAATCGCAGTCGAAATCGTAATTTCGATTCTGACGACGATACAACAAAACcaaaagaaaaatcaaataCCAATAAGAATGATTTATTCGATAAATTACTGAACGATATTCCGAGTAGTACTAAACAATCGAATAATCGAAAACGTAGTCGTAGTATAGATTCTGATGATGACAATATTCGGAAAGtacaaaaagttaaaagaattaatgataataatgatgacgatgatgatgatgatgatgatgatgatgatattttaTCCTTCGCAAAGAGAACGTTACAACACACGGACACTGAAGACATTACGAGTGGTTTAGCTCAACGAGCTGTAAATAGCG\n",
      "4063471 4063538\n",
      "2 0\n",
      "Getting Gene\n",
      "Getting Gene\n",
      "22 *\n",
      ">Heliconius_charithonia_Exon_22_CM060814.1_4063471_4063538_left_2_right_0\n",
      "ATAGCGAGGAAGAAATGGCGCCCGTTAAGAAACACAAGAAACTGGTAATTAGCGACGATTCAGACTGA\n",
      "Heliconius_nattereri\n",
      "Hestina_assimilis\n",
      "Hypolimnas_misippus\n",
      "Limenitis_camilla\n",
      "Mechanitis_mazaeus\n",
      "Mechanitis_messenoides\n",
      "Melinaea_marsaeus_rileyi\n",
      "Melinaea_menophilus\n",
      "Melitaea_cinxia\n",
      "Mellicta_athalia\n",
      "Napeogenes_inachia\n",
      "Napeogenes_sylphis\n",
      "Nymphalis_c-album\n",
      "Nymphalis_io\n",
      "Nymphalis_polychloros\n",
      "Nymphalis_urticae\n",
      "Philaethria_dido\n",
      "Vanessa_atalanta\n",
      "Vanessa_cardui\n",
      "Vanessa_tameamea\n",
      "\n",
      "\n",
      "\n",
      "Baronia_brevicornis\n",
      "Battus_philenor\n",
      "Iphiclides_podalirius\n",
      "Luehdorfia_chinensis\n",
      "Ornithoptera_alexandrae\n",
      "Ornithoptera_priamus\n",
      "Papilio_demoleus\n",
      "Papilio_gigon\n",
      "Papilio_machaon\n",
      "Papilio_memnon\n",
      "Papilio_polytes\n",
      "Papilio_protenor\n",
      "Papilio_xuthus\n",
      "Parnassius_glacialis\n",
      "Sericinus_montela\n",
      "Teinopalpus_imperialis\n",
      "Troides_aeacus\n",
      "Troides_oblongomaculatus\n",
      "\n",
      "Aricia_agestis\n",
      "Aricia_artaxerxes\n",
      "Calycopis_cecrops\n",
      "Celastrina_argiolus\n",
      "Cyaniris_semiargus\n",
      "Eumaeus_atala\n",
      "Glaucopsyche_alexis\n",
      "Helleia_helle\n",
      "Lycaena_phlaeas\n",
      "Lysandra_bellargus\n",
      "Lysandra_coridon\n",
      "Phengaris_arion\n",
      "Plebejus_argus\n",
      "Polyommatus_icarus\n",
      "Polyommatus_iphigenia\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "out_test = ''\n",
    "family_group_list = [\"3.Satyrine\",\"4.Pierinae\",\"5.Coliadinae\",\"6.Heliconiinae_Danainae_Nymphalinae\",\"7.Papilionidae\",\"8.Lycaenidae\"]\n",
    "# family_group_list = [\"3.Satyrine_exon1_dup\"]\n",
    "for family_group in family_group_list:\n",
    "    output_location = f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/15.Timeout_Exon_Analysis/{family_group}\"\n",
    "    make_output_folder(output_location)\n",
    "    genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "    list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "    if 'desktop.ini' in list_of_species:\n",
    "        list_of_species.remove(\"desktop.ini\")\n",
    "    # print(list_of_species)\n",
    "    # list_of_species = [\"Heliconius_charithonia\"]\n",
    "    # list_of_species = [\"Maniola_hyperantus\"]\n",
    "    error_exons = ''\n",
    "    for species in list_of_species:\n",
    "        print(species)\n",
    "        if species != \"Heliconius_charithonia\":\n",
    "            continue\n",
    "        genome_file = get_genome_file(genome_location,species)\n",
    "        print(genome_file)\n",
    "        previous_coordinate_start = ''\n",
    "        previous_scaffold = ''\n",
    "        with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "            coordinate_file_list = coordinate_final_file.readlines()\n",
    "        \n",
    "        for i in range(1,len(coordinate_file_list)):\n",
    "            \n",
    "            coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "            \n",
    "                \n",
    "    #         print(coodinate_line_split[5])\n",
    "            if coordinate_line_split[5] == \"N\":\n",
    "                if previous_coordinate_start == '':\n",
    "                    previous_coordinate_start = coordinate_line_split[2]\n",
    "                    previous_scaffold = coordinate_line_split[1]\n",
    "                complement = coordinate_line_split[4]\n",
    "                start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "                print(start,stop)\n",
    "                query_name = coordinate_line_split[6]\n",
    "                scaffold = coordinate_line_split[1]\n",
    "                exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "                left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "                print(left_oh, right_oh)\n",
    "    #             print(complement)\n",
    "                if complement == \"0\":\n",
    "                    if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                        print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                        prompt = input(\"Continue (Y/N)?\")\n",
    "                        if prompt[0].lower() == \"n\":\n",
    "                            assert False\n",
    "                elif complement == \"1\":\n",
    "                    if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                        print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                        prompt = input(\"Continue (Y/N)?\")\n",
    "                        if prompt[0].lower() == \"n\":\n",
    "                            assert False\n",
    "                else:\n",
    "                    print(\"ERROR\")\n",
    "                \n",
    "            elif coordinate_line_split[5] == \"Y\":\n",
    "                query_name = coordinate_line_split[6]\n",
    "                exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "                scaffold = coordinate_line_split[1]\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                continue\n",
    "                \n",
    "            gene_sequence = get_gene_sequence(genome_location, \n",
    "                                              species, \n",
    "                                              genome_file, \n",
    "                                              scaffold,\n",
    "                                              start,\n",
    "                                              stop,\n",
    "                                              complement,\n",
    "                                              output_location)\n",
    "            \n",
    "            gene_sequence_2 = get_gene_sequence(genome_location, \n",
    "                                              species, \n",
    "                                              genome_file, \n",
    "                                              scaffold,\n",
    "                                              int(start)-2,\n",
    "                                              int(stop)+2,\n",
    "                                              complement,\n",
    "                                              output_location)\n",
    "        \n",
    "            current_exon = query_name.split('_')[-1]\n",
    "            test = ''\n",
    "            if current_exon != \"1\":        \n",
    "                if gene_sequence_2[:2].lower() != \"ag\":\n",
    "                    print(species, current_exon)\n",
    "                    print(gene_sequence)\n",
    "                    print(gene_sequence_2)\n",
    "                    print(\"Not AG\")\n",
    "                    test = input(\"quit? (y,n)\")\n",
    "                    if test == \"y\":\n",
    "                        assert False\n",
    "                    # assert False\n",
    "            print(current_exon, gene_sequence[-3:].translate())\n",
    "            if gene_sequence[-3:].translate() != \"*\":\n",
    "                \n",
    "                print(gene_sequence_2[-2:].lower())\n",
    "                if (gene_sequence_2[-2:].lower() != \"gt\") and (gene_sequence_2[-2:].lower() != \"gc\"):\n",
    "                    print(current_exon)\n",
    "                    print(len(coordinate_file_list))\n",
    "                    print(species, current_exon)\n",
    "                    print(gene_sequence)\n",
    "                    print(gene_sequence_2)\n",
    "                    print(\"Not gt\")\n",
    "                    test = input(\"quit? (y,n)\")\n",
    "                    if test == \"y\":\n",
    "                        assert False\n",
    "            if current_exon == \"1\":\n",
    "                if complement == 1:\n",
    "                    translated = gene_sequence.reverse_complement.translate()\n",
    "                else:\n",
    "                    translated = gene_sequence.translate()\n",
    "                print(translated)\n",
    "                if \"*\" in translated:\n",
    "                    error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                else:\n",
    "                    exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                    print(exon_output)\n",
    "            \n",
    "                    write_exon_output(output_location,species,exon,exon_output)\n",
    "            elif current_exon == \"16\":\n",
    "                if complement == 1:\n",
    "                    translated = gene_sequence[:-int(left_oh)].reverse_complement.translate()\n",
    "                    print(gene_sequence)\n",
    "                else:\n",
    "                    translated = gene_sequence[int(left_oh):].translate()\n",
    "                    print(gene_sequence)\n",
    "           \n",
    "                \n",
    "                if \"*\" in translated[:-1]:\n",
    "                    error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                    print(translated)\n",
    "                else:\n",
    "                    \n",
    "                    exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                    print(exon_output)\n",
    "                    write_exon_output(output_location,species,exon,exon_output)\n",
    "    #         assert False\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "    \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "            if test == \"n\":\n",
    "                out_test += (f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\")\n",
    "    #         assert False\n",
    "            previous_coordinate_start = start\n",
    "            previous_scaffold = scaffold\n",
    "        error_exons += \"\\n\\n\"\n",
    "    #     print(coordinate_file_list)\n",
    "    #     break\n",
    "    coordinate_file = ''\n",
    "    print(error_exons)\n",
    "    with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "        error_out_file.write(error_exons)\n",
    "print(out_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db6a97d9-863f-4fcd-bdcf-a098c9c0cd6e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
