{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c0786cde-8632-4a2f-b5b3-de07ad86347b",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "chrom column should be numeric. Do you have 'X', 'Y', 'MT', etc? If so change to numbers and try again.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[20], line 16\u001b[0m\n\u001b[1;32m      6\u001b[0m df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_csv(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/61.New_alignment_colias/7.Manhatten/dataset.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      8\u001b[0m \u001b[38;5;66;03m# n_chr = 23  # number of chromosome pairs in humans\u001b[39;00m\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m# assert 'CHR' in df.columns\u001b[39;00m\n\u001b[1;32m     10\u001b[0m \u001b[38;5;66;03m# assert df['CHR'].max() == n_chr\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;66;03m# DATASET = df.groupby('CHR').apply(lambda u: u.head(50))\u001b[39;00m\n\u001b[1;32m     14\u001b[0m \u001b[38;5;66;03m# DATASET = DATASET.droplevel('CHR').reset_index(drop=True)\u001b[39;00m\n\u001b[0;32m---> 16\u001b[0m manhattanplot \u001b[38;5;241m=\u001b[39m \u001b[43mdashbio\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mManhattanPlot\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     17\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdataframe\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     18\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchrm\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mchrom\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     19\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mBP\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     20\u001b[0m \u001b[43m    \u001b[49m\u001b[43mp\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mskane.lulea.fst\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     21\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhighlight_color\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m#00FFAA\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\n\u001b[1;32m     22\u001b[0m \u001b[43m)\u001b[49m\n\u001b[1;32m     24\u001b[0m \u001b[38;5;66;03m# dcc.Graph(figure=manhattanplot)\u001b[39;00m\n\u001b[1;32m     26\u001b[0m plotly\u001b[38;5;241m.\u001b[39moffline\u001b[38;5;241m.\u001b[39mplot(manhattanplot, image\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpng\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/miniconda3/envs/genomics_work/lib/python3.11/site-packages/dash_bio/component_factory/_manhattan.py:127\u001b[0m, in \u001b[0;36mManhattanPlot\u001b[0;34m(dataframe, chrm, bp, p, snp, gene, annotation, logp, title, showgrid, xlabel, ylabel, point_size, showlegend, col, suggestiveline_value, suggestiveline_color, suggestiveline_width, genomewideline_value, genomewideline_color, genomewideline_width, highlight, highlight_color)\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mManhattanPlot\u001b[39m(\n\u001b[1;32m     16\u001b[0m         dataframe,\n\u001b[1;32m     17\u001b[0m         chrm\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCHR\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     38\u001b[0m         highlight_color\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mred\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     39\u001b[0m ):\n\u001b[1;32m     40\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Returns a figure for a manhattan plot.\u001b[39;00m\n\u001b[1;32m     41\u001b[0m \n\u001b[1;32m     42\u001b[0m \u001b[38;5;124;03mKeyword arguments:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    124\u001b[0m \n\u001b[1;32m    125\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 127\u001b[0m     mh \u001b[38;5;241m=\u001b[39m \u001b[43m_ManhattanPlot\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    128\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdataframe\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    129\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchrm\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchrm\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    130\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    131\u001b[0m \u001b[43m        \u001b[49m\u001b[43mp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    132\u001b[0m \u001b[43m        \u001b[49m\u001b[43msnp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msnp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    133\u001b[0m \u001b[43m        \u001b[49m\u001b[43mgene\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mgene\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    134\u001b[0m \u001b[43m        \u001b[49m\u001b[43mannotation\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mannotation\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    135\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlogp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlogp\u001b[49m\n\u001b[1;32m    136\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    138\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m mh\u001b[38;5;241m.\u001b[39mfigure(\n\u001b[1;32m    139\u001b[0m         title\u001b[38;5;241m=\u001b[39mtitle,\n\u001b[1;32m    140\u001b[0m         showgrid\u001b[38;5;241m=\u001b[39mshowgrid,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    153\u001b[0m         highlight_color\u001b[38;5;241m=\u001b[39mhighlight_color\n\u001b[1;32m    154\u001b[0m     )\n",
      "File \u001b[0;32m~/miniconda3/envs/genomics_work/lib/python3.11/site-packages/dash_bio/component_factory/_manhattan.py:221\u001b[0m, in \u001b[0;36m_ManhattanPlot.__init__\u001b[0;34m(self, x, chrm, bp, p, snp, gene, annotation, logp)\u001b[0m\n\u001b[1;32m    219\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    220\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m is_numeric_dtype(x[chrm]\u001b[38;5;241m.\u001b[39mdtype):\n\u001b[0;32m--> 221\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m column should be numeric. Do you have \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    222\u001b[0m                         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mX\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mY\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMT\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, etc? If so change to \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    223\u001b[0m                         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnumbers and try again.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m chrm)\n\u001b[1;32m    225\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m bp \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m x\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mvalues:\n\u001b[1;32m    226\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mColumn \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m not found in \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m data.frame\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m bp)\n",
      "\u001b[0;31mTypeError\u001b[0m: chrom column should be numeric. Do you have 'X', 'Y', 'MT', etc? If so change to numbers and try again."
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from dash import dcc\n",
    "import dash_bio as dashbio\n",
    "import plotly\n",
    "\n",
    "df = pd.read_csv(\"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/61.New_alignment_colias/7.Manhatten/dataset.csv\")\n",
    "\n",
    "# n_chr = 23  # number of chromosome pairs in humans\n",
    "# assert 'CHR' in df.columns\n",
    "# assert df['CHR'].max() == n_chr\n",
    "\n",
    "# Trim down the data\n",
    "# DATASET = df.groupby('CHR').apply(lambda u: u.head(50))\n",
    "# DATASET = DATASET.droplevel('CHR').reset_index(drop=True)\n",
    "\n",
    "manhattanplot = dashbio.ManhattanPlot(\n",
    "    dataframe=df,\n",
    "    chrm = \"chrom\",\n",
    "    bp = \"BP\",\n",
    "    p = \"skane.lulea.fst\",\n",
    "    highlight_color='#00FFAA'\n",
    ")\n",
    "\n",
    "# dcc.Graph(figure=manhattanplot)\n",
    "\n",
    "plotly.offline.plot(manhattanplot, image='png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c47a1166-b8a1-477e-8dc4-5875ac2ab437",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mDEPRECATION: Loading egg at /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages/ete3-3.1.3-py3.10.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330\u001b[0m\u001b[33m\n",
      "\u001b[0mCollecting dash_bio\n",
      "  Downloading dash_bio-1.0.2.tar.gz (10.2 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m10.2/10.2 MB\u001b[0m \u001b[31m8.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m0:01\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting dash>=1.6.1 (from dash_bio)\n",
      "  Downloading dash-3.0.4-py3-none-any.whl.metadata (10 kB)\n",
      "Requirement already satisfied: pandas in /home/saurav/.local/lib/python3.11/site-packages (from dash_bio) (2.1.2)\n",
      "Requirement already satisfied: scipy in /home/saurav/.local/lib/python3.11/site-packages (from dash_bio) (1.14.0)\n",
      "Requirement already satisfied: scikit-learn>=0.20.1 in /home/saurav/.local/lib/python3.11/site-packages (from dash_bio) (1.3.2)\n",
      "Requirement already satisfied: biopython>=1.77 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash_bio) (1.84)\n",
      "Collecting colour (from dash_bio)\n",
      "  Downloading colour-0.1.5-py2.py3-none-any.whl.metadata (18 kB)\n",
      "Collecting GEOparse>=1.1.0 (from dash_bio)\n",
      "  Downloading GEOparse-2.0.4-py3-none-any.whl.metadata (6.5 kB)\n",
      "Requirement already satisfied: jsonschema in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash_bio) (4.22.0)\n",
      "Collecting parmed (from dash_bio)\n",
      "  Downloading parmed-4.3.0.tar.gz (20.2 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m20.2/20.2 MB\u001b[0m \u001b[31m9.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0mm\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting periodictable (from dash_bio)\n",
      "  Downloading periodictable-2.0.2-py3-none-any.whl.metadata (5.4 kB)\n",
      "Requirement already satisfied: requests in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash_bio) (2.28.1)\n",
      "Requirement already satisfied: numpy in /home/saurav/.local/lib/python3.11/site-packages (from biopython>=1.77->dash_bio) (1.26.1)\n",
      "Collecting Flask<3.1,>=1.0.4 (from dash>=1.6.1->dash_bio)\n",
      "  Downloading flask-3.0.3-py3-none-any.whl.metadata (3.2 kB)\n",
      "Collecting Werkzeug<3.1 (from dash>=1.6.1->dash_bio)\n",
      "  Downloading werkzeug-3.0.6-py3-none-any.whl.metadata (3.7 kB)\n",
      "Requirement already satisfied: plotly>=5.0.0 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash>=1.6.1->dash_bio) (6.1.1)\n",
      "Collecting importlib-metadata (from dash>=1.6.1->dash_bio)\n",
      "  Downloading importlib_metadata-8.7.0-py3-none-any.whl.metadata (4.8 kB)\n",
      "Requirement already satisfied: typing-extensions>=4.1.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash>=1.6.1->dash_bio) (4.12.2)\n",
      "Collecting retrying (from dash>=1.6.1->dash_bio)\n",
      "  Downloading retrying-1.3.4-py3-none-any.whl.metadata (6.9 kB)\n",
      "Requirement already satisfied: nest-asyncio in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash>=1.6.1->dash_bio) (1.6.0)\n",
      "Requirement already satisfied: setuptools in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from dash>=1.6.1->dash_bio) (70.1.1)\n",
      "Requirement already satisfied: tqdm>=4.31.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from GEOparse>=1.1.0->dash_bio) (4.66.4)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/saurav/.local/lib/python3.11/site-packages (from pandas->dash_bio) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from pandas->dash_bio) (2022.1)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /home/saurav/.local/lib/python3.11/site-packages (from pandas->dash_bio) (2024.1)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from requests->dash_bio) (2.1.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from requests->dash_bio) (3.3)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from requests->dash_bio) (1.26.20)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from requests->dash_bio) (2022.6.15)\n",
      "Collecting joblib>=1.1.1 (from scikit-learn>=0.20.1->dash_bio)\n",
      "  Downloading joblib-1.5.1-py3-none-any.whl.metadata (5.6 kB)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /home/saurav/.local/lib/python3.11/site-packages (from scikit-learn>=0.20.1->dash_bio) (3.5.0)\n",
      "Requirement already satisfied: attrs>=22.2.0 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from jsonschema->dash_bio) (23.2.0)\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from jsonschema->dash_bio) (2023.12.1)\n",
      "Requirement already satisfied: referencing>=0.28.4 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from jsonschema->dash_bio) (0.35.1)\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from jsonschema->dash_bio) (0.18.1)\n",
      "Requirement already satisfied: pyparsing in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from periodictable->dash_bio) (3.2.1)\n",
      "Requirement already satisfied: Jinja2>=3.1.2 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from Flask<3.1,>=1.0.4->dash>=1.6.1->dash_bio) (3.1.4)\n",
      "Collecting itsdangerous>=2.1.2 (from Flask<3.1,>=1.0.4->dash>=1.6.1->dash_bio)\n",
      "  Downloading itsdangerous-2.2.0-py3-none-any.whl.metadata (1.9 kB)\n",
      "Requirement already satisfied: click>=8.1.3 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from Flask<3.1,>=1.0.4->dash>=1.6.1->dash_bio) (8.1.3)\n",
      "Collecting blinker>=1.6.2 (from Flask<3.1,>=1.0.4->dash>=1.6.1->dash_bio)\n",
      "  Downloading blinker-1.9.0-py3-none-any.whl.metadata (1.6 kB)\n",
      "Requirement already satisfied: narwhals>=1.15.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from plotly>=5.0.0->dash>=1.6.1->dash_bio) (1.40.0)\n",
      "Requirement already satisfied: packaging in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from plotly>=5.0.0->dash>=1.6.1->dash_bio) (23.2)\n",
      "Requirement already satisfied: six>=1.5 in /home/saurav/.local/lib/python3.11/site-packages (from python-dateutil>=2.8.2->pandas->dash_bio) (1.16.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.1.1 in /home/saurav/miniconda3/envs/genomics_work/lib/python3.11/site-packages (from Werkzeug<3.1->dash>=1.6.1->dash_bio) (2.1.5)\n",
      "Collecting zipp>=3.20 (from importlib-metadata->dash>=1.6.1->dash_bio)\n",
      "  Downloading zipp-3.22.0-py3-none-any.whl.metadata (3.6 kB)\n",
      "Downloading dash-3.0.4-py3-none-any.whl (7.9 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m7.9/7.9 MB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hDownloading GEOparse-2.0.4-py3-none-any.whl (29 kB)\n",
      "Downloading colour-0.1.5-py2.py3-none-any.whl (23 kB)\n",
      "Downloading periodictable-2.0.2-py3-none-any.whl (811 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m811.1/811.1 kB\u001b[0m \u001b[31m9.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hDownloading flask-3.0.3-py3-none-any.whl (101 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m101.7/101.7 kB\u001b[0m \u001b[31m7.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading joblib-1.5.1-py3-none-any.whl (307 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m307.7/307.7 kB\u001b[0m \u001b[31m6.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hDownloading werkzeug-3.0.6-py3-none-any.whl (227 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m228.0/228.0 kB\u001b[0m \u001b[31m7.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading importlib_metadata-8.7.0-py3-none-any.whl (27 kB)\n",
      "Downloading retrying-1.3.4-py3-none-any.whl (11 kB)\n",
      "Downloading blinker-1.9.0-py3-none-any.whl (8.5 kB)\n",
      "Downloading itsdangerous-2.2.0-py3-none-any.whl (16 kB)\n",
      "Downloading zipp-3.22.0-py3-none-any.whl (9.8 kB)\n",
      "Building wheels for collected packages: dash_bio, parmed\n",
      "  Building wheel for dash_bio (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for dash_bio: filename=dash_bio-1.0.2-py3-none-any.whl size=10338084 sha256=ba9ac0a654d833f6e79c24587be112954807379a5c3b63300a0356d07c4f925a\n",
      "  Stored in directory: /home/saurav/.cache/pip/wheels/5c/bb/b1/b4d9fe4356712f296a1ffa8c87f6c106ef74b5e6e390ff3f24\n",
      "  Building wheel for parmed (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for parmed: filename=ParmEd-4.3.0-cp311-cp311-linux_x86_64.whl size=19201984 sha256=a4aa030ffc71ec555b17c92e0e7fcf14c9c46c1220a1b46110c330b4fce655a7\n",
      "  Stored in directory: /home/saurav/.cache/pip/wheels/5d/5f/0a/e88e403aba1e99f00bdf228e123bbf9543a881e568079fc065\n",
      "Successfully built dash_bio parmed\n",
      "Installing collected packages: colour, zipp, Werkzeug, retrying, periodictable, parmed, joblib, itsdangerous, blinker, importlib-metadata, Flask, GEOparse, dash, dash_bio\n",
      "  Attempting uninstall: joblib\n",
      "    Found existing installation: joblib 1.1.0\n",
      "    Uninstalling joblib-1.1.0:\n",
      "      Successfully uninstalled joblib-1.1.0\n",
      "Successfully installed Flask-3.0.3 GEOparse-2.0.4 Werkzeug-3.0.6 blinker-1.9.0 colour-0.1.5 dash-3.0.4 dash_bio-1.0.2 importlib-metadata-8.7.0 itsdangerous-2.2.0 joblib-1.5.1 parmed-4.3.0 periodictable-2.0.2 retrying-1.3.4 zipp-3.22.0\n"
     ]
    }
   ],
   "source": [
    "!pip3 install dash_bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "46d4b731-f48b-4175-9b2c-3321bf90e7c3",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"snp argument specified as SNP but column not found in 'x' data.frame\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 6\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mnp\u001b[39;00m\n\u001b[1;32m      3\u001b[0m dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(\n\u001b[1;32m      4\u001b[0m     np\u001b[38;5;241m.\u001b[39mrandom\u001b[38;5;241m.\u001b[39mrandint(\u001b[38;5;241m0\u001b[39m,\u001b[38;5;241m100\u001b[39m,size\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m100\u001b[39m, \u001b[38;5;241m3\u001b[39m)),\n\u001b[1;32m      5\u001b[0m     columns\u001b[38;5;241m=\u001b[39m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mP\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCHR\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mBP\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[0;32m----> 6\u001b[0m fig \u001b[38;5;241m=\u001b[39m \u001b[43mdashbio\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mManhattanPlot\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdataframe\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtitle\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mXYZ Manhattan plot\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      8\u001b[0m plotly\u001b[38;5;241m.\u001b[39moffline\u001b[38;5;241m.\u001b[39mplot(fig, image\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpng\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/miniconda3/envs/genomics_work/lib/python3.11/site-packages/dash_bio/component_factory/_manhattan.py:127\u001b[0m, in \u001b[0;36mManhattanPlot\u001b[0;34m(dataframe, chrm, bp, p, snp, gene, annotation, logp, title, showgrid, xlabel, ylabel, point_size, showlegend, col, suggestiveline_value, suggestiveline_color, suggestiveline_width, genomewideline_value, genomewideline_color, genomewideline_width, highlight, highlight_color)\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mManhattanPlot\u001b[39m(\n\u001b[1;32m     16\u001b[0m         dataframe,\n\u001b[1;32m     17\u001b[0m         chrm\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCHR\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     38\u001b[0m         highlight_color\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mred\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     39\u001b[0m ):\n\u001b[1;32m     40\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Returns a figure for a manhattan plot.\u001b[39;00m\n\u001b[1;32m     41\u001b[0m \n\u001b[1;32m     42\u001b[0m \u001b[38;5;124;03mKeyword arguments:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    124\u001b[0m \n\u001b[1;32m    125\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 127\u001b[0m     mh \u001b[38;5;241m=\u001b[39m \u001b[43m_ManhattanPlot\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    128\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdataframe\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    129\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchrm\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchrm\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    130\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    131\u001b[0m \u001b[43m        \u001b[49m\u001b[43mp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    132\u001b[0m \u001b[43m        \u001b[49m\u001b[43msnp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msnp\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    133\u001b[0m \u001b[43m        \u001b[49m\u001b[43mgene\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mgene\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    134\u001b[0m \u001b[43m        \u001b[49m\u001b[43mannotation\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mannotation\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    135\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlogp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlogp\u001b[49m\n\u001b[1;32m    136\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    138\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m mh\u001b[38;5;241m.\u001b[39mfigure(\n\u001b[1;32m    139\u001b[0m         title\u001b[38;5;241m=\u001b[39mtitle,\n\u001b[1;32m    140\u001b[0m         showgrid\u001b[38;5;241m=\u001b[39mshowgrid,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    153\u001b[0m         highlight_color\u001b[38;5;241m=\u001b[39mhighlight_color\n\u001b[1;32m    154\u001b[0m     )\n",
      "File \u001b[0;32m~/miniconda3/envs/genomics_work/lib/python3.11/site-packages/dash_bio/component_factory/_manhattan.py:243\u001b[0m, in \u001b[0;36m_ManhattanPlot.__init__\u001b[0;34m(self, x, chrm, bp, p, snp, gene, annotation, logp)\u001b[0m\n\u001b[1;32m    240\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m snp \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    241\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m snp \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m x\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mvalues:\n\u001b[1;32m    242\u001b[0m         \u001b[38;5;66;03m# Warn if you don't have a snp column\u001b[39;00m\n\u001b[0;32m--> 243\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\n\u001b[1;32m    244\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msnp argument specified as \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m but column not found in \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    245\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m data.frame\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m snp)\n\u001b[1;32m    246\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    247\u001b[0m         \u001b[38;5;66;03m# If the input DataFrame has a snp column, add it to the new\u001b[39;00m\n\u001b[1;32m    248\u001b[0m         \u001b[38;5;66;03m# DataFrame\u001b[39;00m\n\u001b[1;32m    249\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdata[snp] \u001b[38;5;241m=\u001b[39m x[snp]\n",
      "\u001b[0;31mKeyError\u001b[0m: \"snp argument specified as SNP but column not found in 'x' data.frame\""
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "dataframe = pd.DataFrame(\n",
    "    np.random.randint(0,100,size=(100, 3)),\n",
    "    columns=['P', 'CHR', 'BP'])\n",
    "fig = dashbio.ManhattanPlot(dataframe, title='XYZ Manhattan plot')\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
