{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ea859f2c-4d0b-4928-8b51-45e01ca5cd0e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['clock.fas.best.fas', 'cry1a.fas.best.fas', 'cry1b.fas.best.fas', 'cry2.fas.best.fas', 'cwo.fas.best.fas', 'cycle.fas.best.fas', 'period.fas.best.fas', 'timeless.fas.best.fas', 'timeout.fas.best.fas']\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/clock.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/clock.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.9884\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/clock.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 5694\n",
      "Number of sites kept: 4152\n",
      "Number of sites trimmed: 1542\n",
      "Percentage of alignment trimmed: 27.081%\n",
      "\n",
      "Execution time: 0.166s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/cry1a.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry1a.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.9875\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry1a.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 1674\n",
      "Number of sites kept: 1626\n",
      "Number of sites trimmed: 48\n",
      "Percentage of alignment trimmed: 2.867%\n",
      "\n",
      "Execution time: 0.059s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/cry1b.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry1b.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.9884\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry1b.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 1656\n",
      "Number of sites kept: 1554\n",
      "Number of sites trimmed: 102\n",
      "Percentage of alignment trimmed: 6.159%\n",
      "\n",
      "Execution time: 0.058s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/cry2.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry2.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.8966\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cry2.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 4164\n",
      "Number of sites kept: 3081\n",
      "Number of sites trimmed: 1083\n",
      "Percentage of alignment trimmed: 26.009%\n",
      "\n",
      "Execution time: 0.099s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/cwo.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cwo.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.9767\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cwo.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 2313\n",
      "Number of sites kept: 1839\n",
      "Number of sites trimmed: 474\n",
      "Percentage of alignment trimmed: 20.493%\n",
      "\n",
      "Execution time: 0.058s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/cycle.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cycle.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.8929\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/cycle.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 4398\n",
      "Number of sites kept: 3030\n",
      "Number of sites trimmed: 1368\n",
      "Percentage of alignment trimmed: 31.105%\n",
      "\n",
      "Execution time: 0.091s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/period.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/period.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.8876\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/period.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 7527\n",
      "Number of sites kept: 5121\n",
      "Number of sites trimmed: 2406\n",
      "Percentage of alignment trimmed: 31.965%\n",
      "\n",
      "Execution time: 0.164s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/timeless.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/timeless.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.8824\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/timeless.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 7176\n",
      "Number of sites kept: 4656\n",
      "Number of sites trimmed: 2520\n",
      "Percentage of alignment trimmed: 35.117%\n",
      "\n",
      "Execution time: 0.149s\n",
      "\n",
      "\n",
      "-------------\n",
      "| Arguments |\n",
      "-------------\n",
      "Input file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/8.Merged_aligned/timeout.fas.best.fas (format: fasta)\n",
      "Output file: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/timeout.fas.best.fas.ClipKIT.fas (format: fasta)\n",
      "Sequence type: Nucleotides\n",
      "Gaps threshold: 0.9518\n",
      "Gap characters: ['-', '?', '*', 'X', 'x', 'N', 'n']\n",
      "Trimming mode: smart-gap\n",
      "Create complementary output: False\n",
      "Process as codons: True\n",
      "Create log file: False\n",
      "\n",
      "\n",
      "------------------------\n",
      "| Writing output files |\n",
      "------------------------\n",
      "Trimmed alignment: /mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/9.Clipkit/timeout.fas.best.fas.ClipKIT.fas\n",
      "Complement file: False\n",
      "Log file: False\n",
      "\n",
      "\n",
      "---------------------\n",
      "| Output Statistics |\n",
      "---------------------\n",
      "Original length: 5907\n",
      "Number of sites kept: 4911\n",
      "Number of sites trimmed: 996\n",
      "Percentage of alignment trimmed: 16.861%\n",
      "\n",
      "Execution time: 0.128s\n",
      "\n"
     ]
    }
   ],
   "source": [
    "location = \"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias\"\n",
    "\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "list_of_files = os.listdir(f\"{location}/8.Merged_aligned\")\n",
    "if \"desktop.ini\" in list_of_files:\n",
    "    list_of_files.remove(\"desktop.ini\")\n",
    "\n",
    "print(list_of_files)\n",
    "\n",
    "for i, file_name in enumerate(list_of_files):\n",
    "    # print(i)\n",
    "\tsubprocess.run(f\"clipkit '{location}/8.Merged_aligned/{file_name}' --codon -o  '{location}/9.Clipkit/{file_name}.ClipKIT.fas'\", shell = True)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c8398e7-1cef-42ce-ae79-c084e3621d63",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
