{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e614fa45-2a27-4518-894c-0d1b28464d33",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_anno_dic = {}\n",
    "with open(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/20.RERconverge/2.Modified_diapause/correlation_with_annotation.csv\", 'r') as cor_file:\n",
    "    for lines in cor_file.readlines()[1:]:\n",
    "        gene_name = lines.split(\",\")[0].split(\"_\")[0]\n",
    "        # print(gene_name)\n",
    "        anno_name = lines.split(\",\")[-1].replace('\"', '')\n",
    "        gene_anno_dic[gene_name]= anno_name.strip()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d5e12e00-4dd5-4656-a6d9-19ddde6d6529",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2905at7088 per\n",
      "12510at7088 CHMP1B\n",
      "13263at7088 DSTN\n",
      "16035at7088 ATP6V1G1\n",
      "2560at7088 LETM1\n",
      "3165at7088 TH\n",
      "3362at7088 NOP14\n",
      "3811at7088 LPCAT3\n",
      "4043at7088 ACAT1\n",
      "4842at7088 HDAC2\n",
      "5326at7088 NPFFR2\n",
      "5423at7088 QRSL1\n",
      "5474at7088 CLOCK\n",
      "5679at7088 5679at7088\n",
      "6081at7088 6081at7088\n",
      "8044at7088 Met\n",
      "8751at7088 8751at7088\n",
      "8809at7088 ENAH\n",
      "10001at7088 NTHL1\n",
      "1390at7088 USP2\n",
      "1422at7088 tim\n",
      "1709at7088 CSDE1\n",
      "1846at7088 SLC6A19\n",
      "2036at7088 LINS\n"
     ]
    }
   ],
   "source": [
    "with open(\"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/61.New_alignment_colias/6.GSEA_rho/3.Positive_modi/gene_names.txt\", 'r') as gene_name_file:\n",
    "    for lines in gene_name_file.readlines()[1:]:\n",
    "        print(lines.strip(), gene_anno_dic[lines.strip()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d08d724-b1a8-42d4-86bf-3c563fae4b6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "location = \"/mnt/griffin/saubar/busco_runs_CDS_files/BUSCO_GCF_947172395.1_ilBicAnyn1.1_genomic.fna/run_lepidoptera_odb10/busco_sequences/single_copy_busco_sequences\"\n",
    "\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "list_of_files = os.listdir(location)\n",
    "output = \"\"\n",
    "for i,file_name in enumerate(list_of_files):\n",
    "    print(i, end = \",\")\n",
    "    if file_name.endswith(\".faa\"):\n",
    "        fasta_file = SeqIO.parse(f\"{location}/{file_name}\", 'fasta')\n",
    "        for records in fasta_file:\n",
    "            output += f\">{records.id}\\n{records.seq}\\n\\n\"\n",
    "\n",
    "with open(\"/mnt/griffin/saubar/busco_runs_CDS_files/BUSCO_GCF_947172395.1_ilBicAnyn1.1_genomic.fna/just_aa.fas\", 'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
