{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "ee2e0de5-6752-4f3f-bad1-affa1b5370f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "fasta_files = SeqIO.to_dict(SeqIO.parse(\"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/5.Genome/1.Colias_eurytheme/1.Genome/Ce_fixed_chromosome_assembly.fa\", 'fasta'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "03109c17-be1c-4380-b211-9c76430fdfe4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cry2_query.fas\n",
      "ATGTCAGCTGCTGCAGAGACGCTGCCGGCGTCCAGCTCGCGTAATCCTGGCCCTGCACCTACTTCTCCCACTCCCCCAACTCGTCAACCAGCTCTCAAACACACAGTGCATTGGTTTCGAAAAGGTTTACGTCTCCACGATAATCCCGCTTTACGAGAGGGATTAGTTGATGCAGTCACTTTTCGATGCGTGTTTATAATTGATCCGTGGTTCGCAAGTTCCTCGAATGTCGGTATTAATAAATGGAG\n",
      "0\n",
      "ATTCCTATTACAATGTCTGGAAGATTTGGATAGTAGCTAAGAAAAATTAAACTCTCGCCTTTTCGTGGTACGCGGACAGCCTGCAGATGCGTTACCTAAACTATTTCGTGAGTGGGGCACAACGGCGCTCACTTTCGAAGAGGATCCTGAGCCCTATGGCAGGGTCCGGGATCATATATTATGACGAAATGTCGCGAAGGTCGGTATAACCGTTATTTCGCGCGTTTCACATACTTTATACAAACTTGACCA\n",
      "1\n",
      "1.Colias_eurytheme,CHR3,8073425,8073676,0,N,Colias_croceus_XM_045647888.1_Frame_1_rightoh_2_query_Exon_2,1,83,83\n",
      " ID: CHR3\n",
      "Name: CHR3\n",
      "Description: CHR3\n",
      "Number of features: 0\n",
      "Seq('TT') Donor ERROR\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[49], line 74\u001b[0m\n\u001b[1;32m     72\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m (donor\u001b[38;5;241m.\u001b[39mseq\u001b[38;5;241m.\u001b[39mupper() \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGT\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;129;01mand\u001b[39;00m (donor\u001b[38;5;241m.\u001b[39mseq\u001b[38;5;241m.\u001b[39mupper() \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGC\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m     73\u001b[0m         \u001b[38;5;28mprint\u001b[39m(lines, donor,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mDonor\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mERROR\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 74\u001b[0m         \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m     75\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m complement \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m1\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m     76\u001b[0m     donor \u001b[38;5;241m=\u001b[39m fasta_files[scaffold][\u001b[38;5;28mint\u001b[39m(start)\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m3\u001b[39m:\u001b[38;5;28mint\u001b[39m(start) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "location  = \"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/5.Genome\"\n",
    "\n",
    "query_location = f\"{location}/4.Queries\"\n",
    "query_transcript_list = os.listdir(f\"{query_location}\")\n",
    "\n",
    "species = \"1.Colias_eurytheme\"\n",
    "query_transcript_list = [\"cry2_query.fas\"]\n",
    "for query_transcript in query_transcript_list:\n",
    "    print(query_transcript)\n",
    "    gene_name = query_transcript.split(\"_\")[0]\n",
    "    # print(gene_name)\n",
    "    # assert False\n",
    "    output = \"\"\n",
    "    full_sequence = \"\"\n",
    "    \n",
    "    with open(f\"{location}/{species}/{gene_name}_coordinates_old.csv\",'r') as coordinate_file:\n",
    "        coordinate_list = (coordinate_file.readlines()[1:])\n",
    "    seq_dictionary_exon = {}\n",
    "    for i,lines in enumerate(coordinate_list):\n",
    "        lines_split = lines.strip().split(\",\")\n",
    "        exon = lines_split[6].split(\"Exon_\")[1]\n",
    "        # print(exon)\n",
    "        # assert False\n",
    "        scaffold,start, stop, complement = lines_split[1],lines_split[2],lines_split[3],lines_split[4]\n",
    "        if gene_name == \"clock111\" and exon == \"3\":\n",
    "            sequence.seq = \"AAATAACCGTAAGGTCTCGCGCTCATGATGTACAAGAGGATTGGAAACCTGCTTTTCTATCAAATGAGGAATTCACTTATCTTGTTTTGGAGGCCTTGGAGGGTTTTGTTATGGTTTTCTCGGCAACTGGACGAATTTACTACGTATCTGAGGGAATAACCTCATTACTTGGACACAATCcg\".upper()\n",
    "            complement = \"0\"\n",
    "            # print(\"here\")\n",
    "            # assert False\n",
    "            # pass\n",
    "        elif gene_name == \"clock111\" and exon == \"10\":\n",
    "            sequence.seq = \"ccGCCGTCCCTGAAGTCCGCGTCGCACACGCGCTACTCGACCGGCTACGAGAACAGCATGGTGTCGGCTGACTCGCGTTCCTCGCAGCGCTCCAGCTCGAGAgac\".upper()\n",
    "            complement = \"0\"\n",
    "        else:\n",
    "            sequence = fasta_files[scaffold][int(start)-1:int(stop)]\n",
    "            print(sequence.seq)\n",
    "            print(i)\n",
    "            if i == 0:\n",
    "                if complement == \"0\":\n",
    "                    donor = fasta_files[scaffold][int(stop):int(stop) +2]\n",
    "                    if donor.seq.upper() != \"GT\":\n",
    "                        print(lines, \"Donor\", \"ERROR\")\n",
    "                        assert False\n",
    "                if complement == \"1\":\n",
    "                    donor = fasta_files[scaffold][int(start)-3:int(start) -1]\n",
    "                    donor = donor.seq.reverse_complement()\n",
    "                    if donor.upper() != \"GT\":\n",
    "                        print(lines, \"Donor\", \"ERROR\")\n",
    "                        assert False\n",
    "    \n",
    "            elif i == len(coordinate_list)-1:\n",
    "                # print(lines)\n",
    "                if complement == \"0\":\n",
    "                    acceptor = fasta_files[scaffold][int(start)-3:int(start) -1]\n",
    "                    if acceptor.seq.upper() != \"AG\":\n",
    "                        print(lines, \"Acceptor\", \"ERROR\")\n",
    "                        assert False\n",
    "                if complement == \"1\":\n",
    "                    acceptor = fasta_files[scaffold][int(stop):int(stop) +2]\n",
    "                    acceptor = acceptor.seq.reverse_complement()\n",
    "                    if acceptor.upper() != \"AG\":\n",
    "                        print(lines, \"Acceptor\", \"ERROR\")\n",
    "                        assert False\n",
    "            else:\n",
    "                if complement == \"0\":\n",
    "                    donor = fasta_files[scaffold][int(stop):int(stop) +2]\n",
    "                    acceptor = fasta_files[scaffold][int(start)-3:int(start) -1]\n",
    "                    if acceptor.seq.upper() != \"AG\":\n",
    "                        print(lines, acceptor,\"Acceptor\", \"ERROR\")\n",
    "                        assert False\n",
    "                    if (donor.seq.upper() != \"GT\") and (donor.seq.upper() != \"GC\"):\n",
    "                        print(lines, donor,\"Donor\", \"ERROR\")\n",
    "                        assert False\n",
    "                if complement == \"1\":\n",
    "                    donor = fasta_files[scaffold][int(start)-3:int(start) -1]\n",
    "                    donor = donor.seq.reverse_complement()\n",
    "                    if donor.upper() != \"GT\":\n",
    "                        print(lines, donor,\"Donor\", \"ERROR\")\n",
    "                        assert False\n",
    "                    acceptor = fasta_files[scaffold][int(stop):int(stop) +2]\n",
    "                    acceptor = acceptor.seq.reverse_complement()\n",
    "                    if acceptor.upper() != \"AG\":\n",
    "                        print(lines, acceptor,\"Acceptor\", \"ERROR\")\n",
    "                        assert False\n",
    "        if complement == \"1\":\n",
    "            full_sequence += sequence.seq.reverse_complement()\n",
    "            # print(sequence.seq.reverse_complement())\n",
    "        else:\n",
    "            full_sequence += sequence.seq\n",
    "            # print(sequence.seq)   \n",
    "        # print(donor, acceptor)\n",
    "        # print(sequence.seq)\n",
    "    with open(f\"/mnt/c/Users/sauba/Desktop/20.RERConverge_trimmed/3.New_colias/6.Sequences/{gene_name}_{species}.fas\", \"w\") as out_file:\n",
    "        output = f\">{species}__{gene_name}\\n{full_sequence}\\n\"\n",
    "        out_file.write(output)\n",
    "    print(full_sequence.translate())\n",
    "    assert False\n",
    "        # output_file.write(Output_Sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6d9ec99-1900-4d2c-a75a-095f4d3facbb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
