{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0d321c49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_folder_error_exon(output_location,species, error_exon):\n",
    "    import subprocess\n",
    "    list_of_folders = os.listdir(f\"{output_location}/1.Blast_result/{species}\")\n",
    "    if \"Error_exon_processing\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing\")\n",
    "        \n",
    "    list_of_folder_2 = os.listdir(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing\")\n",
    "    if error_exon in list_of_folder_2:\n",
    "        subprocess.run(f'rm -r \"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\")\n",
    "    \n",
    "    list_of_folders_3 = os.listdir(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\")\n",
    "    if \"Temp_query\" not in list_of_folders_3:\n",
    "        os.mkdir(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}/Temp_query\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8456dd2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_query(output_location,error_exon,species):\n",
    "    from Bio import SeqIO\n",
    "    import random\n",
    "    import os\n",
    "    exon = error_exon.split(\"Error\")[1][1:]\n",
    "    print(exon)\n",
    "#     print(\"Here Here\")\n",
    "    initial_list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    \n",
    "    \n",
    "    try:\n",
    "        initial_list_of_species.remove(\"desktop.ini\")\n",
    "    except:\n",
    "        pass\n",
    "    list_of_species = []\n",
    "#     print(initial_list_of_species)\n",
    "    for species_name in initial_list_of_species:\n",
    "        list_of_files_in_species_folder = os.listdir(f\"{output_location}/2.Final_output/{species_name}\")\n",
    "#         print(list_of_files_in_species_folder)\n",
    "#         print(f\"{exon}.fa\")\n",
    "        if f\"{exon}.fa\" in os.listdir(f\"{output_location}/2.Final_output/{species_name}\"):\n",
    "            list_of_species.append(species_name)\n",
    "    \n",
    "    \n",
    "    if len(list_of_species) > 6:\n",
    "        list_of_species = random.sample(list_of_species, 5)\n",
    "    \n",
    "    for query_species in list_of_species:\n",
    "        print(\"current_query :\", query_species)\n",
    "        list_of_exons = os.listdir(f\"{output_location}/2.Final_output/{query_species}\")\n",
    "        try:\n",
    "            list_of_exons.remove(\"desktop.ini\")\n",
    "        except:\n",
    "            pass\n",
    "#         print(query_species,list_of_exons)\n",
    "        if f\"{exon}.fa\" in list_of_exons:\n",
    "            print (f\"{exon}.fa\")\n",
    "            exon_file = SeqIO.parse(f\"{output_location}/2.Final_output/{query_species}/{exon}.fa\", 'fasta')\n",
    "            for records in exon_file:\n",
    "                print(records.id)\n",
    "                print(records.seq[int(records.id.split(\"_\")[-3]):].translate())\n",
    "                left_oh = records.id.split(\"_\")[-3]\n",
    "                right_oh = records.id.split(\"_\")[-1]\n",
    "    #             print(left_oh)\n",
    "                query_sequence = records.seq[int(records.id.split(\"_\")[-3]):].translate()\n",
    "            with open(f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}/Temp_query/{query_species}_{exon}.fa\", \"w\") as out_file:\n",
    "                output = f\">{records.id}\\n{query_sequence}\"\n",
    "                out_file.write(output)\n",
    "    return(left_oh,right_oh)\n",
    "# left_oh,right_oh = make_query(output_location,error_exon,species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b7aff1cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genomic_coordinates(output_location,error_exon,species):\n",
    "    error_exon_number = error_exon.split(\"_\")[-1]\n",
    "#     print(error_exon_number)\n",
    "#     return\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as final_coordinate_file:\n",
    "        final_coordinate_file_lines = final_coordinate_file.readlines()\n",
    "    for i in range(1, len(final_coordinate_file_lines)):\n",
    "#         print(final_coordinate_file_lines[i])\n",
    "        current_position = i\n",
    "        next_position = i\n",
    "#         print(final_coordinate_file_lines[i].split(\",\")[6].split(\"Exon\")[1][1:].strip())\n",
    "#         print(error_exon_number)\n",
    "        scaffold = final_coordinate_file_lines[current_position-1].split(\",\")[1]\n",
    "\n",
    "        original_query_name =f\"{'_'.join(final_coordinate_file_lines[current_position-1].split(',')[6].split('_')[:-1])}_{current_position}\"\n",
    "        \n",
    "        \n",
    "        if final_coordinate_file_lines[i].split(\",\")[6].split(\"Exon\")[1][1:].strip() == error_exon_number:\n",
    "#             print(current_position, next_position)\n",
    "    \n",
    "            if error_exon_number == \"1\":\n",
    "                current_position = current_position\n",
    "                while final_coordinate_file_lines[next_position+1].split(\",\")[5] == \"Y\":\n",
    "                        next_position += 1\n",
    "                complement = final_coordinate_file_lines[current_position].split(\",\")[4]\n",
    "                \n",
    "                if complement == \"0\":\n",
    "                    fragment_start = int(final_coordinate_file_lines[current_position].split(\",\")[3])-1000\n",
    "                    fragment_end = final_coordinate_file_lines[next_position+1].split(\",\")[2]\n",
    "                if complement == \"1\":\n",
    "                    fragment_start = final_coordinate_file_lines[next_position+1].split(\",\")[3]\n",
    "                    fragment_end = int(final_coordinate_file_lines[current_position].split(\",\")[2]) + 1000\n",
    "\n",
    "                scaffold = final_coordinate_file_lines[current_position].split(\",\")[1]\n",
    "                \n",
    "                \n",
    "                \n",
    "            elif error_exon_number == \"16\":\n",
    "                next_position = next_position\n",
    "                while final_coordinate_file_lines[current_position-1].split(\",\")[5] == \"Y\":\n",
    "                    current_position = current_position-1\n",
    "                complement = final_coordinate_file_lines[current_position].split(\",\")[4]\n",
    "                \n",
    "                if complement == \"0\":\n",
    "                    fragment_start = int(final_coordinate_file_lines[current_position-1].split(\",\")[3])\n",
    "                    fragment_end = int(final_coordinate_file_lines[next_position].split(\",\")[2])+1000\n",
    "                if complement == \"1\":\n",
    "                    fragment_start = int(final_coordinate_file_lines[next_position].split(\",\")[3])-1000\n",
    "                    fragment_end = final_coordinate_file_lines[current_position-1].split(\",\")[2] \n",
    "                \n",
    "                scaffold = final_coordinate_file_lines[current_position].split(\",\")[1]\n",
    "                original_query_name =f\"{'_'.join(final_coordinate_file_lines[current_position].split(',')[6].split('_')[:-1])}_{current_position}\"\n",
    "                \n",
    "            else: \n",
    "                while final_coordinate_file_lines[current_position-1].split(\",\")[5] == \"Y\":\n",
    "                    current_position = current_position-1\n",
    "                while final_coordinate_file_lines[next_position+1].split(\",\")[5] == \"Y\":\n",
    "                    next_position += 1\n",
    "\n",
    "                complement = final_coordinate_file_lines[current_position-1].split(\",\")[4]\n",
    "                original_query_name =f\"{'_'.join(final_coordinate_file_lines[current_position-1].split(',')[6].split('_')[:-1])}_{current_position}\"\n",
    "\n",
    "                if complement == \"0\":\n",
    "                    fragment_start = final_coordinate_file_lines[current_position-1].split(\",\")[3]\n",
    "                    fragment_end = final_coordinate_file_lines[next_position+1].split(\",\")[2]\n",
    "                if complement == \"1\":\n",
    "                    fragment_start = final_coordinate_file_lines[next_position+1].split(\",\")[3]\n",
    "                    fragment_end = final_coordinate_file_lines[current_position-1].split(\",\")[2]\n",
    "            \n",
    "                scaffold = final_coordinate_file_lines[current_position-1].split(\",\")[1]\n",
    "\n",
    "                original_query_name =f\"{'_'.join(final_coordinate_file_lines[current_position-1].split(',')[6].split('_')[:-1])}_{current_position}\"\n",
    "                \n",
    "    #             query_name =f\"{''.join(final_coordinate_file_lines[current_position-1].split(''))}\"\n",
    "    #             print(query_name)                       \n",
    "                                   \n",
    "            return(fragment_start, fragment_end, scaffold,complement,original_query_name)\n",
    "# print(get_genomic_coordinates(output_location,error_exon,species))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "09eaa662",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file) \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species,                       \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    print(list_of_folders)\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "# get_gene_sequence(genome_location, \n",
    "#                           species,                       \n",
    "#                           scaffold,\n",
    "#                           fragment_start,\n",
    "#                           fragment_end,\n",
    "#                           complement,\n",
    "#                           output_location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f36f8190",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,output_location,species,error_exon):\n",
    "    \n",
    "    \n",
    "    error_exon_location = f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"\n",
    "    list_of_folders = os.listdir(error_exon_location)\n",
    "    if \"for_alignment\" not in list_of_folders:\n",
    "        \n",
    "        os.mkdir(f\"{error_exon_location}/for_alignment\")\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{error_exon_location}/for_alignment\")\n",
    "    for file in list_of_files:\n",
    "        os.remove(f\"{error_exon_location}/for_alignment/{file}\")\n",
    "\n",
    "    \n",
    "    query_location = f\"{error_exon_location}/Temp_query/\"\n",
    "    list_of_queries = os.listdir(query_location)\n",
    "    print(list_of_queries)\n",
    "#     return\n",
    "    query_fasta_sequence = ''\n",
    "    query_length = 0\n",
    "    \n",
    "    for query in list_of_queries:\n",
    "        \n",
    "        query_fasta_file = SeqIO.parse(f\"{query_location}/{query}\", 'fasta')\n",
    "        for records in query_fasta_file:\n",
    "            \n",
    "            query_fasta_sequence = f\">{records.id}\\n{records.seq}\\n\\n\"\n",
    "            query_length_new = len(records.seq)\n",
    "            if query_length == 0:\n",
    "                query_length = len(records.seq)\n",
    "            if query_length_new < query_length:\n",
    "                query_length = query_length_new\n",
    "                \n",
    "            query_species = \"_\".join(records.id.split(\"_\")[:2])\n",
    "#             print(query_species)\n",
    "#             return\n",
    "            print(error_exon)\n",
    "        #     return\n",
    "\n",
    "        #     query_length = query_length/3\n",
    "            for offset in range(3):\n",
    "                translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "                for i in range(len(translated_sequence)):\n",
    "\n",
    "                    if len(translated_sequence[i])> 0.8*query_length and  str(translated_sequence[i]).count(\"X\") < 5:\n",
    "                        sequence_set = f\">set{i+1}_frame{offset}_{query_species}\\n{translated_sequence[i]}\\n\\n\"\n",
    "        #                 print(i+1, offset)\n",
    "\n",
    "                        with open(f\"{error_exon_location}/for_alignment/{error_exon}_query_{query_species}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                            output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                            out_file.write(output)\n",
    "\n",
    "# make_raw_files_for_alignment(gene_sequence,output_location,species,error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "730727e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_mafft(output_location,species,error_exon):\n",
    "    error_exon_location = f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"\n",
    "    location = f'{error_exon_location}/for_alignment/'\n",
    "    list_of_files_to_run_mafft_on = os.listdir(location)\n",
    "    for file in list_of_files_to_run_mafft_on:\n",
    "        if file.endswith(\".fa\"):\n",
    "            command = f'\"mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "            # print(command)\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            # os.system(f'{command}')\n",
    "            command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/{file}\" > \"{location}/alignment_clustal_{file}.txt\"'\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            \n",
    "            # os.system(f'{command}')\n",
    "    return(location)\n",
    "# run_mafft(output_location,species,error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a78e0d08",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def process_mafft_output(mafft_run_location, error_exon):\n",
    "    error_exon = error_exon.split(\"Error\")[1][1:]\n",
    "    list_of_files_in_mafft_run_folder = os.listdir(mafft_run_location)\n",
    "    score_output = []\n",
    "    score = 99\n",
    "    min_score_sequence = ''\n",
    "    alignment_file = ''\n",
    "    for file in list_of_files_in_mafft_run_folder:\n",
    "        if file.endswith(\".fa.hat2\"):\n",
    "            with open(f\"{mafft_run_location}/{file}\", 'r') as dist_matrix_file:\n",
    "                dist_matrix_list = dist_matrix_file.readlines()\n",
    "            \n",
    "            sequence_name = f'set{file.split(\"_\")[-2]}_{file.split(\"_\")[-1].split(\".\")[0]}'\n",
    "            query_species_name =f'{file.split(\"_\")[4]}_{file.split(\"_\")[5]}'\n",
    "            distance_score = float(dist_matrix_list[2].strip())\n",
    "            if len(score_output) < 10:\n",
    "                alignment_file_now = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "                score_output.append( [sequence_name,distance_score,alignment_file_now])\n",
    "            else:\n",
    "                for i in range(len(score_output)):\n",
    "                    score_at_this_index = score_output[i][1]\n",
    "                    if distance_score < score_at_this_index:\n",
    "                        alignment_file_now = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "                        score_output[i] = [sequence_name,distance_score,alignment_file_now]\n",
    "                        break\n",
    "            if distance_score < score:\n",
    "                score = distance_score\n",
    "                min_score_sequence = sequence_name\n",
    "                alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "#     print(score_output)\n",
    "    print(f\"min = {min_score_sequence}, {score}\" )\n",
    "    \n",
    "    # score_out_merged = '\\n'.join(score_output)\n",
    "    print(f\"5 top scores:\\n{score_output}\")\n",
    "    \n",
    "    for i in range(len(score_output)):\n",
    "        temp_align_file = score_output[i][2]\n",
    "        clustal_alignment_file = temp_align_file.replace(\"alignment_\",\"alignment_clustal_\")\n",
    "        with open(f\"{mafft_run_location}{clustal_alignment_file}\", 'r') as clustal_file_open:\n",
    "            print(\"\".join(clustal_file_open.readlines()))\n",
    "    \n",
    "    print(f\"{mafft_run_location}/{alignment_file}\")\n",
    "#     return\n",
    "    \n",
    "    print(f\"Alignmnet file: {alignment_file}\")\n",
    "    clustal_alignment_file = alignment_file.replace(\"alignment_\",\"alignment_clustal_\")\n",
    "    print(clustal_alignment_file)\n",
    "    alignment_file = SeqIO.parse(f\"{mafft_run_location}/{alignment_file}\", 'fasta')\n",
    "    # print (records.id)\n",
    "    fasta_start_position = 0\n",
    "    fasta_end_position = 0\n",
    "    start_switch = 0\n",
    "    end_switch = 0\n",
    "    alignment_name = ''\n",
    "    for records in alignment_file:\n",
    "        \n",
    "        if start_switch == 0 and end_switch == 0:\n",
    "            print(error_exon)\n",
    "            if error_exon in records.id:\n",
    "                print(records.id)\n",
    "                gap_counter = 0\n",
    "                base_counter = 0\n",
    "                for current_position in range(len(records.seq)):\n",
    "                    sequence_length = len(records.seq) - records.seq.count('-')\n",
    "#                     print(f\"fasta_end_position {fasta_end_position} fasta_start_position {fasta_start_position}\")\n",
    "                    # print(f\"current_position = {current_position}, {len(records.seq)}\")\n",
    "                    # print(records.seq[current_position])\n",
    "                    # print(\"fasta_start_position\",fasta_start_position)\n",
    "#                     print(\"start_switch\",start_switch)\n",
    "\n",
    "                    # print(gap_counter, base_counter)\n",
    "                    # print(5,0.2*sequence_length)\n",
    "#                     print((len(records.seq[:current_position]) - records.seq[:current_position].count('-')), 0.1*sequence_length)\n",
    "                    if start_switch == 1 and records.seq[current_position] == \"-\" and gap_counter > 3 and (len(records.seq[:current_position]) - records.seq[:current_position].count('-')) < 0.1*sequence_length:\n",
    "                        # print(\"\\n\\nhere\\n\\n\")\n",
    "                        start_switch = 0\n",
    "                        gap_counter = 0\n",
    "                    \n",
    "                    if \"-\" not in records.seq[current_position] and start_switch == 0 :\n",
    "                        \n",
    "                        fasta_start_position = current_position\n",
    "                        start_switch = 1\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "                    if end_switch == 1 and \"-\" not in records.seq[current_position] :\n",
    "                        end_switch = 0\n",
    "                        gap_counter = 0\n",
    "\n",
    "#                     print((len(records.seq[current_position:]) - records.seq[current_position:].count('-')), 0.1*sequence_length) \n",
    "#                     print((start_switch == 1),records.seq[current_position],(len(records.seq[current_position:]) - records.seq[current_position:].count('-')) < 0.1*sequence_length)\n",
    "                    if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0 and (len(records.seq[current_position:]) - records.seq[current_position:].count('-')) < 0.1*sequence_length:\n",
    "                        # print(f\"base_counter {base_counter}\")\n",
    "                        # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "\n",
    "                        # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "\n",
    "                        fasta_end_position = current_position\n",
    "                        end_switch = 1\n",
    "                    if \"-\" in records.seq[current_position]:\n",
    "                        gap_counter += 1\n",
    "                    else:\n",
    "                        gap_counter = 0\n",
    "                        base_counter += 1\n",
    "        else:\n",
    "            if end_switch == 0:\n",
    "                end_switch = 1\n",
    "                fasta_end_position = current_position\n",
    "            start_switch = 1\n",
    "    #         break\n",
    "        print(\"here here\", start_switch,end_switch,min_score_sequence,records.id,fasta_start_position, fasta_end_position )\n",
    "        if start_switch == 1 and end_switch == 1 and min_score_sequence in records.id :\n",
    "            alignment_name = records.id\n",
    "#             print(fasta_start_position, fasta_end_position)\n",
    "            gene_sequence = records.seq[fasta_start_position:fasta_end_position]\n",
    "            print(f\"{records.id}\\n{gene_sequence}\")\n",
    "#     print(\"_\".join(alignment_name.split(\"_\")[:2]).replace(\"set\", \"sequence\"))\n",
    "#     clustal_alignment_file = f'alignment_clustal_Error_{error_exon}_translated_genomic_{(\"_\".join(alignment_name.split(\"_\")[:2]).replace(\"set\", \"sequence_\"))}.fa.txt'\n",
    "    print(clustal_alignment_file)\n",
    "    with open(f\"{mafft_run_location}{clustal_alignment_file}\", 'r') as clustal_file_open:\n",
    "        return(gene_sequence, alignment_name,clustal_file_open.readlines() )\n",
    "\n",
    "# query_sequence, alignment_name, alignment_file = process_mafft_output(mafft_run_location, error_exon)\n",
    "# print(query_sequence, alignment_name)\n",
    "# print(\"\".join(alignment_file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e78ed650",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_blast(gene_sequence,query_sequence,alignment_name,output_location,species, error_exon):\n",
    "    error_exon_location = f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"\n",
    "    list_of_folders = os.listdir(error_exon_location)\n",
    "    if \"Run_Blast\" not in list_of_folders:\n",
    "        os.mkdir(f\"{error_exon_location}/Run_Blast\")\n",
    "    \n",
    "    with open(f\"{error_exon_location}/Run_Blast/local_db.txt\",'w') as db_file:\n",
    "        output = f\">genome_fragment\\n{gene_sequence}\"\n",
    "        db_file.write(output)\n",
    "        \n",
    "    with open(f\"{error_exon_location}/Run_Blast/query.txt\",'w') as query_file:\n",
    "        output = f\">{alignment_name}\\n{query_sequence}\"\n",
    "        query_file.write(output)\n",
    "    \n",
    "    makeblast_command = f'cd \"{error_exon_location}/Run_Blast/\"\\nmakeblastdb -in local_db.txt -dbtype nucl'\n",
    "    subprocess.run(f'{makeblast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    blast_command = f'cd \"{error_exon_location}/Run_Blast/\"\\ntblastn -seg no -query query.txt -db local_db.txt -num_alignments 3 -out blast_out_genome_fragment.htm -html\\ntblastn -seg no -query query.txt -db local_db.txt -num_alignments 3 -out blast_out_genome_fragment.txt'\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "# run_blast(gene_sequence,query_sequence,alignment_name,output_location,species, error_exon)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ae0d3e81",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def process_genome_fragment_blast_file(output_location,\n",
    "#                                         annotated_genome_location, \n",
    "#                                        species_name,\n",
    "#                                        error_exon,\n",
    "#                                        left_oh,right_oh,\n",
    "                                                                                                                                      \n",
    "#                                        original_query_name,\n",
    "# #                                       query_species_original ):\n",
    "    \n",
    "def process_genome_fragment_blast_file(output_location,left_oh,right_oh,species,genome_location):\n",
    "    error_exon_location = f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"\n",
    "    blast_location = f\"{error_exon_location}/Run_Blast\"\n",
    "    \n",
    "    with open(f\"{blast_location}/query.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_oh),int(right_oh)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "#     species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        print(query_name)\n",
    "#         return\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out_genome_fragment.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        print(start_coor,stop_coor)\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "        genome_file = SeqIO.parse(f\"{blast_location}/local_db.txt\", 'fasta')\n",
    "        print(\"reached here\")\n",
    "        \n",
    "        for records in genome_file:\n",
    "            old_start = start\n",
    "            old_stop = stop\n",
    "            ag = \"N\"\n",
    "            gt = \"N\"\n",
    "            stop_counter = 0\n",
    "            while True:\n",
    "                print(f\"sequence:\\n{records.seq[start+start_modifier-1:stop]}\")\n",
    "                translated_sequence = records.seq[start+start_modifier-1:stop].translate()\n",
    "                print(f\"sequence:\\n{translated_sequence}\")\n",
    "                if \"*\" in translated_sequence:\n",
    "                    stop_counter +=1\n",
    "                    if ag == \"N\":\n",
    "                        start = old_start + 3*stop_counter\n",
    "                    if gt ==\"N\":\n",
    "                        stop = old_stop - 3*stop_counter\n",
    "                if stop_counter > 90:\n",
    "                    return (\"Error!! Too many stops\")\n",
    "                print(f\"left = {records.seq[start-3:start-1]}, right ={(records.seq[stop:stop + 2])}\"  ), \n",
    "                if (records.seq[start-3:start-1]).lower() == \"ag\" and ag != \"Y\":\n",
    "                    \n",
    "                    ag = \"Y\"\n",
    "                    \n",
    "                elif ag != \"Y\":\n",
    "                    start -= 3\n",
    "                \n",
    "                if (records.seq[stop:stop + 2]).lower() == \"gt\" and gt != \"Y\":\n",
    "                    gt = \"Y\"\n",
    "                elif gt != \"Y\":\n",
    "                    stop +=3\n",
    "                if old_start - start > 1000 or stop - old_stop > 1000:\n",
    "                    break\n",
    "                if gt == \"Y\" and ag == \"Y\":\n",
    "                    gt_ag = \"Y\"\n",
    "                    break\n",
    "        \n",
    "        query_location = f\"{blast_location}/new_query_spliced.txt\"\n",
    "        with open(query_location , 'w') as query_file_new:\n",
    "\n",
    "            sequence_translated = records.seq[start+start_modifier-1:stop].translate()\n",
    "            print(sequence_translated)\n",
    "            # proceed_test = input(\"Proceed with this?\")\n",
    "            # while True:\n",
    "            #     if proceed_test.lower()[0] == \"n\":\n",
    "            #         assert False\n",
    "            #     elif proceed_test.lower()[0] == \"y\":\n",
    "            #         break\n",
    "            if \"*\" in sequence_translated:\n",
    "                print(\"Errror in Spliced query\")\n",
    "                assert False\n",
    "            \n",
    "            output = f\">{query_name}\\n{sequence_translated}\"\n",
    "            query_file_new.write(output)\n",
    "\n",
    "        genome_file = get_genome_file(genome_location,species)\n",
    "        genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "        out_location = f\"{blast_location}\"\n",
    "    \n",
    "        \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.htm\" -html'\n",
    "        # print(blast_command)\n",
    "        # subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.txt\"'\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "\n",
    "                \n",
    "        # acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name)\n",
    "\n",
    "#         if acceptor == \"Y\" and donor  == \"Y\":\n",
    "#             splice_prediction = \"Y\"\n",
    "#         else:\n",
    "#             splice_prediction = \"N\"\n",
    "            \n",
    "        return(gt_ag) \n",
    "        # output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaffold) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        # print(output_format)\n",
    "\n",
    "# ag_gt = process_genome_fragment_blast_file(output_location,left_oh,right_oh,species,genome_location)\n",
    "# print(ag_gt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "fbf3f985",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, ag_gt, splice_prediction,original_query_name, query_species ): ):\n",
    "def process_genome_blast_file(output_location,left_oh,right_oh,species,genome_location,ag_gt,query_name_original):\n",
    "    error_exon_location = f\"{output_location}/1.Blast_result/{species}/Error_exon_processing/{error_exon}\"\n",
    "    blast_location = f\"{error_exon_location}/Run_Blast\"\n",
    "\n",
    "    with open(f\"{blast_location}/new_query_spliced.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_oh),int(right_oh)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "    \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = species\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "            # print(lines)\n",
    "            # print(query_name)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        print(f\"start_coordinate : {start_coor},stop_coordinate : {stop_coor}\")\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "        splice_prediction = \"Y\"   \n",
    "        output_format = str(species_name)+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name_original)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \",\" + ag_gt + \",\" + splice_prediction +\"\\n\"  \n",
    "        # print(Output_Sequence)\n",
    "        return(output_format)\n",
    "# process_genome_blast_file(output_location,left_oh,right_oh,species,genome_location,ag_gt,query_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e71b515e",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brenthis_daphne,OX352476.1,000,000,0,Y,Error_Exon_15,00,00,00\n",
      "\n",
      "Exon_15\n",
      "current_query : Mechanitis_messenoides\n",
      "Exon_15.fa\n",
      "Mechanitis_messenoides_Exon_15_OY365759.1_13946136_13946264_left_1_right_2\n",
      "SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "current_query : Fabriciana_adippe\n",
      "Exon_15.fa\n",
      "Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2\n",
      "TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQAELEYFNN\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current_query : Mechanitis_mazaeus\n",
      "Exon_15.fa\n",
      "Mechanitis_mazaeus_Exon_15_OY365750.1_5004849_5004977_left_1_right_2\n",
      "SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "current_query : Vanessa_cardui\n",
      "Exon_15.fa\n",
      "Vanessa_cardui_Exon_15_NC_061129.1_7377675_7377866_left_1_right_2\n",
      "SIIPVEEDTMKDLSPQEITHVPLEVSQLGTLESSSLILPTIMSVKQETSAFSTRLTDSDYYNN\n",
      "current_query : Nymphalis_polychloros\n",
      "Exon_15.fa\n",
      "Nymphalis_polychloros_Exon_15_HG992244.1_7305544_7305741_left_1_right_2\n",
      "SIIPIEENTKKDSSPEEITHVPLHMSSQQLGPLESSSLILPTIMSVKQEASAFNTPLTDPDYYND\n",
      "Gene_start = 22694297, Gene_end = 22698602\n",
      "Getting Gene\n",
      "GCA_947093115.1_ES_BD_1141.v2_0_genomic.fna\n",
      "['1.Query', '1.Blast_result', 'tblastn_Danaus_plexippus.sh', 'temp', '2.Final_output', '0.Temp', 'error_exons.txt', 'desktop.ini']\n",
      "['Mechanitis_messenoides_Exon_15.fa', 'Fabriciana_adippe_Exon_15.fa', 'Mechanitis_mazaeus_Exon_15.fa', 'Vanessa_cardui_Exon_15.fa', 'Nymphalis_polychloros_Exon_15.fa']\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "min = set20_frame2, 1.699\n",
      "5 top scores:\n",
      "[['set20_frame2', 1.699, 'alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_20_frame2.fa.txt'], ['set16_frame2', 4.227, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_16_frame2.fa.txt'], ['set91_frame1', 4.328, 'alignment_Error_Exon_15_query_Nymphalis_polychloros_translated_genomic_sequence_91_frame1.fa.txt'], ['set11_frame0', 4.325, 'alignment_Error_Exon_15_query_Mechanitis_messenoides_translated_genomic_sequence_11_frame0.fa.txt'], ['set82_frame2', 4.374, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_82_frame2.fa.txt'], ['set91_frame2', 4.387, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_91_frame2.fa.txt'], ['set29_frame0', 4.366, 'alignment_Error_Exon_15_query_Mechanitis_messenoides_translated_genomic_sequence_29_frame0.fa.txt'], ['set21_frame2', 4.414, 'alignment_Error_Exon_15_query_Vanessa_cardui_translated_genomic_sequence_21_frame2.fa.txt'], ['set53_frame0', 4.432, 'alignment_Error_Exon_15_query_Vanessa_cardui_translated_genomic_sequence_53_frame0.fa.txt'], ['set60_frame0', 4.445, 'alignment_Error_Exon_15_query_Vanessa_cardui_translated_genomic_sequence_60_frame0.fa.txt']]\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Fabriciana_adip ------------------TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQA\n",
      "set20_frame2_Fa ESDLHFILYLFKINKISRTIIAENDELSKKPNKDNSKLPMLPQVSLPAGVQQLPSFIYQD\n",
      "                                  ****:.**:.******:..:***** ****.*:* ***  * \n",
      "\n",
      "Fabriciana_adip ELEYFNN-----------------------------------------------------\n",
      "set20_frame2_Fa ELECYNNRFELFHLFFFLFFSSKVINDFVLRSKKYLRPVRQDDCLNFKISISRSIIDNYG\n",
      "                *** :**                                                     \n",
      "\n",
      "Fabriciana_adip ---------------------\n",
      "set20_frame2_Fa MLRELIIRFIISVFFLIFYSQ\n",
      "                                     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza --------------------------SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGF\n",
      "set16_frame2_Me CRRQCSSFLGFRTRRVPTNPIIKSPLSVCLSEGCIKTAIRRKLK-----FLQSV------\n",
      "                                          *:  .* *::**  . *.      *:**      \n",
      "\n",
      "Mechanitis_maza VNNDSEYN\n",
      "set16_frame2_Me ------YF\n",
      "                      * \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Nymphalis_polyc SIIPIEENTKKDSSPEEITHVPLHMSSQQLGPLESSSLILPTIMSVKQEASAFNTPLTDP\n",
      "set91_frame1_Ny GVSVVKSCVRF-----------YHLCGN----------------------------VVSP\n",
      "                .:  ::. .:             *:..:                            :..*\n",
      "\n",
      "Nymphalis_polyc DYYND----------\n",
      "set91_frame1_Ny DFHSDRDRVHLLAYY\n",
      "                *::.*          \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_mess ------------SIITEEVCVRTAAVEALSPVLLPSLE--------SVKQDAGFVNNDSE\n",
      "set11_frame0_Me LKMTSYQRSQIRTIQNCQCCRKCPSPRASSSYLLSSTKMNWNVIITGLSYFICFFFYFSA\n",
      "                            :* . : * : .: .* *. **.* :        .:.    *.   * \n",
      "\n",
      "Mechanitis_mess YN\n",
      "set11_frame0_Me VK\n",
      "                 :\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza S------------------IITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "set82_frame2_Me EFLPSIRVLALLNPASDFIIYVEMLSHRIFIVTGIECTCWPTTKSIVL----------FP\n",
      "                .                  * .* :. *   * .:. .  *: :*:            : \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza SIITEE------------------VCVRTAAVEALSPVL--LPSLES---VKQDAGFVNN\n",
      "set91_frame2_Me NVPSDEPERSEGEPLVNNNLTIIIICIHISVCLLLSIILFIIPTITTCLTVKEN--IVKK\n",
      "                .: ::*                  :*:: :.   ** :*  :*:: :   **::  :*::\n",
      "\n",
      "Mechanitis_maza DS------------------------EYN\n",
      "set91_frame2_Me PACLRELYKFSKVMCSPPTRTWPAWWTMA\n",
      "                 :                           \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_mess SIITEEVCVRTAAVEALSPVLLPS------LESVKQDAGFVNNDSEYN\n",
      "set29_frame0_Me KINLTLIC-----QVALELLILPSSFHNHIIYGLKTRA--LNN---LL\n",
      "                .*    :*       **. ::***      : .:*  *  :**     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Vanessa_cardui_ SIIPVEEDTMKDLSPQEITHVPLEVSQLGTLESSSLILPTIMSVKQETSAFSTRLTDSDY\n",
      "set21_frame2_Va N--------------------------------------RILAIKKVSEIHGAYISFTVF\n",
      "                .                                       *:::*: :. ..: :: : :\n",
      "\n",
      "Vanessa_cardui_ YN---------N\n",
      "set21_frame2_Va FSIFIHWNFFFI\n",
      "                :.          \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Vanessa_cardui_ SIIPVEEDTMKDLSPQEITHVPLEVSQLGTLESSSLILPTIMSVKQETSAFSTRLTDSDY\n",
      "set53_frame0_Va C---------------------------GNRQTGK---------NSKNYRFGLCIDNNVY\n",
      "                .                           *. ::..         :.:.  *.  : :. *\n",
      "\n",
      "Vanessa_cardui_ YNN-------------------\n",
      "set53_frame0_Va YNNIFPKIFNVQIGLVTILLYV\n",
      "                ***                   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Vanessa_cardui_ SIIPVEEDTMKDLSPQEITHVPLEVSQL-----GTLESSSLILPTIMSVKQETSAFSTRL\n",
      "set60_frame0_Va ------------------CKSPIQIFKVLKTGRGLDESCTIWIDSTSRMDQK--------\n",
      "                                   : *::: ::     *  **.:: : :   :.*:        \n",
      "\n",
      "Vanessa_cardui_ TDSDYYNN\n",
      "set60_frame0_Va --------\n",
      "                        \n",
      "\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_daphne/Error_exon_processing/Error_Exon_15/for_alignment//alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_20_frame2.fa.txt\n",
      "Alignmnet file: alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_20_frame2.fa.txt\n",
      "alignment_clustal_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_20_frame2.fa.txt\n",
      "Exon_15\n",
      "Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2\n",
      "here here 1 1 set20_frame2 Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2 18 67\n",
      "here here 1 1 set20_frame2 set20_frame2_Fabriciana_adippe 18 67\n",
      "set20_frame2_Fabriciana_adippe\n",
      "TIIAENDELSKKPNKDNSKLPMLPQVSLPAGVQQLPSFIYQDELECYNN\n",
      "alignment_clustal_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_20_frame2.fa.txt\n",
      "Alignment File: CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Fabriciana_adip ------------------TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQA\n",
      "set20_frame2_Fa ESDLHFILYLFKINKISRTIIAENDELSKKPNKDNSKLPMLPQVSLPAGVQQLPSFIYQD\n",
      "                                  ****:.**:.******:..:***** ****.*:* ***  * \n",
      "\n",
      "Fabriciana_adip ELEYFNN-----------------------------------------------------\n",
      "set20_frame2_Fa ELECYNNRFELFHLFFFLFFSSKVINDFVLRSKKYLRPVRQDDCLNFKISISRSIIDNYG\n",
      "                *** :**                                                     \n",
      "\n",
      "Fabriciana_adip ---------------------\n",
      "set20_frame2_Fa MLRELIIRFIISVFFLIFYSQ\n",
      "                                     \n",
      "\n",
      "Query OK?y\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/11/2024 06:09:04\n",
      "New DB name:   /mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_daphne/Error_exon_processing/Error_Exon_15/Run_Blast/local_db.txt\n",
      "New DB title:  local_db.txt\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.00775909 seconds.\n",
      "[[1, 2]]\n",
      "set20_frame2_Fabriciana_adippe\n",
      "645 791\n",
      "reached here\n",
      "sequence:\n",
      "aCCATAATAGCTGAAAATGACGAGTTATCAAAGAAGCCAAATAAGGACAATTCAAAATTGCCAATGCTGCCGCAAGTGTCCCTCCCCGCGGGCGTCCAGCAGTTACCTTCTTTCATCTACCAAGATGAATTGGAATGTTATAATAACAG\n",
      "sequence:\n",
      "TIIAENDELSKKPNKDNSKLPMLPQVSLPAGVQQLPSFIYQDELECYNN\n",
      "left = ag, right =GT\n",
      "TIIAENDELSKKPNKDNSKLPMLPQVSLPAGVQQLPSFIYQDELECYNN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "start_coordinate : 22697958,stop_coordinate : 22697812\n",
      "Brenthis_daphne,OX352476.1,22697810,22697959,1,N,Danaus_plexippus_XM_032654785.2_Frame_1_rightoh_2_query_Exon_15,1,49,49,Y,Y\n",
      "\n",
      "Fix Overhang!! Proceed?y\n",
      "Brenthis_hecate,OX638268.1,000,000,0,Y,Error_Exon_15,00,00,00\n",
      "\n",
      "Exon_15\n",
      "current_query : Limenitis_camilla\n",
      "Exon_15.fa\n",
      "Limenitis_camilla_Exon_15_LR990227.1_8793248_8793424_left_1_right_2\n",
      "PIITDAESVPTTSCLSKSNVPLPMPDVSSLILPTILNIDQKPKSFDNNISTESDCCNN\n",
      "current_query : Fabriciana_adippe\n",
      "Exon_15.fa\n",
      "Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2\n",
      "TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQAELEYFNN\n",
      "current_query : Hypolimnas_misippus\n",
      "Exon_15.fa\n",
      "Hypolimnas_misippus_Exon_15_SGBE01000384.1_8316073_8316270_left_1_right_2\n",
      "SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVKQETSNFILGQADSEYFNN\n",
      "current_query : Eueides_isabella\n",
      "Exon_15.fa\n",
      "Eueides_isabella_Exon_15_CM032384.1_9631882_9632067_left_1_right_2\n",
      "SIIAEDENVSKKLTKDKPILPILPQVSIPSAPSLLLPTIMSVKQEPTAFAINQAESEYYNN\n",
      "current_query : Melinaea_marsaeus_rileyi\n",
      "Exon_15.fa\n",
      "Melinaea_marsaeus_rileyi_Exon_15_OU911089.1_25515051_25515158_left_1_right_2\n",
      "SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "Gene_start = 300637, Gene_end = 310169\n",
      "Getting Gene\n",
      "GCA_951806755.1_ES_BH_1412.v2_0_genomic.fna\n",
      "['1.Query', '1.Blast_result', 'tblastn_Danaus_plexippus.sh', 'temp', '2.Final_output', '0.Temp', 'error_exons.txt', 'desktop.ini']\n",
      "['Limenitis_camilla_Exon_15.fa', 'Fabriciana_adippe_Exon_15.fa', 'Hypolimnas_misippus_Exon_15.fa', 'Eueides_isabella_Exon_15.fa', 'Melinaea_marsaeus_rileyi_Exon_15.fa']\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "min = set16_frame2, 1.972\n",
      "5 top scores:\n",
      "[['set16_frame2', 1.972, 'alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_16_frame2.fa.txt'], ['set16_frame2', 3.374, 'alignment_Error_Exon_15_query_Eueides_isabella_translated_genomic_sequence_16_frame2.fa.txt'], ['set160_frame2', 4.164, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_160_frame2.fa.txt'], ['set150_frame2', 4.211, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_150_frame2.fa.txt'], ['set146_frame2', 4.23, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_146_frame2.fa.txt'], ['set152_frame2', 4.279, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_152_frame2.fa.txt'], ['set204_frame2', 4.352, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_204_frame2.fa.txt'], ['set51_frame2', 4.383, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_51_frame2.fa.txt'], ['set148_frame2', 4.383, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_148_frame2.fa.txt'], ['set19_frame2', 4.387, 'alignment_Error_Exon_15_query_Melinaea_marsaeus_translated_genomic_sequence_19_frame2.fa.txt']]\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Fabriciana_adip ------TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQAELEYFNN-----\n",
      "set16_frame2_Fa VNKISRTIIAENEELPKKPNKDNPKLPM--QVSLPAGVQQLPSFINQDELECYNNRFELF\n",
      "                      ****:.:*: ******:*.:**  * ****.*:* *** ** *** :**     \n",
      "\n",
      "Fabriciana_adip ------------------------------------------------------------\n",
      "set16_frame2_Fa HLFFFLFFSKVIKDFVLRSKKYLRPVRQDDCLNFKMSISRSIIDNYGMLRELIIRFIISV\n",
      "                                                                            \n",
      "\n",
      "Fabriciana_adip --\n",
      "set16_frame2_Fa FF\n",
      "                  \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Eueides_isabell ------SIIAEDENVSKKLTKDKPILPILPQVSIPSAPSLLLPTIMSVKQEPTAFAINQA\n",
      "set16_frame2_Eu VNKISRTIIAENEELPKKPNKDNPKLPM--QVSLPAG----------VQQLPS--FINQD\n",
      "                      :****:*::.** .**:* **:  ***:*:.          *:* *:   *** \n",
      "\n",
      "Eueides_isabell ESEYYNN-----------------------------------------------------\n",
      "set16_frame2_Eu ELECYNNRFELFHLFFFLFFSKVIKDFVLRSKKYLRPVRQDDCLNFKMSISRSIIDNYGM\n",
      "                * * ***                                                     \n",
      "\n",
      "Eueides_isabell --------------\n",
      "set16_frame2_Eu LRELIIRFIISVFF\n",
      "                              \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae SILTE------EVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "set160_frame2_M --YTELALGIFATKASLSPKTNATPPSLKET--------VG\n",
      "                   **       . .* * :: * **: . *         .\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae S---------------ILTEEVCVSTSVEAPALPPAKDGTGDAHND-----------SEY\n",
      "set150_frame2_M AAGGGGADGASLDLWDFLTGCPCGDSSSERVLLMERSDSGFEASNRYPIALSSAKVLADS\n",
      "                :               :**   * .:* *   *   .*.  :* *            :: \n",
      "\n",
      "Melinaea_marsae N\n",
      "set150_frame2_M S\n",
      "                .\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae SILT----------------EEVCVSTSVEAPALPPAK-------DGTGDAHNDSEY---\n",
      "set146_frame2_M GNLCGVPVVPLVPATGATPGAAVPAGEGAAALALAPAAAAPEREVDGTGP---EAEVPGL\n",
      "                . *                   * .. .. * **.**        ****    ::*    \n",
      "\n",
      "Melinaea_marsae ----------N\n",
      "set146_frame2_M GVGWLMPLIRN\n",
      "                          *\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae SILTEEVCVS-------------------------TSVEAPALPPAKDGTGDAHNDSEYN\n",
      "set152_frame2_M ----EEVAAGLPVGDSGLVEAPIAISVPAVTSIGCADIWAVSSAPIVSGMRSWHNSSRVC\n",
      "                    ***...                         :.: * : .*  .*  . **.*.  \n",
      "\n",
      "Melinaea_marsae ----------------\n",
      "set152_frame2_M DVVTSFLFLLFWSLLI\n",
      "                                \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae SILTEEVCVS------------------TSVEAPALPP--AKDGTGDAHNDSEYN\n",
      "set204_frame2_M RKTCEETCMSEGTLQVFESHVQSANPHLASVMDYGLIPLVSKGGPCQVLG----R\n",
      "                    **.*:*                  :**   .* *  :*.*. :. .    .\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae SILTEEVCVS---------TSVEAPALPPAKDGTGDA-HNDSEYN\n",
      "set51_frame2_Me A-----CCFAIPQNGSGPHSGVSAPLLANASAFSFPSILSDPGPI\n",
      "                :      *.:         :.*.** *. *.  :  :  .*.   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae -----------------------------------SILTEEVCVSTSVEA----------\n",
      "set148_frame2_M AFRSVFWRVSTDWSSMERVSVIPPFSAFLLRTFHSSVLRPSFCIRRQESAWSTSLDSGSG\n",
      "                                                   *:*  ..*:  . .*          \n",
      "\n",
      "Melinaea_marsae ----PALPPAKD----------GTG-------DAH----------------NDSEYN---\n",
      "set148_frame2_M MYTTPTFGLGKSLSLMRVSLAPGHGCRVGIQFSSHSLAYWLIQVMSSQPGLNDSPPKAGP\n",
      "                    *::  .*.          * *       .:*                ***  :   \n",
      "\n",
      "Melinaea_marsae --------------\n",
      "set148_frame2_M LPFSEVVILATIAL\n",
      "                              \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_marsae -----------------SIL---------------------------------TEEVCVS\n",
      "set19_frame2_Me KKISEIHGAYISFTVFFSIFIHWKLFFFNTKMRLKVSAHCTVSYHDRATNVSGTARTTMP\n",
      "                                 **:                                 * .. :.\n",
      "\n",
      "Melinaea_marsae TSVEAPALPPAKDGTGDAHND--------------SEYN\n",
      "set19_frame2_Me TRTYQQRIIPVHHGFSAAYDHHCAAIVTAPCATRGAAGR\n",
      "                * .    : *.:.* . *::.              :  .\n",
      "\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_hecate/Error_exon_processing/Error_Exon_15/for_alignment//alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_16_frame2.fa.txt\n",
      "Alignmnet file: alignment_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_16_frame2.fa.txt\n",
      "alignment_clustal_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_16_frame2.fa.txt\n",
      "Exon_15\n",
      "Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2\n",
      "here here 1 1 set16_frame2 Fabriciana_adippe_Exon_15_FR989986.1_10037911_10038060_left_1_right_2 6 55\n",
      "here here 1 1 set16_frame2 set16_frame2_Fabriciana_adippe 6 55\n",
      "set16_frame2_Fabriciana_adippe\n",
      "TIIAENEELPKKPNKDNPKLPM--QVSLPAGVQQLPSFINQDELECYNN\n",
      "alignment_clustal_Error_Exon_15_query_Fabriciana_adippe_translated_genomic_sequence_16_frame2.fa.txt\n",
      "Alignment File: CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Fabriciana_adip ------TIIAKSDEFKKKPNKDKPTIPMLPQESLPASVKQEPSFTNQAELEYFNN-----\n",
      "set16_frame2_Fa VNKISRTIIAENEELPKKPNKDNPKLPM--QVSLPAGVQQLPSFINQDELECYNNRFELF\n",
      "                      ****:.:*: ******:*.:**  * ****.*:* *** ** *** :**     \n",
      "\n",
      "Fabriciana_adip ------------------------------------------------------------\n",
      "set16_frame2_Fa HLFFFLFFSKVIKDFVLRSKKYLRPVRQDDCLNFKMSISRSIIDNYGMLRELIIRFIISV\n",
      "                                                                            \n",
      "\n",
      "Fabriciana_adip --\n",
      "set16_frame2_Fa FF\n",
      "                  \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query OK?y\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/11/2024 06:28:24\n",
      "New DB name:   /mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_hecate/Error_exon_processing/Error_Exon_15/Run_Blast/local_db.txt\n",
      "New DB title:  local_db.txt\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0107799 seconds.\n",
      "[[1, 2]]\n",
      "set16_frame2_Fabriciana_adippe\n",
      "669 809\n",
      "reached here\n",
      "sequence:\n",
      "aCCATAATAGCTGAAAATGAAGAGTTACCAAAGAAGCCAAATAAGGACAATCCAAAATTGCCAATGCAAGTGTCCCTCCCCGCGGGCGTCCAGCAGTTACCTTCTTTCATCAACCAAGATGAATTGGAATGTTATAATAACAG\n",
      "sequence:\n",
      "TIIAENEELPKKPNKDNPKLPMQVSLPAGVQQLPSFINQDELECYNN\n",
      "left = ag, right =GT\n",
      "TIIAENEELPKKPNKDNPKLPMQVSLPAGVQQLPSFINQDELECYNN\n",
      "[[1, 2]]\n",
      "start_coordinate : 309501,stop_coordinate : 309361\n",
      "Brenthis_hecate,OX638268.1,309359,309502,1,N,Danaus_plexippus_XM_032654785.2_Frame_1_rightoh_2_query_Exon_15,1,47,47,Y,Y\n",
      "\n",
      "Fix Overhang!! Proceed?y\n",
      "Brenthis_ino,OV170226.1,000,000,0,Y,Error_Exon_15,00,00,00\n",
      "\n",
      "Exon_15\n",
      "current_query : Napeogenes_sylphis\n",
      "Exon_15.fa\n",
      "Napeogenes_sylphis_Exon_15_OY365786.1_24697014_24697157_left_1_right_2\n",
      "SIITEEVCVATAAVEAPAHALPPALLPALEAAKQDVAGFVNNDSEYN\n",
      "current_query : Dryadula_phaetusa\n",
      "Exon_15.fa\n",
      "Dryadula_phaetusa_Exon_15_JAOYRS010000003.1_7325634_7325816_left_1_right_2\n",
      "SIIAEEENCTKKVNKEKPLPISPQVSIPSAPSLLLPTIMSVKHEPTAFAVNQAESEYYNN\n",
      "current_query : Vanessa_cardui\n",
      "Exon_15.fa\n",
      "Vanessa_cardui_Exon_15_NC_061129.1_7377675_7377866_left_1_right_2\n",
      "SIIPVEEDTMKDLSPQEITHVPLEVSQLGTLESSSLILPTIMSVKQETSAFSTRLTDSDYYNN\n",
      "current_query : Mechanitis_mazaeus\n",
      "Exon_15.fa\n",
      "Mechanitis_mazaeus_Exon_15_OY365750.1_5004849_5004977_left_1_right_2\n",
      "SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "current_query : Melinaea_menophilus\n",
      "Exon_15.fa\n",
      "Melinaea_menophilus_Exon_15_OU911048.1_12888007_12888114_left_1_right_2\n",
      "SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "Gene_start = 7909279, Gene_end = 7915749\n",
      "Getting Gene\n",
      "GCA_921882275.1_SP_BI_364.v2_0_genomic.fna\n",
      "['1.Query', '1.Blast_result', 'tblastn_Danaus_plexippus.sh', 'temp', '2.Final_output', '0.Temp', 'error_exons.txt', 'desktop.ini']\n",
      "['Napeogenes_sylphis_Exon_15.fa', 'Dryadula_phaetusa_Exon_15.fa', 'Vanessa_cardui_Exon_15.fa', 'Mechanitis_mazaeus_Exon_15.fa', 'Melinaea_menophilus_Exon_15.fa']\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "min = set56_frame2, 3.442\n",
      "5 top scores:\n",
      "[['set56_frame2', 3.442, 'alignment_Error_Exon_15_query_Dryadula_phaetusa_translated_genomic_sequence_56_frame2.fa.txt'], ['set40_frame1', 4.148, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_40_frame1.fa.txt'], ['set54_frame2', 4.205, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_54_frame2.fa.txt'], ['set121_frame2', 4.213, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_121_frame2.fa.txt'], ['set126_frame2', 4.284, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_126_frame2.fa.txt'], ['set37_frame2', 4.336, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_37_frame2.fa.txt'], ['set107_frame2', 4.347, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_107_frame2.fa.txt'], ['set58_frame1', 4.418, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_58_frame1.fa.txt'], ['set32_frame2', 4.424, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_32_frame2.fa.txt'], ['set17_frame2', 4.434, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_17_frame2.fa.txt']]\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Dryadula_phaetu ------------------------------SIIAEEENCTKKVNKEKP-LPISPQVSIPS\n",
      "set56_frame2_Dr KNRLNGTEPFVCESDLHFILYLFKINKISRTIIAENEELPKKPNKDNPKLPMLPQVSLPA\n",
      "                                              :****:*: .** **::* **: ****:*:\n",
      "\n",
      "Dryadula_phaetu APSLLLPTIMSVKHEPTAFAVNQAESEYYNN---------------\n",
      "set56_frame2_Dr ----------SVQQLPS--FIYQDELECYNNRFELFHLFFFSIFQQ\n",
      "                          **:: *:   : * * * ***               \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza ------------------------------------------------------------\n",
      "set40_frame1_Me GTFCMYVCMLYSALFNVPLLGTGLASVRGGSGNSPPRWPSAGWWTSHIFRKLARFCQTCR\n",
      "                                                                            \n",
      "\n",
      "Mechanitis_maza ---SIITEEVCVRTAAVEALSPVLLPSLE---------------------SVKQD-----\n",
      "set40_frame1_Me FSFTVQTSGNCVRNDSHMKISEVRDPGFELASLGFPSERVTTGIPLPRHLLLKENLPFPR\n",
      "                   :: *.  ***. :   :* *  *.:*                      :*::     \n",
      "\n",
      "Mechanitis_maza ----AGFVNNDSEY--------------------N\n",
      "set40_frame1_Me DNTITGFIESSLETFCWLSLKTYIHLNNILKNCRS\n",
      "                    :**::.. *                     .\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph --SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "set54_frame2_Me RIEIFTE--CVSSAALA----TNNKKTKMAKFKKAAM\n",
      "                  .*:**  ***::. *    . :. *  *: ..   \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph ----------------------SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "set121_frame2_M EFLPSIGVLALLNPASDFTIYVEMLSHRIFIVKGIECACWPTTKSIV-------LFP\n",
      "                                      .:*:..: : ..:*... *.:*. .        : \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph S------------ILT---------------------EEVCVSTSV----------EAPA\n",
      "set126_frame2_M SDYYYNLYTCKCMFITIYHTVYHSNYYHLSYGEGKHREETCMSEGILQVFESHVQSANPH\n",
      "                *            ::*                     **.*:* .:            * \n",
      "\n",
      "Melinaea_menoph LPPAKDGTGDAHNDSEY----------------N\n",
      "set126_frame2_M LASVVD----------YGLISLVSKGGPCQVLGG\n",
      "                *... *          *                .\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph SI---------LTEEVCVSTSVEAPAL----PPAKDGTGDAHNDSEYN\n",
      "set37_frame2_Me KIVGLEKSFRKILNSICLSSKLHSQYLFLSIPIIKLSQIDVRV-SRKK\n",
      "                .*         : :.:*:*:.:.:  *    *  * .  *.:  *. :\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph SILTEE----------VCVSTSVEAPALPPAKDGTGDAHNDSE--------YN\n",
      "set107_frame2_M CGNRQTGKNSKNYRFCLCIDNNVYCNNIFPKKI-------NVQIGLVTILLYV\n",
      "                .   :           :*:...* .  : * *        : :        * \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph ------------SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEY--N\n",
      "set58_frame1_Me LFGISYPKGANGTLLLSLRCPSVCQRAVSKP-------------QYVEN\n",
      "                            ::* .  * *.. .* : *             :*  *\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph SILTEEV----CVSTSVEAPALPPAKDGTGDAHNDSEY-----------N\n",
      "set32_frame2_Me KYQRCETRDLNLLHSVSQASALP-----LGYHCLDTFYLKKIFLSHGIIQ\n",
      "                .    *.     : :  :*.***      *    *: *           :\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph SIL-------------TEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN---\n",
      "set17_frame2_Me SDLQSKSEKKRSNENVSYERCL--------------------NDERYMIMF\n",
      "                * *             : * *:                    **..*    \n",
      "\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_ino/Error_exon_processing/Error_Exon_15/for_alignment//alignment_Error_Exon_15_query_Dryadula_phaetusa_translated_genomic_sequence_56_frame2.fa.txt\n",
      "Alignmnet file: alignment_Error_Exon_15_query_Dryadula_phaetusa_translated_genomic_sequence_56_frame2.fa.txt\n",
      "alignment_clustal_Error_Exon_15_query_Dryadula_phaetusa_translated_genomic_sequence_56_frame2.fa.txt\n",
      "Exon_15\n",
      "Dryadula_phaetusa_Exon_15_JAOYRS010000003.1_7325634_7325816_left_1_right_2\n",
      "here here 1 1 set56_frame2 Dryadula_phaetusa_Exon_15_JAOYRS010000003.1_7325634_7325816_left_1_right_2 30 91\n",
      "here here 1 1 set56_frame2 set56_frame2_Dryadula_phaetusa 30 91\n",
      "set56_frame2_Dryadula_phaetusa\n",
      "TIIAENEELPKKPNKDNPKLPMLPQVSLPA----------SVQQLPS--FIYQDELECYNN\n",
      "alignment_clustal_Error_Exon_15_query_Dryadula_phaetusa_translated_genomic_sequence_56_frame2.fa.txt\n",
      "Alignment File: CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Dryadula_phaetu ------------------------------SIIAEEENCTKKVNKEKP-LPISPQVSIPS\n",
      "set56_frame2_Dr KNRLNGTEPFVCESDLHFILYLFKINKISRTIIAENEELPKKPNKDNPKLPMLPQVSLPA\n",
      "                                              :****:*: .** **::* **: ****:*:\n",
      "\n",
      "Dryadula_phaetu APSLLLPTIMSVKHEPTAFAVNQAESEYYNN---------------\n",
      "set56_frame2_Dr ----------SVQQLPS--FIYQDELECYNNRFELFHLFFFSIFQQ\n",
      "                          **:: *:   : * * * ***               \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query OK?y\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/11/2024 06:49:02\n",
      "New DB name:   /mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Brenthis_ino/Error_exon_processing/Error_Exon_15/Run_Blast/local_db.txt\n",
      "New DB title:  local_db.txt\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.00137305 seconds.\n",
      "[[1, 2]]\n",
      "set56_frame2_Dryadula_phaetusa\n",
      "3207 3353\n",
      "reached here\n",
      "sequence:\n",
      "aCCATAATAGCTGAAAATGAAGAGTTACCAAAGAAGCCAAATAAGGACAATCCAAAATTGCCAATGCTGCCGCAAGTGTCCCTCCCCGCGAGCGTCCAGCAGTTACCTTCTTTCATCTACCAAGATGAATTGGAATGCTATAATAACAG\n",
      "sequence:\n",
      "TIIAENEELPKKPNKDNPKLPMLPQVSLPASVQQLPSFIYQDELECYNN\n",
      "left = ag, right =GT\n",
      "TIIAENEELPKKPNKDNPKLPMLPQVSLPASVQQLPSFIYQDELECYNN\n",
      "[[1, 2]]\n",
      "start_coordinate : 7912485,stop_coordinate : 7912631\n",
      "Brenthis_ino,OV170226.1,7912484,7912633,0,N,Danaus_plexippus_XM_032654785.2_Frame_1_rightoh_2_query_Exon_15,1,49,49,Y,Y\n",
      "\n",
      "Fix Overhang!! Proceed?y\n",
      "Melitaea_cinxia,NC_059396.1,000,000,0,Y,Error_Exon_15,00,00,00\n",
      "\n",
      "Exon_15\n",
      "current_query : Argynnis_bischoffii_washingtonia\n",
      "Exon_15.fa\n",
      "Argynnis_bischoffii_washingtonia_Exon_15_CM066609.1_9671129_9671278_left_1_right_2\n",
      "TIIAESEEFKKKPNKDKPTLPMLPQQSLPAIFKQEPSFTNQAELEYSNN\n",
      "current_query : Vanessa_cardui\n",
      "Exon_15.fa\n",
      "Vanessa_cardui_Exon_15_NC_061129.1_7377675_7377866_left_1_right_2\n",
      "SIIPVEEDTMKDLSPQEITHVPLEVSQLGTLESSSLILPTIMSVKQETSAFSTRLTDSDYYNN\n",
      "current_query : Melinaea_menophilus\n",
      "Exon_15.fa\n",
      "Melinaea_menophilus_Exon_15_OU911048.1_12888007_12888114_left_1_right_2\n",
      "SILTEEVCVSTSVEAPALPPAKDGTGDAHNDSEYN\n",
      "current_query : Mechanitis_mazaeus\n",
      "Exon_15.fa\n",
      "Mechanitis_mazaeus_Exon_15_OY365750.1_5004849_5004977_left_1_right_2\n",
      "SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "current_query : Boloria_selene\n",
      "Exon_15.fa\n",
      "Boloria_selene_Exon_15_HG993134.1_8456739_8456906_left_1_right_2\n",
      "TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSVQEEPSYSNRDDSESYNN\n",
      "Gene_start = 9522588, Gene_end = 9528464\n",
      "Getting Gene\n",
      "GCF_905220565.1_ilMelCinx1.1_genomic.fna\n",
      "['1.Query', '1.Blast_result', 'tblastn_Danaus_plexippus.sh', 'temp', '2.Final_output', '0.Temp', 'error_exons.txt', 'desktop.ini']\n",
      "['Argynnis_bischoffii_washingtonia_Exon_15.fa', 'Vanessa_cardui_Exon_15.fa', 'Melinaea_menophilus_Exon_15.fa', 'Mechanitis_mazaeus_Exon_15.fa', 'Boloria_selene_Exon_15.fa']\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "min = set92_frame2, 3.523\n",
      "5 top scores:\n",
      "[['set92_frame2', 3.523, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_92_frame2.fa.txt'], ['set19_frame2', 4.163, 'alignment_Error_Exon_15_query_Melinaea_menophilus_translated_genomic_sequence_19_frame2.fa.txt'], ['set28_frame0', 4.167, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_28_frame0.fa.txt'], ['set60_frame1', 4.214, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_60_frame1.fa.txt'], ['set63_frame1', 4.291, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_63_frame1.fa.txt'], ['set97_frame1', 4.334, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_97_frame1.fa.txt'], ['set8_frame1', 4.374, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_8_frame1.fa.txt'], ['set26_frame2', 4.377, 'alignment_Error_Exon_15_query_Mechanitis_mazaeus_translated_genomic_sequence_26_frame2.fa.txt'], ['set58_frame2', 4.402, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_58_frame2.fa.txt'], ['set125_frame0', 4.415, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_125_frame0.fa.txt']]\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSV-QEEPSYSNRDDSESYNN-\n",
      "set92_frame2_Bo --------RWSGRSVT-----------GTPLLPTVMSVKQETSSFGIAQDSEDCNNR\n",
      "                        : * :. *           .: ***::*** ** .*:.  :***. ** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Melinaea_menoph -----------------------------SILTEEVCVSTSVEAPALPPAKDGTGDAHND\n",
      "set19_frame2_Me DFPERVPAHAVKSLGGVNEAHQFCECVRISCLTLTIC--SLVPRPLTKPACS-SGICHCR\n",
      "                                             * **  :*  : *  *   ** . :* .*  \n",
      "\n",
      "Melinaea_menoph SEYN-----\n",
      "set19_frame2_Me NAFNLLLMT\n",
      "                . :*     \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza SIITEEVCVRTAAVEALSP---VLLPSLESVKQDAGFVNNDSEYN\n",
      "set28_frame0_Me D-FNINLLIKNDSNPKLIRYLLVILPTLVEV--------------\n",
      "                . :. :: ::. :   *     *:**:* .*              \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSVQEEPSYSNRDDSESYNN\n",
      "set60_frame1_Bo L-------QGHTSNTTFVINHITSPKWLIAFSRKIKKV-----------------\n",
      "                        :  ..:.*::: *:: *     :  .* .*                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMS--LPAISSLLLPSIMSVQEEPSYSNRDDSESYNN\n",
      "set63_frame1_Bo STFSNIVEYPIKTLLMIPILHMTGTLGNIS---------------------------\n",
      "                : .::* *   *.   * : **:  *  **                           \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza -----------------SIITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "set97_frame1_Me KTRMVLIETKKIEIFLRAVTTLKYVLYTGKLKIRNETILSYARRNCTAAQFQNTVSNKN\n",
      "                                 :: * :  : *. ::  . .:*.  .     * * *. *: *\n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSL--PAISSLLLPSIM--------------SVQEEPSY\n",
      "set8_frame1_Bol T------------SRFFVLLHIFLFCHIRVSLLIADLAISDRRRIFFSQPPWSVSSEPRY\n",
      "                *            .: ::* *: *      ***:..:               **..** *\n",
      "\n",
      "Boloria_selene_ SNRD-DSESYNN---------------------------------\n",
      "set8_frame1_Bol KNSEANSRSLYSPVLGSVPALSTIKALVFALLTIRPTARFRIGLF\n",
      "                .* : :*.*  .                                 \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Mechanitis_maza SI------------------ITEEVCVRTAAVEALSPVLLPSLESVKQDAGFVNNDSEYN\n",
      "set26_frame2_Me SASYFGLQYLRFKVVRVYEQIIFRLVIVTIGVEALT-IRSPPF------GHFVN----FC\n",
      "                *                   *  .: : * .****: :  *.:      . ***    : \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILP--HMSLPAISSLLLPSIMSVQEEPSYSNRDDSESYNN\n",
      "set58_frame2_Bo Y----PKKHHNKKNYMTLCPVEHIQY----NLTL--------------------YSL\n",
      "                      ::::.**:   : *  *:.     .* *                    *. \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ ------TIVAEIEEK--------------------KSKKDKTIILPHMS--LPAISSLLL\n",
      "set125_frame0_B YNFEKLSIVIQMSRKKLILQGVYWLFFGLKIAHSTKKKKKKVHVTSHYNPFVFVIFCDVL\n",
      "                      :** ::..*                    *.**.*. : .* .  : .* . :*\n",
      "\n",
      "Boloria_selene_ PSIMSVQEEPSYSN-RDDSESYNN\n",
      "set125_frame0_B PTAPPFYFEPHVINGRTLRPVVSR\n",
      "                *:  ..  **   * *      ..\n",
      "\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Melitaea_cinxia/Error_exon_processing/Error_Exon_15/for_alignment//alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_92_frame2.fa.txt\n",
      "Alignmnet file: alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_92_frame2.fa.txt\n",
      "alignment_clustal_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_92_frame2.fa.txt\n",
      "Exon_15\n",
      "Boloria_selene_Exon_15_HG993134.1_8456739_8456906_left_1_right_2\n",
      "here here 1 1 set92_frame2 Boloria_selene_Exon_15_HG993134.1_8456739_8456906_left_1_right_2 0 56\n",
      "here here 1 1 set92_frame2 set92_frame2_Boloria_selene 0 56\n",
      "set92_frame2_Boloria_selene\n",
      "--------RWSGRSVT-----------GTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "alignment_clustal_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_92_frame2.fa.txt\n",
      "Alignment File: CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSV-QEEPSYSNRDDSESYNN-\n",
      "set92_frame2_Bo --------RWSGRSVT-----------GTPLLPTVMSVKQETSSFGIAQDSEDCNNR\n",
      "                        : * :. *           .: ***::*** ** .*:.  :***. ** \n",
      "\n",
      "Query OK?n\n",
      "Add new query :PLLPTVMSVKQETSSFGIAQDSEDC\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/11/2024 07:03:37\n",
      "New DB name:   /mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Melitaea_cinxia/Error_exon_processing/Error_Exon_15/Run_Blast/local_db.txt\n",
      "New DB title:  local_db.txt\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.00149107 seconds.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "set92_frame2_Boloria_selene\n",
      "4251 4325\n",
      "reached here\n",
      "sequence:\n",
      "CCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAA\n",
      "sequence:\n",
      "PLLPTVMSVKQETSSFGIAQDSEDC\n",
      "left = AC, right =CA\n",
      "sequence:\n",
      "ACGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAACAA\n",
      "sequence:\n",
      "TPLLPTVMSVKQETSSFGIAQDSEDCN\n",
      "left = GG, right =CA\n",
      "sequence:\n",
      "GGCACGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAACAACAG\n",
      "sequence:\n",
      "GTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "left = AC, right =GT\n",
      "sequence:\n",
      "ACGGGCACGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAACAACAG\n",
      "sequence:\n",
      "TGTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTCACGGGCACGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAACAACAG\n",
      "sequence:\n",
      "VTGTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "left = TC, right =GT\n",
      "sequence:\n",
      "TCCGTCACGGGCACGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGACCAGCAGCTTCGGCATCGCCCAGGACTCCGAGGACTGCAACAACAG\n",
      "sequence:\n",
      "SVTGTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "left = AG, right =GT\n",
      "SVTGTPLLPTVMSVKQETSSFGIAQDSEDCNN\n",
      "[[1, 2]]\n",
      "start_coordinate : 9526823,stop_coordinate : 9526918\n",
      "Melitaea_cinxia,NC_059396.1,9526822,9526920,0,N,Danaus_plexippus_XM_032654785.2_Frame_1_rightoh_2_query_Exon_15,1,32,32,Y,Y\n",
      "\n",
      "Fix Overhang!! Proceed?y\n",
      "Mellicta_athalia,HG992179.1,000,000,0,Y,Error_Exon_15,00,00,00\n",
      "\n",
      "Exon_15\n",
      "current_query : Boloria_euphrosyne\n",
      "Exon_15.fa\n",
      "Boloria_euphrosyne_Exon_15_OX637873.1_7248353_7248517_left_1_right_2\n",
      "TIIAEIEEKKSKKDKTNILPHVSLPALSSLLRPSMSVQDEPSYSNRDDSESYNN\n",
      "current_query : Danaus_plexippus\n",
      "Exon_15.fa\n",
      "Danaus_plexippus_Exon_15_NC_083537.1_5585774_5585923_left_1_right_2\n",
      "SILTEEVVVSKSSVEAPPVAVLPPALVLPSAINMKQDGSGYVNNDSEYN\n",
      "current_query : Nymphalis_io\n",
      "Exon_15.fa\n",
      "Nymphalis_io_Exon_15_NC_065890.1_6887128_6887325_left_1_right_2\n",
      "SIISVENDTMKVCVPEKVTHVPLDMTSHQLGPLESSSLILPTIMSVNQEASAFRTRLTDPDYYNN\n",
      "current_query : Boloria_selene\n",
      "Exon_15.fa\n",
      "Boloria_selene_Exon_15_HG993134.1_8456739_8456906_left_1_right_2\n",
      "TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSVQEEPSYSNRDDSESYNN\n",
      "current_query : Hypolimnas_misippus\n",
      "Exon_15.fa\n",
      "Hypolimnas_misippus_Exon_15_SGBE01000384.1_8316073_8316270_left_1_right_2\n",
      "SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVKQETSNFILGQADSEYFNN\n",
      "Gene_start = 11131605, Gene_end = 11138135\n",
      "Getting Gene\n",
      "GCA_905220545.2_ilMelAtha1.2_genomic.fna\n",
      "['1.Query', '1.Blast_result', 'tblastn_Danaus_plexippus.sh', 'temp', '2.Final_output', '0.Temp', 'error_exons.txt', 'desktop.ini']\n",
      "['Boloria_euphrosyne_Exon_15.fa', 'Danaus_plexippus_Exon_15.fa', 'Nymphalis_io_Exon_15.fa', 'Boloria_selene_Exon_15.fa', 'Hypolimnas_misippus_Exon_15.fa']\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "Error_Exon_15\n",
      "min = set110_frame0, 3.034\n",
      "5 top scores:\n",
      "[['set110_frame0', 3.034, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_110_frame0.fa.txt'], ['set110_frame0', 3.594, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_110_frame0.fa.txt'], ['set112_frame0', 4.322, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_112_frame0.fa.txt'], ['set65_frame2', 4.376, 'alignment_Error_Exon_15_query_Boloria_selene_translated_genomic_sequence_65_frame2.fa.txt'], ['set88_frame2', 4.385, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_88_frame2.fa.txt'], ['set98_frame0', 3.988, 'alignment_Error_Exon_15_query_Boloria_euphrosyne_translated_genomic_sequence_98_frame0.fa.txt'], ['set121_frame0', 4.449, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_121_frame0.fa.txt'], ['set118_frame2', 4.453, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_118_frame2.fa.txt'], ['set101_frame2', 4.485, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_101_frame2.fa.txt'], ['set119_frame0', 4.517, 'alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_119_frame0.fa.txt']]\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVKQETSNFILGQADS\n",
      "set110_frame0_H QV---------------------AVVGRSV--AGAP--LLPTVMSVKQEASSFGIAH-DS\n",
      "                .:                      *  *..  *.:*  :***:******:*.* :.: **\n",
      "\n",
      "Hypolimnas_misi EYFNN-\n",
      "set110_frame0_H EYCNNR\n",
      "                ** ** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSLPAISSLLLPSIMSV-QEEPSYSNRDDSESYNN-\n",
      "set110_frame0_B QVAV-----------------VGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNNR\n",
      "                 :..                 :. .. .: ***::*** ** .*:.  .***  ** \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ TIVAEIEEKKSKKDKTIILPHMSLPAISSL--LLPSIMSVQEEPSYSNRDDSESYNN\n",
      "set112_frame0_B KTIQYVQ-------QFLVLHTICRPFVSSARGMMPPLAHPLATPLCSERNE------\n",
      "                . :  ::       : ::*  :. * :**   ::*.:      *  *:*::      \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_selene_ T--IVAEIEEKKSK-------------------KDKTIILPHM-----------------\n",
      "set65_frame2_Bo NNLLIISMSYIGSKVLLFYFLCNSLKQSFDFLAHKCTLHLLHIVTVFKLQDGNMHRILSG\n",
      "                .  :: .:.   **                   :. *: * *:                 \n",
      "\n",
      "Boloria_selene_ ----SLPAIS--------------------------------------------------\n",
      "set65_frame2_Bo VLICSCPTPSSMTSLGGKTWHAFPFLIFGGGPSKIALELGRPLGVGVEYLHNVSARSDRK\n",
      "                    * *: *                                                  \n",
      "\n",
      "Boloria_selene_ ------SLLLPSIMSVQEEPSYSNRDDSESYNN---------------------------\n",
      "set65_frame2_Bo FARFMISLLIQKGFSVQ-NPGIDNREIQKVVENSHVPLSTFYADSVPAKMGIHEVNASHV\n",
      "                      ***: . :*** :*. .**: .:  :*                           \n",
      "\n",
      "Boloria_selene_ ----------------------------------------------\n",
      "set65_frame2_Bo GVVLFHQLWARYSNPSFRLFIVAFYCFGWFSAYVGRSQNDLFVIFP\n",
      "                                                              \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi SII---------PDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVKQETS\n",
      "set88_frame2_Hy HILYFPFIHVCNP---------IQSKLPSKPKRTKPKP-----------------KKEGK\n",
      "                 *:         *          *.*:* .* .. *:*                 *:* .\n",
      "\n",
      "Hypolimnas_misi NFI---LGQADSEYF----------NN\n",
      "set88_frame2_Hy NYTCKHVGSFSAQTYLFDFRGYLVKKI\n",
      "                *:    :*. .:: :          : \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Boloria_euphros TIIAEIEEKKSKKDKTN---ILPHVSLPALSSLLRPSMSVQDEPSYSNRDDSESYNN---\n",
      "set98_frame0_Bo ------SSKSSPSDSISSDEISPFSSMRSISSIFLSSMSVIQYKASCQANQSEQNQNQKK\n",
      "                      ..*.* .*. .   * *. *: ::**:: .**** :  : .: ::**. :*   \n",
      "\n",
      "Boloria_euphros -------------------------\n",
      "set98_frame0_Bo REKITPVSMWGRFRHKHIYLILGGI\n",
      "                                         \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi ---------------------SIIPDDE-----DTDKDLSQQKIPRTPLEVPPRPAVPAA\n",
      "set121_frame0_H YTHKLYGMVLMFKLTFKYYYESFVWEYRKLLFLDFFRDF-----------FSLRKNHPRT\n",
      "                                     *:: : .     *  :*:           .. *   * :\n",
      "\n",
      "Hypolimnas_misi SPSLVLPTIMSVKQETSNFILGQADSEYFNN--------\n",
      "set121_frame0_H SRNIIKKELAKSVQPLSRFALSNTFSDSFLYILIIIIII\n",
      "                * .::   : .  *  *.* *.:: *: *          \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLP----TIMSVKQETSNFILG\n",
      "set118_frame2_H -------------------KKQFEIATDNIVYLNSEKSFFPIHLYRMILIKTKNLGFSLG\n",
      "                                   :. :*:..   *   * . .:*     :: :* :. .* **\n",
      "\n",
      "Hypolimnas_misi QADSEYFNN\n",
      "set118_frame2_H R--------\n",
      "                :        \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi ----SIIPDDEDTDKD-------------------------------------------L\n",
      "set101_frame2_H ESFKSVV-----TSRGGRQVRRGRAAAAHRHERQAGGQQLRHRPRLRVLQQQVAALRFLF\n",
      "                    *::     *.:.                                           :\n",
      "\n",
      "Hypolimnas_misi SQQKIPRTPL------EVPPRPAVPAASPSL----VLPTIMSVKQETSNFILGQADSEYF\n",
      "set101_frame2_H SNRTYTRTFLSRQWRSELNSRPGHHTRPPSLENHTICPAIFSFAYHLSPLC---------\n",
      "                *::. .** *      *: .**.  : .***    : *:*:*.  . * :          \n",
      "\n",
      "Hypolimnas_misi NN\n",
      "set101_frame2_H --\n",
      "                  \n",
      "\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVK--QETSNFILGQA\n",
      "set119_frame0_H ML------------QRRIGKK--------LGSYSSPTLE-KTVLYIKIGKTTLNRHI---\n",
      "                 :            *::* :.          . :**:*   *:: :*  : * *  :   \n",
      "\n",
      "Hypolimnas_misi DSEYFNN\n",
      "set119_frame0_H ---FINN\n",
      "                   ::**\n",
      "\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Mellicta_athalia/Error_exon_processing/Error_Exon_15/for_alignment//alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_110_frame0.fa.txt\n",
      "Alignmnet file: alignment_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_110_frame0.fa.txt\n",
      "alignment_clustal_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_110_frame0.fa.txt\n",
      "Exon_15\n",
      "Hypolimnas_misippus_Exon_15_SGBE01000384.1_8316073_8316270_left_1_right_2\n",
      "here here 1 1 set110_frame0 Hypolimnas_misippus_Exon_15_SGBE01000384.1_8316073_8316270_left_1_right_2 0 65\n",
      "here here 1 1 set110_frame0 set110_frame0_Hypolimnas_misippus 0 65\n",
      "set110_frame0_Hypolimnas_misippus\n",
      "QV---------------------AVVGRSV--AGAP--LLPTVMSVKQEASSFGIAH-DSEYCNN\n",
      "alignment_clustal_Error_Exon_15_query_Hypolimnas_misippus_translated_genomic_sequence_110_frame0.fa.txt\n",
      "Alignment File: CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Hypolimnas_misi SIIPDDEDTDKDLSQQKIPRTPLEVPPRPAVPAASPSLVLPTIMSVKQETSNFILGQADS\n",
      "set110_frame0_H QV---------------------AVVGRSV--AGAP--LLPTVMSVKQEASSFGIAH-DS\n",
      "                .:                      *  *..  *.:*  :***:******:*.* :.: **\n",
      "\n",
      "Hypolimnas_misi EYFNN-\n",
      "set110_frame0_H EYCNNR\n",
      "                ** ** \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query OK?n\n",
      "Add new query :GRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCN\n",
      "\n",
      "\n",
      "Building a new DB, current time: 05/11/2024 07:15:18\n",
      "New DB name:   /mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Mellicta_athalia/Error_exon_processing/Error_Exon_15/Run_Blast/local_db.txt\n",
      "New DB title:  local_db.txt\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.00128102 seconds.\n",
      "[[1, 2]]\n",
      "set110_frame0_Hypolimnas_misippus\n",
      "4918 5016\n",
      "reached here\n",
      "sequence:\n",
      "GGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAA\n",
      "sequence:\n",
      "GRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCN\n",
      "left = GT, right =CA\n",
      "sequence:\n",
      "GTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GC, right =GT\n",
      "sequence:\n",
      "GCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "AVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = CA, right =GT\n",
      "sequence:\n",
      "CAAGTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "QVAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = TA, right =GT\n",
      "sequence:\n",
      "TAACAAGTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "*QVAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GG, right =GT\n",
      "sequence:\n",
      "GGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "GRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GC, right =GT\n",
      "sequence:\n",
      "GCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "AVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = GT, right =GT\n",
      "sequence:\n",
      "GTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "VAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = CA, right =GT\n",
      "sequence:\n",
      "CAAGTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "QVAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = TA, right =GT\n",
      "sequence:\n",
      "TAACAAGTCGCGGTGGTCGGCAGGTCCGTCGCGGGCGCGCCGCTGCTGCCCACCGTCATGAGCGTCAAGCAGGAGGCCAGCAGCTTCGGCATCGCCCACGACTCCGAGTACTGCAACAACAG\n",
      "sequence:\n",
      "*QVAVVGRSVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "left = AG, right =GT\n",
      "SVAGAPLLPTVMSVKQEASSFGIAHDSEYCNN\n",
      "[[1, 2]]\n",
      "start_coordinate : 11136528,stop_coordinate : 11136623\n",
      "Mellicta_athalia,HG992179.1,11136527,11136625,0,N,Danaus_plexippus_XM_032654785.2_Frame_1_rightoh_2_query_Exon_15,1,32,32,Y,Y\n",
      "\n",
      "Fix Overhang!! Proceed?y\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"6.Heliconiinae_Danainae_Nymphalinae\"\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "\n",
    "with open(f\"{output_location}/error_exons.txt\",'r') as error_file:\n",
    "    error_file_lines = error_file.readlines()\n",
    "    \n",
    "for line in error_file_lines:\n",
    "    if len(line) != 1:\n",
    "        print(line)\n",
    "        line_split = line.split(\",\")\n",
    "        species = line_split[0]\n",
    "        error_exon = line_split[6]\n",
    "        \n",
    "        make_folder_error_exon(output_location,species,error_exon)\n",
    "        left_oh,right_oh = make_query(output_location,error_exon,species)\n",
    "        fragment_start, fragment_end, scaffold, complement,query_name = get_genomic_coordinates(output_location,error_exon,species)\n",
    "        print(f\"Gene_start = {fragment_start}, Gene_end = {fragment_end}\")\n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                          species,                       \n",
    "                          scaffold,\n",
    "                          fragment_start,\n",
    "                          fragment_end,\n",
    "                          complement,\n",
    "                          output_location)\n",
    "        \n",
    "        make_raw_files_for_alignment(gene_sequence,output_location,species,error_exon)\n",
    "        mafft_run_location = run_mafft(output_location,species,error_exon)\n",
    "        \n",
    "        query_sequence, alignment_name,alignment_file = process_mafft_output(mafft_run_location, error_exon)\n",
    "        \n",
    "        print(f'Alignment File: {\"\".join(alignment_file)}')\n",
    "        query_check = input(\"Query OK?\")\n",
    "        if query_check[0].lower() == 'n':\n",
    "            query_sequence = input(\"Add new query :\")\n",
    "            if query_sequence == '':\n",
    "                print(species, error_exon)\n",
    "                assert False\n",
    "        run_blast(gene_sequence,query_sequence,alignment_name,output_location,species, error_exon)\n",
    "        ag_gt = process_genome_fragment_blast_file(output_location,left_oh,right_oh,species,genome_location)\n",
    "        if ag_gt != \"Error!! Too many stops\":\n",
    "            coordinate_output = process_genome_blast_file(output_location,left_oh,right_oh,species,genome_location,ag_gt,query_name)\n",
    "            print(coordinate_output)\n",
    "        else:\n",
    "            print(ag_gt)\n",
    "        input(\"Fix Overhang!! Proceed?\")\n",
    "#         print(gene_sequence)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "0698f5d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "mafft_run_location = run_mafft(output_location,species,error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b1cb49b",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    \n",
    "#         print()\n",
    "        \n",
    "get_genomic_coordinates(output_location,error_exon,species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12854358",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length):\n",
    "    for offset in range(3):\n",
    "        translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "        for i in range(len(translated_sequence)):\n",
    "            if len(translated_sequence[i])> 0.8*query_length:\n",
    "                sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "                # print(i+1, offset)\n",
    "                \n",
    "                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                    output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fba374ae",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
