{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b1818357",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "45\n",
      "38\n",
      "PQG--VKRS--SKNTDDWGT-N--KHRC--SSARTTRRRF---TE---PP---NNPPLSATYN\n",
      "--|--||||--|---||----|--|-|---|--|--|.-----|----|----||-------|\n",
      "--GHGVKRSRDS---DD---ENSYK-R-GKS--R--RK--SCHT-HSAP-QSLNN-------N\n",
      "\n"
     ]
    }
   ],
   "source": [
    "species1 = \"5.Bicyclus_anynana\"\n",
    "\n",
    "transcript1 = \"XM_024088150.2\"\n",
    "\n",
    "location = \"H:/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\"\n",
    "seq_modi = [[0,2],[1,2],[1,1],[2,0],[0,0],[0,0],[0,1],[2,1],[2,2],[1,0],[0,0],[0,0],[0,2],[1,2],[1,2],[1,0],[0,1],[2,1],[2,1],[2,0],[0,0],[0,0],[0,0],[0,0]]\n",
    "\n",
    "\n",
    "\n",
    "# print(overhang)\n",
    "\n",
    "\n",
    "\n",
    "from Bio import Align\n",
    "from Bio.Align import substitution_matrices\n",
    "substitution_matrices.load()\n",
    "matrix = substitution_matrices.load(\"BLOSUM62\")\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.substitution_matrix = matrix\n",
    "# aligner.mode = \"local\"\n",
    "\n",
    "from Bio import SeqIO\n",
    "species2 = \"33.Pararge_aegeria\"\n",
    "transcript2 = \"XM_039907241.1\"\n",
    "exon_1 = \"Exon17\"\n",
    "exon_2 = \"Exon17\"\n",
    "exon_number = int(exon_1.split(\"Exon\")[1])\n",
    "overhang = seq_modi[exon_number-1][0]\n",
    "\n",
    "exon_1_data = SeqIO.parse(f\"{location}/{species1}/Transcripts/{transcript1}/{exon_1}.fa\",\"fasta\")\n",
    "for exon_1_info in exon_1_data:\n",
    "    exon_1_sequence = exon_1_info.seq[overhang:].translate()\n",
    "    print(len(exon_1_sequence))\n",
    "exon_2_data = SeqIO.parse(f\"{location}/{species2}/Transcripts/{transcript2}/{exon_2}.fa\",\"fasta\")\n",
    "for exon_2_info in exon_2_data:\n",
    "    exon_2_sequence = exon_2_info.seq[overhang:].translate()\n",
    "    print(len(exon_2_sequence))\n",
    "alignment = aligner.align(exon_1_sequence, exon_2_sequence)\n",
    "print(alignment[0])\n",
    "# print(exon_1_data)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "0c3e600d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pairwise sequence aligner with parameters\n",
      "  match_score: 1.000000\n",
      "  mismatch_score: 0.000000\n",
      "  target_internal_open_gap_score: 0.000000\n",
      "  target_internal_extend_gap_score: 0.000000\n",
      "  target_left_open_gap_score: 0.000000\n",
      "  target_left_extend_gap_score: 0.000000\n",
      "  target_right_open_gap_score: 0.000000\n",
      "  target_right_extend_gap_score: 0.000000\n",
      "  query_internal_open_gap_score: 0.000000\n",
      "  query_internal_extend_gap_score: 0.000000\n",
      "  query_left_open_gap_score: 0.000000\n",
      "  query_left_extend_gap_score: 0.000000\n",
      "  query_right_open_gap_score: 0.000000\n",
      "  query_right_extend_gap_score: 0.000000\n",
      "  mode: global\n",
      "\n",
      "ATGC--AA-GGAA\n",
      "|--|--||-|-||\n",
      "A--CGTAAAG-AA\n",
      "\n",
      "24.0\n"
     ]
    }
   ],
   "source": [
    "from Bio import Align\n",
    "from Bio.Align import substitution_matrices\n",
    "substitution_matrices.load()\n",
    "# matrix = substitution_matrices.load(\"BLOSUM62\")\n",
    "aligner = Align.PairwiseAligner()\n",
    "# aligner.substitution_matrix = matrix\n",
    "print(aligner)\n",
    "alignment = aligner.align(\"ATGCAAGGAA\", \"ACGTAAAGAA\")\n",
    "print(alignment[0])\n",
    "print(score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93279735",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
