{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "667f63f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(species):\n",
    "    list_of_genomes = os.listdir(\"/mnt/f/Genomes_2023-12-26\")\n",
    "    for genome in list_of_genomes:\n",
    "        if species == genome:\n",
    "            files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\n",
    "            for file in files_in_genome_folder:\n",
    "                if file.endswith(\"_genomic.fna\"):\n",
    "#                     print(file)\n",
    "                    return(file)\n",
    "#                     break\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d8adb7bf",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Parnassius_glacialis', 'Sericinus_montela', 'Teinopalpus_imperialis', 'Troides_aeacus', 'Troides_oblongomaculatus', 'Papilio_polytes', 'Papilio_glaucus', 'Battus_philenor', 'Baronia_brevicornis']\n",
      "Parnassius_glacialis\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "\n",
      "mdh\n",
      "\n",
      "idh\n",
      "\n",
      "DDC\n",
      "\n",
      "gapdh\n",
      "\n",
      "ef1\n",
      "\n",
      "hcl\n",
      "\n",
      "CAT\n",
      "\n",
      "cad\n",
      "\n",
      "Sericinus_montela\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "\n",
      "mdh\n",
      "\n",
      "idh\n",
      "\n",
      "DDC\n",
      "\n",
      "gapdh\n",
      "\n",
      "ef1\n",
      "\n",
      "hcl\n",
      "\n",
      "CAT\n",
      "\n",
      "cad\n",
      "\n",
      "Teinopalpus_imperialis\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "\n",
      "mdh\n",
      "\n",
      "idh\n",
      "\n",
      "DDC\n",
      "\n",
      "gapdh\n",
      "\n",
      "ef1\n",
      "\n",
      "hcl\n",
      "\n",
      "CAT\n",
      "\n",
      "cad\n",
      "\n",
      "Troides_aeacus\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "\n",
      "mdh\n",
      "\n",
      "idh\n",
      "\n",
      "DDC\n",
      "\n",
      "gapdh\n",
      "\n",
      "ef1\n",
      "\n",
      "hcl\n",
      "\n",
      "CAT\n",
      "\n",
      "cad\n",
      "\n",
      "Troides_oblongomaculatus\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "\n",
      "mdh\n",
      "\n",
      "idh\n",
      "\n",
      "DDC\n",
      "\n",
      "gapdh\n",
      "\n",
      "ef1\n",
      "\n",
      "hcl\n",
      "\n",
      "CAT\n",
      "\n",
      "cad\n",
      "\n",
      "Papilio_polytes\n",
      "rps5\n",
      "\n",
      "wg\n",
      "\n",
      "rps2\n",
      "SGPNH*IEKFFLEWRTQLXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXWVPVTKLGRLVREGKIDKLESIYLFSLPIKEFEIIDFFLGASLNDEVLKIMPVQKQTRAGQRTRFKAFVAIGDNNGHIGLGVKCSKEVATAIRGAIILAKLSVLPVRRGYWGNKIGKPHTVPCK\n",
      "mdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "idh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "DDC\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "gapdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "ef1\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "hcl\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "CAT\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "cad\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "Papilio_glaucus\n",
      "rps5\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "wg\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "rps2\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "mdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "\n",
      "idh\n",
      "FKCLICT*TKHKXXXXXXXXIWDLIKEKLILPFLDIELHTYDLGMEYRDQTEDQVTIDCANAIKKYNVGIKCATITPDENRVEEFKLKKMWKSPNGTIRNILGGTVFREAIICKNIPRLVTGWEKPIIIGRHAHADQYKATDFVVPGEGKLELIWTPPSGEPIKHVVNDFKGAGVALGMFNTDASIVDFAHSSFKYALDRKYPLYLSTKNTILKKYDGRFKDIFQEIYDNKYKTEFEAAGIWYEHRLIDDMVAYAMKSEGGFVWACKNYDGDVQSDSVAQGYGSLGLMTSVLICPDGKTVEAEAAHGTVTRHYRFYQQGKETSTNPIASIFAWTRGLLHRAKLDNNDELKKFAETLESVCINTIESGIMTKDLAICIKGMNNVKRADYYETFEFMNKLAENLKKSLGK\n",
      "DDC\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "gapdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "ef1\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "hcl\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "CAT\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "cad\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "Battus_philenor\n",
      "rps5\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "wg\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "rps2\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "\n",
      "mdh\n",
      "*GSCYWSCRSDCLLSAVSNCFWRCIWT*PASISSPS*YCSHDGCFGRCGNGAC*LCTAIAHWCSSHC*SRRGF*RCCCSFLGWSYAKKGRDGAKRFTFSQCSHLQGTRSST**GS*KRCQSTCSRKPCKHQCLNLFKVCSIYSKENFTAMTRLDQNRAQSQLAAKIGVPVQDVKNVIIWGNHSSTQFPDASYAIVKIKGAEKPVPAAINDEEYLKTSFVTTVQKRGAAVIAARKMSSALSAAKAASDHMRDWFLGTGDRWVSMGVVSDGSYGTPRDVVFSFPVTVSNGKWKIVEGLTISDFARQMLDNTGKELAEEKQDALDVCKD\n",
      "idh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "DDC\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "gapdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "ef1\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "hcl\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "CAT\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "cad\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "Baronia_brevicornis\n",
      "rps5\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "wg\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "rps2\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "\n",
      "mdh\n",
      "SQFKMSKML*YGAITHQLNFPMLLMLLQKLMVLKNLFQQQSMMMNI*NLLLFQLYKSVELLLLQREKCHLHCLQQKLLQTT*EIGFWVLVIIGLVWGLFLMDHMVHPVMLSSLSLLL*VMENGRLLKALLFLTLPVKCWIIQVKN*LKRNKMHWMFAK\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "idh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "DDC\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "gapdh\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "ef1\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "hcl\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "CAT\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n",
      "cad\n",
      "Papilio_polytes\trps2\n",
      "\n",
      "Papilio_glaucus\tidh\n",
      "\n",
      "Battus_philenor\tmdh\n",
      "\n",
      "Baronia_brevicornis\tmdh\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "family_group = \"7.Papilionidae\"\n",
    "\n",
    "\n",
    "working_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}\"\n",
    "blast_output_location = f\"{working_location}/1.Blast_result\"\n",
    "output_location = f\"{working_location}/3.Extracted Sequences\"\n",
    "\n",
    "list_of_working_folders = os.listdir(working_location)\n",
    "# if \"3.Extracted Sequences\" in list_of_working_folders:\n",
    "#     subprocess.run(f'rm -r \"{output_location}\"', shell = True)\n",
    "# os.mkdir(output_location)\n",
    "\n",
    "\n",
    "list_of_species = os.listdir(blast_output_location)\n",
    "list_of_errors = ''\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "list_of_species= ['Parnassius_glacialis', 'Sericinus_montela', 'Teinopalpus_imperialis', 'Troides_aeacus', 'Troides_oblongomaculatus', 'Papilio_polytes', 'Papilio_glaucus', 'Battus_philenor','Baronia_brevicornis']\n",
    "print(list_of_species)\n",
    "list_of_errors = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    with open(f\"{blast_output_location}/{species}/{species}_coordinates.csv\", 'r') as coordinates_file:\n",
    "        coordinates_list = coordinates_file.readlines()\n",
    "        \n",
    "    for i in range(1,len(coordinates_list)):\n",
    "        coordinates_split = coordinates_list[i].split(\",\")\n",
    "        \n",
    "        gene = coordinates_split[6].split(\"_\")[-1]\n",
    "        \n",
    "        list_of_folders_in_extracted = os.listdir(f\"{output_location}\")\n",
    "        if gene not in list_of_folders_in_extracted:\n",
    "            os.mkdir(f\"{output_location}/{gene}\")\n",
    "        \n",
    "        \n",
    "        scaffold,start,stop,complement = coordinates_split[1],coordinates_split[2],coordinates_split[3],coordinates_split[4]\n",
    "#         genome_file = get_genome_file(species)\n",
    "        \n",
    "        genome_fasta = SeqIO.parse(f\"/mnt/f/Genomes_2023-12-26/{species}/{get_genome_file(species)}\", 'fasta')\n",
    "        \n",
    "        for records in genome_fasta:\n",
    "            if scaffold in records.id:\n",
    "                sequence = (records.seq[int(start)-1:int(stop)])\n",
    "                break\n",
    "        if complement == \"1\":\n",
    "            sequence = sequence.reverse_complement()\n",
    "            \n",
    "        \n",
    "        print(gene)\n",
    "#         print(sequence,\"\\n\")\n",
    "        if \"*\" in sequence.translate():\n",
    "            list_of_errors += f\"{species}\\t{gene}\\n\\n\"\n",
    "            print(sequence.translate())\n",
    "            continue\n",
    "        \n",
    "        print(list_of_errors)\n",
    "            \n",
    "        with open(f\"{output_location}/{gene}/{species}.txt\", 'w') as out_file:\n",
    "            output = f\">{species}_{gene}\\n{sequence}\"\n",
    "            out_file.write(output)\n",
    "        \n",
    "#         print(coordinates_list[i])\n",
    "#     assert False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d8ec914",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e872e2c8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
