{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "be5b7270-c0c4-4779-b72f-0c5d6e5e9a53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bicyclus_anynana\n",
      "Coenonympha_glycerion\n",
      "Erebia_aethiops\n",
      "Erebia_ligea\n",
      "Hipparchia_semele\n",
      "Lasiommata_megera\n",
      "Maniola_hyperantus\n",
      "Maniola_jurtina\n",
      "Melanargia_galathea\n",
      "Oeneis_ivallda\n",
      "Pararge_aegeria\n",
      "Anthocharis_cardamines\n",
      "Aporia_crataegi\n",
      "Leptophobia_aripa\n",
      "Pieris_brassicae\n",
      "Pieris_mannii\n",
      "Pieris_melete\n",
      "Pieris_napi\n",
      "Pieris_rapae\n",
      "Colias_croceus\n",
      "Colias_eurytheme\n",
      "Colias_nastes\n",
      "Eurema_hecabe\n",
      "Phoebis_sennae\n",
      "Zerene_cesonia\n",
      "Leptidea_juvernica\n",
      "Leptidea_reali\n",
      "Leptidea_sinapis\n",
      "Danaus_chrysippus\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta\n",
      "Mechanitis_mazaeus\n",
      "Mechanitis_messenoides\n",
      "Melinaea_marsaeus_rileyi\n",
      "Melinaea_menophilus\n",
      "Napeogenes_inachia\n",
      "Napeogenes_sylphis\n",
      "Argynnis_bischoffii_washingtonia\n",
      "Boloria_euphrosyne\n",
      "Boloria_selene\n",
      "Brenthis_daphne\n",
      "Brenthis_hecate\n",
      "Brenthis_ino\n",
      "Dryadula_phaetusa\n",
      "Dryas_iulia_moderata\n",
      "Eueides_isabella\n",
      "Fabriciana_adippe\n",
      "Heliconius_charithonia\n",
      "Heliconius_nattereri\n",
      "Heliconius_sara\n",
      "Philaethria_dido\n",
      "Euphydryas_editha\n",
      "Hypolimnas_misippus\n",
      "Melitaea_cinxia\n",
      "Mellicta_athalia\n",
      "Nymphalis_c-album\n",
      "Nymphalis_io\n",
      "Nymphalis_polychloros\n",
      "Nymphalis_urticae\n",
      "Vanessa_atalanta\n",
      "Vanessa_cardui\n",
      "Vanessa_tameamea\n",
      "Battus_philenor\n",
      "Ornithoptera_alexandrae\n",
      "Ornithoptera_priamus\n",
      "Papilio_dardanus_tibullus\n",
      "Papilio_demoleus\n",
      "Papilio_elwesi\n",
      "Papilio_glaucus\n",
      "Papilio_machaon\n",
      "Papilio_memnon\n",
      "Papilio_polytes\n",
      "Papilio_protenor\n",
      "Papilio_xuthus\n",
      "Parnassius_glacialis\n",
      "Teinopalpus_imperialis\n",
      "Troides_aeacus\n",
      "Troides_oblongomaculatus\n",
      "Aricia_agestis\n",
      "Aricia_artaxerxes\n",
      "Calycopis_cecrops\n",
      "Celastrina_argiolus\n",
      "Cyaniris_semiargus\n",
      "Glaucopsyche_alexis\n",
      "Lysandra_bellargus\n",
      "Lysandra_coridon\n",
      "Phengaris_arion\n",
      "Plebejus_argus\n",
      "Polyommatus_icarus\n",
      "Polyommatus_iphigenia\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis\"\n",
    "family_group_list = [\"3.Satyrine\",\"4.Pierinae\",\"5.Coliadinae\",\"6.Leptidea\",\"8.Heliconiinae_Danainae\",\"9.Papilionidae\",\"10.Lycaenidae\"]\n",
    "gene = \"period\"\n",
    "output = ''\n",
    "output_2 = 'Species\\tExons'\n",
    "\n",
    "for family_group in family_group_list:\n",
    "    list_of_species = os.listdir(f\"{location}/{family_group}/2.Final_output/\")\n",
    "    \n",
    "    if \"desktop.ini\" in list_of_species:\n",
    "        list_of_species.remove(\"desktop.ini\")\n",
    "    \n",
    "    \n",
    "    for species in list_of_species:\n",
    "        print(species)\n",
    "        output_2 += f\"\\n{species}\"\n",
    "        individual_species_sequence = ''\n",
    "        list_of_exon_files = os.listdir(f\"{location}/{family_group}/2.Final_output/{species}\")\n",
    "        for i in range(1, 30):\n",
    "            try:\n",
    "                exon_file = SeqIO.parse(f\"{location}/{family_group}/2.Final_output/{species}/Exon_{i}.fa\", 'fasta')\n",
    "                output_2 +=f\"\\tExon_{i}\" \n",
    "            except:\n",
    "                output_2 +=f\"\\t\" \n",
    "                continue\n",
    "            for records in exon_file:\n",
    "                # print(records.id,records.seq)\n",
    "                individual_species_sequence += records.seq \n",
    "        # print(list_of_exon_files)\n",
    "        if \"*\" in individual_species_sequence.translate()[:-1]:\n",
    "            print(individual_species_sequence.translate())\n",
    "            assert False\n",
    "        output += f\">{species}-{gene}\\n{individual_species_sequence}\\n\\n\"\n",
    "        \n",
    "        \n",
    "    # print(output_2)\n",
    "    # assert False\n",
    "with open(f\"{location}/11.Alignment/{gene}.fas\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "\n",
    "with open(f\"{location}/11.Alignment/exon_details.txt\", 'w') as out_file_2:\n",
    "    out_file_2.write(output_2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c035353f-c1cb-44a1-8513-6abe7452db7f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
