{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "999fc4cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3fe06edc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51286a38",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "location = \"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/1.Bombyx_mori/Transcripts\"\n",
    "species = \"Bombyx mori\"\n",
    "\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "list_of_transcripts = os.listdir(location)\n",
    "max_exons = 0\n",
    "longest_transcript = ''\n",
    "data_storage_dictionary = {}\n",
    "\n",
    "for transcript in list_of_transcripts:\n",
    "    if \"desktop\" not in transcript:\n",
    "        data_storage_dictionary[transcript] = {}\n",
    "#         print(transcript)\n",
    "        list_of_exons = os.listdir(f\"{location}/{transcript}\")\n",
    "        \n",
    "        if len(list_of_exons) > max_exons:\n",
    "            max_exons = len(list_of_exons)\n",
    "            longest_transcript = transcript\n",
    "        for exon_file_name in list_of_exons:\n",
    "            if \"desktop\" not in exon_file_name:\n",
    "#                 print(exon_file_name)\n",
    "                exon = exon_file_name.split(\".\")[0]\n",
    "                exon_data = SeqIO.parse(f\"{location}/{transcript}/{exon_file_name}\", 'fasta')\n",
    "                for records in exon_data:\n",
    "#                     print(records)\n",
    "                    data_storage_dictionary[transcript][exon] = len(records.seq)\n",
    "            \n",
    "            \n",
    "# print(data_storage_dictionary)\n",
    "output = f\"Species\\tTranscript\"\n",
    "list_of_exons = []\n",
    "for i in range(max_exons-1):\n",
    "    list_of_exons.append(f\"Exon{i+1}\")\n",
    "    output += f\"\\tExon{i+1}\"\n",
    "output += f\"\\n{species}\\t{longest_transcript}\"\n",
    "# print(output)\n",
    "for exon in list_of_exons:\n",
    "    output += f\"\\t{data_storage_dictionary[longest_transcript][exon]}\"\n",
    "#     print(key,value)\n",
    "\n",
    "list_of_transcripts.remove(longest_transcript)\n",
    "list_of_transcripts.remove(\"desktop.ini\")\n",
    "\n",
    "for transcript in list_of_transcripts:\n",
    "    print(transcript)\n",
    "#     print(data_storage_dictionary[transcript])\n",
    "    output += f\"\\n{species}\\t{transcript}\"\n",
    "    for key,value in data_storage_dictionary[transcript].items():\n",
    "#         print(key,value)\n",
    "        output += f\"\\t{value}\"\n",
    "    \n",
    "                \n",
    "        \n",
    "#     break\n",
    "\n",
    "\n",
    "print(output)\n",
    "# print(list_of_exons)\n",
    "                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa00096c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
