{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a56f83a5",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "XM_038016647.1\n",
      "202 3672\n",
      "ATGGATAACCTTGAAAACTCTGAGAACACTGACAAAATATCAGATTCGGCATACTCAAACAGCTGCAGTTACAGTCAATCAAGAAGAAG\n",
      "89\n",
      "not there\n"
     ]
    }
   ],
   "source": [
    "species = \"Bombyx mori\"\n",
    "location = \"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/1.Per period[Bombyx mori]\"\n",
    "\n",
    "import os\n",
    "\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'saubaralwork@gmail.com'\n",
    "\n",
    "with open(location + \"/Gene_table_processed1.tsv\", 'r') as gene_table_file:\n",
    "    gene_table_list = gene_table_file.readlines()\n",
    "\n",
    "with open(\"C:/Users/sauba/Desktop/TBD_Desktop/genomic.gff\", 'r') as gff_file:\n",
    "    gff_file_list = gff_file.readlines()\n",
    "    \n",
    "mrna_accession = ''\n",
    "# mrna_accession_extracted = \"new\"\n",
    "for i in range(1, len(gene_table_list)):\n",
    "    mrna_accession_extracted = gene_table_list[i].split(\"\\t\")[0]\n",
    "    if mrna_accession != mrna_accession_extracted:\n",
    "        mrna_accession = mrna_accession_extracted\n",
    "        print(mrna_accession)\n",
    "    \n",
    "    mrna_detail = Entrez.efetch(db='nucleotide',id=mrna_accession, rettype = 'gb', retmode= 'text')\n",
    "    mrna_detail_list = (mrna_detail.read().split(\"\\n\"))\n",
    "    for lines in mrna_detail_list:\n",
    "        if  lines.strip().startswith(\"CDS\"):\n",
    "            start,stop = (lines.split(\" \")[-1].split(\"..\"))\n",
    "            print(start,stop)\n",
    "    mrna_sequence = Entrez.efetch(db='nucleotide',id=mrna_accession, rettype = 'fasta', retmode= 'text')\n",
    "    sequence = \"\".join(mrna_sequence.read().split(\"\\n\")[1:])[int(start)-1:int(stop)]\n",
    "    exon = gene_table_list[i].split(\"\\t\")[1]\n",
    "    exon_length = int(gene_table_list[i].split(\"\\t\")[2])\n",
    "    \n",
    "    exon_sequence = sequence[:exon_length]\n",
    "    print (exon_sequence)\n",
    "    print(len(exon_sequence))\n",
    "    list_of_folders = os.listdir(location)\n",
    "    if mrna_accession not in list_of_folders:\n",
    "        print(\"not there\")\n",
    "        os.mkdir(f\"{location}/{mrna_accession}\")\n",
    "    \n",
    "    with open(f\"{location}/{mrna_accession}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(f\">{species}_{mrna_accession}_{exon}\\n{exon_sequence}\")\n",
    "        \n",
    "    \n",
    "#     mrna_access\n",
    "    break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e2af10f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "def singleEntry(singleID):   #the singleID is the accession number\n",
    "    handle = Entrez.efetch(db='nucleotide',id=singleID, rettype = 'fasta', retmode= 'text')\n",
    "    f = open('%s.fasta' % singleID, 'w')\n",
    "    f.write(handle.read())\n",
    "    handle.close()\n",
    "    f.close()\n",
    "\n",
    "\n",
    "#get an id list: this makes a big search and gets a list of id \n",
    "\n",
    "handle = Entrez.esearch(db='nucleotide', term = [\"Poaceae[Orgn] AND als[Gene]\"])\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "print (record[\"IdList\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "47edb168",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "XP_037872575.1\n",
      "12531727-12531550\t13-190\t178\t8865\n",
      "\n",
      "4\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_9664\\2087226368.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     27\u001b[0m         \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_table_line\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     28\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_table_line\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m4\u001b[0m \u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 29\u001b[1;33m             \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_table_line\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m5\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     30\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     31\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_table_line\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m7\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_table_line\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m6\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "import re \n",
    "species = \"1.Per period[Bombyx mori]\"\n",
    "switch = 0\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table.txt\",\"r\") as gene_table_file:\n",
    "    gene_table_list = gene_table_file.readlines()\n",
    "    \n",
    "while i < (len(gene_table_list)):\n",
    "    \n",
    "    if \"Exon table for  mRNA  \" in gene_table_list[i]:\n",
    "        gene_table_line = gene_table_list[i]\n",
    "        while re.search(\"[ ]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"  \",\" \")\n",
    "            \n",
    "        mrna_accession = gene_table_line.split(\" \")[4]\n",
    "        protein_accession = gene_table_line.split(\" \")[7].rstrip()\n",
    "        print(protein_accession)\n",
    "        switch = 1\n",
    "        i += 3\n",
    "    if switch == 1:\n",
    "\n",
    "        gene_table_line = gene_table_list[i]\n",
    "#         print(gene_table_line)\n",
    "        while re.search(\"[\\t]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"\\t\\t\",\"\\t\")\n",
    "            print(gene_table_line)\n",
    "        print(len(gene_table_line.split(\"\\t\")))\n",
    "        if (len(gene_table_line.split(\"\\t\")) == 4 ):\n",
    "            print(gene_table_line.split(\"\\t\")[5])\n",
    "        \n",
    "        if len(gene_table_line.split(\"\\t\")) == 7 or len(gene_table_line.split(\"\\t\")) == 6:\n",
    "            \n",
    "            coding_exon_length =  ((gene_table_line.split(\"\\t\")[5]))\n",
    "            genome_start_point,genome_end_point = ((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "#             print((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "#             output += f\"{mrna_accession}\\t{exon_list[exon_counter]}\\t{coding_exon_length.rstrip()}\\t{genome_start_point}\\t{genome_end_point}\\n\"\n",
    "            \n",
    "        \n",
    "#             exon_counter += 1\n",
    "        \n",
    "        if switch == 1 and re.search(\"^mRNA\",gene_table_list[i]):\n",
    "            switch = 0\n",
    "            print(exon_counter)\n",
    "#             output2 += f\"{mrna_accession}\\t{protein_accession}\\t{exon_counter}\\t{alignment_score}\\n\"\n",
    "#             output += \"\\n\\n\"\n",
    "            exon_counter = 0\n",
    "            \n",
    "    i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c0ec13a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "202..3672\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.efetch(db='nucleotide',id=\"XM_038016647.1\", rettype = 'gb', retmode= 'text')\n",
    "handle_list = (handle.read().split(\"\\n\"))\n",
    "for lines in handle_list:\n",
    "    if  lines.strip().startswith(\"CDS\"):\n",
    "        print(lines.split(\" \")[-1])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
