{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "81fcfca7-185d-4284-bb3f-d800723fc4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7b257780-eea0-47d9-b846-654cff8fa8e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9993f28b-5774-4b4f-a38f-63c393b15b08",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length):\n",
    "    for offset in range(3):\n",
    "        translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "        for i in range(len(translated_sequence)):\n",
    "            if len(translated_sequence[i])> 0.8*query_length:\n",
    "                sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "                # print(i+1, offset)\n",
    "                \n",
    "                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                    output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "236f726f-7cfc-4839-871c-505018689085",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_mafft(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    location = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\n",
    "    list_of_files_to_run_mafft_on = os.listdir(location)\n",
    "    for file in list_of_files_to_run_mafft_on:\n",
    "        if file.endswith(\".fa\"):\n",
    "            command = f'\"mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "            # print(command)\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            # os.system(f'{command}')\n",
    "            command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/{file}\" > \"{location}/alignment_clustal_{file}.txt\"'\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            \n",
    "            # os.system(f'{command}')\n",
    "    return(location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "df06ca7c-0540-4e5e-a02b-2da57d4eecfb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def process_mafft_output(mafft_run_folder, error_exon):\n",
    "    list_of_files_in_mafft_run_folder = os.listdir(mafft_run_folder)\n",
    "    score_output = []\n",
    "    score = 99\n",
    "    min_score_sequence = ''\n",
    "    alignment_file = ''\n",
    "    for file in list_of_files_in_mafft_run_folder:\n",
    "        if file.endswith(\".fa.hat2\"):\n",
    "            with io.open(f\"{mafft_run_folder}/{file}\", 'r') as dist_matrix_file:\n",
    "                dist_matrix_list = dist_matrix_file.readlines()\n",
    "            \n",
    "            sequence_name = dist_matrix_list[-2].rstrip().split(\"=\")[1]\n",
    "            distance_score = float(dist_matrix_list[-1].rstrip())\n",
    "            if len(score_output) < 5:\n",
    "                score_output.append( [sequence_name,distance_score])\n",
    "            else:\n",
    "                for i in range(len(score_output)):\n",
    "                    score_at_this_index = score_output[i][1]\n",
    "                    if distance_score < score_at_this_index:\n",
    "                        score_output[i] = [sequence_name,distance_score]\n",
    "                        break\n",
    "            if distance_score < score:\n",
    "                score = distance_score\n",
    "                min_score_sequence = sequence_name\n",
    "                alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "    # print(score_output)\n",
    "    print(f\"min = {min_score_sequence}, {score}\" )\n",
    "    # score_out_merged = '\\n'.join(score_output)\n",
    "    print(f\"5 top scores:\\n{score_output}\")\n",
    "    print(f\"{mafft_run_folder}/{alignment_file}\")\n",
    "    \n",
    "    alignment_file = SeqIO.parse(f\"{mafft_run_folder}/{alignment_file}\", 'fasta')\n",
    "    # print (records.id)\n",
    "    fasta_start_position = 0\n",
    "    fasta_end_position = 0\n",
    "    start_switch = 0\n",
    "    end_switch = 0\n",
    "    alignment_name = ''\n",
    "    for records in alignment_file:\n",
    "        \n",
    "        \n",
    "        \n",
    "        if error_exon in records.id:\n",
    "            gap_counter = 0\n",
    "            base_counter = 0\n",
    "            for current_position in range(len(records.seq)):\n",
    "                sequence_length = len(records.seq) - records.seq.count('-')\n",
    "                # print(f\"fasta_end_position {fasta_end_position} fasta_start_position {fasta_start_position}\")\n",
    "                # print(f\"current_position = {current_position}, {len(records.seq)}\")\n",
    "                # print(records.seq[current_position])\n",
    "                # print(\"fasta_start_position\",fasta_start_position)\n",
    "                # print(\"start_switch\",start_switch)\n",
    "                \n",
    "                # print(gap_counter, base_counter)\n",
    "                # print(5,0.2*sequence_length)\n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and gap_counter > 3 and (len(records.seq[:current_position]) - records.seq[:current_position].count('-')) < (0.1*sequence_length):\n",
    "                #  :\n",
    "                    # print(\"\\n\\nhere\\n\\n\")\n",
    "                    start_switch = 0\n",
    "                    gap_counter = 0\n",
    "\n",
    "                if \"-\" not in records.seq[current_position] and start_switch == 0 :\n",
    "                    fasta_start_position = current_position\n",
    "                    start_switch = 1\n",
    "                \n",
    "                    \n",
    "                \n",
    "                    \n",
    "                if end_switch == 1 and \"-\" not in records.seq[current_position] and (len(records.seq[current_position:]) - records.seq[current_position:].count('-')) > 0.1*sequence_length :\n",
    "                    end_switch = 0\n",
    "                    gap_counter = 0\n",
    "                \n",
    "                    \n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0:\n",
    "                    # print(f\"base_counter {base_counter}\")\n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    fasta_end_position = current_position\n",
    "                    end_switch = 1\n",
    "                if \"-\" in records.seq[current_position]:\n",
    "                    gap_counter += 1\n",
    "                else:\n",
    "                    gap_counter = 0\n",
    "                    base_counter += 1\n",
    "        else:\n",
    "            alignment_name = records.id\n",
    "            if fasta_end_position == 0:\n",
    "                \n",
    "                fasta_end_position = current_position\n",
    "            end_switch = 1\n",
    "            start_switch = 1\n",
    "\n",
    "        if start_switch == 1 and end_switch == 1:\n",
    "            print(fasta_start_position, fasta_end_position)\n",
    "            gene_sequence = records.seq[fasta_start_position:fasta_end_position]\n",
    "            print(f\"{records.id}\\n{gene_sequence}\")\n",
    "    return(gene_sequence, alignment_name)\n",
    "\n",
    "# process_mafft_output(mafft_run_folder, error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "87615294-93f0-483b-b9f7-fad66730cd1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n",
    "def check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    \n",
    "    list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/\")\n",
    "    \n",
    "    if error_exon not in list_of_exon_directories:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "    elif error_exon in list_of_exon_directories:\n",
    "        list_of_folders_1 = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        if \"for_alignment\" not in list_of_folders_1:\n",
    "            os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        for file in list_of_files:\n",
    "            os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "50742e4c-f9ab-433d-96a1-bde1f9d1c255",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "    print(\"Getting Gene\")\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp\")\n",
    "\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    # subprocess.run(\"pwd\")\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3cb097ff-9369-40fd-9ae5-7587c792ba31",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mafft_process(previous_exon_coordinates,\n",
    "                  next_exon_coordinates,\n",
    "                  current_exon_coordinates,\n",
    "                  query_species,\n",
    "                  query_transcript,\n",
    "                  query_location,\n",
    "                  annotated_genome_location,\n",
    "                 genome_location):\n",
    "    query_exon = current_exon_coordinates[6].split(\"query\")[-1][1:]\n",
    "    error_exon = query_exon\n",
    "    # print(\"AA\")\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    upstream_exon_line, downstream_exon_line = previous_exon_coordinates.split(\",\"),next_exon_coordinates.split(\",\")\n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    \n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "        original_query = query_file_list[1]\n",
    "        \n",
    "    query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "    \n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "    # print(annotated_species_name)\n",
    "\n",
    "    # print(annotated_genome_location)\n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    # assert False\n",
    "    check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon)\n",
    "    \n",
    "    \n",
    "    make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length)\n",
    "    \n",
    "    return(annotated_species_name, error_exon,species, genome_file,gene_sequence,left_overhang,right_overhang,scaffold, original_query_name,original_query )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e3790f71-89c6-4326-a178-d4147df81ed0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990231.1,6338200,6338283,1,Y,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_2_query_Exon_2,10,22,27\n",
      "\n",
      "1\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set3_frame0, 0.394\n",
      "5 top scores:\n",
      "[['set1_frame0', 1.655], ['set3_frame0', 0.394], ['set5_frame1', 1.655], ['set28_frame1', 1.677], ['set88_frame1', 1.688]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_2/for_alignment/alignment_Exon_2_translated_genomic_sequence_3_frame0.fa.txt\n",
      "18 45\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_2\n",
      "HSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "18 45\n",
      "set3_frame0\n",
      "HSSKSTHSESNSSGSSGYGGKPDTETS\n",
      "Original query = HSSKSTHSGSNSSGSSGYGGKPSTSGY\n",
      "New query = HSSKSTHSESNSSGSSGYGGKPDTETS\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990231.1,0,0,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_1_query_Exon_3,0,0,53\n",
      "\n",
      "1\n",
      "5\n",
      "6\n",
      "7\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Getting Gene\n",
      "min = set20_frame0, 0.992\n",
      "5 top scores:\n",
      "[['set16_frame1', 1.796], ['set20_frame0', 0.992], ['set44_frame0', 1.762], ['set28_frame1', 1.808], ['set86_frame1', 1.844]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_3/for_alignment/alignment_Exon_3_translated_genomic_sequence_20_frame0.fa.txt\n",
      "47 96\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_1_query_Exon_3\n",
      "GQPPEKRTKDKEAKKKKKVKIETNPTETKKENEWPPAPVFESPKEDIKD\n",
      "47 96\n",
      "set20_frame0\n",
      "SSNQENRIKDKEVKKKKKMLIDIQSTGVEKENEWPLPPVLEAPKED-KD\n",
      "Original query = NNLGQPPEKRTKDKEAKKKKKVKIETNPTETKKENEWPPAPVFESPKEDIKD\n",
      "New query = SSNQENRIKDKEVKKKKKMLIDIQSTGVEKENEWPLPPVLEAPKED-KD\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990231.1,0,0,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_0_query_Exon_4,0,0,31\n",
      "\n",
      "2\n",
      "1\n",
      "6\n",
      "7\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Getting Gene\n",
      "min = set5_frame1, 1.59\n",
      "5 top scores:\n",
      "[['set5_frame1', 1.59], ['set64_frame2', 1.64], ['set28_frame1', 1.708], ['set89_frame2', 1.708], ['set82_frame1', 1.716]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_4/for_alignment/alignment_Exon_4_translated_genomic_sequence_5_frame1.fa.txt\n",
      "8 38\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_0_query_Exon_4\n",
      "ARTPPTSTV-QVEKGPESMEIISPEKTCGK\n",
      "8 38\n",
      "set5_frame1\n",
      "ARSPHTQKVIQAEVAAMAANLILKHPNTGR\n",
      "Original query = ARTPPTSTVQVEKGPESMEIISPEKTCGKDE\n",
      "New query = ARSPHTQKVIQAEVAAMAANLILKHPNTGR\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990231.1,0,0,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_5,0,0,25\n",
      "\n",
      "3\n",
      "2\n",
      "1\n",
      "7\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Getting Gene\n",
      "min = set9_frame1, 1.495\n",
      "5 top scores:\n",
      "[['set9_frame1', 1.495], ['set63_frame0', 1.538], ['set80_frame0', 1.549], ['set84_frame1', 1.567], ['set66_frame2', 1.607]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_5/for_alignment/alignment_Exon_5_translated_genomic_sequence_9_frame1.fa.txt\n",
      "0 17\n",
      "set9_frame1\n",
      "----NTNVSISNQIQEE\n",
      "Original query = SSSCNIPIPVSAHVTSRRNSVHSS\n",
      "New query = ----NTNVSISNQIQEE\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6307745,6307876,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_6,2,44,44\n",
      "\n",
      "4\n",
      "3\n",
      "2\n",
      "1\n",
      "8\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6308572,6308775,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_1_query_Exon_8,5,66,67\n",
      "\n",
      "Getting Gene\n",
      "min = set87_frame1, 0.468\n",
      "5 top scores:\n",
      "[['set87_frame1', 0.468], ['set99_frame2', 1.732], ['set96_frame1', 1.772], ['set13_frame2', 1.795], ['set40_frame2', 1.803]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_6/for_alignment/alignment_Exon_6_translated_genomic_sequence_87_frame1.fa.txt\n",
      "0 48\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_6\n",
      "D----GFSCVISMQDGVVMYTASSITTALGFPKDMWVGRSFIDFVHPR\n",
      "0 48\n",
      "set87_frame1\n",
      "NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "Original query = DGFSCVISMQDGVVMYTASSITTALGFPKDMWVGRSFIDFVHPR\n",
      "New query = NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "5\n",
      "4\n",
      "3\n",
      "2\n",
      "1\n",
      "9\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6310006,6310234,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_2_query_Exon_9,3,75,75\n",
      "\n",
      "Getting Gene\n",
      "min = set99_frame2, 0.684\n",
      "5 top scores:\n",
      "[['set99_frame2', 0.684], ['set121_frame2', 1.569], ['set87_frame1', 1.617], ['set89_frame2', 1.634], ['set138_frame2', 1.653]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_7/for_alignment/alignment_Exon_7_translated_genomic_sequence_99_frame2.fa.txt\n",
      "24 49\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_1_query_Exon_7\n",
      "DRSTFASQITRGLAVPKNVNGTQAK\n",
      "24 49\n",
      "set99_frame2\n",
      "DRTTFAAQITNGLNMPKTNNASQVK\n",
      "Original query = DRSTFASQITRGLAVPKNVNGTQAK\n",
      "New query = DRTTFAAQITNGLNMPKTNNASQVK\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6308572,6308775,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_1_query_Exon_8,5,66,67\n",
      "\n",
      "6\n",
      "5\n",
      "4\n",
      "3\n",
      "2\n",
      "1\n",
      "10\n",
      "Limenitis_camilla,LR990224.1,6304231,6304319,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_1,1,29,29\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6310465,6310576,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "Getting Gene\n",
      "min = set107_frame2, 0.894\n",
      "5 top scores:\n",
      "[['set107_frame2', 0.894], ['set44_frame0', 1.824], ['set66_frame0', 1.793], ['set87_frame1', 1.835], ['set138_frame2', 1.845]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_8/for_alignment/alignment_Exon_8_translated_genomic_sequence_107_frame2.fa.txt\n",
      "81 144\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_8\n",
      "GNHVCSMVCRIRRYRGLSSGFGVKDRMVTFMPFMLKFYFKNINDEEGHVIYLVIQATPFFSAF\n",
      "81 144\n",
      "set107_frame2\n",
      "DESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAY\n",
      "Original query = PVLGNHVCSMVCRIRRYRGLSSGFGVKDRMVTFMPFMLKFYFKNINDEEGHVIYLVIQATPFFSAFK\n",
      "New query = DESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAY\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6310006,6310234,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_2_query_Exon_9,3,75,75\n",
      "\n",
      "7\n",
      "11\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_7,1,25,25\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6310992,6311111,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "Getting Gene\n",
      "min = set40_frame0, 0.694\n",
      "5 top scores:\n",
      "[['set9_frame0', 1.851], ['set40_frame0', 0.694], ['set31_frame1', 1.839], ['set37_frame2', 1.858], ['set67_frame2', 1.849]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_9/for_alignment/alignment_Exon_9_translated_genomic_sequence_40_frame0.fa.txt\n",
      "31 106\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_2_query_Exon_9\n",
      "PNEAISNAVPFIIRHSSSGSLEYIDPESVPYLGYLPQDIVEKDALQLYHPADLVYLRQVYETIVKEGVVSRSKPY\n",
      "31 106\n",
      "set40_frame0\n",
      "PNEIVTHPTPFIIKHSAIGNIEHIDIDAVRYLGYLPQEILGKDVLQFYHPGDLVQIRNIYENIIQNGFAPRSKPY\n",
      "Original query = PNEAISNAVPFIIRHSSSGSLEYIDPESVPYLGYLPQDIVEKDALQLYHPADLVYLRQVYETIVKEGVVSRSKPY\n",
      "New query = PNEIVTHPTPFIIKHSAIGNIEHIDIDAVRYLGYLPQEILGKDVLQFYHPGDLVQIRNIYENIIQNGFAPRSKPY\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6310465,6310576,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "8\n",
      "12\n",
      "Limenitis_camilla,LR990224.1,6308572,6308775,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_1_query_Exon_8,5,66,67\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6311464,6311613,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_12,2,44,50\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set49_frame1, 0.388\n",
      "5 top scores:\n",
      "[['set49_frame1', 0.388], ['set36_frame2', 1.642], ['set46_frame1', 1.732], ['set32_frame2', 1.803], ['set43_frame1', 1.816]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_10/for_alignment/alignment_Exon_10_translated_genomic_sequence_49_frame1.fa.txt\n",
      "15 52\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_0_query_Exon_10\n",
      "MMAQNGHYLKLETEWSSFINPWSKKLEFVIGKHFITE\n",
      "15 52\n",
      "set49_frame1\n",
      "MITQNGHYLILESEWSSSVNPWSKRLEFIIGKHYILE\n",
      "Original query = MMAQNGHYLKLETEWSSFINPWSKKLEFVIGKHFITE\n",
      "New query = MITQNGHYLILESEWSSSVNPWSKRLEFIIGKHYILE\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6310992,6311111,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "9\n",
      "13\n",
      "Limenitis_camilla,LR990224.1,6310006,6310234,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_2_query_Exon_9,3,75,75\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Getting Gene\n",
      "min = set23_frame2, 0.73\n",
      "5 top scores:\n",
      "[['set23_frame2', 0.73], ['set74_frame2', 1.697], ['set49_frame0', 1.767], ['set64_frame0', 1.814], ['set66_frame0', 1.799]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_11/for_alignment/alignment_Exon_11_translated_genomic_sequence_23_frame2.fa.txt\n",
      "2 42\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_11\n",
      "GPINPDVFQSPDSENNLKLSEEEKNMAQTLRQNITRIMNE\n",
      "2 42\n",
      "set23_frame2\n",
      "GPSNPDVFQPPDFNNVLKLSEEEIRITVALKQSIMRVMNE\n",
      "Original query = GPINPDVFQSPDSENNLKLSEEEKNMAQTLRQNITRIMNE\n",
      "New query = GPSNPDVFQPPDFNNVLKLSEEEIRITVALKQSIMRVMNE\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6311464,6311613,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_12,2,44,50\n",
      "\n",
      "10\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "Limenitis_camilla,LR990224.1,6310465,6310576,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_0_query_Exon_10,1,37,37\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set41_frame0, 1.009\n",
      "5 top scores:\n",
      "[['set41_frame0', 1.009], ['set96_frame0', 1.776], ['set127_frame0', 1.79], ['set125_frame2', 1.792], ['set58_frame2', 1.816]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_12/for_alignment/alignment_Exon_12_translated_genomic_sequence_41_frame0.fa.txt\n",
      "0 50\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_12\n",
      "ALTKPAEAAKQQMSQRCQDLASFMETLMEDTPKGDEELRLEIHDPDHSCY\n",
      "0 50\n",
      "set41_frame0\n",
      "NLTTPAQAFKQQIDKRNQDIANFVDQFMEDKPKPEEELRVDV---DNCCF\n",
      "Original query = ALTKPAEAAKQQMSQRCQDLASFMETLMEDTPKGDEELRLEIHDPDHSCY\n",
      "New query = NLTTPAQAFKQQIDKRNQDIANFVDQFMEDKPKPEEELRVDV---DNCCF\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "11\n",
      "15\n",
      "16\n",
      "17\n",
      "Limenitis_camilla,LR990224.1,6310992,6311111,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_11,1,40,40\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set52_frame0, 0.236\n",
      "5 top scores:\n",
      "[['set96_frame0', 1.702], ['set37_frame2', 1.757], ['set36_frame1', 1.782], ['set40_frame0', 1.764], ['set52_frame0', 0.236]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_13/for_alignment/alignment_Exon_13_translated_genomic_sequence_52_frame0.fa.txt\n",
      "15 54\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_2_query_Exon_13\n",
      "ERDSVMLGGISPHHDYNDSKSSTETPISYNQLNYNENLQ\n",
      "15 54\n",
      "set52_frame0\n",
      "ERDSVMLGGISPHHD-NDSKSSTETPLSYNQLNYNENLE\n",
      "Original query = ERDSVMLGGISPHHDYNDSKSSTETPISYNQLNYNENLQ\n",
      "New query = ERDSVMLGGISPHHD-NDSKSSTETPLSYNQLNYNENLE\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990248.1,8736156,8736248,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_2_query_Exon_14,3,25,32\n",
      "\n",
      "12\n",
      "16\n",
      "17\n",
      "Limenitis_camilla,LR990224.1,6311464,6311613,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_12,2,44,50\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set47_frame0, 1.304\n",
      "5 top scores:\n",
      "[['set47_frame0', 1.304], ['set84_frame0', 1.668], ['set16_frame0', 1.639], ['set59_frame0', 1.7], ['set24_frame1', 1.718]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_14/for_alignment/alignment_Exon_14_translated_genomic_sequence_47_frame0.fa.txt\n",
      "8 36\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_14\n",
      "YFDSHEPFEEYTVPLPDTKMDMKKPGGA\n",
      "8 36\n",
      "set47_frame0\n",
      "FFKSHEPYEEY---------DFDSGGGA\n",
      "Original query = YFDSHEPFEEYTVPLPDTKMDMKKPGGAS\n",
      "New query = FFKSHEPYEEY---------DFDSGGGA\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990236.1,11950341,11950478,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_2_query_Exon_15,11,43,45\n",
      "\n",
      "13\n",
      "17\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set26_frame1, 1.475\n",
      "5 top scores:\n",
      "[['set57_frame1', 1.778], ['set56_frame0', 1.767], ['set18_frame2', 1.81], ['set17_frame1', 1.796], ['set26_frame1', 1.475]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_15/for_alignment/alignment_Exon_15_translated_genomic_sequence_26_frame1.fa.txt\n",
      "0 53\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_15\n",
      "NF---LSSMGCSSGDYVEGTSMSDSSAVV-LASCSQTSMDDYHQVRLTESLLS\n",
      "0 53\n",
      "set26_frame1\n",
      "HFSKNLSSLEGSTID-----NLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "Original query = NFLSSMGCSSGDYVEGTSMSDSSAVVLASCSQTSMDDYHQVRLTESLLS\n",
      "New query = HFSKNLSSLEGSTID-----NLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6314491,6314662,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_1_rightoh_0_query_Exon_16,1,56,56\n",
      "\n",
      "14\n",
      "13\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set44_frame2, 0.808\n",
      "5 top scores:\n",
      "[['set44_frame2', 0.808], ['set56_frame0', 1.725], ['set227_frame2', 1.726], ['set275_frame2', 1.764], ['set292_frame2', 1.811]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_16/for_alignment/alignment_Exon_16_translated_genomic_sequence_44_frame2.fa.txt\n",
      "19 78\n",
      "00.Heliconius_melpomene_g16970_Frame_1_rightoh_0_query_Exon_16\n",
      "HNAEMEKELVKMHRENRSSNKGDRDKSSNETRQKKKEHLARCNAFYQPPTAALSTLESQ\n",
      "19 78\n",
      "set44_frame2\n",
      "HNADMEKELVKIHNKNRITSKSDREKALHDSLQNKMEHFARCNALYLPTNAELSSSEIQ\n",
      "Original query = HNAEMEKELVKMHRENRSSNKGDRDKSSNETRQKKKEHLARCNAFYQPPTAALSTLESQ\n",
      "New query = HNADMEKELVKIHNKNRITSKSDREKALHDSLQNKMEHFARCNALYLPTNAELSSSEIQ\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "15\n",
      "14\n",
      "13\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set56_frame0, 1.239\n",
      "5 top scores:\n",
      "[['set178_frame2', 1.665], ['set271_frame2', 1.67], ['set56_frame0', 1.239], ['set386_frame1', 1.716], ['set18_frame2', 1.725]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_17/for_alignment/alignment_Exon_17_translated_genomic_sequence_56_frame0.fa.txt\n",
      "0 42\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_1_query_Exon_17\n",
      "PHGIKRASKQLEEPSGHKHRCSPPRLRRRQSPNLVNVSQPTT\n",
      "0 42\n",
      "set56_frame0\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAAT\n",
      "Original query = PHGIKRASKQLEEPSGHKHRCSPPRLRRRQSPNLVNVSQPTTVSPT\n",
      "New query = SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAAT\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990248.1,3517640,3517747,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_1_query_Exon_18,4,32,35\n",
      "\n",
      "16\n",
      "15\n",
      "14\n",
      "13\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_2_query_Exon_13,1,39,39\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set71_frame2, 1.162\n",
      "5 top scores:\n",
      "[['set71_frame2', 1.162], ['set123_frame2', 1.687], ['set20_frame2', 1.718], ['set78_frame2', 1.719], ['set185_frame2', 1.721]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_18/for_alignment/alignment_Exon_18_translated_genomic_sequence_71_frame2.fa.txt\n",
      "9 42\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_18\n",
      "SSPWAGTSVNGMNTFILGLGIPQQMSIMSPMHV\n",
      "9 42\n",
      "set71_frame2\n",
      "APTWQGNTVNNANAF--ALGVPQPISILSPIPM\n",
      "Original query = SSPWAGTSVNGMNTFILGLGIPQQMSIMSPMHVP\n",
      "New query = APTWQGNTVNNANAF--ALGVPQPISILSPIPM\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990248.1,0,0,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_1_query_Exon_19,0,0,36\n",
      "\n",
      "17\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set52_frame1, 1.242\n",
      "5 top scores:\n",
      "[['set52_frame1', 1.242], ['set137_frame2', 1.527], ['set225_frame0', 1.661], ['set160_frame2', 1.67], ['set181_frame2', 1.697]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_19/for_alignment/alignment_Exon_19_translated_genomic_sequence_52_frame1.fa.txt\n",
      "14 47\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_19\n",
      "MFPMYYSPAVNRPISTTEASSGNVNTTQFQGFA\n",
      "14 47\n",
      "set52_frame1\n",
      "VIPVYFGP-ISHPVPAGNSDT-NVNTLQFQAFA\n",
      "Original query = MFPMYYSPAVNRPISTTEASSGNVNTTQFQGFA\n",
      "New query = VIPVYFGP-ISHPVPAGNSDT-NVNTLQFQAFA\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6317815,6317963,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_2_rightoh_0_query_Exon_20,1,46,46\n",
      "\n",
      "18\n",
      "17\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set62_frame0, 0.89\n",
      "5 top scores:\n",
      "[['set62_frame0', 0.89], ['set161_frame0', 1.682], ['set318_frame1', 1.727], ['set124_frame2', 1.762], ['set144_frame2', 1.772]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_20/for_alignment/alignment_Exon_20_translated_genomic_sequence_62_frame0.fa.txt\n",
      "19 69\n",
      "00.Heliconius_melpomene_g16970_Frame_2_rightoh_0_query_Exon_20\n",
      "PCMMFGQAVYGSPFMYSSVDPQVHAMQQNFVNPNIQNFQPLGLSGRNYEE\n",
      "19 69\n",
      "set62_frame0\n",
      "PCMMIGHAIYSSPIIYSAMEPQVYATQQNCTTQN-ANVHNIGLSNRNYEE\n",
      "Original query = PCMMFGQAVYGSPFMYSSVDPQVHAMQQNFVNPNIQNFQPLGLSGRNYEE\n",
      "New query = PCMMIGHAIYSSPIIYSAMEPQVYATQQNCTTQN-ANVHNIGLSNRNYEE\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990225.1,14699432,14699548,1,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_21,7,27,39\n",
      "\n",
      "19\n",
      "18\n",
      "17\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set88_frame0, 1.614\n",
      "5 top scores:\n",
      "[['set88_frame0', 1.614], ['set271_frame0', 1.738], ['set173_frame2', 1.743], ['set199_frame2', 1.771], ['set238_frame2', 1.784]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_21/for_alignment/alignment_Exon_21_translated_genomic_sequence_88_frame0.fa.txt\n",
      "14 65\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_21\n",
      "ACKLTLPMKTGKWFTTNTR--REKTAESQSNVQDGIKTCDVRHSSVGTSNR\n",
      "14 65\n",
      "set88_frame0\n",
      "ACKI-MSMKIGEIQIPELECLKENAGETFQNIPG--QSCDISYSSIYTMDL\n",
      "Original query = ACKLTLPMKTGKWFTTNTRREKTAESQSNVQDGIKTCDVRHSSVGTSNR\n",
      "New query = ACKI-MSMKIGEIQIPELECLKENAGETFQNIPG--QSCDISYSSIYTMDL\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990225.1,0,0,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_22,0,0,53\n",
      "\n",
      "20\n",
      "19\n",
      "18\n",
      "17\n",
      "24\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set173_frame2, 1.088\n",
      "5 top scores:\n",
      "[['set173_frame2', 1.088], ['set223_frame2', 1.721], ['set238_frame2', 1.726], ['set130_frame2', 1.789], ['set166_frame2', 1.806]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_22/for_alignment/alignment_Exon_22_translated_genomic_sequence_173_frame2.fa.txt\n",
      "29 86\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_22\n",
      "VQDQLKRKSHRGNSDGTIEKTDEESSYSSFYSSFFKTESGS-TEESDVRNTNREDGM\n",
      "29 86\n",
      "set173_frame2\n",
      "TKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "Original query = CTERPVQDQLKRKSHRGNSDGTIEKTDEESSYSSFYSSFFKTESGSTEESDVRNTNREDGM\n",
      "New query = TKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990237.1,13692223,13692348,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_23,4,31,42\n",
      "\n",
      "21\n",
      "20\n",
      "19\n",
      "18\n",
      "17\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set225_frame0, 1.417\n",
      "5 top scores:\n",
      "[['set225_frame0', 1.417], ['set243_frame2', 1.697], ['set11_frame2', 1.722], ['set119_frame2', 1.732], ['set17_frame2', 1.736]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_23/for_alignment/alignment_Exon_23_translated_genomic_sequence_225_frame0.fa.txt\n",
      "9 41\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_23\n",
      "KKRRMGDTSLMSNIDTGSTNEPHPMKVTRMEQ\n",
      "9 41\n",
      "set225_frame0\n",
      "RKKRRTDSSL-SDFNTAQPNNPQPRTDNAKEQ\n",
      "Original query = LWQKKRRMGDTSLMSNIDTGSTNEPHPMKVTRMEQ\n",
      "New query = RKKRRTDSSL-SDFNTAQPNNPQPRTDNAKEQ\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6326170,6326271,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_24,1,34,34\n",
      "\n",
      "22\n",
      "21\n",
      "20\n",
      "19\n",
      "18\n",
      "17\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set285_frame1, 0.288\n",
      "5 top scores:\n",
      "[['set285_frame1', 0.288], ['set173_frame2', 1.56], ['set211_frame1', 1.646], ['set120_frame2', 1.656], ['set142_frame2', 1.674]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_24/for_alignment/alignment_Exon_24_translated_genomic_sequence_285_frame1.fa.txt\n",
      "1 35\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_24\n",
      "VCLTSELIYKYQILTKNMEEVLTRDKEKLQKFEQ\n",
      "1 35\n",
      "set285_frame1\n",
      "VCLTSELIYKYQIVTKNMEEVLASDKKKLQSFQQ\n",
      "Original query = VCLTSELIYKYQILTKNMEEVLTRDKEKLQKFEQ\n",
      "New query = VCLTSELIYKYQIVTKNMEEVLASDKKKLQSFQQ\n",
      "Scaffold = LR990224.1, start = 6304231, end = 6304319\n",
      "Limenitis_camilla,LR990224.1,6327527,6327667,0,Y,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_25,19,46,47\n",
      "\n",
      "23\n",
      "22\n",
      "21\n",
      "20\n",
      "19\n",
      "18\n",
      "17\n",
      "25\n",
      "26\n",
      "Limenitis_camilla,LR990224.1,6315012,6315162,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_1_query_Exon_17,7,48,48\n",
      "\n",
      "Limenitis_camilla,LR990224.1,6328366,6328470,0,N,13.Danaus_plexippus_XM_061526465.1_Frame_0_rightoh_0_query_Exon_26,1,31,34\n",
      "\n",
      "Getting Gene\n",
      "min = set238_frame2, 0.594\n",
      "5 top scores:\n",
      "[['set238_frame2', 0.594], ['set166_frame2', 1.717], ['set217_frame2', 1.745], ['set173_frame2', 1.774], ['set115_frame2', 1.792]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result/Limenitis_camilla/Period_gene_genomic_sequence_individual_exon_00.Heliconius_melpomene/Exon_25/for_alignment/alignment_Exon_25_translated_genomic_sequence_238_frame2.fa.txt\n",
      "4 50\n",
      "00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_25\n",
      "PSMVNEQLGQLYLDLQLEGVAARLTLEEGITSSSSSGEETPAKSSK\n",
      "4 50\n",
      "set238_frame2\n",
      "PLLVKQQLNQLQLDLQLQGVTARLTLEEGITSSSSSEAETLSKSNK\n",
      "Original query = PSMVNEQLGQLYLDLQLEGVAARLTLEEGITSSSSSGEETPAKSSK\n",
      "New query = PLLVKQQLNQLQLDLQLQGVTARLTLEEGITSSSSSEAETLSKSNK\n"
     ]
    }
   ],
   "source": [
    "\n",
    "blast_output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result\"\n",
    "# blast_output_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "species_list = os.listdir(blast_output_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "\n",
    "species_list = [\"Argynnis_bischoffii_washingtonia\",\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "species_list = [\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "species_list = [\"Limenitis_camilla\"]\n",
    "for species in species_list:\n",
    "    annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result\"\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Query\"\n",
    "    list_of_query_species = os.listdir(query_location)\n",
    "    if \"desktop.ini\" in list_of_query_species:\n",
    "        list_of_query_species.remove(\"desktop.ini\")\n",
    "    list_of_query_species = [\"00.Heliconius_melpomene\"]\n",
    "    for query_species in list_of_query_species:\n",
    "        list_of_query_transcripts = os.listdir(f\"{query_location}/{query_species}\")\n",
    "        if \"desktop.ini\" in list_of_query_transcripts:\n",
    "            list_of_query_transcripts.remove(\"desktop.ini\")\n",
    "        for query_transcript in list_of_query_transcripts:\n",
    "\n",
    "\n",
    "            genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "\n",
    "\n",
    "\n",
    "            output_coordinate_file = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "            exons_to_check_list = [\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\",\"Exon_13\",\"Exon_14\",\"Exon_15\",\"Exon_16\",\"Exon_17\",\"Exon_18\",\"Exon_19\",\"Exon_20\",\"Exon_21\",\"Exon_22\",\"Exon_23\",\"Exon_24\",\"Exon_25\"]\n",
    "#             exons_to_check_list = [\"Exon_21\"]\n",
    "\n",
    "            list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "            coordinate_file_name = ''\n",
    "        #     print(list_of_files_in_species_folder)\n",
    "            for file_names in list_of_files_in_species_folder:\n",
    "\n",
    "                if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                    coordinate_file_name = file_names\n",
    "            if coordinate_file_name =='':\n",
    "                print(f\"Coordinate file error\")\n",
    "                assert False\n",
    "\n",
    "            with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "                coordinate_file_lines = temp_file_open.readlines()\n",
    "            # print(coordinate_file_lines[0])\n",
    "            for exons_to_check in exons_to_check_list:\n",
    "\n",
    "                for i in range(1,len(coordinate_file_lines)):\n",
    "                    coordinate_file_lines_split = coordinate_file_lines[i].split(\",\")\n",
    "                    current_exon_coordinates = coordinate_file_lines[i].split(\",\")\n",
    "                    if i == 1:\n",
    "                        scaffold_first_exon = coordinate_file_lines[i].split(\",\")[1]\n",
    "                        first_exon_start, first_exon_end = coordinate_file_lines[i].split(\",\")[2],coordinate_file_lines[i].split(\",\")[3]\n",
    "                    # if coordinate_file_lines_split[5] == \"Y\":\n",
    "                    if  coordinate_file_lines_split[6].endswith(exons_to_check):\n",
    "                        if i == 1:\n",
    "                            print(f'First Exon has errors\\n{coordinate_file_lines[i]}')\n",
    "                            assert False\n",
    "                        else:\n",
    "                            print(f\"Scaffold = {scaffold_first_exon}, start = {first_exon_start}, end = {first_exon_end}\")\n",
    "                            print(coordinate_file_lines[i])\n",
    "                            # process_current_exon = input(\"Error Found! Process?\")\n",
    "                            process_current_exon = 'y'\n",
    "                            if process_current_exon.lower()[0] == \"y\":\n",
    "                                if i>2:\n",
    "                                    previous_exon_number = i-2\n",
    "                                else:\n",
    "                                    previous_exon_number = i-1\n",
    "                                if i+2 <=int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = i+2\n",
    "                                else:\n",
    "                                    next_exon_number = i+1\n",
    "                                if i >= int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = int(exons_to_check_list[-1].split(\"_\")[1])\n",
    "#                                     continue\n",
    "\n",
    "                                '''\n",
    "                                the above code ensures that missing exons do not cause issues\n",
    "                                '''\n",
    "\n",
    "                                previous_exon_coordinates = ''\n",
    "                                next_exon_coordinates = ''\n",
    "                                while True:\n",
    "                                    print(previous_exon_number)\n",
    "                                    if coordinate_file_lines[previous_exon_number].split(\",\")[5] == \"N\":\n",
    "                                        previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                        break\n",
    "                                    # else:\n",
    "                                    #     proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[previous_exon_number]}\\nProceed?\")\n",
    "                                    #     if proceed_prompt.lower()[0] == \"y\":\n",
    "                                    #         previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                    #         break\n",
    "                                    previous_exon_number -= 1\n",
    "                                while True:\n",
    "                                    print(next_exon_number)\n",
    "                                    try:\n",
    "                                        if coordinate_file_lines[next_exon_number].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                                            break\n",
    "                                    except:\n",
    "                                        if coordinate_file_lines[next_exon_number-1].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number-1]\n",
    "                                            break\n",
    "                                   \n",
    "                                    next_exon_number += 1\n",
    "\n",
    "\n",
    "                                print(previous_exon_coordinates)\n",
    "                                print(next_exon_coordinates)\n",
    "                                annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query = mafft_process(previous_exon_coordinates,\n",
    "                                                                                                                                  next_exon_coordinates,\n",
    "                                                                                                                                  current_exon_coordinates,\n",
    "                                                                                                                                  query_species,\n",
    "                                                                                                                                  query_transcript,\n",
    "                                                                                                                                  query_location,\n",
    "                                                                                                                                  annotated_genome_location,\n",
    "                                                                                                                                     genome_location)\n",
    "\n",
    "                                mafft_run_folder = run_mafft(annotated_genome_location,annotated_species_name,error_exon)\n",
    "\n",
    "                                possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder, error_exon)\n",
    "\n",
    "                                print(f\"Original query = {original_query}\")\n",
    "                                print(f\"New query = {possible_gene_sequence}\")\n",
    "                                \n",
    "                                list_of_folders_inside_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "                                if \"for_blast\" not in list_of_folders_inside_exon_folder:\n",
    "                                    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                else:\n",
    "                                    list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                    for file in list_of_files:\n",
    "                                        os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/{file}\")\n",
    "\n",
    "\n",
    "                                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\",'w') as query_file:\n",
    "                                    query = f\">Query_{error_exon}_{alignment_name}\\n{possible_gene_sequence}\"\n",
    "                                    query_file.write(query)\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccdf583e-4fa4-4fac-a4d4-98a069f1d391",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307c7dce-21b5-432f-926a-024803e6dad6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
