{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36d22ad3-a85e-44b3-85e6-fc9cb5b48684",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_blast_with_new_query(annotated_genome_location,\n",
    "                             annotated_species_name,\n",
    "                             error_exon, \n",
    "                            query_species,\n",
    "                             genome_location, \n",
    "                             species):\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\"\n",
    "    genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "    out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "\n",
    "    with open(f\"{out_location}/new_query.txt\", 'r') as query_file_open:\n",
    "        query_file_lines = query_file_open.readlines()\n",
    "    if (len(query_file_lines)) == 1:\n",
    "        return (\"Query_error\")\n",
    "    print(len(query_file_lines[1])) \n",
    "    if (len(query_file_lines[1])) < 5:\n",
    "        return (\"Query_error\")\n",
    "    # print(len(query_file_lines[1]))\n",
    "    # genome_fragment_out = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/local_genomic_fragment.fa\"\n",
    "    # with io.open(genome_fragment_out,'w') as out_file:\n",
    "    #     output = f\">Genome_fragment_{error_exon}\\n{gene_sequence}\"\n",
    "    #     out_file.write(output)\n",
    "\n",
    "    local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "    # cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "    # # os.system(f'{cd_command}')\n",
    "    # subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # # print(mkdb_command)\n",
    "    # # os.system(f'{mkdb_command}')\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.htm -html'\n",
    "    # print(blast_command)\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # subprocess.run(f'{blast_command}', shell = True)\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.txt'\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6e7d0eb2-6d4d-44b8-90ab-76c864362837",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bd69b123-ff19-4205-83ee-039f63b0fc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_query_info(query_location,query_species,query_transcript,query_exon):\n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "    return(left_overhang,right_overhang,original_query_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "37864dc3-b190-4ffc-bd3c-ec61c8d7d4d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                       annotated_species_name,\n",
    "                                       error_exon,\n",
    "                                       left_overhang,\n",
    "                                       right_overhang,                                                                                               \n",
    "                                       original_query_name,\n",
    "                                      query_species_original ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "    with open(f\"{blast_location}/new_query.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out_genome_fragment.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "        genome_file = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa\", 'fasta')\n",
    "        print(\"reached here\")\n",
    "        print(f\"Before splice: {start},{stop}\")\n",
    "        acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name,query_species_original)\n",
    "        print(f\"after splice: {start},{stop}\")\n",
    "        for records in genome_file:\n",
    "            old_start = start\n",
    "            old_stop = stop\n",
    "            ag = \"N\"\n",
    "            gt = \"N\"\n",
    "            stop_counter = 0\n",
    "            while True:\n",
    "                print(f\"sequence:\\n{records.seq[start+start_modifier-1:stop]}\")\n",
    "                translated_sequence = records.seq[start+start_modifier-1:stop].translate()\n",
    "                print(f\"sequence:\\n{translated_sequence}\")\n",
    "                if \"*\" in translated_sequence:\n",
    "                    stop_counter +=1\n",
    "                    if ag == \"N\":\n",
    "                        start = old_start + 3*stop_counter\n",
    "                    if gt ==\"N\":\n",
    "                        stop = old_stop - 3*stop_counter\n",
    "                print(f\"left = {records.seq[start-3:start-1]}, right ={(records.seq[stop:stop + 2])}, stop_counter = {stop_counter}\"  ), \n",
    "                if (records.seq[start-3:start-1]).lower() == \"ag\" and ag != \"Y\":\n",
    "                    \n",
    "                    ag = \"Y\"\n",
    "                    \n",
    "                elif ag != \"Y\":\n",
    "                    start -= 3\n",
    "                    \n",
    "                if (records.seq[stop:stop + 2]).lower() == \"gt\" and gt != \"Y\":\n",
    "                    gt = \"Y\"\n",
    "                elif gt != \"Y\":\n",
    "                    stop +=3\n",
    "                if old_start - start > 1000 or stop - old_stop > 1000:\n",
    "                    break\n",
    "                if gt == \"Y\" and ag == \"Y\":\n",
    "                    gt_ag = \"Y\"\n",
    "                    break\n",
    "                if stop_counter > 10:\n",
    "                    gt = \"Y\"\n",
    "                    ag = \"Y\"\n",
    "                \n",
    "        \n",
    "        query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast/new_query_spliced.txt\"\n",
    "        with open(query_location , 'w') as query_file_new:\n",
    "            # sequence_translated = records.seq[start+start_modifier-1:stop]\n",
    "            # print(sequence_translated)\n",
    "            \n",
    "            sequence_translated = records.seq[start+start_modifier-1:stop].translate()\n",
    "            print(sequence_translated)\n",
    "            # proceed_test = input(\"Proceed with this?\")\n",
    "            # while True:\n",
    "            #     if proceed_test.lower()[0] == \"n\":\n",
    "            #         assert False\n",
    "            #     elif proceed_test.lower()[0] == \"y\":\n",
    "            #         break\n",
    "            if \"*\" in sequence_translated:\n",
    "                print(\"Errror in Spliced query\")\n",
    "                assert False\n",
    "            \n",
    "            output = f\">{original_query_name[1:]}\\n{sequence_translated}\"\n",
    "            query_file_new.write(output)\n",
    "\n",
    "        genome_file = get_genome_file(genome_location,species)\n",
    "        genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "        out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "        \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.htm\" -html'\n",
    "        # print(blast_command)\n",
    "        # subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.txt\"'\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "\n",
    "                \n",
    "        # acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name)\n",
    "\n",
    "        if acceptor == \"Y\" and donor  == \"Y\":\n",
    "            splice_prediction = \"Y\"\n",
    "        else:\n",
    "            splice_prediction = \"N\"\n",
    "            \n",
    "        return(start,stop,gt_ag, splice_prediction) \n",
    "        # output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaffold) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        # print(output_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f27f9cc5-c2a8-4cb7-83e1-7104ee982cd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_spiceator_result(start_coordinate, stop_coordinate,annotated_genome_location,annotated_species_name,query_species_original):\n",
    "    results_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/\"\n",
    "#     list_of_files_here = os.listdir(results_location)\n",
    "#     spliceator_results_file = ''\n",
    "#     for files in list_of_files_here:\n",
    "#         if files.startswith(\"Spliceator_results\"):\n",
    "#             spliceator_results_file = files\n",
    "#     if spliceator_results_file == '':\n",
    "#         print(\"Splice file missing\")\n",
    "#         assert False\n",
    "\n",
    "#     with open(f\"{results_location}/{spliceator_results_file}\", 'r') as splice_file:\n",
    "#         splice_file_content = splice_file.readlines()\n",
    "\n",
    "    splice_acceptor_presence = \"N\"\n",
    "    splice_donor_presence = \"N\"\n",
    "    splice_donor = ''\n",
    "    splice_acceptor = ''\n",
    "    start_coordinate = 0\n",
    "    stop_coordinate = 0\n",
    "#     for lines in splice_file_content:\n",
    "#         line_split = lines.split(\";\")\n",
    "#         if line_split[0]==\"Acceptor\" and splice_acceptor_presence == \"N\":\n",
    "#             acceptor_start = int(line_split[1])\n",
    "#             acceptor_end = acceptor_start + len(line_split[3])\n",
    "            \n",
    "#             if start_coordinate >=acceptor_start and start_coordinate <=acceptor_end:\n",
    "#                 splice_acceptor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if start_coordinate >= acceptor_start:\n",
    "#                 #         start_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_acceptor = lines\n",
    "#         if line_split[0]==\"Donor\" and splice_donor_presence == \"N\":\n",
    "#             donor_start = int(line_split[1])\n",
    "#             donor_end = donor_start + len(line_split[3])\n",
    "#             if stop_coordinate >=donor_start and stop_coordinate <=donor_end:\n",
    "#                 splice_donor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if stop_coordinate >=donor_start:\n",
    "#                 #         stop_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_donor = lines\n",
    "\n",
    "    return(splice_acceptor_presence, splice_donor_presence, splice_donor, splice_acceptor)\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "50c956e3-7a49-4f1f-b0e6-bac420e663ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, ag_gt, splice_prediction,original_query_name, query_species ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "    with open(f\"{blast_location}/new_query_spliced.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "    \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "            # print(lines)\n",
    "            # print(query_name)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        print(f\"start_coordinate : {start_coor},stop_coordinate : {stop_coor}\")\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "            \n",
    "        output_format = str(species_name)+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \",\" + ag_gt + \",\" + splice_prediction +\"\\n\"  \n",
    "        # print(Output_Sequence)\n",
    "        return(output_format)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4a79568c-ea02-45a4-88d2-b10f23f091e7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24', 'Exon_25']\n",
      "Limenitis_camilla\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 165,248\n",
      "after splice: 165,248\n",
      "sequence:\n",
      "CATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGA\n",
      "sequence:\n",
      "HSSKSTHSESNSSGSSGYGGKPDTETS\n",
      "left = ag, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "CATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATA\n",
      "sequence:\n",
      "HSSKSTHSESNSSGSSGYGGKPDTETSE\n",
      "left = ag, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "CATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAG\n",
      "sequence:\n",
      "HSSKSTHSESNSSGSSGYGGKPDTETSEY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "HSSKSTHSESNSSGSSGYGGKPDTETSEY\n",
      "[[1, 2]]\n",
      "start_coordinate : 6304396,stop_coordinate : 6304482\n",
      "165 254 Y N\n",
      "Limenitis_camilla,LR990224.1,6304395,6304484,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_2,1,29,29,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_3\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "49\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 714,859\n",
      "after splice: 714,859\n",
      "sequence:\n",
      "agcaGTAATCAAGAAAACAGAATCAAGGATAAAgaagttaagaaaaaaaagaaaatgctaATTGATATTCAATCAACTGGAGTTGAAAAAGAAAACGAATGGCCTCTCCCGCCTGTTTTGGAAGCACCGAAAGAAGACAAGGATG\n",
      "sequence:\n",
      "SSNQENRIKDKEVKKKKKMLIDIQSTGVEKENEWPLPPVLEAPKEDKD\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "SSNQENRIKDKEVKKKKKMLIDIQSTGVEKENEWPLPPVLEAPKEDKD\n",
      "[[1, 1]]\n",
      "start_coordinate : 6304945,stop_coordinate : 6305088\n",
      "714 859 Y N\n",
      "Limenitis_camilla,LR990224.1,6304944,6305089,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_1_query_Exon_3,1,48,48,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_4\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "30\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 168,259\n",
      "after splice: 168,259\n",
      "sequence:\n",
      "GCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGG\n",
      "sequence:\n",
      "ARSPHTQKVIQAEVAAMAANLILKHPNTGR\n",
      "left = CA, right =TA, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTAc\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRY\n",
      "left = ag, right =tc, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActct\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYS\n",
      "left = ag, right =ta, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActcttat\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSY\n",
      "left = ag, right =tt, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActcttatttt\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSYF\n",
      "left = ag, right =aa, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActcttattttaaa\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSYFK\n",
      "left = ag, right =ta, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActcttattttaaatac\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSYFKY\n",
      "left = ag, right =aa, stop_counter = 0\n",
      "sequence:\n",
      "ATAGCTCGAAGTCCACACACTCAGAAAGTAATTCAAGCGGAAGTAGCGGCTATGGCGGCAAACCTGATACTGAAACATCCGAATACAGGTAGGTActcttattttaaatacaaa\n",
      "sequence:\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSYFKYK\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "IARSPHTQKVIQAEVAAMAANLILKHPNTGRYSYFKYK\n",
      "[[2, 0]]\n",
      "start_coordinate : 6304397,stop_coordinate : 6304510\n",
      "165 280 Y N\n",
      "Limenitis_camilla,LR990224.1,6304395,6304510,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_0_query_Exon_4,1,38,38,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_5\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "17\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 500,538\n",
      "after splice: 500,538\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = at, right =GA, stop_counter = 1\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 1\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = ct, right =GA, stop_counter = 2\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 2\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 2\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = at, right =CA, stop_counter = 3\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 3\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 3\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 3\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = ta, right =AT, stop_counter = 4\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 4\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 4\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 4\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 4\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = GC, right =CA, stop_counter = 5\n",
      "sequence:\n",
      "AGCATATCCAATCAA\n",
      "sequence:\n",
      "SISNQ\n",
      "left = ta, right =AT, stop_counter = 5\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 5\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 5\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 5\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 5\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = TA, right =AA, stop_counter = 6\n",
      "sequence:\n",
      "ATATCCAAT\n",
      "sequence:\n",
      "ISN\n",
      "left = GC, right =CA, stop_counter = 6\n",
      "sequence:\n",
      "AGCATATCCAATCAA\n",
      "sequence:\n",
      "SISNQ\n",
      "left = ta, right =AT, stop_counter = 6\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 6\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 6\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 6\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 6\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = CC, right =TC, stop_counter = 7\n",
      "sequence:\n",
      "TCC\n",
      "sequence:\n",
      "S\n",
      "left = TA, right =AA, stop_counter = 7\n",
      "sequence:\n",
      "ATATCCAAT\n",
      "sequence:\n",
      "ISN\n",
      "left = GC, right =CA, stop_counter = 7\n",
      "sequence:\n",
      "AGCATATCCAATCAA\n",
      "sequence:\n",
      "SISNQ\n",
      "left = ta, right =AT, stop_counter = 7\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 7\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 7\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 7\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 7\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = AT, right =AT, stop_counter = 8\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = CC, right =TC, stop_counter = 8\n",
      "sequence:\n",
      "TCC\n",
      "sequence:\n",
      "S\n",
      "left = TA, right =AA, stop_counter = 8\n",
      "sequence:\n",
      "ATATCCAAT\n",
      "sequence:\n",
      "ISN\n",
      "left = GC, right =CA, stop_counter = 8\n",
      "sequence:\n",
      "AGCATATCCAATCAA\n",
      "sequence:\n",
      "SISNQ\n",
      "left = ta, right =AT, stop_counter = 8\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 8\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 8\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 8\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 8\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = AA, right =AG, stop_counter = 9\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = AT, right =AT, stop_counter = 9\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = CC, right =TC, stop_counter = 9\n",
      "sequence:\n",
      "TCC\n",
      "sequence:\n",
      "S\n",
      "left = TA, right =AA, stop_counter = 9\n",
      "sequence:\n",
      "ATATCCAAT\n",
      "sequence:\n",
      "ISN\n",
      "left = GC, right =CA, stop_counter = 9\n",
      "sequence:\n",
      "AGCATATCCAATCAA\n",
      "sequence:\n",
      "SISNQ\n",
      "left = ta, right =AT, stop_counter = 9\n",
      "sequence:\n",
      "gtaAGCATATCCAATCAAATA\n",
      "sequence:\n",
      "VSISNQI\n",
      "left = at, right =CA, stop_counter = 9\n",
      "sequence:\n",
      "aatgtaAGCATATCCAATCAAATACAA\n",
      "sequence:\n",
      "NVSISNQIQ\n",
      "left = ct, right =GA, stop_counter = 9\n",
      "sequence:\n",
      "actaatgtaAGCATATCCAATCAAATACAAGAA\n",
      "sequence:\n",
      "TNVSISNQIQE\n",
      "left = at, right =GA, stop_counter = 9\n",
      "sequence:\n",
      "aatactaatgtaAGCATATCCAATCAAATACAAGAAGAA\n",
      "sequence:\n",
      "NTNVSISNQIQEE\n",
      "left = aa, right =AT, stop_counter = 9\n",
      "sequence:\n",
      "taaaatactaatgtaAGCATATCCAATCAAATACAAGAAGAAATG\n",
      "sequence:\n",
      "*NTNVSISNQIQEEM\n",
      "left = TA, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = AA, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = AT, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = CC, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = TA, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = GC, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = ta, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = at, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "aat\n",
      "sequence:\n",
      "N\n",
      "left = ct, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "actaat\n",
      "sequence:\n",
      "TN\n",
      "left = at, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "aatactaat\n",
      "sequence:\n",
      "NTN\n",
      "left = aa, right =gt, stop_counter = 10\n",
      "sequence:\n",
      "taaaatactaat\n",
      "sequence:\n",
      "*NTN\n",
      "left = AA, right =gt, stop_counter = 11\n",
      "sequence:\n",
      "\n",
      "sequence:\n",
      "\n",
      "left = TA, right =gt, stop_counter = 11\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 0]]\n",
      "start_coordinate : 0,stop_coordinate : 0\n",
      "Error in Blast\n",
      "Limenitis_camilla\n",
      "Processing Exon_6\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "48\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 3503,3646\n",
      "after splice: 3503,3646\n",
      "sequence:\n",
      "aattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = aa, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "taaaattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "*NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = at, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "aattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = aa, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "taaaattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "*NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = at, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "tatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "YLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = at, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "aattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = aa, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "taaaattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "*NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = ta, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "ttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "LQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = at, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "tatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "YLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = at, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "aattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = aa, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "taaaattatttacaggGCGCATTCTCGTGTGTCATGTCGTTGAACGACGGCATAGTCGTCTATGCGCCGCTCTCTATCACAACAACATTGGGTTATCCTAAGGAGATGTGGATTGGGAGATCCTTTATTGACTTTGTTCATCCAAGG\n",
      "sequence:\n",
      "*NYLQGAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "left = ag, right =GT, stop_counter = 4\n",
      "GAFSCVMSLNDGIVVYAPLSITTTLGYPKEMWIGRSFIDFVHPR\n",
      "[[0, 0]]\n",
      "start_coordinate : 6307745,stop_coordinate : 6307876\n",
      "3515 3646 Y N\n",
      "Limenitis_camilla,LR990224.1,6307745,6307876,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_7\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "25\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 3723,3798\n",
      "after splice: 3723,3798\n",
      "sequence:\n",
      "GATCGTACCACTTTCGCAGCACAGATAACCAACGGTCTAAATATGCCGAAAACTAACAATGCTTCTCAAGTTAaag\n",
      "sequence:\n",
      "DRTTFAAQITNGLNMPKTNNASQVK\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "DRTTFAAQITNGLNMPKTNNASQVK\n",
      "[[0, 1]]\n",
      "start_coordinate : 6307953,stop_coordinate : 6308027\n",
      "3723 3798 Y N\n",
      "Limenitis_camilla,LR990224.1,6307953,6308028,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_1_query_Exon_7,1,25,25,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_8\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "63\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 4351,4542\n",
      "after splice: 4351,4542\n",
      "sequence:\n",
      "GACGAAAGTATATGCTCCATAATTTGTCGTATAAGGCTATTCAAGGGCCTCACTACAGGATTTGGTGTCAAAAGTCGAGCAGTGGGCTTCTTGCCTTACCTGATCAAATTCTTCTTAAAAAACATCATTAGTGACAACGGTGCGGACGTGTATTTAGTCATCATGGCTATTCCTATCTTTTCGGCTTATA\n",
      "sequence:\n",
      "DESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAY\n",
      "left = TC, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "CAAGACGAAAGTATATGCTCCATAATTTGTCGTATAAGGCTATTCAAGGGCCTCACTACAGGATTTGGTGTCAAAAGTCGAGCAGTGGGCTTCTTGCCTTACCTGATCAAATTCTTCTTAAAAAACATCATTAGTGACAACGGTGCGGACGTGTATTTAGTCATCATGGCTATTCCTATCTTTTCGGCTTATAATA\n",
      "sequence:\n",
      "QDESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAYN\n",
      "left = GC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "CCTCAAGACGAAAGTATATGCTCCATAATTTGTCGTATAAGGCTATTCAAGGGCCTCACTACAGGATTTGGTGTCAAAAGTCGAGCAGTGGGCTTCTTGCCTTACCTGATCAAATTCTTCTTAAAAAACATCATTAGTGACAACGGTGCGGACGTGTATTTAGTCATCATGGCTATTCCTATCTTTTCGGCTTATAATA\n",
      "sequence:\n",
      "PQDESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAYN\n",
      "left = tC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "CAGCCTCAAGACGAAAGTATATGCTCCATAATTTGTCGTATAAGGCTATTCAAGGGCCTCACTACAGGATTTGGTGTCAAAAGTCGAGCAGTGGGCTTCTTGCCTTACCTGATCAAATTCTTCTTAAAAAACATCATTAGTGACAACGGTGCGGACGTGTATTTAGTCATCATGGCTATTCCTATCTTTTCGGCTTATAATA\n",
      "sequence:\n",
      "QPQDESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAYN\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "QPQDESICSIICRIRLFKGLTTGFGVKSRAVGFLPYLIKFFLKNIISDNGADVYLVIMAIPIFSAYN\n",
      "[[2, 1]]\n",
      "start_coordinate : 6308574,stop_coordinate : 6308774\n",
      "4342 4545 Y N\n",
      "Limenitis_camilla,LR990224.1,6308572,6308775,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_8,1,67,67,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_9\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 5776,6004\n",
      "after splice: 5776,6004\n",
      "sequence:\n",
      "CCTAATGAGATTGTGACACATCCTACaccatttattataaaacattcagcAATTGGTAATATCGAGCATATAGATATTGATGCTGTGCGTTACCTTGGCTACTTGCCCCAAGAAATTCTAGGGAAAGATGTTCTGCAGTTTTACCATCCAGGGGATTTGGTCCAGATACGAAATATATATGAGAATATTATTCAAAATGGATTTGCCCCACGTTCAAAACCTTACAg\n",
      "sequence:\n",
      "PNEIVTHPTPFIIKHSAIGNIEHIDIDAVRYLGYLPQEILGKDVLQFYHPGDLVQIRNIYENIIQNGFAPRSKPY\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "PNEIVTHPTPFIIKHSAIGNIEHIDIDAVRYLGYLPQEILGKDVLQFYHPGDLVQIRNIYENIIQNGFAPRSKPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 6310008,stop_coordinate : 6310232\n",
      "5776 6004 Y N\n",
      "Limenitis_camilla,LR990224.1,6310006,6310234,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_10\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 6235,6346\n",
      "after splice: 6235,6346\n",
      "sequence:\n",
      "ATGATAACACAAAATGGTCACTATCTAATATTAGAATCTGAATGGTCCTCCTCAGTAAATCCCTGGTCAAAAAGACTTGAATTCATCATTGGTAAACATTACATCCTCGAG\n",
      "sequence:\n",
      "MITQNGHYLILESEWSSSVNPWSKRLEFIIGKHYILE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "MITQNGHYLILESEWSSSVNPWSKRLEFIIGKHYILE\n",
      "[[1, 0]]\n",
      "start_coordinate : 6310466,stop_coordinate : 6310576\n",
      "6235 6346 Y N\n",
      "Limenitis_camilla,LR990224.1,6310465,6310576,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_11\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 6762,6881\n",
      "after splice: 6762,6881\n",
      "sequence:\n",
      "GGTCCATCAAACCCAGACGTATTTCAGCCACCAGATTTtaacaatgttttaaaattatcagaGGAAGAAATAAGAATAACAGTAGCATTGAAGCAAAGCATTATGAGAGTAATGAATGAG\n",
      "sequence:\n",
      "GPSNPDVFQPPDFNNVLKLSEEEIRITVALKQSIMRVMNE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "GPSNPDVFQPPDFNNVLKLSEEEIRITVALKQSIMRVMNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 6310992,stop_coordinate : 6311111\n",
      "6762 6881 Y N\n",
      "Limenitis_camilla,LR990224.1,6310992,6311111,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_12\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 7234,7374\n",
      "after splice: 7234,7374\n",
      "sequence:\n",
      "AATCTGACAACACCCGCACAAGCTTTTAAACAGCAAATAGACAAACGTAATCAGGATATCGCTAACTTTGTGGATCAATTTATGGAGGATAAACCCAAGCCAGAAGAAGAACTTCGTGTAGATGTGGATAATTGTTGCTTC\n",
      "sequence:\n",
      "NLTTPAQAFKQQIDKRNQDIANFVDQFMEDKPKPEEELRVDVDNCCF\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "NLTTPAQAFKQQIDKRNQDIANFVDQFMEDKPKPEEELRVDVDNCCF\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 0]]\n",
      "start_coordinate : 6311464,stop_coordinate : 6311604\n",
      "7234 7374 Y N\n",
      "Limenitis_camilla,LR990224.1,6311464,6311604,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_12,1,47,47,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_13\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 8670,8785\n",
      "after splice: 8670,8785\n",
      "sequence:\n",
      "GAACGCGATTCAGTGATGCTTGGAGGTATTTCACCTCACCATGATAACGATAGCAAGTCTAGCACTGAAACGCCTCTTAGCTACAATCAACtgaattataatgaaaatttggagag\n",
      "sequence:\n",
      "ERDSVMLGGISPHHDNDSKSSTETPLSYNQLNYNENLE\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "ERDSVMLGGISPHHDNDSKSSTETPLSYNQLNYNENLE\n",
      "[[0, 2]]\n",
      "start_coordinate : 6312900,stop_coordinate : 6313013\n",
      "8670 8785 Y N\n",
      "Limenitis_camilla,LR990224.1,6312900,6313015,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_2_query_Exon_13,1,38,38,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_14\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "28\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 9252,9311\n",
      "after splice: 9252,9311\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTG\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGA\n",
      "left = AG, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAA\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAW\n",
      "left = AG, right =AG, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGG\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWK\n",
      "left = AG, right =AA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAAT\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKG\n",
      "left = AG, right =AA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAAC\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGI\n",
      "left = AG, right =AA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAAT\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGIT\n",
      "left = AG, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAATGAA\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGITM\n",
      "left = AG, right =TA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAATGAATAA\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGITMN\n",
      "left = AG, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAATGAATAACAA\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGITMNN\n",
      "left = AG, right =TG, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAATGAATAACAATGA\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGITMNNN\n",
      "left = AG, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "TTTTTCAAAAGCCACGAACCATATGAAGAATACGACTTTGATTCTGGAGGTGGAGCCTGGAAAGGAATAACAATGAATAACAATGACAG\n",
      "sequence:\n",
      "FFKSHEPYEEYDFDSGGGAWKGITMNNND\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "FFKSHEPYEEYDFDSGGGAWKGITMNNND\n",
      "[[1, 2]]\n",
      "start_coordinate : 6313483,stop_coordinate : 6313569\n",
      "9252 9341 Y N\n",
      "Limenitis_camilla,LR990224.1,6313482,6313571,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_14,1,29,29,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_15\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "53\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 9675,9821\n",
      "after splice: 9675,9821\n",
      "sequence:\n",
      "catttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = tg, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "tgacatttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "*HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = ca, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "catttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = tg, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "tgacatttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "*HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = tt, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "ttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "FSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = ca, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "catttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = tg, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "tgacatttcagtaaaaatttaTCATCATTGGAGGGCAGTACAATTGATAACTTGTCTGACTCGAGTGGAGTTATTAGGATTTCTTCCACCAACGAAGATACTAACGATGAAAGTGATCAATTCAGATTAACAAAGTATTTACTTCAAAA\n",
      "sequence:\n",
      "*HFSKNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "left = ag, right =GT, stop_counter = 3\n",
      "KNLSSLEGSTIDNLSDSSGVIRISSTNEDTNDESDQFRLTKYLLQ\n",
      "[[1, 2]]\n",
      "start_coordinate : 6313915,stop_coordinate : 6314049\n",
      "9684 9821 Y N\n",
      "Limenitis_camilla,LR990224.1,6313914,6314051,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_2_query_Exon_15,1,45,45,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_16\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "59\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 10261,10438\n",
      "after splice: 10261,10438\n",
      "sequence:\n",
      "CACAATGCAGACATGGAAAAAGAATTGGTTAAGATACACAATAAGAATAGAATAACTAGTAAAAGCGATAGAGAAAAGGCTTTACATGACTCCTTACAGAACAAGATGGAGCATTTCGCACGATGCAACGCGCTCTACCTCCCAACCAACGCTGAATTATCTTCATCGGAAATTCAG\n",
      "sequence:\n",
      "HNADMEKELVKIHNKNRITSKSDREKALHDSLQNKMEHFARCNALYLPTNAELSSSEIQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "HNADMEKELVKIHNKNRITSKSDREKALHDSLQNKMEHFARCNALYLPTNAELSSSEIQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 6314492,stop_coordinate : 6314668\n",
      "10261 10438 Y N\n",
      "Limenitis_camilla,LR990224.1,6314491,6314668,0,N,00.Heliconius_melpomene_g16970_Frame_1_rightoh_0_query_Exon_16,1,59,59,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_17\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "42\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 10797,10923\n",
      "after splice: 10797,10923\n",
      "sequence:\n",
      "TCTCAAGGAGTGAAAAGGTCTTCAACACCTGACGCAGAAGGAAGCGCAAACAAACAGCGGTGTTCATCTACTCGAATCAATAGAACTCAGAGTTCAAATTTGTCAAATGCTCGATCTGCAGCTACTA\n",
      "sequence:\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAAT\n",
      "left = AG, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "TCTCAAGGAGTGAAAAGGTCTTCAACACCTGACGCAGAAGGAAGCGCAAACAAACAGCGGTGTTCATCTACTCGAATCAATAGAACTCAGAGTTCAAATTTGTCAAATGCTCGATCTGCAGCTACTAATA\n",
      "sequence:\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAATN\n",
      "left = AG, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "TCTCAAGGAGTGAAAAGGTCTTCAACACCTGACGCAGAAGGAAGCGCAAACAAACAGCGGTGTTCATCTACTCGAATCAATAGAACTCAGAGTTCAAATTTGTCAAATGCTCGATCTGCAGCTACTAATACAA\n",
      "sequence:\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAATNT\n",
      "left = AG, right =CC, stop_counter = 0\n",
      "sequence:\n",
      "TCTCAAGGAGTGAAAAGGTCTTCAACACCTGACGCAGAAGGAAGCGCAAACAAACAGCGGTGTTCATCTACTCGAATCAATAGAACTCAGAGTTCAAATTTGTCAAATGCTCGATCTGCAGCTACTAATACAACCA\n",
      "sequence:\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAATNTT\n",
      "left = AG, right =TT, stop_counter = 0\n",
      "sequence:\n",
      "TCTCAAGGAGTGAAAAGGTCTTCAACACCTGACGCAGAAGGAAGCGCAAACAAACAGCGGTGTTCATCTACTCGAATCAATAGAACTCAGAGTTCAAATTTGTCAAATGCTCGATCTGCAGCTACTAATACAACCATTG\n",
      "sequence:\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAATNTTI\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SQGVKRSSTPDAEGSANKQRCSSTRINRTQSSNLSNARSAATNTTI\n",
      "[[0, 1]]\n",
      "start_coordinate : 6315027,stop_coordinate : 6315164\n",
      "10797 10935 Y N\n",
      "Limenitis_camilla,LR990224.1,6315027,6315165,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_1_query_Exon_17,1,46,46,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_18\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "33\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 11664,11759\n",
      "after splice: 11664,11759\n",
      "sequence:\n",
      "GCCCCAACTTGGCAAGGGAATACTGTTAACAATGCGAACGCTTTTGCACTAGGTGTTCCACAGCCAATATCTATTCTGAGCCCCATACCTATGC\n",
      "sequence:\n",
      "APTWQGNTVNNANAFALGVPQPISILSPIPM\n",
      "left = AG, right =Aa, stop_counter = 0\n",
      "sequence:\n",
      "GCCCCAACTTGGCAAGGGAATACTGTTAACAATGCGAACGCTTTTGCACTAGGTGTTCCACAGCCAATATCTATTCTGAGCCCCATACCTATGCAag\n",
      "sequence:\n",
      "APTWQGNTVNNANAFALGVPQPISILSPIPMQ\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "APTWQGNTVNNANAFALGVPQPISILSPIPMQ\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[2, 1]]\n",
      "start_coordinate : 6315896,stop_coordinate : 6315991\n",
      "11664 11762 Y N\n",
      "Limenitis_camilla,LR990224.1,6315894,6315992,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_18,1,32,32,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_19\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "33\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 12602,12697\n",
      "after splice: 12602,12697\n",
      "sequence:\n",
      "GTGATTCCTGTGTATTTCGGGCCGATATCACACCCCGTTCCTGCAGGAAACAGTGACACCAACGTAAACACATTACAGTTCCAAGCCTTTGCca\n",
      "sequence:\n",
      "VIPVYFGPISHPVPAGNSDTNVNTLQFQAFA\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "VIPVYFGPISHPVPAGNSDTNVNTLQFQAFA\n",
      "[[2, 1]]\n",
      "start_coordinate : 6316834,stop_coordinate : 6316926\n",
      "12602 12697 Y N\n",
      "Limenitis_camilla,LR990224.1,6316832,6316927,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_1_query_Exon_19,1,31,31,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_20\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 13585,13733\n",
      "after splice: 13585,13733\n",
      "sequence:\n",
      "CCATGCATGATGATTGGACATGCTATTTACAGCTCACCAATAATATACTCCGCAATGGAACCACAAGTTTATGCCACGCAACAAAATTGTACCACACAAAATGCAAACGTACATAACATAGGGTTATCGAACAGAAACTATGAAGAA\n",
      "sequence:\n",
      "PCMMIGHAIYSSPIIYSAMEPQVYATQQNCTTQNANVHNIGLSNRNYEE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PCMMIGHAIYSSPIIYSAMEPQVYATQQNCTTQNANVHNIGLSNRNYEE\n",
      "[[2, 0]]\n",
      "start_coordinate : 6317817,stop_coordinate : 6317963\n",
      "13585 13733 Y N\n",
      "Limenitis_camilla,LR990224.1,6317815,6317963,0,N,00.Heliconius_melpomene_g16970_Frame_2_rightoh_0_query_Exon_20,1,49,49,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_21\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "51\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 14856,14999\n",
      "after splice: 14856,14999\n",
      "sequence:\n",
      "GCGTGTAAAATAATGTCTATGAAGATTGGCGAGATCCAAATCCCCGAATTAGAGTGTTTGAAAGAAAATGCCGGAGAAACTTTCCAGAATATTCCCGGGCAATCCTGCGACATCAGTTATTCTTCTATCTATACTATGGATTTG\n",
      "sequence:\n",
      "ACKIMSMKIGEIQIPELECLKENAGETFQNIPGQSCDISYSSIYTMDL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ACKIMSMKIGEIQIPELECLKENAGETFQNIPGQSCDISYSSIYTMDL\n",
      "[[0, 0]]\n",
      "start_coordinate : 6319086,stop_coordinate : 6319229\n",
      "14856 14999 Y N\n",
      "Limenitis_camilla,LR990224.1,6319086,6319229,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_21,1,48,48,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_22\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "57\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 20540,20710\n",
      "after splice: 20540,20710\n",
      "sequence:\n",
      "ACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "TKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = TC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "CTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "LTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = TT, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "GTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "VLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = CA, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "ACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "TVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = CC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "TCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "STVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = CA, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "ACATCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "TSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = GA, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "CGAACATCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "RTSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = TA, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "GTACGAACATCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "VRTSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = CA, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "ACAGTACGAACATCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "TVRTSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = CC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "GCCACAGTACGAACATCCACAGTTCTCACCAAAGATCGTACTAGTAGGAAAACCCGCCGATGTGAAAATGAAAGGGAAACCACTAGAACTGATGGAGAATCAAACTACTCTTCATTTTATTCATCATTCTTTAAAACTGAATCCAGCAGTACTACTGATGATTGTGATGTTAGAAATAGAAACAGGAATAGTAAACTG\n",
      "sequence:\n",
      "ATVRTSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ATVRTSTVLTKDRTSRKTRRCENERETTRTDGESNYSSFYSSFFKTESSSTTDDCDVRNRNRNSKL\n",
      "[[0, 0]]\n",
      "start_coordinate : 6324743,stop_coordinate : 6324940\n",
      "20513 20710 Y N\n",
      "Limenitis_camilla,LR990224.1,6324743,6324940,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_22,1,66,66,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_23\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "32\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 21438,21530\n",
      "after splice: 21438,21530\n",
      "sequence:\n",
      "CGGAAGAAACGGCGTACTGACTCATCGTTGAGCGATTTTAATACTGCACAACCTAACAACCCCCAGCCTCGGACAGATAATGCGAAGGAACAA\n",
      "sequence:\n",
      "RKKRRTDSSLSDFNTAQPNNPQPRTDNAKEQ\n",
      "left = GT, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "TGTCGGAAGAAACGGCGTACTGACTCATCGTTGAGCGATTTTAATACTGCACAACCTAACAACCCCCAGCCTCGGACAGATAATGCGAAGGAACAA\n",
      "sequence:\n",
      "CRKKRRTDSSLSDFNTAQPNNPQPRTDNAKEQ\n",
      "left = TC, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "TTCTGTCGGAAGAAACGGCGTACTGACTCATCGTTGAGCGATTTTAATACTGCACAACCTAACAACCCCCAGCCTCGGACAGATAATGCGAAGGAACAA\n",
      "sequence:\n",
      "FCRKKRRTDSSLSDFNTAQPNNPQPRTDNAKEQ\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "FCRKKRRTDSSLSDFNTAQPNNPQPRTDNAKEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 6325662,stop_coordinate : 6325760\n",
      "21432 21530 Y N\n",
      "Limenitis_camilla,LR990224.1,6325662,6325760,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_23,1,33,33,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_24\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 21940,22041\n",
      "after splice: 21940,22041\n",
      "sequence:\n",
      "GTTTGCCTCACTTCTGAACTTATTTACAAATATCAAATCGTTACCAAAAATATGGAGGAAGTTCTAGCAAGCGATAAGAAGAAATTACAAAGTTTTCAACAA\n",
      "sequence:\n",
      "VCLTSELIYKYQIVTKNMEEVLASDKKKLQSFQQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "VCLTSELIYKYQIVTKNMEEVLASDKKKLQSFQQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 6326170,stop_coordinate : 6326271\n",
      "21940 22041 Y N\n",
      "Limenitis_camilla,LR990224.1,6326170,6326271,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_24,1,34,34,Y,N\n",
      "\n",
      "Limenitis_camilla\n",
      "Processing Exon_25\n",
      "Limenitis_camilla_coordinates_00.Heliconius_melpomene.csv\n",
      "1 query coordinate\n",
      "46\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 23297,23434\n",
      "after splice: 23297,23434\n",
      "sequence:\n",
      "CCTTTATTAGTTAAACAGCAACTTAACCAACTCCAGCTTGACTTACAATTACAAGGTGTTACGGCTAGACTTACCTTAGAAGAAGGAATAACAAGTTCGAGTAGTTCTGAAGCAGAAACTCTGTCTAAATCGAATAAG\n",
      "sequence:\n",
      "PLLVKQQLNQLQLDLQLQGVTARLTLEEGITSSSSSEAETLSKSNK\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "PLLVKQQLNQLQLDLQLQGVTARLTLEEGITSSSSSEAETLSKSNK\n",
      "[[0, 0]]\n",
      "start_coordinate : 6327527,stop_coordinate : 6327664\n",
      "23297 23434 Y N\n",
      "Limenitis_camilla,LR990224.1,6327527,6327664,0,N,00.Heliconius_melpomene_g16970_Frame_0_rightoh_0_query_Exon_25,1,46,46,Y,N\n",
      "\n",
      "Limenitis_camilla,Exon_5\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess\n",
    "annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Blast_result\"\n",
    "# annotated_genome_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "# species_list = [\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Bicyclus_anynana\"]\n",
    "\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "species_list = [\"Argynnis_bischoffii_washingtonia\",\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "species_list = [\"Hestina_assimilis\"]\n",
    "species_list = [\"Limenitis_camilla\"]\n",
    "blast_error = []\n",
    "# query_species = \n",
    "for annotated_species_name in species_list:\n",
    "    # annotated_species_name = \"Lasiommata_megera\"\n",
    "    species = annotated_species_name\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/8.Heliconiinae_Danainae/1.Query\"\n",
    "    # query_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Query\"\n",
    "    query_species = \"00.Heliconius_melpomene\" \n",
    "    query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "    if \"desktop.ini\" in query_transcript_list:\n",
    "        query_transcript_list.remove(\"desktop.ini\")\n",
    "    \n",
    "    query_transcript = query_transcript_list[0]\n",
    "    \n",
    "    genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "    # genome_location = \"/mnt/g/Genomes_2023-12-26\"\n",
    "    list_of_exons_folders = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    # print(list_of_exons_folders)\n",
    "    error_exon_list = []\n",
    "    for folders in list_of_exons_folders:\n",
    "        \n",
    "        if folders.startswith(\"Exon\"):\n",
    "            if int(folders.split(\"_\")[1]) > 1:\n",
    "                error_exon_list.append(folders)\n",
    "    print(error_exon_list)\n",
    "    # break\n",
    "    \n",
    "    \n",
    "#     error_exon_list = [\"Exon_5\"]\n",
    "    for error_exon in error_exon_list:\n",
    "        print(annotated_species_name)\n",
    "        print(f\"Processing {error_exon}\")\n",
    "        coordinate_location = f\"{annotated_genome_location}/{species}\"\n",
    "        files_in_coordinate_location = os.listdir(coordinate_location)\n",
    "        coordinate_backup = 0\n",
    "        query_coordinate = 0\n",
    "        # print(files_in_coordinate_location)\n",
    "        for file_names in files_in_coordinate_location:\n",
    "            if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                coordinate_backup = 1\n",
    "            if file_names.endswith(f\"_coordinates_{query_species}.csv\"):\n",
    "                print(file_names)\n",
    "                query_coordinate = 1\n",
    "        print(query_coordinate, \"query coordinate\")\n",
    "        if query_coordinate == 0:\n",
    "            print(file_names)\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates_old.csv\" \"{coordinate_location}/{species}_coordinates_{query_species}.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        if coordinate_backup == 0:\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates.csv\" \"{coordinate_location}/{species}_coordinates_old.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        \n",
    "        query_error = run_blast_with_new_query(annotated_genome_location,\n",
    "                                     annotated_species_name,\n",
    "                                     error_exon, \n",
    "                                     query_species,\n",
    "                                     genome_location, \n",
    "                                     species )\n",
    "    \n",
    "        if query_error == \"Query_error\":\n",
    "            print(query_error)\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        left_overhang,right_overhang,original_query_name = get_query_info(query_location,query_species,query_transcript,error_exon)\n",
    "        \n",
    "        try:\n",
    "            start_coordinate,stop_coordinate,gt_ag, splice_prediction = process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                                                                                       annotated_species_name,\n",
    "                                                                                                       error_exon,\n",
    "                                                                                                       left_overhang,\n",
    "                                                                                                       right_overhang,                                                                                               \n",
    "                                                                                                       original_query_name,\n",
    "                                                                                                      query_species)\n",
    "\n",
    "            new_coordinate_file_line = process_genome_blast_file(annotated_genome_location, \n",
    "                                                         annotated_species_name,\n",
    "                                                         error_exon,\n",
    "                                                         left_overhang,\n",
    "                                                         right_overhang,\n",
    "                                                         gt_ag, \n",
    "                                                         splice_prediction,\n",
    "                                                         original_query_name,\n",
    "                                                            query_species)\n",
    "        except:\n",
    "            print(\"Error in Blast\")\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        print(start_coordinate,stop_coordinate,gt_ag, splice_prediction)\n",
    "        print(new_coordinate_file_line)\n",
    "        output = ''\n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'r') as open_coordinate_file:\n",
    "            coordinate_file_list = open_coordinate_file.readlines()\n",
    "            # print(coordinate_file_list)\n",
    "            # assert False\n",
    "        for lines in coordinate_file_list:\n",
    "            if lines.split(\",\")[6].endswith(error_exon):\n",
    "                lines = new_coordinate_file_line\n",
    "            output += lines\n",
    "        \n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'w') as out_coordinate_file:\n",
    "            out_coordinate_file.write(output)\n",
    "\n",
    "\n",
    "print(\"\\n\".join(blast_error))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c490af29-b58c-4481-8429-8779374d7ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, \"1\", splice_prediction,original_query_name, query_species )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4fff37c-451f-48f4-8a15-eba962efc380",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
