{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2b2fddd4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus\n",
      "GCF_905220415.1_ilColCroc2.1_genomic.fna\n",
      "13001361 13001509\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_2_NC_059568.1_13001361_13001509_left_1_right_1\n",
      "GAGGACTCGCAACTTAAGTGAAAAGAAACGTCGAGACCAGTTCAACATGCTTGTGAATGAGTTGGGTGCAATGGTGTCCACAAACAACAGAAAAATGGACAAGTCAACTGTTCTGAAATCAACTATTTCATTTTTGAAGAACCATAATG\n",
      "13002193 13002374\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_3_NC_059568.1_13002193_13002374_left_2_right_0\n",
      "AAATAACCGTGAGGTCTCGCGCTCATGATGTACAAGAGGATTGGAAACCTGCTTTTCTATCAAATGAGGAATTCACTTATCTTGTTTTGGAGGCCTTGGAGGGTTTTGTTATGGTTTTCTCGGCAACTGGACGAATTTACTACGTATCCGAGGGAATAACCTCATTACTTGGACACAATCCG\n",
      "13003284 13003401\n",
      "0 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_4_NC_059568.1_13003284_13003401_left_0_right_1\n",
      "AGCGATATTCTGAATACTACAATATTTGATCTTACGTATGAAGATGACCGACCGAATTTGTTCAATCTTCTACAGAATTCAGGGCCGACTGTAGATCCAAATCAGCCAATAACAGAAG\n",
      "13005760 13005855\n",
      "2 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_5_NC_059568.1_13005760_13005855_left_2_right_1\n",
      "AGAATGAAATCAAGTTCCAATGTCATTTACAACGGGGCACattggatttccgggatgagaTTACATATGAGCTTGTGGAGTTCAATGGACATTTTC\n",
      "13006241 13006298\n",
      "2 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_6_NC_059568.1_13006241_13006298_left_2_right_2\n",
      "GTGTCAACACGAAACAAATAGATGGCGACGATATGTACCAACTGGATCCCGACTCTAG\n",
      "13006461 13006699\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_7_NC_059568.1_13006461_13006699_left_1_right_1\n",
      "GCTGCTATTCATATGCACTGGAAGGTTGTGCACGCCTCAACTGATCCGAGACGTGTCACTGGTTGACACGAGCCGCAACGAGTTCACTTCGCGGCACAGCCTCGAGTGGAAGTTCCTGTTCCTGGACCATCGCGCTCCGCCTATCATTGGATACTTGCCGTTTGAGGTCTTGGGCACTTCTGGCTATGACTACTACCACTTTGACGACTTGGAGAAGGTCGTCACCTGCCATGAAGCCT\n",
      "13006885 13007032\n",
      "2 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_8_NC_059568.1_13006885_13007032_left_2_right_2\n",
      "TAATGCAAAAGGGCGAGTTGACGTCGTGCTACTACAGATTCCTAACCAAAGGCCAACAATGGATCTGGCTGCAGACACGGTTCTACATCACATACCATCAGTGGAACTCCAAGCCAGAGTTTGTTATGTGTACACATCGAGTTGTGAG\n",
      "13007249 13007429\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_9_NC_059568.1_13007249_13007429_left_1_right_0\n",
      "TTATGCTGACATACAAAAGAGCACAAAACAAGAAATAACAGACATGGAAGTGGCGGATCAGCCAGAAGTCAATCAAACGGTTCTGAAGGAAGCTCCGGAAGAAACCATGGTGGCTATGTCACCATCTTATATGTCGGAGACGAGCGAGGCGTATGGGAACTCCTACCATTCACTATCGCAG\n",
      "13011223 13011399\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_10_NC_059568.1_13011223_13011399_left_0_right_0\n",
      "CCGCCGTCCGTGAAGTCCGCGTCAGCCGGCAGCACGAGCGGCACGCACGCGCACGTGGGCGCGCACACGGCGAGCACGTCGTGGTCGCGCGCGTCGCACACGCGCTATTCGGCCGGCTACGAGAACAGCATGCTGTCGGCCGACTCGCGCTCCTCGCAGCGCTCCAGCTCGAGAgac\n",
      "13012288 13012509\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_11_NC_059568.1_13012288_13012509_left_0_right_0\n",
      "AGCAAAGAAGTGCCGAGCCGCATGGAGGCCCGCGAGACGGACTTCCACAACCGCACGATGTTCGCGCCGCACAAGGCGGTGGAGCCGGCAGTGGTGCCGCAGCACGGGATCGGTGCCCAGTACCTGGAGCCGGCCCCGTACGTTGGGGTCACGGTGCCCGGCGTGCTGCCGATACTGCCCCCGCTGCCGGTCATCGTGTCCTCCGACCAGGCGCAGATACAG\n",
      "13017903 13018037\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_12_NC_059568.1_13017903_13018037_left_0_right_0\n",
      "TGGCCGTGGCAGGAGCAGTTGCAGCGCAAACACGAGGAGTTGCAGCAGATGATCGTGCGCCAGCAGGAGGAGCTGCGGCAAGTGAAGGAGCAGCTGCTGCTCGCCCGCCTAGGGATGCTGCAACCGCTTATCAAT\n",
      "13019595 13019753\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_13_NC_059568.1_13019595_13019753_left_0_right_0\n",
      "GTCCAAAATCCGTACGGGCCAGAGGACGGGCAACAGAACCAGCGCATGCCAGCACAAATCATGTACGAGGGGTCTCCGAGCCGCAGCGTGACGTACCCACAGCCGCCGGGACACTTGCCACACACGCACTCCAACCAGCAGCAGCATCACAATGAGTAA\n",
      "Colias_eurytheme\n",
      "GCA_907164685.1_Colias_eurytheme_alba_genomic.fna\n",
      "176809 176957\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_2_CAJQZN010042375.1_176809_176957_left_1_right_1\n",
      "GAGGACTCGCAACTTAAGTGAAAAGAAACGTCGAGATCAGTTCAACATGCTTGTGAACGAGTTGGGTGCAATGGTGTCCACAAACAACAGAAAAATGGACAAGTCAACAGTTCTGAAATCGACTATTTCATTTTTGAAGAACCATAAGG\n",
      "176126 176307\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_3_CAJQZN010042375.1_176126_176307_left_2_right_0\n",
      "AAATAACCGTAAGGTCTCGCGCTCATGATGTACAAGAGGATTGGAAACCTGCTTTTCTATCAAATGAGGAATTCACTTATCTTGTTTTGGAGGCCTTGGAGGGTTTTGTTATGGTTTTCTCGGCAACTGGACGAATTTACTACGTATCTGAGGGAATAACCTCATTACTTGGACACAATCcg\n",
      "175893 176010\n",
      "0 1\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_4_CAJQZN010042375.1_175893_176010_left_0_right_1\n",
      "AGCGATATTCTGAATACTACAATATTTGATCTTACGTATGAAGATGACCGACCGAATTTGTTCAATCTTCTACAGAATTCAGGGCCGACTGTAGATCCAAATCAGCCAATTACAGAAG\n",
      "173416 173511\n",
      "2 1\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_5_CAJQZN010042375.1_173416_173511_left_2_right_1\n",
      "AGAATGAAATCAAGTTCCAATGTCATTTACAACGGGGCAcattggatttccgggatgagaTTACATATGAGCTTGTTGAGTTCATTGGACATTTTC\n",
      "172938 172995\n",
      "2 2\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_6_CAJQZN010042375.1_172938_172995_left_2_right_2\n",
      "GTGCCAACACGAAACAAACAGATGGCGACGATATGTACCAACTGGATCCCGACTCTAG\n",
      "172356 172594\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_7_CAJQZN010042375.1_172356_172594_left_1_right_1\n",
      "GCTGCTATTCATATGCACTGGTAGGTTGTGCACGCCTCAACTGATCCGGGACGTGTCACTGGTTGACTCGAGCCGCAACGAGTTCACTTCGCGGCACAGCCTCGAGTGGAAGTTCCTGTTCTTGGACCATCGTGCTCCGCCTATTATTGGGTACTTGCCGTTCGAAGTGCTGGGCACTTCTGGCTACGACTACTACCACTTTGACGACTTGGAGAAGGTTGTCACCTGCCATGAAGCTT\n",
      "166180 166327\n",
      "2 2\n",
      "Getting Gene\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 67\u001b[0m\n\u001b[1;32m     64\u001b[0m     error_exons \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mscaffold\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,000,000,0,Y,Error_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mexon\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m,00,00,00\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     65\u001b[0m     \u001b[38;5;28;01mcontinue\u001b[39;00m\n\u001b[0;32m---> 67\u001b[0m gene_sequence \u001b[38;5;241m=\u001b[39m \u001b[43mget_gene_sequence\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     68\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     69\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mgenome_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     70\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mscaffold\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     71\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mstart\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     72\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     73\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43mcomplement\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     74\u001b[0m \u001b[43m                                  \u001b[49m\u001b[43moutput_location\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     77\u001b[0m current_exon \u001b[38;5;241m=\u001b[39m query_name\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m     78\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m current_exon \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m1\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "Cell \u001b[0;32mIn[1], line 27\u001b[0m, in \u001b[0;36mget_gene_sequence\u001b[0;34m(genome_location, species, genome_file, scaffold, gene_start, gene_end, complement, output_location)\u001b[0m\n\u001b[1;32m     23\u001b[0m         os\u001b[38;5;241m.\u001b[39mmkdir(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00moutput_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/0.Temp\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     25\u001b[0m \u001b[38;5;66;03m#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\u001b[39;00m\n\u001b[1;32m     26\u001b[0m \u001b[38;5;66;03m#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\u001b[39;00m\n\u001b[0;32m---> 27\u001b[0m     \u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43msamtools faidx \u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgenome_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgenome_file\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m \u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mscaffold\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m:\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgene_start\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m-\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mgene_end\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m > \u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43moutput_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/0.Temp/temp_genome.fa\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshell\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstderr\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDEVNULL\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     29\u001b[0m     genome \u001b[38;5;241m=\u001b[39m SeqIO\u001b[38;5;241m.\u001b[39mparse(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00moutput_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/0.Temp/temp_genome.fa\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfasta\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     30\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m entries \u001b[38;5;129;01min\u001b[39;00m genome:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:491\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    489\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Popen(\u001b[38;5;241m*\u001b[39mpopenargs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs) \u001b[38;5;28;01mas\u001b[39;00m process:\n\u001b[1;32m    490\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 491\u001b[0m         stdout, stderr \u001b[38;5;241m=\u001b[39m \u001b[43mprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcommunicate\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    492\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m TimeoutExpired \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[1;32m    493\u001b[0m         process\u001b[38;5;241m.\u001b[39mkill()\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1016\u001b[0m, in \u001b[0;36mPopen.communicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1014\u001b[0m         stderr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mread()\n\u001b[1;32m   1015\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mclose()\n\u001b[0;32m-> 1016\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1017\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1018\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1079\u001b[0m, in \u001b[0;36mPopen.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1077\u001b[0m     endtime \u001b[38;5;241m=\u001b[39m _time() \u001b[38;5;241m+\u001b[39m timeout\n\u001b[1;32m   1078\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1079\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1080\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyboardInterrupt\u001b[39;00m:\n\u001b[1;32m   1081\u001b[0m     \u001b[38;5;66;03m# https://bugs.python.org/issue25942\u001b[39;00m\n\u001b[1;32m   1082\u001b[0m     \u001b[38;5;66;03m# The first keyboard interrupt waits briefly for the child to\u001b[39;00m\n\u001b[1;32m   1083\u001b[0m     \u001b[38;5;66;03m# exit under the common assumption that it also received the ^C\u001b[39;00m\n\u001b[1;32m   1084\u001b[0m     \u001b[38;5;66;03m# generated SIGINT and will exit rapidly.\u001b[39;00m\n\u001b[1;32m   1085\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1804\u001b[0m, in \u001b[0;36mPopen._wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1802\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1803\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m  \u001b[38;5;66;03m# Another thread waited.\u001b[39;00m\n\u001b[0;32m-> 1804\u001b[0m (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_try_wait\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1805\u001b[0m \u001b[38;5;66;03m# Check the pid and loop as waitpid has been known to\u001b[39;00m\n\u001b[1;32m   1806\u001b[0m \u001b[38;5;66;03m# return 0 even without WNOHANG in odd situations.\u001b[39;00m\n\u001b[1;32m   1807\u001b[0m \u001b[38;5;66;03m# http://bugs.python.org/issue14396.\u001b[39;00m\n\u001b[1;32m   1808\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m pid \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid:\n",
      "File \u001b[0;32m/usr/lib/python3.8/subprocess.py:1762\u001b[0m, in \u001b[0;36mPopen._try_wait\u001b[0;34m(self, wait_flags)\u001b[0m\n\u001b[1;32m   1760\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"All callers to this function MUST hold self._waitpid_lock.\"\"\"\u001b[39;00m\n\u001b[1;32m   1761\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1762\u001b[0m     (pid, sts) \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwaitpid\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpid\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwait_flags\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1763\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mChildProcessError\u001b[39;00m:\n\u001b[1;32m   1764\u001b[0m     \u001b[38;5;66;03m# This happens if SIGCLD is set to be ignored or waiting\u001b[39;00m\n\u001b[1;32m   1765\u001b[0m     \u001b[38;5;66;03m# for child processes has otherwise been disabled for our\u001b[39;00m\n\u001b[1;32m   1766\u001b[0m     \u001b[38;5;66;03m# process.  This child is dead, we can't get the status.\u001b[39;00m\n\u001b[1;32m   1767\u001b[0m     pid \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpid\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"5.Coliadinae\"\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}\"\n",
    "make_output_folder(output_location)\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "if 'desktop.ini' in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Heliconius_charithonia\"]\n",
    "# list_of_species = [\"Maniola_hyperantus\"]\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "        \n",
    "            \n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "            print(left_oh, right_oh)\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            query_name = coordinate_line_split[6]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            continue\n",
    "            \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "       \n",
    "        current_exon = query_name.split('_')[-1]\n",
    "        if current_exon == \"1\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "            print(translated)\n",
    "            if \"*\" in translated:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "        \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "        elif current_exon == \"16\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence[:-int(left_oh)].reverse_complement.translate()\n",
    "                print(gene_sequence)\n",
    "            else:\n",
    "                translated = gene_sequence[int(left_oh):].translate()\n",
    "                print(gene_sequence)\n",
    "\n",
    "            if \"*\" in translated[:-1]:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                print(translated)\n",
    "            else:\n",
    "                \n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "#         assert False\n",
    "        else:\n",
    "            exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "            print(exon_output)\n",
    "\n",
    "            write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
