{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3247b82e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "086c0ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    print(list_of_files_in_genome_folder)\n",
    "    if \"Temp\" in list_of_files_in_genome_folder:\n",
    "        subprocess.run(f'rm -r \"{genome_location}/{species}/Temp\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{genome_location}/{species}/Temp\")\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9d4c6ece",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_query_folder(annotated_species_location,gene_name,species):\n",
    "    import time\n",
    "\n",
    "\n",
    "    list_of_files = os.listdir(annotated_species_location)\n",
    "    if species in list_of_files:\n",
    "        subprocess.run(f'rm -r \"{annotated_species_location}/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    time.sleep(2.5)\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts\")\n",
    "    os.mkdir(f\"{annotated_species_location}/{species}/Transcripts/{gene_name}\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cb40a07",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCF_905147365.1_ilAriAges1.1_genomic.fna', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nhr', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nin', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nsq', 'GCF_905147365.1_ilAriAges1.1_genomic.gff', 'MD5SUMS']\n",
      "['GCF_905147365.1_ilAriAges1.1_genomic.fna', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nhr', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nin', 'GCF_905147365.1_ilAriAges1.1_genomic.fna.nsq', 'GCF_905147365.1_ilAriAges1.1_genomic.gff', 'MD5SUMS', 'Temp']\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "\n",
    "scaffold = \"\"\n",
    "gene_name = \"XM_042131467.1\"\n",
    "species = \"Aricia_agestis\"\n",
    "\n",
    "genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "genome_file = get_genome_file(genome_location,species)\n",
    "annotated_species_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\"\n",
    "query_species = species\n",
    "create_query_folder(annotated_species_location,gene_name,species)\n",
    "list_of_files = os.listdir(f\"{genome_location}/{species}\")\n",
    "print(list_of_files)\n",
    "for file_names in list_of_files:\n",
    "    if file_names.endswith(\"_genomic.gff\"):\n",
    "        gff3_file = file_names\n",
    "\n",
    "with open(f\"{genome_location}/{species}/{gff3_file}\") as gff_file_open:\n",
    "    gff_file_list = gff_file_open.readlines()\n",
    "complement = ''\n",
    "exon_count = 1\n",
    "gene_found = 0\n",
    "exon_dictionary = {}\n",
    "for line in gff_file_list:\n",
    "    if gene_name in line:\n",
    "        if \"CDS\" in line.split(\"\\t\"):\n",
    "\n",
    "            print(line)\n",
    "            gene_found = 1\n",
    "            if complement == '':\n",
    "                complement = line.split(\"\\t\")[6]\n",
    "            exon_start = line.split(\"\\t\")[3]\n",
    "            exon_stop = line.split(\"\\t\")[4]\n",
    "            scaffold = line.split(\"\\t\")[0]\n",
    "            exon_dictionary[f\"Exon_{exon_count}\"] = [exon_start,exon_stop]\n",
    "            exon_count += 1\n",
    "        if \"exon\" in line.split(\"\\t\") and gene_found == 1:\n",
    "            break\n",
    "print(complement)\n",
    "print(exon_dictionary)\n",
    "if complement == \"+\":\n",
    "    exon_counter = 1\n",
    "elif complement == \"-\":\n",
    "    exon_counter = len(exon_dictionary)\n",
    "else:\n",
    "    print(\"Error!!\")\n",
    "    assert False\n",
    "\n",
    "# list_of_query_species = os.listdir(query_location)\n",
    "# if query_species not in list_of_query_species:\n",
    "#     os.mkdir(f\"{annotated_species_location}/{query_species}\")\n",
    "# list_of_folders_here = os.listdir(f\"{annotated_species_location}/{query_species}\")\n",
    "# if \"Transcripts\" not\n",
    "#     os.mkdir(f\"{annotated_species_location}/{query_species}/{gene_name}\")\n",
    "\n",
    "print_out = ''\n",
    "print(\"exon counter\", exon_counter)        \n",
    "for i in range(1,len(exon_dictionary)+1):\n",
    "    \n",
    "    exon_start, exon_stop = exon_dictionary[f\"Exon_{exon_counter}\"]\n",
    "#     print(exon_start,exon_stop)\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{exon_start}-{exon_stop} > \"{genome_location}/{species}/Temp/temp_gene.fa\"', shell = True)\n",
    "    exon_sequence_file = SeqIO.parse(f\"{genome_location}/{species}/Temp/temp_gene.fa\", \"fasta\")\n",
    "    sequence = ''\n",
    "    \n",
    "    for sequences in exon_sequence_file:\n",
    "#         print(sequences.id)\n",
    "        if complement == \"-\":\n",
    "            sequence = sequences.seq.reverse_complement() \n",
    "            \n",
    "#             print(sequences.seq.reverse_complement().translate())\n",
    "        else:\n",
    "            sequence = sequences.seq\n",
    "#             print(sequences.seq)\n",
    "#             print(sequences.seq.translate())\n",
    "        print_out += str(len(sequence))+\"\\t\"\n",
    "        output = f\">{query_species}_{gene_name}_Exon{i}\\n{sequence}\"\n",
    "    with open(f\"{annotated_species_location}/{query_species}/Transcripts/{gene_name}/Exon{i}.fa\", 'w') as outfile:\n",
    "        outfile.write(output)\n",
    "    print(output)\n",
    "    if complement == \"-\":\n",
    "        exon_counter -= 1\n",
    "    else:\n",
    "        exon_counter += 1\n",
    "                                            \n",
    "#         if gene_location in line.split(\"\\t\"):\n",
    "#           print(line)\n",
    "#           break\n",
    "print(print_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "111273de",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
