{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d6d836d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "35e8295b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "93710a78",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_output_location = \"H:/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/1.Blast_output\"\n",
    "annotated_genome_location = \"H:/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\"\n",
    "reblast_location = \"H:/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast\"\n",
    "\n",
    "species = \"Pararge_aegeria\"\n",
    "coordinate_file = f\"{species}_coordinates.csv\"\n",
    "good_exon_upstream = 1\n",
    "good_exon_downstream = 3\n",
    "error_exon = \"Exon_2\"\n",
    "\n",
    "genome_location = \"F:/Genomes_2023-12-26\"\n",
    "list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "for file in list_of_files_in_genome_folder:\n",
    "    if file.endswith(\"_genomic.fna\"):\n",
    "        genome_file = file\n",
    "# print(genome_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c2e69b6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16927135 16934367\n",
      "['Pararge_aegeria', 'NC_053208.1', '16934367', '16934455', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_1', '1', '29', '29\\n']\n",
      "['Pararge_aegeria', 'NC_053208.1', '16926960', '16927135', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_3', '1', '57', '58\\n']\n"
     ]
    }
   ],
   "source": [
    "with open(f\"{blast_output_location}/{species}/{coordinate_file}\", 'r') as coordinate_open_file:\n",
    "    coordinate_file_lines = coordinate_open_file.readlines()\n",
    "    \n",
    "upstream_exon_line, downstream_exon_line = coordinate_file_lines[good_exon_upstream].split(\",\"), coordinate_file_lines[good_exon_downstream].split(\",\")\n",
    "if (\"Y\" in upstream_exon_line[5]) or (\"Y\" in downstream_exon_line[5]):\n",
    "    print(\"ERROR!!!\")\n",
    "    print(f\"{coordinate_file_lines[0]}\\n{coordinate_file_lines[1]}\\n{coordinate_file_lines[-1]}\")\n",
    "    assert False\n",
    "          \n",
    "\n",
    "complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "if complement == \"0\":\n",
    "    gene_start = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    gene_end = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "\n",
    "if complement == \"1\":\n",
    "    gene_start = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    gene_end = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "\n",
    "print(gene_start, gene_end)\n",
    "\n",
    "\n",
    "print(upstream_exon_line)\n",
    "print(downstream_exon_line)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f02904da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_053208.1\n"
     ]
    }
   ],
   "source": [
    "genome = SeqIO.parse(f\"{genome_location}/{species}/{genome_file}\", \"fasta\")\n",
    "for entries in genome:\n",
    "    if scaffold == entries.id:\n",
    "        print(scaffold)\n",
    "        gene_sequence = entries.seq[gene_start-1:gene_end]\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7508f49f",
   "metadata": {},
   "outputs": [],
   "source": [
    "list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "annotated_species_name = ''\n",
    "for annotated_species in list_of_annotated_genomes:\n",
    "    if annotated_species.endswith(species):\n",
    "        annotated_species_name = (annotated_species)\n",
    "        break\n",
    "if annotated_species_name == '':\n",
    "    print(f\"Error with annotated species name\")\n",
    "    assert False\n",
    "    \n",
    "list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "\n",
    "with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}_genomic_sequence.fa\",'w') as out_file:\n",
    "    output = f\">{species}_genome_{scaffold}_{gene_start}_{gene_end}\\n{gene_sequence}\"\n",
    "    out_file.write(output)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e6dc7553",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/5.Bicyclus_anynana/XM_024088150.2/query_protein.fa\" -db \"Exon_2_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/Pararge_aegeria/Pararge_aegeria_Exon_2_re_blast_out.htm\" -html \n",
      "\n",
      " tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/5.Bicyclus_anynana/XM_024088150.2/query_protein.fa\" -db \"Exon_2_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/Pararge_aegeria/Pararge_aegeria_Exon_2_re_blast_out.txt\"\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "query_species = \"5.Bicyclus_anynana\"\n",
    "query_transcript = \"XM_024088150.2\"\n",
    "\n",
    "list_of_species_folder_in_re_blast = os.listdir(reblast_location)\n",
    "\n",
    "if species not in list_of_species_folder_in_re_blast:\n",
    "    os.mkdir(f\"{reblast_location}/{species}\")\n",
    "\n",
    "# output_tblastn = f'tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/{query_species}/{query_transcript}/query_protein.fa\" -db \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species/{annotated_species_name}/Period_gene_genomic_sequence/period_gene_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/{species}/{species}_re_blast_out.htm\" -html \\n\\n tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/{query_species}/{query_transcript}/query_protein.fa\" -db \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species/{annotated_species_name}/Period_gene_genomic_sequence/period_gene_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/{species}/{species}_re_blast_out.txt\"\\n\\n'\n",
    "\n",
    "output_tblastn = f'tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/{query_species}/{query_transcript}/query_protein.fa\" -db \"{error_exon}_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/{species}/{species}_{error_exon}_re_blast_out.htm\" -html \\n\\n tblastn -seg no -query \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query/{query_species}/{query_transcript}/query_protein.fa\" -db \"{error_exon}_genomic_sequence.fa\" -num_alignments 3 -out \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/4.Re_Blast/{species}/{species}_{error_exon}_re_blast_out.txt\"\\n\\n'\n",
    "print(output_tblastn)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
