{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "619b7f03",
   "metadata": {},
   "outputs": [],
   "source": [
    "family_group = \"8.Heliconiinae_Danainae\"\n",
    "pad_seq = 10000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e7832546",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "   \n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp\")\n",
    "\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\"')\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/temp/temp_genome.fa\", \"fasta\")\n",
    "    # genome = SeqIO.parse(\"/mnt/c/Users/sauba/Downloads/Compressed/ncbi_dataset_7/ncbi_dataset/data/GCA_917051295.2/GCA_917051295.2_ilEreLige1.2_genomic.fna\", 'fasta')\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    return (gene_sequence)\n",
    "\n",
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)   \n",
    "\n",
    "def get_sequence(blast_output_location,genome_location, species, genome_file,annotated_genome_location,annotated_species_name):\n",
    "    list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "    coordinate_file_name = ''\n",
    "    for file_names in list_of_files_in_species_folder:\n",
    "        if file_names.endswith(\"_coordinates.csv\"):\n",
    "            coordinate_file_name = file_names\n",
    "\n",
    "    with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "        coordinate_file_lines = temp_file_open.readlines()\n",
    "\n",
    "    upstream_exon_line =coordinate_file_lines[1].split(\",\")\n",
    "    if upstream_exon_line[5] == \"Y\":\n",
    "        print(\"First Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    downstream_exon_line = coordinate_file_lines[-1].split(\",\")\n",
    "    if downstream_exon_line[5] == \"Y\":\n",
    "        print(\"Last Exon Error!!\")\n",
    "        assert False\n",
    "\n",
    "    \n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        \n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        \n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    padding_error = ''\n",
    "    \n",
    "    if gene_start < pad_seq:\n",
    "        print(f\"Error{gene_start}\\n Changed Padding\")\n",
    "        print(species)\n",
    "        assert False\n",
    "    gene_start -= pad_seq\n",
    "    gene_end +=  pad_seq\n",
    "    \n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    print(\"BB\")\n",
    "    return(gene_sequence,scaffold,gene_start,gene_end,padding_error )\n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2b83123a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_species_folder(output_location, species):\n",
    "    list_of_folders = os.listdir(f\"{output_location}/5.For Email_padded/\")\n",
    "    if species not in list_of_folders:\n",
    "#         subprocess.run(f'rm -r \"{output_location}/5.For Email_padded/{species}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#         import time\n",
    "\n",
    "#         time.sleep(5)\n",
    "        os.mkdir(f\"{output_location}/5.For Email_padded/{species}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0414369",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Danaus_chrysippus\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Danaus_plexippus\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Dircenna_loreta\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Mechanitis_mazaeus\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Mechanitis_messenoides\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Melinaea_marsaeus_rileyi\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Melinaea_menophilus\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Napeogenes_inachia\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Napeogenes_sylphis\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Argynnis_bischoffii_washingtonia\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BB\n",
      "Boloria_euphrosyne\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
      "samtools: /home/saurav/miniconda3/envs/ncbi_datasets/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import io\n",
    "import subprocess\n",
    "genome_location =  \"/mnt/f/Genomes_2023-12-26\"\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(f\"{output_location}\")\n",
    "if \"5.For Email_padded\" not in list_of_folders:\n",
    "    os.mkdir(f\"{output_location}/5.For Email_padded\")\n",
    "\n",
    "blast_output_location = f\"{output_location}/1.Blast_result\"\n",
    "\n",
    "list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove('desktop.ini')\n",
    "\n",
    "error_species = ''\n",
    "# list_of_species.remove(\"Colias_nastes\")\n",
    "# list_of_species = [\"Colias_nastes\"]\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    make_species_folder(output_location, species)\n",
    "    annotated_genome_location = f\"{output_location}/1.Blast_result\"\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "\n",
    "    gene_sequence,scaffold, gene_start, gene_end, padding_error = get_sequence(blast_output_location,genome_location, species, genome_file,\n",
    "                                                               annotated_genome_location,annotated_species_name)\n",
    "    error_species += f\"{padding_error}\\n\"\n",
    "#     print(gene_sequence,scaffold, gene_start, gene_end)\n",
    "    with open(f\"{output_location}/5.For Email_padded/{species}/Genome_fragment.fa\",'w') as out_file:\n",
    "        output = f\">{species}_{scaffold}_{gene_start}_{gene_end}\\n{gene_sequence}\"\n",
    "        out_file.write(output)\n",
    "\n",
    "with open(f\"{output_location}/5.For Email_padded/error_species.txt\",'w') as error_out_file:\n",
    "    error_out_file.write(error_species)\n",
    "    \n",
    "    \n",
    "#     local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "#     cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "#         # # os.system(f'{cd_command}')\n",
    "#     subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "#         # # print(mkdb_command)\n",
    "#         # # os.system(f'{mkdb_command}')\n",
    "#     # print(gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4b9d370",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "# list_of_species = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "# if \"desktop.ini\" in list_of_species:\n",
    "#     list_of_species.remove('desktop.ini')\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Anthocharis_cardamines\"]\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "#     print(gene_start)\n",
    "    output = ''\n",
    "    genome_fragment_file = SeqIO.parse(f\"{output_location}/5.For Email_padded/{species}/Genome_fragment.fa\", \"fasta\")\n",
    "    for sequences in genome_fragment_file:\n",
    "        genome_name = sequences.id\n",
    "        fragment_length = len(sequences.seq)\n",
    "        print(fragment_length)\n",
    "    list_of_variant_files = os.listdir(f\"{output_location}/4.Bed File/{species}/\")\n",
    "    if 'desktop.ini' in list_of_variant_files:\n",
    "        list_of_variant_files.remove('desktop.ini')\n",
    "#     start = 0\n",
    "    \n",
    "    for variant in list_of_variant_files:    \n",
    "        with open(f\"{output_location}/4.Bed File/{species}/{variant}\", 'r') as var_file:\n",
    "            var_list = var_file.readlines()\n",
    "        \n",
    "        if int((var_list[0].split(\"\\t\")[1])) <int((var_list[1].split(\"\\t\")[1])):\n",
    "            complement = 0\n",
    "            gene_start = int((var_list[0].split(\"\\t\")[1]))\n",
    "        else:\n",
    "            complement = 1\n",
    "            gene_start = int((var_list[-1].split(\"\\t\")[1])) \n",
    "        output = ''\n",
    "#         print(var_list)\n",
    "        for gene_line in  var_list:\n",
    "            \n",
    "            gene_line_split = gene_line.split(\"\\t\")\n",
    "#             print(gene_line, var_list)\n",
    "            gene_line_split[0] = genome_name\n",
    "            if complement == 0:\n",
    "                gene_line_split[1] = str(int(gene_line_split[1]) - gene_start + pad_seq)\n",
    "                gene_line_split[2] = str(int(gene_line_split[2]) - gene_start + pad_seq)\n",
    "            if complement == 1:\n",
    "                old_gene_line_split_1 = gene_line_split[1]\n",
    "                gene_line_split[1] = str(fragment_length - (int(gene_line_split[2]) - gene_start + pad_seq))\n",
    "                gene_line_split[2] = str(fragment_length - (int(old_gene_line_split_1) - gene_start + pad_seq))\n",
    "            output += \"\\t\".join(gene_line_split)\n",
    "        \n",
    "#         print(output)\n",
    "        with open(f\"{output_location}/5.For Email_padded/{species}/{variant}\", 'w') as out_file:\n",
    "            out_file.write(output)\n",
    "#         print(complement, gene_start)\n",
    "#         assert False\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f157c3a4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "098ed3a0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
