{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c14aca49",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_gene_sequence(genome_location, \n",
    "                      species, \n",
    "                      genome_file, \n",
    "                      scaffold,\n",
    "                      gene_start,\n",
    "                      gene_end,\n",
    "                      complement,\n",
    "                      output_location):\n",
    "\n",
    "    \n",
    "    print(\"Getting Gene\")\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"0.Temp\" not in list_of_folders:\n",
    "        os.mkdir(f\"{output_location}/0.Temp\")\n",
    "\n",
    "#     subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "#     print(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end}')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{output_location}/0.Temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "    genome = SeqIO.parse(f\"{output_location}/0.Temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)\n",
    "def make_output_folder(output_location):\n",
    "    list_of_folders = os.listdir(output_location)\n",
    "    if \"2.Final_output\" in list_of_folders:\n",
    "        subprocess.run(f'rm -r \"{output_location}/2.Final_output\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    os.mkdir(f\"{output_location}/2.Final_output\")\n",
    "\n",
    "def write_exon_output(output_location,species,exon,exon_output):\n",
    "    \n",
    "    \n",
    "    list_of_folders_in_final_output = os.listdir(f\"{output_location}/2.Final_output\")\n",
    "    if species not in list_of_folders_in_final_output:\n",
    "        os.mkdir(f\"{output_location}/2.Final_output/{species}\")\n",
    "    \n",
    "    with open(f\"{output_location}/2.Final_output/{species}/{exon}.fa\", 'w') as out_file:\n",
    "        out_file.write(exon_output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b2fddd4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_behrii\n",
      "GCA_029959065.1_ilColBehr1.0.hap2_genomic.fna\n",
      "6804607 6804703\n",
      "0 1\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MEWVLRGPQIHSTFSSLGFQHAEGYHINENCN\n",
      ">Colias_behrii_Exon_1_JARWMC010000003.1_6804607_6804703_left_0_right_1\n",
      "ATGGAGTGGGTTCTTAGGGGACCTCAGATTCATAGTACATTCAGTAGTCTTGGATTTCAACATGCAGAAGGATATCACATAAATGAAAATTGCAATG\n",
      "6805578 6806125\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_2_JARWMC010000003.1_6805578_6806125_left_2_right_0\n",
      "CTGCGTTGGAGTCCATTCTTCATAACATTCTAACGGAGGATAAAAATTTACGCACTTATCGTAGAAGCATAAGTTTTGGACAAAACATCAAGAAAGATATCATACCCCTTTTAATAAATGTTAAAGATAATAAAACGATCGAGTTACTCATCAAGATACTTGTGAATTTAACTATACCTGTTGAATGTTTACTTTCCGTGGAAACGATATCACACACCGATATCGGAAGGCACACTATATTTGAAATCAACAGCCTTCTCGCTTCTACTAAATCTGCATTTATTGACCATCGGGCTACGAAAGTGGTCATAGAGTTTCTGAAGAAGAACTCGGATGTTGAACAGAAGCCGAAATTATCTTCAGAGAGTTGTACTAACATCAGTAACACTCTACTGTTCTTACGGAATATTCTTCATATCCCAGAAGACGTGAACAGTATTTCAAATTATAACAATCCTCCACATACTATTCAAAATCAAATATTATGGAATATTTTTAGTCAAAGCATTGACAAAATTCTCATTAAATTAATGACGATCCCAGATTCG\n",
      "6806635 6806788\n",
      "0 1\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_3_JARWMC010000003.1_6806635_6806788_left_0_right_1\n",
      "ataaattgGGGTGTGACGATGGTTCAGTTGATAGCTCTGCTTTATAAAGATCAGCATGTTACAACATTACATAAACTTCTCAACATGTGGCTTGAAGCATCGCTATCGGGGAGTTCGGAGGATAATGAGAGTAATACTTCGCCACTAAATAGAG\n",
      "6806922 6807151\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_4_JARWMC010000003.1_6806922_6807151_left_2_right_0\n",
      "GTAGTGATGACTCGTCCCCAATGCTGACATCGGACCCGACCTCTGATTCCTCCGACACGGGTGGAAGCGGTAAAATGAACGACGAATCCAACTCTCCCAGTAATAGTTGGGACACATCGAGGAATATAAGTGAACAAAATTTCCCATCAGATAGGAGAGAGGATAAATCATCGAGTCATATGCTAGATTCCGGTCAAGAAAATTCATACGACGACATGAAACAGTCGAAT\n",
      "6807307 6807602\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_5_JARWMC010000003.1_6807307_6807602_left_0_right_2\n",
      "GAAAGGATGATCACGGATAATCAAATGTCAAATAGTCAATTAATGGATAACAAAAAACATGAGGAaggtatgaataaaaaaatgaattcgGAAACGTCTGATTGTGGTTATGGAACACAAATTGAAAATCAAGAAGTTATATCGACTTCAAGTAACGAAGATGAATTACCATCAAAAAAGCCGGTACATCAAAAACCTCACATTCCCAAACAGAGACCGAATAACAAAGTTCGTGCAAGTGTTACGTTGCAAGAGAGAAAAAGGAAGAAAATTGTCAAAAGAGGAAAAGCTAATAT\n",
      "6807688 6808048\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_6_JARWMC010000003.1_6807688_6808048_left_1_right_0\n",
      "AATAAACGTACAAGGGCTGACGCACAATACGCCAACGGATGACGACATATCTAACGTCTTGAAAGAGTTCACGGTAGATTTCCTTTTGAAGGGATACAATTCATTGGTTCGAACGCTTCACACGCAGATCCTGACAAATTTGCAGTTGAAAATAGATACCTCACATTTTTTCTGGCTTGTCACGTATTTTCTTAAATTCGCGACTCAGATAGACATAGACTTGGAGCACGTCTCGTCTGTTTTATCCTTCGATATTGTTTTATACCTAACAGCGGAAGGAGTCAATTTGTGCGAACAATTTGAACTAGCCATTAAATTAGACGGCAACGATTTGAAGCCAAGTATTCGCAGACTGCATTTG\n",
      "6808344 6808535\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_7_JARWMC010000003.1_6808344_6808535_left_0_right_0\n",
      "GTGGTGACAGCTATACGAGAGTTTGTCCAAGCAATTGAAGTTTATAAAAAGTCGGTTCACATTTGCCAACATGATAAGAACATATTGATCAACTTGCAAATGAAAATGTGTGAAACGGAAGAGCTACGTTCATTACTTGTACTTCTATTGCGTCATTATAATCCCAAGTACCATTCCAAGCAATATTTACAG\n",
      "6808638 6808738\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_8_JARWMC010000003.1_6808638_6808738_left_0_right_2\n",
      "GACCTAGTAGTTACGAATCACATTTTATTGATGTTTTTGGACAGTGTAATGAAATTACCGGATTACAAGGGATCTGCGAAATTAATTGACCACATAAAGCA\n",
      "6809014 6809212\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_9_JARWMC010000003.1_6809014_6809212_left_1_right_0\n",
      "GTTCGCTACACCAGAGGTAATGTACCAGTATGGACTGCTATTAGAAGATTATGAAGGCAACGGAGAGTTTGTAAACGATTGTGTTTTCACGTTGATGCACCATGTCGGCGGTGATTTGGGCTGTTTGATTTCACTCTTCCAACCAAAAATACTGAAAATTTTTACATCTATATGGAAATCTGAGTTTGAAATTTGCGAT\n",
      "6809381 6809586\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_10_JARWMC010000003.1_6809381_6809586_left_0_right_2\n",
      "GATTGGTCCGATCTAATAGAATACGTTATAAATACATTTGTAAAGAAACCGCACACGTTACAACGTACGGCAAGCTTCCGATTGGATACAGAAAGTTTCGATGACgacaaaattataaaagaacttattgaaaatgaaGAGAAACCAAATGAAGATGTGTCCTCTAGTAGAGTCATTGTCAACTATGATGCAAGCGTAAAGGAcaa\n",
      "6809807 6810297\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_11_JARWMC010000003.1_6809807_6810297_left_1_right_1\n",
      "GTGGACAGATGACGAACTATCAGCATTTAATTGGATATATATGCAAAGTAGCACATCAACAGATACTGTGGGTGAAATTGTAAGATTATTCAAAGAAGATGGCATTATCAAAACGCGCGACGCAGTCATTAGGCAGCTCCATAAACAATCAATTATAGACAAAAATGAATTTGATAAGCTCATAAAGGAAGAATCCGGAAAAAGTTCTAAGAGTTTCGAAATTAAAGAAGTGAAAGATGGCGATAATATAGGAAAACTATGTGAATACATGACGAAGGATGGAAAATCTAAATTTTTGAATTGGGTTCAAAGTGTCCTGCTCGAGACgtgttttacaaaaatatatttagaaaaaatgaaaaaagcCAAAAAAGCTGCTGAGAGTGAAACTGTCGCTAAGGACACCACTAAAACGCAAAAAATGTATCATGCAGTGCCCAAAAAGGTTGAAGAATTGCCTGTCATGTCATCCGTTTCGTATCATTCCCTTT\n",
      "6810396 6810596\n",
      "2 1\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_12_JARWMC010000003.1_6810396_6810596_left_2_right_1\n",
      "TGTTCAATCAGTCTGTGCCTCTAGTAACATGGAATTGTGAACAAGCGGCCATATGTAAGGATCTGAAGTTCCTACAGCTTTTACATAAATTAGGTTTCTATATGCCCGTGGATACTGGTAAAGTTTTCATCAGAATACCTTATTTCTGGACGCCCGACTTACTCTATGAAGTTGCAAGCAAGATTTCATCTATTGACGCAT\n",
      "6810753 6810921\n",
      "2 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_13_JARWMC010000003.1_6810753_6810921_left_2_right_2\n",
      "CAAAACTAAAGTTCTCTTTGAGCGATATATCGGAAAGTAGCGTTTTGAGCTCATCTGTGAAAACAAGAACCGCCATTCCGGAGCCAAAGGACATTCCAATGACGTCTATTCCGGAAAATTTCTTTCACAAACAGAAACATATCGCATCTATCGTCAATTTTACACCCAT\n",
      "6812385 6812489\n",
      "1 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_14_JARWMC010000003.1_6812385_6812489_left_1_right_2\n",
      "GCCTGGTTCATCATTTAACATAGCGTGTGACGAACGGGCCGTTCCTAGTTGGTTGGACGTTGTACAAAAATCTCAAGAATTTAAACTTACTCTGGAGCTGGaaag\n",
      "6812823 6813026\n",
      "1 2\n",
      "Getting Gene\n",
      ">Colias_behrii_Exon_15_JARWMC010000003.1_6812823_6813026_left_1_right_2\n",
      "GACGATTATTCAAAGTGAAATACCTGCGAATAAGCCAGATGGGCAAGAAACTACGGCGAAGACATCGAGAACACCAGAGATAAATAAAGAGGAGTGTTATGTAGCAGCTGAGTCTACTCTTTTGATCCATACCGACAATGAAGTCAAAGGAGAGCCTTCTTACTTCAATATCAGCCAAGCGGAATCAGAATATTGTAATAACAG\n",
      "6813178 6813361\n",
      "1 0\n",
      "Getting Gene\n",
      "TTCCTGCGAAACTGCATCGGTCGCTTCAGATCTCACAAGAATGTACGTTTCGGACGAAGACGAGAAGCCCGAAGTGATACACCGACCAGCGGCTGATAAACATTTTAATGACGATCAAAATTCCAGCGATCTAGACAGCAGCGCCGGGAAACGTGCACGCATGCATTTCTCTCCACCTTTTTGA\n",
      ">Colias_behrii_Exon_16_JARWMC010000003.1_6813178_6813361_left_1_right_0\n",
      "TTCCTGCGAAACTGCATCGGTCGCTTCAGATCTCACAAGAATGTACGTTTCGGACGAAGACGAGAAGCCCGAAGTGATACACCGACCAGCGGCTGATAAACATTTTAATGACGATCAAAATTCCAGCGATCTAGACAGCAGCGCCGGGAAACGTGCACGCATGCATTTCTCTCCACCTTTTTGA\n",
      "Colias_croceus\n",
      "GCF_905220415.1_ilColCroc2.1_genomic.fna\n",
      "6060936 6061032\n",
      "0 1\n",
      "Getting Gene\n",
      "MEWVLRGPQIHSTFSNLGFQHAEGYHINENCN\n",
      ">Colias_croceus_Exon_1_NC_059539.1_6060936_6061032_left_0_right_1\n",
      "ATGGAGTGGGTTCTTAGGGGACCTCAGATTCATAGTACATTCAGTAATCTTGGATTTCAACATGCAGAAGGATATCACATAAATGAAAATTGCAatg\n",
      "6061165 6061712\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_2_NC_059539.1_6061165_6061712_left_2_right_0\n",
      "CTGCGTTGGAGTCCATCCTTCATAACATTTTAACGGAGGATAAGAATTTACGCACTTATCGTAGAAGCATAAGTTTTGGACAAAACATCAAGAAAGATATAATACCccttttaataaatgttaaagaTGATAAAACTATCGAGTTACTCATCAAGATACTTGTGAATTTAACTATACCTGTTGAATGTTTACTTTGCGTGGAAACGATATCACACACTGATATCGGAAGGCACACTATATTTGAAATCAACAGCCTTCTCGCTTCTACAAAATCTGCATTTATTGACCATCGGGCGACGAAAGTGGTCATAGAGTTTCTGAAGAAGAACTCTGATGTTGAACAGAAGCCGAAGTTATCTTCAGAGAGTTGTGCTAACATCAGTAACACTCTACTGTTCTTACGGAATATTCTTCATATCCCAGAAGACGTGAACagtatttcaaattataacaATCCTCCACATACtattcaaaatcaaatattatggAATATTTTTAGTCAAAGCATTGACAAAATTCTCATTAAATTAATGACGATCCCAGATTCG\n",
      "6062218 6062371\n",
      "0 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_3_NC_059539.1_6062218_6062371_left_0_right_1\n",
      "ataaattggGGTGTGACGATGGTTCAGTTGATAGCTTTACTTTATAAAGATCAGCATGTTACAACATTACATAAACTTCTCAACATGTGGCTTGAAGCATCGCTATCGGAGAGTTCGGAGGATAATGAGAGTAATACTTCGCCACCAAATAGAG\n",
      "6062506 6062735\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_4_NC_059539.1_6062506_6062735_left_2_right_0\n",
      "gtagtgATGACTCGTCTCCAATGCTGACATCGGACCCGACCTCTGATTCCTCTGACACGGGTGGAAGCGGTAAAATGAACGACGAATCCAACTCTCCCAATAATAGTTGGGACACATCGAGGAATATAAGTGAACAAAATTTCCCATCAGATAGGAGGGAGGATAAATCATCCAGCCATATGCTAGATTCCGGTCAAGAAAATTCATATGACGACATGAAACAGTCGAAT\n",
      "6062827 6063119\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_5_NC_059539.1_6062827_6063119_left_0_right_2\n",
      "GAAAGGATGATCACGGATAATCAAATGACAAATAATCAAATggataacaaaaaacatgaggaaggtatgaataaaaaaatgaattcgGAAACGTCTGATTGTGGTTATGGAACACAAATTGAAAATCAAGAGGTTATATCGACTTCAAGTAATGAAGATGAATTACCATCAAAAAAGCCGGTACATCAAAAACCTCACAATCCAAAACAGAGACCGAATAACAAAGTTCGTGCAAGTGTTACGTTGCAAGAGAGAAAAAGGAAGAAAATTGTCAAAAGAGGAAAAGCTAATAT\n",
      "6063419 6063779\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_6_NC_059539.1_6063419_6063779_left_1_right_0\n",
      "aataAACGTACAAGGGCTGACGCACAATACGCCAACGGATGACGACATATCTAACGTCTTGAAAGAGTTCACAGTAGATTTCCTTTTAAAGGGATACAATTCATTGGTTCGAACGCTACACACGCAGATCCTGACAAATTTGCAATTGCAAATAGATACCTCACATTTTTTCTGGCTTGTCAcgtattttcttaaattcGCGACTCAGATAGACATAGACTTGGAGCACGTCTCGTCTGTTTTATCCTTCGATATTGTGTTATACCTAACAGCGGAAGGAGTCAATTTGTGCGAACAATTTGAACTAGCCATTAAATTAGACGGCAACGATTTAAAGCCAAGTATTCGCAGACTGCATTTG\n",
      "6063957 6064148\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_7_NC_059539.1_6063957_6064148_left_0_right_0\n",
      "GTGGTGACAGCTATACGAGAGTTCGTCCAAGCAATTGAAGTTTATAAAAAGTCGGTTCACATTTGCCAACATGATAAGAATATATTGATCAACTTGCAAATGAAAATGTGTGAAACGGAAGAGCTACGTTCATTACTTGTACTTCTATTGCGTCATTATAATCCGAAGTACCATTCCAAGCAATATTTACAG\n",
      "6064225 6064325\n",
      "0 2\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Colias_croceus_Exon_8_NC_059539.1_6064225_6064325_left_0_right_2\n",
      "GACCTCGTGGTTACGAATCACATTTTATTGATGTTTTTGGACAGTGTTATGAAATTACCAGATTACAAGGGATCTGCGAAATTAATTGACCACATCAAGCA\n",
      "6064592 6064790\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_9_NC_059539.1_6064592_6064790_left_1_right_0\n",
      "gtTCGCTACACCAGAGGTTATGTACCAGTATGGACTCTTGTTAGAAGATTACGAAGGCAACGGAGAATTTGTAAATGATTGCGTATTCACGTTGATGCACCATGTTGGCGGTGACTTGGGCTGTTTGATTTCACTCTTCCAAccgaaaatattgaaaactttTACATCTATATGGAAATCTGAGTTTGAAATTTGCGAT\n",
      "6064946 6065136\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_10_NC_059539.1_6064946_6065136_left_0_right_2\n",
      "gatTGGTCCGATCTAATAGAATatgttataaatacatttgtaaaGAAACCGCATACGTTACAACGTACGGCAAGCTTCCGATTGGATACAGAAAGTTTTGATGACGACAAAATTCTTAAAGAACttattgaaaatgaagatGTGTCCTCTAGTAGGGCCATTGTCAACTATGATGCAAGCGTAAAGGACAg\n",
      "6065343 6065833\n",
      "1 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_11_NC_059539.1_6065343_6065833_left_1_right_1\n",
      "GTGGACAGACGACGAATTATCAGCATTTAATTGGATATACATGCAAAGTAGCACATCAACAGATACTGTGGGTGAAATTGTAAGATTATTTAAAGAAGATGGCATTATCAAAACTCGCGACGCGGTCATTAGGCAGCTCCATAAACAATCAATTATAGACAAAAATGAATTTGACAAGCTTATAAAGGAAGAATCCGGAAAAAGTTCAAAGAATTCCGAAATCAGAGAAGTGAAAGATGACGATAATATAGGAAAACTATGTGAATACATGACGAAGGATGGAAAATCGAAGTTTTTGAATTGGGTTCAAAGTGTCCTGCTCGAAACgtgttttacaaaaatatatttagaaaaaaagaaaatagccaAAAAAGCTGCTGAGAGTGAAACTGCCGTTAAGGACACCactaaaacacaaaaaatgtatCATGCAGTGCCCAAAAAGATTGAAGAATTGCCTGTCATGTCATCCGTTTCGTATCATTCCCTTT\n",
      "6065937 6066137\n",
      "2 1\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_12_NC_059539.1_6065937_6066137_left_2_right_1\n",
      "TGTTCAATCAGTCTGTGCCTCTAGTACCATGGAATTGCGAACAAGCGGCCATATGTAAGGATCTGAAGTTCCTGCAGcttttacataaattaggtTTCTATATGCCCGTTGATACTGGTAAAGTTTTCATAAGAATACCTTATTTCTGGACACCCGACTTACTCTATGAAGTTGCAAGCAAAATTTCGTCTATTGACGCAt\n",
      "6066302 6066467\n",
      "2 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_13_NC_059539.1_6066302_6066467_left_2_right_2\n",
      "CAAATCTAAAGTTCTCCTTGAGCGATATATCGGAAAGTAGCGTTTTGAGCTCATCTGTGACCGCCATTCCCGAGCCAAAGGACATTCCAATGACGTCTATTCCGGAAAATTTCTTTCAAATTCACAAACAGAAACATATCGCATCTATCGTCAATTTTACACCCAT\n",
      "6074184 6074288\n",
      "1 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_14_NC_059539.1_6074184_6074288_left_1_right_2\n",
      "GCCTGGTTCATCATTTAACATGGCGTGTGACGAACACGCAGTTCCTAGTTGGTTGGACGTTGTACAAAAATCTCAAGAATTTAAACTTACTCTGGAGTTGGAAAG\n",
      "6074947 6075150\n",
      "1 2\n",
      "Getting Gene\n",
      ">Colias_croceus_Exon_15_NC_059539.1_6074947_6075150_left_1_right_2\n",
      "GACGATTATTCAAAGTGAAATATCTGCGAACAAGCCAGATGGGCAAGAAACTACGGCGAAGACATCTAGAACAACAGAGATAAATAAAGAGGAGCGTGATGTAGCAGCCGAGTCTACTCTTTTGATCCCTACCGACAATGAAGTCAAGGAAGAGCCTTCTAACTTCAATATCAGTCAAGCGGAATCagaatattgtaataacag\n",
      "6075298 6075481\n",
      "1 0\n",
      "Getting Gene\n",
      "TTCCTGCGAAACTGCATCGGTCGCTTCAGATCTCACAAGAATGTACGTATCGGACGAAGACGAGAAGCCCGAAGTGATACACCGACCAGCGGctgataaacattttaatgacGATCAAAATTCTAGCGATCTAGACAACAGCGCCGGGAAACGTGCACGCGTGCATTTCTCTCCacctttttga\n",
      ">Colias_croceus_Exon_16_NC_059539.1_6075298_6075481_left_1_right_0\n",
      "TTCCTGCGAAACTGCATCGGTCGCTTCAGATCTCACAAGAATGTACGTATCGGACGAAGACGAGAAGCCCGAAGTGATACACCGACCAGCGGctgataaacattttaatgacGATCAAAATTCTAGCGATCTAGACAACAGCGCCGGGAAACGTGCACGCGTGCATTTCTCTCCacctttttga\n",
      "Colias_eurytheme\n",
      "GCA_907164685.1_Colias_eurytheme_alba_genomic.fna\n",
      "79574 79670\n",
      "0 1\n",
      "Getting Gene\n",
      "MEWVLRGPQIHSTFSSLGFQHAEGYHINENCN\n",
      ">Colias_eurytheme_Exon_1_CAJQZN010043793.1_79574_79670_left_0_right_1\n",
      "ATGGAGTGGGTTCTTAGGGGACCTCAGATTCATAGTACATTCAGTAGTCTTGGATTTCAACATGCAGAAGGATATCACATAAATGAAAATTGCAATg\n",
      "80204 80751\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_2_CAJQZN010043793.1_80204_80751_left_2_right_0\n",
      "CTGCGTTGGAGTCCATTCTTCATAACATTCTAACGGAGGATAAAAATTTACGCACTTATCGTAGAAGCATAAGCTTTGGACAAAACATCAAGAAAGATATCATACCccttttaataaatgttaaagatGATAAAACGATCGAGTTACTCATCAAGATACTTGTGAATTTAACTATACCCGTTGAATGTTTACTTTCCGTGGAAACGATATCACACACCGATATCGGAAGGCACACTATATTTGAAATCAACAGCCTTCTCGCTTCTACTAAATCTGCATTTATTGACCATCGGGCTACGAAAGTGGTCATAGAGTTTCTGAAGAAGAACTCGGATGTTGAACAGAAGCCGAAATTATCTTCAGAGAGTTGTACTAACATCAGTAACACTCTACTGTTCTTACGGAATATTCTTCATATCCCAGAAGACGTGAAcagtatttcaaattataacaaTCCTCCATATactattcaaaatcaaatattatggAATATTTTTAGTCAAAGCATTGACAAAATTCTCATTAAGTTAATGACGATTCCAGATTCG\n",
      "81235 81388\n",
      "0 1\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_3_CAJQZN010043793.1_81235_81388_left_0_right_1\n",
      "ataaattgGGGAGTGACGATGGTTCAGTTGATAGCTTTACTTTATAAAGATCAGCATGTTACAACATTACATAAACTTCTCAACATGTGGCTTGAAGCATCGCTATCGGAGAGTTCGGAGGATAATGAGAGTAATACTTCCCCACTAAATAGAG\n",
      "81474 81703\n",
      "2 0\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_4_CAJQZN010043793.1_81474_81703_left_2_right_0\n",
      "GTAGTGATGACTCGTCTCCAATGCTGACATCGGACCCGACCTCTGATTCCTCTGACACGGGTGGAAGCGGTAAAATGAACGACGAATCCAACTCTCCCAATAATAGTTGGGACACATCGAGGAATATAAGTGAACAAAATTTCCCATCAGATAGGAGGGAGGATAAATCATCGAGCCATATGCTAGATTCCGGTCAAGAAAATTCATACGACGACATGAAACAGTCGAAT\n",
      "81847 82139\n",
      "0 2\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_5_CAJQZN010043793.1_81847_82139_left_0_right_2\n",
      "GAAAGGATGATCACGGATAATCAAATGTCAAATAGTCAATTAATGGATAACAAAAAACATGAGgaaggtaataaaaaaatgaattcgGAAACGTCTGATTGTGGTTATGGAACACAAATTGAAAATCAGGAAGTTATATCGACTTCAAGTAACGAAGATGAATTACCATCAAAAAAGCCGGTACATCAAAAACCTCACAATCCCAAACAGAGACCGAATAACAAAGTTCGTGCAAGTGTTACGTTGCAAGAGAGAAAAAGGAAGAAAATTGTCAAAAGAGGAAAAGCTAATAT\n",
      "82225 82585\n",
      "1 0\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_6_CAJQZN010043793.1_82225_82585_left_1_right_0\n",
      "AATAAACGTACAAGGGCTGACGCACAATACGCCAACGGATGACGACATATCTAACGTCTTGAAAGAGTTCACGGTAGATTTCCTGTTGAAGGGATACAATTCATTGGTTCGAACGCTTCACACGCAGATCCTGACAAATTTGCAATTGCAAATAGATACCTCACATTTTTTCTGGCTTGTCACGTATTTTCTTAAATTCGCGACTCAGATAGACATAGACTTGGAGCACGTCTCGTCTGTTTTATCCTTCGATATTGTTTTATACCTAACAGCGGAAGGAGTCAATTTGTGCGAACAATTTGAACTAGCTATTAAATTAGACGGCAACGATTTGAAGCCAAGTATTCGCAGACTGCATTTg\n",
      "82892 83083\n",
      "0 0\n",
      "Getting Gene\n",
      ">Colias_eurytheme_Exon_7_CAJQZN010043793.1_82892_83083_left_0_right_0\n",
      "GTGGTAACAGCTATACGAGAGTTTGTCCAAGCAATTGAAGTTTATAAAAAGTCGGTTCACATTTGCCAATATGATAAGAATATATTGATCAACTTGCAAATGAAAATGTGTGAAACGGAAGAGCTTCGTTCATTACTTGTACTTCTATTGCGTCATTATAATCCCAAGTACCATTCCAAGCAATATTTACAG\n",
      "83183 83283\n",
      "0 2\n",
      "Getting Gene\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "\n",
    "family_group = \"5.Coliadinae\"\n",
    "\n",
    "output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "make_output_folder(output_location)\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_species = os.listdir(f\"{output_location}/1.Blast_result\")\n",
    "if 'desktop.ini' in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "# print(list_of_species)\n",
    "# list_of_species = [\"Heliconius_charithonia\"]\n",
    "# list_of_species = [\"Maniola_hyperantus\"]\n",
    "error_exons = ''\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    print(genome_file)\n",
    "    previous_coordinate_start = ''\n",
    "    previous_scaffold = ''\n",
    "    with open(f\"{output_location}/1.Blast_result/{species}/final_coordinates.csv\", 'r') as coordinate_final_file:\n",
    "        coordinate_file_list = coordinate_final_file.readlines()\n",
    "    \n",
    "    for i in range(1,len(coordinate_file_list)):\n",
    "        \n",
    "        coordinate_line_split = coordinate_file_list[i].split(\",\")\n",
    "        \n",
    "            \n",
    "#         print(coodinate_line_split[5])\n",
    "        if coordinate_line_split[5] == \"N\":\n",
    "            if previous_coordinate_start == '':\n",
    "                previous_coordinate_start = coordinate_line_split[2]\n",
    "                previous_scaffold = coordinate_line_split[1]\n",
    "            complement = coordinate_line_split[4]\n",
    "            start,stop = coordinate_line_split[2],coordinate_line_split[3]\n",
    "            print(start,stop)\n",
    "            query_name = coordinate_line_split[6]\n",
    "            scaffold = coordinate_line_split[1]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            left_oh, right_oh = query_name.split('_')[-6], query_name.split('_')[-4]\n",
    "            print(left_oh, right_oh)\n",
    "#             print(complement)\n",
    "            if complement == \"0\":\n",
    "                if start < previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            elif complement == \"1\":\n",
    "                if start > previous_coordinate_start or scaffold != previous_scaffold:\n",
    "                    print(f\"Error in Coordinates\\nPrevious_coordinate = {coordinate_file_list[i-1]}\\nCurrent_coordinate = {coordinate_file_list[i]}\")\n",
    "                    prompt = input(\"Continue (Y/N)?\")\n",
    "                    if prompt[0].lower() == \"n\":\n",
    "                        assert False\n",
    "            else:\n",
    "                print(\"ERROR\")\n",
    "            \n",
    "        elif coordinate_line_split[5] == \"Y\":\n",
    "            query_name = coordinate_line_split[6]\n",
    "            exon = f\"Exon_{query_name.split('_')[-1]}\"\n",
    "            error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            continue\n",
    "            \n",
    "        gene_sequence = get_gene_sequence(genome_location, \n",
    "                                          species, \n",
    "                                          genome_file, \n",
    "                                          scaffold,\n",
    "                                          start,\n",
    "                                          stop,\n",
    "                                          complement,\n",
    "                                          output_location)\n",
    "        \n",
    "       \n",
    "        current_exon = query_name.split('_')[-1]\n",
    "        if current_exon == \"1\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence.reverse_complement.translate()\n",
    "            else:\n",
    "                translated = gene_sequence.translate()\n",
    "            print(translated)\n",
    "            if \"*\" in translated:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "            else:\n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "        \n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "        elif current_exon == \"16\":\n",
    "            if complement == 1:\n",
    "                translated = gene_sequence[:-int(left_oh)].reverse_complement.translate()\n",
    "                print(gene_sequence)\n",
    "            else:\n",
    "                translated = gene_sequence[int(left_oh):].translate()\n",
    "                print(gene_sequence)\n",
    "\n",
    "            if \"*\" in translated[:-1]:\n",
    "                error_exons += f\"{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n\"\n",
    "                print(translated)\n",
    "            else:\n",
    "                \n",
    "                exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "                print(exon_output)\n",
    "                write_exon_output(output_location,species,exon,exon_output)\n",
    "#         assert False\n",
    "        else:\n",
    "            exon_output = f\">{species}_{exon}_{scaffold}_{start}_{stop}_left_{left_oh}_right_{right_oh}\\n{gene_sequence}\"\n",
    "            print(exon_output)\n",
    "\n",
    "            write_exon_output(output_location,species,exon,exon_output)\n",
    "        \n",
    "#         assert False\n",
    "        previous_coordinate_start = start\n",
    "        previous_scaffold = scaffold\n",
    "    error_exons += \"\\n\\n\"\n",
    "#     print(coordinate_file_list)\n",
    "#     break\n",
    "coordinate_file = ''\n",
    "print(error_exons)\n",
    "with open(f\"{output_location}/error_exons.txt\", 'w') as error_out_file:\n",
    "    error_out_file.write(error_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "150d08ac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912c1f49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
