{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"5.Coliadinae\"\n",
    "number_of_exons = 16\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E ME-WVLRGPQIHSTFSSLGFQHAEGYHINENCN\n",
      "Colias_eurythem ME-WVLRGPQIHSTFSSLGFQHAEGYHINENCN\n",
      "Colias_croceus_ ME-WVLRGPQIHSTFSNLGFQHAEGYHINENCN\n",
      "Eurema_hecabe_E ME-WVLRGPQIHSTFSDLGYAHPEGYHVNENCN\n",
      "Zerene_cesonia_ ME-WVLRGPQIHSTFSNLGFAHPEGYHINENCN\n",
      "Leptidea_juvern MDGWILRGPLIHSTFSSLGFSDSKGYCINENCN\n",
      "Leptidea_reali_ MDGWILRGPLIHSTFSSLGFSDSKGYCINENCN\n",
      "Leptidea_sinapi MDGWILRGPLIHSTFSSLGFSDSKGYCINENCN\n",
      "                *: *:**** ******.**: ..:** :*****\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E ALESILHNILTEDKNLRTYRRSISFGQNIKKDIIPLLINVKDNKTIELLIKILVNLTIPV\n",
      "Colias_eurythem ALESILHNILTEDKNLRTYRRSISFGQNIKKDIIPLLINVKDDKTIELLIKILVNLTIPV\n",
      "Colias_croceus_ ALESILHNILTEDKNLRTYRRSISFGQNIKKDIIPLLINVKDDKTIELLIKILVNLTIPV\n",
      "Zerene_cesonia_ ALESILHNILTEDKNLRTYRRSISFGQNIKKDIIPLLINVKDEKTIELLIKILVNLTIPV\n",
      "Eurema_hecabe_E TLESILHNILNEDKNLRTYRRSINFGQNINKDIIPLLINVKEDKTTELLIQILVNLTIPV\n",
      "Leptidea_juvern ALECILHNILTEDRNLRTYRRAISFGQNIQKDLIPLLINTKEDKTVELLIKILVNLTIPV\n",
      "Leptidea_reali_ ALECILHNILTEDRNLRTYRRAISFGQNIQKDLIPLLINTKEDKTVELLIKILVNLTIPV\n",
      "Leptidea_sinapi ALECILHNILTEDRNLRTYRRAISFGQNIQKDLIPLLINTKEDKTVELLIKILVNLTIPV\n",
      "                :**.******.**:*******:*.*****:**:******.*::** ****:*********\n",
      "\n",
      "Colias_behrii_E ECLLSVETISHTDIGRHTIFEINSLLASTKSAFIDHRATKVVIEFLKKNSD------VEQ\n",
      "Colias_eurythem ECLLSVETISHTDIGRHTIFEINSLLASTKSAFIDHRATKVVIEFLKKNSD------VEQ\n",
      "Colias_croceus_ ECLLCVETISHTDIGRHTIFEINSLLASTKSAFIDHRATKVVIEFLKKNSD------VEQ\n",
      "Zerene_cesonia_ ECLLSVETISNTDIGRHTIFEINSLLASTKSAFIDHRATKVIIEFLKKNSD------VEQ\n",
      "Eurema_hecabe_E ECLLSVETISNTDIGRHTIFEINGLLGSAKSAFIDHRATKVVIEFLKKNSD------VEQ\n",
      "Leptidea_juvern ECLLSIKTASNADVGRHTIFEINSLLASTKSAFKDDRTTKVVIEFLKKNAE----LNVEQ\n",
      "Leptidea_reali_ ECLLSIKTASNADVGRHTIFEINSLLASTKSAFKDDRTTKVVIEFLKKNAELDVELNVEQ\n",
      "Leptidea_sinapi ECLLSIKTASNADVGRHTIFEINSLLASTKSAFKDDRTTKVVIEFLKKNAELDVELNVEQ\n",
      "                ****.::* *::*:*********.**.*:**** *.*:***:*******::      ***\n",
      "\n",
      "Colias_behrii_E KPKLSSESCTNISNTLLFLRNILHIPEDVNSI-SNYNNPPHTIQNQILWNIFSQSIDKIL\n",
      "Colias_eurythem KPKLSSESCTNISNTLLFLRNILHIPEDVNSI-SNYNNPPYTIQNQILWNIFSQSIDKIL\n",
      "Colias_croceus_ KPKLSSESCANISNTLLFLRNILHIPEDVNSI-SNYNNPPHTIQNQILWNIFSQSIDKIL\n",
      "Zerene_cesonia_ KSKLSSEQCTNISNTLLFLRNILHIPEDVNNISSNYNSPPHTIQNQILWNIFSQSIDKIL\n",
      "Eurema_hecabe_E TAKLSAEQCTNISNTLLFLRNILHIPEDVNSISTNSNGPPHTIQNKILWNLFSQSIDRIL\n",
      "Leptidea_juvern KSKLSLEQCTNITNTLLLLRNILHIPEEISEIFANYNGVPHTIQNQIIWNLFSQRIDKAI\n",
      "Leptidea_reali_ KSKLSLEQCTNITNTLLLLRNILHIPEEISEIFANYNGVPHTIQNQIIWNLFSQRIDKAI\n",
      "Leptidea_sinapi KSKLSLEQCTNITNTLLLLRNILHIPEEISEIFANYNGVPHTIQNQIIWNLFSQRIDKAI\n",
      "                ..*** *.*:**:****:*********::..* :* *. *:****:*:**:*** **: :\n",
      "\n",
      "Colias_behrii_E IKLMTIPDS\n",
      "Colias_eurythem IKLMTIPDS\n",
      "Colias_croceus_ IKLMTIPDS\n",
      "Zerene_cesonia_ IKLMTIPDS\n",
      "Eurema_hecabe_E IKLMTIPDS\n",
      "Leptidea_juvern IKLVTIPES\n",
      "Leptidea_reali_ IKLVTIPES\n",
      "Leptidea_sinapi IKLVTIPES\n",
      "                ***:***:*\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSGSSEDNESNTSPLNR\n",
      "Colias_croceus_ INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPPNR\n",
      "Zerene_cesonia_ INWGVTMVQLIALLYKDQHVTTLHKLLNMWLETSLSESSEDNESNTSPPNR\n",
      "Colias_eurythem INWGVTMVQLIALLYKDQHVTTLHKLLNMWLEASLSESSEDNESNTSPLNR\n",
      "Eurema_hecabe_E INWGVTMVQLIALLYKDQHVTTLHRLLNVWLETSVSESSDDNESNTSPPKR\n",
      "Leptidea_juvern INWCVAMVQLIALLYKDQHVTVLQKLLNSWLESSQFDSSEDNESNTSQHSR\n",
      "Leptidea_reali_ INWCVAMVQLIALLYKDQHVTVLQKLLNSWLESSHFDSSEDNESNTSQHSR\n",
      "Leptidea_sinapi INWCVAMVQLIALLYKDQHVTVLQKLLNSWLESSHFDSSEDNESNTSQHSR\n",
      "                *** *:***************.*::*** ***:*   **:*******  .*\n",
      "\n",
      "continue?y\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E SDDSSPMLTSDPTSDSSDTGGSGKMNDESNSPSNSWDT----SRNISEQNFPSD-RREDK\n",
      "Colias_croceus_ SDDSSPMLTSDPTSDSSDTGGSGKMNDESNSPNNSWDT----SRNISEQNFPSD-RREDK\n",
      "Colias_eurythem SDDSSPMLTSDPTSDSSDTGGSGKMNDESNSPNNSWDT----SRNISEQNFPSD-RREDK\n",
      "Zerene_cesonia_ SDDSSPMLTSDPTSDSSDTGGSGKMNDESNSPNNGWDT----SRNISEQNFPSD-RMEEK\n",
      "Eurema_hecabe_E SDDSSPMLT-DPTSDSSDTCGNGKINNDSKSPNNDWETA---SKNVNDQNLQTD-TMEKK\n",
      "Leptidea_juvern SSDS-PMLSSGPTSDCSDNEGSAKMKGQSDSPTNSWDSVIRSTNNDTEKLYKSNNTMKPK\n",
      "Leptidea_reali_ SSDS-PMLSSGPTSDCSDNEGSAKMKGQSDSPTNSWDSVIRSTNNDTEKLYKSNNTMKPK\n",
      "Leptidea_sinapi SSDL-PMLSSGPTSDCSDNEGSAKMKGQSDSPTNSWDSVIRSTNNDTEKLYKSNNTMKPK\n",
      "                *.*  ***: .****.**. *..*::.:*.**.*.*::    :.* .::   ::   : *\n",
      "\n",
      "Colias_behrii_E SSSHMLDSGQENSYDDMK-QSN\n",
      "Colias_croceus_ SSSHMLDSGQENSYDDMK-QSN\n",
      "Colias_eurythem SSSHMLDSGQENSYDDMK-QSN\n",
      "Zerene_cesonia_ TSNHMLDSGQENSYDEMK-KTD\n",
      "Eurema_hecabe_E TASPS-DSSQNNSSDINK-ETN\n",
      "Leptidea_juvern PFNKS-VSAMDISLSFDENQSD\n",
      "Leptidea_reali_ PFNKS-VSAMDISLSFDENQSD\n",
      "Leptidea_sinapi PFNKS-VSAMDISLSFDENQSD\n",
      "                . .    *. : * .  : :::\n",
      "\n",
      "continue?y\n",
      "Exon_5.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E ERMITDNQMS-NSQLMDNKKHEEGMNKKMNSETSDCGYGTQIENQEVISTSSNEDELPSK\n",
      "Colias_eurythem ERMITDNQMS-NSQLMDNKKHEEG-NKKMNSETSDCGYGTQIENQEVISTSSNEDELPSK\n",
      "Colias_croceus_ ERMITDNQMT-NNQ-MDNKKHEEGMNKKMNSETSDCGYGTQIENQEVISTSSNEDELPSK\n",
      "Zerene_cesonia_ ERMITDNQMS-NGQ-LENKKHEEGMNKKMNSETSDCGYGTQIENQEVISTSSNEDELPTK\n",
      "Eurema_hecabe_E -----DNIMNKDDSPMEDKEHDQTLNKKMNSETSDCGYGTQIENQAVISTSSNEDELPEK\n",
      "Leptidea_juvern ------NITFGTIEELNIESQRNPKRKMLTSENWHCAYGSEDKNKELVSTSSSDDE-PYK\n",
      "Leptidea_reali_ ------NITYGTIEELNIESQRNPKRKMLTSENWHCAYGSEDKNKELVSTSSNDDE-PYK\n",
      "Leptidea_sinapi ------NITYGTIEELNIESQRNPKRKMLTSENWHCAYGSEDKNKELVSTSSNDDE-PYK\n",
      "                      *      . :: :.: :  .* :.**. .*.**:: :*: ::****.:** * *\n",
      "\n",
      "Colias_behrii_E KPVHQKPHIPKQRPNNKVRASVT--LQERKRKKIVKRGKAN\n",
      "Colias_eurythem KPVHQKPHNPKQRPNNKVRASVT--LQERKRKKIVKRGKAN\n",
      "Colias_croceus_ KPVHQKPHNPKQRPNNKVRASVT--LQERKRKKIVKRGKAN\n",
      "Zerene_cesonia_ KPVHQKPHNPKQRPNNKVRSSVT--LQERKRKKIVKRGKAN\n",
      "Eurema_hecabe_E KLVHQKPHNPKQKPNNKVRTSAS--LQDRKKKKIVKRGKTN\n",
      "Leptidea_juvern KPVHQKPYNPKQRPNTKLRIGLTESLQTRRRKKVMKRATSN\n",
      "Leptidea_reali_ KPVHQKPYNPKQRPNTKLRIGLTESLQTRRRKKVMKRATSN\n",
      "Leptidea_sinapi KPVHQKPYNPKQRPNTKLRIGLTESLQTRRRKKVMKRATSN\n",
      "                * *****: ***:**.*:* . :  ** *::**::**..:*\n",
      "\n",
      "continue?y\n",
      "Exon_6.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E INVQGLTHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNLQLKIDTSHFFWLV\n",
      "Colias_croceus_ INVQGLTHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNLQLQIDTSHFFWLV\n",
      "Colias_eurythem INVQGLTHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNLQLQIDTSHFFWLV\n",
      "Zerene_cesonia_ INVQGLTHNTPTDDDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNLQLEIDTSHFFWLV\n",
      "Eurema_hecabe_E INIQGLSHNSITDDDISNVLKEFTVDFLLRGYNSLVKTLHTQILTNFQLQIDTSHFFWLV\n",
      "Leptidea_juvern INVRGLSHSTPTNDDIACVMKEFTVDFLLKGYNALVKTLHTQILTNFQLEIDISHFFWLV\n",
      "Leptidea_reali_ INVRGLSHSTPTNDDIACVMKEFTVDFLLKGYNALVKTLHTQILTNFQLEIDISHFFWLV\n",
      "Leptidea_sinapi INVRGLSHSTPTNDDIACVMKEFTVDFLLKGYNALVKTLHTQILTNFQLEIDISHFFWLV\n",
      "                **::**:*.: *:***: *:*********:***:**:*********:**:** *******\n",
      "\n",
      "Colias_behrii_E TYFLKFATQIDIDLEHVSSVLSFDIVLYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Colias_croceus_ TYFLKFATQIDIDLEHVSSVLSFDIVLYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Colias_eurythem TYFLKFATQIDIDLEHVSSVLSFDIVLYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Zerene_cesonia_ TYFLKFATQIELDLEHVSSVLSFDIVLYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Eurema_hecabe_E TYFLKFATQIELDLEHVCSVLSYDIALYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Leptidea_juvern TYFLKFATQIELDLSIICSVISFDMVLYLTAEGVHLCEQFELAIKLKKNDLKPIKRRLHL\n",
      "Leptidea_reali_ TYFLKFATQIELDLSIICSVISFDMVLYLTAEGVHLCEQFELAIKLKKNDLKPIKRRLHL\n",
      "Leptidea_sinapi TYFLKFATQIELDLSIICSVISFDMVLYLTAEGVHLCEQFELAIKLKKNDLKPIKRRLHL\n",
      "                **********::**. :.**:*:*:.********:***********. *****  *****\n",
      "\n",
      "continue?y\n",
      "Exon_7.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Colias_behrii_E VVTAIREFVQAIEVYKKSVHICQHDKNILINLQMKMCETEELRSLLVLLLRHYNPKYHSK\n",
      "Colias_croceus_ VVTAIREFVQAIEVYKKSVHICQHDKNILINLQMKMCETEELRSLLVLLLRHYNPKYHSK\n",
      "Colias_eurythem VVTAIREFVQAIEVYKKSVHICQYDKNILINLQMKMCETEELRSLLVLLLRHYNPKYHSK\n",
      "Zerene_cesonia_ VVTAIREFVQAIEVYKKSVHICQHDKDILLDLQMKMCETEELRSLLVLLLRHYNPKYHSK\n",
      "Eurema_hecabe_E VVTAIREFVQAIEVYKKSIHISQHDKNILLELQMKMCETEELRSLFVLLLRNYNPIHHSK\n",
      "Leptidea_juvern VVAAIREFVQAIHVYKKSDNIREEDRMSLLKLQVKMCETEELRSLLVLLLRHYNPNYHTK\n",
      "Leptidea_reali_ VVAAIREFVQAIHVYKKSDNIREEDRMSLLKLQVKMCETEELRSLLVLLLRHYNPNYHTK\n",
      "Leptidea_sinapi VVAAIREFVQAIHVYKKSDNIREEDRMSLLKLQVKMCETEELRSLLVLLLRHYNPNYHTK\n",
      "                **:*********.***** :* : *:  *:.**:***********:*****:*** :*:*\n",
      "\n",
      "Colias_behrii_E QYLQ\n",
      "Colias_croceus_ QYLQ\n",
      "Colias_eurythem QYLQ\n",
      "Zerene_cesonia_ QYLQ\n",
      "Eurema_hecabe_E QYLQ\n",
      "Leptidea_juvern QYLQ\n",
      "Leptidea_reali_ QYLQ\n",
      "Leptidea_sinapi QYLQ\n",
      "                ****\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
