{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"4.Pierinae\"\n",
    "number_of_exons = 26\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car -----MDNLDDNAKVSDSAYSNSCSNSQAR\n",
      "Leptophobia_ari MNIDNMDDSENTAKISDSAYSNSCSNSQAR\n",
      "Pieris_brassica --MGNIDDAENNAKISDSAYTNSCSNSQAR\n",
      "Pieris_mannii_E --MGNIDDSENTAKISDSAYSNSCSNSQAR\n",
      "Pieris_rapae_Ex --MGNIDDSENNAKISDSAYSNSCSNSQAR\n",
      "Pieris_melete_E --MDNIDDSENNAKVSDSAYSNSCSNSQAR\n",
      "Pieris_napi_Exo --MDNIDDSENNAKVSDSAYSNSCSNSQAR\n",
      "Aporia_crataegi ---MNMDESENNAKISDSAYSNSCSNSLAR\n",
      "                     :*: ::.**:*****:****** **\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car RSSQSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Pieris_mannii_E RSSQSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Pieris_melete_E RSSQSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Pieris_napi_Exo RSSQSTHSGSHSSGSSGYGGKPSTSGY\n",
      "Leptophobia_ari RSSHSTHSGSHTSGSSGYGGKPSTSGY\n",
      "Pieris_rapae_Ex RSSHSTLSGSHSSGSSGYGGKPSTSGY\n",
      "Pieris_brassica RSSHSTHSESHSSGSSGYGGKPSTSGY\n",
      "Aporia_crataegi RSSHSTRSGSHSSRSSGYGGRLSTLGY\n",
      "                ***:** * **:* ******: ** **\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car --NKVC--QPPEIRCKIKKKKKQPD-ESETIQEEKPSAPEPEPVVEPAKEVCVE-\n",
      "Aporia_crataegi GNNIFS--QPPEIRCKIKKKKKQVEVKNVVIQEEKPTVLEPEPKVQPAEEQCLD-\n",
      "Leptophobia_ari -SNNFS--QPSEIRSKTKKKKKLAETEAVVIQEEKQIIPEPVPEVQPIEEQCID-\n",
      "Pieris_brassica -SNNVS--QPPEIRCKVKKKKKQLELDTVVIHEEKQTVPDPEPEIKPIEEQRND-\n",
      "Pieris_melete_E -SNNVS--QPPDIRCKIKKKKKQLEIDTVVIQEEKQTVPDPEPEIKPIEEQNNDD\n",
      "Pieris_napi_Exo -SNNVS--QAPDIRCKVKKKKKQLEIDSVAIQEENQIVPDPEPEIKPIEKQSNDD\n",
      "Pieris_mannii_E -SNNIS--KPPEIHCKVKKKKKQLELDNVAIQEEKQTVPEPEPEIKQIEENRKD-\n",
      "Pieris_rapae_Ex -SNNVSQIQPPEIRCKVKKKKKQQEVDNTVIQEEKQAAPEPEPEIKQIEEKEND-\n",
      "                  * ..  :..:*:.* *****  : .  .*:**:    :* * ::  ::   : \n",
      "\n",
      "continue?y\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car VSSPIVQQAESAQLETLEISTTDRSSVKDK---\n",
      "Aporia_crataegi VGYQDVQQTESVQAEKIDLTTTDRSSVKDK---\n",
      "Leptophobia_ari ASSPDPQQMQLMQTEHIDLTTIDRCSTKDK---\n",
      "Pieris_brassica ASSTGVQQKDVSQVEINDLSTIDRNSSKDK---\n",
      "Pieris_melete_E ASSPSVQQNDLSQVEINDQPTIDRSSSKDK---\n",
      "Pieris_napi_Exo ASSPGVQKNDLLQVEINDQSTIDKSSSKDK---\n",
      "Pieris_mannii_E ASTPAVQQKDVLQVEISEPPTTDRNNSKDK---\n",
      "Pieris_rapae_Ex TSSPAVPQKDVLQVEISDTPTTERNNSKEKKEK\n",
      "                ..     : :  * *  : .* :: . *:*   \n",
      "\n",
      "continue?y\n",
      "Exon_5.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car TSTCNLPVARTPNRTLAKSC\n",
      "Pieris_brassica TSSCNIPI-QTPTKCK----\n",
      "Pieris_mannii_E TSSCNIPI-QTPIKTFL---\n",
      "Pieris_rapae_Ex TSTCNIPI-QTPIKTLL---\n",
      "Pieris_napi_Exo TSLCNIPV-QTPVKTLLTNN\n",
      "Pieris_melete_E TSLCNIPI-QSPIKTLLANS\n",
      "Leptophobia_ari ISTFNFPV-HTQLRTLVSN-\n",
      "Aporia_crataegi ASSCNFPI-SV----FFLNW\n",
      "                 *  *:*:            \n",
      "\n",
      "continue?y\n",
      "Exon_6.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car DGFSCIISMQDGVVMYTTNSLTNSLGFPRDMWIGRSFIDFVHPR\n",
      "Pieris_brassica DGFSCVISMQDGVVIYTTNSLTNTLGFPRDMWIGRSFIDFVHPR\n",
      "Pieris_melete_E DGFSCVISMQDGVVMYTTNSLTNTLGYPRDMWIGRSFIDFVHPR\n",
      "Pieris_napi_Exo DGFSCVISMQDGVVMYTTNSLTNTLGYPRDMWIGRSFIDFVHPR\n",
      "Pieris_mannii_E DGFSCVISMQDGVVMYTTNSLTNTLGFPSDMWIGRSFIDFVHPR\n",
      "Pieris_rapae_Ex DGFSCVISMQDGVVMYTTNSLTNTLGFPSDMWIGRSFIDFVHPR\n",
      "Leptophobia_ari DGFSCVISMQDGVVMYTTHSITNALGYPRDMWIGRSFIDFIHPR\n",
      "Aporia_crataegi DGFSCVISMQDGVVMYTTYSLTNTLGFPRNMWIGRSFIDFVHPR\n",
      "                *****:********:*** *:**:**:* :**********:***\n",
      "\n",
      "continue?y\n",
      "Exon_7.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car DRNTFASQITNGLVIPKNVNGTHKK\n",
      "Aporia_crataegi DRNTFASQITNGIAIPKTVNSALIE\n",
      "Pieris_melete_E DRNTFASQITNGVAIPKTVHRTHHK\n",
      "Pieris_napi_Exo DRNTFASQITNGVAIPKIVHRTHHK\n",
      "Pieris_brassica DRNTFASQITNDVATPKILNGTPNE\n",
      "Pieris_mannii_E DRNTFASQITNSVAMPEIVNGTHNK\n",
      "Pieris_rapae_Ex DRNTFASQITNSVAMPEILNGSHNK\n",
      "Leptophobia_ari DRNTFGSQITNGVAILKNVHGTLKK\n",
      "                *****.*****.:.  : :: :  :\n",
      "\n",
      "continue?y\n",
      "Exon_8.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car HY---YSIGNILSTMVCRIRRYRGMATGFGVKDRVVAFMPFLLKLSFRNVCDEEGEVIYL\n",
      "Aporia_crataegi ---------STLSTLVCRIRIYRGLKTGFGIKNRLVSFMPFLLKLCFRNICDEGGNVIYL\n",
      "Leptophobia_ari --------TNTLSTMVCRIRRYRGLSSGFGVKTRTLSFMPFLLKLSFRNICDEEGDVIYL\n",
      "Pieris_brassica NIKLSFSKGSTLSTMVCRIRQYRGLDSGFGVKNQMLSFMPFLLKLSFRNIIDEEGEVIYL\n",
      "Pieris_mannii_E SIKLSFSKGSTLSTMVCRIRKYRGLGGGYGVKNKMVSFLPFLLKLSFRNISDEEGEVIYL\n",
      "Pieris_rapae_Ex SIKLSFCKGSTLSTMVCRIRKYRGLGGGYGVKNKMVSFLPFLLKLSFRNITDEEGEVIYL\n",
      "Pieris_melete_E NIKLSFSKSSTLSTMVCRLRQYRGLGVGFGVKARIVAFMPFLLKLSFKNITDEEGEVIYL\n",
      "Pieris_napi_Exo NIKLSFSKSSTLSTMVCRIRLYRGLGVGFGVKARIVSFMPFLLKLSFRNISDEDGEVMYL\n",
      "                         . ***:***:* ***:  *:*:* : ::*:******.*:*: ** *:*:**\n",
      "\n",
      "Anthocharis_car LLQAIPLFPAYK\n",
      "Aporia_crataegi LVQATPFFSAFK\n",
      "Leptophobia_ari LMQATPFYSTFK\n",
      "Pieris_brassica LVQATPFFPAFK\n",
      "Pieris_mannii_E LVQATPFFPAFK\n",
      "Pieris_rapae_Ex LLQATPFFPAFK\n",
      "Pieris_melete_E LVQATPFFPAFK\n",
      "Pieris_napi_Exo LVQATPFFPAFK\n",
      "                *:** *::.::*\n",
      "\n",
      "continue?Aporia_crataegi,Leptophobia_ari\n",
      "Aporia_crataegi_Exon_8_OU538732.1_3370881_3371072_left_2_right_1\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
