{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1ae39c59",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Get_genome_name(species, genome_location):\n",
    "    import os\n",
    "    list_of_folders = os.listdir(genome_location)\n",
    "    while True:\n",
    "        if species in list_of_folders:\n",
    "            return(species)\n",
    "        else:\n",
    "            input(\"species_missing?\")\n",
    "            break\n",
    "def Genome_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            return(file)\n",
    "\n",
    "def Get_gff_file_name(genome_species_location):\n",
    "    import os\n",
    "    list_of_files = os.listdir(genome_species_location)\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\"_genomic.gff\"):\n",
    "            return(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "5f7fd19c",
   "metadata": {},
   "outputs": [
    {
     "ename": "OSError",
     "evalue": "[Errno 19] No such device: '/mnt/f/Genomes_2023-12-26'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[63], line 22\u001b[0m\n\u001b[1;32m     19\u001b[0m     stop \u001b[38;5;241m=\u001b[39m coordinate_lines[\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m3\u001b[39m]\n\u001b[1;32m     20\u001b[0m scaffold \u001b[38;5;241m=\u001b[39m coordinate_lines[\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m1\u001b[39m]\n\u001b[0;32m---> 22\u001b[0m genome_folder_name \u001b[38;5;241m=\u001b[39m \u001b[43mGet_genome_name\u001b[49m\u001b[43m(\u001b[49m\u001b[43mspecies\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     24\u001b[0m genome_species_location \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mgenome_location\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mspecies\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     25\u001b[0m genome_file_name \u001b[38;5;241m=\u001b[39m Genome_file_name(genome_species_location)\n",
      "Cell \u001b[0;32mIn[20], line 3\u001b[0m, in \u001b[0;36mGet_genome_name\u001b[0;34m(species, genome_location)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mGet_genome_name\u001b[39m(species, genome_location):\n\u001b[1;32m      2\u001b[0m     \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mos\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m     list_of_folders \u001b[38;5;241m=\u001b[39m \u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlistdir\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgenome_location\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      4\u001b[0m     \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m      5\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m species \u001b[38;5;129;01min\u001b[39;00m list_of_folders:\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 19] No such device: '/mnt/f/Genomes_2023-12-26'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "\n",
    "location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/4.Pierinae\"\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "species = \"Pieris_brassicae\"\n",
    "transcript = \"XM_045675385.1\"\n",
    "\n",
    "with open(f\"{location}/1.Blast_result/{species}/Pieris_brassicae_coordinates_old.csv\",'r') as coordinate_file:\n",
    "    coordinate_lines = coordinate_file.readlines()\n",
    "\n",
    "\n",
    "complement = coordinate_lines[1].split(\",\")[4]\n",
    "if complement == \"0\":\n",
    "    start = coordinate_lines[1].split(\",\")[2]\n",
    "    stop = coordinate_lines[-1].split(\",\")[3]\n",
    "if complement == \"1\":\n",
    "    start = coordinate_lines[-1].split(\",\")[2]\n",
    "    stop = coordinate_lines[1].split(\",\")[3]\n",
    "scaffold = coordinate_lines[1].split(\",\")[1]\n",
    "\n",
    "genome_folder_name = Get_genome_name(species, genome_location)\n",
    "\n",
    "genome_species_location = f\"{genome_location}/{species}\"\n",
    "genome_file_name = Genome_file_name(genome_species_location)\n",
    "\n",
    "\n",
    "gff_file_name = Get_gff_file_name(genome_species_location)\n",
    "print(gff_file_name)\n",
    "\n",
    "with open(f\"{genome_species_location}/{gff_file_name}\", 'r') as gff_file:\n",
    "    gff_file_lines = gff_file.readlines()\n",
    "\n",
    "coordinate_switch = 0\n",
    "gene_found = 0\n",
    "exon_dictionary = {}\n",
    "exon_number = 0\n",
    "output = ''\n",
    "chromosome_start = 0\n",
    "# print(start)\n",
    "for line in gff_file_lines:\n",
    "    line_split = line.split(\"\\t\")\n",
    "    if scaffold in line_split:\n",
    "        if \"gene\" in line_split[2]:\n",
    "            if (int(line_split[4]) > int(start)) and (int(line_split[3]) < int(start)):\n",
    "                coordinate_switch = 1\n",
    "#                 print(line)\n",
    "#                 break\n",
    "    if coordinate_switch == 1:\n",
    "        if transcript in line:\n",
    "            if \"CDS\" in line_split[2]:\n",
    "                gene_found = 1\n",
    "                exon_number +=1\n",
    "\n",
    "                exon_dictionary[f\"Exon_{exon_number}\"] = line.split(\"\\t\")[0]+\"\\t\"+line.split(\"\\t\")[3]+\"\\t\"+line.split(\"\\t\")[4]\n",
    "    #             output += line.split(\"\\t\")[0]+\"\\t\"+line.split(\"\\t\")[3]+\"\\t\"+line.split(\"\\t\")[4]+\"\\n\"\n",
    "            if gene_found == 1 and \"CDS\" not in line_split[2]:\n",
    "                break\n",
    "\n",
    "if complement == \"0\":\n",
    "    chromosome_start = exon_dictionary[\"Exon_1\"].split(\"\\t\")[1]\n",
    "    chromosome_stop = exon_dictionary[f\"Exon_{exon_number}\"].split(\"\\t\")[2]\n",
    "if complement == \"1\":\n",
    "    chromosome_start = exon_dictionary[f\"Exon_{exon_number}\"].split(\"\\t\")[1]\n",
    "    chromosome_stop = exon_dictionary[\"Exon_1\"].split(\"\\t\")[2]\n",
    "print(start,stop)\n",
    "print(chromosome_start,chromosome_stop)\n",
    "\n",
    "for i in range(1,len(exon_dictionary)+1):\n",
    "    output += exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[0]+\"\\t\"+str(int(exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[1])-(int(chromosome_start))+100000)+\"\\t\"+str(int(exon_dictionary[f\"Exon_{i}\"].split(\"\\t\")[2])-(int(chromosome_start))+100000)+\"\\t\"+f\"Exon_{i}\"+\"\\n\"\n",
    "\n",
    "print(output)\n",
    "\n",
    "with open(f\"{location}/2.TOGA/1.Query_genomes/{species}/gff_fragment.txt\", 'w') as out_file:\n",
    "    out_file.write(output)\n",
    "    \n",
    "genome_start = int(chromosome_start) - 100000\n",
    "genome_stop = int(chromosome_stop) + 100000\n",
    "\n",
    "\n",
    "\n",
    "subprocess.run(f'samtools faidx \"{genome_species_location}/{genome_file_name}\" {scaffold}:{genome_start}-{genome_stop} > \"{location}/2.TOGA/1.Query_genomes/{species}/query_genome.fa\"', shell = True)\n",
    "\n",
    "print(start,stop)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "5e4e4568",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_905147105.1_ilPieBrab1.1_genomic.gff\n",
      "4554119\n",
      "{'Exon_1': 'NC_059680.1\\t4552213\\t4552234', 'Exon_2': 'NC_059680.1\\t4554093\\t4554204', 'Exon_3': 'NC_059680.1\\t4554742\\t4554825', 'Exon_4': 'NC_059680.1\\t4556586\\t4556740', 'Exon_5': 'NC_059680.1\\t4558509\\t4558600', 'Exon_6': 'NC_059680.1\\t4559038\\t4559082', 'Exon_7': 'NC_059680.1\\t4559287\\t4559418', 'Exon_8': 'NC_059680.1\\t4561142\\t4561217', 'Exon_9': 'NC_059680.1\\t4561627\\t4561845', 'Exon_10': 'NC_059680.1\\t4562078\\t4562306', 'Exon_11': 'NC_059680.1\\t4562761\\t4562872', 'Exon_12': 'NC_059680.1\\t4563542\\t4563655', 'Exon_13': 'NC_059680.1\\t4566676\\t4566825', 'Exon_14': 'NC_059680.1\\t4567504\\t4567637', 'Exon_15': 'NC_059680.1\\t4568226\\t4568342', 'Exon_16': 'NC_059680.1\\t4568891\\t4569031', 'Exon_17': 'NC_059680.1\\t4569618\\t4569795', 'Exon_18': 'NC_059680.1\\t4571131\\t4571266', 'Exon_19': 'NC_059680.1\\t4571578\\t4571670', 'Exon_20': 'NC_059680.1\\t4571786\\t4571893', 'Exon_21': 'NC_059680.1\\t4571999\\t4572231', 'Exon_22': 'NC_059680.1\\t4575221\\t4575358', 'Exon_23': 'NC_059680.1\\t4575876\\t4576028', 'Exon_24': 'NC_059680.1\\t4576780\\t4576884', 'Exon_25': 'NC_059680.1\\t4577160\\t4577261', 'Exon_26': 'NC_059680.1\\t4577809\\t4577952', 'Exon_27': 'NC_059680.1\\t4581533\\t4581631'}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "    \n",
    "print(output)\n",
    "            \n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "348fd6b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "129418 100000\n",
      "NC_059680.1\t100000\t129418\tPeriod\t0\t+\t100000\t129418\t0\t27\t21,111,83,154,91,44,131,75,218,228,111,113,149,133,116,140,177,135,92,107,232,137,152,104,101,143,98,\t0,1880,2529,4373,6296,6825,7074,8929,9414,9865,10548,11329,14463,15291,16013,16678,17405,18918,19365,19573,19786,23008,23663,24567,24947,25596,29320,\n"
     ]
    }
   ],
   "source": [
    "with open(f\"{location}/2.TOGA/1.Query_genomes/{species}/gff_fragment.txt\", 'r') as input_file:\n",
    "    input_bed_file = input_file.readlines()\n",
    "\n",
    "chromosome_start = input_bed_file[0].split(\"\\t\")[1]\n",
    "chromosome = input_bed_file[0].split(\"\\t\")[0]\n",
    "# print(input_bed_file[-1])\n",
    "chromosome_end = input_bed_file[-1].split(\"\\t\")[2].rstrip()\n",
    "print(chromosome_end,chromosome_start)\n",
    "\n",
    "output = f\"{chromosome}\\t{chromosome_start}\\t{chromosome_end}\\tPeriod\\t0\\t+\\t{chromosome_start}\\t{chromosome_end}\\t0\\t{len(input_bed_file)}\\t\"\n",
    "\n",
    "# print(output)\n",
    "size_dictionary = {}\n",
    "sizes = ''\n",
    "pos = ''\n",
    "pos_dictionary = {}\n",
    "for i in range(0,len(input_bed_file)):\n",
    "    line_split = input_bed_file[i].split(\"\\t\")\n",
    "    sizes += f\"{int(line_split[2])-int(line_split[1])},\"\n",
    "    size_dictionary[f\"Exon_{i+1}\"] = int(line_split[2])-int(line_split[1])\n",
    "    pos_dictionary = int(line_split[1]) - 100000\n",
    "    pos += f\"{int(line_split[1]) - 100000},\"\n",
    "#     print(input_bed_file[i])\n",
    "\n",
    "# print(sizes,pos)\n",
    "    \n",
    "output += f\"{sizes}\\t{pos}\"\n",
    "print(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "919b57f0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
