{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3795c17b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1826b335",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import io"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1e064f82",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "69f4b3dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_folder = \"10.Cycle Exon Analysis\"\n",
    "\n",
    "\n",
    "list_of_species = [\"Baronia_brevicornis\",\"Battus_philenor\",\"Iphiclides_podalirius\",\"Luehdorfia_chinensis\",\"Ornithoptera_alexandrae\",\"Ornithoptera_priamus\",\"Papilio_dardanus_tibullus\",\"Papilio_demoleus\",\"Papilio_elwesi\",\"Papilio_gigon\",\"Papilio_glaucus\",\"Papilio_machaon\",\"Papilio_memnon\",\"Papilio_polytes\",\"Papilio_protenor\",\"Papilio_xuthus\",\"Parnassius_glacialis\",\"Sericinus_montela\",\"Teinopalpus_imperialis\",\"Troides_aeacus\",\"Troides_oblongomaculatus\"]\n",
    "query_species = \"Papilio_xuthus\"\n",
    "family_group = \"7.Papilionidae\"\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/{gene_folder}/{family_group}/1.Query\"\n",
    "query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "if \"desktop.ini\" in query_transcript_list:\n",
    "    query_transcript_list.remove(\"desktop.ini\")\n",
    "\n",
    "query_transcript = query_transcript_list[0]\n",
    "\n",
    "list_of_genomes = os.listdir(\"/mnt/f/Genomes_2023-12-26\")\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/{gene_folder}/{family_group}/1.Query/{query_species}/{query_transcript}/query_protein.fa\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4dee5727",
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_file = ''\n",
    "output = ''\n",
    "for species in list_of_species:\n",
    "    species = species.replace(\" \", \"_\")\n",
    "    # if \"\n",
    "    # species = species_folder.split(\".\")[1]\n",
    "    blast_output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/{gene_folder}/{family_group}/1.Blast_result\"\n",
    "    try:\n",
    "        os.mkdir(blast_output_location)\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "    blast_output_folders = os.listdir(blast_output_location)\n",
    "    if species not in blast_output_folders:\n",
    "        os.mkdir(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/{gene_folder}/{family_group}/1.Blast_result/{species}\")\n",
    "    \n",
    "    genome_location = f\"/mnt/f/Genomes_2023-12-26\"\n",
    "#     for genome in list_of_genomes:\n",
    "#         if species == genome:\n",
    "#             files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\n",
    "#             for file in files_in_genome_folder:\n",
    "#                 if re.search('_genomic.fna$', file):\n",
    "#                     genome_file = file\n",
    "#                     break\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    output += f'echo {species}\\ntblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.htm\" -html \\n\\n tblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.txt\"\\n\\n'\n",
    "    \n",
    "with io.open(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/{gene_folder}/{family_group}/tblastn_{query_species}.sh\", \"w\", newline='\\n') as outfile:\n",
    "    outfile.write(output)\n",
    "#             print(\"here\",genome)\n",
    "#     print(species_folder)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00391970",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
