{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "216b3398",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ac483e9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import io\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "220ae581",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "253abc00",
   "metadata": {},
   "outputs": [],
   "source": [
    "# list_of_species = os.listdir(\"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\")\n",
    "list_of_species = [\"Anthocharis_cardamines\",\"Aporia_crataegi\",\"Leptophobia_aripa\",\"Pieris_brassicae\",\"Pieris_mannii\",\"Pieris_melete\",\"Pieris_napi\",\"Pieris_rapae\"]\n",
    "query_species = \"Pieris_brassicae\"\n",
    "family_group = \"4.Pierinae\"\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/1.Query\"\n",
    "\n",
    "query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/1.Query/query.txt\"\n",
    "list_of_genomes = os.listdir(\"/mnt/f/Genomes_2023-12-26\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fe6edab1",
   "metadata": {},
   "outputs": [],
   "source": [
    "working_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}\"\n",
    "output_location = f\"{working_location}/1.Blast_result\"\n",
    "list_of_working_folders = os.listdir(working_location)\n",
    "if \"1.Blast_result\" in list_of_working_folders:\n",
    "    subprocess.run(f'rm -r \"{output_location}\"', shell = True)\n",
    "os.mkdir(output_location)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "genome_file = ''\n",
    "output = ''\n",
    "for species in list_of_species:\n",
    "    # if \"\n",
    "    # species = species_folder.split(\".\")[1]\n",
    "    blast_output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/1.Blast_result\"\n",
    "    blast_output_folders = os.listdir(blast_output_location)\n",
    "    if species not in blast_output_folders:\n",
    "        os.mkdir(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/1.Blast_result/{species}\")\n",
    "    \n",
    "    \n",
    "    for genome in list_of_genomes:\n",
    "        if species == genome:\n",
    "            files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\n",
    "            for file in files_in_genome_folder:\n",
    "                if re.search('_genomic.fna$', file):\n",
    "                    genome_file = file\n",
    "                    break\n",
    "    \n",
    "    output += f'echo {species}\\ntblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.htm\" -html \\n\\n tblastn -seg no -query \"{query_location}\" -db \"/mnt/f/Genomes_2023-12-26/{species}/{genome_file}\" -num_alignments 3 -out \"{blast_output_location}/{species}/{species}_blast_out.txt\"\\n\\n'\n",
    "    \n",
    "with io.open(f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/tblastn_{query_species}.sh\", \"w\", newline='\\n') as outfile:\n",
    "    outfile.write(output)\n",
    "#             print(\"here\",genome)\n",
    "#     print(species_folder)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5a3f687f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Anthocharis_cardamines\n",
      "Aporia_crataegi\n",
      "Leptophobia_aripa\n",
      "Pieris_brassicae\n",
      "Pieris_mannii\n",
      "Pieris_melete\n",
      "Pieris_napi\n",
      "Pieris_rapae\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='sh \"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/4.Pierinae/tblastn_Pieris_brassicae.sh\"', returncode=0)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run(f'sh \"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}/tblastn_{query_species}.sh\"', shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "018a36f0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
