{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f27d4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    out_Seq = ''\n",
    "    \n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = open((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'r')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in SeqIO.parse(fasta_file,\"fasta\"):\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = out_Seq.reverse_complement()\n",
    "                \n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    fasta_file.close()\n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4900aae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d454c091",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989953.1,8403228,8403884,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989955.1,10530615,10531139,1,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989957.1,10287581,10288099,1,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989958.1,3630396,3631379,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989954.1,8450646,8451869,1,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989955.1,5636655,5637359,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989966.1,3414093,3415088,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989952.1,2573769,2575154,0,N,Pieris_brassicae_ef1,1,461,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989961.1,482260,483036,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989951.1,16294806,16296263,1,N,Pieris_brassicae_cat,3,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Anthocharis_cardamines,FR989954.1,6983619,6986756,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Anthocharis_cardamines,FR989953.1,8403228,8403884,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "Anthocharis_cardamines,FR989955.1,10530615,10531139,1,N,Pieris_brassicae_wg,1,175,175\n",
      "Anthocharis_cardamines,FR989957.1,10287581,10288099,1,N,Pieris_brassicae_rps2,1,173,173\n",
      "Anthocharis_cardamines,FR989958.1,3630396,3631379,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "Anthocharis_cardamines,FR989954.1,8450646,8451869,1,N,Pieris_brassicae_idh,1,408,408\n",
      "Anthocharis_cardamines,FR989955.1,5636655,5637359,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "Anthocharis_cardamines,FR989966.1,3414093,3415088,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Anthocharis_cardamines,FR989952.1,2573769,2575154,0,N,Pieris_brassicae_ef1,1,461,462\n",
      "Anthocharis_cardamines,FR989961.1,482260,483036,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "Anthocharis_cardamines,FR989951.1,16294806,16296263,1,N,Pieris_brassicae_cat,3,489,489\n",
      "Anthocharis_cardamines,FR989954.1,6983619,6986756,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538748.1,5303996,5304652,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538736.1,5404408,5404932,0,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538740.1,5474508,5475026,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538743.1,6947935,6948918,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538743.1,4557626,4558849,0,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538729.1,2590129,2590833,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538738.1,5110808,5111803,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538745.1,5768851,5770236,0,N,Pieris_brassicae_ef1,1,461,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538748.1,358574,359350,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538732.1,9739609,9741066,1,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Aporia_crataegi,OU538743.1,5647198,5650335,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Aporia_crataegi,OU538748.1,5303996,5304652,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "Aporia_crataegi,OU538736.1,5404408,5404932,0,N,Pieris_brassicae_wg,1,175,175\n",
      "Aporia_crataegi,OU538740.1,5474508,5475026,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Aporia_crataegi,OU538743.1,6947935,6948918,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "Aporia_crataegi,OU538743.1,4557626,4558849,0,N,Pieris_brassicae_idh,1,408,408\n",
      "Aporia_crataegi,OU538729.1,2590129,2590833,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "Aporia_crataegi,OU538738.1,5110808,5111803,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Aporia_crataegi,OU538745.1,5768851,5770236,0,N,Pieris_brassicae_ef1,1,461,462\n",
      "Aporia_crataegi,OU538748.1,358574,359350,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "Aporia_crataegi,OU538732.1,9739609,9741066,1,N,Pieris_brassicae_cat,1,489,489\n",
      "Aporia_crataegi,OU538743.1,5647198,5650335,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637270.1,3897296,3897952,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637269.1,3205405,3205929,1,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637279.1,3565921,3566439,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637276.1,4970476,4971459,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637264.1,956188,957411,0,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637274.1,3600667,3601371,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637281.1,1258937,1259932,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637280.1,1870602,1871990,1,N,Pieris_brassicae_ef1,1,460,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637263.1,2538775,2539551,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637275.1,978613,980079,0,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Leptophobia_aripa,OX637264.1,2059369,2062506,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Leptophobia_aripa,OX637270.1,3897296,3897952,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "Leptophobia_aripa,OX637269.1,3205405,3205929,1,N,Pieris_brassicae_wg,1,175,175\n",
      "Leptophobia_aripa,OX637279.1,3565921,3566439,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Leptophobia_aripa,OX637276.1,4970476,4971459,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "Leptophobia_aripa,OX637264.1,956188,957411,0,N,Pieris_brassicae_idh,1,408,408\n",
      "Leptophobia_aripa,OX637274.1,3600667,3601371,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "Leptophobia_aripa,OX637281.1,1258937,1259932,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Leptophobia_aripa,OX637280.1,1870602,1871990,1,N,Pieris_brassicae_ef1,1,460,462\n",
      "Leptophobia_aripa,OX637263.1,2538775,2539551,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "Leptophobia_aripa,OX637275.1,978613,980079,0,N,Pieris_brassicae_cat,1,489,489\n",
      "Leptophobia_aripa,OX637264.1,2059369,2062506,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059667.1,5865643,5866299,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059668.1,18414171,18414695,0,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059670.1,16698198,16698716,1,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059676.1,3823265,3824248,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059670.1,10565748,10566971,1,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059671.1,16510769,16511473,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059672.1,7078648,7079643,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059673.1,3233600,3234985,1,N,Pieris_brassicae_ef1,1,462,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059666.1,19523577,19524353,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059680.1,12085055,12086521,1,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_brassicae,NC_059670.1,9123621,9126758,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_brassicae,NC_059667.1,5865643,5866299,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "Pieris_brassicae,NC_059668.1,18414171,18414695,0,N,Pieris_brassicae_wg,1,175,175\n",
      "Pieris_brassicae,NC_059670.1,16698198,16698716,1,N,Pieris_brassicae_rps2,1,173,173\n",
      "Pieris_brassicae,NC_059676.1,3823265,3824248,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "Pieris_brassicae,NC_059670.1,10565748,10566971,1,N,Pieris_brassicae_idh,1,408,408\n",
      "Pieris_brassicae,NC_059671.1,16510769,16511473,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "Pieris_brassicae,NC_059672.1,7078648,7079643,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Pieris_brassicae,NC_059673.1,3233600,3234985,1,N,Pieris_brassicae_ef1,1,462,462\n",
      "Pieris_brassicae,NC_059666.1,19523577,19524353,0,N,Pieris_brassicae_hcl,1,259,259\n",
      "Pieris_brassicae,NC_059680.1,12085055,12086521,1,N,Pieris_brassicae_cat,1,489,489\n",
      "Pieris_brassicae,NC_059670.1,9123621,9126758,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054787.1,5505469,5506125,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054785.1,4672695,4673219,1,N,Pieris_brassicae_wg,1,175,175\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054793.1,5735886,5736404,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054790.1,7902404,7903387,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054782.1,1264517,1265740,0,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054791.1,5387521,5388225,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054796.1,2093618,2094613,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054795.1,6928094,6929479,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054784.1,3842609,3843385,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054800.1,12138753,12140219,1,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_mannii,CM054782.1,2876928,2880065,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_mannii,CM054787.1,5505469,5506125,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "Pieris_mannii,CM054785.1,4672695,4673219,1,N,Pieris_brassicae_wg,1,175,175\n",
      "Pieris_mannii,CM054793.1,5735886,5736404,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Pieris_mannii,CM054790.1,7902404,7903387,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "Pieris_mannii,CM054782.1,1264517,1265740,0,N,Pieris_brassicae_idh,1,408,408\n",
      "Pieris_mannii,CM054791.1,5387521,5388225,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "Pieris_mannii,CM054796.1,2093618,2094613,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Pieris_mannii,CM054795.1,6928094,6929479,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "Pieris_mannii,CM054784.1,3842609,3843385,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "Pieris_mannii,CM054800.1,12138753,12140219,1,N,Pieris_brassicae_cat,1,489,489\n",
      "Pieris_mannii,CM054782.1,2876928,2880065,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000002.1,1073394,1074050,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000358.1,1267113,1267637,1,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000345.1,5382465,5382983,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000040.1,2212625,2213608,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000046.1,937209,938432,1,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000028.1,1284362,1285066,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000087.1,2028013,2029008,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000147.1,447237,448619,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000326.1,3018858,3019634,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000240.1,928509,929975,0,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_melete,CAVNZK010000016.1,539900,543037,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_melete,CAVNZK010000002.1,1073394,1074050,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "Pieris_melete,CAVNZK010000358.1,1267113,1267637,1,N,Pieris_brassicae_wg,1,175,175\n",
      "Pieris_melete,CAVNZK010000345.1,5382465,5382983,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Pieris_melete,CAVNZK010000040.1,2212625,2213608,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "Pieris_melete,CAVNZK010000046.1,937209,938432,1,N,Pieris_brassicae_idh,1,408,408\n",
      "Pieris_melete,CAVNZK010000028.1,1284362,1285066,1,N,Pieris_brassicae_ddc,1,235,235\n",
      "Pieris_melete,CAVNZK010000087.1,2028013,2029008,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Pieris_melete,CAVNZK010000147.1,447237,448619,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "Pieris_melete,CAVNZK010000326.1,3018858,3019634,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "Pieris_melete,CAVNZK010000240.1,928509,929975,0,N,Pieris_brassicae_cat,1,489,489\n",
      "Pieris_melete,CAVNZK010000016.1,539900,543037,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062247.1,6746143,6746799,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062240.1,8651111,8651635,0,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062252.1,5650797,5651315,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062246.1,4357957,4358940,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062243.1,11902911,11904134,1,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062248.1,5725730,5726434,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062255.1,8628452,8629447,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062253.1,3592449,3593831,1,N,Pieris_brassicae_ef1,1,462,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062242.1,3993201,3993977,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062259.1,12962946,12964412,1,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_napi,NC_062243.1,10352662,10355799,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_napi,NC_062247.1,6746143,6746799,0,N,Pieris_brassicae_rps5,1,219,219\n",
      "Pieris_napi,NC_062240.1,8651111,8651635,0,N,Pieris_brassicae_wg,1,175,175\n",
      "Pieris_napi,NC_062252.1,5650797,5651315,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Pieris_napi,NC_062246.1,4357957,4358940,1,N,Pieris_brassicae_mdh,1,328,328\n",
      "Pieris_napi,NC_062243.1,11902911,11904134,1,N,Pieris_brassicae_idh,1,408,408\n",
      "Pieris_napi,NC_062248.1,5725730,5726434,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "Pieris_napi,NC_062255.1,8628452,8629447,0,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Pieris_napi,NC_062253.1,3592449,3593831,1,N,Pieris_brassicae_ef1,1,462,462\n",
      "Pieris_napi,NC_062242.1,3993201,3993977,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "Pieris_napi,NC_062259.1,12962946,12964412,1,N,Pieris_brassicae_cat,1,489,489\n",
      "Pieris_napi,NC_062243.1,10352662,10355799,1,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059520.1,4903369,4904025,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059518.1,4080185,4080709,1,N,Pieris_brassicae_wg,1,175,175\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059526.1,4841164,4841682,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059523.1,6841917,6842900,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059515.1,1204730,1205953,0,N,Pieris_brassicae_idh,1,408,408\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059524.1,4665326,4666030,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059529.1,1822835,1823830,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059528.1,6212279,6213664,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059517.1,3364878,3365654,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059534.1,11102485,11103951,1,N,Pieris_brassicae_cat,1,489,489\n",
      "\n",
      "Pieris_brassicae\n",
      "Pieris_rapae,NC_059515.1,2594664,2597801,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Pieris_rapae,NC_059520.1,4903369,4904025,1,N,Pieris_brassicae_rps5,1,219,219\n",
      "Pieris_rapae,NC_059518.1,4080185,4080709,1,N,Pieris_brassicae_wg,1,175,175\n",
      "Pieris_rapae,NC_059526.1,4841164,4841682,0,N,Pieris_brassicae_rps2,1,173,173\n",
      "Pieris_rapae,NC_059523.1,6841917,6842900,0,N,Pieris_brassicae_mdh,1,328,328\n",
      "Pieris_rapae,NC_059515.1,1204730,1205953,0,N,Pieris_brassicae_idh,1,408,408\n",
      "Pieris_rapae,NC_059524.1,4665326,4666030,0,N,Pieris_brassicae_ddc,1,235,235\n",
      "Pieris_rapae,NC_059529.1,1822835,1823830,1,N,Pieris_brassicae_gapdh,1,332,332\n",
      "Pieris_rapae,NC_059528.1,6212279,6213664,0,N,Pieris_brassicae_ef1,1,462,462\n",
      "Pieris_rapae,NC_059517.1,3364878,3365654,1,N,Pieris_brassicae_hcl,1,259,259\n",
      "Pieris_rapae,NC_059534.1,11102485,11103951,1,N,Pieris_brassicae_cat,1,489,489\n",
      "Pieris_rapae,NC_059515.1,2594664,2597801,0,N,Pieris_brassicae_cad,1,1046,1046\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#%reset -f\n",
    "blast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/4.Pierinae\"\n",
    "query_species = \"Pieris_brassicae\"\n",
    "\n",
    "\n",
    "species_list = os.listdir(f\"{blast_location}/1.Blast_result\")\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "query_file = SeqIO.parse(f\"{blast_location}/1.Query/query.txt\", 'fasta')\n",
    "query_name_list = []\n",
    "\n",
    "for records in query_file:\n",
    "    query_name_list.append(records.id)\n",
    "\n",
    "\n",
    "\n",
    "for species in species_list:\n",
    "    # print(species_name)\n",
    "    # species = species_name.split(\".\")[1]\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    \n",
    "#Run for each query in the query list\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        with open(f\"{blast_location}/1.Blast_result/{species}/{species}_blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "#                 print(query_species_split)\n",
    "                query_species = str(query_species_split[0]+\"_\"+query_species_split[1].rstrip())\n",
    "                print(query_species)\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "#                     if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "#                         error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "\n",
    "\n",
    "# #Check for Met or STOP at the start and end\n",
    "#         while True and (start != 0 or stop != 0):\n",
    "#             if (query_name == query_name_list[0] or query_name == query_name_list[-1]):\n",
    "#                 frame = 1\n",
    "#                 translated_sequence = (sequence_extractor(species, scaff, int(complement), start, stop,frame ))\n",
    "\n",
    "#                 if translated_sequence == old_trans:\n",
    "#                     break\n",
    "                    \n",
    "# #Check for Met at the beginning of first exon\n",
    "\n",
    "#                 if query_name == query_name_list[0] :\n",
    "#                     if translated_sequence[0] != \"M\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             start = int(start) - 3\n",
    "#                         if complement == \"1\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if \"*\" in translated_sequence:\n",
    "#                         error = \"Y\"\n",
    "#                         break\n",
    "#                     if translated_sequence[0] == \"M\":\n",
    "#                         break\n",
    "\n",
    "# #Check for stop at the end of last exon\n",
    "\n",
    "#                 if query_name == query_name_list[-1]:\n",
    "#                     if translated_sequence[-1] != \"*\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         if complement == \"1\":\n",
    "#                             start = int(start) - 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if translated_sequence[-1] == \"*\":\n",
    "\n",
    "#                         break\n",
    "#             else:\n",
    "#                 break\n",
    "        \n",
    "        \n",
    "#Add offset at the beginning and end of each exon\n",
    "#         if (start != 0 or stop != 0):\n",
    "            \n",
    "#             start_modifier = int(query_name.split(\"Frame\")[1][1])\n",
    "#             stop_modifier = int(query_name.split(\"rightoh\")[1][1])\n",
    "#         else:\n",
    "        start_modifier = 0\n",
    "        stop_modifier = 0            \n",
    "\n",
    "        seq_length = query_length\n",
    "        \n",
    "        \n",
    "#Get exon start and end by adding or removing codons\n",
    "        \n",
    "    #if not the beginning\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "#             if old_end != 0 and old_end > stop:\n",
    "\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "#             if old_end != 0 and old_end < stop:\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "            \n",
    "           \n",
    "        output_format = str(species.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        print(output_format)\n",
    "#         assert False\n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "    output_file = open(f\"{blast_location}/1.Blast_result/{species}/{species}_coordinates.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fa07f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465fa823",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
