{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "78df7b31",
   "metadata": {},
   "source": [
    "# Parse from JSON file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e36396fc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import json\n",
    "with open (\"F:/genome_test/ncbi_dataset/data/assembly_data_report.jsonl\", 'r') as json_file:\n",
    "    json_file_as_list = json_file.readlines()\n",
    "\n",
    "output = \"Accession\\tNumber of Scaffolds\\tContig N50\\n\"\n",
    "for accession_detail_line in json_file_as_list:\n",
    "    accession_detail = json.loads(accession_detail_line)\n",
    "    contig_n50 = (int((accession_detail[\"assemblyStats\"][\"contigN50\"]))/1000000)\n",
    "    accession_number = ((accession_detail[\"accession\"]))\n",
    "    scaffold_number = int(accession_detail[\"assemblyStats\"][\"numberOfScaffolds\"])\n",
    "#     print(accession_number,scaffold_number,contig_n50 )\n",
    "    output = output + (f\"{accession_number}\\t{scaffold_number}\\t{contig_n50}\\n\")\n",
    "    \n",
    "with open (\"F:/genome_test/temp_details.txt\",'w') as out_file:\n",
    "    out_file.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f56063a7",
   "metadata": {},
   "source": [
    "# Parse from Txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95dd231d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "\n",
    "with open(\"J:/genome_test/test2/gcf.txt\",'r') as list_file:\n",
    "    \n",
    "    list_of_accessions = list_file.readlines()\n",
    "\n",
    "output = \"Accession\\tScaffold_number\\tContig_N50\\n\"\n",
    "for i in range(len(list_of_accessions)):\n",
    "    accession = list_of_accessions[i].rstrip()\n",
    "#     print(accession)\n",
    "#     break\n",
    "    \n",
    "    location = f\"J:/genome_test/test2/genbank/invertebrate/{accession}\"\n",
    "    try:\n",
    "        files_in_accession_folder = os.listdir(location)\n",
    "    except FileNotFoundError:\n",
    "        output += f\"{accession}\\tN/A\\tN/A\\n\"\n",
    "        continue\n",
    "    for file in files_in_accession_folder:\n",
    "        if \"assembly_stats\" in file:\n",
    "           with open(f\"{location}/{file}\", 'r') as assembly_stat_file:\n",
    "                stat_file_list = assembly_stat_file.readlines()\n",
    "                break\n",
    "    \n",
    "    for lines in stat_file_list:\n",
    "        if \"all\tall\tall\tall\tscaffold-count\" in lines:\n",
    "            scaffold_count = lines.split(\"\t\")[-1].rstrip()\n",
    "        if \"all\tall\tall\tall\tcontig-N50\" in lines:\n",
    "            contig_n50 = int(lines.split(\"\t\")[-1].rstrip())/1000000\n",
    "#     print(scaffold_count, contig_n50)\n",
    "    output += f\"{accession}\\t{scaffold_count}\\t{contig_n50}\\n\"\n",
    "    \n",
    "with open(\"J:/genome_test/test2/accession_details.txt\",'w') as output_file:\n",
    "    output_file.write(output)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b52d3e58",
   "metadata": {},
   "source": [
    "# Code for extraction and makeblastdb\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de32354e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "location = \"J:/Genomes_2023-12-26\"\n",
    "\n",
    "list_of_species = os.listdir(location)\n",
    "output = '#!/bin/zsh\\n'\n",
    "for species in list_of_species:\n",
    "    files_list = os.listdir(f\"{location}/{species}\")\n",
    "    for files in files_list:\n",
    "        if \".gz\" in files:\n",
    "            output += f\"echo {species} \\ngzip -d /mnt/j/Genomes_2023-12-26/{species}/{files}\\n\"\n",
    "            \n",
    "#             print(species)\n",
    "\n",
    "print(output)\n",
    "            \n",
    "# with open(\"J:/My Drive/Circadian Rhythm Genes Project/1.Codes/bash scripts/unzip.sh\", \"w\") as out_file:\n",
    "#     out_file.write(output)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ec95918",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "location = \"J:/Genomes_2023-12-26\"\n",
    "\n",
    "list_of_species = os.listdir(location)\n",
    "output = '#!/bin/zsh\\n'\n",
    "for species in list_of_species:\n",
    "    files_list = os.listdir(f\"{location}/{species}\")\n",
    "    for files in files_list:\n",
    "        if \"genomic.fna\" in files[-11:]:\n",
    "            output += f\"echo {species} \\nmakeblastdb -in /mnt/j/Genomes_2023-12-26/{species}/{files} -dbtype nucl\\n\"\n",
    "    \n",
    "#             print(species)\n",
    "\n",
    "# print(output)\n",
    "            \n",
    "with open(\"K:/My Drive/Circadian Rhythm Genes Project/1.Codes/bash scripts/makeblastdb.sh\", \"w\") as out_file:\n",
    "    out_file.write(output)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f09b92b4",
   "metadata": {},
   "source": [
    "# Parse gene table from NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fd2bc12b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "26\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12490887', '12490759']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511908']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "24\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12483156', '12482956']\n",
      "['12482268', '12482134']\n",
      "['12481645', '12481544']\n",
      "['12478465', '12478217']\n",
      "25\n",
      "['12522661', '12522573']\n",
      "['12521246', '12521163']\n",
      "['12520999', '12520845']\n",
      "['12514423', '12514329']\n",
      "['12513351', '12513274']\n",
      "['12512245', '12512114']\n",
      "['12511968', '12511893']\n",
      "['12510243', '12510040']\n",
      "['12508701', '12508473']\n",
      "['12507855', '12507744']\n",
      "['12506233', '12506114']\n",
      "['12505865', '12505716']\n",
      "['12503906', '12503788']\n",
      "['12503481', '12503392']\n",
      "['12502301', '12502149']\n",
      "['12501492', '12501303']\n",
      "['12498481', '12498334']\n",
      "['12497509', '12497396']\n",
      "['12497300', '12497286']\n",
      "['12496602', '12496486']\n",
      "['12494243', '12494059']\n",
      "['12492407', '12492237']\n"
     ]
    }
   ],
   "source": [
    "import re \n",
    "from Bio import Entrez, SeqIO, Align\n",
    "from Bio.Align import substitution_matrices\n",
    "from io import StringIO\n",
    "\n",
    "aligner = Align.PairwiseAligner()\n",
    "aligner.mode = 'global'\n",
    "aligner.substitution_matrix = substitution_matrices.load(\"BLOSUM62\")\n",
    "aligner.open_gap_score = -10\n",
    "aligner.extend_gap_score = -0.5\n",
    "aligner.target_end_gap_score = 0.0\n",
    "aligner.query_end_gap_score = 0.0\n",
    "\n",
    "\n",
    "Entrez.email = 'saubaral@gmail.com'\n",
    "\n",
    "\n",
    "def get_protein_sequence(ncbi_protein_id):\n",
    "    handle = Entrez.efetch(db=\"protein\", id=ncbi_protein_id, rettype=\"fasta\", retmode=\"text\")\n",
    "    record = handle.read()\n",
    "    fasta_io = StringIO(record) \n",
    "    protein_sequence_element = SeqIO.parse(fasta_io, \"fasta\")\n",
    "    for record in protein_sequence_element:\n",
    "        return(record)\n",
    "\n",
    "\n",
    "species = \"1.Per period[Bombyx mori]\"\n",
    "bombyx_protein_accession = \"NP_001036975.1\"\n",
    "bombyx_protein = get_protein_sequence(bombyx_protein_accession)\n",
    "\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table.txt\",\"r\") as gene_table_file:\n",
    "    gene_table_list = gene_table_file.readlines()\n",
    "\n",
    "    \n",
    "exon_list = \"Exon1,Exon2,Exon3,Exon4,Exon5,Exon6,Exon7,Exon8,Exon9,Exon10,Exon11,Exon12,Exon13,Exon14,Exon15,Exon16,Exon17,Exon18,Exon19,Exon20,Exon21,Exon22,Exon23,Exon24,Exon25,Exon26,Exon27,Exon28,Exon29,Exon30,Exon31,Exon32,Exon33,Exon34\".split(\",\")\n",
    "i = 0\n",
    "switch = 0\n",
    "output = \"mRNA Accession\\tExon Number\\tCoding Length\\tGenome Start Point\\tGenome Stop Point\\n\" \n",
    "output2 = \"mRNA Accession\\tProtein Accession\\tExon Number\\tAlignment Score with Bombyx mori\\n\"\n",
    "exon_counter = 0\n",
    "\n",
    "while i < (len(gene_table_list)):\n",
    "    \n",
    "    if \"Exon table for  mRNA  \" in gene_table_list[i]:\n",
    "        gene_table_line = gene_table_list[i]\n",
    "        while re.search(\"[ ]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"  \",\" \")\n",
    "            \n",
    "        mrna_accession = gene_table_line.split(\" \")[4]\n",
    "        protein_accession = gene_table_line.split(\" \")[7].rstrip()\n",
    "#         print(get_protein_sequence(protein_accession))\n",
    "#         alignment_score = aligner.score(bombyx_protein.seq, get_protein_sequence(protein_accession).seq)\n",
    "#         print(alignment_score)\n",
    "        \n",
    "#         for alignment in alignments:\n",
    "#             print(\"Score = %.1f:\" % alignment.score)\n",
    "#         assert False\n",
    "        \n",
    "#         print(f\"{mrna_accession}\\t{protein_accession}\\t{alignment_score}\", end = \"\\t\")\n",
    "        \n",
    "        switch = 1\n",
    "        i += 3\n",
    "    if switch == 1:\n",
    "\n",
    "        gene_table_line = gene_table_list[i]\n",
    "        while re.search(\"[\\t]{2,}\", gene_table_line):\n",
    "            gene_table_line = gene_table_line.replace(\"\\t\\t\",\"\\t\")\n",
    "#         print(len(gene_table_line.split(\"\\t\")))\n",
    "        if len(gene_table_line.split(\"\\t\")) == 7 or len(gene_table_line.split(\"\\t\")) == 6:\n",
    "            \n",
    "            coding_exon_length =  ((gene_table_line.split(\"\\t\")[5]))\n",
    "            genome_start_point,genome_end_point = ((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "            print((gene_table_line.split(\"\\t\")[1]).split(\"-\"))\n",
    "            output += f\"{mrna_accession}\\t{exon_list[exon_counter]}\\t{coding_exon_length.rstrip()}\\t{genome_start_point}\\t{genome_end_point}\\n\"\n",
    "            \n",
    "        \n",
    "            exon_counter += 1\n",
    "        \n",
    "        if switch == 1 and re.search(\"^mRNA\",gene_table_list[i]):\n",
    "            switch = 0\n",
    "            print(exon_counter)\n",
    "#             output2 += f\"{mrna_accession}\\t{protein_accession}\\t{exon_counter}\\t{alignment_score}\\n\"\n",
    "            output += \"\\n\\n\"\n",
    "            exon_counter = 0\n",
    "            \n",
    "    i += 1\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table_processed1.tsv\",'w') as out_file:\n",
    "    out_file.write(output)\n",
    "\n",
    "with open(f\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/0.Species/{species}/Gene_table_alignment_score1.tsv\",'w') as out_file:\n",
    "    out_file.write(output2)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "351cc700",
   "metadata": {},
   "source": [
    "# Download ncbi sequence using accession"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "dd002047",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">XM_022976811.1 PREDICTED: Spodoptera litura period circadian protein (LOC111360701), transcript variant X3, mRNA\n",
      "GCTCCGGCTAGCTGAACGTTTTGTGAGAAGTAACTCGTAACTCATTCGTTTTGTCGTTTGTTCTTTTATT\n",
      "AGGTACTGTTAAACTTATATATTCGTCGATCAATATTAGATGTCGAAAAGTTTAAAAGTGCTATAATTTG\n",
      "TTACGATAATAACGTACTAATTACTGTTTGAAACTTGATTAACTAAAGTGACCCTTGTGAAAACGGAATA\n",
      "TTCACCTACTCTAAACAAGGATAACGTGATTTCGGCATTGTTTCTTTTTAAAGACCATCAATGCAACATG\n",
      "GACAACTTGGATGATTCAGAGAACAATGCGAAGATTTCAGATTCGGCATACTCCAACAGTTGCAGTAATA\n",
      "GTCAATCTAGAAGGAGTCATAGTTCTAAGTCAACACATTCTGGAAGCAATTCCAGTGGCAGCAGTGGCTA\n",
      "TGGCGGGAAACCCTCCACATCTGGAAGCAGCAACAACTTGGGTCAACCACCTAAAAAAGATAAGGAACCC\n",
      "AAGAAAAAGAAGCCTCAAATTGAAACAATGGTCCCAGATGTAGTTGTGGAAACGGAGGCAAGGGCGCCTG\n",
      "AGCCGGCCCCTACATTTGATGCTCCGAAAGAAGATCAGCTAGATGTTATCCCGTCGCTACCACCTGCGCA\n",
      "TGCGCAATCAGAGAAGGGAGTGGAGAACATGGACATATGTACCCCTGAGTTGAATACTTTAAAAGATGAG\n",
      "GTGGTTTCCTGCAACATATCTCAGGTTAATCCTGCTTCAAGCTTTGTTACCGGTCGTCCCCTGCAAGCTT\n",
      "CTTATTGCTCGGTACAAAGTAAACGATACCAAGAGATTGGCCGCTTCGAGATGCGATGTGAGAGCTCTGA\n",
      "ACAGTTGCGTTGCAAGGATGGATTTTCATGCGTGATATCTATGCACGATGGTGTCGTAATGTACGCAACG\n",
      "TCATCATTGACGACTACTCTTGGTTTCCCCAAAGACATGTGGATTGGAAGATCATTCATCGACTTCGTTC\n",
      "ATCCAAGGGATCGCAACACCTTTGCATCACAAATCACGAATGGTTTGGCCGTCCCTAAAATTGTTAATGG\n",
      "TACGCAAGAAAAAGTTCAATCTCCAGCTAACTCGGTGTCAACAATGGTTTGTCGGATACGGCGGTACAGA\n",
      "GGCCTCACTACAGGGTTCGGTGTTAAAGAGAGAGTCGTGGGGTTCATGCCGTTCCTGCTCAAACTCACGT\n",
      "TCAAAAACATAAGCGATGAAGAGGGTAAAGTCATCTACTTGGTTATACAGGCGACACCATTTTTCTCTGC\n",
      "GTTTAAAATGCCGAACGAGGTGGTGGTAAAAGCGGTGCCTTTTGTCATAAGGCATGCGGCTAACGGTCAC\n",
      "ATTGAATATGTGGATCCTGAGTCTGTTCCGTACCTGGGGTACTTGCCTCAAGACCTCGCTGACAAGGACG\n",
      "CGCTGATGCTATACCATCCTGATGACTTGATGTATCTACGACAAGTGTATGATACCATTGTCAAAGAAGG\n",
      "AGGTCTCCCTAGATCTAAGGCTTATCGTATGATTACACAAAACGGTGACTACATTCGGCTGGAAACGGAG\n",
      "TGGTCATCGTTCATCAATCCTTGGTCTAAGAAACTGGAGTTCGTTATTGGAAAACACCATATTTTAGAGG\n",
      "GCCCGACCAACCCAGACGTCTTTCAATGTCCTGAGCCGGAAAAATCAATGAAAGTTCCCGAAGAAGAAAA\n",
      "AAACAAGGCACAAATGCTGAGAGAGACCATTATTCGAACAATGAATGAGGCTCTGACTAAACCCGCGGAA\n",
      "ATAGCGAAGCAACAGATGAGCAAGCGTTGTCAAGATCTTGCTTCTTTCATGGAGAGTCTAATGGAGGAAG\n",
      "TGCCCAAGACTGATGAGGAACTGCGCTTAGAGATTCACGATCCTGATCACAGCTATTATGAAAGGGACTC\n",
      "GGTGATGCTTGGTGGGATTTCCCCACACCATGACTATTATGACAGCAAGTCTAGTACTGAGACTCCGTTG\n",
      "AGTTACAACCAATTGAACTATAATGAAACGCTGCAGAGGTATTTCGATAGTCACCAACCTATCTCCTATG\n",
      "AGGACTACAACACGGTCACCGGCGAAAACATTCTCGGACTGAAAGATCCGAGACCAATCATAAATCATTG\n",
      "CCTGTCTCCGATGGCGCAGCATTCGGGAGACTCTGGGGAAATGACCTGTTCGTCTGATTCCAACGGCATG\n",
      "GTCATGGGCAGTAACAGTCCCGTGATGCCCCTCGGAGACTACCAACCAATACGTCTCACTGAGGCGTTGC\n",
      "TTAGCAAGCACAATGCTGAGATGGAAAAAGAGTTAATGAAGATGCATAGAGAGACTCGTTCTAGCAGTAA\n",
      "GGGGGATCGGGAGAAGAACTCAAATGAGACTAGGAAGAAGAAGAAGGAGCACTTGGCGCGTTGCAACGCG\n",
      "TCCTACCATCCTACTAGTGCTGGCGCCACCAATACTGAACATCAGCAACCTCACGGTGTAAAGCGTACAT\n",
      "CTAAAATGATGGACGCTGAGGCCGGCGCTCACAAGCACCACTGCCCGTCGCCGCGTCAGGCGCGTCGCAA\n",
      "TAAACGCACGTTGTCTACCAACGCGGCCGTCGCCCAACCATCTACCAGCGTTACCACGACCACTGTCGGT\n",
      "CAGTTGGCTACAGCATCGAATCATTGGCCAACCCCTCCTGCAAACAATATGAACACGTTCATCTTGGGCG\n",
      "TAGGAATCCCTCAGCAGATGTCTATAATGAGCCCCATGCCTCATCCTATGCAGGCCATGCACACTATGCC\n",
      "GGCGGTGCCAGGCATGTTCCCGATGTACTACGCGCCGGCGGCAGCGCCGCAACCGATGCCAACAAGCTCA\n",
      "GCGCATCATCACATTCCTTCAAGCAGCCAGCATCAGTTTCCTGCACCCATGATGATGTACAGCCAGCCAA\n",
      "TGTACGGAGCTCCTTTCATGTACTCGCCAATGGCCCAACAAATGTCTTACCCCATGCAGCAATCTCACAT\n",
      "GATGGCACAGTCCATGCAACAATACACCAATACCATGAACCCTTTAGGGTTGACCAGCAGCAACTATGAA\n",
      "GAGGCTTGTAAGCCCAGTTTAACATTGAGGCCAGGCAAGGTGCAAGGCAGTGTGTGGCGCGACAAGAAAA\n",
      "GAGCAGAATCTCAAGGCACATCTTCAAAAACAGACAACGGCAGCGCAGATAGTAACTCATCTACGAATGC\n",
      "GTTGAATCGGGTGTCGGGGACGCAAAAAAGTCACAGTGGGAACGAGAAGTCTACGATATACTCGACTGAC\n",
      "ATCAGATCTTCAAACAGTATGAGCGCTAAGAGCACAGAGGCTGTGCCGAACACTCCACCTTCAGCCAGGG\n",
      "AGAGTCGCCTCAGCAAGCTGAATCGGCTGGGCAACAGCGAGGAGACCCCCGACAAGACTGATGGAGAGTC\n",
      "TAGCTACTCGTCCTTCTACTCTTCCTTTTTTAAGACTGAGTCCGGTAGTGCCGAGGATAGTGGTGACGGC\n",
      "AAGAATGGCAAGGACAAGCAAGCTAGGAGCCCCATAAATCTGACATCTTCATACCATGTATCTCACCCAA\n",
      "CGAAGAAGGTTGCCAGAAGGAAAATGGAACCGCCATGGTTGGAACAGGTATGCGTGACATCAGAGTTAGT\n",
      "CTACAAGTACCAAATTCTTACCAAGTCTATGGAGGAGACTTTATCCTCTGACAAACAGAAGATGAAAACT\n",
      "CTGGAACAGCCTTCACTAGTGAACGAGCAGCTGAGCCAGTTGTACCTAGACCTTCAGCTGGAAGGTGCTG\n",
      "CGGCGAGGCTGACCTTGGAGGAAGGTATCACTAGTTCCAGTAGTTCCGGTGAAGAAACCACCACAGTTAC\n",
      "AAAGCCTGCTTCTAACAGAGCCTCAACTTCTGGAAAAAGCGAAGGAAAGAGACATACGACTTCACGTAGG\n",
      "AAACGGGAGTACAGCAAGTTAGTTATGATTTACGAAGAAGATGCGCCGTTGCCGCCGCCTGCTGAAGACG\n",
      "TCGTGGAAGGGGGTGAGGCATCCAGCTCGTCCACTTGTTAAATTAGGACCGTCAAAGGTCTAGTCGCAAG\n",
      "TCTTCGTCACCCTCGTCACATTATGTCGTAGCAATGCACAATAATAGTAAATTAATTTGCATTTAAGCGT\n",
      "CATGATAAAACGCACGACGGTTCAGGGAATATTAAACTTCAAATATTCGTAAATAGACATATAGTTGTGT\n",
      "GTAAGTATGAACACGTGTCGTAGCTGTGTGTTTGACGGGCCTTTCTGTACTATGATGAAGATATCGTGCT\n",
      "TGAGCTCGTGTCTTCGAACCCAGATCTCATTCAGTTCTCGAGTTAACTTGAATGCAGTAGATGTGTGGTG\n",
      "TGATTCTAACTGTAACTTAAGTCTCTGTGTTACATTTATCTTATGTATAGACTGCCATTAGCACCAAATA\n",
      "ATAGTGATCGTAGACTACTTTGCTTAATTGATAAATATGGAAATTGAATTATTTTTGTTACGATTTTGTT\n",
      "GCGTGTGTTATTATTTTGTTTGTATTTTATGAAACAGATTCTAATACTGAATTTTAAAATATTTGATATA\n",
      "AACGTGAGTTAATCTGGGAATGTTAAGAATACAAGATTTTGTTTGTCCTATTCGAATCTGTGATACTCAA\n",
      "ATTCCTATTACAGATAGGTAGTACAACTAGATAAAAAAGAGATTTATTAATATAATTATGTATCTATGTC\n",
      "ATTTAATGTAAATGACTTATTAATATTATTATATTGGGGAAAGTAGTGAACTCAGTTTGTGCGCTTATTG\n",
      "TATCTTTAGTTCTCTTAACATTCCGTTTGGTGCTCTGTTGCTCACAATGGCTTGAAAAGTATTTTAACAG\n",
      "GGTAAAGTATTATGATGAATCAGAATCAAATATTATATTTTCATAATAGGCAAGTTTCCGACTAGTCAAG\n",
      "ATAAATACTTTTTTGCAAAGTCAAAGGATATTTTTTTTTGATTTTGTAGGTACGAGATTGGGTGCTGTGA\n",
      "CATAACAGCATACAGTTTTCTTATGTTTCTAAGAAAATACGAAAATTTCGTATTAAAGAAAATGACTTTT\n",
      "TTTTTTAATTTTACTGTATTATAAAGTCAAATTACTGACTAAGGTATTCTATGAGTTTTGTAGATACCTA\n",
      "CAAATCTTCGCAGTCTGTTTGTTGTTATGTCACAATAATTCCACGTCAAGTACTGCCAAATAATGCACTC\n",
      "ATTTATTTAAAAATATTTGAAAATTAAGTAAATTAATTACATGATCGAATCAATTAATGAAAGTATGATT\n",
      "ATTTTTAATATTCTATTGTATTAATTGGTCGAAATAATGTATTGTTGACTGTGGAAACTACCCTACTTTA\n",
      "TTTATACACATAAATATATATATATGAGCGTTATGTTTATTGAATTATACTTACAGTCAAAGACTTATTG\n",
      "CATTTACTCGATCACGCAAGACATTGCTATGTTGGTTGTGCGTTGTCCTATTTATGCATTTGTAATAAAT\n",
      "AAAATAAAGAGCTGGACTGACTTGAACGTATTGTGATATGAAAGAATTGTCAGATTTATTATCATTTGTT\n",
      "GTATACTTAAGTGGTAATTTTTGCGTTATCGAGTGATTATTATGTTAAACGTGTCTTAGTTTTATTTTAT\n",
      "GAAACAAAGTTGTCAAAATGAACCCATCATCGTAACAGGTGCCTTAGAATCGATCTTTAAACTATTGTTC\n",
      "AGTGTCAATTTAATCGATCAAAATTACAATAAATTGATTTTATTTTGTTTCACGATGTCATTAGAGTAGG\n",
      "GTGTGTTGATTAAGCTGTTGGCAAAAGTCCGTGCAGGGTTTGACATTATCAACTTTTGTTTAGTATTTCC\n",
      "TACTAAAAGAATAACAATCTTTTTGAGATATCTCTTTATATTTAAATCGAGTTTGTAATTAAATAATTGA\n",
      "TATAATGCATTTATTATAAATGGTATATGTGATTGACGAATTGATGAGTTTTTAATGTGTTGAAAGTCTA\n",
      "GTTTTGTATGACTCGAAATATATTTTTTAAAAGAGAAATTTTAAAAGTTATGGAATTGTCGTGCGGTCTT\n",
      "TAAATGCCAAGGTTACCAAATTGATGTAGGTTATAGAAAATTTGAATACTTAAATGAGGCATTCATTCCT\n",
      "ACGTTGAGAATCAGAACAGACAGAATAGGTGACTGTGGTTACCTTAGTCCTAGTTAAGATATACAGTAAG\n",
      "TGTGATTAACGTTAAGTTTGATACGATAGTTATTTGTAAATATTATTTGTGCACTAACGATTACAATAAG\n",
      "TTGGTGTTTACTATTATGTTAAATTAAAGTTATCCGTTTCTAATAGAGACGTATAAGAGAATTTTATCAC\n",
      "TAAATTGCAATATAGTATAGTATTCATGCTGTTTTATAGTTGAATACATAGTTTATTACATGCATAGATG\n",
      "TTTCTTTTTATTTTTATCTTGTGTGTGGTAACTGTCCTACCATTTTTGTCAATAAAAGAGAAAGAAATGT\n",
      "TGAATTTATTACATGCCAGTAATTTTATTCTTATTTTGAATTAAAGTGGCGATTAAATAGAAGCGGTATT\n",
      "TTTTTTAATGGTAAATTCTAGTTTCCTTTTAATAAACCGTGCCGTTATTTTGTCGTCTATCACTTTAGTC\n",
      "TTCTGTCTAATTTAGATTATTAAAGCTTATAAGGGAGGGGAGGAGAATTACGTTTTTCTTAAAATCATTG\n",
      "CCATTAAATAATTAATTTTATTTGCTAATTATTACGATTCTGAAAACGAATTGGCGTTTATTTTATTGAT\n",
      "GTTGCTTATTATTTTTAAAATAATTGTATTATGAGAAAATTGAAATGTTACGAATAATTCACCAATATAT\n",
      "GACATTTCTGTGAAATAAATGATTGGAAAGTGAATA\n",
      "\n",
      "\n",
      "XM_022976811.1\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez, SeqIO\n",
    "from io import StringIO\n",
    "\n",
    "Entrez.email = 'saubaral@gmail.com'\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"XM_022976811.1\", rettype=\"fasta\", retmode=\"text\")\n",
    "# print (handle)\n",
    "record = handle.read()\n",
    "print(record)\n",
    "fasta_io = StringIO(record) \n",
    "record1 = SeqIO.parse(fasta_io, \"fasta\")\n",
    "for sequences in record1:\n",
    "    print(sequences.id)\n",
    "# record.write(record.rstrip('\\n'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b53dc81",
   "metadata": {},
   "source": [
    "# Parse BLAST result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "e5196c32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bombyx mori period (Per), mRNA\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "\n",
    "file_list = {}\n",
    "enumerator = 0\n",
    "with open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/T0260275016-Alignment-Descriptions.csv\", 'r') as blast_out_file:\n",
    "    blast_out_file_content = csv.reader(blast_out_file, delimiter=',', quotechar='\"')\n",
    "\n",
    "    \n",
    "    for data in blast_out_file_content:\n",
    "        file_list[str(enumerator)] = data\n",
    "        enumerator += 1\n",
    "#         break\n",
    "\n",
    "output = (\"\\t\".join(file_list[str(0)]))\n",
    "\n",
    "\n",
    "for i in range(1,len(file_list)-1):\n",
    "    try:\n",
    "        for k in file_list[str(i)]:\n",
    "#             print (k)\n",
    "            if \"NM_\" in k:\n",
    "                print(file_list[str(i)][0])\n",
    "                output+= \"\\n\"+((\"\\t\".join(file_list[str(i)])))\n",
    "    #         break\n",
    "    except:\n",
    "        print(f\"Error{file_list[str(i)]}\")\n",
    "\n",
    "# output = \"\\n\".join(out_list)\n",
    "\n",
    "# with open(\"H:/My Drive/Circadian Rhythm Genes Project/3.Genes from NCBI/1.Period/1.Blast_output/filtered_blast.tsv\", 'w') as blast_output_file:\n",
    "#     blast_output_file.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4714dfa8",
   "metadata": {},
   "source": [
    "# Genomes with annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "78271f0b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species\tAnnotation\n",
      "Abrostola_tripartita\tNO\n",
      "Abrostola_triplasia\tNO\n",
      "Acentria_ephemerella\tNO\n",
      "Achlya_flavicornis\tNO\n",
      "Acleris_cristana\tNO\n",
      "Acleris_emargana\tNO\n",
      "Acleris_holmiana\tNO\n",
      "Acleris_literana\tNO\n",
      "Acleris_sparsana\tNO\n",
      "Acrobasis_consociella\tNO\n",
      "Acrobasis_repandana\tNO\n",
      "Acrobasis_suavella\tNO\n",
      "Acronicta_aceris\tNO\n",
      "Acronicta_leporina\tNO\n",
      "Acronicta_psi\tNO\n",
      "Adoxophyes_honmai\tNO\n",
      "Agonopterix_arenella\tNO\n",
      "Agonopterix_heracliana\tNO\n",
      "Agonopterix_subpropinquella\tNO\n",
      "Agriopis_aurantiaria\tNO\n",
      "Agriopis_leucophaearia\tNO\n",
      "Agriopis_marginaria\tNO\n",
      "Agriphila_geniculea\tNO\n",
      "Agriphila_straminella\tNO\n",
      "Agriphila_tristella\tNO\n",
      "Agrochola_circellaris\tNO\n",
      "Agrochola_litura\tNO\n",
      "Agrochola_lota\tNO\n",
      "Agrochola_lychnidis\tNO\n",
      "Agrochola_macilenta\tNO\n",
      "Agrotis_clavis\tNO\n",
      "Agrotis_exclamationis\tNO\n",
      "Agrotis_ipsilon\tNO\n",
      "Agrotis_puta\tNO\n",
      "Alcis_repandata\tNO\n",
      "Allophyes_oxyacanthae\tNO\n",
      "Alsophila_aescularia\tNO\n",
      "Amorpha_juglandis\tNO\n",
      "Amphipoea_lucens\tNO\n",
      "Amphipoea_oculea\tNO\n",
      "Amphipyra_berbera\tNO\n",
      "Amphipyra_tragopoginis\tNO\n",
      "Amyelois_transitella\tYes\n",
      "Anania_crocealis\tNO\n",
      "Anania_fuscalis\tNO\n",
      "Anania_hortulata\tNO\n",
      "Anarsia_innoxiella\tNO\n",
      "Anorthoa_munda\tNO\n",
      "Antheraea_mylitta\tNO\n",
      "Antheraea_pernyi\tNO\n",
      "Anthocharis_cardamines\tNO\n",
      "Anticlea_derivata\tNO\n",
      "Antitype_chi\tNO\n",
      "Apamea_anceps\tNO\n",
      "Apamea_crenata\tNO\n",
      "Apamea_epomidion\tNO\n",
      "Apamea_monoglypha\tNO\n",
      "Apamea_sordens\tNO\n",
      "Apeira_syringaria\tNO\n",
      "Aplocera_efformata\tNO\n",
      "Apocheima_hispidaria\tNO\n",
      "Apoda_limacodes\tNO\n",
      "Apomyelois_bistriatella\tNO\n",
      "Aporia_crataegi\tNO\n",
      "Aporophyla_lueneburgensis\tNO\n",
      "Aporophyla_nigra\tNO\n",
      "Apotomis_betuletana\tNO\n",
      "Apotomis_capreana\tNO\n",
      "Apotomis_turbidana\tNO\n",
      "Aproaerema_taeniolella\tNO\n",
      "Archips_crataeganus\tNO\n",
      "Archips_xylosteana\tNO\n",
      "Arctia_plantaginis\tYes\n",
      "Argynnis_bischoffii_washingtonia\tNO\n",
      "Argyresthia_albistria\tNO\n",
      "Argyresthia_goedartella\tNO\n",
      "Aricia_agestis\tYes\n",
      "Aricia_artaxerxes\tNO\n",
      "Asteroscopus_sphinx\tNO\n",
      "Atethmia_centrago\tNO\n",
      "Athetis_lepigone\tNO\n",
      "Athrips_mouffetella\tNO\n",
      "Autographa_gamma\tNO\n",
      "Autographa_pulchrina\tNO\n",
      "Baronia_brevicornis\tNO\n",
      "Battus_philenor\tNO\n",
      "Bedellia_somnulentella\tNO\n",
      "Bembecia_ichneumoniformis\tNO\n",
      "Bicyclus_anynana\tYes\n",
      "Biston_betularia\tNO\n",
      "Biston_stratarius\tNO\n",
      "Blastobasis_adustella\tNO\n",
      "Blastobasis_lacticolella\tNO\n",
      "Boloria_euphrosyne\tNO\n",
      "Boloria_selene\tNO\n",
      "Bombyx_mandarina\tNO\n",
      "Bombyx_mori\tYes\n",
      "Brachylomia_viminalis\tNO\n",
      "Brenthis_daphne\tNO\n",
      "Brenthis_hecate\tNO\n",
      "Brenthis_ino\tYes\n",
      "Cabera_pusaria\tNO\n",
      "Calamotropha_paludella\tNO\n",
      "Calephelis_nemesis\tNO\n",
      "Calephelis_virginiensis\tNO\n",
      "Callimorpha_dominula\tNO\n",
      "Calycopis_cecrops\tNO\n",
      "Campaea_margaritaria\tNO\n",
      "Camptogramma_bilineatum\tNO\n",
      "Caradrina_clavipalpis\tNO\n",
      "Caradrina_kadenii\tNO\n",
      "Carcina_quercana\tNO\n",
      "Carpatolechia_fugitivella\tNO\n",
      "Carposina_sasakii\tNO\n",
      "Carterocephalus_palaemon\tNO\n",
      "Catocala_fraxini\tNO\n",
      "Catocala_nupta\tNO\n",
      "Catocala_sponsa\tNO\n",
      "Catoptria_pinella\tNO\n",
      "Celastrina_argiolus\tNO\n",
      "Ceramica_pisi\tNO\n",
      "Cerastis_leucographa\tNO\n",
      "Cerastis_rubricosa\tNO\n",
      "Charanyca_ferruginea\tNO\n",
      "Charaxes_varanes\tNO\n",
      "Chesias_legatella\tNO\n",
      "Chilo_suppressalis\tYes\n",
      "Chloroclysta_siterata\tNO\n",
      "Chloroclystis_v-ata\tNO\n",
      "Choreutis_nemorana\tNO\n",
      "Choristoneura_fumiferana\tYes\n",
      "Chrysodeixis_includens\tYes\n",
      "Chrysoteuchia_culmella\tNO\n",
      "Clepsis_dumicolana\tNO\n",
      "Clostera_curtula\tNO\n",
      "Cnaphalocrocis_medinalis\tNO\n",
      "Coenonympha_glycerion\tNO\n",
      "Coleophora_deauratella\tNO\n",
      "Coleophora_flavipennella\tNO\n",
      "Colias_behrii\tNO\n",
      "Colias_croceus\tYes\n",
      "Colias_eurytheme\tNO\n",
      "Colias_nastes\tNO\n",
      "Colostygia_pectinataria\tNO\n",
      "Conistra_vaccinii\tNO\n",
      "Conogethes_punctiferalis\tNO\n",
      "Conopomorpha_cramerella\tNO\n",
      "Cosmia_pyralina\tNO\n",
      "Cosmia_trapezina\tNO\n",
      "Cosmorhoe_ocellata\tNO\n",
      "Crambus_lathoniellus\tNO\n",
      "Craniophora_ligustri\tNO\n",
      "Crocallis_elinguaria\tNO\n",
      "Cyaniris_semiargus\tNO\n",
      "Cybosia_mesomella\tNO\n",
      "Cyclargus_thomasi\tNO\n",
      "Cyclophora_albipunctata\tNO\n",
      "Cyclophora_punctaria\tNO\n",
      "Cydalima_perspectalis\tNO\n",
      "Cydia_amplana\tNO\n",
      "Cydia_fagiglandana\tNO\n",
      "Cydia_pomonella\tNO\n",
      "Cydia_splendana\tNO\n",
      "Cydia_strobilella\tNO\n",
      "Danaus_chrysippus\tNO\n",
      "Danaus_plexippus\tYes\n",
      "Dasypolia_templi\tNO\n",
      "Deilephila_elpenor\tNO\n",
      "Deilephila_porcellus\tNO\n",
      "Dendrolimus_kikuchii\tYes\n",
      "Dendrolimus_pini\tNO\n",
      "Dendrolimus_punctatus\tNO\n",
      "Diachrysia_chrysitis\tNO\n",
      "Diaphora_mendica\tNO\n",
      "Diarsia_brunnea\tNO\n",
      "Diarsia_dahlii\tNO\n",
      "Diarsia_mendica\tNO\n",
      "Diarsia_rubi\tNO\n",
      "Diatraea_saccharalis\tYes\n",
      "Dicycla_oo\tNO\n",
      "Diloba_caeruleocephala\tNO\n",
      "Dircenna_loreta\tNO\n",
      "Ditula_angustiorana\tNO\n",
      "Drepana_arcuata\tNO\n",
      "Drepana_falcataria\tNO\n",
      "Dryadula_phaetusa\tNO\n",
      "Dryas_iulia_moderata\tNO\n",
      "Drymonia_ruficornis\tNO\n",
      "Dryobota_labecula\tNO\n",
      "Dryobotodes_eremita\tNO\n",
      "Ecliptopera_silaceata\tNO\n",
      "Ectropis_crepuscularia\tNO\n",
      "Ectropis_grisescens\tNO\n",
      "Eilema_caniola\tNO\n",
      "Eilema_depressum\tNO\n",
      "Eilema_griseolum\tNO\n",
      "Eilema_sororculum\tNO\n",
      "Electrophaes_corylata\tNO\n",
      "Elegia_similella\tNO\n",
      "Elophila_nymphaeata\tNO\n",
      "Elymnias_hypermnestra\tNO\n",
      "Emmelina_monodactyla\tNO\n",
      "Endotricha_flammealis\tNO\n",
      "Ennomos_erosarius\tNO\n",
      "Ennomos_fuscantarius\tNO\n",
      "Ennomos_quercinarius\tNO\n",
      "Ephestia_elutella\tNO\n",
      "Epiblema_foenella\tNO\n",
      "Epinotia_bilunana\tNO\n",
      "Epinotia_brunnichana\tNO\n",
      "Epinotia_demarniana\tNO\n",
      "Epinotia_nisella\tNO\n",
      "Epinotia_ramella\tNO\n",
      "Epirrhoe_alternata\tNO\n",
      "Epirrhoe_tristata\tNO\n",
      "Epirrita_christyi\tNO\n",
      "Erannis_defoliaria\tNO\n",
      "Erebia_aethiops\tNO\n",
      "Erebia_ligea\tNO\n",
      "Eremobia_ochroleuca\tNO\n",
      "Erynnis_tages\tNO\n",
      "Esperia_sulphurella\tNO\n",
      "Ethmia_dodecea\tNO\n",
      "Euclidia_mi\tNO\n",
      "Eucosma_cana\tNO\n",
      "Eudemis_profundana\tNO\n",
      "Eudonia_lacustrata\tNO\n",
      "Eudonia_mercurella\tNO\n",
      "Eudonia_truncicolella\tNO\n",
      "Eueides_isabella\tNO\n",
      "Eugnorisma_glareosa\tNO\n",
      "Eulithis_prunata\tNO\n",
      "Eulithis_testata\tNO\n",
      "Eumaeus_atala\tNO\n",
      "Eumeta_japonica\tYes\n",
      "Euphydryas_editha\tYes\n",
      "Eupithecia_abbreviata\tNO\n",
      "Eupithecia_centaureata\tNO\n",
      "Eupithecia_dodoneata\tNO\n",
      "Eupithecia_exiguata\tNO\n",
      "Eupithecia_insigniata\tNO\n",
      "Eupithecia_inturbata\tNO\n",
      "Eupithecia_subfuscata\tNO\n",
      "Eupithecia_subumbrata\tNO\n",
      "Eupithecia_tripunctaria\tNO\n",
      "Eupithecia_vulgata\tNO\n",
      "Euplagia_quadripunctaria\tNO\n",
      "Euplexia_lucipara\tNO\n",
      "Euproctis_similis\tNO\n",
      "Eupsilia_transversa\tNO\n",
      "Eurema_hecabe\tNO\n",
      "Eurois_occulta\tNO\n",
      "Euspilapteryx_auroguttella\tNO\n",
      "Euzophera_pinguis\tNO\n",
      "Fabriciana_adippe\tNO\n",
      "Falcaria_lacertinaria\tNO\n",
      "Fissipunctia_ypsillon\tNO\n",
      "Furcula_furcula\tNO\n",
      "Galleria_mellonella\tNO\n",
      "Gandaritis_pyraliata\tNO\n",
      "Glaucopsyche_alexis\tNO\n",
      "Globia_sparganii\tNO\n",
      "Gortyna_flavago\tNO\n",
      "Grapholita_molesta\tNO\n",
      "Griposia_aprilina\tNO\n",
      "Gymnoscelis_rufifasciata\tNO\n",
      "Habrosyne_pyritoides\tNO\n",
      "Hebomoia_glaucippe\tNO\n",
      "Hecatera_dysodea\tNO\n",
      "Hedya_salicella\tNO\n",
      "Heliconius_charithonia\tNO\n",
      "Heliconius_nattereri\tNO\n",
      "Heliconius_sara\tNO\n",
      "Helicoverpa_armigera\tYes\n",
      "Helicoverpa_assulta\tNO\n",
      "Helicoverpa_zea\tYes\n",
      "Heliothis_peltigera\tNO\n",
      "Heliothis_virescens\tNO\n",
      "Helleia_helle\tNO\n",
      "Hemaris_fuciformis\tNO\n",
      "Hemistola_chrysoprasaria\tNO\n",
      "Hemithea_aestivaria\tNO\n",
      "Heortia_vitessoides\tNO\n",
      "Herminia_tarsipennalis\tNO\n",
      "Hesperia_comma\tNO\n",
      "Hestina_assimilis\tNO\n",
      "Hipparchia_semele\tNO\n",
      "Hofmannophila_pseudospretella\tNO\n",
      "Hoplodrina_ambigua\tNO\n",
      "Hoplodrina_blanda\tNO\n",
      "Horisme_vitalbata\tNO\n",
      "Hydraecia_micacea\tNO\n",
      "Hydriomena_furcata\tNO\n",
      "Hylaea_fasciaria\tNO\n",
      "Hyles_euphorbiae\tNO\n",
      "Hyles_vespertilio\tNO\n",
      "Hypena_proboscidalis\tNO\n",
      "Hypercompe_scribonia\tNO\n",
      "Hyphantria_cunea\tNO\n",
      "Hypolimnas_misippus\tNO\n",
      "Hypomecis_punctinalis\tNO\n",
      "Hyposmocoma_kahamanoa\tYes\n",
      "Hyppa_rectilinea\tNO\n",
      "Hypsopygia_costalis\tNO\n",
      "Idaea_aversata\tNO\n",
      "Idaea_biselata\tNO\n",
      "Idaea_dimidiata\tNO\n",
      "Idaea_straminata\tNO\n",
      "Incurvaria_masculella\tNO\n",
      "Iphiclides_podalirius\tYes\n",
      "Lacanobia_oleracea\tNO\n",
      "Lacanobia_wlatinum\tNO\n",
      "Lampropteryx_suffumata\tNO\n",
      "Laothoe_populi\tNO\n",
      "Lapara_coniferarum\tNO\n",
      "Lasiommata_megera\tNO\n",
      "Laspeyria_flexula\tNO\n",
      "Lathronympha_strigana\tNO\n",
      "Leguminivora_glycinivorella\tYes\n",
      "Leptidea_juvernica\tNO\n",
      "Leptidea_reali\tNO\n",
      "Leptidea_sinapis\tYes\n",
      "Leptophobia_aripa\tNO\n",
      "Lerema_accius\tNO\n",
      "Leucania_comma\tNO\n",
      "Leucinodes_orbonalis\tNO\n",
      "Leucoma_salicis\tNO\n",
      "Ligdia_adustata\tNO\n",
      "Limenitis_camilla\tNO\n",
      "Lithophane_leautieri\tNO\n",
      "Lithophane_ornitopus\tNO\n",
      "Lithophane_semibrunnea\tNO\n",
      "Lithophane_socia\tNO\n",
      "Lithosia_quadra\tNO\n",
      "Lobophora_halterata\tNO\n",
      "Lomographa_bimaculata\tNO\n",
      "Luehdorfia_chinensis\tNO\n",
      "Luffia_ferchaultella\tNO\n",
      "Luperina_nickerlii\tNO\n",
      "Luperina_testacea\tNO\n",
      "Lycaena_phlaeas\tNO\n",
      "Lycia_hirtaria\tNO\n",
      "Lycophotia_porphyrea\tNO\n",
      "Lymantria_dispar\tNO\n",
      "Lymantria_monacha\tNO\n",
      "Lysandra_bellargus\tNO\n",
      "Lysandra_coridon\tNO\n",
      "Macaria_notata\tNO\n",
      "Malacosoma_neustria\tNO\n",
      "Mamestra_brassicae\tNO\n",
      "Mamestra_configurata\tYes\n",
      "Manduca_sexta\tYes\n",
      "Maniola_hyperantus\tYes\n",
      "Maniola_jurtina\tYes\n",
      "Marasmarcha_lunaedactyla\tNO\n",
      "Marasmia_exigua\tNO\n",
      "Mechanitis_mazaeus\tNO\n",
      "Mechanitis_messenoides\tNO\n",
      "Mecyna_flavalis\tNO\n",
      "Meganola_albula\tNO\n",
      "Megathymus_violae\tNO\n",
      "Melanargia_galathea\tNO\n",
      "Melanchra_persicariae\tNO\n",
      "Melinaea_marsaeus_rileyi\tNO\n",
      "Melinaea_menophilus_n._ssp._AW-2005\tNO\n",
      "Melitaea_cinxia\tYes\n",
      "Mellicta_athalia\tNO\n",
      "Mesoligia_furuncula\tNO\n",
      "Metalampra_italica\tNO\n",
      "Micropterix_aruncella\tNO\n",
      "Miltochrista_miniata\tNO\n",
      "Mimas_tiliae\tNO\n",
      "Monopis_laevigella\tNO\n",
      "Musotima_nitidalis\tNO\n",
      "Mythimna_albipuncta\tNO\n",
      "Mythimna_ferrago\tNO\n",
      "Mythimna_impura\tNO\n",
      "Mythimna_lalbum\tNO\n",
      "Mythimna_loreyi\tYes\n",
      "Mythimna_pallens\tNO\n",
      "Mythimna_separata\tYes\n",
      "Mythimna_vitellina\tNO\n",
      "Napeogenes_inachia\tNO\n",
      "Napeogenes_sylphis\tNO\n",
      "Nematopogon_swammerdamellus\tNO\n",
      "Noctua_comes\tNO\n",
      "Noctua_fimbriata\tNO\n",
      "Noctua_janthe\tNO\n",
      "Noctua_janthina\tNO\n",
      "Noctua_pronuba\tNO\n",
      "Nomophila_noctuella\tNO\n",
      "Notocelia_uddmanniana\tNO\n",
      "Notodonta_dromedarius\tNO\n",
      "Notodonta_ziczac\tNO\n",
      "Nudaria_mundana\tNO\n",
      "Nycteola_revayana\tNO\n",
      "Nymphalis_c-album\tNO\n",
      "Nymphalis_io\tYes\n",
      "Nymphalis_polychloros\tNO\n",
      "Nymphalis_urticae\tNO\n",
      "Nymphula_nitidulata\tNO\n",
      "Ochlodes_sylvanus\tNO\n",
      "Ochropacha_duplaris\tNO\n",
      "Ochropleura_leucogaster\tNO\n",
      "Ochropleura_plecta\tNO\n",
      "Oegoconia_quadripuncta\tNO\n",
      "Oeneis_ivallda\tNO\n",
      "Oligia_fasciuncula\tNO\n",
      "Oligia_latruncula\tNO\n",
      "Oligia_strigilis\tNO\n",
      "Omphaloscelis_lunosa\tNO\n",
      "Operophtera_brumata\tNO\n",
      "Opisthograptis_luteolata\tNO\n",
      "Orgyia_antiqua\tNO\n",
      "Ornithoptera_alexandrae\tNO\n",
      "Ornithoptera_priamus\tNO\n",
      "Orthosia_gothica\tNO\n",
      "Orthosia_gracilis\tNO\n",
      "Orthosia_incerta\tNO\n",
      "Ostrinia_furnacalis\tYes\n",
      "Ostrinia_nubilalis\tNO\n",
      "Pammene_aurita\tNO\n",
      "Pammene_fasciana\tNO\n",
      "Pandemis_cinnamomeana\tNO\n",
      "Pandemis_corylana\tNO\n",
      "Pandemis_heparana\tNO\n",
      "Papilio_dardanus_tibullus\tNO\n",
      "Papilio_demoleus\tNO\n",
      "Papilio_elwesi\tNO\n",
      "Papilio_gigon\tNO\n",
      "Papilio_glaucus\tNO\n",
      "Papilio_machaon\tYes\n",
      "Papilio_memnon\tNO\n",
      "Papilio_polytes\tYes\n",
      "Papilio_protenor\tNO\n",
      "Papilio_xuthus\tYes\n",
      "Parapoynx_stratiotata\tNO\n",
      "Pararge_aegeria\tYes\n",
      "Parascotia_fuliginaria\tNO\n",
      "Parnassius_glacialis\tNO\n",
      "Pasiphila_rectangulata\tNO\n",
      "Pectinophora_gossypiella\tYes\n",
      "Pelosia_muscerda\tNO\n",
      "Peribatodes_rhomboidaria\tNO\n",
      "Perizoma_affinitatum\tNO\n",
      "Perizoma_flavofasciatum\tNO\n",
      "Petrophora_chlorosata\tNO\n",
      "Phalera_bucephala\tNO\n",
      "Phengaris_arion\tNO\n",
      "Pheosia_gnoma\tNO\n",
      "Pheosia_tremula\tNO\n",
      "Philaethria_dido\tNO\n",
      "Philereme_vetulata\tNO\n",
      "Phlogophora_meticulosa\tNO\n",
      "Phoebis_sennae\tNO\n",
      "Phragmatobia_fuliginosa\tNO\n",
      "Pieris_brassicae\tYes\n",
      "Pieris_mannii\tNO\n",
      "Pieris_melete\tNO\n",
      "Pieris_napi\tYes\n",
      "Pieris_rapae\tYes\n",
      "Plagodis_dolabraria\tNO\n",
      "Plagodis_pulveraria\tNO\n",
      "Plebejus_argus\tNO\n",
      "Plemyria_rubiginata\tNO\n",
      "Plodia_interpunctella\tYes\n",
      "Plusia_festucae\tNO\n",
      "Plutella_xylostella\tYes\n",
      "Polia_nebulosa\tNO\n",
      "Polymixis_flavicincta\tNO\n",
      "Polymixis_lichenea\tNO\n",
      "Polyommatus_icarus\tNO\n",
      "Polyommatus_iphigenia\tNO\n",
      "Polyploca_ridens\tNO\n",
      "Protodeltote_pygarga\tNO\n",
      "Pseudoips_prasinana\tNO\n",
      "Ptilodon_capucinus\tNO\n",
      "Pyralis_farinalis\tNO\n",
      "Pyrausta_aurata\tNO\n",
      "Pyrausta_nigrata\tNO\n",
      "Pyrgus_malvae\tNO\n",
      "Pyrrhia_umbra\tNO\n",
      "Rhodophaea_formosa\tNO\n",
      "Samia_ricini\tNO\n",
      "Saturnia_japonica\tNO\n",
      "Saturnia_pavonia\tNO\n",
      "Schrankia_costaestrigalis\tNO\n",
      "Scotopteryx_bipunctaria\tNO\n",
      "Scrobipalpa_costella\tNO\n",
      "Selenia_dentaria\tNO\n",
      "Sericinus_montela\tNO\n",
      "Sesamia_nonagrioides\tNO\n",
      "Sesia_apiformis\tNO\n",
      "Sesia_bembeciformis\tNO\n",
      "Shargacucullia_verbasci\tNO\n",
      "Sphinx_pinastri\tNO\n",
      "Spilarctia_lutea\tNO\n",
      "Spilosoma_lubricipeda\tNO\n",
      "Spodoptera_exigua\tYes\n",
      "Spodoptera_frugiperda\tYes\n",
      "Spodoptera_littoralis\tYes\n",
      "Spodoptera_litura\tYes\n",
      "Stenoptilia_bipunctidactyla\tNO\n",
      "Subacronicta_megacephala\tNO\n",
      "Synanthedon_andrenaeformis\tNO\n",
      "Synanthedon_formicaeformis\tNO\n",
      "Synanthedon_myopaeformis\tNO\n",
      "Synanthedon_tipuliformis\tNO\n",
      "Synanthedon_vespiformis\tNO\n",
      "Tachystola_acroxantha\tNO\n",
      "Teinopalpus_imperialis\tNO\n",
      "Teleiodes_luculella\tNO\n",
      "Tethea_ocularis\tNO\n",
      "Tetheella_fluctuosa\tNO\n",
      "Thalpophila_matura\tNO\n",
      "Thera_britannica\tNO\n",
      "Thera_obeliscata\tNO\n",
      "Theretra_japonica\tNO\n",
      "Tholera_decimalis\tNO\n",
      "Thumatha_senex\tNO\n",
      "Thyatira_batis\tNO\n",
      "Thymelicus_acteon\tNO\n",
      "Thymelicus_sylvestris\tNO\n",
      "Tiliacea_aurago\tNO\n",
      "Timandra_comae\tNO\n",
      "Tinea_pellionella\tNO\n",
      "Tinea_semifulvella\tNO\n",
      "Tinea_trinotella\tNO\n",
      "Tortricodes_alternella\tNO\n",
      "Tortrix_viridana\tNO\n",
      "Trichoplusia_ni\tYes\n",
      "Trilocha_varians\tNO\n",
      "Trisateles_emortualis\tNO\n",
      "Troides_aeacus\tNO\n",
      "Troides_oblongomaculatus\tNO\n",
      "Tuta_absoluta\tNO\n",
      "Tyria_jacobaeae\tNO\n",
      "Udea_ferrugalis\tNO\n",
      "Udea_olivalis\tNO\n",
      "Urbanus_simplicius\tNO\n",
      "Vanessa_atalanta\tYes\n",
      "Vanessa_cardui\tYes\n",
      "Vanessa_tameamea\tYes\n",
      "Venusia_cambrica\tNO\n",
      "Watsonalla_binaria\tNO\n",
      "Xanthia_icteritia\tNO\n",
      "Xanthia_togata\tNO\n",
      "Xanthorhoe_designata\tNO\n",
      "Xanthorhoe_spadicearia\tNO\n",
      "Xestia_ashworthii\tNO\n",
      "Xestia_c-nigrum\tNO\n",
      "Xestia_rhomboidea\tNO\n",
      "Xestia_sexstrigata\tNO\n",
      "Xestia_xanthographa\tNO\n",
      "Xylocampa_areola\tNO\n",
      "Yponomeuta_cagnagella\tNO\n",
      "Yponomeuta_malinellus\tNO\n",
      "Yponomeuta_plumbellus\tNO\n",
      "Yponomeuta_sedellus\tNO\n",
      "Ypsolopha_scabrella\tNO\n",
      "Ypsolopha_sequella\tNO\n",
      "Zelleria_hepariella\tNO\n",
      "Zerene_cesonia\tYes\n",
      "Zeuzera_pyrina\tNO\n",
      "Zygaena_filipendulae\tNO\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "list_of_genomes = os.listdir(\"J:/Genomes_2023-12-26\")\n",
    "output = \"Species\\tAnnotation\\n\"\n",
    "for species in list_of_genomes:\n",
    "    list_of_files= os.listdir(f\"J:/Genomes_2023-12-26/{species}\")\n",
    "    annotation = \"NO\"\n",
    "    for file in list_of_files:\n",
    "        if \"genomic.gff\" in file:\n",
    "            annotation = \"Yes\"\n",
    "#             print(species)\n",
    "    \n",
    "    output += f\"{species}\\t{annotation}\\n\"\n",
    "\n",
    "print(output)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7a56c32",
   "metadata": {},
   "source": [
    "# Parse fasta query file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "e9e74324",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon 1, Length = 89\n",
      "Name:NC_051358.1:c12522661-12522573\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = MDNLENSENTDKISDSAYSNSCSYSQSRR\n",
      "translated_length = 29\n",
      "\n",
      "Exon 2, Length = 84\n",
      "Name:NC_051358.1:c12521246-12521163\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = HSSKSTHSGSYSSGSSGYGGRHVTGTN\n",
      "translated_length = 27\n",
      "\n",
      "Exon 3, Length = 155\n",
      "Name:NC_051358.1:c12520999-12520845\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = NKLAQSPNKDKELKKKKQSQTELTTTKEQNEGESGPQESVNLETLEEEKHD\n",
      "translated_length = 51\n",
      "\n",
      "Exon 4, Length = 95\n",
      "Name:NC_051358.1:c12514423-12514329\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = TSAPSPAHVQVEKGPENMDIASTDTEGDKVE\n",
      "translated_length = 31\n",
      "\n",
      "Exon 5, Length = 78\n",
      "Name:NC_051358.1:c12513351-12513274\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = VLSCDLQPQNSTVDVVTSRSNLICYT\n",
      "translated_length = 26\n",
      "\n",
      "Exon 6, Length = 132\n",
      "Name:NC_051358.1:c12512245-12512114\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = DGFSCVISMHDGIVMYTTSSLTAILGFPKDMWIGRSFIDFVHPR\n",
      "translated_length = 44\n",
      "\n",
      "Exon 7, Length = 76\n",
      "Name:NC_051358.1:c12511968-12511893\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = DRSTFASQITSGLAVPKTANGTQEK\n",
      "translated_length = 25\n",
      "\n",
      "Exon 8, Length = 204\n",
      "Name:NC_051358.1:c12510243-12510040\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = QSPGNSGSTMVCRIRRYRGLSTGFGVKERVVTFMPFLLKFTFKNISDEEGNVIYLVIQATPFFSAFK\n",
      "translated_length = 67\n",
      "\n",
      "Exon 9, Length = 229\n",
      "Name:NC_051358.1:c12508701-12508473\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = SFEILPKVNPFVMRHSANGNLEYLDPESVPYLGYLPQDVQDKDALQLYHPEDLEYLQQVYEVIVKDGGMPRSKTY\n",
      "translated_length = 75\n",
      "\n",
      "Exon 10, Length = 112\n",
      "Name:NC_051358.1:c12507855-12507744\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = MMTQNGDYIKIETEWSSFINPWSKKLEFVIGKHYIIE\n",
      "translated_length = 37\n",
      "\n",
      "Exon 11, Length = 120\n",
      "Name:NC_051358.1:c12506233-12506114\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = GPENPDVFQSQDPEKHTKLCDEQIKKSMVFRENIVKLMNE\n",
      "translated_length = 40\n",
      "\n",
      "Exon 12, Length = 150\n",
      "Name:NC_051358.1:c12505865-12505716\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = ALTKPAEVAKQQMSKRCQDLASFMESLMEEPPKNDEELRLEIQDPDHSYY\n",
      "translated_length = 50\n",
      "\n",
      "Exon 13, Length = 119\n",
      "Name:NC_051358.1:c12503906-12503788\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = ERDSVMLGGISPHHDYNDSKSSTETPLSYNQLNYNETLQ\n",
      "translated_length = 39\n",
      "\n",
      "Exon 14, Length = 90\n",
      "Name:NC_051358.1:c12503481-12503392\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = YFDSHEPFSSENFNPLIVAKEPNNLKSIL\n",
      "translated_length = 29\n",
      "\n",
      "Exon 15, Length = 153\n",
      "Name:NC_051358.1:c12502301-12502149\n",
      "\n",
      "\n",
      "1\n",
      "\n",
      "translated = TCLSPMAQNSGDSTELNSSSDSSSMALGGVSPVGILGDYHHVRLTEFLLT\n",
      "translated_length = 50\n",
      "\n",
      "Exon 16, Length = 190\n",
      "Name:NC_051358.1:c12501492-12501303\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = HNDEMEKELINMHRESRSNSKGERDKTSNETRQKKKEHLARCKASFHPTATSTTPVDKEVYKK\n",
      "translated_length = 63\n",
      "\n",
      "Exon 17, Length = 148\n",
      "Name:NC_051358.1:c12498481-12498334\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = PHGVKRASKHIETETVSHKYHCPSPRASRPRQTTSAAPVQTSSTLTTSV\n",
      "translated_length = 49\n",
      "\n",
      "Exon 18, Length = 114\n",
      "Name:NC_051358.1:c12497509-12497396\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = STTWPPSTNAAGNMNTFILGVGMAPQMSLISPVPPMA\n",
      "translated_length = 37\n",
      "\n",
      "Exon 19, Length = 117\n",
      "Name:NC_051358.1:c12496602-12496486\n",
      "\n",
      "\n",
      "2\n",
      "\n",
      "translated = MLPLYYTPMATMAPVPSTSEAANHQNLHNNPQQYAPPP\n",
      "translated_length = 38\n",
      "\n",
      "Exon 20, Length = 185\n",
      "Name:NC_051358.1:c12494243-12494059\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = QCVLYGQPIYGQPMYSSPFVYSPMNPHTNYPMQQTTPQPNAQFTPTNTMNPLCLANSNYEE\n",
      "translated_length = 61\n",
      "\n",
      "Exon 21, Length = 201\n",
      "Name:NC_051358.1:c12483156-12482956\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = RTEAGHATTPSKDIKKTSRLGNTEEIVDKTDGESSYSSFYSSFFKTESGSAEESGESKKRDTRNHVK\n",
      "translated_length = 67\n",
      "\n",
      "Exon 22, Length = 135\n",
      "Name:NC_051358.1:c12482268-12482134\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = FWDKNYRGDDGMAMPGSTEVRQTFSDHHIPAKVPRRNIEPPWMEQ\n",
      "translated_length = 45\n",
      "\n",
      "Exon 23, Length = 102\n",
      "Name:NC_051358.1:c12481645-12481544\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = VCLTSELVYNYQIRTKTMEEVLSSDKQKISNLEQ\n",
      "translated_length = 34\n",
      "\n",
      "Exon 24, Length = 249\n",
      "Name:NC_051358.1:c12478465-12478217\n",
      "\n",
      "\n",
      "0\n",
      "\n",
      "translated = PSLVNEQLGQLYLDLQLEGVAARLTLEEGITSSSSSGEETTTSSKRKREYSKLVMIYEENAPFPPPSSEEIQEIPTANQSES*\n",
      "translated_length = 83\n",
      "\n",
      ">Exon_1_Bombyx_mori_NM_001043510.1\n",
      "MDNLENSENTDKISDSAYSNSCSYSQSRR\n",
      "\n",
      ">Exon_2_Bombyx_mori_NM_001043510.1\n",
      "HSSKSTHSGSYSSGSSGYGGRHVTGTN\n",
      "\n",
      ">Exon_3_Bombyx_mori_NM_001043510.1\n",
      "NKLAQSPNKDKELKKKKQSQTELTTTKEQNEGESGPQESVNLETLEEEKHD\n",
      "\n",
      ">Exon_4_Bombyx_mori_NM_001043510.1\n",
      "TSAPSPAHVQVEKGPENMDIASTDTEGDKVE\n",
      "\n",
      ">Exon_5_Bombyx_mori_NM_001043510.1\n",
      "VLSCDLQPQNSTVDVVTSRSNLICYT\n",
      "\n",
      ">Exon_6_Bombyx_mori_NM_001043510.1\n",
      "DGFSCVISMHDGIVMYTTSSLTAILGFPKDMWIGRSFIDFVHPR\n",
      "\n",
      ">Exon_7_Bombyx_mori_NM_001043510.1\n",
      "DRSTFASQITSGLAVPKTANGTQEK\n",
      "\n",
      ">Exon_8_Bombyx_mori_NM_001043510.1\n",
      "QSPGNSGSTMVCRIRRYRGLSTGFGVKERVVTFMPFLLKFTFKNISDEEGNVIYLVIQATPFFSAFK\n",
      "\n",
      ">Exon_9_Bombyx_mori_NM_001043510.1\n",
      "SFEILPKVNPFVMRHSANGNLEYLDPESVPYLGYLPQDVQDKDALQLYHPEDLEYLQQVYEVIVKDGGMPRSKTY\n",
      "\n",
      ">Exon_10_Bombyx_mori_NM_001043510.1\n",
      "MMTQNGDYIKIETEWSSFINPWSKKLEFVIGKHYIIE\n",
      "\n",
      ">Exon_11_Bombyx_mori_NM_001043510.1\n",
      "GPENPDVFQSQDPEKHTKLCDEQIKKSMVFRENIVKLMNE\n",
      "\n",
      ">Exon_12_Bombyx_mori_NM_001043510.1\n",
      "ALTKPAEVAKQQMSKRCQDLASFMESLMEEPPKNDEELRLEIQDPDHSYY\n",
      "\n",
      ">Exon_13_Bombyx_mori_NM_001043510.1\n",
      "ERDSVMLGGISPHHDYNDSKSSTETPLSYNQLNYNETLQ\n",
      "\n",
      ">Exon_14_Bombyx_mori_NM_001043510.1\n",
      "YFDSHEPFSSENFNPLIVAKEPNNLKSIL\n",
      "\n",
      ">Exon_15_Bombyx_mori_NM_001043510.1\n",
      "TCLSPMAQNSGDSTELNSSSDSSSMALGGVSPVGILGDYHHVRLTEFLLT\n",
      "\n",
      ">Exon_16_Bombyx_mori_NM_001043510.1\n",
      "HNDEMEKELINMHRESRSNSKGERDKTSNETRQKKKEHLARCKASFHPTATSTTPVDKEVYKK\n",
      "\n",
      ">Exon_17_Bombyx_mori_NM_001043510.1\n",
      "PHGVKRASKHIETETVSHKYHCPSPRASRPRQTTSAAPVQTSSTLTTSV\n",
      "\n",
      ">Exon_18_Bombyx_mori_NM_001043510.1\n",
      "STTWPPSTNAAGNMNTFILGVGMAPQMSLISPVPPMA\n",
      "\n",
      ">Exon_19_Bombyx_mori_NM_001043510.1\n",
      "MLPLYYTPMATMAPVPSTSEAANHQNLHNNPQQYAPPP\n",
      "\n",
      ">Exon_20_Bombyx_mori_NM_001043510.1\n",
      "QCVLYGQPIYGQPMYSSPFVYSPMNPHTNYPMQQTTPQPNAQFTPTNTMNPLCLANSNYEE\n",
      "\n",
      ">Exon_21_Bombyx_mori_NM_001043510.1\n",
      "RTEAGHATTPSKDIKKTSRLGNTEEIVDKTDGESSYSSFYSSFFKTESGSAEESGESKKRDTRNHVK\n",
      "\n",
      ">Exon_22_Bombyx_mori_NM_001043510.1\n",
      "FWDKNYRGDDGMAMPGSTEVRQTFSDHHIPAKVPRRNIEPPWMEQ\n",
      "\n",
      ">Exon_23_Bombyx_mori_NM_001043510.1\n",
      "VCLTSELVYNYQIRTKTMEEVLSSDKQKISNLEQ\n",
      "\n",
      ">Exon_24_Bombyx_mori_NM_001043510.1\n",
      "PSLVNEQLGQLYLDLQLEGVAARLTLEEGITSSSSSGEETTTSSKRKREYSKLVMIYEENAPFPPPSSEEIQEIPTANQSES*\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\Bio\\Seq.py:2334: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "expected str, bytes or os.PathLike object, not tuple",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[1;32mIn [23]\u001b[0m, in \u001b[0;36m<cell line: 72>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     68\u001b[0m     exon_counter \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;28mprint\u001b[39m(output_protein_query)\n\u001b[1;32m---> 72\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m \u001b[49m\u001b[43m(\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mH:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_protein.txt\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mw\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m out_file:\n\u001b[0;32m     73\u001b[0m     out_file\u001b[38;5;241m.\u001b[39mwrite(output_protein_query)\n",
      "\u001b[1;31mTypeError\u001b[0m: expected str, bytes or os.PathLike object, not tuple"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "def print_codon_translated(exon_sequence,exon_counter,exon_overhang):\n",
    "    counter = 0\n",
    "    codon_counter = 0\n",
    "    protein_print_overhang = 0\n",
    "    if exon_overhang != 0:\n",
    "        print(f\"{exon_sequence[counter : counter + exon_overhang]} \", end = '')\n",
    "        protein_print_overhang = 1\n",
    "        \n",
    "    counter = counter + exon_overhang\n",
    "    start_position = counter\n",
    "    while counter < len(exon_sequence):\n",
    "        print(f\"{exon_sequence[counter : counter + 3]} \", end = '')\n",
    "        counter += 3\n",
    "        codon_counter += 1\n",
    "        if codon_counter%5 == 0:\n",
    "            print(\"\\n\")\n",
    "            if protein_print_overhang == 1:\n",
    "                print (f\"{' '*exon_overhang} \", end = '')\n",
    "                protein_print_overhang = 0\n",
    "            \n",
    "            while start_position < counter:\n",
    "                print(f\" {Seq(exon_sequence[start_position : start_position + 3]).translate()}  \", end = '')\n",
    "                start_position += 3\n",
    "            print(\"\\n\")\n",
    "    \n",
    "    print(\"\\n\")\n",
    "    if protein_print_overhang == 1:\n",
    "        print (f\"{' '*exon_overhang} \", end = '')\n",
    "        protein_print_overhang = 0\n",
    "\n",
    "    while start_position < counter:\n",
    "        print(f\" {Seq(exon_sequence[start_position : start_position + 3]).translate()}  \", end = '')\n",
    "        start_position += 3\n",
    "    print(\"\\n\")\n",
    "    return\n",
    "\n",
    "\n",
    "query_rna_file = SeqIO.parse(\"H:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_rna.txt\", 'fasta')\n",
    "exon_counter = 1\n",
    "full_sequence = ''\n",
    "exon_overhang = 0\n",
    "exon_3_overhang = 0\n",
    "exon_5_overhang = 0\n",
    "\n",
    "output_protein_query = ''\n",
    "\n",
    "for sequences in query_rna_file:\n",
    "    \n",
    "    exon_sequence = str(sequences.seq)\n",
    "    print(f\"Exon {exon_counter}, Length = {len(sequences.seq)}\\nName:{sequences.id}\\n\\n\")\n",
    "    \n",
    "    print_codon_translated(exon_sequence,exon_counter,exon_overhang)\n",
    "    \n",
    "    exon_without_overhang_length = len(exon_sequence) - exon_overhang\n",
    "    exon_5_overhang = exon_overhang\n",
    "    exon_overhang = 3 - exon_without_overhang_length % 3\n",
    "    if exon_overhang == 3:\n",
    "        exon_overhang = 0\n",
    "   \n",
    "    print(exon_overhang)\n",
    "    translated = (sequences.seq[exon_5_overhang:]).translate()\n",
    "    print (f\"\\ntranslated = {translated}\\ntranslated_length = {len(translated)}\\n\")\n",
    "    \n",
    "    output_protein_query += f\">Exon_{exon_counter}_Bombyx_mori_NM_001043510.1\\n{translated}\\n\\n\"\n",
    "    exon_counter += 1\n",
    "    \n",
    "# print(output_protein_query)\n",
    "\n",
    "with open ((\"H:/My Drive/Circadian Rhythm Genes Project/4.Queries for Blast/1.Period/0.Species/1.Bombyx mori/Query_protein.txt\", 'w')) as out_file:\n",
    "    out_file.write(output_protein_query)\n",
    "    \n",
    "    \n",
    "    \n",
    "#     break\n",
    "#     counter = 0\n",
    "# #     while counter < len(exon_sequence):\n",
    "        \n",
    "    \n",
    "    \n",
    "#     if len(sequences.seq)%3 != 0:\n",
    "#         print(sequences.seq, len(sequences.seq))\n",
    "#         break\n",
    "# #         my_seq = Seq(full_sequence)\n",
    "\n",
    "# #     print(my_seq.translate())\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6af45e47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2dcfa6b3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f15d7870",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea7409c1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8e769e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41a91958",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
