{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "36d22ad3-a85e-44b3-85e6-fc9cb5b48684",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_blast_with_new_query(annotated_genome_location,\n",
    "                             annotated_species_name,\n",
    "                             error_exon, \n",
    "                            query_species,\n",
    "                             genome_location, \n",
    "                             species):\n",
    "    \n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\"\n",
    "    genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "    out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "\n",
    "    with open(f\"{out_location}/new_query.txt\", 'r') as query_file_open:\n",
    "        query_file_lines = query_file_open.readlines()\n",
    "    if (len(query_file_lines)) == 1:\n",
    "        return (\"Query_error\")\n",
    "    print(len(query_file_lines[1])) \n",
    "    if (len(query_file_lines[1])) < 5:\n",
    "        return (\"Query_error\")\n",
    "    # print(len(query_file_lines[1]))\n",
    "    # genome_fragment_out = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/local_genomic_fragment.fa\"\n",
    "    # with io.open(genome_fragment_out,'w') as out_file:\n",
    "    #     output = f\">Genome_fragment_{error_exon}\\n{gene_sequence}\"\n",
    "    #     out_file.write(output)\n",
    "\n",
    "    local_genomic_fragment_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\"\n",
    "    # cd_command = f'cd \"{local_genomic_fragment_location}\"\\nmakeblastdb -in gene_sequence_all.fa -dbtype nucl\\n'\n",
    "    # # os.system(f'{cd_command}')\n",
    "    # subprocess.run(f'{cd_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # # print(mkdb_command)\n",
    "    # # os.system(f'{mkdb_command}')\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.htm -html'\n",
    "    # print(blast_command)\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # subprocess.run(f'{blast_command}', shell = True)\n",
    "    blast_command = f'cd \"{out_location}\"\\ntblastn -seg no -query new_query.txt -db ../../../Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa -num_alignments 3 -out blast_out_genome_fragment.txt'\n",
    "    # os.system(f'{blast_command}')\n",
    "    subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6e7d0eb2-6d4d-44b8-90ab-76c864362837",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bd69b123-ff19-4205-83ee-039f63b0fc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_query_info(query_location,query_species,query_transcript,query_exon):\n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "    return(left_overhang,right_overhang,original_query_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "37864dc3-b190-4ffc-bd3c-ec61c8d7d4d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                       annotated_species_name,\n",
    "                                       error_exon,\n",
    "                                       left_overhang,\n",
    "                                       right_overhang,                                                                                               \n",
    "                                       original_query_name,\n",
    "                                      query_species_original ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "    with open(f\"{blast_location}/new_query.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out_genome_fragment.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "        genome_file = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/gene_sequence_all.fa\", 'fasta')\n",
    "        print(\"reached here\")\n",
    "        print(f\"Before splice: {start},{stop}\")\n",
    "        acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name,query_species_original)\n",
    "        print(f\"after splice: {start},{stop}\")\n",
    "        for records in genome_file:\n",
    "            old_start = start\n",
    "            old_stop = stop\n",
    "            ag = \"N\"\n",
    "            gt = \"N\"\n",
    "            stop_counter = 0\n",
    "            while True:\n",
    "                print(f\"sequence:\\n{records.seq[start+start_modifier-1:stop]}\")\n",
    "                translated_sequence = records.seq[start+start_modifier-1:stop].translate()\n",
    "                print(f\"sequence:\\n{translated_sequence}\")\n",
    "                if \"*\" in translated_sequence:\n",
    "                    stop_counter +=1\n",
    "                    if ag == \"N\":\n",
    "                        start = old_start + 3*stop_counter\n",
    "                    if gt ==\"N\":\n",
    "                        stop = old_stop - 3*stop_counter\n",
    "                print(f\"left = {records.seq[start-3:start-1]}, right ={(records.seq[stop:stop + 2])}, stop_counter = {stop_counter}\"  ), \n",
    "                if (records.seq[start-3:start-1]).lower() == \"ag\" and ag != \"Y\":\n",
    "                    \n",
    "                    ag = \"Y\"\n",
    "                    \n",
    "                elif ag != \"Y\":\n",
    "                    start -= 3\n",
    "                    \n",
    "                if (records.seq[stop:stop + 2]).lower() == \"gt\" and gt != \"Y\":\n",
    "                    gt = \"Y\"\n",
    "                elif gt != \"Y\":\n",
    "                    stop +=3\n",
    "                if old_start - start > 1000 or stop - old_stop > 1000:\n",
    "                    break\n",
    "                if gt == \"Y\" and ag == \"Y\":\n",
    "                    gt_ag = \"Y\"\n",
    "                    break\n",
    "                if stop_counter > 10:\n",
    "                    gt = \"Y\"\n",
    "                    ag = \"Y\"\n",
    "                \n",
    "        \n",
    "        query_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast/new_query_spliced.txt\"\n",
    "        with open(query_location , 'w') as query_file_new:\n",
    "            # sequence_translated = records.seq[start+start_modifier-1:stop]\n",
    "            # print(sequence_translated)\n",
    "            \n",
    "            sequence_translated = records.seq[start+start_modifier-1:stop].translate()\n",
    "            print(sequence_translated)\n",
    "            # proceed_test = input(\"Proceed with this?\")\n",
    "            # while True:\n",
    "            #     if proceed_test.lower()[0] == \"n\":\n",
    "            #         assert False\n",
    "            #     elif proceed_test.lower()[0] == \"y\":\n",
    "            #         break\n",
    "            if \"*\" in sequence_translated:\n",
    "                print(\"Errror in Spliced query\")\n",
    "                assert False\n",
    "            \n",
    "            output = f\">{original_query_name[1:]}\\n{sequence_translated}\"\n",
    "            query_file_new.write(output)\n",
    "\n",
    "        genome_file = get_genome_file(genome_location,species)\n",
    "        genome = f\"{genome_location}/{species}/{genome_file}\"\n",
    "        out_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/{error_exon}/for_blast\"\n",
    "    \n",
    "        \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.htm\" -html'\n",
    "        # print(blast_command)\n",
    "        # subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "        blast_command = f'tblastn -seg no -query \"{query_location}\" -db \"{genome}\" -num_alignments 3 -out \"{out_location}/blast_out.txt\"'\n",
    "        subprocess.run(f'{blast_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "    \n",
    "\n",
    "                \n",
    "        # acceptor, donor, don_line, acc_line = process_spiceator_result(start, stop,annotated_genome_location,annotated_species_name)\n",
    "\n",
    "        if acceptor == \"Y\" and donor  == \"Y\":\n",
    "            splice_prediction = \"Y\"\n",
    "        else:\n",
    "            splice_prediction = \"N\"\n",
    "            \n",
    "        return(start,stop,gt_ag, splice_prediction) \n",
    "        # output_format = str(species_name.split(\"\\n\")[0])+\",\" + str(scaffold) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        # print(output_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f27f9cc5-c2a8-4cb7-83e1-7104ee982cd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_spiceator_result(start_coordinate, stop_coordinate,annotated_genome_location,annotated_species_name,query_species_original):\n",
    "    results_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species_original}/\"\n",
    "#     list_of_files_here = os.listdir(results_location)\n",
    "#     spliceator_results_file = ''\n",
    "#     for files in list_of_files_here:\n",
    "#         if files.startswith(\"Spliceator_results\"):\n",
    "#             spliceator_results_file = files\n",
    "#     if spliceator_results_file == '':\n",
    "#         print(\"Splice file missing\")\n",
    "#         assert False\n",
    "\n",
    "#     with open(f\"{results_location}/{spliceator_results_file}\", 'r') as splice_file:\n",
    "#         splice_file_content = splice_file.readlines()\n",
    "\n",
    "    splice_acceptor_presence = \"N\"\n",
    "    splice_donor_presence = \"N\"\n",
    "    splice_donor = ''\n",
    "    splice_acceptor = ''\n",
    "    start_coordinate = 0\n",
    "    stop_coordinate = 0\n",
    "#     for lines in splice_file_content:\n",
    "#         line_split = lines.split(\";\")\n",
    "#         if line_split[0]==\"Acceptor\" and splice_acceptor_presence == \"N\":\n",
    "#             acceptor_start = int(line_split[1])\n",
    "#             acceptor_end = acceptor_start + len(line_split[3])\n",
    "            \n",
    "#             if start_coordinate >=acceptor_start and start_coordinate <=acceptor_end:\n",
    "#                 splice_acceptor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if start_coordinate >= acceptor_start:\n",
    "#                 #         start_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_acceptor = lines\n",
    "#         if line_split[0]==\"Donor\" and splice_donor_presence == \"N\":\n",
    "#             donor_start = int(line_split[1])\n",
    "#             donor_end = donor_start + len(line_split[3])\n",
    "#             if stop_coordinate >=donor_start and stop_coordinate <=donor_end:\n",
    "#                 splice_donor_presence = \"Y\"\n",
    "#                 # while True:\n",
    "#                 #     if stop_coordinate >=donor_start:\n",
    "#                 #         stop_coordinate -= 3\n",
    "#                 #     else:\n",
    "#                 #         break\n",
    "#                 splice_donor = lines\n",
    "\n",
    "    return(splice_acceptor_presence, splice_donor_presence, splice_donor, splice_acceptor)\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50c956e3-7a49-4f1f-b0e6-bac420e663ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, ag_gt, splice_prediction,original_query_name, query_species ):\n",
    "    blast_location = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\"\n",
    "    with open(f\"{blast_location}/new_query_spliced.txt\", 'r') as query_file:\n",
    "        query_name_list = [query_file.readlines()[0][1:].rstrip()]\n",
    "        seq_modi = [[int(left_overhang),int(right_overhang)]]\n",
    "        print(seq_modi)\n",
    "\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "    \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    species_name = annotated_species_name\n",
    "\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        \n",
    "        with open(f\"{blast_location}/blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "        gt_ag = \"N\"\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "            # print(lines)\n",
    "            # print(query_name)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "                query_species = str(query_species_split[1]+\"_\"+query_species_split[2].rstrip())\n",
    "\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "                    if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "                        error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "        \n",
    "        print(f\"start_coordinate : {start_coor},stop_coordinate : {stop_coor}\")\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        seq_length = query_length\n",
    "        if (start != 0 or stop != 0):\n",
    "            start_modifier = seq_modi[i][0]\n",
    "            stop_modifier = seq_modi[i][1]\n",
    "        else:\n",
    "            start_modifier = 0\n",
    "            stop_modifier = 0  \n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "            if old_end != 0 and old_end > stop:\n",
    "\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "            if old_end != 0 and old_end < stop:\n",
    "                error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "\n",
    "            \n",
    "        output_format = str(species_name)+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \",\" + ag_gt + \",\" + splice_prediction +\"\\n\"  \n",
    "        # print(Output_Sequence)\n",
    "        return(output_format)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a79568c-ea02-45a4-88d2-b10f23f091e7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24', 'Exon_25']\n",
      "Colias_croceus\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 2132,2215\n",
      "after splice: 2132,2215\n",
      "sequence:\n",
      "CGTAGTTCACACTCTACGCATTCGGGGAGCCACTCCAGCGGCAGTAGTGGCTATGGCGGAAAGCCATCCACATCTGGATACAG\n",
      "sequence:\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSGY\n",
      "left = AG, right =Gt, stop_counter = 0\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSGY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "start_coordinate : 12340827,stop_coordinate : 12340907\n",
      "2132 2215 Y N\n",
      "Colias_croceus,NC_059568.1,12340826,12340909,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_3\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "51\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 6557,6711\n",
      "after splice: 6557,6711\n",
      "sequence:\n",
      "AGCAACAATGTCAGCCAGCCGCCCGTGGAGCTCCGTTGTAAAgttaaaaagaagaagaagcaGCCGGAGGAGACCCTCGAGGTACAGGAGGAGGAGCCGAGCCCTCCCGAGCCGGAGCCCGAGGTGAAGCCAGTGGAGGAGGAATGCGTTGATG\n",
      "sequence:\n",
      "SNNVSQPPVELRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKPVEEECVD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SNNVSQPPVELRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKPVEEECVD\n",
      "[[1, 1]]\n",
      "start_coordinate : 12345252,stop_coordinate : 12345404\n",
      "6557 6711 Y N\n",
      "Colias_croceus,NC_059568.1,12345251,12345405,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_1_query_Exon_3,1,51,51,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_4\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 7720,7808\n",
      "after splice: 7720,7808\n",
      "sequence:\n",
      "GCCAGCTCGCCCGCGCAGCCAGCTGAGGCCTCGTGCGTGGAGAGCATGGACATATCCACGCCGGAACGAAGCAGCGTTAAGGATAAG\n",
      "sequence:\n",
      "ASSPAQPAEASCVESMDISTPERSSVKDK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ASSPAQPAEASCVESMDISTPERSSVKDK\n",
      "[[2, 0]]\n",
      "start_coordinate : 12346416,stop_coordinate : 12346502\n",
      "7720 7808 Y N\n",
      "Colias_croceus,NC_059568.1,12346414,12346502,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_0_query_Exon_4,1,29,29,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_5\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "19\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 8294,8350\n",
      "after splice: 8294,8350\n",
      "sequence:\n",
      "GCGAGCACGTGTAACTTCCCGGTCAACACAACCAACAGGCCTTTTGCGAATATTGCC\n",
      "sequence:\n",
      "ASTCNFPVNTTNRPFANIA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ASTCNFPVNTTNRPFANIA\n",
      "[[0, 0]]\n",
      "start_coordinate : 12346988,stop_coordinate : 12347044\n",
      "8294 8350 Y N\n",
      "Colias_croceus,NC_059568.1,12346988,12347044,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_5,1,19,19,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_6\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 9432,9563\n",
      "after splice: 9432,9563\n",
      "sequence:\n",
      "GACGGATTCTCATGTGTGATCTCCATGCAAGATGGAGTCGTGATGTACACGACGTCGTCTCTCACCAACACGCTGGGTTTTCCAAAGGACATGTGGATTGGCCGGTCGTTTATCGATTTCGTTCATCCAAGG\n",
      "sequence:\n",
      "DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "[[0, 0]]\n",
      "start_coordinate : 12348126,stop_coordinate : 12348257\n",
      "9432 9563 Y N\n",
      "Colias_croceus,NC_059568.1,12348126,12348257,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_7\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "25\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 10841,10916\n",
      "after splice: 10841,10916\n",
      "sequence:\n",
      "GACCGCAACACATTCGCGTCCCTTATAACTAACGGCTTGGCGGCGCCGAAAACTGTTAATGGTACGCAAAAGAAAG\n",
      "sequence:\n",
      "DRNTFASLITNGLAAPKTVNGTQKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DRNTFASLITNGLAAPKTVNGTQKK\n",
      "[[0, 1]]\n",
      "start_coordinate : 12349535,stop_coordinate : 12349609\n",
      "10841 10916 Y N\n",
      "Colias_croceus,NC_059568.1,12349535,12349610,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_1_query_Exon_7,1,25,25,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_8\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "63\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 12992,13183\n",
      "after splice: 12992,13183\n",
      "sequence:\n",
      "AATACTGTATCAACGATGGTGTGCCGTATTCGCCGGTACAGAGGTCTGACTTCAGGCTTCGGGATAAAGGAAAGAATCGTGACGTTCATGCCGTTCCTACTGAAACTGTCCTTTAGAAATATATGCGATGAGGAAGGCGAGGTTATATACCTGTTAGTGCAGGCGACACCGTTCTTTTCAGCGTTTAAAT\n",
      "sequence:\n",
      "NTVSTMVCRIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "NTVSTMVCRIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "[[2, 1]]\n",
      "start_coordinate : 12351688,stop_coordinate : 12351876\n",
      "12992 13183 Y N\n",
      "Colias_croceus,NC_059568.1,12351686,12351877,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_8,1,63,63,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_9\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 13397,13625\n",
      "after splice: 13397,13625\n",
      "sequence:\n",
      "CCCAACGAGTTAGTATCGAAAGCTGTGCCATTCGTAATACGACATGCAGCGAACGGTGACATTAAGTACATTGACCCAGAGTCTGTGCCCTATCTGGGCTATTCGCCACAAGACATCGTCAATAAAGACGCCCTGCATTTATACCATCCTCACGATTTGGCGTATTTGAGACAGGTTTATGAAACTATTGTGAAAGAAGGGGGAGTACCACGCTCAAAACCCTATAG\n",
      "sequence:\n",
      "PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 12352093,stop_coordinate : 12352317\n",
      "13397 13625 Y N\n",
      "Colias_croceus,NC_059568.1,12352091,12352319,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_10\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 14457,14568\n",
      "after splice: 14457,14568\n",
      "sequence:\n",
      "ATGATGGCTCAAAATGGCGACTATGTCAAATTAGAGACGGAGTGGTCTTCGTTCATTAATCCATGGTCGAGGAAACTGGAATTTGTTATTGGAAAACATTTGTTGATTGAA\n",
      "sequence:\n",
      "MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "[[1, 0]]\n",
      "start_coordinate : 12353152,stop_coordinate : 12353262\n",
      "14457 14568 Y N\n",
      "Colias_croceus,NC_059568.1,12353151,12353262,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_11\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 14686,14805\n",
      "after splice: 14686,14805\n",
      "sequence:\n",
      "ggACCTAATAATCCAGATGTATTCCAATCTGCAGACATAGAAAAGTCTTTTAAGTTCTCTGAAGATGAAAAGAACAAAGCTCAAACTCTCAGGGAGAGTATTGTGAGGATTATGAATGAG\n",
      "sequence:\n",
      "GPNNPDVFQSADIEKSFKFSEDEKNKAQTLRESIVRIMNE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "GPNNPDVFQSADIEKSFKFSEDEKNKAQTLRESIVRIMNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 12353380,stop_coordinate : 12353499\n",
      "14686 14805 Y N\n",
      "Colias_croceus,NC_059568.1,12353380,12353499,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_12\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 15144,15293\n",
      "after splice: 15144,15293\n",
      "sequence:\n",
      "CTATTAACAAAGCCGGCCGAAGCAGCGAAACAACAGATGACCAAGCGTTGCCAAGACCTCGCCTCCTTCATGGAGAGTCTCATGGAGGAAGCTCCTAAGACTGACGATGAAATGTGGCTAGATATAAAGGAACCTGATAATggatattat\n",
      "sequence:\n",
      "LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "[[0, 0]]\n",
      "start_coordinate : 12353838,stop_coordinate : 12353987\n",
      "15144 15293 Y N\n",
      "Colias_croceus,NC_059568.1,12353838,12353987,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_13\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 15641,15759\n",
      "after splice: 15641,15759\n",
      "sequence:\n",
      "GAGAGAGATTCAGTGATGCTCGGTGGTATTTCCCCACACCACGACTGCAATGACAGTAAGTCAAGTACTGAGACACAAATAAGCTACAATCAACTTAATTATAACGAAACATTACAACg\n",
      "sequence:\n",
      "ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "[[0, 2]]\n",
      "start_coordinate : 12354335,stop_coordinate : 12354451\n",
      "15641 15759 Y N\n",
      "Colias_croceus,NC_059568.1,12354335,12354453,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_14\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "32\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 15892,15990\n",
      "after splice: 15892,15990\n",
      "sequence:\n",
      "tatttcGACAGTCATTTACTATTTTCTGAAGAGAACAACTCTGATACTCCATCTGATAGGCCGTCCGGCAATAAGAACATAAAAACGAATTCTAGCGG\n",
      "sequence:\n",
      "YFDSHLLFSEENNSDTPSDRPSGNKNIKTNSS\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "YFDSHLLFSEENNSDTPSDRPSGNKNIKTNSS\n",
      "[[1, 2]]\n",
      "start_coordinate : 12354587,stop_coordinate : 12354682\n",
      "15892 15990 Y N\n",
      "Colias_croceus,NC_059568.1,12354586,12354684,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_14,1,32,32,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_15\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "53\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 16385,16537\n",
      "after splice: 16385,16537\n",
      "sequence:\n",
      "TTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TG, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "TGATTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "*LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "TTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TG, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "TGATTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "*LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "TTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "FFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "TTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TG, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "TGATTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "*LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "TTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "FSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "TTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "FFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TT, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "TTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = TG, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "TGATTGTTTTTCAGTATGTCACCGATGGCCCAAAACTCCGGTGACTTTGGTGACATGACGAGCTCATGTGACTCCAGCGCACTCGCTCTCTCTAATTCCACACCTATGGGTGATTATTATCAACCCGTCAGGCTAACTGAGTCTGCTTTGAGCAA\n",
      "sequence:\n",
      "*LFFSMSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n",
      "left = AG, right =GT, stop_counter = 4\n",
      "MSPMAQNSGDFGDMTSSCDSSALALSNSTPMGDYYQPVRLTESALS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "start_coordinate : 12355092,stop_coordinate : 12355229\n",
      "16397 16537 Y N\n",
      "Colias_croceus,NC_059568.1,12355091,12355231,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_15,1,46,46,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_16\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "59\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 18106,18283\n",
      "after splice: 18106,18283\n",
      "sequence:\n",
      "CACAACGCAGTAATGGAGAAGGAATTAATGAAATTACACCGTGAAATAAAAACCGTAACCAAAGATGACAGGAAGAAGGCATCAAAGGAATCTAGGCAGAAGAAGAAGGAACATCTTGCCCGTTGTAATGCGTCTTTCTTGCCTACGTCAGCTGGATTATCTACTGTTGATTCTCAA\n",
      "sequence:\n",
      "HNAVMEKELMKLHREIKTVTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLSTVDSQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "HNAVMEKELMKLHREIKTVTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLSTVDSQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 12356801,stop_coordinate : 12356977\n",
      "18106 18283 Y N\n",
      "Colias_croceus,NC_059568.1,12356800,12356977,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_0_query_Exon_16,1,59,59,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_17\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "53\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 18711,18870\n",
      "after splice: 18711,18870\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacata\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEH\n",
      "left = ag, right =at, stop_counter = 0\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacataata\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHN\n",
      "left = ag, right =tt, stop_counter = 0\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacataatattt\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHNI\n",
      "left = ag, right =ta, stop_counter = 0\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacataatattttaa\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHNIL\n",
      "left = ag, right =ta, stop_counter = 0\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacataatattttaataa\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHNILI\n",
      "left = ag, right =tt, stop_counter = 0\n",
      "sequence:\n",
      "ccTCACGGTTTAAAGCGGTCGTCAAGGCACTCCGACGACTCCGAAACGGGTTATAAATACCGTTACTCTACAGCCAAGTTGCCCAGACGGAAGCAGAGCACAAGCGCTGCGCCAGCGCATCCATCAAACACACTTACTAATACTCTTGGTaaggaacataatattttaataattt\n",
      "sequence:\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHNILII\n",
      "left = ag, right =gt, stop_counter = 0\n",
      "PHGLKRSSRHSDDSETGYKYRYSTAKLPRRKQSTSAAPAHPSNTLTNTLGKEHNILII\n",
      "[[0, 1]]\n",
      "start_coordinate : 12357405,stop_coordinate : 12357578\n",
      "18711 18885 Y N\n",
      "Colias_croceus,NC_059568.1,12357405,12357579,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_1_query_Exon_17,1,58,58,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_18\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 19416,19529\n",
      "after splice: 19416,19529\n",
      "sequence:\n",
      "TCATCACCAATGCCGACAAATTCAATGAACATGAACAATATGAATACTTTCTTCCTCGGAGTTAGTATTCCTCAACAGATGTCGATGATGAGCCCAGTCGCTGCGATGCCTG\n",
      "sequence:\n",
      "SSPMPTNSMNMNNMNTFFLGVSIPQQMSMMSPVAAMP\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SSPMPTNSMNMNNMNTFFLGVSIPQQMSMMSPVAAMP\n",
      "[[2, 1]]\n",
      "start_coordinate : 12358112,stop_coordinate : 12358222\n",
      "19416 19529 Y N\n",
      "Colias_croceus,NC_059568.1,12358110,12358223,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_18,1,37,37,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_19\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 20170,20277\n",
      "after splice: 20170,20277\n",
      "sequence:\n",
      "ATGTTCCCGATATGCTATGCTATGCCCGCATCATCCCAGCAAATGATGCAAAATGATACGGAAATGCCAAACCAGGGACCGTCAACTAGCAATCATCCGTTTACca\n",
      "sequence:\n",
      "MFPICYAMPASSQQMMQNDTEMPNQGPSTSNHPFT\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "MFPICYAMPASSQQMMQNDTEMPNQGPSTSNHPFT\n",
      "[[2, 1]]\n",
      "start_coordinate : 12358866,stop_coordinate : 12358970\n",
      "20170 20277 Y N\n",
      "Colias_croceus,NC_059568.1,12358864,12358971,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_19,1,35,35,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_20\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 20764,20927\n",
      "after splice: 20764,20927\n",
      "sequence:\n",
      "CCCTATGTAATGTACGGTCAGCCGGTATATGGATCGCCGTTCATGTATTCGCCGATAAACGCTCAAATGGGTATGCAATACCCGATGCAGCCGGGTTTCGTCGCGCAAGGTATGAACGCGGCGATGCACCCACTGGGATTGTCCAGCGGGAACTACCAGGAG\n",
      "sequence:\n",
      "PYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "[[2, 0]]\n",
      "start_coordinate : 12359460,stop_coordinate : 12359621\n",
      "20764 20927 Y N\n",
      "Colias_croceus,NC_059568.1,12359458,12359621,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_0_query_Exon_20,1,54,54,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_21\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "48\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 23366,23509\n",
      "after splice: 23366,23509\n",
      "sequence:\n",
      "GCTTGTAAACTCACTTTCCCAACGAAGAGCAAGTTGGTCCCTTCGTCCACCTGGCGAGAGAAACGGGCGGAACAGCTAGCGGCCGCGGCTAAAAAAGACAAAAGCAGCGGTACGAGTAACTCGTCGGCTGCCGTTTCTAATCTG\n",
      "sequence:\n",
      "ACKLTFPTKSKLVPSSTWREKRAEQLAAAAKKDKSSGTSNSSAAVSNL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ACKLTFPTKSKLVPSSTWREKRAEQLAAAAKKDKSSGTSNSSAAVSNL\n",
      "[[0, 0]]\n",
      "start_coordinate : 12362060,stop_coordinate : 12362203\n",
      "23366 23509 Y N\n",
      "Colias_croceus,NC_059568.1,12362060,12362203,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_21,1,48,48,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_22\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 27259,27420\n",
      "after splice: 27259,27420\n",
      "sequence:\n",
      "AGTGCAAATGCAGCTACAGAGAACCGAAGCAAGGGCAAAGAGCGTCTCGGCAACAGCGACGCTACAAACGAAAAGACTGACGAGGAATCCAGCTATTCCTCGTTCTATTCTAACTTCTTTAAGAGCGAATCTGGCAGCGCTGAGGAAAGTGATGTTAAGAAG\n",
      "sequence:\n",
      "SANAATENRSKGKERLGNSDATNEKTDEESSYSSFYSNFFKSESGSAEESDVKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SANAATENRSKGKERLGNSDATNEKTDEESSYSSFYSNFFKSESGSAEESDVKK\n",
      "[[0, 0]]\n",
      "start_coordinate : 12365953,stop_coordinate : 12366114\n",
      "27259 27420 Y N\n",
      "Colias_croceus,NC_059568.1,12365953,12366114,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_22,1,54,54,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_23\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 31785,31871\n",
      "after splice: 31785,31871\n",
      "sequence:\n",
      "CGGTCGATGGATCCCATTCCACCTTATGAAAGTCAGCAAGCGAAAAAGGTAGCCCGAAGGAAAATGGATCCGCCATGGATGGAACAT\n",
      "sequence:\n",
      "RSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "RSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "[[0, 0]]\n",
      "start_coordinate : 12370479,stop_coordinate : 12370565\n",
      "31785 31871 Y N\n",
      "Colias_croceus,NC_059568.1,12370479,12370565,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_23,1,29,29,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_24\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 32163,32264\n",
      "after splice: 32163,32264\n",
      "sequence:\n",
      "GTATGTGTGACATCAGAGCTgatatacaaatatcaagtGCTACCCAAGACAATGGAAGAGGTTTTATCTTGCGACAAGAAGAGGCTAGAGACCTTAGAACAA\n",
      "sequence:\n",
      "VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 12370857,stop_coordinate : 12370958\n",
      "32163 32264 Y N\n",
      "Colias_croceus,NC_059568.1,12370857,12370958,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_24,1,34,34,Y,N\n",
      "\n",
      "Colias_croceus\n",
      "Processing Exon_25\n",
      "Colias_croceus_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "47\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 32605,32745\n",
      "after splice: 32605,32745\n",
      "sequence:\n",
      "CCAACACTAGTGAACGAGCAGTTGGGTCAGTTATACCTAGACCTTCAGTTAGAGGGCGTTGCCGCGCGGCTCACGCTCGAGGAAGGGCTCACCAGCTCTAGCAGCTCCGGGGAAGAGAACTTCTCCAGGAGCCCTAAGGTC\n",
      "sequence:\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFSRSPKV\n",
      "left = AG, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "CCAACACTAGTGAACGAGCAGTTGGGTCAGTTATACCTAGACCTTCAGTTAGAGGGCGTTGCCGCGCGGCTCACGCTCGAGGAAGGGCTCACCAGCTCTAGCAGCTCCGGGGAAGAGAACTTCTCCAGGAGCCCTAAGGTCCAG\n",
      "sequence:\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFSRSPKVQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFSRSPKVQ\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 0]]\n",
      "start_coordinate : 12371299,stop_coordinate : 12371442\n",
      "32605 32748 Y N\n",
      "Colias_croceus,NC_059568.1,12371299,12371442,0,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_25,1,48,48,Y,N\n",
      "\n",
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24', 'Exon_25']\n",
      "Colias_eurytheme\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 905,988\n",
      "after splice: 905,988\n",
      "sequence:\n",
      "CGTAGTTCACACTCTACGCATTCGGGGAGCCACTCCAGCGGTAGTAGTGGGTATGGCGGAAAGCCATCCACATCTGAATACAG\n",
      "sequence:\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSEY\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSEY\n",
      "[[1, 2]]\n",
      "start_coordinate : 185383,stop_coordinate : 185303\n",
      "905 988 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,185301,185384,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_3\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "51\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 6204,6358\n",
      "after splice: 6204,6358\n",
      "sequence:\n",
      "AACAACAATGTCAGCCAGCCGCCTGTGGAGATCCGGTGTAAAgttaaaaagaagaagaagcaGCCGGAGGAGACCCTCGAGGTGCAGGAGGAGGAGCCGAGCCCGCCCGAGCCGGAGCCCGAGGTGAAGTCGGCGGAGGAGGAATGCGTTGATG\n",
      "sequence:\n",
      "NNNVSQPPVEIRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKSAEEECVD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "NNNVSQPPVEIRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKSAEEECVD\n",
      "[[1, 1]]\n",
      "start_coordinate : 180084,stop_coordinate : 179932\n",
      "6204 6358 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,179931,180085,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_1_query_Exon_3,1,51,51,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_4\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 7452,7537\n",
      "after splice: 7452,7537\n",
      "sequence:\n",
      "GCCAGCTCGCCCACGCAGCCAGCTGAAGCGTCGTGCGTTGAGAGCATGGACATATCCACGCCGGAACGAAGCGTTAAGGATAAG\n",
      "sequence:\n",
      "ASSPTQPAEASCVESMDISTPERSVKDK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ASSPTQPAEASCVESMDISTPERSVKDK\n",
      "[[2, 0]]\n",
      "start_coordinate : 178835,stop_coordinate : 178752\n",
      "7452 7537 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,178752,178837,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_0_query_Exon_4,1,28,28,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_5\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "19\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 8046,8102\n",
      "after splice: 8046,8102\n",
      "sequence:\n",
      "TCGAGCACGTGTAACTTCCCGGTCAACACACCCAACAGGCCTTTTGCGAATATTGCT\n",
      "sequence:\n",
      "SSTCNFPVNTPNRPFANIA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SSTCNFPVNTPNRPFANIA\n",
      "[[0, 0]]\n",
      "start_coordinate : 178243,stop_coordinate : 178187\n",
      "8046 8102 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,178187,178243,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_5,1,19,19,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_6\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "44\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 8337,8468\n",
      "after splice: 8337,8468\n",
      "sequence:\n",
      "GACGGATTCTCATGTGTGATCTCCATGCAAGATGGAGTCGTGATGTATACGACGTCGTCTCTCACCAACACGCTCGGTTTTCCAAAGGACATGTGGATTGGCCGGTCGTTTATCGACTTCGTTCATCCAAGG\n",
      "sequence:\n",
      "DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DGFSCVISMQDGVVMYTTSSLTNTLGFPKDMWIGRSFIDFVHPR\n",
      "[[0, 0]]\n",
      "start_coordinate : 177952,stop_coordinate : 177821\n",
      "8337 8468 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,177821,177952,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_6,1,44,44,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_7\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "25\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 9528,9603\n",
      "after splice: 9528,9603\n",
      "sequence:\n",
      "GACCGCAACACATTCGCGTCCCTTATAACGAACGGGTTGGCGGCGCCGAAAACTGTTAATGGTACGGAAAAGAAAG\n",
      "sequence:\n",
      "DRNTFASLITNGLAAPKTVNGTEKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "DRNTFASLITNGLAAPKTVNGTEKK\n",
      "[[0, 1]]\n",
      "start_coordinate : 176761,stop_coordinate : 176687\n",
      "9528 9603 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,176686,176761,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_1_query_Exon_7,1,25,25,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_8\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "63\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 11266,11457\n",
      "after splice: 11266,11457\n",
      "sequence:\n",
      "AATATAGTATCAACGATGGTGGCTCGTATTCGCCGGTACAGAGGTCTGACTTCAGGCTTCGGAATAAAGGAAAGGATCGTGACGTTCATGCCGTTCCTACTGAAATTGTCCTTTAGAAACATATGCGATGAGGAAGGCGAGGTTATATACCTGTTAGTGCAGGCAACACCGTTCTTTTCAGCGTTTAAAT\n",
      "sequence:\n",
      "NIVSTMVARIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "NIVSTMVARIRRYRGLTSGFGIKERIVTFMPFLLKLSFRNICDEEGEVIYLLVQATPFFSAFK\n",
      "[[2, 1]]\n",
      "start_coordinate : 175021,stop_coordinate : 174833\n",
      "11266 11457 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,174832,175023,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_8,1,63,63,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_9\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "75\n",
      "[[2, 2]]\n",
      "reached here\n",
      "Before splice: 11666,11894\n",
      "after splice: 11666,11894\n",
      "sequence:\n",
      "CCCAACGAGTTAGTATCGAAAGCTGTGCCATTCGTAATAAGACATGCAGCGAACGGTGATATTAAGTACATTGATCCAGAGTCTGTGCCTTATCTGGGCTATTCACCACAAGACATAGTCAATAAAGACGCCCTACATTTATACCATCCTCACGACTTGGCGTATTTGAGACAGGTTTATGAAACTATTGTGAAAGAGGGGGGAGTACCACGCTCAAAACCTTATAG\n",
      "sequence:\n",
      "PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PNELVSKAVPFVIRHAANGDIKYIDPESVPYLGYSPQDIVNKDALHLYHPHDLAYLRQVYETIVKEGGVPRSKPY\n",
      "[[2, 2]]\n",
      "start_coordinate : 174621,stop_coordinate : 174397\n",
      "11666 11894 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,174395,174623,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_2_query_Exon_9,1,75,75,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_10\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "37\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 12768,12879\n",
      "after splice: 12768,12879\n",
      "sequence:\n",
      "atGATGGCTCAAAATGGCGACTATGTCAAATTAGAGACGGAGTGGTCTTCGTTCATTAATCCATGGTCGAGGAAACTGGAGTTCGTTATTGGAAAACATTTGTTGATTGAA\n",
      "sequence:\n",
      "MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "MMAQNGDYVKLETEWSSFINPWSRKLEFVIGKHLLIE\n",
      "[[1, 0]]\n",
      "start_coordinate : 173520,stop_coordinate : 173410\n",
      "12768 12879 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,173410,173521,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_0_query_Exon_10,1,37,37,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_11\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "40\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 12992,13111\n",
      "after splice: 12992,13111\n",
      "sequence:\n",
      "GGACCTAATAATCCAGATGTATTTCAATCTGCAGACATTGAAAAGTCTATTAAGTTGTCTGAAGATGAAAAGAATAAAGCTCAAACTATCAGGGAGAGTATTGTCAGGATTATGAATGAG\n",
      "sequence:\n",
      "GPNNPDVFQSADIEKSIKLSEDEKNKAQTIRESIVRIMNE\n",
      "left = aG, right =GT, stop_counter = 0\n",
      "GPNNPDVFQSADIEKSIKLSEDEKNKAQTIRESIVRIMNE\n",
      "[[0, 0]]\n",
      "start_coordinate : 173297,stop_coordinate : 173178\n",
      "12992 13111 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,173178,173297,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_11,1,40,40,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_12\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "50\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 15556,15705\n",
      "after splice: 15556,15705\n",
      "sequence:\n",
      "CTGTTAACAAAGCCGGCCGAAGCAGCCAAACAACAGATGACCAAGCGTTGCCAAGACCTCGCCTCGTTCATGGAGAGTCTCATGGAGGAAGCTCCTAAGACTGACGATGAAATGTGGCTAGATATCAAGGAACCTGATAATGgatattat\n",
      "sequence:\n",
      "LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "LLTKPAEAAKQQMTKRCQDLASFMESLMEEAPKTDDEMWLDIKEPDNGYY\n",
      "[[0, 0]]\n",
      "start_coordinate : 170733,stop_coordinate : 170584\n",
      "15556 15705 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,170584,170733,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_12,1,50,50,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_13\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "39\n",
      "[[0, 2]]\n",
      "reached here\n",
      "Before splice: 16076,16194\n",
      "after splice: 16076,16194\n",
      "sequence:\n",
      "GAGAGAGATTCAGTGATGCTCGGTGGTATATCCCCACACCACGATTGCAATGACAGTAAGTCAAGTACTGAGACACAAATAAGCTACAACCAACTTAATTATAACGAAACACTACAAcg\n",
      "sequence:\n",
      "ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "ERDSVMLGGISPHHDCNDSKSSTETQISYNQLNYNETLQ\n",
      "[[0, 2]]\n",
      "start_coordinate : 170213,stop_coordinate : 170097\n",
      "16076 16194 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,170095,170213,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_2_query_Exon_13,1,39,39,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_14\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 17104,17202\n",
      "after splice: 17104,17202\n",
      "sequence:\n",
      "TATTTCGACAGCCATTTACTGTTTTCCGAAGAGAACAACTCTGATACTCCTTCTGATAGGCCAGCCgaaaataagaatataaaagCGAATTCTAGCGG\n",
      "sequence:\n",
      "YFDSHLLFSEENNSDTPSDRPAENKNIKANSS\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "YFDSHLLFSEENNSDTPSDRPAENKNIKANSS\n",
      "[[1, 2]]\n",
      "start_coordinate : 169184,stop_coordinate : 169089\n",
      "17104 17202 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,169087,169185,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_14,1,32,32,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_15\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "53\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 17504,17656\n",
      "after splice: 17504,17656\n",
      "sequence:\n",
      "ttggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tg, right =GT, stop_counter = 0\n",
      "sequence:\n",
      "tgattggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "*LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tt, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "ttggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tg, right =GT, stop_counter = 1\n",
      "sequence:\n",
      "tgattggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "*LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = gT, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "gTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "VFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tt, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "ttggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tg, right =GT, stop_counter = 2\n",
      "sequence:\n",
      "tgattggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "*LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = TT, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "TTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "FSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = gT, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "gTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "VFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tt, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "ttggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = tg, right =GT, stop_counter = 3\n",
      "sequence:\n",
      "tgattggTTTTCAGTATGTCACCGATGGCCCAAAACTCAGGTGACTTAGGTGACATGACGAGCTCATGTGACTCCAGTGCACTCACTCTCTCTAATTCCACTCCTATGGGTGATTACTATCAACCCGTAAGGCTAACTGAGTCTGCTTTGACCAA\n",
      "sequence:\n",
      "*LVFSMSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "left = AG, right =GT, stop_counter = 4\n",
      "MSPMAQNSGDLGDMTSSCDSSALTLSNSTPMGDYYQPVRLTESALT\n",
      "[[1, 2]]\n",
      "start_coordinate : 168772,stop_coordinate : 168635\n",
      "17516 17656 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,168633,168773,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_15,1,46,46,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_16\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "59\n",
      "[[1, 0]]\n",
      "reached here\n",
      "Before splice: 17963,18140\n",
      "after splice: 17963,18140\n",
      "sequence:\n",
      "CACAACGCAGTTATGGAGAAAGAATTAATGAAACTACACCGTGAAATCAAAACCGGTACCAAAGACGACAGAAAGAAGGCATCAAAAGAATCTAGGCAGAAGAAGAAGGAACATCTTGCCCGTTGCAATGCATCTTTCTTGCCTACGTCGGCTGGATTATCTACTGTTGATTCTCAA\n",
      "sequence:\n",
      "HNAVMEKELMKLHREIKTGTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLSTVDSQ\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "HNAVMEKELMKLHREIKTGTKDDRKKASKESRQKKKEHLARCNASFLPTSAGLSTVDSQ\n",
      "[[1, 0]]\n",
      "start_coordinate : 168325,stop_coordinate : 168149\n",
      "17963 18140 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,168149,168326,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_0_query_Exon_16,1,59,59,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_17\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "53\n",
      "[[0, 1]]\n",
      "reached here\n",
      "Before splice: 18816,18975\n",
      "after splice: 18816,18975\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEH\n",
      "left = ag, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHN\n",
      "left = ag, right =TT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTG\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNI\n",
      "left = ag, right =TA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTAT\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIV\n",
      "left = ag, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCAT\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVS\n",
      "left = ag, right =TA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAG\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSL\n",
      "left = ag, right =TT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTT\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLV\n",
      "left = ag, right =AT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVY\n",
      "left = ag, right =GA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGAT\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYR\n",
      "left = ag, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRS\n",
      "left = ag, right =AC, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACG\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSN\n",
      "left = ag, right =AA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNE\n",
      "left = ag, right =TT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAATTC\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEI\n",
      "left = ag, right =TA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAATTCTAC\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEIL\n",
      "left = ag, right =AA, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAATTCTACAAA\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEILQ\n",
      "left = ag, right =GG, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAATTCTACAAAGGC\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEILQR\n",
      "left = ag, right =CT, stop_counter = 0\n",
      "sequence:\n",
      "ccTCATGGTTTAAAGCGGTCATCGAGGCATTCCGAAGACTCTGAATCGGGCTATAAATACCGTTACTCTACGGCCAAATTGCCCAGACGTAAACAGAGCACAAGCGCTGCGCCTGCGCATCCATCAACCACACTTACTAATACTCTTGGTAAGGAACACAATATTGTATCATTAGTTTATAGATCAAACGAAATTCTACAAAGGCCTT\n",
      "sequence:\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEILQRP\n",
      "left = ag, right =GT, stop_counter = 0\n",
      "PHGLKRSSRHSEDSESGYKYRYSTAKLPRRKQSTSAAPAHPSTTLTNTLGKEHNIVSLVYRSNEILQRP\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0, 1]]\n",
      "start_coordinate : 167473,stop_coordinate : 167267\n",
      "18816 19023 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,167266,167473,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_1_query_Exon_17,1,69,69,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_18\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 19307,19414\n",
      "after splice: 19307,19414\n",
      "sequence:\n",
      "TCATCCCCAATGCCGACAAATTCAATGAACAACATGAATACTTTCTTCCTCGGAGTCAGTATTCCTCAACAGATGTCGATGATGAGCCCAGTGGCTGCGATGCCTG\n",
      "sequence:\n",
      "SSPMPTNSMNNMNTFFLGVSIPQQMSMMSPVAAMP\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SSPMPTNSMNNMNTFFLGVSIPQQMSMMSPVAAMP\n",
      "[[2, 1]]\n",
      "start_coordinate : 166980,stop_coordinate : 166876\n",
      "19307 19414 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,166875,166982,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_18,1,35,35,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_19\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "35\n",
      "[[2, 1]]\n",
      "reached here\n",
      "Before splice: 20095,20202\n",
      "after splice: 20095,20202\n",
      "sequence:\n",
      "ATGTTCCCGATATGCTATACTATGCCTGCATCATCCCAGCAAATGATGCAAAATGATGTGGAAATGCCAAACCAGGAACCGTCAACTAGCAATCATCCGTTTACca\n",
      "sequence:\n",
      "MFPICYTMPASSQQMMQNDVEMPNQEPSTSNHPFT\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "MFPICYTMPASSQQMMQNDVEMPNQEPSTSNHPFT\n",
      "[[2, 1]]\n",
      "start_coordinate : 166192,stop_coordinate : 166088\n",
      "20095 20202 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,166087,166194,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_1_query_Exon_19,1,35,35,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_20\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 20688,20851\n",
      "after splice: 20688,20851\n",
      "sequence:\n",
      "CCCTATGTAATGTACGGTCAACCGGTATATGGATCGCCGTTCATGTATTCGCCGATAAACGCTCAAATGGGTATGCAATACCCTATGCAGCCGGGTTTCGTCGCGCAAGGTATGAACGCGGCGATGCACCCACTGGGATTGTCCAGCGGGAACTACCAGGAG\n",
      "sequence:\n",
      "PYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PYVMYGQPVYGSPFMYSPINAQMGMQYPMQPGFVAQGMNAAMHPLGLSSGNYQE\n",
      "[[2, 0]]\n",
      "start_coordinate : 165599,stop_coordinate : 165438\n",
      "20688 20851 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,165438,165601,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_0_query_Exon_20,1,54,54,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_21\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "48\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 23289,23432\n",
      "after splice: 23289,23432\n",
      "sequence:\n",
      "GCTTGTAAACTCACTTTCCCAACGAAGAGCAAGTTGGTCCCTTCGGCCACGTGGCGAGAGAAACGGGCGGAACAGCTAGCGGCCGCGGCTAAAAAAGATAAAAGCAGCGGCACGAGTAACTCGTCGGCTGCCGTTTCTAATCTG\n",
      "sequence:\n",
      "ACKLTFPTKSKLVPSATWREKRAEQLAAAAKKDKSSGTSNSSAAVSNL\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ACKLTFPTKSKLVPSATWREKRAEQLAAAAKKDKSSGTSNSSAAVSNL\n",
      "[[0, 0]]\n",
      "start_coordinate : 163000,stop_coordinate : 162857\n",
      "23289 23432 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,162857,163000,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_21,1,48,48,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_22\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "54\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 24616,24777\n",
      "after splice: 24616,24777\n",
      "sequence:\n",
      "AGTGCAAATGCAGCTACAGAGAACCAAAGCAAGGGCAAAGAGCGTCTCGGCAACAGCGACGCCACAATCGACAAGACTGACGAGGAATCCAGCTACTCCTCATTCTATTCTAACTTCTTTAAGAGCGAATCTGGCAGCGCTGAGGAAAGTGATGTTAAGAAG\n",
      "sequence:\n",
      "SANAATENQSKGKERLGNSDATIDKTDEESSYSSFYSNFFKSESGSAEESDVKK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SANAATENQSKGKERLGNSDATIDKTDEESSYSSFYSNFFKSESGSAEESDVKK\n",
      "[[0, 0]]\n",
      "start_coordinate : 161673,stop_coordinate : 161512\n",
      "24616 24777 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,161512,161673,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_22,1,54,54,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_23\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 25001,25087\n",
      "after splice: 25001,25087\n",
      "sequence:\n",
      "CGGTCGATGGATCCCATTCCACCTTATGAAAGTCAGCAAGCCAAAAAGGTAGCCCGAAGGAAAATGGATCCGCCATGGATGGAACAT\n",
      "sequence:\n",
      "RSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "RSMDPIPPYESQQAKKVARRKMDPPWMEH\n",
      "[[0, 0]]\n",
      "start_coordinate : 161288,stop_coordinate : 161202\n",
      "25001 25087 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,161202,161288,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_23,1,29,29,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_24\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "34\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 25202,25303\n",
      "after splice: 25202,25303\n",
      "sequence:\n",
      "GTATGTGTGACATCAGAGCTGATATACAAATATCAAGTGCTACCCAAGACAATGGAAGAGGTTTTATCTTGCGATAAGAAGAGGCTAGAGACTTTAGAACAA\n",
      "sequence:\n",
      "VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "VCVTSELIYKYQVLPKTMEEVLSCDKKRLETLEQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 161087,stop_coordinate : 160986\n",
      "25202 25303 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,160986,161087,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_24,1,34,34,Y,N\n",
      "\n",
      "Colias_eurytheme\n",
      "Processing Exon_25\n",
      "Colias_eurytheme_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "47\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 25967,26107\n",
      "after splice: 25967,26107\n",
      "sequence:\n",
      "CCAACACTAGTGAACGAGCAGTTGGGTCAGTTGTACCTGGACCTCCAGTTGGAGGGCGTGGCCGCGCGGCTCACGCTCGAGGAAGGGCTCACTAGCTCCAGCAGTTCGGGGGAAGAGAACTTCGCCAGGAGCCCTAAGGTC\n",
      "sequence:\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFARSPKV\n",
      "left = AG, right =CA, stop_counter = 0\n",
      "sequence:\n",
      "CCAACACTAGTGAACGAGCAGTTGGGTCAGTTGTACCTGGACCTCCAGTTGGAGGGCGTGGCCGCGCGGCTCACGCTCGAGGAAGGGCTCACTAGCTCCAGCAGTTCGGGGGAAGAGAACTTCGCCAGGAGCCCTAAGGTCCAG\n",
      "sequence:\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFARSPKVQ\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "PTLVNEQLGQLYLDLQLEGVAARLTLEEGLTSSSSSGEENFARSPKVQ\n",
      "[[0, 0]]\n",
      "start_coordinate : 160322,stop_coordinate : 160179\n",
      "25967 26110 Y N\n",
      "Colias_eurytheme,CAJQZN010043737.1,160179,160322,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_0_rightoh_0_query_Exon_25,1,48,48,Y,N\n",
      "\n",
      "['Exon_2', 'Exon_3', 'Exon_4', 'Exon_5', 'Exon_6', 'Exon_7', 'Exon_8', 'Exon_9', 'Exon_10', 'Exon_11', 'Exon_12', 'Exon_13', 'Exon_14', 'Exon_15', 'Exon_16', 'Exon_17', 'Exon_18', 'Exon_19', 'Exon_20', 'Exon_21', 'Exon_22', 'Exon_23', 'Exon_24', 'Exon_25']\n",
      "Colias_nastes\n",
      "Processing Exon_2\n",
      "0 query coordinate\n",
      "desktop.ini\n",
      "27\n",
      "[[1, 2]]\n",
      "reached here\n",
      "Before splice: 958,1041\n",
      "after splice: 958,1041\n",
      "sequence:\n",
      "CGTAGTTCACACTCTACGCATTCGGGGAGCCACTCCAGCGGCAGTAGTGGTTATGGCGGAAAGCCATCCACATCTGGATACAG\n",
      "sequence:\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSGY\n",
      "left = AG, right =gt, stop_counter = 0\n",
      "RSSHSTHSGSHSSGSSGYGGKPSTSGY\n",
      "[[1, 2]]\n",
      "start_coordinate : 32284,stop_coordinate : 32204\n",
      "958 1041 Y N\n",
      "Colias_nastes,CAJQZM010035413.1,32202,32285,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_2_query_Exon_2,1,27,27,Y,N\n",
      "\n",
      "Colias_nastes\n",
      "Processing Exon_3\n",
      "Colias_nastes_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "51\n",
      "[[1, 1]]\n",
      "reached here\n",
      "Before splice: 8427,8581\n",
      "after splice: 8427,8581\n",
      "sequence:\n",
      "AGCAACAATGTCAGCCAGCCGCCCGTGGAGCTGCGTTGTAAAgttaaaaagaagaagaagcaGCCGGAGGAAACCCTCGAGGTGCAGGAGGAGGAGCCGAGCCCTCCCGAGCCGGAGCCCGAGGTGAAGCCTGCGGAGGAGGAATGCGTTGATG\n",
      "sequence:\n",
      "SNNVSQPPVELRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKPAEEECVD\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "SNNVSQPPVELRCKVKKKKKQPEETLEVQEEEPSPPEPEPEVKPAEEECVD\n",
      "[[1, 1]]\n",
      "start_coordinate : 24815,stop_coordinate : 24663\n",
      "8427 8581 Y N\n",
      "Colias_nastes,CAJQZM010035413.1,24662,24816,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_1_rightoh_1_query_Exon_3,1,51,51,Y,N\n",
      "\n",
      "Colias_nastes\n",
      "Processing Exon_4\n",
      "Colias_nastes_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "29\n",
      "[[2, 0]]\n",
      "reached here\n",
      "Before splice: 11358,11446\n",
      "after splice: 11358,11446\n",
      "sequence:\n",
      "GCCAGCTCGCCCGCGCAGCCAGTTGAAGCGTCGTGCGTCGAGAGCATGGACATATCCACGCCAGAACGAAGCAGCGTTAAGGATAAG\n",
      "sequence:\n",
      "ASSPAQPVEASCVESMDISTPERSSVKDK\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ASSPAQPVEASCVESMDISTPERSSVKDK\n",
      "[[2, 0]]\n",
      "start_coordinate : 21883,stop_coordinate : 21797\n",
      "11358 11446 Y N\n",
      "Colias_nastes,CAJQZM010035413.1,21797,21885,1,N,14.Zerene_cesonia_XM_038360763.1_Frame_2_rightoh_0_query_Exon_4,1,29,29,Y,N\n",
      "\n",
      "Colias_nastes\n",
      "Processing Exon_5\n",
      "Colias_nastes_coordinates_14.Zerene_cesonia.csv\n",
      "1 query coordinate\n",
      "19\n",
      "[[0, 0]]\n",
      "reached here\n",
      "Before splice: 11967,12023\n",
      "after splice: 11967,12023\n",
      "sequence:\n",
      "GCGAGCACATATAACTTCCCGGTCAACACACCCAACAGGCCTTTTGCGAATATTGCC\n",
      "sequence:\n",
      "ASTYNFPVNTPNRPFANIA\n",
      "left = AG, right =GT, stop_counter = 0\n",
      "ASTYNFPVNTPNRPFANIA\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess\n",
    "annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Blast_result\"\n",
    "# annotated_genome_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Blast_result\"\n",
    "# species_list = [\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Bicyclus_anynana\"]\n",
    "\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "    \n",
    "blast_error = []\n",
    "# query_species = \n",
    "for annotated_species_name in species_list:\n",
    "    # annotated_species_name = \"Lasiommata_megera\"\n",
    "    species = annotated_species_name\n",
    "    query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/5.Coliadinae/1.Query\"\n",
    "    # query_location = \"/mnt/j/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/3.Satyrine/1.Query\"\n",
    "    query_species = \"32.Colias_croceus\" \n",
    "    query_transcript_list = os.listdir(f\"{query_location}/{query_species}\")\n",
    "    if \"desktop.ini\" in query_transcript_list:\n",
    "        query_transcript_list.remove(\"desktop.ini\")\n",
    "    \n",
    "    query_transcript = query_transcript_list[0]\n",
    "    \n",
    "    genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "    # genome_location = \"/mnt/g/Genomes_2023-12-26\"\n",
    "    list_of_exons_folders = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    # print(list_of_exons_folders)\n",
    "    error_exon_list = []\n",
    "    for folders in list_of_exons_folders:\n",
    "        \n",
    "        if folders.startswith(\"Exon\"):\n",
    "            if int(folders.split(\"_\")[1]) > 1:\n",
    "                error_exon_list.append(folders)\n",
    "    print(error_exon_list)\n",
    "    # break\n",
    "    \n",
    "    \n",
    "#     error_exon_list = [\"Exon_5\"]\n",
    "    for error_exon in error_exon_list:\n",
    "        print(annotated_species_name)\n",
    "        print(f\"Processing {error_exon}\")\n",
    "        coordinate_location = f\"{annotated_genome_location}/{species}\"\n",
    "        files_in_coordinate_location = os.listdir(coordinate_location)\n",
    "        coordinate_backup = 0\n",
    "        query_coordinate = 0\n",
    "        # print(files_in_coordinate_location)\n",
    "        for file_names in files_in_coordinate_location:\n",
    "            if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                coordinate_backup = 1\n",
    "            if file_names.endswith(f\"_coordinates_{query_species}.csv\"):\n",
    "                print(file_names)\n",
    "                query_coordinate = 1\n",
    "        print(query_coordinate, \"query coordinate\")\n",
    "        if query_coordinate == 0:\n",
    "            print(file_names)\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates_old.csv\" \"{coordinate_location}/{species}_coordinates_{query_species}.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        if coordinate_backup == 0:\n",
    "            copy_command = f'cp \"{coordinate_location}/{species}_coordinates.csv\" \"{coordinate_location}/{species}_coordinates_old.csv\"'\n",
    "            subprocess.run(f'{copy_command}', shell = True, stderr = subprocess.DEVNULL)\n",
    "        \n",
    "        query_error = run_blast_with_new_query(annotated_genome_location,\n",
    "                                     annotated_species_name,\n",
    "                                     error_exon, \n",
    "                                     query_species,\n",
    "                                     genome_location, \n",
    "                                     species )\n",
    "    \n",
    "        if query_error == \"Query_error\":\n",
    "            print(query_error)\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        left_overhang,right_overhang,original_query_name = get_query_info(query_location,query_species,query_transcript,error_exon)\n",
    "        \n",
    "        try:\n",
    "            start_coordinate,stop_coordinate,gt_ag, splice_prediction = process_genome_fragment_blast_file(annotated_genome_location, \n",
    "                                                                                                       annotated_species_name,\n",
    "                                                                                                       error_exon,\n",
    "                                                                                                       left_overhang,\n",
    "                                                                                                       right_overhang,                                                                                               \n",
    "                                                                                                       original_query_name,\n",
    "                                                                                                      query_species)\n",
    "\n",
    "            new_coordinate_file_line = process_genome_blast_file(annotated_genome_location, \n",
    "                                                         annotated_species_name,\n",
    "                                                         error_exon,\n",
    "                                                         left_overhang,\n",
    "                                                         right_overhang,\n",
    "                                                         gt_ag, \n",
    "                                                         splice_prediction,\n",
    "                                                         original_query_name,\n",
    "                                                            query_species)\n",
    "        except:\n",
    "            print(\"Error in Blast\")\n",
    "            blast_error.append(f\"{annotated_species_name},{error_exon}\")\n",
    "            continue\n",
    "        \n",
    "        print(start_coordinate,stop_coordinate,gt_ag, splice_prediction)\n",
    "        print(new_coordinate_file_line)\n",
    "        output = ''\n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'r') as open_coordinate_file:\n",
    "            coordinate_file_list = open_coordinate_file.readlines()\n",
    "            # print(coordinate_file_list)\n",
    "            # assert False\n",
    "        for lines in coordinate_file_list:\n",
    "            if lines.split(\",\")[6].endswith(error_exon):\n",
    "                lines = new_coordinate_file_line\n",
    "            output += lines\n",
    "        \n",
    "        with open(f\"{coordinate_location}/{species}_coordinates_{query_species}.csv\", 'w') as out_coordinate_file:\n",
    "            out_coordinate_file.write(output)\n",
    "\n",
    "\n",
    "print(\"\\n\".join(blast_error))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c490af29-b58c-4481-8429-8779374d7ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "process_genome_blast_file(annotated_genome_location, annotated_species_name,error_exon,left_overhang,right_overhang, \"1\", splice_prediction,original_query_name, query_species )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4fff37c-451f-48f4-8a15-eba962efc380",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
