{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f27d4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sequence_extractor(Species, Scaff, reverse_c, start, end, frame = 1, trans = 1 ):\n",
    "    from Bio import SeqIO\n",
    "    from Bio.Seq import Seq\n",
    "    import os\n",
    "    \n",
    "    out_Seq = ''\n",
    "    \n",
    "    \n",
    "    #Genome folder\n",
    "    entries = os.listdir(\"/mnt/f/Genomes_2023-12-26/\"+Species)\n",
    "    \n",
    "    #Get genome file from Genome folder\n",
    "    for file_names  in entries:\n",
    "        if \".nhr\" in file_names:\n",
    "            Genome_name = file_names[:-4]\n",
    "            break\n",
    "    \n",
    "    #Read the genome file\n",
    "    fasta_file = open((\"/mnt/f/Genomes_2023-12-26/\"+Species+\"/\"+Genome_name),'r')\n",
    "    \n",
    "    #Extract the sequence \n",
    "    for record in SeqIO.parse(fasta_file,\"fasta\"):\n",
    "        if record.id == Scaff:\n",
    "            sequence = str(record.seq)\n",
    "            out_Seq = Seq(sequence[start-1:end])\n",
    "            if reverse_c == 1:\n",
    "                out_Seq = out_Seq.reverse_complement()\n",
    "                \n",
    "    if len(out_Seq) < 10000: #fixing error due to mistake in typing the coordinate, change this for longer sequence\n",
    "#        print (out_Seq)\n",
    "        if frame == 1 :\n",
    "            out_trans = out_Seq[0:]\n",
    "        if frame == 2 :\n",
    "            out_trans = out_Seq[1:]\n",
    "        if frame == 3 :\n",
    "            out_trans = out_Seq[2:]\n",
    "    else:\n",
    "        print (\"too long\")\n",
    "        assert(False)\n",
    "        \n",
    "        \n",
    "    #    break\n",
    "    fasta_file.close()\n",
    "    if trans == 1:\n",
    "        return (out_trans.translate())\n",
    "    else:\n",
    "        return(out_Seq)\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4900aae5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d454c091",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000002.1,5897125,5897781,0,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000003.1,3641209,3641733,0,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000005.1,3729620,3730138,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000007.1,3880985,3881971,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000004.1,5810214,5811437,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000003.1,8767552,8768256,1,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000016.1,8440708,8441703,1,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000002.1,10794961,10796310,0,N,Colias_croceus_ef1,2,441,462\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000012.1,12489455,12490231,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000006.1,2553521,2554978,1,N,Colias_croceus_cat,14,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Colias_behrii,JARWMC010000004.1,7417143,7420280,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Colias_behrii,JARWMC010000002.1,5897125,5897781,0,N,Colias_croceus_rps5,1,219,219\n",
      "Colias_behrii,JARWMC010000003.1,3641209,3641733,0,N,Colias_croceus_wg,1,175,175\n",
      "Colias_behrii,JARWMC010000005.1,3729620,3730138,0,N,Colias_croceus_rps2,1,173,173\n",
      "Colias_behrii,JARWMC010000007.1,3880985,3881971,1,N,Colias_croceus_mdh,1,329,329\n",
      "Colias_behrii,JARWMC010000004.1,5810214,5811437,0,N,Colias_croceus_idh,1,408,408\n",
      "Colias_behrii,JARWMC010000003.1,8767552,8768256,1,N,Colias_croceus_ddc,1,235,235\n",
      "Colias_behrii,JARWMC010000016.1,8440708,8441703,1,N,Colias_croceus_gapdh,1,332,332\n",
      "Colias_behrii,JARWMC010000002.1,10794961,10796310,0,N,Colias_croceus_ef1,2,441,462\n",
      "Colias_behrii,JARWMC010000012.1,12489455,12490231,1,N,Colias_croceus_hcl,1,259,259\n",
      "Colias_behrii,JARWMC010000006.1,2553521,2554978,1,N,Colias_croceus_cat,14,482,482\n",
      "Colias_behrii,JARWMC010000004.1,7417143,7420280,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059538.1,5218796,5219452,0,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059539.1,3245110,3245634,0,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059541.1,9371257,9371775,1,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059544.1,3284986,3285972,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059540.1,5183156,5184379,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059539.1,7726033,7726737,1,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059551.1,3289895,3290890,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059537.1,3519697,3521082,0,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059549.1,10547352,10548128,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059568.1,239261,240706,0,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Colias_croceus,NC_059540.1,6484268,6487405,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Colias_croceus,NC_059538.1,5218796,5219452,0,N,Colias_croceus_rps5,1,219,219\n",
      "Colias_croceus,NC_059539.1,3245110,3245634,0,N,Colias_croceus_wg,1,175,175\n",
      "Colias_croceus,NC_059541.1,9371257,9371775,1,N,Colias_croceus_rps2,1,173,173\n",
      "Colias_croceus,NC_059544.1,3284986,3285972,1,N,Colias_croceus_mdh,1,329,329\n",
      "Colias_croceus,NC_059540.1,5183156,5184379,0,N,Colias_croceus_idh,1,408,408\n",
      "Colias_croceus,NC_059539.1,7726033,7726737,1,N,Colias_croceus_ddc,1,235,235\n",
      "Colias_croceus,NC_059551.1,3289895,3290890,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Colias_croceus,NC_059537.1,3519697,3521082,0,N,Colias_croceus_ef1,1,462,462\n",
      "Colias_croceus,NC_059549.1,10547352,10548128,1,N,Colias_croceus_hcl,1,259,259\n",
      "Colias_croceus,NC_059568.1,239261,240706,0,N,Colias_croceus_cat,1,482,482\n",
      "Colias_croceus,NC_059540.1,6484268,6487405,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010000045.1,70360,71016,0,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010043569.1,22621,23145,1,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010023122.1,4320,4838,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010017460.1,17347,18333,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010040646.1,21285,22508,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010037531.1,4533,5237,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010021018.1,8091,9086,1,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010033949.1,2135,3520,1,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010038386.1,11097,11873,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010040486.1,77581,79026,0,N,Colias_croceus_cat,1,422,482\n",
      "\n",
      "Colias_croceus\n",
      "Colias_eurytheme,CAJQZN010030549.1,5763,8900,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Colias_eurytheme,CAJQZN010000045.1,70360,71016,0,N,Colias_croceus_rps5,1,219,219\n",
      "Colias_eurytheme,CAJQZN010043569.1,22621,23145,1,N,Colias_croceus_wg,1,175,175\n",
      "Colias_eurytheme,CAJQZN010023122.1,4320,4838,0,N,Colias_croceus_rps2,1,173,173\n",
      "Colias_eurytheme,CAJQZN010017460.1,17347,18333,1,N,Colias_croceus_mdh,1,329,329\n",
      "Colias_eurytheme,CAJQZN010040646.1,21285,22508,0,N,Colias_croceus_idh,1,408,408\n",
      "Colias_eurytheme,CAJQZN010037531.1,4533,5237,0,N,Colias_croceus_ddc,1,235,235\n",
      "Colias_eurytheme,CAJQZN010021018.1,8091,9086,1,N,Colias_croceus_gapdh,1,332,332\n",
      "Colias_eurytheme,CAJQZN010033949.1,2135,3520,1,N,Colias_croceus_ef1,1,462,462\n",
      "Colias_eurytheme,CAJQZN010038386.1,11097,11873,1,N,Colias_croceus_hcl,1,259,259\n",
      "Colias_eurytheme,CAJQZN010040486.1,77581,79026,0,N,Colias_croceus_cat,1,422,482\n",
      "Colias_eurytheme,CAJQZN010030549.1,5763,8900,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010011753.1,1750,2406,1,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010038627.1,-156,368,1,Y,Colias_croceus_wg,1,122,175\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010024979.1,411,929,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010043258.1,13470,14456,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010038817.1,802,2025,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010026943.1,1556,2260,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010040790.1,25700,26695,1,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010040432.1,3465,4850,1,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010023028.1,2957,3733,0,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010027421.1,5166,6611,1,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Colias_nastes,CAJQZM010039989.1,38654,41791,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Colias_nastes,CAJQZM010011753.1,1750,2406,1,N,Colias_croceus_rps5,1,219,219\n",
      "Colias_nastes,CAJQZM010038627.1,-156,368,1,Y,Colias_croceus_wg,1,122,175\n",
      "Colias_nastes,CAJQZM010024979.1,411,929,0,N,Colias_croceus_rps2,1,173,173\n",
      "Colias_nastes,CAJQZM010043258.1,13470,14456,1,N,Colias_croceus_mdh,1,329,329\n",
      "Colias_nastes,CAJQZM010038817.1,802,2025,0,N,Colias_croceus_idh,1,408,408\n",
      "Colias_nastes,CAJQZM010026943.1,1556,2260,0,N,Colias_croceus_ddc,1,235,235\n",
      "Colias_nastes,CAJQZM010040790.1,25700,26695,1,N,Colias_croceus_gapdh,1,332,332\n",
      "Colias_nastes,CAJQZM010040432.1,3465,4850,1,N,Colias_croceus_ef1,1,462,462\n",
      "Colias_nastes,CAJQZM010023028.1,2957,3733,0,N,Colias_croceus_hcl,1,259,259\n",
      "Colias_nastes,CAJQZM010027421.1,5166,6611,1,N,Colias_croceus_cat,1,482,482\n",
      "Colias_nastes,CAJQZM010039989.1,38654,41791,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010001663.1,4923751,4924407,1,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010006787.1,7634851,7635375,1,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010005989.1,7043045,7043563,1,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010003811.1,6720632,6721618,0,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010001810.1,4057413,4058636,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010003939.1,9891,10595,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010005346.1,2877667,2878662,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010006098.1,2376019,2377404,0,N,Colias_croceus_ef1,1,370,462\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010006502.1,1827049,1827825,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010005835.1,8900536,8901981,0,N,Colias_croceus_cat,1,482,482\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus\n",
      "Eurema_hecabe,JADANM010001810.1,5074844,5077981,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Eurema_hecabe,JADANM010001663.1,4923751,4924407,1,N,Colias_croceus_rps5,1,219,219\n",
      "Eurema_hecabe,JADANM010006787.1,7634851,7635375,1,N,Colias_croceus_wg,1,175,175\n",
      "Eurema_hecabe,JADANM010005989.1,7043045,7043563,1,N,Colias_croceus_rps2,1,173,173\n",
      "Eurema_hecabe,JADANM010003811.1,6720632,6721618,0,N,Colias_croceus_mdh,1,329,329\n",
      "Eurema_hecabe,JADANM010001810.1,4057413,4058636,0,N,Colias_croceus_idh,1,408,408\n",
      "Eurema_hecabe,JADANM010003939.1,9891,10595,0,N,Colias_croceus_ddc,1,235,235\n",
      "Eurema_hecabe,JADANM010005346.1,2877667,2878662,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Eurema_hecabe,JADANM010006098.1,2376019,2377404,0,N,Colias_croceus_ef1,1,370,462\n",
      "Eurema_hecabe,JADANM010006502.1,1827049,1827825,1,N,Colias_croceus_hcl,1,259,259\n",
      "Eurema_hecabe,JADANM010005835.1,8900536,8901981,0,N,Colias_croceus_cat,1,482,482\n",
      "Eurema_hecabe,JADANM010001810.1,5074844,5077981,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01001926.1,76803,77459,1,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01001975.1,71177,71701,0,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000558.1,162575,163093,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01001073.1,116430,117416,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000129.1,861114,862337,1,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000843.1,374770,375474,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000077.1,551566,552561,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01001215.1,18972,20357,0,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01002131.1,125052,125828,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000462.1,30726,32171,0,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Phoebis_sennae,LQNK01000765.1,359185,362322,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Phoebis_sennae,LQNK01001926.1,76803,77459,1,N,Colias_croceus_rps5,1,219,219\n",
      "Phoebis_sennae,LQNK01001975.1,71177,71701,0,N,Colias_croceus_wg,1,175,175\n",
      "Phoebis_sennae,LQNK01000558.1,162575,163093,0,N,Colias_croceus_rps2,1,173,173\n",
      "Phoebis_sennae,LQNK01001073.1,116430,117416,1,N,Colias_croceus_mdh,1,329,329\n",
      "Phoebis_sennae,LQNK01000129.1,861114,862337,1,N,Colias_croceus_idh,1,408,408\n",
      "Phoebis_sennae,LQNK01000843.1,374770,375474,0,N,Colias_croceus_ddc,1,235,235\n",
      "Phoebis_sennae,LQNK01000077.1,551566,552561,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Phoebis_sennae,LQNK01001215.1,18972,20357,0,N,Colias_croceus_ef1,1,462,462\n",
      "Phoebis_sennae,LQNK01002131.1,125052,125828,1,N,Colias_croceus_hcl,1,259,259\n",
      "Phoebis_sennae,LQNK01000462.1,30726,32171,0,N,Colias_croceus_cat,1,482,482\n",
      "Phoebis_sennae,LQNK01000765.1,359185,362322,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052112.1,4286458,4287114,0,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052102.1,8675941,8676465,1,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052118.1,7948359,7948877,1,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052120.1,7751590,7752576,0,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052121.1,6582173,6583396,1,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052102.1,4569070,4569774,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052106.1,2573127,2574122,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052111.1,331274,332659,0,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052119.1,441722,442498,0,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052122.1,12383770,12385215,1,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Zerene_cesonia,NC_052121.1,5468793,5471930,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Zerene_cesonia,NC_052112.1,4286458,4287114,0,N,Colias_croceus_rps5,1,219,219\n",
      "Zerene_cesonia,NC_052102.1,8675941,8676465,1,N,Colias_croceus_wg,1,175,175\n",
      "Zerene_cesonia,NC_052118.1,7948359,7948877,1,N,Colias_croceus_rps2,1,173,173\n",
      "Zerene_cesonia,NC_052120.1,7751590,7752576,0,N,Colias_croceus_mdh,1,329,329\n",
      "Zerene_cesonia,NC_052121.1,6582173,6583396,1,N,Colias_croceus_idh,1,408,408\n",
      "Zerene_cesonia,NC_052102.1,4569070,4569774,0,N,Colias_croceus_ddc,1,235,235\n",
      "Zerene_cesonia,NC_052106.1,2573127,2574122,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Zerene_cesonia,NC_052111.1,331274,332659,0,N,Colias_croceus_ef1,1,462,462\n",
      "Zerene_cesonia,NC_052119.1,441722,442498,0,N,Colias_croceus_hcl,1,259,259\n",
      "Zerene_cesonia,NC_052122.1,12383770,12385215,1,N,Colias_croceus_cat,1,482,482\n",
      "Zerene_cesonia,NC_052121.1,5468793,5471930,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000027.1,7541472,7542128,1,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000008.1,1842767,1843291,1,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000035.1,5707006,5707524,1,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000024.1,11248833,11249819,0,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000023.1,6089363,6090586,1,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000009.1,14119752,14120456,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000014.1,7870658,7871653,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000025.1,6044626,6046011,0,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000005.1,3991705,3992481,0,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000001.1,12166361,12167809,0,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_juvernica,CATIXQ010000005.1,19028075,19031212,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Leptidea_juvernica,CATIXQ010000027.1,7541472,7542128,1,N,Colias_croceus_rps5,1,219,219\n",
      "Leptidea_juvernica,CATIXQ010000008.1,1842767,1843291,1,N,Colias_croceus_wg,1,175,175\n",
      "Leptidea_juvernica,CATIXQ010000035.1,5707006,5707524,1,N,Colias_croceus_rps2,1,173,173\n",
      "Leptidea_juvernica,CATIXQ010000024.1,11248833,11249819,0,N,Colias_croceus_mdh,1,329,329\n",
      "Leptidea_juvernica,CATIXQ010000023.1,6089363,6090586,1,N,Colias_croceus_idh,1,408,408\n",
      "Leptidea_juvernica,CATIXQ010000009.1,14119752,14120456,0,N,Colias_croceus_ddc,1,235,235\n",
      "Leptidea_juvernica,CATIXQ010000014.1,7870658,7871653,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Leptidea_juvernica,CATIXQ010000025.1,6044626,6046011,0,N,Colias_croceus_ef1,1,462,462\n",
      "Leptidea_juvernica,CATIXQ010000005.1,3991705,3992481,0,N,Colias_croceus_hcl,1,259,259\n",
      "Leptidea_juvernica,CATIXQ010000001.1,12166361,12167809,0,N,Colias_croceus_cat,1,482,482\n",
      "Leptidea_juvernica,CATIXQ010000005.1,19028075,19031212,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000025.1,3048472,3049128,1,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000006.1,25439228,25439752,0,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000001.1,22098147,22098665,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000026.1,5615869,5616855,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000009.1,24193222,24194445,0,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000001.1,21488711,21489415,1,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000012.1,12464475,12465470,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000020.1,890335,891720,0,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000019.1,4911732,4912508,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000017.1,11967778,11969226,0,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_reali,CATIXK010000014.1,7387126,7390263,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Leptidea_reali,CATIXK010000025.1,3048472,3049128,1,N,Colias_croceus_rps5,1,219,219\n",
      "Leptidea_reali,CATIXK010000006.1,25439228,25439752,0,N,Colias_croceus_wg,1,175,175\n",
      "Leptidea_reali,CATIXK010000001.1,22098147,22098665,0,N,Colias_croceus_rps2,1,173,173\n",
      "Leptidea_reali,CATIXK010000026.1,5615869,5616855,1,N,Colias_croceus_mdh,1,329,329\n",
      "Leptidea_reali,CATIXK010000009.1,24193222,24194445,0,N,Colias_croceus_idh,1,408,408\n",
      "Leptidea_reali,CATIXK010000001.1,21488711,21489415,1,N,Colias_croceus_ddc,1,235,235\n",
      "Leptidea_reali,CATIXK010000012.1,12464475,12465470,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Leptidea_reali,CATIXK010000020.1,890335,891720,0,N,Colias_croceus_ef1,1,462,462\n",
      "Leptidea_reali,CATIXK010000019.1,4911732,4912508,1,N,Colias_croceus_hcl,1,259,259\n",
      "Leptidea_reali,CATIXK010000017.1,11967778,11969226,0,N,Colias_croceus_cat,1,482,482\n",
      "Leptidea_reali,CATIXK010000014.1,7387126,7390263,1,N,Colias_croceus_cad,1,1046,1046\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066272.1,11751465,11752121,0,N,Colias_croceus_rps5,1,219,219\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066268.1,15254203,15254727,0,N,Colias_croceus_wg,1,175,175\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066299.1,9779759,9780277,0,N,Colias_croceus_rps2,1,173,173\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066278.1,14120943,14121929,1,N,Colias_croceus_mdh,1,329,329\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066274.1,1233308,1234531,1,N,Colias_croceus_idh,1,408,408\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066281.1,10843002,10843706,0,N,Colias_croceus_ddc,1,235,235\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066273.1,4059238,4060233,0,N,Colias_croceus_gapdh,1,332,332\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066308.1,8168777,8170162,1,N,Colias_croceus_ef1,1,462,462\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066298.1,7138307,7139083,1,N,Colias_croceus_hcl,1,259,259\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066312.1,23945829,23947277,1,N,Colias_croceus_cat,1,482,482\n",
      "\n",
      "Colias_croceus\n",
      "Leptidea_sinapis,NC_066270.1,8425936,8429073,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Leptidea_sinapis,NC_066272.1,11751465,11752121,0,N,Colias_croceus_rps5,1,219,219\n",
      "Leptidea_sinapis,NC_066268.1,15254203,15254727,0,N,Colias_croceus_wg,1,175,175\n",
      "Leptidea_sinapis,NC_066299.1,9779759,9780277,0,N,Colias_croceus_rps2,1,173,173\n",
      "Leptidea_sinapis,NC_066278.1,14120943,14121929,1,N,Colias_croceus_mdh,1,329,329\n",
      "Leptidea_sinapis,NC_066274.1,1233308,1234531,1,N,Colias_croceus_idh,1,408,408\n",
      "Leptidea_sinapis,NC_066281.1,10843002,10843706,0,N,Colias_croceus_ddc,1,235,235\n",
      "Leptidea_sinapis,NC_066273.1,4059238,4060233,0,N,Colias_croceus_gapdh,1,332,332\n",
      "Leptidea_sinapis,NC_066308.1,8168777,8170162,1,N,Colias_croceus_ef1,1,462,462\n",
      "Leptidea_sinapis,NC_066298.1,7138307,7139083,1,N,Colias_croceus_hcl,1,259,259\n",
      "Leptidea_sinapis,NC_066312.1,23945829,23947277,1,N,Colias_croceus_cat,1,482,482\n",
      "Leptidea_sinapis,NC_066270.1,8425936,8429073,0,N,Colias_croceus_cad,1,1046,1046\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#%reset -f\n",
    "blast_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/5.Coliadinae\"\n",
    "query_species = \"Colias_croceus\"\n",
    "\n",
    "\n",
    "species_list = os.listdir(f\"{blast_location}/1.Blast_result\")\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "query_file = SeqIO.parse(f\"{blast_location}/1.Query/query.txt\", 'fasta')\n",
    "query_name_list = []\n",
    "\n",
    "for records in query_file:\n",
    "    query_name_list.append(records.id)\n",
    "\n",
    "\n",
    "\n",
    "for species in species_list:\n",
    "    # print(species_name)\n",
    "    # species = species_name.split(\".\")[1]\n",
    "    header = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length\\n\" \n",
    "    Output_Sequence = header\n",
    "    scaff = \"Intial_value\"\n",
    "    scaff_old = \"Intial_value\"\n",
    "    old_end = 0\n",
    "    \n",
    "#Run for each query in the query list\n",
    "    for i in range(len(query_name_list)):\n",
    "        query_name = query_name_list[i]\n",
    "        Length_switch = \"0\"\n",
    "        with open(f\"{blast_location}/1.Blast_result/{species}/{species}_blast_out.txt\",'r') as tblast_out:\n",
    "            lines_in_file = tblast_out.readlines()\n",
    "        #print(lines_in_file)\n",
    "\n",
    "        result_section_switch = 0\n",
    "        start_coor_switch = 0\n",
    "        query_start_coor_switch = 0\n",
    "        stop_coor_switch = 0\n",
    "        error = \"N\"\n",
    "        break_switch = 0\n",
    "\n",
    "        start = 0\n",
    "        stop = 0\n",
    "        start_coor = 0\n",
    "        stop_coor = 0\n",
    "        query_length = 0\n",
    "\n",
    "        for lines in lines_in_file:\n",
    "\n",
    "#             print(lines)\n",
    "            if query_name in lines:\n",
    "            #Initialize that results can now be checked\n",
    "                result_section_switch = 1\n",
    "                query_species_split = lines.split(\" \")[1].split(\"_\")\n",
    "#                 print(query_species_split)\n",
    "                query_species = str(query_species_split[0]+\"_\"+query_species_split[1].rstrip())\n",
    "                print(query_species)\n",
    "            if result_section_switch == 1 and \"Lambda\" in lines:\n",
    "            #This block indicates end of the results block in blast output\n",
    "                result_section_switch == 0\n",
    "                \n",
    "                break\n",
    "\n",
    "            if result_section_switch == 1:\n",
    "            #While checking the result\n",
    "                if \"Length=\" in lines and Length_switch == \"0\":\n",
    "                #Get query length from the blast output\n",
    "                    \n",
    "                    query_length = int(lines.split(\"=\")[1].rstrip())\n",
    "                    \n",
    "                    Length_switch = 1 #Indicated length has been acquired\n",
    "                    \n",
    "                if (\"Score\" in lines or \">\" in lines) and (start_coor_switch == 1):\n",
    "    #                print (lines)\n",
    "                    break\n",
    "        \n",
    "                if \">\" in lines:\n",
    "                #Start of the first result\n",
    "                    scaff = lines.split(\" \")[0][1:] #Scaffold from the result\n",
    "#                     if scaff_old != \"Intial_value\" and scaff_old != scaff:\n",
    "#                         error = \"Y\"\n",
    "                    scaff_old = scaff\n",
    "                    \n",
    "                if \"Query\" in lines and \"=\" not in lines:\n",
    "                #Read the query line in output\n",
    "                    if query_start_coor_switch == 0:\n",
    "#                        print(lines)\n",
    "                        query_start_coor = int(lines.split(\" \")[2])\n",
    "                        query_start_coor_switch = 1\n",
    "                        #Query start coordinate fixed\n",
    "            \n",
    "                    query_stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting query stop coordinates for multiline result\n",
    "    #                print (stop_coor)\n",
    "                    \n",
    "                if \"Sbjct\" in lines:\n",
    "                #Read the blast target line\n",
    "                    if start_coor_switch == 0:\n",
    "                        start_coor = int(lines.split(\" \")[2])\n",
    "                        start_coor_switch = 1\n",
    "                    stop_coor =int(lines.split(\" \")[-1][:-1])\n",
    "                    #Keep getting target stop coordinates for multiline result\n",
    "                \n",
    "                \n",
    "\n",
    "        \n",
    "        if break_switch == 1:\n",
    "            break\n",
    "\n",
    "        if start_coor < stop_coor:\n",
    "            complement = \"0\" #Forward complement\n",
    "            \n",
    "            length = (stop_coor-start_coor)/3\n",
    "            start = start_coor\n",
    "            stop = stop_coor\n",
    "\n",
    "        elif start_coor > stop_coor:\n",
    "            complement = \"1\" #Reverse complement\n",
    "            length = (-stop_coor+start_coor)/3\n",
    "            start = stop_coor\n",
    "            stop = start_coor\n",
    "\n",
    "        else:\n",
    "            error = \"Y\"\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "        #Check if the length of target (blast hit) is significantly smaller than query\n",
    "        if length < query_length - 0.2*query_length:\n",
    "            error = \"Y\"\n",
    "\n",
    "        old_trans = ''\n",
    "#             print(start, stop)\n",
    "\n",
    "\n",
    "# #Check for Met or STOP at the start and end\n",
    "#         while True and (start != 0 or stop != 0):\n",
    "#             if (query_name == query_name_list[0] or query_name == query_name_list[-1]):\n",
    "#                 frame = 1\n",
    "#                 translated_sequence = (sequence_extractor(species, scaff, int(complement), start, stop,frame ))\n",
    "\n",
    "#                 if translated_sequence == old_trans:\n",
    "#                     break\n",
    "                    \n",
    "# #Check for Met at the beginning of first exon\n",
    "\n",
    "#                 if query_name == query_name_list[0] :\n",
    "#                     if translated_sequence[0] != \"M\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             start = int(start) - 3\n",
    "#                         if complement == \"1\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if \"*\" in translated_sequence:\n",
    "#                         error = \"Y\"\n",
    "#                         break\n",
    "#                     if translated_sequence[0] == \"M\":\n",
    "#                         break\n",
    "\n",
    "# #Check for stop at the end of last exon\n",
    "\n",
    "#                 if query_name == query_name_list[-1]:\n",
    "#                     if translated_sequence[-1] != \"*\":\n",
    "#                         print(translated_sequence)\n",
    "# #                             print(start, stop)\n",
    "#                         if complement == \"0\":\n",
    "#                             stop = int(stop) + 3\n",
    "#                         if complement == \"1\":\n",
    "#                             start = int(start) - 3\n",
    "#                         old_trans = translated_sequence\n",
    "#                     if translated_sequence[-1] == \"*\":\n",
    "\n",
    "#                         break\n",
    "#             else:\n",
    "#                 break\n",
    "        \n",
    "        \n",
    "#Add offset at the beginning and end of each exon\n",
    "#         if (start != 0 or stop != 0):\n",
    "            \n",
    "#             start_modifier = int(query_name.split(\"Frame\")[1][1])\n",
    "#             stop_modifier = int(query_name.split(\"rightoh\")[1][1])\n",
    "#         else:\n",
    "        start_modifier = 0\n",
    "        stop_modifier = 0            \n",
    "\n",
    "        seq_length = query_length\n",
    "        \n",
    "        \n",
    "#Get exon start and end by adding or removing codons\n",
    "        \n",
    "    #if not the beginning\n",
    "        if query_start_coor != \"1\" and query_name != query_name_list[0]:\n",
    "            if complement == \"0\":\n",
    "                start = int(start) - 3*(int(query_start_coor)-1)                \n",
    "            if complement == \"1\":\n",
    "                stop = int(stop) + 3*(int(query_start_coor)-1)\n",
    "    \n",
    "    #For the end\n",
    "        if query_stop_coor != str(seq_length) and query_name != query_name_list[-1]:\n",
    "            if complement == \"0\":\n",
    "                stop = int(stop) + 3*(int(seq_length)-int(query_stop_coor))\n",
    "            if complement == \"1\":\n",
    "                \n",
    "\n",
    "                start = int(start) - 3*(int(seq_length)-int(query_stop_coor))\n",
    " \n",
    "        \n",
    "        \n",
    "        \n",
    "#Adding or removing 3' and 5' overhangs for forward and reverse complement\n",
    "    #For forward complement\n",
    "        if complement == \"0\":\n",
    "            start = int(start) - int(start_modifier)\n",
    "            stop = int(stop) +  int(stop_modifier)\n",
    "#             if old_end != 0 and old_end > stop:\n",
    "\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #For reverse complement\n",
    "        if complement == \"1\":\n",
    "            start = int(start) - int(stop_modifier)\n",
    "            stop = int(stop) +  int(start_modifier)\n",
    "#             if old_end != 0 and old_end < stop:\n",
    "#                 error = \"Y\"\n",
    "            old_end = stop\n",
    "\n",
    "    #Simple check for lenghth\n",
    "        if start == 0 or stop == 0:\n",
    "            error = \"Y\"\n",
    "            \n",
    "           \n",
    "        output_format = str(species.split(\"\\n\")[0])+\",\" + str(scaff) +\",\" + str(start)+\",\" + str(stop)+\",\" + str(complement)+\",\" + str(error)+  \",\"+ str(query_name)+\",\"+ str(query_start_coor)+\",\"+str(query_stop_coor)+\",\"+str(query_length)+ \"\\n\"  \n",
    "        print(output_format)\n",
    "#         assert False\n",
    "        Output_Sequence = Output_Sequence + output_format\n",
    "\n",
    "    print(Output_Sequence)\n",
    "    output_file = open(f\"{blast_location}/1.Blast_result/{species}/{species}_coordinates.csv\",'w')\n",
    "    output_file.write(Output_Sequence)\n",
    "    output_file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fa07f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465fa823",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
