{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c17fd178-5fc9-40d1-83e5-56fa358cd304",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "file_name = sys.argv[1]\n",
    "location = \"/mnt/griffin/saubar\"\n",
    "\n",
    "# file_name = f'{family}_{gene}_nostop.fas'\n",
    "gene_name = file_name.split(\".\")[0]\n",
    "\n",
    "original_fasta_file = SeqIO.parse(f\"{location}/2.Individual_gene_files_BUSCO/{file_name}\", 'fasta')\n",
    "\n",
    "initial_output = ''\n",
    "for original_sequences in original_fasta_file:\n",
    "    if original_sequences.seq.translate()[-1] == \"*\":\n",
    "        initial_output += f\">{original_sequences.id}\\n{original_sequences.seq[:-3]}\\n\\n\"\n",
    "    else:\n",
    "        initial_output += f\">{original_sequences.id}\\n{original_sequences.seq}\\n\\n\"\n",
    "    \n",
    "list_of_gene_folders = os.listdir(f\"{location}/3.Busco_processing_individual_genes/\")\n",
    "if gene_name not in list_of_gene_folders:\n",
    "    os.mkdir(f\"{location}/3.Busco_processing_individual_genes/{gene_name}\")\n",
    "\n",
    "with open(f\"{location}/3.Busco_processing_individual_genes/{gene_name}/{gene_name}_nostop.fas\", 'w') as out_file:\n",
    "    out_file.write(initial_output)\n",
    "\n",
    "\n",
    "subprocess.run(f'cd {location}/3.Busco_processing_individual_genes/{gene_name}\\nprank -d={gene_name}_nostop.fas -o={gene_name}_nostop.fas.aligned.fas -codon -iterate=10', shell = True, stderr = subprocess.DEVNULL)\n",
    "new_location = f\"{location}/3.Busco_processing_individual_genes/{gene_name}\"\n",
    "\n",
    "\n",
    "output1 = \"\"\n",
    "output2 = ''\n",
    "\n",
    "\n",
    "untranslated = SeqIO.parse(f\"{new_location}/{gene_name}_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "\n",
    "for records in untranslated:\n",
    "    if records.seq.translate()[-1] == \"*\":\n",
    "        output1 += f\">{records.id}\\n{records.seq.replace('-','')[:-3]}\\n\\n\"\n",
    "        output2 += f\">{records.id}\\n{records.seq.translate()[:-1]+'-'}\\n\\n\"\n",
    "    else:\n",
    "        output1 += f\">{records.id}\\n{records.seq.replace('-','')}\\n\\n\"\n",
    "        output2 += f\">{records.id}\\n{records.seq.translate()}\\n\\n\"\n",
    "# print(output1)\n",
    "#mkdir(f\"{location}/1.trimal_outputs\")\n",
    "try:\n",
    "    os.mkdir(f\"{location}/1.trimal_outputs\")\n",
    "except:\n",
    "   pass\n",
    "with open(f\"{location}/1.trimal_outputs/{gene_name}_nostop.fas.aligned.fas.ungapped.fas\", 'w') as out_file:\n",
    "    out_file.write(output1)\n",
    "\n",
    "with open(f\"{location}/1.trimal_outputs/{gene_name}_nostop.fas.aligned.fas.translated.fas\", 'w') as out_file:\n",
    "    out_file.write(output2)\n",
    "\n",
    "subprocess.run(f'cd {location}/1.trimal_outputs\\ntrimal -colnumbering -htmlout trimal_output.html -in {gene_name}_nostop.fas.aligned.fas.translated.fas -out {gene_name}_nostop.fas.aligned.fas.trimal.fas -automated1 > columns_kept.txt', shell = True)\n",
    "subprocess.run(f'cd {location}/1.trimal_outputs\\ntrimal -backtrans {gene_name}_nostop.fas.aligned.fas.ungapped.fas -htmlout trimal_output.html -in {gene_name}_nostop.fas.aligned.fas.translated.fas -out {gene_name}_nostop.fas.aligned.fas.trimal.fas -automated1', shell = True)\n",
    "\n",
    "print(f'\"prank\" -d={file_name} -o={file_name}.aligned.fas -codon')\n",
    "#assert False\n",
    "\n",
    "try:\n",
    "    os.mkdir(f\"{new_location}/2.iqtree_outputs\")\n",
    "except:\n",
    "   pass\n",
    "\n",
    "subprocess.run(f'cd {location}/2.iqtree_outputs\\ncp ../1.trimal_outputs/{gene_name}_nostop.fas.aligned.fas.trimal.fas ./', shell = True)\n",
    "  \n",
    "subprocess.run(f'cd {new_location}/2.iqtree_outputs\\niqtree -s {gene_name}_nostop.fas.aligned.fas.trimal.fas -m MFP -t RANDOM -bb 1000 -alrt 1000 -bnni -nt AUTO', shell = True)\n",
    "\n",
    "subprocess.run(f'cd {new_location}/2.iqtree_outputs\\ncp ./{gene_name}_nostop.fas.aligned.fas.trimal.fas.treefile ../', shell = True)\n",
    "\n",
    "subprocess.run(f'cd {new_location}\\nHYPHYMPI absrel --alignment {gene_name}_nostop.fas.aligned.fas.trimal.fas --tree {gene_name}_nostop.fas.aligned.fas.trimal.fas.treefile ', shell = True)\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI busted --alignment {file_name}.aligned.fas.trimal.fas --tree {file_name}.aligned.fas.trimal.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "# print(f\"HYPHYMPI busted --alignment {file_name}.aligned.fas.best.fas --tree{file_name}.aligned.fas.best.fas.treefile \")\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI busted --alignment {file_name}.aligned.fas.best.fas --tree {file_name}.aligned.fas.best.fas.treefile ', shell = True)\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI fel --alignment {file_name}.aligned.fas.trimal.fas --tree {file_name}.aligned.fas.trimal.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "# #print(f\"HYPHYMPI busted --alignment {file_name}.aligned.fas.best.fas --tree{file_name}.aligned.fas.best.fas.treefile \")\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI fel --alignment {file_name}.aligned.fas.best.fas --tree {file_name}.aligned.fas.best.fas.treefile ', shell = True)\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI meme --alignment {file_name}.aligned.fas.trimal.fas --tree {file_name}.aligned.fas.trimal.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "# #print(f\"HYPHYMPI busted --alignment {file_name}.aligned.fas.best.fas --tree{file_name}.aligned.fas.best.fas.treefile \")\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI meme --alignment {file_name}.aligned.fas.best.fas --tree {file_name}.aligned.fas.best.fas.treefile ', shell = True)\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI absrel --alignment {file_name}.aligned.fas.trimal.fas --tree {file_name}.aligned.fas.trimal.fas.treefile ', shell = True, stderr = subprocess.DEVNULL)\n",
    "# #print(f\"HYPHYMPI busted --alignment {file_name}.aligned.fas.best.fas --tree{file_name}.aligned.fas.best.fas.treefile \")\n",
    "\n",
    "# subprocess.run(f'cd {location}\\nHYPHYMPI absrel --alignment {file_name}.aligned.fas.best.fas --tree {file_name}.aligned.fas.best.fas.treefile ', shell = True)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
