{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "aa7a4de4-f487-4b7e-860b-58b26519e10e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "busco_gene_dic = {}\n",
    "\n",
    "busco_fasta = SeqIO.parse(\"/mnt/g/My Drive/Circadian Rhythm Genes Project/22.Busco_related/4.Busco_aligned/1422at7088_nostop.fas.aligned.fas.best.fas\", 'fasta')\n",
    "\n",
    "for records in busco_fasta:\n",
    "    busco_gene_dic[records.id] = records.seq.replace(\"-\",'').upper()\n",
    "\n",
    "tim_gene_dic = {}\n",
    "\n",
    "tim_fasta = SeqIO.parse(\"/mnt/g/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/12.Family_specific_tests_timeless/Combined/Combined_timeless.fas\", 'fasta')\n",
    "\n",
    "for records in tim_fasta:\n",
    "    tim_gene_dic[records.id.split(\"-\")[0]] = records.seq.replace(\"-\",'').upper()\n",
    "\n",
    "# for key in tim_gene_dic:\n",
    "#     print(key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "45b940b7-f09d-43b4-ac36-76d1faba5047",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3209.   \n",
      "Segment  23/ 26 3209-3361.   \n",
      "Segment  24/ 26 3361-3512.   \n",
      "Segment  25/ 26 3512-3652.   \n",
      "Segment  26/ 26 3652-3727.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3203.   \n",
      "Segment  23/ 26 3203-3345.   \n",
      "Segment  24/ 26 3345-3496.   \n",
      "Segment  25/ 26 3496-3640.   \n",
      "Segment  26/ 26 3640-3718.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 974.   \n",
      "Segment   8/ 26  974-1152.   \n",
      "Segment   9/ 26 1152-1303.   \n",
      "Segment  10/ 26 1303-1454.   \n",
      "Segment  11/ 26 1454-1605.   \n",
      "Segment  12/ 26 1605-1756.   \n",
      "Segment  13/ 26 1756-1907.   \n",
      "Segment  14/ 26 1907-2058.   \n",
      "Segment  15/ 26 2058-2209.   \n",
      "Segment  16/ 26 2209-2360.   \n",
      "Segment  17/ 26 2360-2511.   \n",
      "Segment  18/ 26 2511-2662.   \n",
      "Segment  19/ 26 2662-2813.   \n",
      "Segment  20/ 26 2813-2964.   \n",
      "Segment  21/ 26 2964-3115.   \n",
      "Segment  22/ 26 3115-3214.   \n",
      "Segment  23/ 26 3214-3367.   \n",
      "Segment  24/ 26 3367-3518.   \n",
      "Segment  25/ 26 3518-3661.   \n",
      "Segment  26/ 26 3661-3739.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3688.   \n",
      "Segment  26/ 26 3688-3751.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3545.   \n",
      "Segment  25/ 25 3545-3616.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3536.   \n",
      "Segment  25/ 25 3536-3598.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3526.   \n",
      "Segment  25/ 25 3526-3577.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3524.   \n",
      "Segment  25/ 25 3524-3574.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3532.   \n",
      "Segment  25/ 25 3532-3589.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3539.   \n",
      "Segment  25/ 25 3539-3604.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3643.   \n",
      "Segment  26/ 26 3643-3661.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 975.   \n",
      "Segment   8/ 26  975-1133.   \n",
      "Segment   9/ 26 1133-1284.   \n",
      "Segment  10/ 26 1284-1435.   \n",
      "Segment  11/ 26 1435-1586.   \n",
      "Segment  12/ 26 1586-1737.   \n",
      "Segment  13/ 26 1737-1888.   \n",
      "Segment  14/ 26 1888-2039.   \n",
      "Segment  15/ 26 2039-2190.   \n",
      "Segment  16/ 26 2190-2309.   \n",
      "Segment  17/ 26 2309-2470.   \n",
      "Segment  18/ 26 2470-2621.   \n",
      "Segment  19/ 26 2621-2772.   \n",
      "Segment  20/ 26 2772-2923.   \n",
      "Segment  21/ 26 2923-3074.   \n",
      "Segment  22/ 26 3074-3196.   \n",
      "Segment  23/ 26 3196-3343.   \n",
      "Segment  24/ 26 3343-3494.   \n",
      "Segment  25/ 26 3494-3635.   \n",
      "Segment  26/ 26 3635-3712.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  89\n",
      "Segment   2/ 26   89- 240.   \n",
      "Segment   3/ 26  240- 391.   \n",
      "Segment   4/ 26  391- 542.   \n",
      "Segment   5/ 26  542- 693.   \n",
      "Segment   6/ 26  693- 789.   \n",
      "Segment   7/ 26  789- 900.   \n",
      "Segment   8/ 26  900-1018.   \n",
      "Segment   9/ 26 1018-1176.   \n",
      "Segment  10/ 26 1176-1327.   \n",
      "Segment  11/ 26 1327-1478.   \n",
      "Segment  12/ 26 1478-1629.   \n",
      "Segment  13/ 26 1629-1780.   \n",
      "Segment  14/ 26 1780-1931.   \n",
      "Segment  15/ 26 1931-2082.   \n",
      "Segment  16/ 26 2082-2233.   \n",
      "Segment  17/ 26 2233-2384.   \n",
      "Segment  18/ 26 2384-2535.   \n",
      "Segment  19/ 26 2535-2686.   \n",
      "Segment  20/ 26 2686-2837.   \n",
      "Segment  21/ 26 2837-2988.   \n",
      "Segment  22/ 26 2988-3139.   \n",
      "Segment  23/ 26 3139-3290.   \n",
      "Segment  24/ 26 3290-3441.   \n",
      "Segment  25/ 26 3441-3549.   \n",
      "Segment  26/ 26 3549-3592.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3200.   \n",
      "Segment  23/ 26 3200-3311.   \n",
      "Segment  24/ 26 3311-3462.   \n",
      "Segment  25/ 26 3462-3603.   \n",
      "Segment  26/ 26 3603-3679.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 28    1-  87\n",
      "Segment   2/ 28   87- 238.   \n",
      "Segment   3/ 28  238- 389.   \n",
      "Segment   4/ 28  389- 540.   \n",
      "Segment   5/ 28  540- 691.   \n",
      "Segment   6/ 28  691- 842.   \n",
      "Segment   7/ 28  842- 993.   \n",
      "Segment   8/ 28  993-1144.   \n",
      "Segment   9/ 28 1144-1295.   \n",
      "Segment  10/ 28 1295-1446.   \n",
      "Segment  11/ 28 1446-1597.   \n",
      "Segment  12/ 28 1597-1748.   \n",
      "Segment  13/ 28 1748-1899.   \n",
      "Segment  14/ 28 1899-2050.   \n",
      "Segment  15/ 28 2050-2201.   \n",
      "Segment  16/ 28 2201-2352.   \n",
      "Segment  17/ 28 2352-2503.   \n",
      "Segment  18/ 28 2503-2654.   \n",
      "Segment  19/ 28 2654-2805.   \n",
      "Segment  20/ 28 2805-2956.   \n",
      "Segment  21/ 28 2956-3107.   \n",
      "Segment  22/ 28 3107-3200.   \n",
      "Segment  23/ 28 3200-3247.   \n",
      "Segment  24/ 28 3247-3270.   \n",
      "Segment  25/ 28 3270-3357.   \n",
      "Segment  26/ 28 3357-3508.   \n",
      "Segment  27/ 28 3508-3625.   \n",
      "Segment  28/ 28 3625-3676.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3684.   \n",
      "Segment  26/ 26 3684-3742.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 993.   \n",
      "Segment   8/ 27  993-1144.   \n",
      "Segment   9/ 27 1144-1295.   \n",
      "Segment  10/ 27 1295-1446.   \n",
      "Segment  11/ 27 1446-1597.   \n",
      "Segment  12/ 27 1597-1748.   \n",
      "Segment  13/ 27 1748-1899.   \n",
      "Segment  14/ 27 1899-2050.   \n",
      "Segment  15/ 27 2050-2201.   \n",
      "Segment  16/ 27 2201-2352.   \n",
      "Segment  17/ 27 2352-2503.   \n",
      "Segment  18/ 27 2503-2654.   \n",
      "Segment  19/ 27 2654-2805.   \n",
      "Segment  20/ 27 2805-2956.   \n",
      "Segment  21/ 27 2956-3107.   \n",
      "Segment  22/ 27 3107-3258.   \n",
      "Segment  23/ 27 3258-3409.   \n",
      "Segment  24/ 27 3409-3560.   \n",
      "Segment  25/ 27 3560-3711.   \n",
      "Segment  26/ 27 3711-3788.   \n",
      "Segment  27/ 27 3788-3799.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 993.   \n",
      "Segment   8/ 27  993-1144.   \n",
      "Segment   9/ 27 1144-1295.   \n",
      "Segment  10/ 27 1295-1446.   \n",
      "Segment  11/ 27 1446-1597.   \n",
      "Segment  12/ 27 1597-1748.   \n",
      "Segment  13/ 27 1748-1899.   \n",
      "Segment  14/ 27 1899-2050.   \n",
      "Segment  15/ 27 2050-2201.   \n",
      "Segment  16/ 27 2201-2352.   \n",
      "Segment  17/ 27 2352-2503.   \n",
      "Segment  18/ 27 2503-2654.   \n",
      "Segment  19/ 27 2654-2805.   \n",
      "Segment  20/ 27 2805-2956.   \n",
      "Segment  21/ 27 2956-3107.   \n",
      "Segment  22/ 27 3107-3258.   \n",
      "Segment  23/ 27 3258-3409.   \n",
      "Segment  24/ 27 3409-3560.   \n",
      "Segment  25/ 27 3560-3711.   \n",
      "Segment  26/ 27 3711-3809.   \n",
      "Segment  27/ 27 3809-3841.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3700.   \n",
      "Segment  26/ 26 3700-3775.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3664.   \n",
      "Segment  26/ 26 3664-3703.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3684.   \n",
      "Segment  26/ 26 3684-3742.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3668.   \n",
      "Segment  26/ 26 3668-3745.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3666.   \n",
      "Segment  26/ 26 3666-3706.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3648.   \n",
      "Segment  26/ 26 3648-3670.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3648.   \n",
      "Segment  26/ 26 3648-3670.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3645.   \n",
      "Segment  26/ 26 3645-3664.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3645.   \n",
      "Segment  26/ 26 3645-3664.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 991.   \n",
      "Segment   8/ 27  991-1181.   \n",
      "Segment   9/ 27 1181-1332.   \n",
      "Segment  10/ 27 1332-1483.   \n",
      "Segment  11/ 27 1483-1634.   \n",
      "Segment  12/ 27 1634-1785.   \n",
      "Segment  13/ 27 1785-1876.   \n",
      "Segment  14/ 27 1876-2102.   \n",
      "Segment  15/ 27 2102-2253.   \n",
      "Segment  16/ 27 2253-2348.   \n",
      "Segment  17/ 27 2348-2485.   \n",
      "Segment  18/ 27 2485-2636.   \n",
      "Segment  19/ 27 2636-2787.   \n",
      "Segment  20/ 27 2787-2938.   \n",
      "Segment  21/ 27 2938-3048.   \n",
      "Segment  22/ 27 3048-3173.   \n",
      "Segment  23/ 27 3173-3324.   \n",
      "Segment  24/ 27 3324-3475.   \n",
      "Segment  25/ 27 3475-3626.   \n",
      "Segment  26/ 27 3626-3703.   \n",
      "Segment  27/ 27 3703-3715.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3657.   \n",
      "Segment  26/ 26 3657-3688.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3657.   \n",
      "Segment  26/ 26 3657-3688.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3652.   \n",
      "Segment  26/ 26 3652-3679.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 23    1-  87\n",
      "Segment   2/ 23   87- 238.   \n",
      "Segment   3/ 23  238- 389.   \n",
      "Segment   4/ 23  389- 540.   \n",
      "Segment   5/ 23  540- 691.   \n",
      "Segment   6/ 23  691- 842.   \n",
      "Segment   7/ 23  842- 956.   \n",
      "Segment   8/ 23  956-1187.   \n",
      "Segment   9/ 23 1187-1338.   \n",
      "Segment  10/ 23 1338-1489.   \n",
      "Segment  11/ 23 1489-1640.   \n",
      "Segment  12/ 23 1640-1791.   \n",
      "Segment  13/ 23 1791-1942.   \n",
      "Segment  14/ 23 1942-2093.   \n",
      "Segment  15/ 23 2093-2244.   \n",
      "Segment  16/ 23 2244-2395.   \n",
      "Segment  17/ 23 2395-2546.   \n",
      "Segment  18/ 23 2546-2697.   \n",
      "Segment  19/ 23 2697-2848.   \n",
      "Segment  20/ 23 2848-2999.   \n",
      "Segment  21/ 23 2999-3150.   \n",
      "Segment  22/ 23 3150-3299.   \n",
      "Segment  23/ 23 3299-3712.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 24    1-  87\n",
      "Segment   2/ 24   87- 238.   \n",
      "Segment   3/ 24  238- 389.   \n",
      "Segment   4/ 24  389- 540.   \n",
      "Segment   5/ 24  540- 691.   \n",
      "Segment   6/ 24  691- 842.   \n",
      "Segment   7/ 24  842- 972.   \n",
      "Segment   8/ 24  972-1145.   \n",
      "Segment   9/ 24 1145-1296.   \n",
      "Segment  10/ 24 1296-1447.   \n",
      "Segment  11/ 24 1447-1598.   \n",
      "Segment  12/ 24 1598-1749.   \n",
      "Segment  13/ 24 1749-1900.   \n",
      "Segment  14/ 24 1900-2051.   \n",
      "Segment  15/ 24 2051-2202.   \n",
      "Segment  16/ 24 2202-2353.   \n",
      "Segment  17/ 24 2353-2504.   \n",
      "Segment  18/ 24 2504-2655.   \n",
      "Segment  19/ 24 2655-2806.   \n",
      "Segment  20/ 24 2806-2957.   \n",
      "Segment  21/ 24 2957-3108.   \n",
      "Segment  22/ 24 3108-3259.   \n",
      "Segment  23/ 24 3259-3343.   \n",
      "Segment  24/ 24 3343-3688.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3658.   \n",
      "Segment  26/ 26 3658-3691.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 23    1-  87\n",
      "Segment   2/ 23   87- 238.   \n",
      "Segment   3/ 23  238- 389.   \n",
      "Segment   4/ 23  389- 540.   \n",
      "Segment   5/ 23  540- 691.   \n",
      "Segment   6/ 23  691- 842.   \n",
      "Segment   7/ 23  842- 986.   \n",
      "Segment   8/ 23  986-1224.   \n",
      "Segment   9/ 23 1224-1375.   \n",
      "Segment  10/ 23 1375-1526.   \n",
      "Segment  11/ 23 1526-1677.   \n",
      "Segment  12/ 23 1677-1828.   \n",
      "Segment  13/ 23 1828-1979.   \n",
      "Segment  14/ 23 1979-2130.   \n",
      "Segment  15/ 23 2130-2281.   \n",
      "Segment  16/ 23 2281-2432.   \n",
      "Segment  17/ 23 2432-2583.   \n",
      "Segment  18/ 23 2583-2734.   \n",
      "Segment  19/ 23 2734-2885.   \n",
      "Segment  20/ 23 2885-3036.   \n",
      "Segment  21/ 23 3036-3187.   \n",
      "Segment  22/ 23 3187-3333.   \n",
      "Segment  23/ 23 3333-3721.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 24    1-  87\n",
      "Segment   2/ 24   87- 238.   \n",
      "Segment   3/ 24  238- 389.   \n",
      "Segment   4/ 24  389- 540.   \n",
      "Segment   5/ 24  540- 691.   \n",
      "Segment   6/ 24  691- 842.   \n",
      "Segment   7/ 24  842- 985.   \n",
      "Segment   8/ 24  985-1220.   \n",
      "Segment   9/ 24 1220-1371.   \n",
      "Segment  10/ 24 1371-1522.   \n",
      "Segment  11/ 24 1522-1673.   \n",
      "Segment  12/ 24 1673-1824.   \n",
      "Segment  13/ 24 1824-1975.   \n",
      "Segment  14/ 24 1975-2126.   \n",
      "Segment  15/ 24 2126-2277.   \n",
      "Segment  16/ 24 2277-2428.   \n",
      "Segment  17/ 24 2428-2579.   \n",
      "Segment  18/ 24 2579-2730.   \n",
      "Segment  19/ 24 2730-2881.   \n",
      "Segment  20/ 24 2881-3032.   \n",
      "Segment  21/ 24 3032-3183.   \n",
      "Segment  22/ 24 3183-3334.   \n",
      "Segment  23/ 24 3334-3414.   \n",
      "Segment  24/ 24 3414-3736.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3337.   \n",
      "Segment  24/ 25 3337-3690.   \n",
      "Segment  25/ 25 3690-3792.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3553.   \n",
      "Segment  25/ 25 3553-3631.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3338.   \n",
      "Segment  24/ 25 3338-3588.   \n",
      "Segment  25/ 25 3588-3665.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 24    1-  87\n",
      "Segment   2/ 24   87- 238.   \n",
      "Segment   3/ 24  238- 389.   \n",
      "Segment   4/ 24  389- 540.   \n",
      "Segment   5/ 24  540- 691.   \n",
      "Segment   6/ 24  691- 842.   \n",
      "Segment   7/ 24  842- 974.   \n",
      "Segment   8/ 24  974-1133.   \n",
      "Segment   9/ 24 1133-1284.   \n",
      "Segment  10/ 24 1284-1435.   \n",
      "Segment  11/ 24 1435-1586.   \n",
      "Segment  12/ 24 1586-1737.   \n",
      "Segment  13/ 24 1737-1888.   \n",
      "Segment  14/ 24 1888-2039.   \n",
      "Segment  15/ 24 2039-2190.   \n",
      "Segment  16/ 24 2190-2341.   \n",
      "Segment  17/ 24 2341-2492.   \n",
      "Segment  18/ 24 2492-2643.   \n",
      "Segment  19/ 24 2643-2794.   \n",
      "Segment  20/ 24 2794-2945.   \n",
      "Segment  21/ 24 2945-3096.   \n",
      "Segment  22/ 24 3096-3247.   \n",
      "Segment  23/ 24 3247-3334.   \n",
      "Segment  24/ 24 3334-3682.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 978.   \n",
      "Segment   8/ 26  978-1136.   \n",
      "Segment   9/ 26 1136-1287.   \n",
      "Segment  10/ 26 1287-1438.   \n",
      "Segment  11/ 26 1438-1589.   \n",
      "Segment  12/ 26 1589-1740.   \n",
      "Segment  13/ 26 1740-1891.   \n",
      "Segment  14/ 26 1891-2042.   \n",
      "Segment  15/ 26 2042-2193.   \n",
      "Segment  16/ 26 2193-2344.   \n",
      "Segment  17/ 26 2344-2495.   \n",
      "Segment  18/ 26 2495-2646.   \n",
      "Segment  19/ 26 2646-2797.   \n",
      "Segment  20/ 26 2797-2948.   \n",
      "Segment  21/ 26 2948-3099.   \n",
      "Segment  22/ 26 3099-3250.   \n",
      "Segment  23/ 26 3250-3401.   \n",
      "Segment  24/ 26 3401-3552.   \n",
      "Segment  25/ 26 3552-3684.   \n",
      "Segment  26/ 26 3684-3751.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 24    1-  87\n",
      "Segment   2/ 24   87- 238.   \n",
      "Segment   3/ 24  238- 389.   \n",
      "Segment   4/ 24  389- 540.   \n",
      "Segment   5/ 24  540- 691.   \n",
      "Segment   6/ 24  691- 842.   \n",
      "Segment   7/ 24  842- 993.   \n",
      "Segment   8/ 24  993-1144.   \n",
      "Segment   9/ 24 1144-1295.   \n",
      "Segment  10/ 24 1295-1446.   \n",
      "Segment  11/ 24 1446-1597.   \n",
      "Segment  12/ 24 1597-1748.   \n",
      "Segment  13/ 24 1748-1899.   \n",
      "Segment  14/ 24 1899-2050.   \n",
      "Segment  15/ 24 2050-2201.   \n",
      "Segment  16/ 24 2201-2352.   \n",
      "Segment  17/ 24 2352-2503.   \n",
      "Segment  18/ 24 2503-2654.   \n",
      "Segment  19/ 24 2654-2805.   \n",
      "Segment  20/ 24 2805-2956.   \n",
      "Segment  21/ 24 2956-3107.   \n",
      "Segment  22/ 24 3107-3258.   \n",
      "Segment  23/ 24 3258-3353.   \n",
      "Segment  24/ 24 3353-3685.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 22    1-  87\n",
      "Segment   2/ 22   87- 238.   \n",
      "Segment   3/ 22  238- 389.   \n",
      "Segment   4/ 22  389- 540.   \n",
      "Segment   5/ 22  540- 691.   \n",
      "Segment   6/ 22  691- 842.   \n",
      "Segment   7/ 22  842- 993.   \n",
      "Segment   8/ 22  993-1144.   \n",
      "Segment   9/ 22 1144-1295.   \n",
      "Segment  10/ 22 1295-1446.   \n",
      "Segment  11/ 22 1446-1597.   \n",
      "Segment  12/ 22 1597-1748.   \n",
      "Segment  13/ 22 1748-1899.   \n",
      "Segment  14/ 22 1899-2050.   \n",
      "Segment  15/ 22 2050-2201.   \n",
      "Segment  16/ 22 2201-2352.   \n",
      "Segment  17/ 22 2352-2503.   \n",
      "Segment  18/ 22 2503-2654.   \n",
      "Segment  19/ 22 2654-2805.   \n",
      "Segment  20/ 22 2805-2956.   \n",
      "Segment  21/ 22 2956-3055.   \n",
      "Segment  22/ 22 3055-3349.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 22    1-  87\n",
      "Segment   2/ 22   87- 238.   \n",
      "Segment   3/ 22  238- 389.   \n",
      "Segment   4/ 22  389- 540.   \n",
      "Segment   5/ 22  540- 691.   \n",
      "Segment   6/ 22  691- 842.   \n",
      "Segment   7/ 22  842- 993.   \n",
      "Segment   8/ 22  993-1144.   \n",
      "Segment   9/ 22 1144-1295.   \n",
      "Segment  10/ 22 1295-1446.   \n",
      "Segment  11/ 22 1446-1597.   \n",
      "Segment  12/ 22 1597-1748.   \n",
      "Segment  13/ 22 1748-1899.   \n",
      "Segment  14/ 22 1899-2050.   \n",
      "Segment  15/ 22 2050-2201.   \n",
      "Segment  16/ 22 2201-2352.   \n",
      "Segment  17/ 22 2352-2503.   \n",
      "Segment  18/ 22 2503-2654.   \n",
      "Segment  19/ 22 2654-2805.   \n",
      "Segment  20/ 22 2805-2956.   \n",
      "Segment  21/ 22 2956-3060.   \n",
      "Segment  22/ 22 3060-3358.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3691.   \n",
      "Segment  26/ 26 3691-3757.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3688.   \n",
      "Segment  26/ 26 3688-3751.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 978.   \n",
      "Segment   8/ 26  978-1136.   \n",
      "Segment   9/ 26 1136-1287.   \n",
      "Segment  10/ 26 1287-1438.   \n",
      "Segment  11/ 26 1438-1589.   \n",
      "Segment  12/ 26 1589-1740.   \n",
      "Segment  13/ 26 1740-1891.   \n",
      "Segment  14/ 26 1891-2042.   \n",
      "Segment  15/ 26 2042-2193.   \n",
      "Segment  16/ 26 2193-2344.   \n",
      "Segment  17/ 26 2344-2495.   \n",
      "Segment  18/ 26 2495-2646.   \n",
      "Segment  19/ 26 2646-2797.   \n",
      "Segment  20/ 26 2797-2948.   \n",
      "Segment  21/ 26 2948-3099.   \n",
      "Segment  22/ 26 3099-3250.   \n",
      "Segment  23/ 26 3250-3401.   \n",
      "Segment  24/ 26 3401-3552.   \n",
      "Segment  25/ 26 3552-3689.   \n",
      "Segment  26/ 26 3689-3760.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 978.   \n",
      "Segment   8/ 27  978-1136.   \n",
      "Segment   9/ 27 1136-1287.   \n",
      "Segment  10/ 27 1287-1438.   \n",
      "Segment  11/ 27 1438-1589.   \n",
      "Segment  12/ 27 1589-1740.   \n",
      "Segment  13/ 27 1740-1853.   \n",
      "Segment  14/ 27 1853-2102.   \n",
      "Segment  15/ 27 2102-2253.   \n",
      "Segment  16/ 27 2253-2345.   \n",
      "Segment  17/ 27 2345-2476.   \n",
      "Segment  18/ 27 2476-2627.   \n",
      "Segment  19/ 27 2627-2778.   \n",
      "Segment  20/ 27 2778-2929.   \n",
      "Segment  21/ 27 2929-3043.   \n",
      "Segment  22/ 27 3043-3173.   \n",
      "Segment  23/ 27 3173-3324.   \n",
      "Segment  24/ 27 3324-3475.   \n",
      "Segment  25/ 27 3475-3626.   \n",
      "Segment  26/ 27 3626-3718.   \n",
      "Segment  27/ 27 3718-3745.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3685.   \n",
      "Segment  26/ 26 3685-3745.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3685.   \n",
      "Segment  26/ 26 3685-3745.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1858.   \n",
      "Segment  14/ 26 1858-2091.   \n",
      "Segment  15/ 26 2091-2242.   \n",
      "Segment  16/ 26 2242-2350.   \n",
      "Segment  17/ 26 2350-2488.   \n",
      "Segment  18/ 26 2488-2639.   \n",
      "Segment  19/ 26 2639-2790.   \n",
      "Segment  20/ 26 2790-2941.   \n",
      "Segment  21/ 26 2941-3048.   \n",
      "Segment  22/ 26 3048-3225.   \n",
      "Segment  23/ 26 3225-3376.   \n",
      "Segment  24/ 26 3376-3527.   \n",
      "Segment  25/ 26 3527-3625.   \n",
      "Segment  26/ 26 3625-3658.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1858.   \n",
      "Segment  14/ 25 1858-2091.   \n",
      "Segment  15/ 25 2091-2242.   \n",
      "Segment  16/ 25 2242-2393.   \n",
      "Segment  17/ 25 2393-2544.   \n",
      "Segment  18/ 25 2544-2695.   \n",
      "Segment  19/ 25 2695-2846.   \n",
      "Segment  20/ 25 2846-2997.   \n",
      "Segment  21/ 25 2997-3148.   \n",
      "Segment  22/ 25 3148-3299.   \n",
      "Segment  23/ 25 3299-3450.   \n",
      "Segment  24/ 25 3450-3543.   \n",
      "Segment  25/ 25 3543-3571.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1858.   \n",
      "Segment  14/ 25 1858-2091.   \n",
      "Segment  15/ 25 2091-2242.   \n",
      "Segment  16/ 25 2242-2393.   \n",
      "Segment  17/ 25 2393-2544.   \n",
      "Segment  18/ 25 2544-2695.   \n",
      "Segment  19/ 25 2695-2846.   \n",
      "Segment  20/ 25 2846-2997.   \n",
      "Segment  21/ 25 2997-3148.   \n",
      "Segment  22/ 25 3148-3299.   \n",
      "Segment  23/ 25 3299-3450.   \n",
      "Segment  24/ 25 3450-3570.   \n",
      "Segment  25/ 25 3570-3625.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1858.   \n",
      "Segment  14/ 25 1858-2091.   \n",
      "Segment  15/ 25 2091-2242.   \n",
      "Segment  16/ 25 2242-2393.   \n",
      "Segment  17/ 25 2393-2544.   \n",
      "Segment  18/ 25 2544-2695.   \n",
      "Segment  19/ 25 2695-2846.   \n",
      "Segment  20/ 25 2846-2997.   \n",
      "Segment  21/ 25 2997-3148.   \n",
      "Segment  22/ 25 3148-3299.   \n",
      "Segment  23/ 25 3299-3450.   \n",
      "Segment  24/ 25 3450-3570.   \n",
      "Segment  25/ 25 3570-3625.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 980.   \n",
      "Segment   8/ 26  980-1142.   \n",
      "Segment   9/ 26 1142-1293.   \n",
      "Segment  10/ 26 1293-1444.   \n",
      "Segment  11/ 26 1444-1595.   \n",
      "Segment  12/ 26 1595-1746.   \n",
      "Segment  13/ 26 1746-1857.   \n",
      "Segment  14/ 26 1857-2091.   \n",
      "Segment  15/ 26 2091-2242.   \n",
      "Segment  16/ 26 2242-2340.   \n",
      "Segment  17/ 26 2340-2468.   \n",
      "Segment  18/ 26 2468-2619.   \n",
      "Segment  19/ 26 2619-2770.   \n",
      "Segment  20/ 26 2770-2921.   \n",
      "Segment  21/ 26 2921-3032.   \n",
      "Segment  22/ 26 3032-3168.   \n",
      "Segment  23/ 26 3168-3319.   \n",
      "Segment  24/ 26 3319-3470.   \n",
      "Segment  25/ 26 3470-3583.   \n",
      "Segment  26/ 26 3583-3631.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1858.   \n",
      "Segment  14/ 25 1858-2091.   \n",
      "Segment  15/ 25 2091-2242.   \n",
      "Segment  16/ 25 2242-2393.   \n",
      "Segment  17/ 25 2393-2544.   \n",
      "Segment  18/ 25 2544-2695.   \n",
      "Segment  19/ 25 2695-2846.   \n",
      "Segment  20/ 25 2846-2993.   \n",
      "Segment  21/ 25 2993-3168.   \n",
      "Segment  22/ 25 3168-3319.   \n",
      "Segment  23/ 25 3319-3470.   \n",
      "Segment  24/ 25 3470-3582.   \n",
      "Segment  25/ 25 3582-3628.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1858.   \n",
      "Segment  14/ 25 1858-2091.   \n",
      "Segment  15/ 25 2091-2242.   \n",
      "Segment  16/ 25 2242-2393.   \n",
      "Segment  17/ 25 2393-2544.   \n",
      "Segment  18/ 25 2544-2695.   \n",
      "Segment  19/ 25 2695-2846.   \n",
      "Segment  20/ 25 2846-2992.   \n",
      "Segment  21/ 25 2992-3165.   \n",
      "Segment  22/ 25 3165-3285.   \n",
      "Segment  23/ 25 3285-3439.   \n",
      "Segment  24/ 25 3439-3578.   \n",
      "Segment  25/ 25 3578-3652.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 23    1-  87\n",
      "Segment   2/ 23   87- 238.   \n",
      "Segment   3/ 23  238- 389.   \n",
      "Segment   4/ 23  389- 540.   \n",
      "Segment   5/ 23  540- 691.   \n",
      "Segment   6/ 23  691- 842.   \n",
      "Segment   7/ 23  842- 993.   \n",
      "Segment   8/ 23  993-1144.   \n",
      "Segment   9/ 23 1144-1295.   \n",
      "Segment  10/ 23 1295-1446.   \n",
      "Segment  11/ 23 1446-1597.   \n",
      "Segment  12/ 23 1597-1748.   \n",
      "Segment  13/ 23 1748-1899.   \n",
      "Segment  14/ 23 1899-2050.   \n",
      "Segment  15/ 23 2050-2201.   \n",
      "Segment  16/ 23 2201-2352.   \n",
      "Segment  17/ 23 2352-2503.   \n",
      "Segment  18/ 23 2503-2654.   \n",
      "Segment  19/ 23 2654-2805.   \n",
      "Segment  20/ 23 2805-2956.   \n",
      "Segment  21/ 23 2956-3107.   \n",
      "Segment  22/ 23 3107-3251.   \n",
      "Segment  23/ 23 3251-3634.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 974.   \n",
      "Segment   8/ 26  974-1124.   \n",
      "Segment   9/ 26 1124-1275.   \n",
      "Segment  10/ 26 1275-1426.   \n",
      "Segment  11/ 26 1426-1577.   \n",
      "Segment  12/ 26 1577-1728.   \n",
      "Segment  13/ 26 1728-1879.   \n",
      "Segment  14/ 26 1879-2030.   \n",
      "Segment  15/ 26 2030-2181.   \n",
      "Segment  16/ 26 2181-2318.   \n",
      "Segment  17/ 26 2318-2509.   \n",
      "Segment  18/ 26 2509-2660.   \n",
      "Segment  19/ 26 2660-2811.   \n",
      "Segment  20/ 26 2811-2962.   \n",
      "Segment  21/ 26 2962-3113.   \n",
      "Segment  22/ 26 3113-3264.   \n",
      "Segment  23/ 26 3264-3415.   \n",
      "Segment  24/ 26 3415-3566.   \n",
      "Segment  25/ 26 3566-3705.   \n",
      "Segment  26/ 26 3705-3778.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3525.   \n",
      "Segment  25/ 25 3525-3649.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3651.   \n",
      "Segment  26/ 26 3651-3682.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3663.   \n",
      "Segment  26/ 26 3663-3706.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 981.   \n",
      "Segment   8/ 27  981-1136.   \n",
      "Segment   9/ 27 1136-1287.   \n",
      "Segment  10/ 27 1287-1438.   \n",
      "Segment  11/ 27 1438-1589.   \n",
      "Segment  12/ 27 1589-1740.   \n",
      "Segment  13/ 27 1740-1891.   \n",
      "Segment  14/ 27 1891-2042.   \n",
      "Segment  15/ 27 2042-2193.   \n",
      "Segment  16/ 27 2193-2344.   \n",
      "Segment  17/ 27 2344-2495.   \n",
      "Segment  18/ 27 2495-2646.   \n",
      "Segment  19/ 27 2646-2797.   \n",
      "Segment  20/ 27 2797-2948.   \n",
      "Segment  21/ 27 2948-3099.   \n",
      "Segment  22/ 27 3099-3250.   \n",
      "Segment  23/ 27 3250-3401.   \n",
      "Segment  24/ 27 3401-3496.   \n",
      "Segment  25/ 27 3496-3615.   \n",
      "Segment  26/ 27 3615-3719.   \n",
      "Segment  27/ 27 3719-3772.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2352.   \n",
      "Segment  17/ 26 2352-2503.   \n",
      "Segment  18/ 26 2503-2654.   \n",
      "Segment  19/ 26 2654-2805.   \n",
      "Segment  20/ 26 2805-2956.   \n",
      "Segment  21/ 26 2956-3107.   \n",
      "Segment  22/ 26 3107-3258.   \n",
      "Segment  23/ 26 3258-3409.   \n",
      "Segment  24/ 26 3409-3560.   \n",
      "Segment  25/ 26 3560-3691.   \n",
      "Segment  26/ 26 3691-3757.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 27    1-  87\n",
      "Segment   2/ 27   87- 238.   \n",
      "Segment   3/ 27  238- 389.   \n",
      "Segment   4/ 27  389- 540.   \n",
      "Segment   5/ 27  540- 691.   \n",
      "Segment   6/ 27  691- 842.   \n",
      "Segment   7/ 27  842- 978.   \n",
      "Segment   8/ 27  978-1138.   \n",
      "Segment   9/ 27 1138-1289.   \n",
      "Segment  10/ 27 1289-1440.   \n",
      "Segment  11/ 27 1440-1591.   \n",
      "Segment  12/ 27 1591-1742.   \n",
      "Segment  13/ 27 1742-1893.   \n",
      "Segment  14/ 27 1893-2044.   \n",
      "Segment  15/ 27 2044-2195.   \n",
      "Segment  16/ 27 2195-2326.   \n",
      "Segment  17/ 27 2326-2491.   \n",
      "Segment  18/ 27 2491-2642.   \n",
      "Segment  19/ 27 2642-2793.   \n",
      "Segment  20/ 27 2793-2944.   \n",
      "Segment  21/ 27 2944-3095.   \n",
      "Segment  22/ 27 3095-3246.   \n",
      "Segment  23/ 27 3246-3397.   \n",
      "Segment  24/ 27 3397-3515.   \n",
      "Segment  25/ 27 3515-3654.   \n",
      "Segment  26/ 27 3654-3740.   \n",
      "Segment  27/ 27 3740-3760.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 981.   \n",
      "Segment   8/ 25  981-1178.   \n",
      "Segment   9/ 25 1178-1329.   \n",
      "Segment  10/ 25 1329-1480.   \n",
      "Segment  11/ 25 1480-1631.   \n",
      "Segment  12/ 25 1631-1782.   \n",
      "Segment  13/ 25 1782-1933.   \n",
      "Segment  14/ 25 1933-2084.   \n",
      "Segment  15/ 25 2084-2235.   \n",
      "Segment  16/ 25 2235-2386.   \n",
      "Segment  17/ 25 2386-2537.   \n",
      "Segment  18/ 25 2537-2688.   \n",
      "Segment  19/ 25 2688-2839.   \n",
      "Segment  20/ 25 2839-2990.   \n",
      "Segment  21/ 25 2990-3134.   \n",
      "Segment  22/ 25 3134-3405.   \n",
      "Segment  23/ 25 3405-3556.   \n",
      "Segment  24/ 25 3556-3665.   \n",
      "Segment  25/ 25 3665-3708.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3536.   \n",
      "Segment  25/ 25 3536-3598.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3545.   \n",
      "Segment  25/ 25 3545-3616.   \n",
      "done 001-001-1  identical.   \n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3532.   \n",
      "Segment  25/ 25 3532-3589.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3532.   \n",
      "Segment  25/ 25 3532-3589.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3530.   \n",
      "Segment  25/ 25 3530-3586.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3539.   \n",
      "Segment  25/ 25 3539-3604.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 25    1-  87\n",
      "Segment   2/ 25   87- 238.   \n",
      "Segment   3/ 25  238- 389.   \n",
      "Segment   4/ 25  389- 540.   \n",
      "Segment   5/ 25  540- 691.   \n",
      "Segment   6/ 25  691- 842.   \n",
      "Segment   7/ 25  842- 993.   \n",
      "Segment   8/ 25  993-1144.   \n",
      "Segment   9/ 25 1144-1295.   \n",
      "Segment  10/ 25 1295-1446.   \n",
      "Segment  11/ 25 1446-1597.   \n",
      "Segment  12/ 25 1597-1748.   \n",
      "Segment  13/ 25 1748-1899.   \n",
      "Segment  14/ 25 1899-2050.   \n",
      "Segment  15/ 25 2050-2201.   \n",
      "Segment  16/ 25 2201-2352.   \n",
      "Segment  17/ 25 2352-2503.   \n",
      "Segment  18/ 25 2503-2654.   \n",
      "Segment  19/ 25 2654-2805.   \n",
      "Segment  20/ 25 2805-2956.   \n",
      "Segment  21/ 25 2956-3107.   \n",
      "Segment  22/ 25 3107-3258.   \n",
      "Segment  23/ 25 3258-3409.   \n",
      "Segment  24/ 25 3409-3530.   \n",
      "Segment  25/ 25 3530-3586.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2292.   \n",
      "Segment  17/ 26 2292-2419.   \n",
      "Segment  18/ 26 2419-2570.   \n",
      "Segment  19/ 26 2570-2721.   \n",
      "Segment  20/ 26 2721-2872.   \n",
      "Segment  21/ 26 2872-3023.   \n",
      "Segment  22/ 26 3023-3174.   \n",
      "Segment  23/ 26 3174-3325.   \n",
      "Segment  24/ 26 3325-3476.   \n",
      "Segment  25/ 26 3476-3558.   \n",
      "Segment  26/ 26 3558-3574.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "distout=h\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "dndpre (nuc) Version 7.525\n",
      "alg=X, model=DNA200 (2), 1.53 (4.59), 0.37 (1.11), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/ 26    1-  87\n",
      "Segment   2/ 26   87- 238.   \n",
      "Segment   3/ 26  238- 389.   \n",
      "Segment   4/ 26  389- 540.   \n",
      "Segment   5/ 26  540- 691.   \n",
      "Segment   6/ 26  691- 842.   \n",
      "Segment   7/ 26  842- 993.   \n",
      "Segment   8/ 26  993-1144.   \n",
      "Segment   9/ 26 1144-1295.   \n",
      "Segment  10/ 26 1295-1446.   \n",
      "Segment  11/ 26 1446-1597.   \n",
      "Segment  12/ 26 1597-1748.   \n",
      "Segment  13/ 26 1748-1899.   \n",
      "Segment  14/ 26 1899-2050.   \n",
      "Segment  15/ 26 2050-2201.   \n",
      "Segment  16/ 26 2201-2289.   \n",
      "Segment  17/ 26 2289-2401.   \n",
      "Segment  18/ 26 2401-2552.   \n",
      "Segment  19/ 26 2552-2703.   \n",
      "Segment  20/ 26 2703-2854.   \n",
      "Segment  21/ 26 2854-3005.   \n",
      "Segment  22/ 26 3005-3156.   \n",
      "Segment  23/ 26 3156-3307.   \n",
      "Segment  24/ 26 3307-3458.   \n",
      "Segment  25/ 26 3458-3565.   \n",
      "Segment  26/ 26 3565-3607.   \n",
      "done 001-001-1  rejected.\n",
      "dvtditr (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-i (Standard)\n",
      " Iterative refinement method (max. 2 iterations)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "for key in busco_gene_dic:\n",
    "    if key in tim_gene_dic:\n",
    "        seq1 = busco_gene_dic[key]\n",
    "        seq2 = tim_gene_dic[key]\n",
    "        # # print(seq1,seq2)\n",
    "        # seq1 = \"GTTTTTAGATATAGCA\"\n",
    "        # seq2 = \"GTTAGATATAGCA\"\n",
    "        \n",
    "        output = f\">busco_{key}\\n{seq1}\\n>tim_{key}\\n{seq2}\"\n",
    "        \n",
    "        with open(f\"/mnt/g/My Drive/Circadian Rhythm Genes Project/22.Busco_related/5.Checking_busco_aligned/1.Timeless_1422at7088/{key}.txt\",'w') as out_file:\n",
    "            out_file.write(output)\n",
    "        subprocess.run(f\"cd '/mnt/g/My Drive/Circadian Rhythm Genes Project/22.Busco_related/5.Checking_busco_aligned/1.Timeless_1422at7088/'\\n mafft --auto {key}.txt > {key}_aligned.txt\", shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ac1b1a1a-a7cc-4cc4-ac46-f01560bbb4df",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "location = \"/mnt/g/My Drive/Circadian Rhythm Genes Project/22.Busco_related/5.Checking_busco_aligned/1.Timeless_1422at7088/\"\n",
    "list_of_files  = os.listdir(location)\n",
    "output = ''\n",
    "for file_name in list_of_files:\n",
    "    if file_name.endswith(\"_aligned.txt\"):\n",
    "        with open(f\"{location}/{file_name}\", 'r') as aligned_file:\n",
    "            output += \"\".join(aligned_file.readlines())\n",
    "with open(\"/mnt/g/My Drive/Circadian Rhythm Genes Project/22.Busco_related/5.Checking_busco_aligned/1.Timeless_1422at7088.txt\", 'w')             as out_file:\n",
    "    out_file.write(output)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
