{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ba9da5af",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Anthocharis_cardamines\n",
      "Aporia_crataegi\n",
      "Argynnis_bischoffii_washingtonia\n",
      "Aricia_agestis\n",
      "Aricia_artaxerxes\n",
      "Baronia_brevicornis\n",
      "Battus_philenor\n",
      "Bicyclus_anynana\n",
      "Boloria_euphrosyne\n",
      "Boloria_selene\n",
      "Brenthis_daphne\n",
      "Brenthis_hecate\n",
      "Brenthis_ino\n",
      "Calephelis_nemesis\n",
      "Calephelis_virginiensis\n",
      "Calycopis_cecrops\n",
      "Celastrina_argiolus\n",
      "Coenonympha_glycerion\n",
      "Colias_behrii\n",
      "Colias_croceus\n",
      "Colias_eurytheme\n",
      "Cyaniris_semiargus\n",
      "Danaus_plexippus\n",
      "Dircenna_loreta\n",
      "Dryadula_phaetusa\n",
      "Dryas_iulia_moderata\n",
      "Elymnias_hypermnestra\n",
      "Erebia_aethiops\n",
      "Erebia_ligea\n",
      "Eueides_isabella\n",
      "Eumaeus_atala\n",
      "Euphydryas_editha\n",
      "Eurema_hecabe\n",
      "Fabriciana_adippe\n",
      "Glaucopsyche_alexis\n",
      "Heliconius_charithonia\n",
      "Heliconius_nattereri\n",
      "Heliconius_sara\n",
      "Helleia_helle\n",
      "Hestina_assimilis\n",
      "Hipparchia_semele\n",
      "Hypolimnas_misippus\n",
      "Lasiommata_megera\n",
      "Leptidea_juvernica\n",
      "Leptidea_reali\n",
      "Leptidea_sinapis\n",
      "Leptophobia_aripa\n",
      "Limenitis_camilla\n",
      "Luehdorfia_chinensis\n",
      "Lycaena_phlaeas\n",
      "Lysandra_bellargus\n",
      "Lysandra_coridon\n",
      "Maniola_hyperantus\n",
      "Maniola_jurtina\n",
      "Mechanitis_mazaeus\n",
      "Mechanitis_messenoides\n",
      "Melanargia_galathea\n",
      "Melinaea_marsaeus_rileyi\n",
      "Melinaea_menophilus\n",
      "Melitaea_cinxia\n",
      "Mellicta_athalia\n",
      "Napeogenes_inachia\n",
      "Napeogenes_sylphis\n",
      "Nymphalis_c-album\n",
      "Nymphalis_io\n",
      "Nymphalis_polychloros\n",
      "Nymphalis_urticae\n",
      "Oeneis_ivallda\n",
      "Ornithoptera_alexandrae\n",
      "Ornithoptera_priamus\n",
      "Papilio_dardanus_tibullus\n",
      "Papilio_demoleus\n",
      "Papilio_elwesi\n",
      "Papilio_gigon\n",
      "Papilio_glaucus\n",
      "Papilio_machaon\n",
      "Papilio_memnon\n",
      "Papilio_protenor\n",
      "Papilio_xuthus\n",
      "Pararge_aegeria\n",
      "Phengaris_arion\n",
      "Philaethria_dido\n",
      "Pieris_brassicae\n",
      "Pieris_mannii\n",
      "Pieris_melete\n",
      "Pieris_napi\n",
      "Pieris_rapae\n",
      "Plebejus_argus\n",
      "Polyommatus_icarus\n",
      "Polyommatus_iphigenia\n",
      "Sericinus_montela\n",
      "Teinopalpus_imperialis\n",
      "Troides_aeacus\n",
      "Troides_oblongomaculatus\n",
      "Vanessa_atalanta\n",
      "Vanessa_cardui\n",
      "Vanessa_tameamea\n",
      "Zerene_cesonia\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "location = \"/mnt/c/Users/sauba/Desktop/Emails/For Email Timeless\"\n",
    "list_of_species = os.listdir(location)\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "    \n",
    "for species in list_of_species:\n",
    "#     input(species)\n",
    "    print(species)\n",
    "    with open(f\"{location}/{species}/Transcript.bed\", 'r') as bed_file:\n",
    "        bed_file_list= bed_file.readlines()\n",
    "    left_overhang = 0\n",
    "    right_overhang = 0\n",
    "    for lines in bed_file_list:\n",
    "        lines_split = lines.split(\"\\t\")\n",
    "        start = int(lines_split[1])\n",
    "        stop = int(lines_split[2])\n",
    "\n",
    "        genome_file = SeqIO.parse(f\"{location}/{species}/Genome_fragment.fa\", \"fasta\")\n",
    "\n",
    "        for records in genome_file:\n",
    "            sequence = (records.seq[start:stop])\n",
    "            \n",
    "            translated_sequence = (sequence[left_overhang:].translate())\n",
    "            \n",
    "            \n",
    "            if \"*\" in translated_sequence[:-1]:\n",
    "                print(sequence)\n",
    "                print(translated_sequence)\n",
    "                print(lines_split[3])\n",
    "                check = input(\"continue?\")\n",
    "                if check == \"n\":\n",
    "                    assert False\n",
    "                \n",
    "            right_overhang = len(sequence[left_overhang:]) % 3\n",
    "#             print(left_overhang,right_overhang)\n",
    "            left_overhang = (3 - right_overhang)%3\n",
    "    #         print(left_overhang,)\n",
    "    #     print(start,stop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c4a3d8a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
