{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bac35abe",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# Make Fasta File\n",
    "\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "family_group = \"4.Pierinae\"\n",
    "number_of_exons = 16\n",
    "\n",
    "location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/7.Timeless Exon Analysis/{family_group}\"\n",
    "\n",
    "list_of_folders = os.listdir(location)\n",
    "if \"3.Test_alignment\" in list_of_folders:\n",
    "    subprocess.run(f'rm -r \"{location}/3.Test_alignment\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "os.mkdir(f\"{location}/3.Test_alignment\")\n",
    "\n",
    "    \n",
    "exons_location = f\"{location}/2.Final_output\"\n",
    "\n",
    "list_of_species = os.listdir(exons_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "\n",
    "\n",
    "for exon_number in range(1,number_of_exons+1):\n",
    "\n",
    "    exon_align = ''\n",
    "    for species in list_of_species:\n",
    "#         print(species, \"Exon\", exon_number)\n",
    "        try:\n",
    "            sequence = SeqIO.parse(f\"{exons_location}/{species}/Exon_{exon_number}.fa\", 'fasta')\n",
    "\n",
    "            for records in sequence:\n",
    "\n",
    "                left_overhang = int(records.id.split(\"_\")[-3])\n",
    "\n",
    "                sequence = records.seq[left_overhang:].translate()\n",
    "\n",
    "                exon_align += f\">{records.id}\\n{sequence}\\n\\n\"\n",
    "\n",
    "        except:\n",
    "            \n",
    "            pass\n",
    "    \n",
    "    \n",
    "    with open(f\"{location}/3.Test_alignment/Exon_{exon_number}.fa\",'w') as out_file:\n",
    "        out_file.write(exon_align)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14fde9ac",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_1.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car MEWVIRGPQIHSTFSNLGFSHPEGYHINENCN\n",
      "Aporia_crataegi MEWVLRGPQIHSTFSNLGFAHSEGYHINENCN\n",
      "Pieris_mannii_E MEWVLRGPQIHSTFSKLGFAHGEGYHINENCN\n",
      "Pieris_rapae_Ex MEWVLRGPQIHSTFSKLGFAHGEGYHINENCN\n",
      "Pieris_melete_E MEWVLRGPQIHSTFGKLGFAHAEGYHINENCN\n",
      "Pieris_napi_Exo MEWVLRGPQIHSTFGKLGFAHAEGYHINENCN\n",
      "Pieris_brassica MEWVLRGPQIHSTFSSLGFAHAEGYHIHENCN\n",
      "Leptophobia_ari MDWVLRGPQIHSTFSNLGFSHQDGYHINENCN\n",
      "                *:**:*********..***:* :****:****\n",
      "\n",
      "continue?y\n",
      "Exon_2.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car ALESILHNILTEDKNLRTYRRSISFGQNIKRDLIPLLIHVKEDKTIELLIKILVNLTIPV\n",
      "Aporia_crataegi ALESILHNILTEDKNLRTYRRSISFGQNIKKDLIPLLIHVKDDKTIELLIKILVNLTIPV\n",
      "Leptophobia_ari ALESILNNILIEDKNLRTYRRSINFGQNIKKDLIPLLIHVKDDKTIELLIKILVNLTIPV\n",
      "Pieris_brassica ALESILNNILTEDKNLRTYRRSISFGQNIKKDLIPLLLHVKDDKTIELLMKILVNLTIPV\n",
      "Pieris_mannii_E ALESILNNILTEDKNLRTYRRSISFGQNIKKDLIPLLIHVKDDKTIELLIKILVNLTIPV\n",
      "Pieris_rapae_Ex ALESILNNILTEDKNLRTYRRSISFGQNIKKDLIPLLIHVKDDKTIELLIKILVNLTIPV\n",
      "Pieris_melete_E ALESILNDILTEDKNLRTYRRGISFGQNIKKDLIPLLIHVKDDKTIELLIKILVNLTIPV\n",
      "Pieris_napi_Exo ALESILNNILTEDKNLRTYRRGISFGQNIKKDLIPLLIHVKDDTTIELLIKILVNLTIPV\n",
      "                ******::** **********.*.******:******:***:*.*****:**********\n",
      "\n",
      "Anthocharis_car ECLLSVDTVSNTTVGKHNIFEINSLLSSTKSAFSDNRATKVILEFLKKNSIVEQTSKLTS\n",
      "Aporia_crataegi ECLLSLDTISNTDVGKHTIFEINSLLSSTKAAFVDHRATKVIVKFLNKNSDLEQNKKLTL\n",
      "Leptophobia_ari ECLLSVDTISHTEVGKHTVFEINSLLSSTKAAFVDNRATKVIVEFLKKNSDLEQKAKLTV\n",
      "Pieris_brassica ECLLSVDSISNSEVGKHTIFEINSLLSSTKAAFVDNRATKVIVEFLKKNSDFKQKTKLTL\n",
      "Pieris_mannii_E ECLLSVDSISNSEVGKHTIFEINSLLSSTKAAFVDNKATKVIVEFLKKNSDCEQPPKLTV\n",
      "Pieris_rapae_Ex ECLLSVESLSTSEVGKHTIFEINSLLTSTKAAFVDNKATKVIVEFLKKNSDCEQKLKLTV\n",
      "Pieris_melete_E ECLLSVDSISNSEVGKHTIFEINNLLSSTKAAFVDNRATKVIVDFLKKNSDFEQNTKLTV\n",
      "Pieris_napi_Exo ECLLSVDSISNSEVGKHTIFEINTLLSSTKAAFVDIRATKVIVEFLKKNSDFEQNTKLTV\n",
      "                *****::::* : ****.:****.**:***:** * :*****:.**:***  :*  *** \n",
      "\n",
      "Anthocharis_car EQCTNISNTLLFLRNILHIPEDIYNMSSSYNGPPHTIQNQILWNLFSQSIDKILIKLMTI\n",
      "Aporia_crataegi EQCTNISNTLLFLRNILHIPEDG-NMSSNYSGPPHTTQNQILWNLFSQSIDKTLMKLMTI\n",
      "Leptophobia_ari EQCANISNTLLFLRNILHIPEDV-NVPYSYSAPPHTIQNQILWNLFSQSIDKTLIKLMTI\n",
      "Pieris_brassica EQCANISNTLLFLRNILHIPEDV-IISSSYSG--HILQNQIIWNLFSQSIDKTLIKLMTI\n",
      "Pieris_mannii_E EQCTNISNTLLFLRNILHIPEDI-IMSSSYSGPPHMLQNQIIWNLFSQSIDKTLIRLMNI\n",
      "Pieris_rapae_Ex EQCTNISNTLLFLRNILHIPEDI-IISSSYNGPPHVLQNQIIWNLFSQSIDKTLMRLMNI\n",
      "Pieris_melete_E EQCTNISNALLFLRNILHIPEDI-IMSSSYSGPPHMLQNQIIWNLFSQSIDKSLIKLMNI\n",
      "Pieris_napi_Exo EQCTNISNALLFLRNILHIPEDN-IMSSSYSGPPHMLQNQIIWNLFSQSIDKSLIKLMNI\n",
      "                ***:****:*************   :. .*..  *  ****:********** *::**.*\n",
      "\n",
      "Anthocharis_car PDS\n",
      "Aporia_crataegi EDS\n",
      "Leptophobia_ari QDS\n",
      "Pieris_brassica QDS\n",
      "Pieris_mannii_E KDS\n",
      "Pieris_rapae_Ex QDS\n",
      "Pieris_melete_E QDS\n",
      "Pieris_napi_Exo QES\n",
      "                 :*\n",
      "\n",
      "continue?y\n",
      "Exon_3.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car IHWGVTMVQLIALLYKDQHVTTLHKLLNMWLETSLSESSEDNESNTSPPINR\n",
      "Aporia_crataegi INWSVTMVQLVALLYKDQHVATLQKLLNIWLETSLTESSEDNESNTSPP-NR\n",
      "Leptophobia_ari INWGVTMVQLIALLYKDQHVTTLHKLLNIWLETSLSESSEDNESNTSPP-NR\n",
      "Pieris_melete_E INWGVTMVQLIALLYKDQQVTTLHKLLNIWLEKSLSESSEDNESNTTPP-NR\n",
      "Pieris_napi_Exo INWGVTMVQLIALLYKDQQVTTLHKLLNIWLETSLSESSEDNESNTTPP-NR\n",
      "Pieris_mannii_E INWGVTMVQLIALLYKDQHVTTIHKLLNIWLGTSLSESSEDNESNTTPP-NR\n",
      "Pieris_rapae_Ex INWGVTMVQLIALLYKNQNVTTIHKLLNMWLGTSLSESSEDNESNTSPP-NR\n",
      "Pieris_brassica INWGVTMVQLVALLYKDQHVTMLHKLLNIWLGTSLSESSEDNESNTSPT-DR\n",
      "                *:*.******:*****:*:*: ::****:** .**:**********:*. :*\n",
      "\n",
      "continue?y\n",
      "Exon_4.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car SDDSSRMLTSDPTSDSSDSGGCGKMNEGSNSADNSWDSATRNMSEQTFQSDTLQEKESSF\n",
      "Aporia_crataegi SDDSSPMLTSDPTSDSSDTGGSGKINEGSNSPNNSWDS-NRNMSEQTYQSDTMQEKTCLF\n",
      "Leptophobia_ari SDNSSPMLTSDPTSDSSDTSGSGKMNEGPDSPNNSWHS-NRNMSEQIFQSDSLQDQTRLL\n",
      "Pieris_brassica RDDSSPMLTSDPTSDSSDTGASGKTNGGSVSPDNSWVS-TRNMGEQTFQLEPMQEKSCFI\n",
      "Pieris_melete_E SNNSSPMLTSDATSDSSDTGRSGKTNEGSNSIDNSWDS-TRNMSEQTFQSDPMQQKSCLM\n",
      "Pieris_napi_Exo SNNSSPMLTSDATSDSSDTGRSGKTNEGSNSLNNSWDS-TRNMSEQTFQSDPMQKKSCRI\n",
      "Pieris_mannii_E SGDSSPMLTSDPISDYSDMGGSGKTNEGSISPNNSWDS-NINMSEPIFQSEPMQEKSGHV\n",
      "Pieris_rapae_Ex SGDSSPMLTSEPISDYSDMGGSGKTNEGSISPNNSWDS-NINMSEPIFQSEPMQEKSGHI\n",
      "                 .:** ****:. ** ** . .** * *. * :*** * . **.*  :* :.:*.:   .\n",
      "\n",
      "Anthocharis_car INTGQ-ENAFDNIDERN\n",
      "Aporia_crataegi INDEQGDKSFDNNKEMN\n",
      "Leptophobia_ari INNEQGDNSFKNMTHIT\n",
      "Pieris_brassica KNNEQGENSYEKDIN--\n",
      "Pieris_melete_E INNEQGENSREKDIN--\n",
      "Pieris_napi_Exo INNEQGENSRDKDIN--\n",
      "Pieris_mannii_E LNIGQRENSSEKNIK--\n",
      "Pieris_rapae_Ex LNIGQRENSCDENIK--\n",
      "                 *  * ::: .:  .  \n",
      "\n",
      "continue?y\n",
      "Exon_5.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car NTISKQKQDQMEIQNHNEPTKLGEKIRASETSDCGYGTQIENQEVISTSSNEDELPTKKP\n",
      "Aporia_crataegi NEMMK-GNEQMDSQSHDEPTGC-DKKMTAETLDCGYSTQIENQEVISTSCNEDELPSKEP\n",
      "Pieris_brassica NEILK-PKEKTETKNRDELIN--DTKMTLEKSDCGYGTQNENQEVISISSNEDDWPFKKP\n",
      "Pieris_melete_E NETLK-PKEKTKKKNRDEQIN--DKNMSSEKSDCGYDTQMEKQEVISTSSNEDEMPCEKP\n",
      "Pieris_napi_Exo KEILK-PKEKTGKKNRDKPIN--DKKMSSEKSDCGYGTQIEKQEVISTSSNDDEMPCKKP\n",
      "Pieris_mannii_E VEILR-PKEKTENKNFDVPIN--DEKMTSKKSNCGYNTKIENQE--STSSNEDELPFKKP\n",
      "Pieris_rapae_Ex GEILK-AEEETDNKNFDKPIN--D-----EKSNCSYDEKIDNQE--STSSNDDELPSKKP\n",
      "Leptophobia_ari GGTLK-PNEHAE--HCEIPIN--VKKLKSE-SECSYGALIENQEVISTSSNDDEMPSKKP\n",
      "                    :  ::.      :            :  :*.*.   ::**  * *.*:*: * ::*\n",
      "\n",
      "Anthocharis_car VHQKPHNPKQRPNNKGRASVPLQDRKRKKMLKRGKAN-\n",
      "Aporia_crataegi VHQKPNNPNQRPDNTVRGSVTFQERQRKKIGKRGKANI\n",
      "Pieris_brassica VHQKPHNPKQKPNNKMRASVTLQERKRKKIVKRGKAK-\n",
      "Pieris_melete_E VHLKPYNPKQRPNNKVPESVTLQERKRKKIVKRGKAN-\n",
      "Pieris_napi_Exo VHQKPQHPKQRPNNKVPESVALQERKRKKIVKRGKAN-\n",
      "Pieris_mannii_E VHQKPHNPKLKPNNKVLASVTLQERKRKKIVKRGKAN-\n",
      "Pieris_rapae_Ex VHQKPHNPKQKPNNNVLSSVTPQESKRKKIVKRGKAN-\n",
      "Leptophobia_ari VHQKLHKP-----HKIRVSVTLQDRKRKKIVKRGKAN-\n",
      "                ** *  :*     :.   **. *: :***: *****: \n",
      "\n",
      "continue?y\n",
      "Exon_6.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car INVQGLTHNTPTAEDISNVLKDFTVDFLLKGYNSLVRTLHTQILTNKQLEIDTSHFFWLV\n",
      "Aporia_crataegi INVQGLTHNTPTDEDVSNVLKEFTIDFLLKGYNSLVRTLHTQIMTNMQLEIDTSHFFWLV\n",
      "Leptophobia_ari INVQGLTQITPTDEDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNMQLEIDTSHFFWLV\n",
      "Pieris_melete_E INVQGLTHNTPTDEDISNVLKEFTLDFLLKGYNSLVRTLHTQILTNMQLEIDTSHFFWLV\n",
      "Pieris_napi_Exo INVQGLTHNTPTDEDISNVLKEFTLDFLLKGYNSLVRTLHTQILTNMQLEIDTSHFFWLV\n",
      "Pieris_brassica INVQGLSHNMPTDEDISNVLKEFTVDFLLRGYNSLVRTLHTQILTNMQLEIDTSHFFWLV\n",
      "Pieris_mannii_E INVQELLHNMPTDEDISNVLKEFTVDFLLKGYNSLVRTLHTQILTNMHLEIDTSHFFWLV\n",
      "Pieris_rapae_Ex INVQGLLHNMPTDDDISNVIKEFTVDFLLKGYNSLVRTLHTQILTNVHLEIDTSHFFWLV\n",
      "                **** * :  ** :*:***:*:**:****:*************:** :************\n",
      "\n",
      "Anthocharis_car TYFLKFATQIELDLEQVKSVVSFDIVMYLTAEGVNLCEQFELAIKLDGNDLKPSIRRLHL\n",
      "Aporia_crataegi TYFLKFATQIELGMEHVSSVLSFDIVLYLTAESVNLCEQFELAVKLDGNDLKPRIRRLHL\n",
      "Leptophobia_ari TYFLKFATQIELDLKHVNSVLSFDIVLYLTAEGVNLCEQFELAVKLDGNNLKPSIRRLHL\n",
      "Pieris_melete_E TYFLKFATQIELNLKHVNSVLSFDIVLYLTAEGVNLCEQFELAIKLDENDLKPSIRRLHL\n",
      "Pieris_napi_Exo TYFLKFATQIELNLKHVNSVLSFDIVLYLTAEGVNLCEQFELAIKLDENDLKPSIRRLHL\n",
      "Pieris_brassica TYFLKFATQIELDLDHVNSVLSYDIVLYLTAEGVNLCEQFELAIKLDGYDLKPSIRRLHL\n",
      "Pieris_mannii_E TYFLKFATQIELDLEYVNSVLSFDIVLYLTAEGVNLCEQFELALKLDGCDLKPSIRRLHL\n",
      "Pieris_rapae_Ex TYFLKFATQIELDLEYVDSVLSFDIVLYLTAEGVNLCEQFELALKLDGCDLKPNIRRLHL\n",
      "                ************.:. *.**:*:***:*****.**********:***  :*** ******\n",
      "\n",
      "continue?y\n",
      "Exon_7.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car VVTAIREFVQAIEVYKKSIHICQKEKEVLLKLQMKMCESEELRSLLVLLLRHYNPKYHSK\n",
      "Aporia_crataegi VVTAIREFVQAIEVYKKSIHICQEDKEVLLQLQIKMCEAEELRSSLVLLLRHYNPKYHSK\n",
      "Leptophobia_ari VVTAIREFVQAIEVYKKSIHVCREDKDILLELQMKMSETEELRSLFVLLLRHYNPKYHSK\n",
      "Pieris_brassica VVTAIREFVQAIEVYQKSNHISQKDKEVLVQLQMKMSETNELRSLFVLLLRHYNPKYHSK\n",
      "Pieris_mannii_E VVTAIREFVQAIEVYKKSNLICQEDKEVLIQLQMKMCETDELRSLLVLLLRHYNPKYHSK\n",
      "Pieris_melete_E VVTAIRELVQAIEVYKKSNHICQKDKEALIQLQMKMSETDELRSLLVLLLRHYNPKYHSK\n",
      "Pieris_napi_Exo VVTAIRELVQAIEVYKKSNHICQEDKEALIQLQMKMSETDELRSLLVLLLRHYNPKYHSK\n",
      "Pieris_rapae_Ex VVTAIREFVQAIEVYKKSSHISQEDKEALVKLQMKMCETDELRSLLVLLLRHYNPKYQSK\n",
      "                *******:*******:**  :.:::*: *::**:**.*::**** :***********:**\n",
      "\n",
      "Anthocharis_car QYLQ\n",
      "Aporia_crataegi QYLQ\n",
      "Leptophobia_ari QYLQ\n",
      "Pieris_brassica QYLQ\n",
      "Pieris_mannii_E QYLQ\n",
      "Pieris_melete_E QYLQ\n",
      "Pieris_napi_Exo QYLQ\n",
      "Pieris_rapae_Ex QYLQ\n",
      "                ****\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "continue?y\n",
      "Exon_8.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car DLIVTNHILLMFVDNVMKLPDYKGSSNMIDHIR\n",
      "Aporia_crataegi DLVVTNHILLMFLDDGMKLPDYKGSGKMIDHIR\n",
      "Leptophobia_ari DLIVTNHILIMFLDSVMSLPEYKGTGKMIDHIR\n",
      "Pieris_brassica DLIVTNHILLMFLDNIMHLPDYKGSGKMIDHLR\n",
      "Pieris_mannii_E DLIVTNHILLMFLDNVMNLPDYKGSGKMIDHLR\n",
      "Pieris_melete_E DLIVTNHILLMFLDNVMNLPDYKGSGKMIDHLR\n",
      "Pieris_napi_Exo DLIVTNHILLMFLDNVMNLPDYKGSGKMIDHLK\n",
      "Pieris_rapae_Ex DLVVTNHILLMFLDTVINLPDYKGSGKMIDYLR\n",
      "                **:******:**:*  : **:***:.:***:::\n",
      "\n",
      "continue?y\n",
      "Exon_9.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car FATPEIMYQYGLLLEDYEGNGEYVNDCVFTLMHHVVGELDCLISLFQPRLLKTFTAIWKS\n",
      "Aporia_crataegi FATPDVMYQYGLLLDEYEGNGEYVNDCVFTLMHHVGGELDSLISLFQPKILRTFTSIWKS\n",
      "Leptophobia_ari FATPEIMYQYGLLLDDYEGNGEYVNDCVFTLMHHVGGELGCLISLFQPKILKTFTSIWKS\n",
      "Pieris_mannii_E FATPQVMYQYGLLLDDYEGNGEYVNDCVFTLMHYVGGELGCLISLFQPKILKTFTSIWKS\n",
      "Pieris_rapae_Ex FATPQVMYQYGLLLDDYMGNGEYVNDCIFTLMHYVGGELGCLISLFQPKILKTFTSIWKS\n",
      "Pieris_melete_E FATPEIMYQYGLLLDDYEVNGEYVNDCVFTLMHYVGGELGCLISLFQPKILKTFTSIWKS\n",
      "Pieris_napi_Exo FATPEIMYQYGLLLDDFEVNGEYVNDCVFTLMHYVGGELGCLISLFQPKILKTFTCIWRS\n",
      "Pieris_brassica FATPEVMHQYGLLLEDYEGNGEYVNDCVFTLMHYVGGELGCLISLYQPKILKTFTSIWKS\n",
      "                ****::*:******:::  ********:*****:* ***..****:**::*:***.**:*\n",
      "\n",
      "Anthocharis_car DFEICD\n",
      "Aporia_crataegi EFDICD\n",
      "Leptophobia_ari EFEICD\n",
      "Pieris_mannii_E EFEICD\n",
      "Pieris_rapae_Ex DFEICD\n",
      "Pieris_melete_E EFEICE\n",
      "Pieris_napi_Exo EFEICD\n",
      "Pieris_brassica EFEICD\n",
      "                :*:**:\n",
      "\n",
      "continue?y\n",
      "Exon_10.fa\n",
      "CLUSTAL format alignment by MAFFT L-INS-i (v7.525)\n",
      "\n",
      "\n",
      "Anthocharis_car DWSDLIEYVINTFIKKPHSLQPTAIFRLETESFDDDKVLKEDEKDEVPKESEDSKPD---\n",
      "Aporia_crataegi DWSDLIEYVLNTFVKKPQNLQLTKYFRLDTETFDDDKVLKDPFENEDSKVVEDTTLDQVI\n",
      "Leptophobia_ari DWSDLIEYVINTFIKKPHSLQRTMSFRLDTESFDDDKVHKEVISE---KAKEDTTADQVM\n",
      "Pieris_brassica DWSDLIEYVINTFIKKPHSLQRTASFRLDTESFDDDKVRKDLIDD----SKE------FM\n",
      "Pieris_mannii_E DWSDLIEYVINTFIKKPYSLQRTASFRLDTESFDDDKILKDIIDN----SKEDTTEDHLM\n",
      "Pieris_rapae_Ex DWSDLIEYVINTFIKKPYSLQRTASFRLDTESFDDDKILKDLIEN----PKEDTTEDQVM\n",
      "Pieris_melete_E DWSDLIEYVINTFIKKPHCLQRTASFRLDTESFDDDKVFKDLIDD----PKKDSAEDQFM\n",
      "Pieris_napi_Exo DWSDLIEYVINTFIKKPHCLQRTASFRLDTESFDDDKVFKELIDD----PQEDTAEDQFM\n",
      "                *********:***:***  ** *  ***:**:*****: *:  .:      :        \n",
      "\n",
      "Anthocharis_car -------\n",
      "Aporia_crataegi LDVEDTD\n",
      "Leptophobia_ari SDVEDND\n",
      "Pieris_brassica LDVKDTD\n",
      "Pieris_mannii_E LDVEDTD\n",
      "Pieris_rapae_Ex LDVEETD\n",
      "Pieris_melete_E SDAEDTD\n",
      "Pieris_napi_Exo SDVEDTD\n",
      "                       \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Align using MAFFT\n",
    "from Bio import SeqIO\n",
    "\n",
    "error_exon_file = ''\n",
    "\n",
    "list_of_files_to_run_mafft_on = os.listdir(f\"{location}/3.Test_alignment/\")\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    print(file)\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/3.Test_alignment/{file}\" > \"{location}/3.Test_alignment/alignment_clustal_{file}.txt\"'\n",
    "        subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "    \n",
    "        with open(f\"{location}/3.Test_alignment/alignment_clustal_{file}.txt\", 'r') as alignment_file:\n",
    "            print(\"\".join(alignment_file.readlines()))\n",
    "\n",
    "        error_exon = input(\"continue?\")\n",
    "        if error_exon != \"y\":\n",
    "\n",
    "            if \",\" not in error_exon:\n",
    "                error_exon_list = [error_exon]\n",
    "            else:\n",
    "                error_exon_list = error_exon.split(\",\")\n",
    "\n",
    "            for species in error_exon_list:\n",
    "                sequence = SeqIO.parse(f\"{location}/3.Test_alignment/{file}\", \"fasta\")\n",
    "\n",
    "                for records in sequence:\n",
    "                    if species in records.id:\n",
    "                        print(records.id)\n",
    "                        species_name = input(\"Species_name?\")\n",
    "                        scaffold = input(\"Scaffold:\")\n",
    "                        exon = file.split(\"_\")[1].split(\".\")[0]\n",
    "\n",
    "                        error_exon_file += f\"{species_name},{scaffold},000,000,0,Y,Error_Exon_{exon},00,00,00\\n\\n\"\n",
    "\n",
    "\n",
    "with open(f\"{location}/3.Test_alignment/error_exons.txt\", 'w') as out_file:\n",
    "    out_file.write(error_exon_file)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f58bc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
