{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "667f63f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(species):\n",
    "    list_of_genomes = os.listdir(\"/mnt/f/Genomes_2023-12-26\")\n",
    "    for genome in list_of_genomes:\n",
    "        if species == genome:\n",
    "            files_in_genome_folder = os.listdir(f\"/mnt/f/Genomes_2023-12-26/{genome}\")\n",
    "            for file in files_in_genome_folder:\n",
    "                if file.endswith(\"_genomic.fna\"):\n",
    "#                     print(file)\n",
    "                    return(file)\n",
    "#                     break\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d8adb7bf",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Colias_behrii', 'Colias_croceus', 'Colias_eurytheme', 'Colias_nastes', 'Eurema_hecabe', 'Phoebis_sennae', 'Zerene_cesonia', 'Leptidea_juvernica', 'Leptidea_reali', 'Leptidea_sinapis']\n",
      "Colias_behrii\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/5.Coliadinae/1.Blast_result/Colias_behrii/Colias_behrii_coordinates.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 28\u001b[0m\n\u001b[1;32m     26\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m species \u001b[38;5;129;01min\u001b[39;00m list_of_species:\n\u001b[1;32m     27\u001b[0m     \u001b[38;5;28mprint\u001b[39m(species)\n\u001b[0;32m---> 28\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mblast_output_location\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mspecies\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m_coordinates.csv\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m coordinates_file:\n\u001b[1;32m     29\u001b[0m         coordinates_list \u001b[38;5;241m=\u001b[39m coordinates_file\u001b[38;5;241m.\u001b[39mreadlines()\n\u001b[1;32m     31\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m1\u001b[39m,\u001b[38;5;28mlen\u001b[39m(coordinates_list)):\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py:284\u001b[0m, in \u001b[0;36m_modified_open\u001b[0;34m(file, *args, **kwargs)\u001b[0m\n\u001b[1;32m    277\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m file \u001b[38;5;129;01min\u001b[39;00m {\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m2\u001b[39m}:\n\u001b[1;32m    278\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m    279\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIPython won\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mt let you open fd=\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfile\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m by default \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    280\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mas it is likely to crash IPython. If you know what you are doing, \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    281\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124myou can use builtins\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m open.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    282\u001b[0m     )\n\u001b[0;32m--> 284\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mio_open\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfile\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/5.Coliadinae/1.Blast_result/Colias_behrii/Colias_behrii_coordinates.csv'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "family_group = \"5.Coliadinae\"\n",
    "\n",
    "\n",
    "working_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/8.For Phylogeny/{family_group}\"\n",
    "blast_output_location = f\"{working_location}/1.Blast_result\"\n",
    "output_location = f\"{working_location}/3.Extracted Sequences\"\n",
    "\n",
    "list_of_working_folders = os.listdir(working_location)\n",
    "if \"3.Extracted Sequences\" in list_of_working_folders:\n",
    "    subprocess.run(f'rm -r \"{output_location}\"', shell = True)\n",
    "os.mkdir(output_location)\n",
    "\n",
    "\n",
    "list_of_species = os.listdir(blast_output_location)\n",
    "\n",
    "if \"desktop.ini\" in list_of_species:\n",
    "    list_of_species.remove(\"desktop.ini\")\n",
    "    \n",
    "print(list_of_species)\n",
    "\n",
    "for species in list_of_species:\n",
    "    print(species)\n",
    "    with open(f\"{blast_output_location}/{species}/{species}_coordinates.csv\", 'r') as coordinates_file:\n",
    "        coordinates_list = coordinates_file.readlines()\n",
    "        \n",
    "    for i in range(1,len(coordinates_list)):\n",
    "        coordinates_split = coordinates_list[i].split(\",\")\n",
    "        \n",
    "        gene = coordinates_split[6].split(\"_\")[-1]\n",
    "        if gene == \"rps2\":\n",
    "            continue\n",
    "        \n",
    "        list_of_folders_in_extracted = os.listdir(f\"{output_location}\")\n",
    "        if gene not in list_of_folders_in_extracted:\n",
    "            os.mkdir(f\"{output_location}/{gene}\")\n",
    "        \n",
    "        \n",
    "        scaffold,start,stop,complement = coordinates_split[1],coordinates_split[2],coordinates_split[3],coordinates_split[4]\n",
    "#         genome_file = get_genome_file(species)\n",
    "        \n",
    "        genome_fasta = SeqIO.parse(f\"/mnt/f/Genomes_2023-12-26/{species}/{get_genome_file(species)}\", 'fasta')\n",
    "        \n",
    "        for records in genome_fasta:\n",
    "            if scaffold in records.id:\n",
    "                sequence = (records.seq[int(start)-1:int(stop)])\n",
    "                break\n",
    "        if complement == \"1\":\n",
    "            sequence = sequence.reverse_complement()\n",
    "            \n",
    "        \n",
    "        print(gene)\n",
    "#         print(sequence,\"\\n\")\n",
    "        if \"*\" in sequence.translate():\n",
    "            print(sequence.translate())\n",
    "            assert False\n",
    "        \n",
    "        with open(f\"{output_location}/{gene}/{species}.txt\", 'w') as out_file:\n",
    "            output = f\">{species}_{gene}\\n{sequence}\"\n",
    "            out_file.write(output)\n",
    "        \n",
    "#         print(coordinates_list[i])\n",
    "#     assert False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d8ec914",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
