{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "57c07d87-9bba-4e3d-92df-b8cc20b952b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b7e1ac4b-92ba-49c1-b327-9c2c73999dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_output_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/1.Blast_output\"\n",
    "species = \"Pararge_aegeria\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "96a2f162-dfcb-4334-907d-388c74826ccb",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def mafft_alignement_and_processing(coordinate_file_list, current_exon_number):\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "eec2ab65-23cf-4ae4-93eb-cd93dfa83757",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "\n",
      "Pararge_aegeria,NC_053208.1,16934367,16934455,1,N,5.Bicyclus_anynana_XM_024088150.2_query_Exon_1,1,29,29\n",
      "\n",
      "Pararge_aegeria,NC_053208.1,16926960,16927135,1,N,5.Bicyclus_anynana_XM_024088150.2_query_Exon_3,1,57,58\n",
      "\n"
     ]
    }
   ],
   "source": [
    "query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query\"\n",
    "\n",
    "list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "coordinate_file_name = ''\n",
    "for file_names in list_of_files_in_species_folder:\n",
    "    if file_names.endswith(\"_coordinates.csv\"):\n",
    "        coordinate_file_name = file_names\n",
    "if coordinate_file_name =='':\n",
    "    print(f\"Coordinate file error\")\n",
    "    assert False\n",
    "\n",
    "with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "    coordinate_file_lines = temp_file_open.readlines()\n",
    "print(coordinate_file_lines[0])\n",
    "for i in range(1,len(coordinate_file_lines)):\n",
    "    coordinate_file_lines_split = coordinate_file_lines[i].split(\",\")\n",
    "    current_exon_coordinates = coordinate_file_lines[i].split(\",\")\n",
    "    if coordinate_file_lines_split[5] == \"Y\":\n",
    "        if i == 1:\n",
    "            print(f'First Exon has errors\\n{coordinate_file_lines[i]}')\n",
    "            assert False\n",
    "        previous_exon_number = i-1\n",
    "        next_exon_number = i+1\n",
    "        previous_exon_coordinates = ''\n",
    "        next_exon_coordinates = ''\n",
    "        while True:\n",
    "            if coordinate_file_lines[previous_exon_number].split(\",\")[5] == \"N\":\n",
    "                previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                break\n",
    "            else:\n",
    "                proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[previous_exon_number]}\\nProceed?\")\n",
    "                if proceed_prompt.lower()[0] == \"y\":\n",
    "                    previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                    break\n",
    "            previous_exon_number -= 1\n",
    "        while True:\n",
    "            if coordinate_file_lines[next_exon_number].split(\",\")[5] == \"N\":\n",
    "                next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                break\n",
    "            else:\n",
    "                proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[next_exon_number]}\\nProceed?\")\n",
    "                if proceed_prompt.lower()[0] == \"y\":\n",
    "                    next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                    break\n",
    "            next_exon_number += 1\n",
    "\n",
    "        \n",
    "        print(previous_exon_coordinates)\n",
    "        print(next_exon_coordinates)\n",
    "        break\n",
    "        # print(coordinate_file_lines[i])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "53161d46-e567-453b-8b74-a4b185c4e0bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_905163445.1_ilParAegt1.1_genomic.fna\n",
      "['Pararge_aegeria', 'NC_053208.1', '16934367', '16934455', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_1', '1', '29', '29\\n']\n",
      "1 NC_053208.1\n",
      "16927135 16934367\n",
      "['Pararge_aegeria', 'NC_053208.1', '16934367', '16934455', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_1', '1', '29', '29\\n']\n",
      "['Pararge_aegeria', 'NC_053208.1', '16926960', '16927135', '1', 'N', '5.Bicyclus_anynana_XM_024088150.2_query_Exon_3', '1', '57', '58\\n']\n"
     ]
    }
   ],
   "source": [
    "previous_exon_coordinates\n",
    "next_exon_coordinates\n",
    "current_exon_coordinates\n",
    "query_species = \"5.Bicyclus_anynana\"\n",
    "query_transcript = \"XM_024088150.2\"\n",
    "query_exon = current_exon_coordinates[6].split(\"query\")[-1][1:]\n",
    "error_exon = query_exon\n",
    "# print(query_exon)\n",
    "# assert False\n",
    "query_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/2.Blast/0.Query\"\n",
    "annotated_genome_location = \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species\"\n",
    "\n",
    "genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "for file in list_of_files_in_genome_folder:\n",
    "    if file.endswith(\"_genomic.fna\"):\n",
    "        genome_file = file\n",
    "\n",
    "print(genome_file)\n",
    "\n",
    "upstream_exon_line, downstream_exon_line = previous_exon_coordinates.split(\",\"),next_exon_coordinates.split(\",\")\n",
    "print(upstream_exon_line)\n",
    "complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "print(complement,scaffold)\n",
    "\n",
    "if complement == \"0\":\n",
    "    gene_start = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    gene_end = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "\n",
    "if complement == \"1\":\n",
    "    gene_start = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    gene_end = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "\n",
    "print(gene_start, gene_end)\n",
    "\n",
    "\n",
    "print(upstream_exon_line)\n",
    "print(downstream_exon_line)\n",
    "\n",
    "# genome = SeqIO.parse(f\"{genome_location}/{species}/{genome_file}\", \"fasta\")\n",
    "# for entries in genome:\n",
    "#     # print(scaffold)\n",
    "#     if scaffold == entries.id:\n",
    "#         print(scaffold)\n",
    "#         gene_sequence = entries.seq[gene_start-1:gene_end]\n",
    "#         if complement == \"1\":\n",
    "#             gene_sequence = gene_sequence.reverse_complement()\n",
    "#         break\n",
    "\n",
    "with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "    \n",
    "    query_fasta_sequence = \"\".join(query_file.readlines())\n",
    "\n",
    "list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "annotated_species_name = ''\n",
    "# print(annotated_species)\n",
    "for annotated_species in list_of_annotated_genomes:\n",
    "    if annotated_species.endswith(species):\n",
    "        annotated_species_name = (annotated_species)\n",
    "        break\n",
    "if annotated_species_name == '':\n",
    "    print(f\"Error with annotated species name\")\n",
    "    assert False\n",
    "\n",
    "list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "if \"Period_gene_genomic_sequence_individual_exon\" not in list_of_files_inside_annotated_species_folder:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon\")\n",
    "query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "\n",
    "\n",
    "list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\")\n",
    "if error_exon not in list_of_exon_directories:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\")\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/for_alignment\")\n",
    "elif error_exon in list_of_exon_directories:\n",
    "    list_of_folders_1 = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\")\n",
    "    if \"for_alignment\" not in list_of_folders_1:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/for_alignment\")\n",
    "    list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/for_alignment\")\n",
    "    for file in list_of_files:\n",
    "        os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/for_alignment/{file}\")\n",
    "\n",
    "# for offset in range(3):\n",
    "# #     print()\n",
    "#     translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "#     for i in range(len(translated_sequence)):\n",
    "#         if len(translated_sequence[i])> 0.8*query_length:\n",
    "#             sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "#             with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "\n",
    "#                 output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "#                 out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6ff3654e-9e4d-4c0f-b760-b8652bc1131f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species/33.Pararge_aegeria/Period_gene_genomic_sequence_individual_exon/Exon_2\n",
      "\"/home/saurav/miniconda3/envs/ncbi_datasets/bin/mafft\" --localpair --maxiterate 16 --reorder --distout \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species/33.Pararge_aegeria/Period_gene_genomic_sequence_individual_exon/Exon_2/Exon_2_translated_genomic_sequence_152_frame2.fa\" > \"/mnt/h/My Drive/Circadian Rhythm Genes Project/6.Period Exon Analysis/1.Annotated Species/33.Pararge_aegeria/Period_gene_genomic_sequence_individual_exon/Exon_2/alignment_Exon_2_translated_genomic_sequence_152_frame2.fa.txt\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.520\n",
      "alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP     1 /1 \n",
      "done.\n",
      "tbfast (aa) Version 7.520\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/  1    1-  71\n",
      "done 001-001-1  identical.   \n",
      "dvtditr (aa) Version 7.520\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "list_of_files_to_run_mafft_on = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\")\n",
    "location = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}'\n",
    "print(location)\n",
    "# print(list_of_files_to_run_mafft_on)\n",
    "output = ''\n",
    "for file in list_of_files_to_run_mafft_on:\n",
    "    if file.endswith(\".fa\"):\n",
    "        command = f'\"/home/saurav/miniconda3/envs/ncbi_datasets/bin/mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "        print(command)\n",
    "        os.system(f'{command}')\n",
    "        # break\n",
    "# print(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "e3467057-f50a-4dff-8228-90edb251d55d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4 31\n",
      "5.Bicyclus_anynana_XM_024088150.2_Frame_1_rightoh_2_query_Exon_2\n",
      "set152_frame2\n",
      "HSSKSTHSESNSSGSSGYGGKPSSSGY\n",
      "set152_frame2 =  0.443\n",
      "\n",
      "min = set152_frame2, 0.443\n"
     ]
    }
   ],
   "source": [
    "mafft_run_folder = f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\"\n",
    "list_of_files_in_mafft_run_folder = os.listdir(mafft_run_folder)\n",
    "score_output = ''\n",
    "score = 99\n",
    "min_score_sequence = ''\n",
    "alignmnet_file = ''\n",
    "for file in list_of_files_in_mafft_run_folder:\n",
    "    if file.endswith(\".fa.hat2\"):\n",
    "        with io.open(f\"{mafft_run_folder}/{file}\", 'r') as dist_matrix_file:\n",
    "            dist_matrix_list = dist_matrix_file.readlines()\n",
    "        \n",
    "        sequence_name = dist_matrix_list[-2].rstrip().split(\"=\")[1]\n",
    "        distance_score = float(dist_matrix_list[-1].rstrip())\n",
    "        score_output += f\"{sequence_name} =  {distance_score}\\n\"\n",
    "        if distance_score < score:\n",
    "            score = distance_score\n",
    "            min_score_sequence = sequence_name\n",
    "            alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "\n",
    "alignment_file = SeqIO.parse(f\"{mafft_run_folder}/{alignment_file}\", 'fasta')\n",
    "# print (records.id)\n",
    "fasta_start_position = 0\n",
    "fasta_end_position = 0\n",
    "start_switch = 0\n",
    "end_switch = 0\n",
    "for records in alignment_file:\n",
    "    \n",
    "    if start_switch == 1 and end_switch == 1:\n",
    "        print(f\"{records.id}\\n{records.seq[fasta_start_position:fasta_end_position]}\")\n",
    "    if error_exon in records.id:\n",
    "        for current_position in range(len(records.seq)):\n",
    "            # print(records.seq[current_position])\n",
    "            if \"-\" not in records.seq[current_position] and start_switch == 0:\n",
    "                fasta_start_position = current_position\n",
    "                start_switch = 1\n",
    "            if end_switch == 1 and \"-\" not in records.seq[current_position]:\n",
    "                end_switch = 0\n",
    "                \n",
    "            if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0:\n",
    "                fasta_end_position = current_position\n",
    "                end_switch = 1\n",
    "\n",
    "        print(fasta_start_position,fasta_end_position)\n",
    "    \n",
    "            \n",
    "                \n",
    "        print (records.id)\n",
    "    # print(alignment_file_open.readlines())\n",
    "#         print()    \n",
    "#         print(dist_matrix_list[-1].rstrip())\n",
    "\n",
    "print(score_output)\n",
    "print(f\"min = {min_score_sequence}, {score}\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "db1b9326-99b1-4d4e-9f03-23e45e0a781c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exon_2\n",
      ">5.Bicyclus_anynana_XM_024088150.2_Frame_1_rightoh_2_query_Exon_2\n",
      "HSSKSTHSSGSSGYGGKQTTSGY\n",
      "33.Pararge_aegeria\n",
      "23\n",
      "GGTAAGATTTTAGCATTAGATAAGAATTAAGCTAATGAGTCATCGCCAAGTCATCACAACGAATGCGTCGTGCTTCATCACTGCGAAAAATTTTTGGCCGTCATTTTGACGTTTCACTTCTgtggaaaaaagaaaaaaaaaaggtcgacttttaactacttaaataaaggTCAATCCACAAAGGACGGAACTACACTAGCGGAACTAGATTGTTTGGCACAACACTGTGTCAATTTACCACTACACCGATTATATGGCACTCACTTTTAAAACTTGCGTACAAGCGAAACTTTTTATAAGTGGTTTTCTTCGCTGTCTTCACTTATGGCTATAAAACGGATTGATTACCAAATTCAAAGTCGCTCAGGGATCTATGAAGAGCTTAGAAGCGCTTATAGCTTAATGGTTGTGACATCGGCTTCACATGAGGGGGCACCGAGTTCGAaaccacctctaacttttcgtagttattCTGAGAGAAGGGCACGGGTGCACCCTTTACCCTATGGGCTGATAAAGGTAAGCTATTGACAGCTGACGgataaaattcgaaattaaaaGATCTGCAGGAGAACTCAAACAAGCTTAAATGGCAATGGGCACTGAGCAGGCCATGCTTATTATAGGTTTGAAGGCAGTTTAAGCGAAAGTCCTACTGGTTAGAACGTGGATAGTAGGCAATGCGAGTTGGACGAGCGATCGAGCGAGTTGGTTCGTTGGTTTGGAGTCCGTTCGTATCGTGTAATTATGGTGGCGCTTCCTAGGAAAGCTCTAAATAGAGGATTTAGCAGGAAACGTGCTCAGGCTCATGATAGATCATGATAGCCTGCCGACCGAGGATAGATAGAAATAGGAAATTCATTTTGCGTTATATTCATGCGGTTAAGCTATTATATCTATtacttgctgttgcccgcgacttcgttcgtgcacgttttcgttttcctGTAGAAAAAGTATATTTAGCTTTGTATCTTGTAAAAGCAATCGGTTGTACTTTTGAATAGAACTTTTTCTATGAAGTATCTCTCCAATAAGATCTTTTTTGTTTAGGTGGTTCTTTAATTTTCTAAAGCCGCGATAACTTCTTAATTAAATGTGAGTGAATTATAAAGTATTATACTGCCATATACCTTTGATGGTTAAACTATTTATTCGGATAAGAATTATTAACGACAGAACACACTTTGTTTCTACCGGATCTAATAAAATCTTAGGACACAGAAGCTGCTATTGCCACTTAACTGTAATATTTGGATATATTGTCCCGTGGACGTTATGTATCAATAATagtcagcgcacgccaagtagtgaattttataggcaaaattttgcaacTTTGGGTTACATCATTGTAATTTTCTTTGGAACctcaaattttttgaaaataatttatagctaatatTACTACTTGATATTAAAGCTTTCTTTTTGTAAAGGAATGGTTAAATTcggtaaaaaacaaataaacaaacatagatttattattagacGGTGATTGCCTAGTTGTCGGGGCATAGACCTGTTTTTCGTGGGGAGAGTTCGATACTCGCGGGTCAAGTCGTCGGTAAAAAGTagtttaattcaataaaaattttcattaattgaaaatattaactcTCGTGCAAGTaaaaaagcggcgatagcctaatggttagagCGTCTGTTTTGGCGGACCGAGTTTTTTTCcatgcacgcacctctaactatcgttatgtgcgtttaaatctATGCAATTGAATATCAAAATGCCTGAGAGTGCACCATAGTGTTCTTTAAGGCACGTGCCGtctacttggccagcgtgttatttatatacttatgagGTAATTCTCAACTTTCCTTGGGCATGTTTTCATTAGTAGAACTGCGACCTTTCGGTTTGAATGGGTAAATGATGACATTTTAAATCGCAAAGGTGGGACCATTATCTCACCCCTTCATCGTTATTCAATCACTTATCCACCTTGCAACGTTACTCCCTCCCAATTAACGCATATAGATAATTGTCACTCGATTTGATTGATTTTCGCGCACCCTATTCTAAATGTTATACACAAATAGAACTATTGAGGCACACTAGCAGAAAATAAACACGTGATTGCAAACTCTGCACGCACACTGACTCAGCTTGGAACGAGAAAGTGTAGTTGGATGAATTGATAGGACATTTATCGGTATTAATTATTGAGGTAATTCATTTTACATATTGACAGAAAGAGttttgcgtgcgtgtgtgtgcgtattTCGTGAACTCGTAATAATCTcgtaataatgatgatgatagggAAAACTGGAGGCGGAGGTCACGAGTTTATTACATTTGAATAGTACCGGAAGGTAATGGAACCTTTATAATCCTATGTAGGTTGTTCAAAAAACAGTTTGAAGCCTgtttgtccgtccgtccgttgTTTTGAAAACTTTCGTGGTGCAGCGTTGAGTGCTGTGGGTTTATAggcggaggtcccgggttcgatccctggcaggggaaattcgggaatttataagtACTGAACTTTTCggaacaggttagtccgctaccatcttagacttcatcatcacggAAAAGAATGCCCATACACGTATTTCTCTGAAATTACTTGAGGTACAGTGTTGAAACATATTTCCTGAATAATAGAGTCAAAAGTCTCCTTTTTGCTATGAAACTTCTAGTTAAACTCAATATAAAGAAATAGGCCATCATAGAGTTTTTTCCCGATCATGATAGTAAAAACTTAACTCCTTAAGATCCTGCAATCTTAAGAACTTCAATACTTTTAGATCTTTGcaaatatgaaaacaataatcGCTTTAAAATCGCTCTATACTTTTACTTTAGCTAAGCTAAATTTCCACATGATGGATGATACTCTACTAATTTCCACTTCGTTCTGTAACGCGTTACGGTGCCAGTCTGTCTGGCTTCCTTTTCTCGAACGCATACGGCGCTGCGAGAAAGGGGGAGGAGTAGGGGGGGAGTAGGAAGGCTAATGAACATATTCAAGAATAGTaactaagaataaaatatattaaataacctttccttacaatattatattaatgcccgttttcactaacgtcgATTATTGCAATGCTTTAGAAAGTAACGTCTAgtctaagaagattcctagcTTCGTTGCTTTTCACCAATCAATTTTCACTAACAAAACTTCGGTAAGCTATGTGTCGTTTATATTGTCTATGAATCCTTGCCACAAGTTGTCAAATTTGATGTATGACTTTGACATTTGTTtcttttgtctttgttttgaCGTTGAACAAGCCATcttgcattttatttttcgtaatgCTTGTACTTAAAGCACGCGAGCGGCCCGTTCACACACTCGTATTATGTCCTATGATAATAAGATTATACCTTAAATTTGGGACCCGAAACACCTAGCcctatatataacaaaaaaaatagcccCGAGAGGTGCCTTTTGCGAGTATGAAGTgaaaacagatttaaattaaaagaaaaagtttttttttttctatcactttacgagttccctaaactttacgacccgtcgaatcggtgtcgtaactttagaagtcgcctaacgtcgttagtaacataagagttggtgaaacgttttttttctctaggaaatCACTAGGCGTCGAAAACGTTTTCTTAGTTTAGTTAATACAACCGCCGTTAAACATTAAATGACACCGAATCTAAGGAATTagtattgcattttatttattctttaattctAATGTTCTTAAATTCGCAAATGTCCTTGTTGGTTTGATAAAGATTACGTGAAGTACCTAAGTAAAAGCCCAAAGACTGGAATTGTTCATAAATgcaatgattgaaaataattatatcaattattaattatctatctatacttataccTAATAAAACAGCTGTAACGGGTCAAACACTGTACAttgtagatatttaaaaaaaaatttttttggtaaACACTATAACCGATATCTACTATATACCGAAAGCAaaactgatatttttttcatcgtgtctttctgtctgtatctggaccgcgcatcacgctggaactactgaatgaattaaaATGGAACTTTTCACGATATGAGTTCATACTACGACGTTGGcaaaagatactttttatcccgtaatTCAGCTCAGTTTCGTTGGGAGAGGGGATGTAAGGGCATGACGATTTACAatcataactccgtcaaatgatttaataaaattttaataattattttcgtactgGATTGGTCATAGTATCACCCCTATTGTCACATCTATGAGGAGTTTTAGGCCTAggactgaataaaaaatgttagaaatcgTCTTGGTTTCTTTTTCTTAACTTTGAGCATACAAATCAACTAActatgcgatttcgattcttcctttaacgattaggTACTTTTCTTTGTTAGTGATATGAGCTAAGACTGATTGATATCAAATTGTGGTTCAAATTTCCAAAGACTGGTACCCATCCATTTAATTACTTGGATCAATGTTGCCACACATTatttgcagtcatttgttatatAGGGTGAAAAAGTATTACAATTAAGGCAAAAAGGTTCGCGGGCCAGTCATTTCCGTTTGGCAATTTTTACGTTATTggaggtttcaaatagtgggtggttaaagctccaagagcaaaaACAAGTATATTATTCGCCCACACCTCAGTCGTTTGATTAgataatatgttgaaatactatctgtacttaatattgTAGGTTTGTATAGTTGGCAGTAcagattgaaaaaactttacgtttagcatttgatttattgaggtaCGTATAGTCTATGTAGCATGTACGCTGTGCCccgtaattaacgcggacgaagccgcagggcacatctagtatatttatttaaattagttttatatttattgataacaCGTCGACCGCACGTTTTCGGCGCTTGACAATTTCTTTGGTCACGTTTACTGCCAGTTCATAACCAATCAGTAGCATATAAAAGTCCACGGCTGTAATAAAGGCAGATAATTTGTATTTGGATGTTTCCAATATTTGTTATCTGTATCCCtttattgtgtgcaataaacaatttatttatctacctattttaaatataatcttcTACATTATTCTAGGCAAAATCTCTCATTTTGTACCCTAATAATTGAGAGGGGTGCATAAAAAATTATTGCTGCGCTAGCTTTTCCAGTCCGCCATATTGTATTTAGAATAATGACACTTCGCTACCTTGCATTGCCATTTTCTAGTTTCATGCCCTAGTTAAACCCTACTACACCAATTTAGAGGTTTGATTTGCATAACATTACTTTTACTATTTTAAGCTCACTATTGCATAATacgttaaataattaataattatttgtataataataaattcaacaTCTTGTTCcgattacataatattaaaaagaaaggtacagttaaattaaagaaaactaaaCAGGTAAGAAGCCTTTATGACGCAAAAAAGAAGTGCTTTCTGAGCGGAACAAAAACTTAGGTACAACGTGACACTGAAAGGGTAAATGGGGTAGGTACGCCTGGGGCATAAACGTGAGCGCTACTTCGGCatctttattaaaaagaaagttaaaatcaaataaaactagAGAATCCATACATATCTTCTTATTTATACCGTAGTATAGTCATAGGTGTAGATTCAAATAAATGTTCTGTCATCACCTTTGTATTATGAGGATTTTAACACCTATCGTTATAAAtgttatcttaaaataaaaccggGACTTAATTAGGATAGGTAATCGCTTAACCTCAAGTAAACAACCTACTAGTTAGCCCTCGATTTTCTAGTCATCCGTATCATAGATCGGTGGAGGTTCGATACGGATCACCTAAGAACGTGCTATTTTGGAAAATATCTATCGTAAATCTCGTAGTTTTGTCGACCCCCTAGCACAGCGGTGAGCGATTTGGACTGAGAGGTCCCGAGTTCAATCATCCGGGCCTGATTTGGTCCCAGTGAAAAttgtaatttcatttatttcaagtaggcccagtttataagcattctgaaacgtcaaatcagtctgtttgaagtgactctagaactgattcggaaggcagaatttcaccgagaagagccggcaagaaactcagcattgctctttttcaaaatcCTTCACAAGTTATATTGGGCGATACTTGTTGAcaaattgacttgcagtttggagtccccctttgtgtgttctatcgactctataacggagtgttccgaagagttgtttgcgttgatacctccttccctgttctctgaccgcacctcacgtcgtagaaataaattccaccctcatcatctggatgcctggtattcttctactgttcgaaatactagatcatttctaccgcgcacttgcaaactaTGGAagaatctcccatctgatgtgtttcctcaaaaatacaatctagggttattcaaggggcggatagacaaattccttaaaagccggcaacgcatcggtggctcctctggtgctgcagatgtttatgggcggcggtgatcacttaccatcaggtgacccacttgctcgtttgcccgctattaatattaaaaaaaagaaacgatTGGGTTTAGCAGCCAGTTTAACTGAATTCtcactttttatataaatttcttcCTATATTAAAACACAAGTCCGCTAATACGTATAAGTCCGGCATGGTGGGTTTGTTTTCGTAGACACGGGGACATGAATTGGCTGTCggcaaaaaaaacttattaacttCTTTCCAGTCACAGCTCCAAATCTACTCATTCCGAGAGCAACTCTAGTGGAAGCAGTGGATATGGCGGAAAACCGTCCTCATCTGGATACAGGttcgaaaatattataatttacattatcGAATTGCTTCAATTTTTTTCGTCTGAATTATCCAtgttagatgttttttttttttgcaatggtCTCCGTCATCTAGACCCATTTCGGTAAATGACAATTTTCTTTATGAGACTTTTCTGTCTTTTtatcaatgaataatttttagaaaattcttaataatgaatatttataacgatgatgatgtcGAAATATTCGGGGTCTAGGAagttcttaatttaattatattacccAGTAACGCATAAAACAATGATATTAGCCATAAGCTTATGCTGACATTTAGACTTCAATGCAAAATTGAATTGAACTCTGACAAATTTATTGAGAGAATCCGTTTAAGGTTATGTTGCCGTGTTTTAATCCTTCCATATATTAATTTAGC\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "print(query_fasta_sequence)\n",
    "# assert False\n",
    "\n",
    "    \n",
    "\n",
    "print(query_length)\n",
    "print((gene_sequence))\n",
    "sequence_set = ''\n",
    "\n",
    "list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/\")\n",
    "if error_exon not in list_of_exon_directories:\n",
    "    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c5dc5f70-9a2a-4ee3-810d-d3026424aa89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 66\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_101_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_102_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 128 Feb 15 14:52 Exon_2_translated_genomic_sequence_102_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 132 Feb 15 14:52 Exon_2_translated_genomic_sequence_104_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 132 Feb 15 14:52 Exon_2_translated_genomic_sequence_106_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 141 Feb 15 14:52 Exon_2_translated_genomic_sequence_106_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 159 Feb 15 14:52 Exon_2_translated_genomic_sequence_107_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 131 Feb 15 14:52 Exon_2_translated_genomic_sequence_107_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_108_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 139 Feb 15 14:52 Exon_2_translated_genomic_sequence_109_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_109_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_10_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_110_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_110_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_111_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_112_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_113_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_114_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 136 Feb 15 14:52 Exon_2_translated_genomic_sequence_115_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_116_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 161 Feb 15 14:52 Exon_2_translated_genomic_sequence_117_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_120_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 149 Feb 15 14:52 Exon_2_translated_genomic_sequence_124_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 159 Feb 15 14:52 Exon_2_translated_genomic_sequence_125_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_126_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_126_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_127_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_130_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_131_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_131_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_132_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 139 Feb 15 14:52 Exon_2_translated_genomic_sequence_136_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_137_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 155 Feb 15 14:52 Exon_2_translated_genomic_sequence_137_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_138_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_13_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 165 Feb 15 14:52 Exon_2_translated_genomic_sequence_140_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 128 Feb 15 14:52 Exon_2_translated_genomic_sequence_140_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 139 Feb 15 14:52 Exon_2_translated_genomic_sequence_140_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 145 Feb 15 14:52 Exon_2_translated_genomic_sequence_141_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 167 Feb 15 14:52 Exon_2_translated_genomic_sequence_142_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_142_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 276 Feb 15 14:52 Exon_2_translated_genomic_sequence_145_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 154 Feb 15 14:52 Exon_2_translated_genomic_sequence_146_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_147_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 151 Feb 15 14:52 Exon_2_translated_genomic_sequence_148_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_150_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_151_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 139 Feb 15 14:52 Exon_2_translated_genomic_sequence_151_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_152_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 178 Feb 15 14:52 Exon_2_translated_genomic_sequence_152_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 143 Feb 15 14:52 Exon_2_translated_genomic_sequence_153_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_154_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 153 Feb 15 14:52 Exon_2_translated_ge"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nomic_sequence_156_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 129 Feb 15 14:52 Exon_2_translated_genomic_sequence_15_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_16_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_17_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_19_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 135 Feb 15 14:52 Exon_2_translated_genomic_sequence_20_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_21_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 145 Feb 15 14:52 Exon_2_translated_genomic_sequence_22_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_24_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 170 Feb 15 14:52 Exon_2_translated_genomic_sequence_25_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 131 Feb 15 14:52 Exon_2_translated_genomic_sequence_26_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 206 Feb 15 14:52 Exon_2_translated_genomic_sequence_27_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_28_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_29_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 185 Feb 15 14:52 Exon_2_translated_genomic_sequence_2_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 126 Feb 15 14:52 Exon_2_translated_genomic_sequence_30_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 131 Feb 15 14:52 Exon_2_translated_genomic_sequence_31_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 130 Feb 15 14:52 Exon_2_translated_genomic_sequence_33_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 129 Feb 15 14:52 Exon_2_translated_genomic_sequence_33_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 129 Feb 15 14:52 Exon_2_translated_genomic_sequence_34_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_34_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_36_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 142 Feb 15 14:52 Exon_2_translated_genomic_sequence_37_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 192 Feb 15 14:52 Exon_2_translated_genomic_sequence_3_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 135 Feb 15 14:52 Exon_2_translated_genomic_sequence_3_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 170 Feb 15 14:52 Exon_2_translated_genomic_sequence_40_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 136 Feb 15 14:52 Exon_2_translated_genomic_sequence_43_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_45_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_47_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 132 Feb 15 14:52 Exon_2_translated_genomic_sequence_48_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_48_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_50_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 136 Feb 15 14:52 Exon_2_translated_genomic_sequence_52_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 141 Feb 15 14:52 Exon_2_translated_genomic_sequence_53_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 135 Feb 15 14:52 Exon_2_translated_genomic_sequence_54_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_55_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_56_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_56_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_59_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 144 Feb 15 14:52 Exon_2_translated_genomic_sequence_5_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_60_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 173 Feb 15 14:52 Exon_2_translated_genomic_sequence_61_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 136 Feb 15 14:52 Exon_2_translated_genomic_sequence_61_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 128 Feb 15 14:52 Exon_2_translated_genomic_sequence_63_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 140 Feb 15 14:52 Exon_2_translated_genomic_sequence_64_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 133 Feb 15 14:52 Exon_2_translated_genomic_sequence_65_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 175 Feb 15 14:52 Exon_2_translated_genomic_sequence_66_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_66_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_67_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_68_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 136 Feb 15 14:52 Exon_2_translated_genomic_sequence_69_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 146 Feb 15 14:52 Exon_2_translated_genomic_sequence_69_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_70_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_74_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 132 Feb 15 14:52 Exon_2_translated_genomic_sequence_76_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 131 Feb 15 14:52 Exon_2_translated_genomic_sequence_76_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_79_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 164 Feb 15 14:52 Exon_2_translated_genomic_sequence_79_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 158 Feb 15 14:52 Exon_2_translated_genomic_sequence_7_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 147 Feb 15 14:52 Exon_2_translated_genomic_sequence_7_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 149 Feb 15 14:52 Exon_2_translated_genomic_sequence_7_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 135 Feb 15 14:52 Exon_2_translated_genomic_sequence_84_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 149 Feb 15 14:52 Exon_2_translated_genomic_sequence_85_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 151 Feb 15 14:52 Exon_2_translated_genomic_sequence_86_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 137 Feb 15 14:52 Exon_2_translated_genomic_sequence_87_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 128 Feb 15 14:52 Exon_2_translated_genomic_sequence_8_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 135 Feb 15 14:52 Exon_2_translated_genomic_sequence_90_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_90_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_91_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 128 Feb 15 14:52 Exon_2_translated_genomic_sequence_92_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 131 Feb 15 14:52 Exon_2_translated_genomic_sequence_93_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 161 Feb 15 14:52 Exon_2_translated_genomic_sequence_95_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 138 Feb 15 14:52 Exon_2_translated_genomic_sequence_95_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 147 Feb 15 14:52 Exon_2_translated_genomic_sequence_96_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 148 Feb 15 14:52 Exon_2_translated_genomic_sequence_96_frame2.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_97_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 151 Feb 15 14:52 Exon_2_translated_genomic_sequence_97_frame1.fa\n",
      "-rwxrwxrwx 0 saurav saurav 127 Feb 15 14:52 Exon_2_translated_genomic_sequence_98_frame0.fa\n",
      "-rwxrwxrwx 0 saurav saurav 134 Feb 15 14:52 Exon_2_translated_genomic_sequence_98_frame2.fa\n"
     ]
    }
   ],
   "source": [
    "os.system(f'ls -l \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon/{error_exon}\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f99a10c7-b1b7-45a3-b993-0fb02dbfb8ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23b5a318-e4dc-4da3-ae81-86e18f942b40",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b65d0f19-0c03-4dae-85f3-a6d92e5f4478",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8ef4e14-c3f7-420a-932d-e975bab294f1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
