{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "81fcfca7-185d-4284-bb3f-d800723fc4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "import io\n",
    "from Bio.Seq import Seq\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7b257780-eea0-47d9-b846-654cff8fa8e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9993f28b-5774-4b4f-a38f-63c393b15b08",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length):\n",
    "    for offset in range(3):\n",
    "        translated_sequence = str(gene_sequence[offset:].translate()).split(\"*\")\n",
    "        for i in range(len(translated_sequence)):\n",
    "            if len(translated_sequence[i])> 0.8*query_length:\n",
    "                sequence_set = f\">set{i+1}_frame{offset}\\n{translated_sequence[i]}\\n\\n\"\n",
    "                # print(i+1, offset)\n",
    "                \n",
    "                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{error_exon}_translated_genomic_sequence_{i+1}_frame{offset}.fa\",'w') as out_file:\n",
    "                    output = f\"{query_fasta_sequence}\\n\\n{sequence_set}\"\n",
    "                    out_file.write(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "236f726f-7cfc-4839-871c-505018689085",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_mafft(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    location = f'{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment'\n",
    "    list_of_files_to_run_mafft_on = os.listdir(location)\n",
    "    for file in list_of_files_to_run_mafft_on:\n",
    "        if file.endswith(\".fa\"):\n",
    "            command = f'\"mafft\" --localpair --maxiterate 16 --reorder --distout \"{location}/{file}\" > \"{location}/alignment_{file}.txt\"'\n",
    "            # print(command)\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            # os.system(f'{command}')\n",
    "            command = f'\"mafft\" --localpair --clustalout --maxiterate 16 --reorder \"{location}/{file}\" > \"{location}/alignment_clustal_{file}.txt\"'\n",
    "            subprocess.run(f'{command}', shell=True, stderr = subprocess.DEVNULL) \n",
    "            \n",
    "            # os.system(f'{command}')\n",
    "    return(location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "df06ca7c-0540-4e5e-a02b-2da57d4eecfb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def process_mafft_output(mafft_run_folder, error_exon):\n",
    "    list_of_files_in_mafft_run_folder = os.listdir(mafft_run_folder)\n",
    "    score_output = []\n",
    "    score = 99\n",
    "    min_score_sequence = ''\n",
    "    alignment_file = ''\n",
    "    for file in list_of_files_in_mafft_run_folder:\n",
    "        if file.endswith(\".fa.hat2\"):\n",
    "            with io.open(f\"{mafft_run_folder}/{file}\", 'r') as dist_matrix_file:\n",
    "                dist_matrix_list = dist_matrix_file.readlines()\n",
    "            \n",
    "            sequence_name = dist_matrix_list[-2].rstrip().split(\"=\")[1]\n",
    "            distance_score = float(dist_matrix_list[-1].rstrip())\n",
    "            if len(score_output) < 5:\n",
    "                score_output.append( [sequence_name,distance_score])\n",
    "            else:\n",
    "                for i in range(len(score_output)):\n",
    "                    score_at_this_index = score_output[i][1]\n",
    "                    if distance_score < score_at_this_index:\n",
    "                        score_output[i] = [sequence_name,distance_score]\n",
    "                        break\n",
    "            if distance_score < score:\n",
    "                score = distance_score\n",
    "                min_score_sequence = sequence_name\n",
    "                alignment_file = f'alignment_{file.replace(\".hat2\",\".txt\")}'\n",
    "    # print(score_output)\n",
    "    print(f\"min = {min_score_sequence}, {score}\" )\n",
    "    # score_out_merged = '\\n'.join(score_output)\n",
    "    print(f\"5 top scores:\\n{score_output}\")\n",
    "    print(f\"{mafft_run_folder}/{alignment_file}\")\n",
    "    \n",
    "    alignment_file = SeqIO.parse(f\"{mafft_run_folder}/{alignment_file}\", 'fasta')\n",
    "    # print (records.id)\n",
    "    fasta_start_position = 0\n",
    "    fasta_end_position = 0\n",
    "    start_switch = 0\n",
    "    end_switch = 0\n",
    "    alignment_name = ''\n",
    "    for records in alignment_file:\n",
    "        \n",
    "        \n",
    "        \n",
    "        if error_exon in records.id:\n",
    "            gap_counter = 0\n",
    "            base_counter = 0\n",
    "            for current_position in range(len(records.seq)):\n",
    "                sequence_length = len(records.seq) - records.seq.count('-')\n",
    "                # print(f\"fasta_end_position {fasta_end_position} fasta_start_position {fasta_start_position}\")\n",
    "                # print(f\"current_position = {current_position}, {len(records.seq)}\")\n",
    "                # print(records.seq[current_position])\n",
    "                # print(\"fasta_start_position\",fasta_start_position)\n",
    "                # print(\"start_switch\",start_switch)\n",
    "                \n",
    "                # print(gap_counter, base_counter)\n",
    "                # print(5,0.2*sequence_length)\n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and gap_counter > 3 and (len(records.seq[:current_position]) - records.seq[:current_position].count('-')) < (0.1*sequence_length):\n",
    "                #  :\n",
    "                    # print(\"\\n\\nhere\\n\\n\")\n",
    "                    start_switch = 0\n",
    "                    gap_counter = 0\n",
    "\n",
    "                if \"-\" not in records.seq[current_position] and start_switch == 0 :\n",
    "                    fasta_start_position = current_position\n",
    "                    start_switch = 1\n",
    "                \n",
    "                    \n",
    "                \n",
    "                    \n",
    "                if end_switch == 1 and \"-\" not in records.seq[current_position] and (len(records.seq[current_position:]) - records.seq[current_position:].count('-')) > 0.1*sequence_length :\n",
    "                    end_switch = 0\n",
    "                    gap_counter = 0\n",
    "                \n",
    "                    \n",
    "                if start_switch == 1 and records.seq[current_position] == \"-\" and end_switch == 0:\n",
    "                    # print(f\"base_counter {base_counter}\")\n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    # print(f\"fasta_end_position {fasta_end_position}\")\n",
    "                    \n",
    "                    fasta_end_position = current_position\n",
    "                    end_switch = 1\n",
    "                if \"-\" in records.seq[current_position]:\n",
    "                    gap_counter += 1\n",
    "                else:\n",
    "                    gap_counter = 0\n",
    "                    base_counter += 1\n",
    "        else:\n",
    "            alignment_name = records.id\n",
    "            if fasta_end_position == 0:\n",
    "                \n",
    "                fasta_end_position = current_position\n",
    "            end_switch = 1\n",
    "            start_switch = 1\n",
    "\n",
    "        if start_switch == 1 and end_switch == 1:\n",
    "            print(fasta_start_position, fasta_end_position)\n",
    "            gene_sequence = records.seq[fasta_start_position:fasta_end_position]\n",
    "            print(f\"{records.id}\\n{gene_sequence}\")\n",
    "    return(gene_sequence, alignment_name)\n",
    "\n",
    "# process_mafft_output(mafft_run_folder, error_exon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "87615294-93f0-483b-b9f7-fad66730cd1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genome_file(genome_location,species):\n",
    "    list_of_files_in_genome_folder = os.listdir(f\"{genome_location}/{species}\")\n",
    "    for file in list_of_files_in_genome_folder:\n",
    "        if file.endswith(\"_genomic.fna\"):\n",
    "            genome_file = file\n",
    "    return(genome_file)  \n",
    "\n",
    "def get_annotated_genome_name(annotated_genome_location, species):\n",
    "    list_of_annotated_genomes = os.listdir(annotated_genome_location)\n",
    "  \n",
    "    # print(annotated_species)\n",
    "    for annotated_species in list_of_annotated_genomes:\n",
    "        if annotated_species.endswith(species):\n",
    "            return(annotated_species)\n",
    "             \n",
    "    if annotated_species_name == '':\n",
    "        print(f\"Error with annotated species name\")\n",
    "        assert False\n",
    "\n",
    "def check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon):\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    \n",
    "    list_of_exon_directories = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/\")\n",
    "    \n",
    "    if error_exon not in list_of_exon_directories:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "    elif error_exon in list_of_exon_directories:\n",
    "        list_of_folders_1 = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "        if \"for_alignment\" not in list_of_folders_1:\n",
    "            os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment\")\n",
    "        for file in list_of_files:\n",
    "            os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_alignment/{file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "50742e4c-f9ab-433d-96a1-bde1f9d1c255",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name):\n",
    "    from Bio import SeqIO\n",
    "    print(\"Getting Gene\")\n",
    "    list_of_files_inside_annotated_species_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}\")\n",
    "    # print(list_of_files_inside_annotated_species_folder)\n",
    "    # print(\n",
    "    if f\"Period_gene_genomic_sequence_individual_exon_{query_species}\" not in list_of_files_inside_annotated_species_folder:\n",
    "        os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "\n",
    "    list_of_files_inside_indiv_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}\")\n",
    "    if \"temp\" not in list_of_files_inside_indiv_exon_folder:\n",
    "         os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp\")\n",
    "\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"')\n",
    "    # subprocess.run(\"pwd\")\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    # os.system(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"')\n",
    "    subprocess.run(f'samtools faidx \"{genome_location}/{species}/{genome_file}\" {scaffold}:{gene_start}-{gene_end} > \"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\"', shell = True, stderr = subprocess.DEVNULL)\n",
    "    genome = SeqIO.parse(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/temp/temp_genome.fa\", \"fasta\")\n",
    "    for entries in genome:\n",
    "        gene_sequence = entries.seq\n",
    "        if complement == \"1\":\n",
    "            gene_sequence = gene_sequence.reverse_complement()\n",
    "        break\n",
    "    # print(gene_sequence)\n",
    "    return (gene_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3cb097ff-9369-40fd-9ae5-7587c792ba31",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mafft_process(previous_exon_coordinates,\n",
    "                  next_exon_coordinates,\n",
    "                  current_exon_coordinates,\n",
    "                  query_species,\n",
    "                  query_transcript,\n",
    "                  query_location,\n",
    "                  annotated_genome_location,\n",
    "                 genome_location):\n",
    "    query_exon = current_exon_coordinates[6].split(\"query\")[-1][1:]\n",
    "    error_exon = query_exon\n",
    "    # print(\"AA\")\n",
    "    genome_file = get_genome_file(genome_location,species)\n",
    "    \n",
    "    upstream_exon_line, downstream_exon_line = previous_exon_coordinates.split(\",\"),next_exon_coordinates.split(\",\")\n",
    "    complement,scaffold = upstream_exon_line[4], upstream_exon_line[1]\n",
    "\n",
    "    if complement == \"0\":\n",
    "        gene_start = min(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "        gene_end = max(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "    \n",
    "    if complement == \"1\":\n",
    "        gene_start = min(int(downstream_exon_line[2]),int(downstream_exon_line[3]))\n",
    "        gene_end = max(int(upstream_exon_line[2]),int(upstream_exon_line[3]))\n",
    "    \n",
    "    \n",
    "    with open(f\"{query_location}/{query_species}/{query_transcript}/query_{query_exon}.fa\", 'r') as query_file:\n",
    "        query_file_list = query_file.readlines()\n",
    "        query_fasta_sequence = \"\".join(query_file_list)\n",
    "        left_overhang = query_file_list[0].split(\"Frame\")[1][1]\n",
    "        right_overhang = query_file_list[0].split(\"rightoh\")[1][1]\n",
    "        original_query_name = query_file_list[0]\n",
    "        original_query = query_file_list[1]\n",
    "        \n",
    "    query_length = len(query_fasta_sequence.split(\"\\n\")[1])\n",
    "    \n",
    "    annotated_species_name = get_annotated_genome_name(annotated_genome_location, species)\n",
    "    # print(annotated_species_name)\n",
    "\n",
    "    # print(annotated_genome_location)\n",
    "    gene_sequence = get_gene_sequence(genome_location, species, genome_file, scaffold,gene_start,gene_end,complement,annotated_genome_location,annotated_species_name)\n",
    "    # assert False\n",
    "    check_and_make_folders(annotated_genome_location,annotated_species_name,error_exon)\n",
    "    \n",
    "    \n",
    "    make_raw_files_for_alignment(gene_sequence,annotated_genome_location,annotated_species_name,error_exon,query_fasta_sequence,query_length)\n",
    "    \n",
    "    return(annotated_species_name, error_exon,species, genome_file,gene_sequence,left_overhang,right_overhang,scaffold, original_query_name,original_query )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce6f54b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3790f71-89c6-4326-a178-d4147df81ed0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930478,1930626,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_2,1,49,49\n",
      "\n",
      "1\n",
      "4\n",
      "Euphydryas_editha,CAKOGL010000026.1,1934478,1934626,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930011,1930128,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4,2,38,38\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/saurav/.local/lib/python3.8/site-packages/Bio/Seq.py:2880: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set98_frame2, 0.0\n",
      "5 top scores:\n",
      "[['set98_frame2', 0.0], ['set30_frame2', 1.732], ['set65_frame2', 1.739], ['set22_frame1', 1.805], ['set99_frame0', 1.818]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_2/for_alignment/alignment_Exon_2_translated_genomic_sequence_98_frame2.fa.txt\n",
      "19 68\n",
      "Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_2\n",
      "RTRNLSEKKRRDQFNMLVNELGSMVSTNNRKMDKSTVLKSTISFLKNHN\n",
      "19 68\n",
      "set98_frame2\n",
      "RTRNLSEKKRRDQFNMLVNELGSMVSTNNRKMDKSTVLKSTISFLKNHN\n",
      "Original query = RTRNLSEKKRRDQFNMLVNELGSMVSTNNRKMDKSTVLKSTISFLKNHN\n",
      "New query = RTRNLSEKKRRDQFNMLVNELGSMVSTNNRKMDKSTVLKSTISFLKNHN\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930208,1930389,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_0_query_Exon_3,4,60,60\n",
      "\n",
      "1\n",
      "5\n",
      "Euphydryas_editha,CAKOGL010000026.1,1934478,1934626,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1925562,1925657,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5,1,31,31\n",
      "\n",
      "Getting Gene\n",
      "min = set99_frame0, 0.208\n",
      "5 top scores:\n",
      "[['set161_frame2', 1.723], ['set170_frame2', 1.75], ['set184_frame2', 1.783], ['set99_frame0', 0.208], ['set120_frame2', 1.806]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_3/for_alignment/alignment_Exon_3_translated_genomic_sequence_99_frame0.fa.txt\n",
      "13 73\n",
      "Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_0_query_Exon_3\n",
      "ITVRSRAHDVQEDWKPAFLSNEEFTYLVLEALEGFVMVFSASGCIYYVSESVTSLLGHTP\n",
      "13 73\n",
      "set99_frame0\n",
      "ITIRSRAHDVQEDWKPVFLSNEEFTYLVLEALEGFVLVFSATGQIYYVSESIMSLLGHNP\n",
      "Original query = ITVRSRAHDVQEDWKPAFLSNEEFTYLVLEALEGFVMVFSASGCIYYVSESVTSLLGHTP\n",
      "New query = ITIRSRAHDVQEDWKPVFLSNEEFTYLVLEALEGFVLVFSATGQIYYVSESIMSLLGHNP\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930011,1930128,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4,2,38,38\n",
      "\n",
      "2\n",
      "6\n",
      "7\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930478,1930626,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_2,1,49,49\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1922276,1922514,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7,1,79,79\n",
      "\n",
      "Getting Gene\n",
      "min = set22_frame0, 0.687\n",
      "5 top scores:\n",
      "[['set22_frame0', 0.687], ['set64_frame2', 1.714], ['set51_frame2', 1.737], ['set69_frame2', 1.744], ['set43_frame0', 1.742]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_4/for_alignment/alignment_Exon_4_translated_genomic_sequence_22_frame0.fa.txt\n",
      "0 39\n",
      "Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4\n",
      "GDIINKSIFDLAFVDDRPNLYNILQN-GGTLDPTQVVTT\n",
      "0 39\n",
      "set22_frame0\n",
      "NDIVNRSIFDLVFEDDRPNLYGLLQNPAAALHPTQTVTS\n",
      "Original query = GDIINKSIFDLAFVDDRPNLYNILQNGGTLDPTQVVTT\n",
      "New query = NDIVNRSIFDLVFEDDRPNLYGLLQNPAAALHPTQTVTS\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1925562,1925657,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5,1,31,31\n",
      "\n",
      "3\n",
      "7\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930208,1930389,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_0_query_Exon_3,4,60,60\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1922276,1922514,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7,1,79,79\n",
      "\n",
      "Getting Gene\n",
      "min = set107_frame0, 0.354\n",
      "5 top scores:\n",
      "[['set107_frame0', 0.354], ['set70_frame1', 1.645], ['set135_frame1', 1.675], ['set96_frame2', 1.695], ['set17_frame0', 1.687]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_5/for_alignment/alignment_Exon_5_translated_genomic_sequence_107_frame0.fa.txt\n",
      "19 50\n",
      "Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5\n",
      "NPISFRCRLQRGTLDFRDEVTYELVQFDGHF\n",
      "19 50\n",
      "set107_frame0\n",
      "NEIRFQCHLQRGTLDFRDETTYELVQFNGHF\n",
      "Original query = NPISFRCRLQRGTLDFRDEVTYELVQFDGHF\n",
      "New query = NEIRFQCHLQRGTLDFRDETTYELVQFNGHF\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,0,0,0,Y,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_6,0,0,19\n",
      "\n",
      "4\n",
      "8\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930011,1930128,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4,2,38,38\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1921657,1921804,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_8,1,48,48\n",
      "\n",
      "Getting Gene\n",
      "min = set126_frame2, 1.199\n",
      "5 top scores:\n",
      "[['set126_frame2', 1.199], ['set118_frame2', 1.508], ['set64_frame2', 1.57], ['set112_frame2', 1.597], ['set114_frame1', 1.639]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_6/for_alignment/alignment_Exon_6_translated_genomic_sequence_126_frame2.fa.txt\n",
      "4 23\n",
      "Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_6\n",
      "KNLESNENGHHSYQDEHES\n",
      "4 23\n",
      "set126_frame2\n",
      "TYMEQNDNSNVSYQQNQDS\n",
      "Original query = KNLESNENGHHSYQDEHES\n",
      "New query = TYMEQNDNSNVSYQQNQDS\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1922276,1922514,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7,1,79,79\n",
      "\n",
      "5\n",
      "9\n",
      "Euphydryas_editha,CAKOGL010000026.1,1925562,1925657,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5,1,31,31\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1916933,1917116,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_0_query_Exon_9,1,62,62\n",
      "\n",
      "Getting Gene\n",
      "min = set46_frame0, 0.146\n",
      "5 top scores:\n",
      "[['set133_frame1', 1.856], ['set163_frame1', 1.859], ['set46_frame0', 0.146], ['set57_frame0', 1.861], ['set41_frame1', 1.88]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_7/for_alignment/alignment_Exon_7_translated_genomic_sequence_46_frame0.fa.txt\n",
      "68 147\n",
      "Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7\n",
      "LLFVCTGRLYMPQLVRDVSLVDTIRSEFTSRHSLEWKFLFLDHRAPPIIGYLPFEVLGTSGYDYYHFDDLEKVVSCHEA\n",
      "68 147\n",
      "set46_frame0\n",
      "LLFVCTGKLCTPQLIRDVSLVDSNRSEFTSRHSLEWKFLFLDHRAPPIIGYLPFEVLGTSGYDYYHFDDLEKVVTCHEA\n",
      "Original query = LLFVCTGRLYMPQLVRDVSLVDTIRSEFTSRHSLEWKFLFLDHRAPPIIGYLPFEVLGTSGYDYYHFDDLEKVVSCHEA\n",
      "New query = LLFVCTGKLCTPQLIRDVSLVDSNRSEFTSRHSLEWKFLFLDHRAPPIIGYLPFEVLGTSGYDYYHFDDLEKVVTCHEA\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1921657,1921804,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_8,1,48,48\n",
      "\n",
      "6\n",
      "5\n",
      "10\n",
      "Euphydryas_editha,CAKOGL010000026.1,1925562,1925657,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5,1,31,31\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1915436,1915624,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_10,2,63,63\n",
      "\n",
      "Getting Gene\n",
      "min = set57_frame0, 0.0\n",
      "5 top scores:\n",
      "[['set57_frame0', 0.0], ['set214_frame2', 1.703], ['set163_frame1', 1.741], ['set83_frame2', 1.744], ['set180_frame1', 1.782]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_8/for_alignment/alignment_Exon_8_translated_genomic_sequence_57_frame0.fa.txt\n",
      "62 110\n",
      "Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_8\n",
      "MQKGELTSCYYRFLTKGQQWIWLQTRFYITYHQWNSKPEFVVCTHRVV\n",
      "62 110\n",
      "set57_frame0\n",
      "MQKGELTSCYYRFLTKGQQWIWLQTRFYITYHQWNSKPEFVVCTHRVV\n",
      "Original query = MQKGELTSCYYRFLTKGQQWIWLQTRFYITYHQWNSKPEFVVCTHRVV\n",
      "New query = MQKGELTSCYYRFLTKGQQWIWLQTRFYITYHQWNSKPEFVVCTHRVV\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1916933,1917116,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_0_query_Exon_9,1,62,62\n",
      "\n",
      "7\n",
      "11\n",
      "Euphydryas_editha,CAKOGL010000026.1,1922276,1922514,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7,1,79,79\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1914585,1914740,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_11,2,54,54\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min = set95_frame2, 0.996\n",
      "5 top scores:\n",
      "[['set95_frame2', 0.996], ['set117_frame2', 1.805], ['set164_frame1', 1.825], ['set143_frame0', 1.773], ['set150_frame0', 1.827]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_9/for_alignment/alignment_Exon_9_translated_genomic_sequence_95_frame2.fa.txt\n",
      "7 69\n",
      "Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_0_query_Exon_9\n",
      "YADIIKSTKQERTETEESVRDCDHNGSSLKDPSTEDAMVPVSPSYMSEASDAFATSYNSMSK\n",
      "7 69\n",
      "set95_frame2\n",
      "YADIDKSTKQENIDA-DTLSETGPNRVDMKENSSEDAMMAMSPSYMSETSDAFGNSYHSISQ\n",
      "Original query = YADIIKSTKQERTETEESVRDCDHNGSSLKDPSTEDAMVPVSPSYMSEASDAFATSYNSMSK\n",
      "New query = YADIDKSTKQENIDA-DTLSETGPNRVDMKENSSEDAMMAMSPSYMSETSDAFGNSYHSISQ\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1915436,1915624,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_10,2,63,63\n",
      "\n",
      "8\n",
      "12\n",
      "Euphydryas_editha,CAKOGL010000026.1,1921657,1921804,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_8,1,48,48\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1913530,1913652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_12,1,41,41\n",
      "\n",
      "Getting Gene\n",
      "min = set105_frame0, 0.749\n",
      "5 top scores:\n",
      "[['set105_frame0', 0.749], ['set77_frame2', 1.76], ['set169_frame2', 1.791], ['set131_frame2', 1.817], ['set68_frame0', 1.82]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_10/for_alignment/alignment_Exon_10_translated_genomic_sequence_105_frame0.fa.txt\n",
      "13 76\n",
      "Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_10\n",
      "ASVKSAATS-GSTSATVATLGTAITTASATWPPRSSYLLYTTGSDTTSVSGGSRSSQRNSSQE\n",
      "13 76\n",
      "set105_frame0\n",
      "ASVKSAATSAGSTCATVATVGTA-TTATGSWHARASHVRYATGSDTASMSGESRSSRRHNSQE\n",
      "Original query = LASVKSAATSGSTSATVATLGTAITTASATWPPRSSYLLYTTGSDTTSVSGGSRSSQRNSSQE\n",
      "New query = ASVKSAATSAGSTCATVATVGTA-TTATGSWHARASHVRYATGSDTASMSGESRSSRRHNSQE\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1914585,1914740,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_11,2,54,54\n",
      "\n",
      "9\n",
      "12\n",
      "Euphydryas_editha,CAKOGL010000026.1,1916933,1917116,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_0_query_Exon_9,1,62,62\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1913530,1913652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_12,1,41,41\n",
      "\n",
      "Getting Gene\n",
      "min = set49_frame0, 0.5\n",
      "5 top scores:\n",
      "[['set49_frame0', 0.5], ['set43_frame2', 1.75], ['set62_frame2', 1.793], ['set35_frame2', 1.824], ['set39_frame2', 1.834]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_11/for_alignment/alignment_Exon_11_translated_genomic_sequence_49_frame0.fa.txt\n",
      "0 56\n",
      "Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_11\n",
      "L--QRLPEPALVPQHGIGAQYLEPAPYVGAVGVPAVLPLSLPPIPVIVAQDQAQLQ\n",
      "0 56\n",
      "set49_frame0\n",
      "LQTQRMPERAVVPQHGIGAQYLEPAPYVAAVGVPGV--LSLHPVPVIVSPDQAQLQ\n",
      "Original query = LQRLPEPALVPQHGIGAQYLEPAPYVGAVGVPAVLPLSLPPIPVIVAQDQAQLQ\n",
      "New query = LQTQRMPERAVVPQHGIGAQYLEPAPYVAAVGVPGV--LSLHPVPVIVSPDQAQLQ\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1913530,1913652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_12,1,41,41\n",
      "\n",
      "10\n",
      "12\n",
      "Euphydryas_editha,CAKOGL010000026.1,1915436,1915624,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_10,2,63,63\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1913530,1913652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_12,1,41,41\n",
      "\n",
      "Getting Gene\n",
      "min = set29_frame1, 0.079\n",
      "5 top scores:\n",
      "[['set29_frame1', 0.079], ['set22_frame2', 1.717], ['set31_frame2', 1.739], ['set24_frame2', 1.786], ['set13_frame1', 1.825]]\n",
      "/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/6.Heliconiinae_Danainae_Nymphalinae/1.Blast_result/Euphydryas_editha/Period_gene_genomic_sequence_individual_exon_Danaus_plexippus/Exon_12/for_alignment/alignment_Exon_12_translated_genomic_sequence_29_frame1.fa.txt\n",
      "4 44\n",
      "set29_frame1\n",
      "EQLQRTHRELQQMIVRQQEELRQVKEQLLLARLGILQPLI\n",
      "Original query = EQLQRTHRELQQMIVRQQEELRQVKEQLLFARLGILQPVIN\n",
      "New query = EQLQRTHRELQQMIVRQQEELRQVKEQLLLARLGILQPLI\n",
      "Scaffold = CAKOGL010000026.1, start = 1934478, end = 1934626\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930478,1930626,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_2,1,49,49\n",
      "\n",
      "1\n",
      "4\n",
      "Euphydryas_editha,CAKOGL010000026.1,1934478,1934626,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930011,1930128,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4,2,38,38\n",
      "\n",
      "Getting Gene\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "KeyboardInterrupt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "family_group = \"6.Heliconiinae_Danainae_Nymphalinae\"\n",
    "blast_output_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "species_list = os.listdir(blast_output_location)\n",
    "\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "species_list = [\"Euphydryas_editha\"]\n",
    "for species in species_list:\n",
    "    annotated_genome_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    query_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Query\"\n",
    "    list_of_query_species = os.listdir(query_location)\n",
    "    if \"desktop.ini\" in list_of_query_species:\n",
    "        list_of_query_species.remove(\"desktop.ini\")\n",
    "#     list_of_query_species = [\"Papilio_xuthus\"]\n",
    "    for query_species in list_of_query_species:\n",
    "        list_of_query_transcripts = os.listdir(f\"{query_location}/{query_species}\")\n",
    "        if \"desktop.ini\" in list_of_query_transcripts:\n",
    "            list_of_query_transcripts.remove(\"desktop.ini\")\n",
    "        for query_transcript in list_of_query_transcripts:\n",
    "\n",
    "\n",
    "            genome_location = \"/mnt/f/Genomes_2023-12-26\"\n",
    "\n",
    "\n",
    "\n",
    "            output_coordinate_file = \"Species,\" + \"Scaffold,\" + \"Start,\" + \"Stop,\" + \"Complement,\" + \"Error,\" + \"Gene,\"+ \"Query_start,\" + \"Query_stop,\"+ \"Query_Length,\" +  \"AG_GT,\" + \"Spliceator_prediction\\n\"\n",
    "            exons_to_check_list = [\"Exon_2\",\"Exon_3\",\"Exon_4\",\"Exon_5\",\"Exon_6\",\"Exon_7\",\"Exon_8\",\"Exon_9\",\"Exon_10\",\"Exon_11\",\"Exon_12\"]\n",
    "#             exons_to_check_list = [\"Exon_9\"]\n",
    "\n",
    "            list_of_files_in_species_folder = os.listdir(f\"{blast_output_location}/{species}\")\n",
    "            coordinate_file_name = ''\n",
    "        #     print(list_of_files_in_species_folder)\n",
    "            for file_names in list_of_files_in_species_folder:\n",
    "\n",
    "                if file_names.endswith(\"_coordinates_old.csv\"):\n",
    "                    coordinate_file_name = file_names\n",
    "            if coordinate_file_name =='':\n",
    "                print(f\"Coordinate file error\")\n",
    "                assert False\n",
    "\n",
    "            with io.open(f\"{blast_output_location}/{species}/{coordinate_file_name}\", 'r') as temp_file_open:\n",
    "                coordinate_file_lines = temp_file_open.readlines()\n",
    "            # print(coordinate_file_lines[0])\n",
    "            for exons_to_check in exons_to_check_list:\n",
    "\n",
    "                for i in range(1,len(coordinate_file_lines)):\n",
    "                    coordinate_file_lines_split = coordinate_file_lines[i].split(\",\")\n",
    "                    current_exon_coordinates = coordinate_file_lines[i].split(\",\")\n",
    "                    if i == 1:\n",
    "                        scaffold_first_exon = coordinate_file_lines[i].split(\",\")[1]\n",
    "                        first_exon_start, first_exon_end = coordinate_file_lines[i].split(\",\")[2],coordinate_file_lines[i].split(\",\")[3]\n",
    "                    # if coordinate_file_lines_split[5] == \"Y\":\n",
    "                    if  coordinate_file_lines_split[6].endswith(exons_to_check):\n",
    "                        if i == 1:\n",
    "                            print(f'First Exon has errors\\n{coordinate_file_lines[i]}')\n",
    "                            continue\n",
    "#                             assert False\n",
    "                        else:\n",
    "                            print(f\"Scaffold = {scaffold_first_exon}, start = {first_exon_start}, end = {first_exon_end}\")\n",
    "                            print(coordinate_file_lines[i])\n",
    "                            # process_current_exon = input(\"Error Found! Process?\")\n",
    "                            process_current_exon = 'y'\n",
    "                            if process_current_exon.lower()[0] == \"y\":\n",
    "                                if i>2:\n",
    "                                    previous_exon_number = i-2\n",
    "                                else:\n",
    "                                    previous_exon_number = i-1\n",
    "                                if i+2 <=int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = i+2\n",
    "                                else:\n",
    "                                    next_exon_number = i+1\n",
    "                                if i >= int(exons_to_check_list[-1].split(\"_\")[1]):\n",
    "                                    next_exon_number = int(exons_to_check_list[-1].split(\"_\")[1])\n",
    "#                                     continue\n",
    "\n",
    "                                '''\n",
    "                                the above code ensures that missing exons do not cause issues\n",
    "                                '''\n",
    "\n",
    "                                previous_exon_coordinates = ''\n",
    "                                next_exon_coordinates = ''\n",
    "                                while True:\n",
    "                                    print(previous_exon_number)\n",
    "                                    if coordinate_file_lines[previous_exon_number].split(\",\")[5] == \"N\":\n",
    "                                        previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                        break\n",
    "                                    # else:\n",
    "                                    #     proceed_prompt = input(f\"Previous exon coordinate\\n{coordinate_file_lines[previous_exon_number]}\\nProceed?\")\n",
    "                                    #     if proceed_prompt.lower()[0] == \"y\":\n",
    "                                    #         previous_exon_coordinates = coordinate_file_lines[previous_exon_number]\n",
    "                                    #         break\n",
    "                                    previous_exon_number -= 1\n",
    "                                while True:\n",
    "                                    print(next_exon_number)\n",
    "                                    try:\n",
    "                                        if coordinate_file_lines[next_exon_number].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number]\n",
    "                                            break\n",
    "                                    except:\n",
    "                                        if coordinate_file_lines[next_exon_number-1].split(\",\")[5] == \"N\":\n",
    "                                            next_exon_coordinates = coordinate_file_lines[next_exon_number-1]\n",
    "                                            break\n",
    "                                   \n",
    "                                    next_exon_number += 1\n",
    "\n",
    "\n",
    "                                print(previous_exon_coordinates)\n",
    "                                print(next_exon_coordinates)\n",
    "                                annotated_species_name, error_exon, species, genome_file, gene_sequence,left_overhang,right_overhang,scaffold,original_query_name, original_query = mafft_process(previous_exon_coordinates,\n",
    "                                                                                                                                  next_exon_coordinates,\n",
    "                                                                                                                                  current_exon_coordinates,\n",
    "                                                                                                                                  query_species,\n",
    "                                                                                                                                  query_transcript,\n",
    "                                                                                                                                  query_location,\n",
    "                                                                                                                                  annotated_genome_location,\n",
    "                                                                                                                                     genome_location)\n",
    "\n",
    "                                mafft_run_folder = run_mafft(annotated_genome_location,annotated_species_name,error_exon)\n",
    "\n",
    "                                possible_gene_sequence, alignment_name = process_mafft_output(mafft_run_folder, error_exon)\n",
    "\n",
    "                                print(f\"Original query = {original_query}\")\n",
    "                                print(f\"New query = {possible_gene_sequence}\")\n",
    "                                \n",
    "                                list_of_folders_inside_exon_folder = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}\")\n",
    "                                if \"for_blast\" not in list_of_folders_inside_exon_folder:\n",
    "                                    os.mkdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                else:\n",
    "                                    list_of_files = os.listdir(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast\")\n",
    "                                    for file in list_of_files:\n",
    "                                        os.remove(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/{file}\")\n",
    "\n",
    "\n",
    "                                with open(f\"{annotated_genome_location}/{annotated_species_name}/Period_gene_genomic_sequence_individual_exon_{query_species}/{error_exon}/for_blast/new_query.txt\",'w') as query_file:\n",
    "                                    query = f\">Query_{error_exon}_{alignment_name}\\n{possible_gene_sequence}\"\n",
    "                                    query_file.write(query)\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccdf583e-4fa4-4fac-a4d4-98a069f1d391",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307c7dce-21b5-432f-926a-024803e6dad6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
