{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "12f98d0a-daf2-4ad6-aef0-7eb1394a8a41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b50aa19f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Euphydryas_editha\n",
      "Exon_1 Match\n",
      "[['Danaus_plexippus', ['1934478', '1934626']], ['Nymphalis_io', ['1934478', '1934626']], ['Vanessa_tameamea', ['1934478', '1934626']]]\n",
      "Exon_2 Match\n",
      "[['Danaus_plexippus', ['1930478', '1930626']], ['Nymphalis_io', ['1930478', '1930626']], ['Vanessa_tameamea', ['1930478', '1930626']]]\n",
      "Exon_3 Match\n",
      "[['Danaus_plexippus', ['1930208', '1930389']], ['Nymphalis_io', ['1930208', '1930389']], ['Vanessa_tameamea', ['1930208', '1930389']]]\n",
      "Exon_4 Match\n",
      "[['Danaus_plexippus', ['1930011', '1930128']], ['Nymphalis_io', ['1930011', '1930128']], ['Vanessa_tameamea', ['1930011', '1930128']]]\n",
      "Exon_5 Match\n",
      "[['Danaus_plexippus', ['1925562', '1925657']], ['Nymphalis_io', ['1925562', '1925657']], ['Vanessa_tameamea', ['1925562', '1925657']]]\n",
      "Exon_6 Match\n",
      "[['Danaus_plexippus', ['1923804', '1923864']], ['Nymphalis_io', ['1923804', '1923864']], ['Vanessa_tameamea', ['1923804', '1923864']]]\n",
      "Exon_7 Match\n",
      "[['Danaus_plexippus', ['1922276', '1922514']], ['Nymphalis_io', ['1922276', '1922514']], ['Vanessa_tameamea', ['1922276', '1922514']]]\n",
      "Exon_8 Match\n",
      "[['Danaus_plexippus', ['1921657', '1921804']], ['Nymphalis_io', ['1921657', '1921804']], ['Vanessa_tameamea', ['1921657', '1921804']]]\n",
      "Exon_9 Match\n",
      "[['Danaus_plexippus', ['1916933', '1917116']], ['Nymphalis_io', ['1916933', '1917116']], ['Vanessa_tameamea', ['1916933', '1917116']]]\n",
      "Exon_10 Match\n",
      "[['Danaus_plexippus', ['1915436', '1915624']], ['Nymphalis_io', ['1915436', '1915624']], ['Vanessa_tameamea', ['1915436', '1915624']]]\n",
      "Exon_11 Match\n",
      "[['Danaus_plexippus', ['1914585', '1914734']], ['Nymphalis_io', ['1914585', '1914734']], ['Vanessa_tameamea', ['1914585', '1914734']]]\n",
      "Exon_12 Match\n",
      "[['Danaus_plexippus', ['1913530', '1913652']], ['Nymphalis_io', ['1913530', '1913652']], ['Vanessa_tameamea', ['1913530', '1913652']]]\n",
      "Exon_13 Match\n",
      "[['Danaus_plexippus', ['1911530', '1911652']], ['Nymphalis_io', ['1911530', '1911652']], ['Vanessa_tameamea', ['1911530', '1911652']]]\n",
      "Species,Scaffold,Start,Stop,Complement,Error,Gene,Query_start,Query_stop,Query_Length\n",
      "Euphydryas_editha,CAKOGL010000026.1,1934478,1934626,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_2_query_Exon_1,0,0,0\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930478,1930626,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_2,1,49,49,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930208,1930389,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_0_query_Exon_3,1,60,60,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1930011,1930128,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_1_query_Exon_4,1,39,39,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1925562,1925657,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_1_query_Exon_5,1,31,31,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1923804,1923864,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_6,1,19,19,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1922276,1922514,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_1_query_Exon_7,1,79,79,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1921657,1921804,1,N,Danaus_plexippus_XM_032670980.2_Frame_2_rightoh_2_query_Exon_8,1,48,48,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1916933,1917116,1,N,Danaus_plexippus_XM_032670980.2_Frame_1_rightoh_0_query_Exon_9,1,61,61,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1915436,1915624,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_10,1,63,63,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1914585,1914734,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_11,1,50,50,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1913530,1913652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_12,1,41,41,Y,N\n",
      "Euphydryas_editha,CAKOGL010000026.1,1911530,1911652,1,N,Danaus_plexippus_XM_032670980.2_Frame_0_rightoh_0_query_Exon_13,0,0,0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "family_group = \"6.Heliconiinae_Danainae_Nymphalinae\"\n",
    "annotated_genome_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "species_list = os.listdir(annotated_genome_location)\n",
    "if \"desktop.ini\" in species_list:\n",
    "    species_list.remove(\"desktop.ini\")\n",
    "# species_list = [\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Limenitis_camilla\"]\n",
    "# species_list = [\"Argynnis_bischoffii_washingtonia\",\"Boloria_euphrosyne\",\"Boloria_selene\",\"Brenthis_daphne\",\"Brenthis_hecate\",\"Brenthis_ino\",\"Dryadula_phaetusa\",\"Dryas_iulia_moderata\",\"Eueides_isabella\",\"Fabriciana_adippe\",\"Heliconius_charithonia\",\"Heliconius_nattereri\",\"Heliconius_sara\",\"Philaethria_dido\",\"Hestina_assimilis\"]\n",
    "# species_list = [\"Bicyclus_anynana\",\"Coenonympha_glycerion\",\"Elymnias_hypermnestra\",\"Erebia_aethiops\",\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "# species_list = [\"Erebia_ligea\",\"Hipparchia_semele\",\"Lasiommata_megera\",\"Maniola_hyperantus\",\"Maniola_jurtina\",\"Melanargia_galathea\",\"Oeneis_ivallda\",\"Pararge_aegeria\"]\n",
    "species_list = [\"Euphydryas_editha\"]\n",
    "for species in species_list:\n",
    "    print(species)\n",
    "    blast_location = f\"/mnt/h/My Drive/Circadian Rhythm Genes Project/9.CLOCK Exon Analysis/{family_group}/1.Blast_result\"\n",
    "    \n",
    "    list_of_files = os.listdir(f\"{blast_location}/{species}\")\n",
    "    query_species_list = []\n",
    "    coordinate_detail_dictionary = {}\n",
    "    coordinate_info_dictionary = {}\n",
    "    scaffold = ''\n",
    "    for file in list_of_files:\n",
    "        if file.endswith(\".csv\") and len(file.split(\"_\")) >= 5 and not(file.endswith(\"_old.csv\")) and not(file.endswith(\"_old_backup.csv\")):\n",
    "#             print(file)\n",
    "            if \".\" in file[:-4]:\n",
    "                query_species = f\"{file.split('_')[-2].split('.')[1]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            else:\n",
    "                query_species = f\"{file.split('_')[-2]}_{file.split('_')[-1].split('.')[0]}\"\n",
    "            query_species_list.append(query_species)\n",
    "            coordinate_detail_dictionary[query_species] = {}\n",
    "            with open(f\"{blast_location}/{species}/{file}\", 'r') as coordinate_file_open:\n",
    "                coordinate_file_lines_list  =  coordinate_file_open.readlines()\n",
    "            if scaffold == '':\n",
    "                scaffold = coordinate_file_lines_list[1].split(\",\")[1]\n",
    "    \n",
    "            for i in range(1,len(coordinate_file_lines_list)):\n",
    "                \n",
    "                coordinate_file_lines_split = coordinate_file_lines_list[i].split(\",\")\n",
    "                start,stop = coordinate_file_lines_split[2],coordinate_file_lines_split[3]\n",
    "                exon = f\"Exon_{i}\"\n",
    "                if exon not in coordinate_info_dictionary:\n",
    "                    coordinate_info_dictionary[exon] = {}\n",
    "                coordinate_info_dictionary[exon][query_species] = [start,stop]\n",
    "                coordinate_detail_dictionary[query_species][exon] = coordinate_file_lines_list[i]\n",
    "            \n",
    "            # break\n",
    "            # print(query_species)\n",
    "            # print(file)\n",
    "#     print(coordinate_info_dictionary)\n",
    "#     print(coordinate_detail_dictionary)\n",
    "    # assert False\n",
    "    output = coordinate_file_lines_list[0]\n",
    "    for exon,species_specific_coordinates in coordinate_info_dictionary.items():\n",
    "        # print(exon,species_specific_coordinates)\n",
    "        coordinates_list = []\n",
    "        previous_coordinates = \"\"\n",
    "        match = 1\n",
    "        for query_species, coordinates in coordinate_info_dictionary[exon].items():\n",
    "            coordinates_list.append([query_species,coordinates])\n",
    "            if previous_coordinates != '':\n",
    "                if coordinates != previous_coordinates:\n",
    "                    match = 0\n",
    "                    \n",
    "            previous_coordinates = coordinates\n",
    "       \n",
    "        \n",
    "        \n",
    "        if match == 0:\n",
    "            print(exon,\"mismatch\")\n",
    "            print(coordinates_list)\n",
    "                \n",
    "            while True:\n",
    "                choice = input(\"Proceed?\")\n",
    "                if choice[0].lower() == \"y\" or choice[0].lower() == \"n\":\n",
    "                    break\n",
    "            \n",
    "            while True:\n",
    "                if choice[0].lower() == \"y\":\n",
    "                    alignment_detail_dictionary = {}\n",
    "                    for query_species_name in query_species_list:\n",
    "                        list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                        for folders in list_of_folders:\n",
    "                            if folders.endswith(query_species_name):\n",
    "                                # print()\n",
    "                                \n",
    "                                with open(f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query.txt\", 'r') as query_file_open:\n",
    "                                    # alignment_detail = query_file_open.readlines()[0]\n",
    "                                    alignment_detail = query_file_open.readlines()[0].split(\"set\")[1][:-1]\n",
    "                                    alignment_detail_dictionary[query_species_name] = alignment_detail\n",
    "                                with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/{exon}_translated_genomic_sequence_{alignment_detail}.fa.hat2\", 'r') as alignment_score_file:\n",
    "                                    distance_score = float(alignment_score_file.readlines()[-1].rstrip())\n",
    "                                try:\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_blast/new_query_spliced.txt\", 'r') as new_query_file:\n",
    "                                        new_query = (new_query_file.readlines()[-1].rstrip())\n",
    "                                except:\n",
    "                                    new_query = \"Issue with query\"\n",
    "                                print(query_species_name,exon, alignment_detail, distance_score,f\"\\n{new_query}\")\n",
    "                    while True:            \n",
    "                        check_alignment = input(\"Check Alignment??\")\n",
    "                        if check_alignment[0].lower() == \"y\" or check_alignment[0].lower() == \"n\":\n",
    "                            break\n",
    "                    if check_alignment[0].lower() == \"y\":\n",
    "                        for query_species_name in query_species_list:\n",
    "                            list_of_folders  = os.listdir(f\"{blast_location}/{species}\")\n",
    "                            for folders in list_of_folders:\n",
    "                                if folders.endswith(query_species_name):\n",
    "                                    with open (f\"{blast_location}/{species}/{folders}/{exon}/for_alignment/alignment_clustal_{exon}_translated_genomic_sequence_{alignment_detail_dictionary[query_species_name]}.fa.txt\", 'r') as clustal_alignment_file:\n",
    "                                        clustal_alignment = clustal_alignment_file.readlines()\n",
    "                                    print(\"\".join(clustal_alignment))\n",
    "                                    \n",
    "                                # assert False\n",
    "                    while True:\n",
    "                        try:\n",
    "                            species_number = (input(f\"Choose Species 1 - {len(query_species_list)}\"))\n",
    "                            \n",
    "                            \n",
    "                            if species_number[0].lower() == 'n':\n",
    "                                output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                                break \n",
    "                            output += coordinate_detail_dictionary[query_species_list[int(species_number)-1]][exon]\n",
    "                            break\n",
    "                        except:\n",
    "                            print(\"Retry\")\n",
    "                    break\n",
    "                elif choice[0].lower() == \"n\":\n",
    "                    output +=  f'{species},{scaffold},000,000,0,Y,Error_{exon},00,00,00\\n'\n",
    "                    break\n",
    "                \n",
    "                \n",
    "        else:\n",
    "            print(exon, \"Match\")\n",
    "            print(coordinates_list)\n",
    "            output += coordinate_detail_dictionary[query_species_list[0]][exon]\n",
    "        \n",
    "       \n",
    "\n",
    "        # assert False\n",
    "    # print(coordinate_detail_dictionary)\n",
    "    print(output)\n",
    "    with open(f\"{blast_location}/{species}/final_coordinates.csv\", \"w\") as out_file:\n",
    "        out_file.write(output)\n",
    "    \n",
    "    # 'Coenonympha_glycerion,OY979626.1,7015875,7015963,0,N,5.Bicyclus_anynana_XM_024088150.2_Frame_0_rightoh_2_query_Exon_1,1,29,29\\n'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7892a6a6-9400-414d-bbaa-cf893036e0d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
